{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ys7t4p5EALTZ",
        "outputId": "f8132a7e-c5ea-442f-afbc-9c961179dce4"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: llama_index in /usr/local/lib/python3.11/dist-packages (0.12.33)\n",
            "Requirement already satisfied: llama-index-agent-openai<0.5.0,>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.4.6)\n",
            "Requirement already satisfied: llama-index-cli<0.5.0,>=0.4.1 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.4.1)\n",
            "Requirement already satisfied: llama-index-core<0.13.0,>=0.12.33 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.12.33.post1)\n",
            "Requirement already satisfied: llama-index-embeddings-openai<0.4.0,>=0.3.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.3.1)\n",
            "Requirement already satisfied: llama-index-indices-managed-llama-cloud>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.6.11)\n",
            "Requirement already satisfied: llama-index-llms-openai<0.4.0,>=0.3.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.3.38)\n",
            "Requirement already satisfied: llama-index-multi-modal-llms-openai<0.5.0,>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.4.3)\n",
            "Requirement already satisfied: llama-index-program-openai<0.4.0,>=0.3.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.3.1)\n",
            "Requirement already satisfied: llama-index-question-gen-openai<0.4.0,>=0.3.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.3.0)\n",
            "Requirement already satisfied: llama-index-readers-file<0.5.0,>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.4.7)\n",
            "Requirement already satisfied: llama-index-readers-llama-parse>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from llama_index) (0.4.0)\n",
            "Requirement already satisfied: nltk>3.8.1 in /usr/local/lib/python3.11/dist-packages (from llama_index) (3.9.1)\n",
            "Requirement already satisfied: openai>=1.14.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-agent-openai<0.5.0,>=0.4.0->llama_index) (1.76.0)\n",
            "Requirement already satisfied: PyYAML>=6.0.1 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (6.0.2)\n",
            "Requirement already satisfied: SQLAlchemy>=1.4.49 in /usr/local/lib/python3.11/dist-packages (from SQLAlchemy[asyncio]>=1.4.49->llama-index-core<0.13.0,>=0.12.33->llama_index) (2.0.40)\n",
            "Requirement already satisfied: aiohttp<4.0.0,>=3.8.6 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (3.11.15)\n",
            "Requirement already satisfied: banks<3.0.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (2.1.2)\n",
            "Requirement already satisfied: dataclasses-json in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (0.6.7)\n",
            "Requirement already satisfied: deprecated>=1.2.9.3 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (1.2.18)\n",
            "Requirement already satisfied: dirtyjson<2.0.0,>=1.0.8 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (1.0.8)\n",
            "Requirement already satisfied: filetype<2.0.0,>=1.2.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (1.2.0)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (2025.3.2)\n",
            "Requirement already satisfied: httpx in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (0.28.1)\n",
            "Requirement already satisfied: nest-asyncio<2.0.0,>=1.5.8 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (1.6.0)\n",
            "Requirement already satisfied: networkx>=3.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (3.4.2)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (2.0.2)\n",
            "Requirement already satisfied: pillow>=9.0.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (11.2.1)\n",
            "Requirement already satisfied: pydantic>=2.8.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (2.11.3)\n",
            "Requirement already satisfied: requests>=2.31.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (2.32.3)\n",
            "Requirement already satisfied: tenacity!=8.4.0,<10.0.0,>=8.2.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (9.1.2)\n",
            "Requirement already satisfied: tiktoken>=0.7.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (0.9.0)\n",
            "Requirement already satisfied: tqdm<5.0.0,>=4.66.1 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (4.67.1)\n",
            "Requirement already satisfied: typing-extensions>=4.5.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (4.13.2)\n",
            "Requirement already satisfied: typing-inspect>=0.8.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (0.9.0)\n",
            "Requirement already satisfied: wrapt in /usr/local/lib/python3.11/dist-packages (from llama-index-core<0.13.0,>=0.12.33->llama_index) (1.17.2)\n",
            "Requirement already satisfied: llama-cloud<0.2.0,>=0.1.13 in /usr/local/lib/python3.11/dist-packages (from llama-index-indices-managed-llama-cloud>=0.4.0->llama_index) (0.1.19)\n",
            "Requirement already satisfied: beautifulsoup4<5.0.0,>=4.12.3 in /usr/local/lib/python3.11/dist-packages (from llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (4.13.4)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (2.2.2)\n",
            "Requirement already satisfied: pypdf<6.0.0,>=5.1.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (5.4.0)\n",
            "Requirement already satisfied: striprtf<0.0.27,>=0.0.26 in /usr/local/lib/python3.11/dist-packages (from llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (0.0.26)\n",
            "Requirement already satisfied: llama-parse>=0.5.0 in /usr/local/lib/python3.11/dist-packages (from llama-index-readers-llama-parse>=0.4.0->llama_index) (0.6.16)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.11/dist-packages (from nltk>3.8.1->llama_index) (8.1.8)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from nltk>3.8.1->llama_index) (1.4.2)\n",
            "Requirement already satisfied: regex>=2021.8.3 in /usr/local/lib/python3.11/dist-packages (from nltk>3.8.1->llama_index) (2024.11.6)\n",
            "Requirement already satisfied: aiohappyeyeballs>=2.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (2.6.1)\n",
            "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (1.3.2)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (25.3.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (1.6.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (6.4.3)\n",
            "Requirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (0.3.1)\n",
            "Requirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.11/dist-packages (from aiohttp<4.0.0,>=3.8.6->llama-index-core<0.13.0,>=0.12.33->llama_index) (1.20.0)\n",
            "Requirement already satisfied: griffe in /usr/local/lib/python3.11/dist-packages (from banks<3.0.0,>=2.0.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (1.7.3)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from banks<3.0.0,>=2.0.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (3.1.6)\n",
            "Requirement already satisfied: platformdirs in /usr/local/lib/python3.11/dist-packages (from banks<3.0.0,>=2.0.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (4.3.7)\n",
            "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.11/dist-packages (from beautifulsoup4<5.0.0,>=4.12.3->llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (2.7)\n",
            "Requirement already satisfied: certifi>=2024.7.4 in /usr/local/lib/python3.11/dist-packages (from llama-cloud<0.2.0,>=0.1.13->llama-index-indices-managed-llama-cloud>=0.4.0->llama_index) (2025.1.31)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.11/dist-packages (from httpx->llama-index-core<0.13.0,>=0.12.33->llama_index) (4.9.0)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.11/dist-packages (from httpx->llama-index-core<0.13.0,>=0.12.33->llama_index) (1.0.9)\n",
            "Requirement already satisfied: idna in /usr/local/lib/python3.11/dist-packages (from httpx->llama-index-core<0.13.0,>=0.12.33->llama_index) (3.10)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.11/dist-packages (from httpcore==1.*->httpx->llama-index-core<0.13.0,>=0.12.33->llama_index) (0.16.0)\n",
            "Requirement already satisfied: llama-cloud-services>=0.6.16 in /usr/local/lib/python3.11/dist-packages (from llama-parse>=0.5.0->llama-index-readers-llama-parse>=0.4.0->llama_index) (0.6.17)\n",
            "Requirement already satisfied: distro<2,>=1.7.0 in /usr/local/lib/python3.11/dist-packages (from openai>=1.14.0->llama-index-agent-openai<0.5.0,>=0.4.0->llama_index) (1.9.0)\n",
            "Requirement already satisfied: jiter<1,>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from openai>=1.14.0->llama-index-agent-openai<0.5.0,>=0.4.0->llama_index) (0.9.0)\n",
            "Requirement already satisfied: sniffio in /usr/local/lib/python3.11/dist-packages (from openai>=1.14.0->llama-index-agent-openai<0.5.0,>=0.4.0->llama_index) (1.3.1)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.8.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.33.1 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.8.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (2.33.1)\n",
            "Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from pydantic>=2.8.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (0.4.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests>=2.31.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (3.4.1)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests>=2.31.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (2.4.0)\n",
            "Requirement already satisfied: greenlet>=1 in /usr/local/lib/python3.11/dist-packages (from SQLAlchemy>=1.4.49->SQLAlchemy[asyncio]>=1.4.49->llama-index-core<0.13.0,>=0.12.33->llama_index) (3.2.1)\n",
            "Requirement already satisfied: mypy-extensions>=0.3.0 in /usr/local/lib/python3.11/dist-packages (from typing-inspect>=0.8.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (1.1.0)\n",
            "Requirement already satisfied: marshmallow<4.0.0,>=3.18.0 in /usr/local/lib/python3.11/dist-packages (from dataclasses-json->llama-index-core<0.13.0,>=0.12.33->llama_index) (3.26.1)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.11/dist-packages (from pandas->llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (2025.2)\n",
            "Requirement already satisfied: python-dotenv<2.0.0,>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from llama-cloud-services>=0.6.16->llama-parse>=0.5.0->llama-index-readers-llama-parse>=0.4.0->llama_index) (1.1.0)\n",
            "Requirement already satisfied: packaging>=17.0 in /usr/local/lib/python3.11/dist-packages (from marshmallow<4.0.0,>=3.18.0->dataclasses-json->llama-index-core<0.13.0,>=0.12.33->llama_index) (24.2)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.8.2->pandas->llama-index-readers-file<0.5.0,>=0.4.0->llama_index) (1.17.0)\n",
            "Requirement already satisfied: colorama>=0.4 in /usr/local/lib/python3.11/dist-packages (from griffe->banks<3.0.0,>=2.0.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (0.4.6)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from jinja2->banks<3.0.0,>=2.0.0->llama-index-core<0.13.0,>=0.12.33->llama_index) (3.0.2)\n",
            "Requirement already satisfied: openai in /usr/local/lib/python3.11/dist-packages (1.76.0)\n",
            "Requirement already satisfied: anyio<5,>=3.5.0 in /usr/local/lib/python3.11/dist-packages (from openai) (4.9.0)\n",
            "Requirement already satisfied: distro<2,>=1.7.0 in /usr/local/lib/python3.11/dist-packages (from openai) (1.9.0)\n",
            "Requirement already satisfied: httpx<1,>=0.23.0 in /usr/local/lib/python3.11/dist-packages (from openai) (0.28.1)\n",
            "Requirement already satisfied: jiter<1,>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from openai) (0.9.0)\n",
            "Requirement already satisfied: pydantic<3,>=1.9.0 in /usr/local/lib/python3.11/dist-packages (from openai) (2.11.3)\n",
            "Requirement already satisfied: sniffio in /usr/local/lib/python3.11/dist-packages (from openai) (1.3.1)\n",
            "Requirement already satisfied: tqdm>4 in /usr/local/lib/python3.11/dist-packages (from openai) (4.67.1)\n",
            "Requirement already satisfied: typing-extensions<5,>=4.11 in /usr/local/lib/python3.11/dist-packages (from openai) (4.13.2)\n",
            "Requirement already satisfied: idna>=2.8 in /usr/local/lib/python3.11/dist-packages (from anyio<5,>=3.5.0->openai) (3.10)\n",
            "Requirement already satisfied: certifi in /usr/local/lib/python3.11/dist-packages (from httpx<1,>=0.23.0->openai) (2025.1.31)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.11/dist-packages (from httpx<1,>=0.23.0->openai) (1.0.9)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.11/dist-packages (from httpcore==1.*->httpx<1,>=0.23.0->openai) (0.16.0)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from pydantic<3,>=1.9.0->openai) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.33.1 in /usr/local/lib/python3.11/dist-packages (from pydantic<3,>=1.9.0->openai) (2.33.1)\n",
            "Requirement already satisfied: typing-inspection>=0.4.0 in /usr/local/lib/python3.11/dist-packages (from pydantic<3,>=1.9.0->openai) (0.4.0)\n",
            "Requirement already satisfied: PyPDF2 in /usr/local/lib/python3.11/dist-packages (3.0.1)\n",
            "Requirement already satisfied: fitz in /usr/local/lib/python3.11/dist-packages (0.0.1.dev2)\n",
            "Requirement already satisfied: configobj in /usr/local/lib/python3.11/dist-packages (from fitz) (5.0.9)\n",
            "Requirement already satisfied: configparser in /usr/local/lib/python3.11/dist-packages (from fitz) (7.2.0)\n",
            "Requirement already satisfied: httplib2 in /usr/local/lib/python3.11/dist-packages (from fitz) (0.22.0)\n",
            "Requirement already satisfied: nibabel in /usr/local/lib/python3.11/dist-packages (from fitz) (5.3.2)\n",
            "Requirement already satisfied: nipype in /usr/local/lib/python3.11/dist-packages (from fitz) (1.10.0)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from fitz) (2.0.2)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.11/dist-packages (from fitz) (2.2.2)\n",
            "Requirement already satisfied: pyxnat in /usr/local/lib/python3.11/dist-packages (from fitz) (1.6.3)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.11/dist-packages (from fitz) (1.15.2)\n",
            "Requirement already satisfied: pyparsing!=3.0.0,!=3.0.1,!=3.0.2,!=3.0.3,<4,>=2.4.2 in /usr/local/lib/python3.11/dist-packages (from httplib2->fitz) (3.2.3)\n",
            "Requirement already satisfied: importlib-resources>=5.12 in /usr/local/lib/python3.11/dist-packages (from nibabel->fitz) (6.5.2)\n",
            "Requirement already satisfied: packaging>=20 in /usr/local/lib/python3.11/dist-packages (from nibabel->fitz) (24.2)\n",
            "Requirement already satisfied: typing-extensions>=4.6 in /usr/local/lib/python3.11/dist-packages (from nibabel->fitz) (4.13.2)\n",
            "Requirement already satisfied: click>=6.6.0 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (8.1.8)\n",
            "Requirement already satisfied: networkx>=2.5 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (3.4.2)\n",
            "Requirement already satisfied: prov>=1.5.2 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (2.0.1)\n",
            "Requirement already satisfied: pydot>=1.2.3 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (3.0.4)\n",
            "Requirement already satisfied: python-dateutil>=2.2 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (2.9.0.post0)\n",
            "Requirement already satisfied: rdflib>=5.0.0 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (6.3.2)\n",
            "Requirement already satisfied: simplejson>=3.8.0 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (3.20.1)\n",
            "Requirement already satisfied: traits>=6.2 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (7.0.2)\n",
            "Requirement already satisfied: filelock>=3.0.0 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (3.18.0)\n",
            "Requirement already satisfied: acres in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (0.3.0)\n",
            "Requirement already satisfied: etelemetry>=0.3.1 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (0.3.1)\n",
            "Requirement already satisfied: looseversion!=1.2 in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (1.3.0)\n",
            "Requirement already satisfied: puremagic in /usr/local/lib/python3.11/dist-packages (from nipype->fitz) (1.28)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.11/dist-packages (from pandas->fitz) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.11/dist-packages (from pandas->fitz) (2025.2)\n",
            "Requirement already satisfied: lxml>=4.3 in /usr/local/lib/python3.11/dist-packages (from pyxnat->fitz) (5.4.0)\n",
            "Requirement already satisfied: requests>=2.20 in /usr/local/lib/python3.11/dist-packages (from pyxnat->fitz) (2.32.3)\n",
            "Requirement already satisfied: pathlib>=1.0 in /usr/local/lib/python3.11/dist-packages (from pyxnat->fitz) (1.0.1)\n",
            "Requirement already satisfied: ci-info>=0.2 in /usr/local/lib/python3.11/dist-packages (from etelemetry>=0.3.1->nipype->fitz) (0.3.0)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.2->nipype->fitz) (1.17.0)\n",
            "Requirement already satisfied: isodate<0.7.0,>=0.6.0 in /usr/local/lib/python3.11/dist-packages (from rdflib>=5.0.0->nipype->fitz) (0.6.1)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests>=2.20->pyxnat->fitz) (3.4.1)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests>=2.20->pyxnat->fitz) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests>=2.20->pyxnat->fitz) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests>=2.20->pyxnat->fitz) (2025.1.31)\n",
            "Requirement already satisfied: PyMuPDF in /usr/local/lib/python3.11/dist-packages (1.25.5)\n",
            "Requirement already satisfied: pymupdf in /usr/local/lib/python3.11/dist-packages (1.25.5)\n",
            "Requirement already satisfied: fuzzywuzzy in /usr/local/lib/python3.11/dist-packages (0.18.0)\n"
          ]
        }
      ],
      "source": [
        "!pip install llama_index\n",
        "!pip install openai\n",
        "!pip install PyPDF2\n",
        "!pip install fitz\n",
        "!pip install PyMuPDF\n",
        "!pip install --upgrade pymupdf\n",
        "!pip install fuzzywuzzy\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "hxbh5pySCLJ4"
      },
      "outputs": [],
      "source": [
        "import os\n",
        "os.environ['OPENAI_API_KEY']=\"\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "w4LACOVgLsBV"
      },
      "outputs": [],
      "source": [
        "from llama_index.core import VectorStoreIndex, SimpleDirectoryReader\n",
        "documents = SimpleDirectoryReader('/content/data').load_data()\n",
        "index = VectorStoreIndex.from_documents(documents)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0y_5NElyMCbI",
        "outputId": "9beb73de-49f0-4d78-b81d-07270bc00969"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "[Document(id_='46323171-d9fe-4cf9-9ce1-4580db9c6b72', embedding=None, metadata={'page_label': '1', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='iScience\\nArticle\\nMultiplex imaging of localized prostate tumors\\nreveals altered spatial organization of AR-positive\\ncells in the microenvironment\\nC¸i /C21gdem Ak,\\nZeynep Sayar,\\nGuillaume\\nThibault, ..., Young\\nHwan Chang,\\nVasilis Stavrinides,\\nSebnem Ece Eksi\\neksi@ohsu.edu\\nHighlights\\nWe provide a detailed\\nspatial map of single cells\\nvia cyclic imaging\\nDistinct mast cell\\npopulations are spatially\\nassociated with M2\\nmacrophages and Tregs\\nGrade-speciﬁc cellular\\nneighborhoods are\\nidentiﬁed based on stromal\\nAR expression\\nStromal AR expression is\\ncorrelated with NF-kB1\\nand HIF1a\\nAk et al., iScience27, 110668\\nSeptember 20, 2024ª 2024\\nThe Authors. Published by\\nElsevier Inc.\\nhttps://doi.org/10.1016/\\nj.isci.2024.110668\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='dbc6d06b-83f3-4d63-a486-953d6ccfb16f', embedding=None, metadata={'page_label': '2', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='iScience\\nArticle\\nMultiplex imaging of localized prostate\\ntumors reveals altered spatial organization\\nof AR-positive cells in the microenvironment\\nC¸i /C21gdem Ak, 1,2,11 Zeynep Sayar, 1,2,8,11 Guillaume Thibault, 2,9 Erik A. Burlingame, 2,10 M.J. Kuykendall, 1\\nJennifer Eng, 2 Alex Chitsazan, 1 Koei Chin, 1 Andrew C. Adey, 1,3 Christopher Boniface, 1 Paul T. Spellman, 1,3\\nGeorge V. Thomas, 1,4 Ryan P. Kopp, 1,5 Emek Demir, 1,7 Young Hwan Chang, 2 Vasilis Stavrinides, 6\\nand Sebnem Ece Eksi 1,2,12,*\\nSUMMARY\\nMapping the spatial interactions of cancer, immune, and stromal cell states presents novel opportunities\\nfor patient stratiﬁcation and for advancing immunotherapy. While single-cell studies revealed signiﬁcant\\nmolecular heterogeneity in prostate cancer cells, the impact of spatial stromal cell heterogeneity remains\\npoorly understood. Here, we used cyclic immunoﬂuorescent imaging on whole-tissue sections to uncover\\nnovel spatial associations between cancer and stromal cells in low- and high-grade prostate tumors\\nand tumor-adjacent normal tissues. Our results provide a spatial map of single cells and recurrent\\ncellular neighborhoods in the prostate tumor microenvironment of treatment-naive patients. We\\nreport unique populations of mast cells that show distinct spatial associations with M2 macrophages\\nand regulatory T cells. Our results show disease-speciﬁc neighborhoods that are primarily driven by\\nandrogen receptor-positive (AR+) stromal cells and identify inﬂammatory gene networks active in AR+\\nprostate stroma.\\nINTRODUCTION\\nProstate tumors are composed of epithelial cells that exhibit molecular and cellular differences and interact with a complex ecosystem of\\nstromal cells: ﬁbroblasts, immune cells, mesenchymal stem cells, smooth muscle, blood vessels, and innervating neurons. 1 The complexity\\nand heterogeneity of the disease are indicated by pathological and single-cell omic evidence. 2–4 It is essential to simultaneously dissect\\nthis cellular heterogeneity and generate a spatial map between cell types and states to better stratify patients into treatment groups and\\nto identify more effective immunotherapy strategies. Not surprisingly, these cell states, cell types, and bidirectional cellular interactions\\nmay not be solely recovered from hematoxylin and eosin (H&E) staining, even with advanced machine learning algorithms. Single-cell\\nomic approaches can identify cell types and states, but not the spatially coordinated behavior of heterogeneous cells. Immunohistochemistry\\nis gaining widespread adoption in clinical practice to mitigate this problem, but it is often deployed for a few markers in a non-systematic,\\nnon-standardized manner.\\nRecent advancements in multiplex proteomic proﬁling and image analysis facilitated the proﬁling of >30 proteins from thousands of single\\ncells on single tissue sections, opening a new path toward predicting disease subtypes based on immune microenvironments, linked to pro-\\ngression.5,6 In breast cancer, structured immune microenvironments linked to survival were identiﬁed. 7–9 In colorectal cancer, distinct cellular\\ncommunities, which can distinguish molecular subtypes and aid in the risk stratiﬁcation of aggressive forms of cancer, were revealed. 10 These\\nstudies show the power of mapping spatial associations in the tumor microenvironment (TME) for predicting patient trajectory and revealing\\nnovel targets for therapeutic interventions. 4\\n1Cancer Early Detection Advanced Research (CEDAR), Knight Cancer Institute, OHSU, Portland, OR 97239, USA\\n2Department of Biomedical Engineering, School of Medicine, OHSU, Portland, OR 97209, USA\\n3Department of Molecular and Medical Genetics, Knight Cancer Institute, OHSU, Portland, OR 97239, USA\\n4Department of Pathology & Laboratory Medicine, School of Medicine, OHSU, Portland, OR 97239, USA\\n5Department of Urology, School of Medicine, Knight Cancer Institute, Portland, OR 97239, USA\\n6Cancer Institute, University College London, London, UK\\n7Division of Oncological Sciences, School of Medicine, OHSU, Portland, OR 97239, USA\\n8Present address: Zeynep Sayar, Parse Biosciences, Seattle, WA, 98109 USA\\n9Present address: Guillaume Thibault, Merck, 126 East Lincoln Avenue PO Box 2000. Rahway, NJ 07065 USA\\n10Present address: Erik Burlingame, Johnson & Johnson Innovative Medicine, New Brunswick, NJ, 08933 USA\\n11These authors contributed equally\\n12Lead contact\\n*Correspondence: eksi@ohsu.edu\\nhttps://doi.org/10.1016/j.isci.2024.110668\\niScience 27, 110668, September 20, 2024ª 2024 The Authors. Published by Elsevier Inc.\\nThis is an open access article under the CC BY-NC-ND license (http://creativecommons.org/licenses/by-nc-nd/4.0/).\\n1\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ece01469-bfa5-45b7-bf6c-414519e9c299', embedding=None, metadata={'page_label': '3', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 1. Single-cell phenotypes of primary prostate tumors identiﬁed via cyclic immunoﬂuorescent imaging\\n(A) Diagram showing the experimental (top) and computational (bottom) pipeline of cyclic immunoﬂuorescent (cycIF) analysis of primary prostate tu mors (image\\ncreated with Biorender.com). Tissue sections were stained, imaged, and quenched repetitively for 10 rounds. Cell segmentation detected 699,461 cells in total\\nthat were annotated into 27 cell types following a multi-step clustering approach. 10 cellular neighborhoods, which were composed of distinct recur rent cell\\ntypes, were identiﬁed across all clinical groups.\\n(B) The antibody library used in the study distinguishes between epithelial (red), immune (yellow), and non-immune stromal cells (teal) and includes several\\nfunctional status markers (pink) for each cell type.\\nll\\nOPEN ACCESS\\n2 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='66311e0b-ef15-4541-a014-e2e71c9c8c28', embedding=None, metadata={'page_label': '4', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='In the prostate TME, dynamic changes in the physical microenvironment of prostate tumors, such as hypoxia, wound healing, and chronic\\ninﬂammation, are postulated to contribute to cancer initiation and progression. 11–13 These processes require coordination between immune\\nand non-immune stromal cells. However, our understanding of non-immune stromal cell types is limited, especially around androgen recep-\\ntor (AR) expression in the prostate TME. A few studies report loss of stromal AR expression that is strongly correlated with the worst patient\\noutcomes.14,15 However, a better delineation of grade-speciﬁc AR-positive (AR+) stromal cell types and of pathways involved in AR regulation\\nin stromal cells is needed. 16\\nHere, we developed an extensive antibody library to probe changes in the prostate TME and identiﬁed 27 epithelial, immune, and non-\\nimmune stromal cell phenotypes that exhibit prostate cancer grade-speciﬁc differences in spatial composition ( Figure 1 A). We used cyclic\\nimmunoﬂuorescent (cycIF) imaging to spatially map 699,461 cells from 13 patients and three clinical groups ( Figure 1A). Further, we analyzed\\ncopy-number aberrations (CNAs) for 10 patients to investigate changes in the spatial coordination of immune and non-immune stromal cells.\\nWe show AR+ stroma-driven neighborhood organization associated with clinical groups and deﬁne transcriptional networks active in AR+\\nstroma using single-cell combinatorial indexing assay for transposase-accessible chromatin sequencing (sci-ATAC-seq) from the same group\\nof patients.17 Our study provides a comprehensive spatial and regulatory map of the prostate TME for patients with low and high-risk localized\\nprostate tumors and highlights the importance of spatial proﬁling in identifying patient-speciﬁc disease subtypes.\\nRESULTS\\nDeveloping a prostate-cancer-speciﬁc antibody library to probe immune and non-immune stromal cell type changes\\nTo study systemic changes in the prostate TME, we developed an extensive antibody library to capture prostate cancer cells in relation to\\nimmune and non-immune stromal cell types. We tested over 150 antibodies and validated more than 100 antibodies following a three-\\nstep veriﬁcation for each antibody ( Figure S1, Data S1 and S2). Our antibody library is speciﬁcally geared toward capturing changes in prostate\\ncancer biology.\\nWe selected 28 targets from this panel for cycIF, which allowed for a thorough classiﬁcation of speciﬁc cell types in the prostate tissue:\\nepithelial, immune, and non-immune stromal cells. We also included molecular and epigenetic markers, functional status markers, and a pro-\\nliferation marker to capture speciﬁc cell states associated with patient outcomes ( Figures 1 B and S2).\\nSpatial proﬁling of 699,461 cells isolated from tumors spanning three clinical groups\\nFormalin-ﬁxed parafﬁn-embedded whole-tissue sections from patients with localized primary prostate tumors were obtained for cycIF anal-\\nysis (Figures 1A and 1C). Whole-slide images varied between 2 and 26 mm 2 in surface area. We did not generate tumor microarrays (TMAs) to\\navoid any sampling bias. 18 Tissue sections were assigned to clinical groups: tumor-adjacent normal (TAN), primary Gleason grade 3 (G3), or\\nprimary Gleason grade 4 (G4), based on their histopathological evaluations. If a tissue section contained >95% normal glands or stroma and\\n<5% G3 glands, it was assigned to the TAN group. If >75% of the section consisted of G3 tumors, it was assigned to the G3 group, and if >75%\\nof the section consisted of G4 tumors, it was assigned to the G4 group ( Figures S3 and S4). The Cancer of the Prostate Risk Assessment Post-\\nSurgical (CAPRA-S) scores for all patients were calculated based on clinical patient data. 19 We identiﬁed ﬁve patients with low-, six patients\\nwith intermediate-, and two patients with high-risk CAPRA-S scores ( Figures S3 and S4).\\nWe analyzed the co-expression of 28 proteins across 13 whole-tissue sections using cycIF. Single-cell segmentation via CellPose identiﬁed\\n699,461 cells, with similar numbers of cells from each clinical group (G3, G4, and TAN) ( Figures 1 D and 1E). 20\\nThe density of epithelial, immune, and non-immune stromal cell types indicates changes in the cellular organization of\\ntissues associated with clinical group\\nOur previous work uncovered a loss of epigenomic heterogeneity in epithelial prostate cancer cells. 17 To further investigate cell type hetero-\\ngeneity, we performed a multi-step clustering analysis for an in-depth classiﬁcation of epithelial, immune, and non-immune stromal cells in\\nprostate tumors. First, we clustered all 699,461 single cells based on the expression of all 28 markers in our panel and detected 49 clusters\\n(Figure 2 A).21 We used uniform manifold approximation and projection (UMAP) to visualize single cells based on Louvain clustering results,\\ncolored by either patient IDs (left) or cell types (right) ( Figure 2A).22 Single-cell clusters on the UMAP space showed patient-speciﬁc epithelial\\ncell clusters ( Figure 2A). In contrast, immune and non-immune stromal cells across all patients formed a single, mixed cluster separate from all\\nepithelial cells ( Figure 2 A).\\nTo delineate cellular heterogeneity, we plotted Z-scored protein expression values as a heatmap and used epithelial (CK5, CK8, ECAD,\\nand chromogranin A), immune (CD3, CD4, CD11b, CD20, CD45, CD66b, CD68, CD163, FOXP3, HLADRB1, and Tryp) and non-immune stro-\\nmal (aSMA and VIM) markers to create three exclusive categories ( Figure S5). Further details about cell phenotyping are provided in the STAR\\nmethods section. This analysis identiﬁed 435,702 epithelial cells (62% of cells), 209,521 stromal cells (30% of cells), and 54,238 (8% of cells)\\nFigure 1.Continued\\n(C) Example cycIF images from TAN, G3, and G4 tissue sections show aSMA (blue), AR (red), and Ki67 (magenta) markers and H&E images from the adjacent\\ntissue section. White arrows show normal (TAN) or cancerous (G3 and G4) glands. Yellow arrows show stroma.\\n(D) CellPose segmentation algorithm determines the boundaries of each nucleus (left) and cell (right). Arrows point out to the deﬁned boundaries of a stromal\\ncell.\\n(E) Numbers of cells were segmented from each clinical group TAN, G3, and G4.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 3\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ac9a4d53-cb1c-4ee7-9fb6-46012b986f7f', embedding=None, metadata={'page_label': '5', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ll\\nOPEN ACCESS\\n4 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c4e31615-688e-45c6-b5b7-0dd4bb9ceebc', embedding=None, metadata={'page_label': '6', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='immune cells (Figure 2B). We analyzed the percentage of cell types across three clinical groups, which showed an increase in epithelial cells by\\n21% and 34% and a decrease in immune and non-immune stromal cells by 15% and 27% in G3 and G4 tumors, respectively, compared to TAN\\nsamples (Figure 2 B).\\nThe combined expression of epithelial markers shows epigenetic differences between clinical groups\\nWe sought to further deﬁne cellular heterogeneity for epithelial, immune, and non-immune cells. First, we investigated epithelial cell hetero-\\ngeneity, by re-clustering only the epithelial cells using the protein markers NCAM, chromogranin A, CD44, VIM, CK5, NKX3.1, AMACR, CK8,\\nECAD, H3K27ac, H3K4me3, and AR, and we identiﬁed ﬁve epithelial cell types: basal, luminal, AMACR-high luminal, neuroendocrine, and\\nVIM+ epithelial cells that are likely in the process of epithelial-to-mesenchymal transition (EMT) (Figure 2 C).\\nAMACR is consistently overexpressed in prostate cancer epithelium and is suggested as a cancer-risk-associated marker. 23,24 We\\nobserved that 86% of all luminal cells express average levels of AMACR, and we found a separate cluster of luminal cells with high\\nAMACR expression in G3 and G4 samples that we labeled as ‘‘AMACR-high’’ (Figure 2 C).\\nWhile we observed an increase in the percentage of proliferating (Ki67+) AMACR-high luminal cells in G4 samples, the density of Ki67+\\nAMACR-high luminal cells was similar in G3 and G4 tumors ( Figure S5). Signiﬁcantly, we found an increase in the density of Ki67+ VIM+/EMT\\ncells in G4 tumors, but not in TAN and G3 ( Figure S5 ).\\nTo quantify the heterogeneity of epithelial cells, we measured Shannon entropy for each clinical group. 21 We observed a non-signiﬁcant\\ndecrease in heterogeneity from TAN to G3 and G3 to G4, suggesting epithelial cells in TAN samples show more differences in epithelial\\nmarker expression, as opposed to G4 samples (Figure 2 D).\\nCell types are static categories that may not reﬂect the heterogeneity of cell states in prostate cancer epithelia. To determine the speciﬁc\\nmarkers associated with the heterogeneity of epithelial cell states, we investigated the combinatorial expression of 12 epithelial markers in\\n699,461 single cells ( Figure S6). Strikingly, we found that the combined expression of AMACR, H3K27ac, H3K4me3, and AR in epithelial cells\\nshowed a signiﬁcant difference across the three clinical groups: 21.5% in G4, 6.4% in G3, and 0.4% in TAN ( Figure S6). This signiﬁcance was not\\nobserved for CK8+ luminal cells: approximately similar percentages of epithelial cells, 20.3% in G4, 18% in G3, and 19.8% cells were positive\\nfor CK8, AMACR, H3K27ac, H3K4me3, and AR ( Figure S6 ). We performed a chi-squared test that showed that the statistical difference was\\nsigniﬁcant between the three clinical groups ( Figure 2E). We conﬁrmed the loss of CK8 in G4 tumors using H&E images of individual TAN, G3,\\nand G4 glands ( Figure 2 F).\\nH3K27ac marks active regulatory regions associated with increased transcription. 25,26 We observed a 10% increase in the number of cells\\nwith H3K27ac expression in G4 tumors ( Figure S6 ). To determine whether the spatial distribution of this marker may contribute to disease\\nprogression, we calculated Moran’s I values for all epithelial markers ( Figure S6) and visualized spatial expression patterns on the tissues ( Fig-\\nure 2 G). Moran’s I uses continuous marker expression across a tissue to determine spatial auto-correlation. 27 We observe broad, low-level\\nexpression of H3K27ac that overlaps with AMACR expression in G4 samples, as opposed to TAN ( Figure 2 G).\\nIdentiﬁcation of Tryp+ mast cell populations that differentially express CD44, CD90, and GZMB\\nWe further stratiﬁed immune cells and identiﬁed 13 myeloid and lymphoid cell types with Louvain clustering based on the expression of 17\\nimmune markers: CD66b, CD11b, CD20, NCAM, PD1, CD163, HLA-DRB1, CD68, CD90, Tryp, CD44, GZMB, CD45, CD3, CD4, FOXP3, and AR\\n(Figure 3 A). Tryp+ mast cells constituted the second largest population of immune cells (18%) following helper T cells (19%), forming two\\ndistinct clusters on the UMAP, which suggested heterogeneous mast cell populations in localized prostate tumors ( Figures 3 A and 3B).\\nEven though the presence of large fractions of Tryp+ mast cells is repeatedly reported in localized prostate tumors, our understanding of\\ntheir role in mediating tumor response is limited. 28 We determined that the majority of mast cells express CD44, a receptor of hyaluronic acid\\nthat is abundant in the extracellular matrix ( Figure 3 C).29 We investigated this further by analyzing the differential expression of immune\\nmarkers in two mast cell populations ( Figure 3D). Our results show a signiﬁcant abundance of mast cluster 2 in G3 patients and mast cluster\\n1 in TAN patients and no difference in G4 patients ( Figure 3 E). Furthermore, we identiﬁed a positive correlation between CD90 and GZMB\\nexpression in mast cluster 2 ( Figure 3 F).\\nMast clusters 1 and 2 showed a striking difference based on the CD90 and GZMB status ( Figures 3 G and 3H). GZMB induces anti-tumor\\nresponses and is traditionally found in effector T cells. 30 However, certain GZMB+ cells, such as myeloid-derived suppressor cells and\\nFigure 2. Combined expression of four markers, AR, AMACR, H3K27ac, and H3K4me3, distinguishes prostate epithelial cells across clinical groups\\n(A) Louvain clustering of 699,481 cells shows clinical-group (left) and cell-type-speciﬁc (right) clusters.\\n(B) The fraction of epithelial cells (purple) increased, while immune (yellow) and non-immune stromal (green) cells decreased in G4 tumors.\\n(C) Annotations of epithelial cell clusters based on the clinical groups (left) and cell types (right).\\n(D) Shannon entropy results measure the distribution of cells expressing epithelial markers across clinical groups, showing an overall decrease in marker\\nheterogeneity with increasing primary tumor grade.\\n(E) The combined expression of four markers (AR, AMACR, H3K27ac, and H3K4me3) distinguishes between G4 samples vs. TAN and G3 samples (chi-square test\\nwith p < 2.2e/C0 16). The mosaic plot shows deviations of observed frequencies as compared to expected. The area of the box is proportional to the difference in\\nobserved and expected frequencies. Pink indicates smaller and blue indicates larger observed frequencies as compared to expected ( p value<2.22 3 10/C0 16).\\n(F) A G4 grade gland detected by an H&E shows expression of AR, AMACR, H3K27ac, and H3K4me3, but does not express CK8 (luminal) or CK5 (basal) markers\\n(left). A G3 grade and a normal gland detected by an H&E show CK8 and CK5 expression.\\n(G) Plot showing the spatial distribution of H3K27ac and AMACR in an example TAN (top) and G4 (bottom) samples.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 5\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0ac14eb3-256c-40a0-a528-c03fa39c5d3f', embedding=None, metadata={'page_label': '7', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ll\\nOPEN ACCESS\\n6 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4d1b74f5-4681-48d8-9c9e-02de9b1b1271', embedding=None, metadata={'page_label': '8', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='mesenchymal stem cells (MSCs), can secrete GZMB to degrade receptors on effector T cells and induce immunosuppressive pro-tumor\\neffects.30 Mast cluster 2 exhibited a higher fraction of GZMB+ mast cells (55%) as opposed to cluster 1 (21%) ( Figure 3 G). We identiﬁed\\ndifferential expression of CD68, AR, PD1, CD44, and CD90 in two clusters of mast cells ( Figure 3 H). We investigated single, dual, and triple\\npositivity for markers CD44, CD90, GZMB, and AR in Tryp+ mast cells and plotted these mast populations across clinical groups ( Figures 3I\\nand S7). We validated the presence of dual and triple-positive mast cell populations in the prostate TME using patient cycIF images\\n(Figure 3 J).\\nGZMB/C0 and AR+ mast cells showed spatial association with M2 macrophages and Tregs in the prostate TME\\nTo investigate whether identiﬁed mast cells are associated with pro- vs. anti-tumor immune cell types, we examined spatial co-occurrence\\npatterns of mast cells in the prostate TME ( Figure 4A). We divided mast cells into two groups based on their GZMB expression and analyzed\\nthe frequency of spatial co-occurrence of GZMB+ or GZMB /C0 mast cells with all identiﬁed epithelial, immune, and non-immune stromal cell\\ntypes.10 GZMB+ and GZMB/C0 mast cells did not show any speciﬁc enrichment in their spatial auto-correlation in TAN samples ( Figures 4B and\\nS8). However, we identiﬁed an enriched spatial co-occurrence between GZMB+ mast cells and M2 macrophages and between GZMB /C0 mast\\ncells and CD90 + MSCs in G3 tumors ( Figure S8 ). GZMB+ mast cells similarly showed highly signiﬁcant spatial association with M2 macro-\\nphages and granulocytes in G4 tumors ( Figures 4C and S8). We did not observe GZMB /C0 cell spatial association with MSCs, but with neuronal\\ncells in G4 prostate TME ( Figure 4 C).\\nFunctional interactions of mast cells with regulatory T cells (Tregs) in colorectal cancer are shown to generate potently immunosuppressive\\nand proinﬂammatory Tregs. 31,32 To test whether expression proﬁles of mast cells impact their spatial correlation, we compared the behavior\\nof mast cells with and without AR expression in the prostate TME. We identiﬁed AR+ mast cells that closely interact with Tregs in G3 and G4\\nprostate tumors as compared to TAN samples (Figures 4 C and S8). Additionally, AR+ mast cells had enriched spatial association with M2\\nmacrophages in G3, but not in G4 tumors ( Figures 4C and S8). AR/C0 mast cells did not show any speciﬁc enrichment in their spatial association\\n(Figure S8). Taken together, our results show changes in the spatial association of mast cells with speciﬁc lymphoid, myeloid, and stromal cells\\nbased on their GZMB and AR expression, suggesting possible functional roles for distinct mast cell states.\\nTo investigate whether mast cell spatial auto-correlation directly inﬂuences these results, we ﬁrst generated Voronoi diagrams of each tis-\\nsue section and then mapped mast cells onto Voronoi neighborhoods ( Figure 4 A).10 Next, we computed Moran’s I values for each patient,\\nindicating the spatial auto-correlation of continuous variables, using mean signal intensity values of all markers in the panel ( Figure 4B). Our\\nresults did not show any differences in mast cell auto-correlation associated with a clinical group (Moran’s I average = 0.08 for all groups). This\\nprovides further evidence that the speciﬁc spatial association of mast cells with M2 macrophages is not an artifact of spatial auto-correlation\\nbut is directly linked to tumor progression in high-grade tumors ( Figures 4 B and 4C).\\nWe validated this result with Ripley’s L statistics, which is similar to Moran’s I but employs discrete values as its input, such as cell types. 33\\nMast cells did not show spatial auto-correlation, whereas CD4 + T cells did ( Figure 4 D). Additionally, we calculated the co-occurrence prob-\\nability score for each cell type in comparison to GZMB+ ( Figure 4 E left) and AR+ ( Figure 4 E right) mast cell populations. GZMB+ mast cells\\nexhibited signiﬁcant spatial co-occurrence with non-APC M2 macrophages, and AR+ mast cells showed signiﬁcant spatial co-occurrence with\\nCD3+CD4/C0 cytotoxic T cells based on increasing distance around each cell in the tissue ( Figure 4 E). We observed that mast cells close to\\nCD163+ M2 macrophages frequently expressed GZMB (GZMB+Tryp+; white box on the left), while mast cells spatially distant from M2 mac-\\nrophages did not express GZMB (GZMB-Tryp+; gray box on the right) ( Figure 4 F).\\nAR+ non-immune stromal cells drive cellular neighborhood organization in tumor samples\\nWe further explored potential predictive factors linked to clinical groups by focusing on non-immune stromal cells. We delineated hetero-\\ngeneous stromal cell populations by clustering them in isolation based on the expression of the following markers: aSMA (smooth muscle),\\nFigure 3. Mast cells in the prostate tumor microenvironment differentially express CD44, CD90, and GZMB\\n(A) UMAP shows immune cell subtype clusters identiﬁed via Louvain clustering. Mast cell clusters 1 and 2 are shown in gray.\\n(B) Fractions of immune cell subtypes show high numbers of mast cells across all clinical groups.\\n(C) Heatmap showing log-normalized signal intensity values of each immune marker for the two mast cell clusters identiﬁed in our dataset. A large frac tion of mast\\ncells in cluster 2 (orange) are GZMB+ (blue).\\n(D) Reclustering of mast cells, UMAPs colored by clinical group (left), and previously identiﬁed mast cell clusters 1 (green) and 2 (orange) (right).\\n(E) The association plot reveals a signiﬁcant abundance of mast cluster 2 in G3 samples and mast cluster 1 in TAN samples and no difference in G4. Rectangles in\\neach row are positioned relative to a baseline indicating independence. If the observed frequency of a cell is greater than the expected one, the box ri ses above\\nthe baseline; otherwise, it falls below. The area of the box is proportional to the difference in observed and expected frequencies ( p value <2.22 3 10/C0 6).\\n(F) The scatterplot shows a positive correlation between GZMB and CD90 expression in mast cells (R = 0.29).\\n(G) Bar plot showing the fraction of GZMB+ and GZMB /C0 cells across mast clusters.\\n(H) Boxplot showing AR, CD44, CD90, PD1, and CD68 expression differs signiﬁcantly in mast clusters 1 and 2 (values represent mean GSEM; t test **** p < 0.001).\\n(I) Percentage distributions of mast cells showing mono-, bi-, and tri-expression of CD44, CD90, GZMB, and AR based on mast cell cluster IDs (top) and c linical\\ngroups (bottom).\\n(J) An example cycIF image that shows heterogeneous mast cell populations, AR+GZMB-CD90 +CD44, AR-GZMB+CD90 +CD44+, AR-GZMB-CD90 +CD44+,\\nAR+GZMB-CD90+CD44+. AR is shown in red, AMACR in white, tryptase (mast cells) in cyan, GZMB in purple, CD90 in green, and CD44 in yellow. The scale\\nbar indicates a length of 100 mm.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 7\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='799e8d3a-9835-4463-b02d-42a4e49e856d', embedding=None, metadata={'page_label': '9', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ll\\nOPEN ACCESS\\n8 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c07d0781-8f94-4065-a587-6aed2691c6d8', embedding=None, metadata={'page_label': '10', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='VIM (mesenchymal ﬁbroblasts and endothelial cells), CD90 and CD44 (MSCs and cancer-associated ﬁbroblasts), NCAM (neurons), HLA-DRB1\\n(inﬁltrating antigen-presenting cells), and AR ( Figures 5 A and 5B). 34,35\\nOur results uncover three populations of AR+ stromal cells: AR+ smooth muscle, AR+ aSMA, and VIM+ and AR+ other stromal cells\\n(Figures 5 A and 5B). We determined a signiﬁcant increase in AR+ stromal cells (27%) in TAN samples, as compared to G3 and G4 tumors\\n(6%) (Figure 5B). Interestingly, despite this increase, we did not observe signiﬁcant enrichment in the spatial auto-correlation of AR+ stromal\\ncell markers in TAN samples, whereas AR+ stromal cells in tumor samples showed signiﬁcant enrichment in their spatial auto-correlation\\n(Figure S9 ).\\nTo determine the impact of AR+ stromal cell populations on tissue organization, we mapped the spatial frequencies of all 27 cell types for\\neach patient and used non-hierarchical clustering to identify recurrent cell neighborhoods (rCNs) 10 (Figure 5C). Each rCN was then mapped\\nback onto the tissue sections to visualize cell type and neighborhood compositions. 10\\nOur results indeed show that AR expression in prostate TME strongly informs the 10 rCNs identiﬁed across all patient samples. Four neigh-\\nborhoods were dominated by AR+ non-immune stromal cells interacting with neuroendocrine (rCN1), mesenchymal/endothelial (rCN8), im-\\nmune (rCN2), and smooth muscle and M2 macrophages (rCN5). The other six neighborhoods were dominated by the lack of AR+ stroma:\\nluminal epithelial (rCN0), EMT/inﬁltrating VIM+ (rCN9), AMACR-high luminal cells interacting with APC immune cells (rCN6 and rCN7), basal\\nand neuroendocrine (rCN3), and basal, neuroendocrine, and luminal cells (rCN4) (Figure 5 C).\\nWe sought to determine if the signiﬁcant heterogeneity of cell types impacted the identiﬁed neighborhoods. We calculated the frequency\\nof each rCN for TAN, G3, and G4 samples. Our results identiﬁed rCN1 and rCN2, which showed a statistically signiﬁcant increase in TAN sam-\\nples (Figure 5D). Both rCN1 and rCN2 are deﬁned by recurrent spatial correlation of AR+ stroma with immune cells ( Figure 5C). The frequency\\nof these neighborhoods was signiﬁcantly lower in both G3 and G4 samples as compared to TAN ( Figure 5 D). We observed that rCN5 was\\ncharacterized by a notable lack of B cells, differentiating it from other stromal neighborhoods ( Figure 5 C). We also identiﬁed a signiﬁcant\\nincrease in the frequency of rCN6, which consists of AMACR-high luminal cells in G3 and G4 tumors ( Figures 5 C and 5D). We repeated\\nthe neighborhood identiﬁcation analysis 13 times, leaving one patient out each time ( Figure 5 E). Our results robustly captured the same\\nrCN composition in each model: AR+ stroma-driven TAN vs. G3 and G4 differences were identiﬁed in all analyses, and AMACR-high-driven\\nrCN change was identiﬁed in 11 out of 13 (Figure 5 E).\\nTo address the effects of spatial variability and sample size, we performed cellular neighborhood analysis on in silicotissues derived from\\neach patient sample, with 10 or 50 tiles per clinical group ( Figure S10). Our ﬁndings revealed consistent neighborhoods for TAN, G3, and G4\\ntissues, highlighting a signiﬁcantly higher AR+ rCN frequency in TAN compared to G3 and G4 tumors ( p value<0.05) (Figure S11). Addition-\\nally, we performed sub-sampling analysis through 100 iterations at four independent values (50%, 75%, 90%, and 99% of the data). The mean\\nAR+ rCN frequency difference between groups (TAN vs. tumors) across these iterations was statistically signiﬁcant (horizontal red line shows\\np = 0.05). In other words, even with 50% data subsampling, there is a statistically signiﬁcant difference for AR+ stromal neighborhoods in TAN\\nvs. G3 (86/100 of the iterations) and TAN vs. G4 (90/100 of the iterations) ( Figure 5F). Combined, these results underscore the signiﬁcance of\\nAR regulation in non-immune stroma as a key factor in prostate cancer progression.\\nLoss of heterogeneity in cellular organization\\nOur results showed that AR+ stromal neighborhoods may distinguish between three clinical groups of prostate samples ( Figures 5 C–\\n5F).12,36–38 To further delineate these clinical groups, we identiﬁed rCNs separately for TAN, G3, and G4 and identiﬁed cell types that are\\ndifferentially enriched within speciﬁc neighborhoods across different clinical groups ( Figures S8 and S10).\\nLooking at the frequency of each rCN in three clinical groups, we found TAN samples consist of neighborhoods that have similar fre-\\nquencies across the tissue section (less than 25% frequency for each neighborhood) ( Figure S9 ). In contrast, G4 samples consisted of one\\nor two dominant neighborhoods ( Figures 5 D and S9). In G3 tumors, the predominant neighborhoods exhibited higher than 25%, but less\\nthan 40% frequency ( Figures S9 and S12). Overall, these results suggest a loss of rCN heterogeneity associated with higher primary Gleason\\ngrade, speciﬁcally as a result of the decrease in AR+ stromal cell neighborhoods ( Figures 5 D and S8).\\nWe calculated the percentage size distribution of each rCN per clinical group and showed a decrease in the actual size of the AR+ stromal\\nneighborhoods in G3 and G4 tumors ( Figure 5G). The size of the AMACR-high neighborhoods showed an increase in G3 and G4 patients as\\nFigure 4. Mast cell subtypes spatially associate with distinct immune cells\\n(A) Voronoi diagrams of cellular neighborhoods for patients TAN_1, G3_3, and G4_1. Each dot shows mast cell distribution across whole tissue section s.\\n(B) Average Moran’s I score of immune cell subtypes across three clinical groups does not show differences based on mast cell marker expression (Tryp) (bottom,\\norange).\\n(C) Heatmaps of likelihood ratios of direct cell-cell contacts between 27 cell types in G4 tumors. Mast cells (Tryp+) were divided into two groups GZMB +//C0 (left)\\nand AR+//C0 (right). GZMB+ mast cells showed highly enriched spatial correlation with M2 macrophages and are closely associated with granulocytes. AR+ mast\\ncells showed highly enriched spatial association with Treg cells.\\n(D) Ripley’s L results of patient G3_3 highlight cell types that are spatially auto-correlated, such as CD4 +CD3+ T cells (red), and others that are spatially dispersed,\\nsuch as granulocytes (brown) and GZMB+ mast cells (black).\\n(E) Co-occurrence probability of each cell type given the presence of GZMB+ (left) and AR+ mast cells (right). The score is computed across increasingdistances\\naround each cell in the tissue. GZMB+ mast cells, non-APC M2 macrophage, AR+ mast cells, and CD4 /C0 CD3+ T cells show co-occurrence in close proximity.\\n(F) cycIF images showing GZMB+ mast cells (gray boxes) near M2 macrophages and GZMB /C0 mast cells (white box). DAPI is shown in white, TRYP in blue, CD163 in\\nyellow, and GZMB in red. The scale bar indicates a length of 50 mm.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 9\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b82b74fb-a43e-4e54-b70b-ed5589a922af', embedding=None, metadata={'page_label': '11', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ll\\nOPEN ACCESS\\n10 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='cb86156a-49f6-4ae5-a762-94d8f487b134', embedding=None, metadata={'page_label': '12', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='compared to TANs, supporting our results ( Figure 5G). Distinct rCNs dominated patients with high-grade prostate tumors, suggesting spe-\\nciﬁc molecular changes underlying the rCN organization ( Figure 5 H).\\nIdentiﬁcation of patient-speciﬁc genomic events\\nTo further delineate patient-speciﬁc disease subtypes, we performed low-pass whole-genome sequencing for 10 patients and identiﬁed com-\\nmon CNAs. G4_1 and G4_2 patients did not have any known CNAs. G4_3 patient exhibited three known ampliﬁcation events: NCOA2, NBN,\\nand EGFR. G4_4 patient showed three known deletion events: PTEN, CDH1, and NKX3.1 (Figures S3 and S4). PTEN loss is associated with an\\nimmunosuppressive microenvironment that is mediated by Tregs and M2 macrophages, 39 which correlates with our neighborhood analysis\\nthat shows a smaller number of APC M2 macrophages in G4 tumor neighborhoods as compared to TAN and G3 ( Figure S13). Finally, G4_5\\npatient showed multiple ampliﬁcation (PIK3CA and EGFR) and deletion (PTEN, CHD1, RB1, FOXO1, TP53, CDKN1B, and NKX3.1) events.\\nOverall, our combined genomic proﬁling and cycIF imaging results further stratify G4 tumor subtypes and exhibit major differences in TME\\norganization that may be associated with known CNAs in prostate cancer, underlining the need for understanding patient-speciﬁc trajectories\\nfor precise treatment approaches (Figure 5 H).\\nAR+ stromal cells in localized prostate tumors show activated NF-kB signaling\\nTo gain further understanding of the biological targets and pathways in AR+ stromal cells, we shifted from imaging and focused on identifying\\npotential gene networks active in AR+ cells in prostate TME. We used sci-ATAC-seq data from ﬂash-frozen tumors collected from the same\\npatients17 and identiﬁed myeloid, lymphoid, and non-immune stromal cells based on cisTopic clustering and gene expression scores 40,41 (Fig-\\nure 6A). We identiﬁed differentially accessible regions of AR+ vs. AR-null (AR0) cells and observed signiﬁcantly enriched accessibility in chro-\\nmatin regions that regulate nuclear factor kB (NF-kB) signaling in AR+ stromal cells ( Figure 6B).42 Gene ontology (GO) analysis was performed\\nusing GREAT and the Panther classiﬁcation system. 42 To gain more mechanistic insights, we investigated transcription factor (TF)-binding\\nmotifs signiﬁcantly enriched in AR+ vs. AR0 stromal cells using both Cistrome and JASPAR 43 (Figure 6C). We identiﬁed signiﬁcantly enriched\\nTFs that are known AR regulators in AR+ cells such as SP1, YY1, and NFYA ( Figure 6 D). Interestingly, we observed that AR+ stromal cells in\\nlocalized prostate tumors showed signiﬁcant enrichment for the hypoxia marker HIF1 a, suggesting a role for stromal AR regulation in hypoxia.\\nWe did not observe any enrichment for CXCR4, a known inﬂammatory marker, nor for FOXA1, a known repressor of HIF1 a (Figure 6E). While\\nthere was no signiﬁcant difference in IFN g accessibility, a subgroup of AR0 cells was accessible for IFN g, as expected ( Figure 6D). Predicted\\ngene scores conﬁrmed that NF- kB1 was signiﬁcantly more accessible in AR+ cells in the prostate TME ( Figure 6D), validating our GREAT GO\\nresults (Figure 6B) and previous ﬁndings. 2 We performed cycIF using AR, HIF1 a, and NF-kB on three additional Gleason score 3 + 3 (G3 clinical\\ngroup) patient samples and identiﬁed AR+ cells that also express HIF1 a and NF-kB( Figure 6F). Overall, our sci-ATAC-seq analysis identiﬁed\\nactive transcriptional networks in both AR+ and AR0 stromal cells and deﬁned utilizing markers that can be used to further investigate AR+\\nstromal cells in cycIF imaging.\\nDISCUSSION\\nIn this work, we integrated multiple single-cell approaches with bulk genomic sequencing and pathology to investigate changes in the pros-\\ntate TME using patient samples across three clinical groups. We present novel spatial patterns within localized prostate tumors that imply\\npatient-speciﬁc spatial proﬁles even within tumors of seemingly identical pathological grades. While 13 specimens are not representative\\nof all known molecular subtypes of localized prostate cancer, our results show striking differences in disease subtypes within a primary Glea-\\nson grade, which would not be detected based on an H&E image. Future studies will involve further delineation of disease subtypes by gener-\\nating an increased number of multi-modal spatial maps, matching H&E and cycIF images, and examining their predictive and prognostic clin-\\nical utility.\\nFigure 5. AR-positive non-immune stromal cells direct cellular neighborhood organization in prostate tumors\\n(A) UMAP shows non-immune stromal cells colored based on clinical grade speciﬁc (left) and subtype (right).\\n(B) Fractions of non-immune stromal cells across clinical groups. TAN samples show high numbers of AR+ stromal cells as compared to G3 and G4 tumors.\\n(C) Recurrent cell neighborhoods (rCNs) are identiﬁed based on 27 cell subtypes. The top four neighborhoods are deﬁned by an increased occurrence of A R+\\nstromal cells, the bottom six neighborhoods are deﬁned by a lack of AR+ stroma.\\n(D) The frequency of each neighborhood is analyzed per clinical group. rCN1 and rCN2 (AR+ stromal cells, neuroendocrine and immune subtypes) frequency is\\nsigniﬁcantly higher in TAN as compared to G4 and G3 tumors (t test p value 0.017 and 0.016), respectively. CN6 (AMACR-high luminal/other APC immune cells)\\nfrequency is signiﬁcantly lower in TAN as compared to G4 tumors (t test ** p value 0.001). See Figure S13 for cell type differential enrichment within rCNs.\\nScatterplot showing the frequency of each neighborhood per patient (bottom). Each colored dot represents a patient. rCN1, rCN2 ,and rCN6 show\\nsigniﬁcant differences between TAN and G3 samples; rCN1 and rCN6 show signiﬁcant differences between TAN and G4 (t test *p value<0.05).\\n(E) Dot plot of p values for the rCN frequency comparing prostate tumors across 13 iterations of individual patient exclusion (horizontal red line shows p = 0.05).\\n(F) Scatterplots of p values for the rCN frequency comparing prostate tumors across 100 iterations of random subsamples of 50% of the spatial data (horizontal red\\nline shows p = 0.05).\\n(G) Bar plots showing the percentages of the Voronoi area of each neighborhood cover in TAN, G3, and G4 samples (right) and the actual Voronoi area of eac h ten\\nCNs per grade (left).\\n(H) Voronoi diagrams of G4_1 and G4_5 patients showing rCN distributions. G4_1 and G4_5 patients show enrichment for very different rCNs, luminal (bl ue) vs.\\nAR+ stroma (orange, green, brown, and yellow), respectively.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 11\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3fdfc939-1ea2-49f2-ab0e-6ebc88b0a93d', embedding=None, metadata={'page_label': '13', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ll\\nOPEN ACCESS\\n12 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e2270104-c988-4f5c-ba78-640d8c204cf6', embedding=None, metadata={'page_label': '14', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Our multi-modal study provides a key resource for the prostate cancer community to better understand the spatial associations between\\ncancer and stromal cells in primary G3 and G4 tumors. While direct data integration was not performed, our approach encompassed multiple\\nexperiments on patient samples, including single-cell assay for transposase-accessible chromatin using sequencing (ATAC-seq), cycIF, H&E,\\nand genomic DNA sequencing, improving our understanding of patient-speciﬁc differences prior to treatment.\\nOur multi-modal analyses underline the need for generating more detailed disease subtypes for patients with localized prostate cancer\\nto improve prognosis. Biological underpinnings measured for each Gleason score 4 + 4 tumor exhibit differences in CNAs, immune\\nresponse, and neighborhood organization. For instance, while we observed an overall increase in the frequency of immune-cold luminal\\nneighborhoods in primary G4 tumors, we identiﬁed a patient dominated by an M2 macrophage-AR+ stromal cell neighborhood, which\\nwas consistent with previous studies showing T cell exclusion in tumors due to defects in antigen presentation, allowing tumors to evade\\nthe immune system. 44,45 We identiﬁed one patient dominated by an M2 macrophage-AR+ stromal cell neighborhood. Interestingly, this\\npatient was the only patient in our cohort with an RB1 deletion, which is a critical event associated with neuroendocrine prostate cancer. 46\\nAlthough implications of AR+ stroma and immunosuppressive TME interactions need to be further characterized in the prostate, it has\\nbeen reported that AR expression is correlated with M2 macrophage polarization and disease severity in infections such as asthma in\\nthe lung. 47\\nImmunotherapies have been largely ineffective for prostate cancer. 48,49 Prior studies have evaluated the enrichment of inﬁltrating mast\\ncells, T cells, natural killer cells, and macrophages as predictors of prostate cancer outcome and better stratiﬁcation of patients into accurate\\ntreatment groups.50–52 Although studies have consistently reported the prostate immune cell microenvironment as a largely immunosuppres-\\nsive one, 53–55 there are conﬂicting reports on the enrichment of speciﬁc cell types, such as mast cells. 28,56–58 We believe that the inconsis-\\ntencies between these reports are due to the lack of spatial information and evaluation of each cell type independently as opposed to\\nthe neighborhoods they are found within. 54,59–64 A recent study, 53 however, has taken spatial information into account by considering the\\ncompartment in which the cells are found (epithelia vs. stroma). Major strengths of this study are the use of multiplex digital pathology to\\nquantify seven immune cell types and their large cohort size consisting of TMAs from over 900 patients. Comparison of immune cell density\\nwithin the tumor epithelia vs. stroma showed a speciﬁc enrichment of Tregs and M1 and M2 macrophages in epithelial tumor regions. Consis-\\ntent with other studies on Tregs 45,55 and M2 macrophages, 65,66 Andersen et al. also found that the enrichment of stromal Tregs and epithelial\\nM2 macrophages were independent predictors of biochemical recurrence (BCR). 53 Mast cells alone were not signiﬁcantly associated with\\nBCR.53 In our study, we used whole-tissue sections instead of TMAs to capture long-distance associations across 27 cell types and avoid sam-\\npling bias associated with the TMAs. 18,67 We observed changes in the spatial association of mast cell populations with M2 macrophages and\\nTregs based on their GZMB and AR status. Our study revealed speciﬁc associations between AR+ mast cells with Tregs in G3 and G4 tumors\\nand with M2 macrophages in G3 tumors. Furthermore, we identiﬁed a speciﬁc association of M2 macrophages with GZMB+ mast cells in G3\\nand G4 tumors. Overall, taken into consideration with the aforementioned studies, interactions of AR or GZMB-expressing mast cells with the\\nimmunosuppressive M2 macrophages and Tregs may be functionally important in promoting a more aggressive, BCR-prone prostate cancer\\nphenotype. Functional interactions between mast cells and Tregs and macrophages mediated by cytokines have been investigated in mouse\\nmodels of colorectal cancer. 31,32 Other studies have shown interactions between M2 macrophages and CD90 + cells to promote a pro-tumor,\\ninvasive phenotype in prostate tumors. 68,69 The interactions between mast cells and Tregs lead to potently immune-suppressive and proin-\\nﬂammatory Tregs, possibly through a positive feedback loop mediated through cytokines. 70,71 Mouse models of prostate cancer cells do not\\nhave abundant mast cell populations, but we show these mast cell populations may play a role in immunosuppressive behavior in human\\nadenocarcinoma. Co-culture studies are needed to investigate the functional interactions between GZMB+ and AR+ mast cell populations\\nwith M2 macrophages.\\nMast cells are a suggested independent prognostic variable for prostate cancer progression. 58 Several studies report associations be-\\ntween low tumor mast cell counts with poor outcomes such as BCR and metastasis. 28,56–58,72 Although not statistically signiﬁcant, our data\\nalso show a lower mast cell density in TAN compared to G3 or G4 samples. CD44 expression is shown to be a critical factor for mast cell\\nproliferation, differentiation, and maturation in other diseases.73,74 Here, we report large fractions of mast cells expressing CD44, with no\\ngrade-speciﬁc changes in localized prostate tumors. In the context of myeloid malignancies, CD44 has been demonstrated as a mast cell\\ninvasion receptor and correlates with disease aggressiveness and tumor formation. 74 It is possible that CD44 expression in mast cells mod-\\nulates mast cell attachment to the extracellular matrix and contributes to extracellular matrix reorganization in high-grade tumors. 29\\nDetailed markers are needed to identify CD44 isoforms that may be heterogeneously active across clinical groups. Overall, our analysis\\nFigure 6. AR+ stromal cells in localized prostate tumors show activation of NF-kB signaling and hypoxia\\n(A) AR+ myeloid, lymphoid, and non-immune stromal cells were identiﬁed in single-cell ATAC-seq data from the same patients. Differentially accessi ble regions\\n(DARs) are identiﬁed in AR+ vs. AR0 cells.\\n(B) GREAT GO results show an enrichment for NIK/NF- kB signaling pathway regulators in AR+ stromal cells.\\n(C) TF motif-binding sites identiﬁed in AR+ stromal cells show an enrichment for known AR regulators.\\n(D) Gene scores of identiﬁed TFs and pathways in single-cell ATAC-seq data in AR+ vs. AR0 cells.\\n(E) Heatmap showing fold enrichment values for DARs in AR+ vs. AR0 stromal cells identiﬁes signiﬁcant HIF1a and NF- kB accessibility in AR+ cells.\\n(F) IF staining of a G3 patient sample showing co-expression of HIF1 a (red), AR (magenta), and NF- kB (green) staining in the prostate stroma. DAPI-marked nuclei\\nare shown in blue on the left. AR+ stromal cells that express HIF1 a (A), HIF1 a, NF- kB (B), and NF- kB (C) are marked. Cell ‘‘A’’ provides an example of nuclear\\ncolocalization of HIF1 a and AR, whereas in cell ‘‘B,’’ HIF1 a and NF- kB are colocalized in the cytoplasm in a cell marked by AR+ expression in the nucleus.\\nCell ‘‘C’’ shows expression of NF-kB in the cytoplasm in an AR+ cell. The scale bar indicates a length of 50 mm.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 13\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3d53c9cb-4cf2-4c61-a4c0-7d31d6efa43e', embedding=None, metadata={'page_label': '15', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='emphasizes the importance of evaluating the spatial context and molecular expression of cells as opposed to only cell counts when\\nanalyzing the TME, and we propose that the lack of consensus on which cell types are associated with adverse outcomes is due to the\\nlack of spatial context in previous studies.\\nThe impact of androgen deprivation therapy (ADT) is primarily considered through its effects on cancer cell proliferation and survival.\\nRecently, the inﬂuence of ADT on T cell inﬁltration and function has gained attention. 75 However, our understanding of AR regulation in other\\nimmune and non-immune stromal cells in prostate tumors, and how ADT may alter stromal response to cancer through AR regulation, remains\\nlimited. Mast cells in the skin have been shown to express AR in large fractions, with studies indicating that testosterone treatment does not\\nimpact degranulation. 76 In cutaneous neuroﬁbroma, activated AR promoted mast cell inﬁltration, and AR inhibition reduced mast cell\\ninﬁltration.77 In prostate cancer, enzalutamide similarly increased mast cell inﬁltration by regulating AR expression, which was linked to neuro-\\nendocrine differentiation.78 These ﬁndings highlight the importance of investigating the impact of ADT on AR+ mast cells and their interac-\\ntions with Tregs and M2 macrophages in creating a protumor microenvironment.\\nWe also identiﬁed similarities between patients for each clinical group that may be used for automated patient stratiﬁcation in future\\nstudies. We show that the combined expression of AMACR, H3K27ac, H3K4me3, and AR in epithelial cells can distinguish between TAN,\\nG3, and G4, independent of CK8 status. Furthermore, we report a loss of heterogeneity in high-grade prostate tumors (G4 clinical group)\\nas compared to low-grade (G3) and TAN samples. Our ﬁrst line of evidence for loss of heterogeneity comes from our previous single-cell\\nepigenomic analyses that demonstrate increased similarity in chromatin accessibility proﬁles of high-grade tumors.17,26,79 Here, we observed\\nincreased epithelial expression of epigenetic markers H3K27ac and H3K4me3, which are associated with activated transcription and acces-\\nsible chromatin, accompanied by a decrease in Shannon entropy in high-grade tumors. Additionally, using cycIF and spatial statistics, we\\nobserved the speciﬁc areas of increased similarity in cancer cells on patient sections, identifying overlapping expression of H3K27ac and\\nAMACR in G4 tumors. 67,80 Finally, we report patient-speciﬁc rCNs in high-grade prostate tumors, showing that decreased heterogeneity\\nis also reﬂected at the tissue-organization level.\\nLoss of stromal AR in advanced prostate cancer is associated with biochemical relapse and poor prognosis. 15,16,36,81 However, there are no\\nprevious reports of delineating non-immune stromal cell types in human adenocarcinoma based on AR expression. While we detected 10\\ndistinct stromal cell populations in this study, we observed many unclassiﬁed non-immune stromal cells that were marked positive for AR.\\nWe report that this ‘‘AR+ other stroma’’ cell population played a major role in neighborhood organization across three clinical groups.\\nWe suspect there will be tremendous value in stratifying these ‘‘AR+ other stroma’’ cells to better understand their role in neighborhood or-\\nganization in low- vs. high-risk patients and in response to AR inhibitors. In future studies, additional markers will be added to cycIF panels to\\nspeciﬁcally stratify AR+ stromal cells, such as markers identiﬁed through sci-ATAC-seq.\\nsci-ATAC-seq provided valuable insights into active pathways in AR+ stromal cells. The number of cells captured via sequencing\\nremains low as compared to the number of single cells captured via imaging. However, the depth of cis-a n d trans-regulator information\\ngained from sci-ATAC-seq is unmatched in multiplex imaging, which is limited to the number of protein markers that can be\\nprobed from a single tissue section. Integrating single-cell sequencing and imaging approaches allowed us to overcome the limitations\\nof each experimental tool and unlocked mechanistic insights related to NF- kB and hypoxia activation in AR+ stromal cells. Our\\nﬁndings validate the previous bulk RNA sequencing, 82 proteomics,83 and single-cell RNA sequencing 2 results that show an increased\\nNF-kB signaling in immune cells isolated from localized prostate tumors that are AR+. We validated the co-expression of AR and\\nNF-kB in prostate tumor stroma from patient samples via immunoﬂuorescent imaging, which further supports the robustness of our\\nﬁndings. Furthermore, this suggests that, despite variations in tissue preparation methods between single-cell ATAC-seq and cycIF im-\\naging, our results reveal consistent gene signatures that are pre served across different modalities. Recent advancements in high-\\nthroughput single-cell omics technologies will allow researche rs to understand the role of stromal AR expression in regulating\\ninﬂammation.\\nLimitations of the study\\nWhile our study presents compelling evidence regarding the role of AR+ stroma in prostate tumor organization supported by robust statis-\\ntical analyses, several limitations should be acknowledged.\\nOur patient cohort consists of 13 individuals, which may appear limited. Although this sample size aligns with similar studies in spatial im-\\naging and single-cell analysis, 2,3,84–86 larger cohorts could provide more generalizable results. We performed extensive power analyses,\\nincluding in silicotissue generation, leave-one-patient-out analysis, subsampling, and permutation tests to mitigate concerns about sample\\nsize and variability. These analyses demonstrated the robustness and stability of our ﬁndings. However, future studies with larger cohorts\\ncould potentially reveal more speciﬁc associations between AR+ stromal changes and genomic aberrations.\\nThe leave-one-patient-out analysis was used to assess the impact of individual variability, which resulted in 11 out of 13 replications, in\\nwhich the role of AR+ stroma in prostate tumor organization was highly signiﬁcant. However, there is still a possibility that speciﬁc, unmea-\\nsured patient characteristics could inﬂuence these results. Our study attempted to mitigate this through rigorous statistical methods, but\\ninherent biological variability remains a challenge.\\nSpatial heterogeneity within tissue samples poses another signiﬁcant challenge. Accurate assessment of disease-relevant tumor structures\\nrequires the statistical power of whole-slide imaging (WSI), as opposed to TMAs, which consist of a small subsection of specimens. 18,67 The\\nrequirement for WSI in a research and diagnostic setting comes with substantial cost. Per-patient datasets are over 100-fold larger than those\\nwith TMAs, and cohorts are more difﬁcult to acquire since whole blocks must be accessed and re-cut. In contrast, 2D WSI (100 K cells per\\nll\\nOPEN ACCESS\\n14 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b70b01f0-209f-4f79-aba2-491e369f32a3', embedding=None, metadata={'page_label': '16', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='specimen) largely overcomes this problem (number of effective cells > 100) for the characterization of local neighborhoods. WSI captures the\\nmost variability to demonstrate the differences in distinct grading and their cell compositions. WSI is also the standard in conventional pa-\\nthology and is regarded by the Food and Drug Administration as a diagnostic necessity.67\\nIn summary, while our study provides signiﬁcant insights into the role of AR+ stroma in prostate tumor organization and employs rigorous\\nstatistical analyses to support these ﬁndings, limitations related to sample size, generalizability, and methodological approaches should be\\nconsidered when interpreting the results. Future studies with larger and more diverse cohorts, along with continued methodological advance-\\nments, could further validate and expand upon our ﬁndings.\\nSTAR+METHODS\\nDetailed methods are provided in the online version of this paper and include the following:\\nd KEY RESOURCES TABLE\\nd RESOURCE AVAILABILITY\\nB Lead contact\\nB Materials availability\\nB Data and code availability\\nd EXPERIMENTAL MODEL AND STUDY PARTICIPANT DETAILS\\nB Human subjects\\nd METHOD DETAILS\\nB Cyclic IF antibody panel\\nB Fluorophore conjugation to antibodies\\nB FFPE tissue processing\\nB Cyclic IF imaging of prostate tissues\\nB Whole slide scanning\\nB Low-pass WGS library preparation and sequencing\\nd QUANTIFICATION AND STATISTICAL ANALYSIS\\nB Single-cell ATAC sequencing analysis\\nB Cyclic IF image analysis workﬂow\\nB Spatial analysis\\nB Spatial heterogeneity and spatial power analysis\\nSUPPLEMENTAL INFORMATION\\nSupplemental information can be found online at https://doi.org/10.1016/j.isci.2024.110668.\\nACKNOWLEDGMENTS\\nWe are grateful to the Biolibrary and the Histopathology Shared Resource (HSR) at OHSU for providing patient samples, speciﬁcally Aletha\\nLetsch and Cheyenne Martin. The Biolibrary and HSR were supported by NIH grants P30 CA069533 and P30 CA069533 13S5 through the\\nKnight Cancer Institute. Our research complies with all relevant ethical regulations of the Knight Cancer Institute at OHSU. Samples from pa-\\ntients undergoing radical prostatectomy were collected with informed consent through Biolibrary under IRB#4918. Participants did not\\nreceive compensation. We thank the Spellman lab and the Human Tumor Atlas Network (HTAN) for engaging in discussions. We thank Ste-\\nfanie Kaech Petrie and her team at the Advanced Light Microscopy Core at OHSU for their technical help. We thank Lindsey Cauthen for her\\nfeedback on the manuscript and her revisions. We also thank Shelley Barton for providing feedback on the manuscript. This project was sup-\\nported by funding (CEDAR3410918) from the Cancer Early Detection Advanced Research Center at Oregon Health & Science University,\\nKnight Cancer Institute (S.E.E.).\\nAUTHOR CONTRIBUTIONS\\nZ.S. conducted cycIF experiments. C.A. and Z.S. analyzed the cycIF imaging data. C.A. and A.C. analyzed the single-cell ATAC-seq data. C.B.\\nand S.E.E. performed CNA experiments and analysis. G.T. reviewed pathology slides. R.P.K. and V.S. reviewed the clinical patient data. Z.S.,\\nC.A., and S.E.E. designed the experiments and drew conclusions from the data. S.E.E. supervised all experiments and analyses. J.E., G.T.,\\nE.A.B., E.D., and Y.H.C. provided analysis pipelines for cycIF registration, segmentation, and clustering steps. A.A. provided experimental\\nand computational pipelines for single-cell ATAC-seq. Z.S., C.A., and S.E.E. wrote the manuscript with input from all authors. S.E.E. reviewed\\nand edited the manuscript.\\nDECLARATION OF INTERESTS\\nThe authors declare no competing interests.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 15\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d7e67f66-78c7-45b9-b47d-edcddff87273', embedding=None, metadata={'page_label': '17', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Received: April 1, 2024\\nRevised: July 19, 2024\\nAccepted: August 1, 2024\\nPublished: August 3, 2024\\nREFERENCES\\n1. Lawson, D.A., Kessenbrock, K., Davis, R.T.,\\nPervolarakis, N., and Werb, Z. (2018). Tumour\\nheterogeneity and metastasis at single-cell\\nresolution. Nat. Cell Biol. 20, 1349–1360.\\nhttps://doi.org/10.1038/s41556-018-0236-7.\\n2. Tuong, Z.K., Loudon, K.W., Berry, B., Richoz,\\nN., Jones, J., Tan, X., Nguyen, Q., George, A.,\\nHori, S., Field, S., et al. (2021). Resolving the\\nimmune landscape of human prostate at a\\nsingle-cell level in health and cancer. Cell\\nRep. 37, 110132. https://doi.org/10.1016/j.\\ncelrep.2021.\\n3. Hirz, T., Mei, S., Sarkar, H., Kfoury, Y., Wu, S.,\\nVerhoeven, B.M., Subtelny, A.O., Zlatev, D.V.,\\nWszolek, M.W., Salari, K., et al. (2023).\\nDissecting the immune suppressive human\\nprostate tumor microenvironment via\\nintegrated single-cell and spatial\\ntranscriptomic analyses. Nat. Commun.14,\\n663. https://doi.org/10.1038/s41467-023-\\n36325-2.\\n4. Rozenblatt-Rosen, O., Regev, A.,\\nOberdoerffer, P., Nawy, T., Hupalowska, A.,\\nRood, J.E., Ashenberg, O., Cerami, E.,\\nCoffey, R.J., Demir, E., et al. (2020). The\\nHuman Tumor Atlas Network: Charting\\nTumor Transitions across Space and Time at\\nSingle-Cell Resolution. Cell 181, 236–249.\\nhttps://doi.org/10.1016/j.cell.2020.03.053.\\n5. Keren, L., Bosse, M., Marquez, D.,\\nAngoshtari, R., Jain, S., Varma, S., Yang, S.R.,\\nKurian, A., Van Valen, D., West, R., et al.\\n(2018). A Structured Tumor-Immune\\nMicroenvironment in Triple Negative Breast\\nCancer Revealed by Multiplexed Ion Beam\\nImaging. Cell 174, 1373–1387. https://doi.\\norg/10.1016/j.cell.2018.08.039.\\n6. Gustafson, K.T., Sayar, Z., Le, H., Gustafson,\\nS.L., Gower, A., Modestino, A., Ibsen, S.,\\nHeller, M.J., Esener, S., and Eksi, S.E. (2022).\\ncyc-DEP: Cyclic immunoﬂuorescence\\nproﬁling of particles collected using\\ndielectrophoresis. Electrophoresis 43, 1784–\\n1798. https://doi.org/10.1002/elps.\\n202200001.\\n7. Bodenmiller, B. (2020). Imaging mass\\ncytometry and multiplatform genomics\\ndeﬁne the phenogenomic landscape of\\nbreast cancer. Nature Cancer 1, 163–175 .\\n8. Danenberg, E., Bardwell, H., Zanotelli, V.R.T.,\\nProvenzano, E., Chin, S.F., Rueda, O.M.,\\nGreen, A., Rakha, E., Aparicio, S., Ellis, I.O.,\\net al. (2022). Breast tumor microenvironment\\nstructures are associated with genomic\\nfeatures and clinical outcome. Nat. Genet.\\n54, 660–669. https://doi.org/10.1038/s41588-\\n022-01041-y.\\n9. Johnson, B.E., Creason, A.L., Stommel, J.M.,\\nKeck, J.M., Parmar, S., Betts, C.B., Blucher, A.,\\nBoniface, C., Bucher, E., Burlingame, E., et al.\\n(2022). An omic and multidimensional spatial\\natlas from serial biopsies of an evolving\\nmetastatic breast cancer. Cell Rep. Med. 3,\\n100525. https://doi.org/10.1016/j.xcrm.2022.\\n100525.\\n10. Schurch, C.M., Bhate, S.S., Barlow, G.L.,\\nPhillips, D.J., Noti, L., Zlobec, I., Chu, P.,\\nBlack, S., Demeter, J., McIlwain, D.R., et al.\\n(2020). Coordinated Cellular Neighborhoods\\nOrchestrate Antitumoral Immunity at the\\nColorectal Cancer Invasive Front. Cell 182,\\n1341–1359. https://doi.org/10.1016/j.cell.\\n2020.07.005.\\n11. Gurel, B., Lucia, M.S., Thompson, I.M., Jr.,\\nGoodman, P.J., Tangen, C.M., Kristal, A.R.,\\nParnes, H.L., Hoque, A., Lippman, S.M.,\\nSutcliffe, S., et al. (2014). Chronic\\ninﬂammation in benign prostate tissue is\\nassociated with high-grade prostate cancer in\\nthe placebo arm of the prostate cancer\\nprevention trial. Cancer Epidemiol.\\nBiomarkers Prev. 23, 847–856. https://doi.\\norg/10.1158/1055-9965.EPI-13-1126.\\n12. De Marzo, A.M., Platz, E.A., Sutcliffe, S., Xu,\\nJ., Gro¨ nberg, H., Drake, C.G., Nakai, Y.,\\nIsaacs, W.B., and Nelson, W.G. (2007).\\nInﬂammation in prostate carcinogenesis. Nat.\\nRev. Cancer 7, 256–269. https://doi.org/10.\\n1038/nrc2090.\\n13. Bhandari, V., Hoey, C., Liu, L.Y., Lalonde, E.,\\nRay, J., Livingstone, J., Lesurf, R., Shiah, Y.J.,\\nVujcic, T., Huang, X., et al. (2019). Molecular\\nlandmarks of tumor hypoxia across cancer\\ntypes. Nat. Genet. 51, 308–318. https://doi.\\norg/10.1038/s41588-018-0318-2.\\n14. Barron, D.A., and Rowley, D.R. (2012). The\\nreactive stroma microenvironment and\\nprostate cancer progression. Endocr. Relat.\\nCancer 19, R187–R204. https://doi.org/10.\\n1530/ERC-12-0085.\\n15. Cioni, B., Zwart, W., and Bergman, A.M.\\n(2018). Androgen receptor moonlighting in\\nthe prostate cancer microenvironment.\\nEndocr. Relat. Cancer 25, R331–R349. https://\\ndoi.org/10.1530/ERC-18-0042.\\n16. Leach, D.A., and Buchanan, G. (2017). Stromal\\nAndrogen Receptor in Prostate Cancer\\nDevelopment and Progression. Cancers9, 10.\\nhttps://doi.org/10.3390/cancers9010010.\\n17. Eksi, S.E., Chitsazan, A., Sayar, Z., Thomas,\\nG.V., Fields, A.J., Kopp, R.P., Spellman, P.T.,\\nand Adey, A.C. (2021). Epigenetic loss of\\nheterogeneity from low to high grade\\nlocalized prostate tumours. Nat. Commun.\\n12, 7292. https://doi.org/10.1038/s41467-\\n021-27615-8.\\n18. Lin, J.R., Wang, S., Coy, S., Chen, Y.A., Yapp,\\nC., Tyler, M., Nariya, M.K., Heiser, C.N., Lau,\\nK.S., Santagata, S., and Sorger, P.K. (2023).\\nMultiplexed 3D atlas of state transitions and\\nimmune interaction in colorectal cancer. Cell\\n186, 363–381. https://doi.org/10.1016/j.cell.\\n2022.12.028.\\n19. Meurs, P., Galvin, R., Fanning, D.M., and\\nFahey, T. (2013). Prognostic value of the\\nCAPRA clinical prediction rule: a systematic\\nreview and meta-analysis. BJU Int. 111,\\n427–436. https://doi.org/10.1111/j.1464-\\n410X.2012.11400.x.\\n20. Stringer, C., Wang, T., Michaelos, M., and\\nPachitariu, M. (2021). Cellpose: a generalist\\nalgorithm for cellular segmentation. Nat.\\nMethods 18, 100–106. https://doi.org/10.\\n1038/s41592-020-01018-x.\\n21. Burlingame, E.A., Eng, J., Thibault, G., Chin,\\nK., Gray, J.W., and Chang, Y.H. (2021).\\nToward reproducible, scalable, and robust\\ndata analysis across multiplex tissue imaging\\nplatforms. Cell Rep. Methods 1, 100053.\\nhttps://doi.org/10.1016/j.crmeth.2021.\\n100053.\\n22. Becht, E., McInnes, L., Healy, J., Dutertre,\\nC.A., Kwok, I.W.H., Ng, L.G., Ginhoux, F., and\\nNewell, E.W. (2019). Dimensionality reduction\\nfor visualizing single-cell data using UMAP.\\nNat Biotechnol. Nature Biotechnol. 37, 38–44.\\nhttps://doi.org/10.1038/nbt.4314.\\n23. Alinezhad, S., Va ¨ a¨ na¨ nen, R.M., Ochoa, N.T.,\\nVertosick, E.A., Bjartell, A., Bostro ¨ m, P.J.,\\nTaimen, P., and Pettersson, K. (2016). Global\\nexpression of AMACR transcripts predicts risk\\nfor prostate cancer - a systematic comparison\\nof AMACR protein and mRNA expression in\\ncancerous and noncancerous prostate. BMC\\nUrol.16, 1–10. https://doi.org/10.1186/\\ns12894-016-0128-8.\\n24. Ananthanarayanan, V., Deaton, R.J., Yang,\\nX.J., Pins, M.R., and Gann, P.H. (2005). Alpha-\\nmethylacyl-CoA racemase (AMACR)\\nexpression in normal prostatic glands and\\nhigh-grade prostatic intraepithelial neoplasia\\n(HGPIN): association with diagnosis of\\nprostate cancer. Prostate 63, 341–346.\\nhttps://doi.org/10.1002/pros.20196.\\n25. Houlahan, K.E., Shiah, Y.J., Gusev, A., Yuan,\\nJ., Ahmed, M., Shetty, A., Ramanand, S.G.,\\nYao, C.Q., Bell, C., O’Connor, E., et al. (2019).\\nGenome-wide germline correlates of the\\nepigenetic landscape of prostate cancer.\\nNat. Med. 25, 1615–1626. https://doi.org/10.\\n1038/s41591-019-0579-z.\\n26. Sinha, A., Huang, V., Livingstone, J., Wang, J.,\\nFox, N.S., Kurganovs, N., Ignatchenko, V.,\\nFritsch, K., Donmez, N., Heisler, L.E., et al.\\n(2019). The Proteogenomic Landscape of\\nCurable Prostate Cancer. Cancer Cell 35,\\n414–427.e6. https://doi.org/10.1016/j.ccell.\\n2019.02.005.\\n27. WaldhO¨ R, T. (1996). The Spatial\\nAutocorrelation Coefﬁcient Moran’si under\\nHeteroscedasticity. Stat. Med. 15, 887–892.\\nhttps://doi.org/10.1002/(sici)1097-\\n0258(19960415)15.\\n28. Johansson, A., Rudolfsson, S., Hammarsten,\\nP., Halin, S., Pietras, K., Jones, J., Stattin, P.,\\nEgevad, L., Granfors, T., Wikstro ¨ m, P., and\\nBergh, A. (2010). Mast cells are novel\\nindependent prognostic markers in prostate\\ncancer and represent a target for therapy.\\nAm. J. Pathol.177, 1031–1041. https://doi.\\norg/10.2353/ajpath.2010.100070.\\n29. Fukui, M., Whittlesey, K., Metcalfe, D.D., and\\nDastych, J. (2000). Human mast cells express\\nthe hyaluronic-acid-binding isoform of CD44\\nand adhere to hyaluronic acid. Clin. Immunol.\\n94, 173–178. https://doi.org/10.1006/clim.\\n1999.4830.\\n30. Wang, W., Zou, R., Qiu, Y., Liu, J., Xin, Y., He,\\nT., and Qiu, Z. (2021). Interaction Networks\\nConverging on Immunosuppressive Roles of\\nGranzyme B: Special Niches Within the\\nTumor Microenvironment. Front. Immunol.\\n12, 670324. https://doi.org/10.3389/ﬁmmu.\\n2021.670324.\\n31. Blatner, N.R., Bonertz, A., Beckhove, P.,\\nCheon, E.C., Krantz, S.B., Strouch, M., Weitz,\\nJ., Koch, M., Halverson, A.L., Bentrem, D.J.,\\nll\\nOPEN ACCESS\\n16 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1bfdb8f5-99cf-4f81-90ff-6182b7e11cde', embedding=None, metadata={'page_label': '18', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='and Khazaie, K. (2010). In colorectal cancer\\nmast cells contribute to systemic regulatory\\nT-cell dysfunction. Proc. Natl. Acad. Sci. USA\\n107, 6430–6435. https://doi.org/10.1073/\\npnas.0913683107.\\n32. Saadalla, A., Lima, M.M., Tsai, F., Osman, A.,\\nSingh, M.P., Linden, D.R., Dennis, K.L.,\\nHaeryfar, S.M.M., Gurish, M.F., Gounari, F.,\\nand Khazaie, K. (2019). Cell Intrinsic\\nDeregulated ss-Catenin Signaling Promotes\\nExpansion of Bone Marrow Derived\\nConnective Tissue Type Mast Cells, Systemic\\nInﬂammation, and Colon Cancer. Front.\\nImmunol. 10, 2777. https://doi.org/10.3389/\\nﬁmmu.2019.02777.\\n33. Palla, G., Spitzer, H., Klein, M., Fischer, D.,\\nSchaar, A.C., Kuemmerle, L.B., Rybakov, S.,\\nIbarra, I.L., Holmberg, O., Virshup, I., et al.\\n(2022). Squidpy: a scalable framework for\\nspatial omics analysis. Nat. Methods19,\\n171–178. https://doi.org/10.1038/s41592-\\n021-01358-2.\\n34. Ridge, S.M., Sullivan, F.J., and Glynn, S.A.\\n(2017). Mesenchymal stem cells: key players in\\ncancer progression. Mol. Cancer 16, 31.\\nhttps://doi.org/10.1186/s12943-017-0597-8.\\n35. Jung, Y., Kim, J.K., Shiozawa, Y., Wang, J.,\\nMishra, A., Joseph, J., Berry, J.E., McGee, S.,\\nLee, E., Sun, H., et al. (2013). Recruitment of\\nmesenchymal stem cells into prostate\\ntumours promotes metastasis. Nat.\\nCommun. 4, 1795. https://doi.org/10.1038/\\nncomms2766.\\n36. Lai, K.P., Yamashita, S., Huang, C.K., Yeh, S.,\\nand Chang, C. (2012). Loss of stromal\\nandrogen receptor leads to suppressed\\nprostate tumourigenesis via modulation of\\npro-inﬂammatory cytokines/chemokines.\\nEMBO Mol. Med. 4, 791–807. https://doi.org/\\n10.1002/emmm.201101140.\\n37. Pederzoli, F., Raffo, M., Pakula, H., Ravera, F.,\\nNuzzo, P.V., and Loda, M. (2023). Stromal cells\\nin prostate cancer pathobiology: friends or\\nfoes? Br. J. Cancer 128, 930–939. https://doi.\\norg/10.1038/s41416-022-02085-x.\\n38. Tang, Q., Cheng, B., Dai, R., and Wang, R.\\n(2021). The Role of Androgen Receptor in\\nCross Talk Between Stromal Cells and\\nProstate Cancer Epithelial Cells. Front. Cell\\nDev. Biol. 9, 729498. https://doi.org/10.3389/\\nfcell.2021.729498.\\n39. Vidotto, T., Saggioro, F.P., Jamaspishvili, T.,\\nChesca, D.L., Picanc¸o de Albuquerque, C.G.,\\nReis, R.B., Graham, C.H., Berman, D.M.,\\nSiemens, D.R., Squire, J.A., and Koti, M.\\n(2019). PTEN-deﬁcient prostate cancer is\\nassociated with an immunosuppressive\\ntumor microenvironment mediated by\\nincreased expression of IDO1 and inﬁltrating\\nFoxP3+ T regulatory cells. Prostate 79,\\n969–979. https://doi.org/10.1002/pros.\\n23808.\\n40. Bravo Gonzalez-Blas, C., Minnoye, L.,\\nPapasokrati, D., Aibar, S., Hulselmans, G.,\\nChristiaens, V., Davie, K., Wouters, J., and\\nAerts, S. (2019). cisTopic: cis-regulatory topic\\nmodeling on single-cell ATAC-seq data. Nat.\\nMethods 16, 397–400. https://doi.org/10.\\n1038/s41592-019-0367-1.\\n41. Fang, R., Preissl, S., Li, Y., Hou, X., Lucero, J.,\\nWang, X., Motamedi, A., Shiau, A.K., Zhou, X.,\\nXie, F., et al. (2021). Comprehensive analysis\\nof single cell ATAC-seq data with SnapATAC.\\nNat. Commun. 12, 1337. https://doi.org/10.\\n1038/s41467-021-21583-9.\\n42. McLean, C.Y., Bristor, D., Hiller, M., Clarke,\\nS.L., Schaar, B.T., Lowe, C.B., Wenger, A.M.,\\nand Bejerano, G. (2010). GREAT improves\\nfunctional interpretation of cis-regulatory\\nregions. Nat. Biotechnol. 28, 495–501.\\nhttps://doi.org/10.1038/nbt.1630.\\n43. Zheng, R., Wan, C., Mei, S., Qin, Q., Wu, Q.,\\nSun, H., Chen, C.H., Brown, M., Zhang, X.,\\nMeyer, C.A., and Liu, X.S. (2019). Cistrome\\nData Browser: expanded datasets and new\\ntools for gene regulatory analysis. Nucleic\\nAcids Res. 47, D729–D735. https://doi.org/\\n10.1093/nar/gky1094.\\n44. Zhang, Y., Guan, X.Y., and Jiang, P. (2020).\\nCytokine and Chemokine Signals of T-Cell\\nExclusion in Tumors. Front. Immunol. 11,\\n594609. https://doi.org/10.3389/ﬁmmu.2020.\\n594609.\\n45. Togashi, Y., Shitara, K., and Nishikawa, H.\\n(2019). Regulatory T cells in cancer\\nimmunosuppression - implications for\\nanticancer therapy. Nat. Rev. Clin. Oncol. 16,\\n356–371. https://doi.org/10.1038/s41571-\\n019-0175-7.\\n46. Tan, H.L., Sood, A., Rahimi, H.A., Wang, W.,\\nGupta, N., Hicks, J., Mosier, S., Gocke, C.D.,\\nEpstein, J.I., Netto, G.J., et al. (2014). Rb loss\\nis characteristic of prostatic small cell\\nneuroendocrine carcinoma. Clin. Cancer Res.\\n20, 890–903. https://doi.org/10.1158/1078-\\n0432.CCR-13-1982.\\n47. Becerra-Diaz, M., Strickland, A.B., Keselman,\\nA., and Heller, N.M. (2018). Androgen and\\nAndrogen Receptor as Enhancers of M2\\nMacrophage Polarization in Allergic Lung\\nInﬂammation. J. Immunol. 201, 2923–2933.\\nhttps://doi.org/10.4049/jimmunol.1800352.\\n48. Stultz, J., and Fong, L. (2021). How to turn up\\nthe heat on the cold immune\\nmicroenvironment of metastatic prostate\\ncancer. Prostate Cancer Prostatic Dis. 24,\\n697–717. https://doi.org/10.1038/s41391-\\n021-00340-5.\\n49. Sfanos, K.S. (2022). Immune cell inﬁltrates and\\nprognosis in localized prostate\\ncancer(dagger). J. Pathol.256, 135–138.\\nhttps://doi.org/10.1002/path.5817.\\n50. Fleischmann, A., Schlomm, T., Ko ¨ llermann, J.,\\nSekulic, N., Huland, H., Mirlacher, M., Sauter,\\nG., Simon, R., and Erbersdobler, A. (2009).\\nImmunological microenvironment in prostate\\ncancer: high mast cell densities are\\nassociated with favorable tumor\\ncharacteristics and good prognosis. Prostate\\n69, 976–981. https://doi.org/10.1002/pros.\\n20948.\\n51. Flammiger, A., Bayer, F., Cirugeda-Ku ¨ hnert,\\nA., Huland, H., Tennstedt, P., Simon, R.,\\nMinner, S., Bokemeyer, C., Sauter, G.,\\nSchlomm, T., and Trepel, M. (2012).\\nIntratumoral T but not B lymphocytes are\\nrelated to clinical outcome in prostate cancer.\\nAPMIS 120, 901–908. https://doi.org/10.\\n1111/j.1600-0463.2012.02924.x.\\n52. Zhao, S.G., Lehrer, J., Chang, S.L., Das, R.,\\nErho, N., Liu, Y., Sjo ¨ stro¨ m, M., Den, R.B.,\\nFreedland, S.J., Klein, E.A., et al. (2019). The\\nImmune Landscape of Prostate Cancer and\\nNomination of PD-L2 as a Potential\\nTherapeutic Target. J. Natl. Cancer Inst. 111,\\n301–310. https://doi.org/10.1093/jnci/djy141.\\n53. Andersen, L.B., Nørgaard, M., Rasmussen,\\nM., Fredsøe, J., Borre, M., Ulhøi, B.P., and\\nSørensen, K.D. (2021). Immune cell analyses\\nof the tumor microenvironment in prostate\\ncancer highlight inﬁltrating regulatory T cells\\nand macrophages as adverse prognostic\\nfactors. J. Pathol. 255, 155–165. https://doi.\\norg/10.1002/path.5757.\\n54. Kaur, H.B., Guedes, L.B., Lu, J., Maldonado,\\nL., Reitz, L., Barber, J.R., De Marzo, A.M.,\\nTosoian, J.J., Tomlins, S.A., Schaeffer, E.M.,\\net al. (2018). Association of tumor-inﬁltrating\\nT-cell density with molecular subtype, racial\\nancestry and clinical outcomes in prostate\\ncancer. Mod. Pathol. 31, 1539–1552. https://\\ndoi.org/10.1038/s41379-018-0083-x.\\n55. Davidsson, S., Ohlson, A.L., Andersson, S.O.,\\nFall, K., Meisner, A., Fiorentino, M., Andre ´ n,\\nO., and Rider, J.R. (2013). CD4 helper T cells,\\nCD8 cytotoxic T cells, and FOXP3(+)\\nregulatory T cells with respect to lethal\\nprostate cancer. Mod. Pathol. 26, 448–455.\\nhttps://doi.org/10.1038/modpathol.\\n2012.164.\\n56. Hempel Sullivan, H., Heaphy, C.M., Kulac, I.,\\nCuka, N., Lu, J., Barber, J.R., De Marzo, A.M.,\\nLotan, T.L., Joshu, C.E., and Sfanos, K.S.\\n(2020). High Extratumoral Mast Cell Counts\\nAre Associated with a Higher Risk of Adverse\\nProstate Cancer Outcomes. Cancer\\nEpidemiol. Biomarkers Prev. 29, 668–675.\\nhttps://doi.org/10.1158/1055-9965.EPI-\\n19-0962.\\n57. Hempel, H.A., Cuka, N.S., Kulac, I., Barber,\\nJ.R., Cornish, T.C., Platz, E.A., De Marzo,\\nA.M., and Sfanos, K.S. (2017). Low\\nIntratumoral Mast Cells Are Associated With\\na Higher Risk of Prostate Cancer Recurrence.\\nProstate 77, 412–424. https://doi.org/10.\\n1002/pros.23280.\\n58. Khazaie, K., Blatner, N.R., Khan, M.W.,\\nGounari, F., Gounaris, E., Dennis, K., Bonertz,\\nA., Tsai, F.N., Strouch, M.J., Cheon, E., et al.\\n(2011). The signiﬁcant role of mast cells in\\ncancer. Cancer Metastasis Rev. 30, 45–60.\\nhttps://doi.org/10.1007/s10555-011-9286-z.\\n59. Valdman, A., Jaraj, S.J., Compe ´ rat, E.,\\nCharlotte, F., Roupret, M., Pisa, P., and\\nEgevad, L. (2010). Distribution of Foxp3-CD4-\\nand CD8-positive lymphocytic cells in benign\\nand malignant prostate tissue. APMIS 118,\\n360–365. https://doi.org/10.1111/j.1600-\\n0463.2010.02604.x.\\n60. Hussein, M.R.A., Al-Assiri, M., and Musalam,\\nA.O. (2009). Phenotypic characterization of\\nthe inﬁltrating immune cells in normal\\nprostate, benign nodular prostatic\\nhyperplasia and prostatic adenocarcinoma.\\nExp. Mol. Pathol. 86, 108–113. https://doi.\\norg/10.1016/j.yexmp.2008.11.010.\\n61. Fujii, T., Shimada, K., Asai, O., Tanaka, N.,\\nFujimoto, K., Hirao, K., and Konishi, N. (2013).\\nImmunohistochemical analysis of\\ninﬂammatory cells in benign and\\nprecancerous lesions and carcinoma of the\\nprostate. Pathobiology 80, 119–126. https://\\ndoi.org/10.1159/000342396.\\n62. Nardone, V., Botta, C., Caraglia, M., Martino,\\nE.C., Ambrosio, M.R., Carfagno, T., Tini, P.,\\nSemeraro, L., Misso, G., Grimaldi, A., et al.\\n(2016). Tumor inﬁltrating T lymphocytes\\nexpressing FoxP3, CCR7 or PD-1 predict the\\noutcome of prostate cancer patients\\nsubjected to salvage radiotherapy after\\nbiochemical relapse. Cancer Biol. Ther.17,\\n1213–1220. https://doi.org/10.1080/\\n15384047.2016.1235666.\\n63. Watanabe, M., Kanao, K., Suzuki, S.,\\nMuramatsu, H., Morinaga, S., Kajikawa, K.,\\nKobayashi, I., Nishikawa, G., Kato, Y.,\\nZennami, K., et al. (2019). Increased\\ninﬁltration of CCR4-positive regulatory T cells\\nin prostate cancer tissue is associated with a\\npoor prognosis. Prostate 79, 1658–1665.\\nhttps://doi.org/10.1002/pros.23890.\\n64. Lanciotti, M., Masieri, L., Raspollini, M.R.,\\nMinervini, A., Mari, A., Comito, G., Giannoni,\\nE., Carini, M., Chiarugi, P., and Serni, S. (2014).\\nThe role of M1 and M2 macrophages in\\nprostate cancer in relation to extracapsular\\ntumor extension and biochemical recurrence\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 17\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='82ce0606-7e9c-4ff2-a06d-7c6270e3a8a3', embedding=None, metadata={'page_label': '19', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='after radical prostatectomy. BioMed Res. Int.\\n2014, 486798. https://doi.org/10.1155/2014/\\n486798.\\n65. Erlandsson, A., Carlsson, J., Lundholm, M.,\\nFa¨ lt, A., Andersson, S.O., Andre ´ n, O., and\\nDavidsson, S. (2019). M2 macrophages and\\nregulatory T cells in lethal prostate cancer.\\nProstate 79, 363–369. https://doi.org/10.\\n1002/pros.23742.\\n66. Sadasivan, S.M., Chen, Y., Gupta, N.S., Han,\\nX., Bobbitt, K.R., Chitale, D.A., Williamson,\\nS.R., Rundle, A.G., Tang, D., and Rybicki, B.A.\\n(2020). The interplay of growth differentiation\\nfactor 15 (GDF15) expression and M2\\nmacrophages during prostate\\ncarcinogenesis. Carcinogenesis 41, 1074–\\n1082. https://doi.org/10.1093/carcin/\\nbgaa065.\\n67. Baker, E.A.G., Schapiro, D., Dumitrascu, B.,\\nVickovic, S., and Regev, A. (2023). In silico\\ntissue generation and power analysis for\\nspatial omics. Nat. Methods 20, 424–431.\\nhttps://doi.org/10.1038/s41592-023-01766-6.\\n68. Comito, G., Giannoni, E., Segura, C.P.,\\nBarcellos-de-Souza, P., Raspollini, M.R.,\\nBaroni, G., Lanciotti, M., Serni, S., and\\nChiarugi, P. (2014). Cancer-associated\\nﬁbroblasts and M2-polarized macrophages\\nsynergize during prostate carcinoma\\nprogression. Oncogene 33, 2423–2431.\\nhttps://doi.org/10.1038/onc.2013.191.\\n69. Allavena, P., Digiﬁco, E., and Belgiovine, C.\\n(2021). Macrophages and cancer stem cells: a\\nmalevolent alliance. Mol. Med. 27, 121.\\nhttps://doi.org/10.1186/s10020-021-00383-3.\\n70. Zhao, Y.B., Yang, S.H., Shen, J., Deng, K., Li,\\nQ., Wang, Y., Cui, W., and Ye, H. (2020).\\nInteraction between regulatory T cells and\\nmast cells via IL-9 and TGF-beta production.\\nOncol. Lett. 20, 360. https://doi.org/10.3892/\\nol.2020.12224.\\n71. Blatner, N.R., Mulcahy, M.F., Dennis, K.L.,\\nScholtens, D., Bentrem, D.J., Phillips, J.D.,\\nHam, S., Sandall, B.P., Khan, M.W., Mahvi,\\nD.M., et al. (2012). Expression of RORgt Marks\\na Pathogenic T Cell Subset in Human Colon\\nCancer. Sci. Transl. Med. 4, 164ra159 .\\n72. Hempel Sullivan, H., Maynard, J.P., Heaphy,\\nC.M., Lu, J., De Marzo, A.M., Lotan, T.L.,\\nJoshu, C.E., and Sfanos, K.S. (2021).\\nDifferential mast cell phenotypes in benign\\nversus cancer tissues and prostate cancer\\noncologic outcomes. J. Pathol. 253, 415–426.\\nhttps://doi.org/10.1002/path.5606.\\n73. Takano, H., Nakazawa, S., Shirata, N., Tamba,\\nS., Furuta, K., Tsuchiya, S., Morimoto, K.,\\nItano, N., Irie, A., Ichikawa, A., et al. (2009).\\nInvolvement of CD44 in mast cell proliferation\\nduring terminal differentiation. Lab. Invest.\\n89, 446–455. https://doi.org/10.1038/\\nlabinvest.2008.159.\\n74. Mueller, N., Wicklein, D., Eisenwort, G.,\\nJawhar, M., Berger, D., Stefanzl, G., Greiner,\\nG., Boehm, A., Kornauth, C., Muellauer, L.,\\net al. (2018). CD44 is a RAS/STAT5-regulated\\ninvasion receptor that triggers disease\\nexpansion in advanced mastocytosis. Blood\\n132, 1936–1950 .\\n75. Guan, X., Polesso, F., Wang, C., Sehrawat, A.,\\nHawkins, R.M., Murray, S.E., Thomas, G.V.,\\nCaruso, B., Thompson, R.F., Wood, M.A.,\\net al. (2022). Androgen receptor activity in\\nT cells limits checkpoint blockade efﬁcacy.\\nNature606, 791–796. https://doi.org/10.\\n1038/s41586-022-04522-6.\\n76. Chen, W., Beck, I., Schober, W., Brockow, K.,\\nEffner, R., Buters, J.T.M., Behrendt, H., and\\nRing, J. (2010). Human mast cells express\\nandrogen receptors but treatment with\\ntestosterone exerts no inﬂuence on IgE-\\nindependent mast cell degranulation elicited\\nby neuromuscular blocking agents. Exp.\\nDermatol. 19, 302–304. https://doi.org/10.\\n1111/j.1600-0625.2009.00969.x.\\n77. Jia, J., Zhang, H., Zhang, H., Liu, W., Du, H.,\\nShu, M., and He, L. (2019). AR facilitates YAP-\\nTEAD interaction with the AM promoter to\\nenhance mast cell inﬁltration into cutaneous\\nneuroﬁbroma. Sci. Rep. 9, 19346. https://doi.\\norg/10.1038/s41598-019-56022-9.\\n78. Dang, Q., Li, L., Xie, H., He, D., Chen, J., Song,\\nW., Chang, L.S., Chang, H.C., Yeh, S., and\\nChang, C. (2015). Anti-androgen\\nenzalutamide enhances prostate cancer\\nneuroendocrine (NE) differentiation via\\naltering the inﬁltrated mast cells–> androgen\\nreceptor (AR)–> miRNA32 signals. Mol.\\nOncol. 9, 1241–1251. https://doi.org/10.\\n1016/j.molonc.2015.02.010.\\n79. Kron, K.J., Murison, A., Zhou, S., Huang, V.,\\nYamaguchi, T.N., Shiah, Y.J., Fraser, M., van\\nder Kwast, T., Boutros, P.C., Bristow, R.G.,\\nand Lupien, M. (2017). TMPRSS2-ERG fusion\\nco-opts master transcription factors and\\nactivates NOTCH signaling in primary\\nprostate cancer. Nat. Genet. 49, 1336–1345.\\nhttps://doi.org/10.1038/ng.3930.\\n80. Rajaram, S., Heinrich, L.E., Gordan, J.D.,\\nAvva, J., Bonness, K.M., Witkiewicz, A.K.,\\nMalter, J.S., Atreya, C.E., Warren, R.S., Wu,\\nL.F., and Altschuler, S.J. (2017). Sampling\\nstrategies to capture single-cell\\nheterogeneity. Nat. Methods 14, 967–970.\\nhttps://doi.org/10.1038/nmeth.4427.\\n81. Liu, Y., Wang, J., Horton, C., Yu, C., Knudsen,\\nB., Stefanson, J., Hu, K., Stefanson, O., Green,\\nJ., Guo, C., et al. (2022). Stromal AR inhibits\\nprostate tumor progression by restraining\\nsecretory luminal epithelial cells. Cell Rep. 39,\\n110848. https://doi.org/10.1016/j.celrep.\\n2022.110848.\\n82. Malinen, M., Niskanen, E.A., Kaikkonen, M.U.,\\nand Palvimo, J.J. (2017). Crosstalk between\\nandrogen and pro-inﬂammatory signaling\\nremodels androgen receptor and NF-kappaB\\ncistrome to reprogram the prostate cancer\\ncell transcriptome. Nucleic Acids Res. 45,\\n619–630. https://doi.org/10.1093/nar/\\ngkw855.\\n83. Zhang, L., Altuwaijri, S., Deng, F., Chen, L.,\\nLal, P., Bhanot, U.K., Korets, R., Wenske, S.,\\nLilja, H.G., Chang, C., et al. (2009). NF-kappaB\\nregulates androgen receptor expression and\\nprostate cancer growth. Am. J. Pathol. 175,\\n489–499. https://doi.org/10.2353/ajpath.\\n2009.080727.\\n84. Nirmal, A.J., Maliga, Z., Vallius, T., Quattrochi,\\nB., Chen, A.A., Jacobson, C.A., Pelletier, R.J.,\\nYapp, C., Arias-Camison, R., Chen, Y.A., et al.\\n(2022). The Spatial Landscape of Progression\\nand Immunoediting in Primary Melanoma at\\nSingle-Cell Resolution. Cancer Discov. 12,\\n1518–1541. https://doi.org/10.1158/2159-\\n8290.CD-21-1357.\\n85. Pourmaleki, M., Jones, C.J., Ariyan, C.E.,\\nZeng, Z., Pirun, M., Navarrete, D.A., Li, Y.,\\nZhang, M., Nandakumar, S., Campos, C.,\\net al. (2022). Tumor MHC Class I Expression\\nAssociates with Intralesional IL2 Response in\\nMelanoma. Cancer Immunol. Res. 10,\\n303–313. https://doi.org/10.1158/2326-6066.\\nCIR-21-1083.\\n86. Song, H., Bucher, S., Rosenberg, K., Tsui, M.,\\nBurhan, D., Hoffman, D., Cho, S.J.,\\nRangaswami, A., Breese, M., Leung, S., et al.\\n(2022). Single-cell analysis of hepatoblastoma\\nidentiﬁes tumor signatures that predict\\nchemotherapy susceptibility using patient-\\nspeciﬁc tumor spheroids. Nat. Commun. 13,\\n4878. https://doi.org/10.1038/s41467-022-\\n32473-z.\\n87. Cooperberg, M.R., Pasta, D.J., Elkin, E.P.,\\nLitwin, M.S., Latini, D.M., Du Chane, J., and\\nCarroll, P.R. (2005). San Francisco Cancer of\\nthe Prostate Risk Assessment score: a\\nstraightforward and reliable preoperative\\npredictor of disease recurrence after radical\\nprostatectomy. J. Urol. 173, 1938–1942.\\nhttps://doi.org/10.1097/01.ju.0000158155.\\n33890.e7.\\n88. Lin, J.R., Izar, B., Wang, S., Yapp, C., Mei, S.,\\nShah, P.M., Santagata, S., and Sorger, P.K.\\n(2018). Highly multiplexed\\nimmunoﬂuorescence imaging of human\\ntissues and tumors using t-CyCIF and\\nconventional optical microscopes. Elife 7,\\ne31657.\\n89. Eng, J., Bucher, E., Hu, Z., Zheng, T., Gibbs,\\nS.L., Chin, K., and Gray, J.W. (2022). A\\nframework for multiplex imaging\\noptimization and reproducible analysis.\\nCommun. Biol. 5, 438. https://doi.org/10.\\n1038/s42003-022-03368-y.\\n90. Adalsteinsson, V.A., Ha, G., Freeman, S.S.,\\nChoudhury, A.D., Stover, D.G., Parsons, H.A.,\\nGydush, G., Reed, S.C., Rotem, D., Rhoades,\\nJ., et al. (2017). Scalable whole-exome\\nsequencing of cell-free DNA reveals high\\nconcordance with metastatic tumors. Nat.\\nCommun. 8, 1324. https://doi.org/10.1038/\\ns41467-017-00965-y.\\n91. Cusanovich, D.A., Daza, R., Adey, A., Pliner,\\nH.A., Christiansen, L., Gunderson, K.L.,\\nSteemers, F.J., Trapnell, C., and Shendure, J.\\n(2015). Multiplex single-cell proﬁling of\\nchromatin accessibility by combinatorial\\ncellular indexing. Science 348, 910–914.\\nhttps://doi.org/10.1126/science.aab1601.\\n92. Buenrostro, J.D., Wu, B., Litzenburger, U.M.,\\nRuff, D., Gonzales, M.L., Snyder, M.P., Chang,\\nH.Y., and Greenleaf, W.J. (2015). Single-cell\\nchromatin accessibility reveals principles of\\nregulatory variation. Nature523, 486–490.\\nhttps://doi.org/10.1038/nature14590.\\n93. Chang, Y.H., Chin, K., Thibault, G., Eng, J.,\\nBurlingame, E., and Gray, J.W. (2020).\\nRESTORE: Robust intEnSiTy nORmalization\\nmEthod for multiplexed imaging. Commun.\\nBiol. 3, 111. https://doi.org/10.1038/s42003-\\n020-0828-1.\\n94. Jackson, H.W., Fischer, J.R., Zanotelli, V.R.T.,\\nAli, H.R., Mechera, R., Soysal, S.D., Moch, H.,\\nMuenst, S., Varga, Z., Weber, W.P., and\\nBodenmiller, B. (2020). The single-cell\\npathology landscape of breast cancer.\\nNature 578, 615–620. https://doi.org/10.\\n1038/s41586-019-1876-x.\\n95. Vallat, R. (2018). Pingouin: statistics in Python.\\nJ. Open Source Softw. 3, 1026. https://doi.\\norg/10.21105/joss.01026.\\n96. Goltsev, Y., Samusik, N., Kennedy-Darling, J.,\\nBhate, S., Hale, M., Vazquez, G., Black, S., and\\nNolan, G.P. (2018). Deep Proﬁling of Mouse\\nSplenic Architecture with CODEX\\nMultiplexed Imaging. Cell 174, 968–981.e15.\\nhttps://doi.org/10.1016/j.cell.2018.07.010.\\nll\\nOPEN ACCESS\\n18 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d29de2ab-0189-435d-bcb3-e9913f5da3c1', embedding=None, metadata={'page_label': '20', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='STAR+METHODS\\nKEY RESOURCES TABLE\\nREAGENT or RESOURCE SOURCE IDENTIFIER\\nAntibodies\\nalphaSMA (1A4) Santa Cruz Biotechnology Cat# sc-32251; RRID:AB_262054\\nAMACR (Polyclonal) Abcam Cat# ab212378; RRID:AB_3096027\\nAR_(CST) (D6F11) Cell Signaling Technology Cat# 8956S; RRID:AB_11129223\\nCD11b_555 (EPR1344) Abcam Cat# ab206616; RRID:AB_2728809\\nCD163_647 (EPR14643-36) Abcam Cat# ab218294; RRID:AB_2943126\\nCD20 (EP459Y) Abcam Cat# ab198943; RRID:AB_2905499\\nCD3_750 (EP449E) Abcam Cat# ab213608; RRID:AB_3096030\\nCD4_647 (EPR6855) Abcam Cat# ab196147; RRID:AB_2923526\\nCD44_555 (EPR1013Y) Abcam Cat# ab216647; RRID:AB_3096031\\nCD45_750 (EP322Y) Abcam Cat# ab214437; RRID:AB_3096032\\nCD66b_750 (G10F5) Fisher Scientiﬁc Cat# NBP2-80664; RRID:AB_3096017\\nCD68_750 (KP1) BioLegend Cat# 916104; RRID:AB_2616797\\nCD90 (EPR3133) Abcam Cat# ab216449; RRID:AB_2889264\\nChromA_750 (EP1030Y) Abcam Cat# ab215276; RRID:AB_3096034\\nCK5_488 (EP1601Y) Abcam Cat# ab193894; RRID:AB_2893023\\nCK8 (EP1628Y) Abcam Cat# ab192468; RRID:AB_2890258\\nECAD (EP700Y) Abcam Cat# ab201499; RRID:AB_2801591\\nFOXP3 (PCH101) Thermo Scientiﬁc Cat# 14-4776-82; RRID:AB_467554\\nGZMB 750 (EPR20129-217) Abcam Cat# ab219803; RRID:AB_2910576\\nH3K27ac (EP16602) Abcam Cat# ab245911; RRID:AB_3096035\\nH3K4me3 (C42D8) Cell Signaling Technology Cat# 11960S; RRID:AB_2797779\\nHIF-1 alpha (ESEE122) Novus Biologicals Cat#NB100-131; RRID:AB_350068\\nHLA DRB1 (EPR6148) Abcam Cat# ab207066; RRID:AB_3096026\\nKi67_CST (D3B5) Cell Signaling Technology Cat# 12075S; RRID:AB_3096029\\nNCAM (EP2567Y) Abcam Cat# ab215981; RRID:AB_3096036\\nNF-kB (D14E12) Cell Signaling Technology Cat#8242S\\nNKX3.1 (3 9) Thermo Fisher Scientiﬁc Cat# 35–9700; RRID:AB_2533232\\nPD1 (EPR4877(2)) Abcam Cat# ab201825; RRID:AB_2728811\\nTryp (AA1) Abcam Cat# ab2378; RRID:AB_303023\\nVIM (EPR3776) Abcam Cat# ab194719; RRID:AB_2909595\\nBiological samples\\nTissue sections from patients OHSU’s Biolibrary IRB#4918\\nChemicals, peptides, and recombinant proteins\\nAlexaFluor (AF) 488, AF555, AF647 and AF750 dyes ThermoFisher Scientiﬁc A20006\\nAlexaFluor NHS ester reagents ThermoFisher Scientiﬁc A20006 and A20000\\nAmicon Ultra spin columns Sigma-Aldrich UFC5010\\nAmpure XP Beads Beckman Coulter A63880\\nCitrate, pH 6 Sigma-Aldrich C-1909\\nIGEPAL CA-630 Sigma-Aldrich I8896-100ML\\nTarget Retrieval Solution, pH 9 Agilent S2367\\n(Continued on next page)\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 19\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a3736479-92d9-4420-b288-934981df3c46', embedding=None, metadata={'page_label': '21', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Continued\\nREAGENT or RESOURCE SOURCE IDENTIFIER\\nPBS, pH 7.4 Fisher Scientiﬁc BP39920\\nNormal Goat Serum (NGS) Fisher Scientiﬁc S-1000\\nBovine Serum Albumin (BSA) ThermoFisher Scientiﬁc AM2616\\nSlowfade Gold DAPI ThermoFisher Scientiﬁc S36938\\n#1.5 thickness coverslip Corning Life Sciences 2980-243 and 2980-24\\nProtease inhibitor tablet Roche Cat. 11873580001\\nKapa HiFi HotStart ReadyMix Fisher Scientiﬁc KK2602\\nBioanalyzer HS dsDNA Kit Agilent 5067–4626\\nQiaQuick PCR puriﬁcation kit Qiagen 28104\\nDeposited data\\nRaw and analyzed data This paper Figshare+ DOI: https://ﬁgshare.com/\\ns/116766067fa229670e17\\nSoftware and algorithms\\nImage Preprocessing – conversion to TIFF Zeiss Zen Blue Microscopy Software\\nImage Preprocessing Quantitative Imaging\\nSystems, LLC\\nQiTissue software\\nNuclei and Cell Segmentation CellPose https://github.com/MouseLand/cellpose\\nSignal Intensity Normalization RESTORE https://gitlab.com/Chang_Lab/cycif_int_norm\\nCell typing with GPU accelerated PhenoGraph Grapheno https://gitlab.com/eburling/grapheno\\nHeatmap visualization https://gitlab.com/eburling/BCTMA/-/\\nblob/master/pyviz_ﬁgs.py\\nCluster Validation https://github.com/zeynepsayar/PCa_\\nimmune/blob/main/mapping.m\\nTissue Area Calculation NIH, Bethesda, Maryland ImageJ software\\nCellular Neighborhood Schu ¨ rch et al.10 https://github.com/nolanlab/\\nNeighborhoodCoordination\\nsquidpy Palla et al. 33 https://github.com/scverse/squidpy\\nBWA-MEM https://github.com/lh3/bwa\\nIchorCNA https://github.com/broadinstitute/ichorCNA\\ncistrome http://cistrome.org/\\nchromVAR https://bioconductor.org/packages/release/\\nbioc/html/chromVAR.html\\nmotifmatchr https://bioconductor.org/packages/release/\\nbioc/html/motifmatchr.html\\nCreating In Silico Tissues and Power Analysis This paper Github: https://github.com/cigdemak/cycIF_prostate\\nZenodo: https://doi.org/10.5281/zenodo.10333970\\nsnapATAC Fang et al. 41 https://github.com/r3fang/SnapATAC\\nOther\\nSingle-cell ATAC sequencing data Eksi et al. 17 GEO: GSE171559\\nJASPAR database https://jaspar2020.genereg.net/\\nCistrome Liu et al. 81 http://cistrome.org/\\nTintoRetriever Pressure Cooker BioSB\\nAxio Scan.Z1 slide scanner Zeiss\\nColibri7 LED light source Zeiss\\nOrca Flash 4.0 V3 CMOS camera Hamamatsu Photonics\\nPlan-apochromat 20x/0.8 objective Zeiss\\n(Continued on next page)\\nll\\nOPEN ACCESS\\n20 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='cf2733d5-c630-4398-b82d-683becdf65eb', embedding=None, metadata={'page_label': '22', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='RESOURCE AVAILABILITY\\nLead contact\\nFurther information and requests for resources and reagents should be directed to and will be fulﬁlled by the lead contact, Ece Eksi ( eksi@\\nohsu.edu).\\nMaterials availability\\nThis study did not generate any new reagents.\\nData and code availability\\n/C15 Imaging data has been deposited to Figshare+ ( https://doi.org/10.25452/ﬁgshare.plus.25199468) and is publicly available as the date\\nof the publication. All cycIF data deposited follows the multiplex imaging data standards published by the Human Tumor Atlas\\nNetwork (HTAN): https://humantumoratlas.org/standard/imaging. This paper analyzes existing, publicly available single-cell ATAC\\nsequencing data (GSE171559). The accession number for the dataset is listed in the key resources table . Any additional information\\nrequired to analyze the data reported in this work is available from the lead contact upon request.\\n/C15 Custom codes have been deposited on Github and Zenodo (Github: https://github.com/cigdemak/cycIF_prostate, https://github.\\ncom/zeynepsayar/PCa_immune; Zenodo: https://doi.org/10.5281/zenodo.10333970) and are publicly available as of the date of the\\npublication.\\n/C15 Any additional information required to reanalyze the data reported in this work is available from the lead contact upon request.\\nEXPERIMENTAL MODEL AND STUDY PARTICIPANT DETAILS\\nHuman subjects\\nThirteen white male patients, aged between 46 and 81, with treatment-naı ¨ve, localized prostate cancer were identiﬁed from the OHSU’s Bio-\\nlibrary using medical records and reviews of H&E images. Tissue samples were obtained from men undergoing radical prostatectomy with\\ninformed consent through Biolibrary under IRB#4918. Participants provided samples voluntarily without compensation. Our research adheres\\nto all pertinent ethical guidelines set forth by the Knight Cancer Institute at OHSU. Five unstained 5-micron FFPE adjacent tissue sections and\\nan adjacent H&E stained tissue section from each patient were obtained. H&E images were reviewed by a pathologist and a Gleason score\\nwas assigned to each patient. Patients were classiﬁed into risk groups based on The Cancer of the Prostate Risk Assessment Post-Surgical\\n(CAPRA-S) scoring system (Figures S3 and S4).87\\nMETHOD DETAILS\\nCyclic IF antibody panel\\nAntibodies used in the study can be found under the Antibodies section in the key resources table and are also listed in Data S1. Validation of\\nantibodies for cycIF to mark cell types in the prostate tumor microenvironment is shown step by step in Figure S1 . Tissue sections used in\\nantibody validations are shown in Data S2. Antibodies were either obtained as commercially conjugated to ﬂuorophores or in a BSA and Azide\\nfree buffer and conjugated to AlexaFluor (AF) 488, AF555, AF647, and AF750 dyes in-house (ThermoFisher Scientiﬁc, A20006) [21].\\nFluorophore conjugation to antibodies\\nAntibodies were either obtained as commercially conjugated to ﬂuorophores or in a BSA and Azide free buffer and conjugated to AlexaFluor\\n(AF) 488, AF555, AF647, and AF750 dyes in-house. AlexaFluor NHS ester reagents (ThermoFisher Scientiﬁc, A20006, and A20000) were dis-\\nsolved in DMSO to a ﬁnal concentration of 10 mM. Antibodies were reacted with ﬂuorophore dyes at a 1:10 antibody-to-dye molar ratio in\\n0.1 M sodium bicarbonate buffer of pH 8.3. Conjugation reactions were carried out on a rocker for 2 h at room temperature in the dark. A\\nbuffer exchange step using 10 kDa Amicon Ultra spin columns (Sigma-Aldrich, UFC5010) was performed to remove the excess Alexa Fluor\\ndye from conjugation reactions.\\nContinued\\nREAGENT or RESOURCE SOURCE IDENTIFIER\\nSonicator for DNA fragmentation Covaris E220 Covaris Inc\\n2100 Bioanalzyser system Agilent Technologies\\nNextSeq 500 Sequencer Illumina\\nBD FacsAria Fusion cell sorter FACSDiva v8.0.3 BD\\nQuantStudio v1.7.1 for RT-PCR ThermoFisher Scientiﬁc\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 21\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='402e8619-642f-4039-8b06-045719355627', embedding=None, metadata={'page_label': '23', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='FFPE tissue processing\\nFFPE tissue sections were deparafﬁnized as follows 17: The parafﬁn embedding was removed from the tissue sections via an overnight incu-\\nbation at 50 /C14 C followed by a 1-h incubation at 65 /C14 C. Slides were immediately transferred into Xylene and sequentially immersed in a fresh\\nXylene solution two times, 5 min each; 100% ethanol two times 5 min each; 95% ethanol two times 2 min each; 70% ethanol two times\\n2 min each and distilled water two times 5 min each. Antigen retrieval was done in a medical decloaking chamber (BioSB, TintoRetriever Pres-\\nsure Cooker). Tissue slides were placed in a Coplin jar containing 10 mM citrate buffer, pH 6 (Sigma-Aldrich, C-1909), and incubated at high\\npressure for 15 min. Each slide was then immersed in hot distilled water and transferred into 1 3 Target Retrieval Solution, pH 9 (Agilent,\\nS2367) for 15 min. Following antigen retrieval, tissues were brieﬂy washed in ddH2O, followed by 5 min in 1 3 phosphate-buffered saline\\n(PBS), pH 7.4 (Fisher, BP39920).\\nCyclic IF imaging of prostate tissues\\nCyclic IF protocol was adapted with minor revisions 88,89 as follows: Before immunoﬂuorescence staining, the slides were incubated in a\\nquenching solution (10% 10 3 PBS, 0.4% 5M NaOH, 3% H2O2) under broad spectrum light for 1 h to reduce inherent tissue autoﬂuorescence.\\nAfter quenching, tissues were blocked with 5% NGS (Fisher Scientiﬁc, S-1000) in 1% BSA (Thermo Scientiﬁc, AM2616) solution for 30 min at\\nroom temperature. For each round of cyclic IF staining, primary antibodies were diluted in 5% NGS in 1% BSA. Slides were stained overnight at\\n4/C14 C in a humid chamber. Antibodies were washed four times in 1 3 PBS for 5 min. Tissue sections were coverslipped using Slowfade Gold\\nDAPI mounting media using a #1.5 thickness coverslip (Corning Life Sciences, 2980-243 and 2980-245) and imaged using a Zeiss Axio\\nScan.Z1 slide scanner. After a successful scan was obtained, the ﬂuorophore signal was quenched as described above under broad-spectrum\\nlight for 1 h. Slides were washed 3 times in 1 3 PBS for 5 min. Quenching was conﬁrmed under a microscope. Primary antibody staining, im-\\naging, and quenching steps were repeated for 10 rounds, using up to four distinct antibodies conjugated to AF488, AF555, AF647, and AF750\\nand DAPI nuclear stain in each round. The order in which each antibody was used in the cyclic IF experiment can be found in Figure S2 B.\\nQuenched signal images were taken following rounds 3 and 10 and before round 1.\\nWhole slide scanning\\nAn Axio Scan.Z1 microscope (Zeiss, Germany) with a Colibri7 LED light source with Cy7 line (Zeiss, Germany) was used for automated high-\\nthroughput whole slide scanning. LED intensities were kept constant for all rounds of imaging, where the LED exposure times were adjusted\\nin each round. Exposure times were kept constant between all tissues in each round. Filter sets for DAPI, GFP, Cy3, Cy5, and Cy7 were used for\\n5-channel imaging in each round of cyclic IF. Images were taken with an Orca Flash 4.0 V3 CMOS camera (Hamamatsu Photonics, Japan) at 203\\nresolution using a plan-apochromat 20 3/0.8 objective (Zeiss, Germany). Image acquisition proﬁles were set using the ZEN Blue edition imag-\\ning software, where the coarse and ﬁne focus strategies and light exposure times were manually adjusted for each round of cyclic IF imaging.\\nLow-pass WGS library preparation and sequencing\\nLow-pass WGS libraries were created using the Kapa Biosystems Hyper Prep kit (KAPA Biosystems, Capetown, South Africa) using 11.8 ng/mL\\nof G4_1; 17.6 ng/mL of G4_3; 13.8 ng/mL of G4_4; G4_4 10.8 ng/mL; 13 ng/mL G4_5; 13.3 ng/mL G3_1; 9.2 ng/mL G3_2; 11.4 ng/mL G3_3; 9.6;\\nG3_5; 11 ng/mL TAN_1; 10.1 ng/mL TAN_2; 10.4 ng/mL TAN_3 DNA as input. DNA was fragmented by sonication to 150 bp using a Covaris\\nE220 (Covaris Inc., Woburn, MA, USA) and ligated to dual-indexed Illumina-compatible sequencing adaptors using KAPA Hyper Prep Re-\\nagents and protocol and then PCR ampliﬁed for 9 cycles using Illumina library ampliﬁcation primers. Ampliﬁed libraries were assessed for\\nsize distribution and concentration using the 2100 Bioanalyzer system (Agilent Technologies, Santa Clara, CA, USA) with HS dsDNA Kits. Sam-\\nples were then sequenced the Illumina NextSeq 500, paired-end 75 bp with dual 14-bp indexing cycles. Raw fastq ﬁles were demultiplexed\\nusing in-house software, aligned using BWA-MEM, and copy number alterations were called using the IchorCNA software package using\\ndefault parameters 90 (https://github.com/broadinstitute/ichorCNA).\\nQUANTIFICATION AND STATISTICAL ANALYSIS\\nSingle-cell ATAC sequencing analysis\\nWe used the single-cell ATAC sequencing data from 13 matching patients over 18 patients that was published previously. 17 Radical prostatec-\\ntomy samples from 13 patients with low- and high-risk prostate cancer were obtained as ﬂash-frozen samples and were prepared for single-cell\\nATAC sequencing using a combinatorial indexing platform. 17,91 Sequencing base calls were converted to fastq ﬁles using bcl2fastq. Reads\\nwere pre-processed using snaptools snap-pre command with default parameters and aligned to the hg19 genome using snaptools align\\npaired-end command using default parameters and bwa speciﬁed as the aligner.41 Peaks were called using runMACS command from snapA-\\nTAC R package with the following MACS2 parameters: -nomodel -shift 100 -ext 200 -qval 5e-2 -B –SPMR.41 Clusters consisting of immune or\\nnon-immune stromal cells were identiﬁed based on cluster IDs and gene expression scores. 17,41 We included all subjects and data without\\nexclusion, maintained blinding to all patient information, and determined sample sizes based on the availability of patient samples.\\nMotif analysis\\nWe used the function ‘‘getJasparMotifs’’from R package chromVar which fetches motifs from the JASPAR database. 92 The function ‘‘match-\\nMotifs’’from the motifmatchr R package ﬁnds which peaks contain which motifs. We found enriched motifs of both AR+ and AR0 cell groups\\nll\\nOPEN ACCESS\\n22 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a846e1d3-f2eb-4d83-af14-fc52cec03d6c', embedding=None, metadata={'page_label': '24', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='using their DARs. One option is the p.cutoff for determining the stringency of motif calling. The default value is set to 0.00005, which gives\\nreasonable numbers of motif matches for human motifs. The function can also return the number of motif matches per peak and the maximum\\nmotif score per peak. We calculated enrichmentp-values comparing motif scores per peak between AR+ and AR0 cell groups using the\\nKolmogorov-Smirnov test implementation of the stats library in R.\\nCyclic IF image analysis workﬂow\\nImage preprocessing\\nAll images were acquired as.czi ﬁles from the Axio Scan Z1 microscope (Zeiss, Germany). ZEN Blue image processing software (Zeiss, Ger-\\nmany) was used to convert.czi images to 64-bit BigTiff images without any compression of the original grayscale data for each channel.\\nThe following steps; registration, segmentation, and autoﬂuorescence subtraction, were performed.\\nA custom image registration algorithm was used to overlay DAPI images from each round of cyclic IF imaging. Registration quality was\\nconﬁrmed by visualizing on QiTissue software (Quantitative Imaging Systems, LLC, Pittsburgh, Pennsylvania) or the Napari application in\\nPython.\\nBackground signal removal\\nImages acquired after the chemical quenching step following the third round of cyclF experiment were selected as the autoﬂuorescence im-\\nage. The autoﬂuorescence image was subtracted from each round. Following autoﬂuorescence subtractions, a background smoothing oper-\\nation was performed to smooth out staining gradients and uneven illumination patterns on the images. This was done by applying white-top-\\nhat operation with the structuring element radius equal to the biggest nucleus radius within each tissue.\\nImage segmentation\\nCellpose was used for the segmentation of individual nuclei. 20 For nuclei segmentation, the z-projection image of all DAPI images from each\\nround of cyclic IF imaging was used. Cytoplasm segmentation mask was created by dilating the nuclear segmentation mask. Average signal\\nintensity values from individual nucleus, cell membrane, cytoplasm (nucleus minus cell membrane), and whole cell (nucleus plus cell mem-\\nbrane) compartments were calculated.\\nTo eliminate parts of tissue that degraded and fell off the slides, we created a binary mask from the DAPI image from the last round of\\nimaging and determined and used only the unique cell labels present in the last round DAPI mask.\\nSignal intensity normalization and thresholding\\nSignal intensities of each marker were normalized by background ﬂuorescence intensity. A multi-step approach was followed for background\\nﬂuorescence determination. First, we used an automated algorithm called RESTORE, which uses mutually exclusive markers, to calculate\\nbackground ﬂuorescence intensities. 93 Cutoff value outputs from RESTORE were visually inspected for each marker in each tissue and manu-\\nally adjusted. Cutoff values were then used to normalize signal intensities by dividing the raw ﬂuorescence intensity values by the cutoff value.\\nTherefore, values above ‘‘1’’ were ‘‘marker positive’’ as their signal intensity values were above the background signal intensity. Based on\\nnormalized intensity values, each cell in the dataset was labeled as proliferating (Ki67+) or non-proliferating (Ki67-) using a threshold of\\nnormalized Ki67 intensity >1. Similarly, each immune cell was identiﬁed as cytotoxic (GZMB+) or non-cytotoxic (GZMB-) and AR expressing\\n(AR+) or not (AR-).\\nClustering and cell type identiﬁcation\\nCell type identiﬁcation analysis was performed using publicly available Python and R packages. 21 Single-cell phenotypes were identiﬁed\\nbased on the normalized signal intensity values from our 28-plex cyclic IF data. To cluster single cells, we used an unsupervised Louvain com-\\nmunity detection algorithm called PhenoGraph (k = 40, to detect rare cell phenotypes).94 Before the application of PhenoGraph to our data,\\nwe performed 99 th percentile normalization and arcsin transformation (cofactor = 5). The GPU-accelerated implementation of PhenoGraph\\nwas implemented from https://gitlab.com/eburling/grapheno.21 For heatmap visualization of single-cell phenotypes and respective marker\\nexpression, ‘‘clustermap’’function from the seaborn package in Python was used from previously developed algorithms available at https://\\ngitlab.com/eburling/BCTMA/-/blob/master/pyviz_ﬁgs.py.\\nFollowing PhenoGraph, clusters were visually conﬁrmed by mapping the location of cells from each cluster onto segmentation masks. We\\nselected random samples to validate cluster annotations by doing side-by-side comparisons of cells mapped onto segmentation masks and\\nIF images. MATLAB code for mapping cells onto segmentation masks is available on GitHub (https://github.com/zeynepsayar/PCa_immune/\\nblob/main/mapping.m).\\nCell type annotation\\nWe selected 28 markers which would allow for the thorough classiﬁcation of speciﬁc cell types in the prostate tissue: epithelial (CK5, CK8,\\nECAD, EPCAM, Chromogranin A (ChromA) and CD44), stromal (aSMA, VIM, CD90, NCAM) and immune cells (CD3, CD4, CD11b, CD20,\\nCD45, CD66b, CD68, CD163, FOXP3, HLADRB1, NCAM, Tryp). We also included molecular and epigenetic markers (AR, p53, ERG, NKX\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 23\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='29ccc519-cbd5-426d-a69b-c8749dad410b', embedding=None, metadata={'page_label': '25', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='3.1, AMACR, PTEN, H3K27ac, H3K4me27), immune cell functional status markers (GZMB, PD1) and a proliferation marker (Ki67). We used\\nTryptase as our mast cell marker. Markers used for each cell type annotation are shown in Figure S2 .\\nWe adopted a 3-step clustering and annotation strategy to obtain the most granular clustering results and be able to detect rare cellular\\nsubtypes. In the ﬁrst step, we performed clustering in Phenograph using 40 K-nearest-neighbors and normalized signal intensity values from\\nall 28 markers listed above, identifying 49 clusters. The 49 clusters were annotated into 2 groups: epithelial, or non-epithelial based on cyto-\\nkeratin (CK5, CK8) and cadherin (ECAD) expressions.\\nAs the second step, we used Phenograph to cluster non-epithelial cells and annotate each cluster as either immune or non-immune stro-\\nmal cells. For this, we used 50 K-nearest-neighbors and all markers in our panel except for epithelial markers (CK5, CK8, ChromA, ECAD,\\nAMACR). We detected 30 clusters and annotated each as either immune or non-immune stroma. Stromal clusters were identiﬁed based\\non the expression of aSMA and VIM, and immune clusters based on the expression of various CD markers.\\nAs the third step, we ran Phenograph separately on epithelial, immune, and non-immune stromal clusters and re-clustered them using\\n50 K-nearest-neighbors to identify speciﬁc cell subtypes within each major cell type. After running each clustering analysis, we grouped\\nclusters with similar protein expression proﬁles under the same annotation identifying a total of 27 distinct cell types in the prostate\\nmicroenvironment.\\nFor re-clustering of epithelial cells, the following epithelial markers were used: H3K27ac, NKX3.1, NCAM, AMACR, CK8, ECAD, H3K4me3,\\nAR, VIM, CK5, CD44 and ChromA ( Figure 2C). For the re-clustering of non-immune stromal cells, the following epithelial markers were used:\\nAR, aSMA, HLA-DRB1, CD90, VIM, CD44, and NCAM. We identiﬁed 23 clusters and classiﬁed them into 10 subtypes ( Figure 5A). For re-clus-\\ntering of immune cells, the following epithelial markers were used: CD66b, CD11b, CD20, PD1, CD163, PD1, HLA-DRB1, AR, CD68, CD90,\\nTryp, GZMB, CD44, FOXP3, CD4, CD45 and CD3. We identiﬁed 27 clusters and classiﬁed them into 12 subtypes (Figure 3 C).\\nClusters were visually conﬁrmed by mapping the location of cells from each cluster onto segmentation masks. We selected random sam-\\nples to validate cluster annotations by doing side-by-side comparisons of cells mapped onto segmentation masks and IF images. MATLAB\\ncode for mapping cells onto segmentation masks is available on GitHub (https://github.com/zeynepsayar/PCa_immune/blob/main/\\nmapping.m).\\nUMAP visualization\\nTo enable the 2-dimensional visualization of our high-plex feature-by-cell data, we used the GPU accelerated, RAPIDS library adaptation of\\nUMAP on 99 th percentile normalized and arcsin transformed (cofactor = 5) data. The default set of parameters was used for both visualization\\nmethods.\\nQuantiﬁcation of epithelial marker heterogeneity using shannon entropy\\nCells from each tissue were labeled as positive for CK8, CK5, AR, AMACR, H3K27ac, H3K4me3 and NKX3.1 if their normalized signal inten-\\nsities were above 1. A Shannon entropy ﬁgure (Figures 2D and S6) was generated using UpSetPlot as implemented in the following algorithm:\\nhttps://gitlab.com/eburling/BCTMA/-/blob/master/rapids015_ﬁgs.py. The heterogeneity of epithelial markers was computed by measuring\\nthe Shannon entropy ( Figures 2D and S6A).21 Statistical differences between marker heterogeneity in TAN, G3, and G4 clinical groups were\\nquantiﬁed using the pairwise_tukey function from pingouin. 95\\nCombined marker expression analysis\\nTo test the signiﬁcance of the association between clinical groups and combined expression of epithelial markers (AR, AMACR, H3K27ac,\\nH3K4me3) we used mosaic plots. Mosaic plots visualize the deviance of observed combined marker expression in each clinical group\\nfrom the expected combined expression frequencies of an independence model. The mosaic plot in Figure 2E was generated with ‘‘mosaic’’\\nfunction from R’s ‘‘vcd’’ package using default parameters. The size of boxes in the mosaic plot is proportional to the difference between\\nexpected and observed frequencies. Boxes are colored in red if observed frequencies are smaller than expected, and blue if observed fre-\\nquencies are greater than expected. We performed a chi-square test with the null hypothesis that grades and combined marker expressions\\nare independent. P-values <0.001 reject the null hypothesis and indicate grades and combined marker expressions are dependent.\\nMast cell differential marker expression analysis\\nNormalized mean intensity values of CD68, AR, PD1, CD44, and CD90 were plotted for each cell in Mast Cluster 1 and Mast Cluster 2 from all\\nsamples in the cohort. t-test was used to identify any statistically signiﬁcant differences between individual and combined marker expressions\\nbetween Mast cluster 1 vs. 2. p-values <0.05 were signiﬁcant.\\nCell density calculations\\nCell counts were calculated for each cluster in epithelial, stromal, and immune populations. Tissue areas were calculated in mm 2 using ImageJ\\nsoftware (NIH, Bethesda, Maryland). The whole tissue region was selected as the region of interest (ROI) and the area was calculated in mm 2\\nusing the measure option and the manually set scale. The fraction of Ki67-positive proliferating cells within each immune, epithelial, and stro-\\nmal cell types was calculated and plotted inFigure 5E. Frequencies of each cell phenotype found in each tissue section were plotted in Fig-\\nure 3 B. The statistical signiﬁcance of differences between cell densities was tested using a t-test ( p < 0.05).\\nll\\nOPEN ACCESS\\n24 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='841a7081-894c-4970-a541-3bc324192a20', embedding=None, metadata={'page_label': '26', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Spatial analysis\\nPairwise cell-cell interactions\\nPairwise cell-cell interactions were calculated using the R Shiny app ‘‘app_CRC_contacts.R’’ ( https://github.com/nolanlab/Neighborhood\\nCoordination).96 Cell types were omitted if the number of unique adjacent cells for that cell type was less than 100 cells. Calculated values\\nwere mapped using the ComplexHeatmap package in R. To calculate pairwise cell-cell contacts, we calculated the likelihood ratios of the 27\\ncell types that we identiﬁed as follows:\\nLoglikelihood ratio = log2\\n/C18 Observed frequency of co-ocurrance\\nTheoretical frequency of co-ocurrance\\n/C19\\nVoronoi diagrams were generated through the following process in https://github.com/nolanlab/NeighborhoodCoordination: FCS ﬁles\\nobtained from VorteX were exported and then processed using a custom Java algorithm. This algorithm was designed speciﬁcally to create\\nVoronoi diagrams and derive contact matrices between individual cells.\\nCell neighborhood identiﬁcation\\nCell neighborhoods were determined by using existing analysis platforms found in https://github.com/nolanlab/NeighborhoodCoordination.\\nThe ten nearest neighbors of each individual cell were determined. Each window was represented as the frequency of each cell type found\\nwithin the ten nearest neighbors. In total we identiﬁed 27 cell types, therefore each window was a vector of length 27. To determine the\\ncell neighborhoods, these windows were clustered using K-means clustering with k = 5 and k = 10 with Python’s scikit-learn library and ‘‘Mini-\\nBatchK-Means\" function. We identiﬁed signiﬁcantly enriched neighborhoods in each clinical group by performing t-tests ( p < 0.05).\\nTo validate the reproducibility of cell neighborhoods identiﬁed from the dataset and their enrichment in speciﬁc clinical groups, we iter-\\natively performed cell neighborhood analysis by leaving out one patient at a time. For example, we identiﬁed cell neighborhoods from all cells\\nin the dataset excluding cells from patient 1 and performed t-test to identify which cell neighborhoods were signiﬁcantly enriched in TAN vs.\\nG3 vs. G4. We repeated this analysis for each patient in the cohort.\\nEnrichment of cell types within cell neighborhoods\\nTo calculate differential cell type enrichment in rCNs, differential enrichment analyses using linear models was conducted. The models were\\nestimated using the following equation: Yn,c = b0+ b1X + b3Yc + e. In this equation, Yc represents the logarithm of the overall frequency of\\ncell type c, X is an indicator variable for the patient group, Yn,c represents the logarithm of the frequency of cell type c in rCN n, bi are co-\\nefﬁcients, and e represents Gaussian noise with a mean of zero. To avoid computational issues, a pseudocount of 1e-3 was added before\\ntaking logarithms. The estimation of these models was performed using the statsmodels Python package. The resulting coefﬁcient estimates\\nand p-values for b1 were extracted and visually presented (signiﬁcant if p < 0.05).\\nSpatially variable markers with Moran’s I\\nMoran’s I evaluate spatial autocorrelation; whether a marker is spatially clustered, dispersed, or random based on the spatial expression of\\nthat marker:\\nI = n\\nW\\nPn\\ni = 1\\nPn\\nj = 1 wi;jzizj\\nPn\\ni = 1 z2\\ni\\nwhere zi is the deviation of the feature from the mean\\n/C0\\nxi /C0 1\\nn\\nPn\\ni = 1 xi\\n/C1\\n, wi;j is the spatial weight between observations, n is the number of\\nspatial units and W is the sum of all wi;j.\\nThe Moran’s I global spatial auto-correlation statistics were calculated using Python’s Squidpy package implementation.33 We computed\\nthe Moran’s I score with ‘‘ squidpy.gr.spatial_autocorr’’ andmode = ’moran’. We ﬁrst generated a spatial graph with ‘‘ squidpy.gr.spatial_\\nneighbors’’ which provides a score on the degree of spatial variability of marker expression. The statistic as well as the p-value were computed\\nfor each marker, and FDR correction was performed (signiﬁcant if p < 0.05). We visualized the top markers after applying Moran’s I spatial\\nautocorrelation statistics.\\nSpatially variable cell types with Ripley’s L\\nRipley’s spatial statistics is a family of spatial analysis methods used to describe whether points with discrete annotation (i.e., cell-types) in a n\\narea of interest follow random, dispersed or clustered patterns. To calculate Ripley’s L, we used ‘‘ squidpy.gr.ripley’’function with mode = ‘L’.\\nCell-type co-occurrence ratio\\nThe co-occurrence ratio of cell types provides a score on the co-occurrence of the cell type of interest across spatial dimensions. It is deﬁned\\nas follows :\\np\\n/C0\\nexpjcond\\n/C1\\np\\n/C0\\nexp\\n/C1\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 25\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='158040fa-561f-4c1d-82ce-3ca9bd725b43', embedding=None, metadata={'page_label': '27', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Where pðexpjcondÞ is the conditional probability of observing a cell-typeexp conditioned on the presence of a cell-type cond, whereas\\npðexpÞ is the probability of observing exp in the radius size of interest. The score was computed across increasing radii size around each\\ncell in the tissue. We used Python’s ‘‘squidpy’’ library, ‘‘squidpy.gr.co_occurrence’’function to calculate the cell-type co-occurrence ratio. 33\\nSpatial heterogeneity and spatial power analysis\\nWe performed several rigorous statistical analyses to ensure the sufﬁcient spatial power and consistency of our results: (1) creating in silico\\ntissue to augment sample size and to address sample variability, (2) leave-one-patient-out analysis to assess the impact of patient-to-patient\\nvariability on the results, (3) subsampling analysis to estimate the variability in our results and showed the stability of our ﬁndings, (4) permu-\\ntation test to show further benchmarking and robustness of our ﬁndings. In all of the analyses, we considered a p-value of 0.05 or less to be\\nstatistically signiﬁcant. All these different tests are different ways to address statistical power to determine the signiﬁcance of the results.\\nCreating in situtissues – ISTs to augment the data\\nSeveral essential parameters are outlined for planning a spatial omics study with adequate statistical power. 67 The authors presented a ﬂex-\\nible approach for generating in silico tissues (IST) and combined it with spatial proﬁling datasets to establish an exploratory computational\\nframework for analyzing spatial power. Using their experimental design, to address effects of spatial variability and sample size, we divided\\neach tumor sample into multiple regions, then performed cellular neighborhood analysis on these multiple regions, and showed the consis-\\ntency of our results.\\nWe divided each whole-tissue image into multiple regions of size 2K-by-2K pixels after testing different size parameters: 500, 1K, 2K, 3K,\\n4K, and 5K. First, we selected the most populated (number of cells in the grid) top 10 ISTs for each grade from 2K-by-2K grids ending up with\\n30 ISTs in total. The number of cells these 30 ISTs contain ranges from 1973 to 3457 cells. We obtained 22556, 28611, and 33440 cells for TAN,\\nG3, and G4 grades respectively, totaling 84607 cells. As a second experiment, we also selected the most populated (number of cells in the\\ngrid) top 50 tiles for each grade from 2K-by-2K grids ending up with 150 IST in total. The number of cells these 150 ISTs contain ranges from\\n1460 to 3929 cells. We obtained 91117, 121734, and 124326 cells for TAN, G3, and G4 grades respectively, totaling 337177 cells.\\nLeave one patient out analysis\\nTo further conﬁrm the robustness of our ﬁndings, we performed an exclusion analysis on the spatial neighborhood data. Leave-one-patient-\\nout analysis helps in understanding how sensitive the results are to the presence or absence of speciﬁc patients. It is particularly useful when\\ndealing with a small dataset and can provide insights into the robustness and stability of the ﬁndings. It also allows us to assess the impact of\\npatient-to-patient variability on the results.\\nFor each round of analysis, we excluded one patient’s data from the imaging dataset. Then, performed rCN analysis, calculated the rCN\\nfrequencies for each patient, and compared AR + rCN frequency between TAN and tumors. We repeated this process iteratively for each\\npatient, leaving one out at a time. Hence, we performed the analysis 13 times, each time excluding a different patient’s data. We evaluated\\nthe consistency of our results across the iterations. We observed that rCN patterns were robust and not overly inﬂuenced by speciﬁc patients.\\nWe also conﬁrmed that the AR + rCN frequency between TAN and tumors (G3 and G4) remained statistically signiﬁcant 10 out of 13 iterations\\nfor TAN vs. G3, and 11 out of 13 iterations for TAN vs. G4.\\nSub-sampling analysis\\nWe performed subsampling to validate that our sample size was sufﬁcient to detect clinically relevant and statistically robust differences in our\\nAR + rCN conclusion. First, 100 iterations of subsampling with different seeds were performed at four independent values; namely 50, 75, 90,\\nand 99% of the data (sub-sampled from each grade data with given sub-sampling percentage). Then we run the rCN algorithm on each sub-\\nsampled data to identify the AR + rCNs. We next tested whether the mean AR + rCN frequency was signiﬁcantly different between TAN and\\ntumors (G3 and G4) within each iteration of the sampled data. The distribution of p-values for the mean AR + rCN frequency differences be-\\ntween patient groups for each of these 100 iterations shows that even with 50% data subsampling, most iterations yield a p-value <0.05\\n(Figures 5 F, S10, and S11). The proportion of statistically signiﬁcant p-values is 86, 82, 83 and 91% (for TAN vs. G3); and 90, 93, 96, and\\n92% (for TAN vs. G4) for subsamples of 50, 75, 90, and 99%, respectively).\\nSpatial signiﬁcance test 1\\nSpatial t-test with permutations: We used scipy.stats.ttest_ind() function in Python with exact permutations; permutations = np.inf during rCN\\nfrequency comparison. This permutes the patient’s grade assignment while comparing rCN frequencies with t-tests. We also separately\\ntested equal variance across grades veriﬁed using Bartlett’s test. AR + rCN (CN1) frequency was signiﬁcantly different between TAN and tu-\\nmors (G3 and G4) with p-values of 0.018 and 0.036, respectively.\\nSpatial signiﬁcance test 2\\nSpatial permutation test with sample mean difference statistic: We performed a permutation test using scipy.stats.permutation_test(). We\\ndeﬁned the difference between the sample means as a test statistic to be used in scipy.stats.permutation_test(). We performed an exact\\ntest, n_resamples = np.inf. The permutation distribution of the test statistic is provided for further investigation, not only for AR + rCN but\\nll\\nOPEN ACCESS\\n26 iScience 27, 110668, September 20, 2024\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='74793e5a-b708-44d2-a89d-2778dec02823', embedding=None, metadata={'page_label': '28', 'file_name': 'ak_prostate.pdf', 'file_path': '/content/data/ak_prostate.pdf', 'file_type': 'application/pdf', 'file_size': 10052254, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='for all rCN frequency comparisons. AR + rCN (rCN1) frequency was signiﬁcantly different between TAN and tumors (G3 and G4) with p-values\\nof 0.036 and 0.036, respectively.\\nSpatial signiﬁcance test 3\\nSpatial negative control by shufﬂing the rCN IDs: In this test, ﬁrst, we calculated the ground truth neighborhood distribution matrix, as we\\ndescribed in cell neighborhood identiﬁcation section above. Then, the rCN assignment labels on the tissue are shufﬂed to relabel tissue\\nby rCNs, preserving the tissue structure. At each shufﬂe, the observed neighborhood frequency per grade was recalculated. We repeated\\nthe shufﬂing 100 times, and the proportion of statistically signiﬁcantp-values is 8 and 29%. This result shows that the fact the AR + rCNs\\nare differentiative of TAN vs. tumors was not an artifact.\\nSpatial signiﬁcance test 4\\nSpatial negative control by both sub-sampling 99% of the data and shufﬂing the rCN IDs: We ﬁrst subsampled our single-cell spatial data at\\n99%, 100 times. Then we identiﬁed the rCNs. Then repeated 3.a., shufﬂed the rCN IDs. At each shufﬂe of the 99% sub-sampled data, the\\nobserved neighborhood frequency per grade was recalculated. The proportion of statistically signiﬁcantp values is 6 and 14%.\\nOther statistical tests\\nAll statistical tests to infer p-value for signiﬁcant differences ( p < 0.05) in mean were performed using the t-test.\\nll\\nOPEN ACCESS\\niScience 27, 110668, September 20, 2024 27\\niScience\\nArticle', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='49d7c702-b819-412a-9b36-bab00b12e812', embedding=None, metadata={'page_label': '1', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ARTICLE OPEN\\nSingle-cell spatial architectures associated with clinical\\noutcome in head and neck squamous cell carcinoma\\nKatie E. Blise 1,2, Shamilene Sivagnanam 2,3, Grace L. Banik 3,4,5, Lisa M. Coussens 2,3 and Jeremy Goecks 1,2 ✉\\nThere is increasing evidence that the spatial organization of cells within the tumor-immune microenvironment (TiME) of solid\\ntumors inﬂuences survival and response to therapy in numerous cancer types. Here, we report results and demonstrate the\\napplicability of quantitative single-cell spatial proteomics analyses in the TiME of primary and recurrent human papillomavirus\\n(HPV)-negative head and neck squamous cell carcinoma (HNSCC) tumors. Single-cell compositions of a nine patient, primary and\\nrecurrent (n = 18), HNSCC cohort is presented, followed by deeper investigation into the spatial architecture of the TiME and its\\nrelationship with clinical variables and progression free survival (PFS). Multiple spatial algorithms were used to quantify the spatial\\nlandscapes of immune cells within TiMEs and demonstrate that neoplastic tumor-immune cell spatial compartmentalization, rather\\nthan mixing, is associated with longer PFS. Mesenchymal (αSMA+) cellular neighborhoods describe distinct immune landscapes\\nassociated with neoplastic tumor-immune compartmentalization and improved patient outcomes. Results from this investigation\\nare concordant with studies in other tumor types, suggesting that trends in TiME cellular heterogeneity and spatial organization\\nmay be shared across cancers and may provide prognostic value in multiple cancer types.\\nnpj Precision Oncology            (2022) 6:10 ; https://doi.org/10.1038/s41698-022-00253-z\\nINTRODUCTION\\nTumor microenvironments, comprising both neoplastic tumor\\ncells and recruited stromal cells of various lineages, including a\\ndiverse assemblage of immune, mesenchymal, and vascular cells,\\nplay a key role in both de novo progression of tumors and\\nregulating response to therapies 1–3. Numerous studies have\\nreported that, in addition to the types and quantities of cells\\npresent in the tumor immune-microenvironment (TiME), the\\nspatial organization of the TiME is prognostic for survival and\\nresponse to therapy in multiple cancer types 4–13. Metrics that\\nquantify this spatial organization can range from simple density\\nratios within speciﬁc tumor regions14, such as the Immunoscore15,\\na now commonly used biomarker for colorectal tumor staging, to\\nmore complex measures that account for the precise locations of\\nspeciﬁc cells relative to other cells, such as mixing scores4 and\\ncellular neighborhood measures7. These more advanced spatial\\nquantiﬁcations are a result of emerging single-cell multiplex tissue\\nimaging modalities4,6,7,16–18, which provide detailed phenotypic\\nand effector proteomic markers for each cell, while maintaining\\nthe spatial architecture of the tissue assayed. Knowing the precise\\nlocations of cells in the TiME enables a deeper understanding of\\nhow cells interact within the tumor, as both direct and indirect cell\\nsignaling mechanisms require cells to be near, if not directly\\nadjacent to one another19. This understanding can aid treatment\\ndecisions, as many therapies require spatial proximity of speciﬁc\\ncell types for efﬁcacy10. Given that single-cell imaging technolo-\\ngies are still relatively new, there is much to be discovered\\nregarding how the spatial organization of cells within the TiME\\nrelates to clinical outcome and may be used for patient\\nstratiﬁcation decisions for therapy.\\nHead and neck squamous cell carcinoma (HNSCC) is the sixth\\nleading form of cancer worldwide20, and it accounts for more than\\n10,000 deaths per year in the US alone21. While patients harboring\\nhuman papillomavirus (HPV) within neoplastic cells tend to exhibit\\na better prognosis, their HPV-negative [HPV( −)] counterparts\\ntypically exhibit T cell suppressive TiMEs and have a signiﬁcantly\\ngreater risk of recurrence and shorter 3-year survival22–24. There is\\na critical need to improve understanding of HNSCC TiMEs to\\nenable better patient stratiﬁcation for therapy, as well as identify\\nnew targets that could be leveraged for therapeutic intervention\\nto improve outcome, particularly for patients with HPV(−) tumors\\nwho currently lack promising therapeutic options. We previously\\ndeveloped a multiplex immunohistochemistry (mIHC) imaging\\nplatform to aid studies investigating the immune contexture of\\nsolid tumors and their response to therapies at the single-cell\\nlevel16,17. Using a sequential antibody staining protocol, detection\\nof 12–30 proteins can be enumerated at single-cell resolution\\nacross a single formalin- ﬁxed parafﬁn-embedded (FFPE) tissue\\nsection. This enables single-cell phenotyping of discrete leukocyte\\nlineages, and importantly, reveals their spatial relationships with\\nother cells in the tissue section. Utilizing this mIHC approach on a\\nsmall cohort of eighteen HPV(−) primary and matched recurrent\\nHNSCC tumor samples collected from nine patients, we previously\\nreported immune contextures associated with disease recurrence,\\nmost notably that myeloid inﬂamed proﬁles in primary tumors\\nexhibited shorter progression-free survival (PFS) compared to\\nlymphoid inﬂamed proﬁles16,17.\\nIn this study, we have signiﬁcantly extended our prior analysis\\nof this cohort, focusing on tumor heterogeneity and composi-\\ntional changes from primary to recurrent tumors, in addition to\\nusing multiple spatial algorithms to quantify the spatial organiza-\\ntion of the TiMEs. We then correlated these spatial features with\\nPFS and identiﬁed TiME architectures that may be important for\\ntherapeutic decision making. Overall, we found increased\\nneoplastic tumor-immune cell spatial compartmentalization in\\nprimary tumors to be associated with longer PFS. These tumors\\n1Department of Biomedical Engineering, Oregon Health & Science University, Portland, OR, USA.2The Knight Cancer Institute, Oregon Health & Science University, Portland, OR,\\nUSA. 3Department of Cell, Developmental & Cancer Biology, Oregon Health & Science University, Portland, OR, USA.4Otolaryngology-Head & Neck Surgery, Oregon Health &\\nScience University, Portland, OR, USA.5Present address: Division of Otolaryngology, Children’s Hospital of Philadelphia, Philadelphia, PA, USA.✉email: goecksj@ohsu.edu\\nwww.nature.com/npjprecisiononcology\\nPublished in partnership with The Hormel Institute, University of Minnesota\\n1234567890():,;', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1b6f60ba-6315-492d-a25a-f02338f33f0f', embedding=None, metadata={'page_label': '2', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='also contained alpha smooth muscle actin ( αSMA+) cells with\\nmore organized structure located near T and B cells, as well as\\nnear cells involved in antigen presentation. Our results are\\nconcordant with those from other studies, indicating that the\\nfeatures identiﬁed herein are likely shared and prognostic across\\ncancer types.\\nRESULTS\\nOne to three regions of 25002 µm2 from each of the nine patients\\nprimary and matched recurrent tumor resections (n = 18) were\\nanalyzed, for a total of 47 regions (Table1 and Fig. 1a). For this\\nstudy, we utilized a gating strategy with thirteen lineage or\\nfunctional protein biomarkers to classify cells as neoplastic tumor\\ncells, stromal cells (mesenchymal), or one of seven different\\nleukocyte subtypes spanning lymphoid and myeloid lineages\\n(Table 2 and Supplementary Fig. 1a). We investigated tumor\\nheterogeneity both within and across patient samples, quantiﬁed\\nthe cellular spatial relationships within the TiME using a mixing\\nscore and performed a neighborhood clustering method to\\ndescribe the association between TiME spatial architecture, clinical\\nfeatures, and PFS.\\nSingle-cell proteomic analyses reveal varying degrees of\\ntumor heterogeneity\\nTo quantify how cellular composition varied across tumor regions,\\nwe assessed tumor heterogeneity at multiple levels, including\\nintra-tumoral, intra-patient, and inter-patient cellular heterogene-\\nity by calculating Kullback-Leibler (KL) divergences for each\\nregion, performing hierarchical clustering, and conducting a\\nprincipal component analysis (PCA). The density of each cell type\\nper region was measured for all eighteen tumor specimens by\\ntaking the count of each cell type divided by the measured tissue\\narea in mm2 (Fig. 1b, c). We then calculated the coefﬁcient of\\nvariation per cell type for each tumor, and averaged these values\\nto quantitatively describe the cell types contributing most to intra-\\ntumoral heterogeneity within the cohort. The coef ﬁcient of\\nvariation is deﬁned as the standard deviation divided by the\\nmean, and it provides a normalized measure of variability for\\ncomparison across cell types with large differences in densities. On\\naverage, B cells exhibited the greatest coef ﬁcient of variation\\nacross the cohort relative to other cell types (Table3). This is likely\\ndue to the fact that B cells were frequently observed to be\\nspatially clustered together, resulting in regions of either high B\\ncell density or low B cell density despite being collected from the\\nsame tumor (Supplementary Fig. 1b).\\nTo further quantify and assess tumor heterogeneity both within\\nand across patients, we calculated the KL divergence of each\\ntumor region from ﬁve average cell type distributions. KL\\ndivergence is a relative measure of how similar two distributions\\nare, with larger values reﬂecting less similarity between the\\ndistributions and smaller values reﬂecting more similarity between\\nthe distributions. This measure has been used previously to\\nquantify tumor heterogeneity6. By calculating and comparing the\\ndivergences of each tumor region from multiple average cell type\\ndistributions, we were able to assess heterogeneity within and\\nacross tumors and patients. Overall, we observed that hetero-\\ngeneity was lower across regions from the same tumor and\\ntumors from the same patient (primary or recurrent), while higher\\nacross tumors from different patients. This is evidenced by smaller\\nintra-tumoral and intra-patient KL divergence values for the\\nmajority of tumor regions (Fig.1d).\\nThe cellular distributions used to calculate the ﬁve KL\\ndivergence values per tumor region were (1) the average cellular\\ndistribution across all regions sampled from the same tumor\\n[“Intra-Tumor (P or R only)”]; (2) the average cellular distribution\\nacross the patient’s primary and recurrent tumors [“Intra-Patient (P\\nTable 1. Patient and tumor characteristics.\\nPatient ID Anatomic site of resection Primary tumor TNM stage Therapy following primary\\nresection\\nGender Race Alcohol\\nhistory\\nSmoking\\nhistory\\nHPV status Progression-free\\nsurvival (days)\\n1 Oral Cavity 2 Cisplatin + Radiation Male White Yes Yes Negative 804\\n2 Oral Cavity 4 Cetuximab + Radiation Male White Yes Yes Negative 123\\n3 Oral Cavity 1 Cisplatin + Radiation Female White Yes No Negative 1447\\n4 Oral Cavity 1 Cisplatin + Radiation Female White No No Negative 246\\n5 Oropharynx 4 Cetuximab + Radiation Female White Yes No Negative 202\\n6 Oropharynx 4 Cisplatin + Radiation Female Asian No No Negative 188\\n7 Larynx 4 Cisplatin + Cetuximab + Radiation Female White Yes Yes Negative 409\\n8 Larynx 3 Cisplatin + Radiation Female White Yes Yes Negative 1033\\n9 Larynx 4 Cisplatin + Radiation Female White Yes Yes Negative 83\\nKE Blise et al.\\n2\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota\\n1234567890():,;', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='30a8f8ab-7dd5-4cd1-a1ac-f156384277bf', embedding=None, metadata={'page_label': '3', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='and R)”]; (3) the average cellular distribution across all tumors in\\nthe cohort collected from the same timepoint [“Inter-Patient (P or\\nR only)”]; (4) the average cellular distribution across all tumors in\\nthe cohort resected from the same anatomic site [“Inter-Patient\\n(Same Anatomic Site)”], and; (5) the average cellular distribution\\nacross all tumors collected from all patients in the cohort,\\nregardless of primary or recurrent status or anatomic site [“Inter-\\nPatient (all)”]. By comparing the relative KL divergence values to\\neach other, we found tumor regions to be more similar to regions\\nsampled from the same tumor and patient than regions collected\\nKE Blise et al.\\n3\\nPublished in partnership with The Hormel Institute, University of Minnesota npj Precision Oncology (2022)    10 ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5bbd7fee-aeef-4e4c-893a-b37af1b837a0', embedding=None, metadata={'page_label': '4', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='from tumors of other patients. Notably, we found no signiﬁcant\\ndifference between “Inter-Patient (Same Anatomic Site) ” and\\n“Inter-Patient (all),” indicating that tumor regions diverged by the\\nsame degree from regions sampled at the same anatomic site as\\nthey did from regions sampled at all three anatomic sites (oral\\ncavity, oropharynx, and larynx) of the head and neck region (Fig.\\n1d). Finally, given the large proportion of neoplastic tumor cells\\ncomprising the TiME for many of the tumor regions, we assessed\\nKL divergence using only the distribution of immune cells present\\nand found similar results (Supplementary Fig. 1c). This indicates\\nthat immune cell composition is more similar within regions from\\nthe same patient than across regions collected from different\\npatients.\\nTo further investigate intra-patient heterogeneity, we per-\\nformed unsupervised hierarchical clustering on the 47 tumor\\nregions based on their normalized density composition (Fig.1e).\\nWe found that two patients (3, green; 7, pink) contained all tumor\\nregions clustering together, independent of primary or recurrent\\nstate. These patients also had the smallest intra-patient KL\\ndivergence values (Fig.1d), indicating that the cell densities of\\nthese patients’primary and recurrent tumors were similar to each\\nother. Three patients (2, orange; 4, red; 9, yellow) contained nearly\\nall regions clustered together. The remaining four patients’tumors\\nexhibited greater degrees of intra-patient heterogeneity, as\\ndemonstrated by the distance between primary and recurrent\\ntumor regions on the clustered heatmap (Fig. 1e). Overall, we\\nfound that regions evaluated from the same patient tended to\\ncluster together more than regions evaluated from different\\npatients (Fig.1e), indicating increased heterogeneity between\\npatients as compared to between samples from the same patient.\\nWe also examined whether tumor regions clustered by the\\nanatomic resection site and found that the clusters formed did not\\ngroup by site. These results, in addition to those of our KL\\ndivergence analyses, indicate that anatomic site was likely not the\\nmain contributor of cellular heterogeneity in this cohort. PCA\\nresults also supported these observations (Fig.1f, g).\\nTiME cellular composition altered by therapy\\nMultiple studies have reported differences in TiME cellular\\nmakeup25,26 and tumor clonal diversity27 between primary and\\nrecurrent tumors. To assess whether any immune contexture\\nchanges occurred following post-operative therapy in our cohort,\\nwe analyzed the cellular composition of primary tumors as\\ncompared to their recurrent tumors. All patients received a\\ncombination therapy of cisplatin and/or cetuximab accompanied\\nby radiation following surgical resections of their primary tumors.\\nWe used the average density of each cell type present across\\nregions for a given tumor and compared primary tumor\\ncomposition to their matched recurrent tumor composition. While\\nwe did not observe any signiﬁcant differences in cell density\\nbetween primary and recurrent tumors (p > 0.112), we didﬁnd\\nthat all patients experienced a decrease in the density of B cells\\nfrom their primary to recurrent tumors (Fig.2a, b). This result is\\nsupported by a recent study that found that a large cohort of\\nHNSCC patients experienced a decline in B cells from primary to\\nrecurrent tumors, and this reduction was dependent upon\\npatients receiving adjuvant chemoradiotherapy28. We found no\\ncommon differences across all patients for any other cell type,\\nalthough eight of the nine patients saw a decrease in the density\\nof neoplastic tumor cells from their primary to recurrent tumors\\n(Fig.2b).\\nWe then assessed whether patients exhibited similar composi-\\ntional differences in primary and recurrent TiMEs by performing\\nunsupervised hierarchical clustering on the normalized average\\nFig. 1 Heterogeneity across patients and tumor regions. aCohort and tissue region selection overview. One to three regions of 25002 µm2\\nwere assayed with mIHC per eighteen tumor resections and are represented by the green boxes in the tissue images. Black scale bar=\\n2500 µm.b Density (cells/mm2) of each cell type present per individual primary (P) and recurrent (R) tumor. Each dot represents a single tumor\\nregion (n = 47). c Density (cells/mm2) of each immune cell type present per individual primary (P) and recurrent (R) tumor. Each dot represents\\na single tumor region (n = 47). d Box plot of the Kullback– Leibler divergences from a single tumor region’s cellular distribution compared to:\\nthe tumor’s average cellular distribution [Intra-Tumor (P or R only)], the patient’s average cellular distribution [Intra-Patient (P and R)], the\\ncohort’s average cellular distribution across tumors of the same timepoint [Inter-Patient (P or R only)], the cohort’s average cellular distribution\\nacross tumors of the same anatomic site [Inter-Patient (Same Anatomic Site)], the cohort’s average cellular distribution across all tumors from\\nall patients [Inter-Patient (all)]. P-values calculated using a one-way ANOVA multi-group signiﬁcance test followed by a Tukey honestly\\nsigniﬁcant difference post-hoc test.e Heatmap of cellular composition across tumor regions. Rows are individual tumor regions that are\\nordered by the hierarchical clustering of their cellular composition. Columns are the cell types used as clustering features. Composition was\\nnormalized using a log10+ 1 transformation before clustering. Leftmost three columns are color coded by patient, tumor timepoint, and\\nanatomic site. f Principal component analysis on cellular density following a log10+ 1 transformation. Each point (n = 47) represents one\\ntumor region and is colored by patient. The shape of each point denotes primary or recurrent status.g Principal component analysis on\\ncellular density following a log10+ 1 transformation. Each dot (n = 47) represents one tumor region and is colored by anatomic resection site.\\nTable 2. mIHC cell phenotype classiﬁcation.\\nCell phenotype Antibody markers\\nCD8+ T Cell CD45 + CD20– CD3+ CD8+\\nCD4+ T helper CD45 + CD20– CD3+ CD8– FOXP3–\\nCD4+ regulatory T cell CD45 + CD20– CD3+ CD8– FOXP3+\\nB Cell CD45 + CD20+\\nMacrophage CD45 + CD20– CD3– CD66B– CD68+\\nGranulocyte CD45 + CD20– CD3– CD66B+\\nAntigen presenting cell CD45+ CD20– CD3– CD66B– CD68– MHCII+\\nOther immune CD45 + CD20– CD3– CD66B– CD68– MHCII–\\nCD8– FOXP3–\\nαSMA+ mesenchymal CD45 – PANCK– αSMA+\\nNeoplastic tumor CD45 – PANCK+\\nFunctional markers PD-1, PD-L1, Ki-67\\nTable 3. Coefﬁcient of variation.\\nCell phenotype Average coef ﬁcient of variation\\nB Cell 0.658\\nMacrophage 0.563\\nαSMA+ mesenchymal 0.545\\nNeoplastic tumor 0.541\\nGranulocyte 0.523\\nAntigen presenting cell 0.519\\nOther immune 0.402\\nCD8+ T cell 0.384\\nCD4+ T helper 0.376\\nCD4+ regulatory T cell 0.311\\nKE Blise et al.\\n4\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5b0073dc-4d1b-4ddb-90c2-5cd89b883629', embedding=None, metadata={'page_label': '5', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='difference in cellular composition for each cell type. This resulted\\nin two groups of patients (Fig.2c). Interestingly, the two patients\\nthat received cetuximab (2, orange; 5, purple), rather than\\ncisplatin, clustered together within one of these groups and were\\nthe only two patients to experience a decrease in the density of\\nevery cell type following therapy. Across the cohort these patients\\nhad the greatest decrease in the density of macrophages,\\ngranulocytes, and other CD45+ immune cells present from their\\nprimary tumors to their recurrent tumors following therapy (Fig.\\n2b, c). Interestingly, the one patient who received both cisplatin\\nand cetuximab (7, pink) was present in the other cluster from the\\ntwo patients who received only cetuximab. This was the only\\npatient to experience an increase in the density of neoplastic\\ntumor cells (Fig.2b, c). This patient also experienced the second\\nlargest increase in CD8+ T cells as well as the greatest decrease in\\nCD4+ regulatory T cells, potentially indicating a pro-inﬂammatory\\na\\nc\\nb\\nd\\nCD8\\n+ T Cell\\nCD4\\n+ T Helper\\nCD4\\n+ Regulatory \\nT Cell\\nB Cell MacrophageGranulocyteAntigen Presenting CellOther ImmuneαSMA\\n+ Mesenchymal\\nNeoplastic\\n Tumor\\n−2000\\n−1000\\n0\\n1000\\n2000\\n3000\\nShort PFS\\nLong PFS\\nCell Phenotype\\nChange in Density (cells/mm2)\\n(Averaged Across Regions)\\nCD8\\n+ T Cell\\nCD4\\n+ T Helper\\nCD4\\n+ Regulatory T Cell\\nB Cell MacrophageGranulocyteAntigen Presenting CellOther ImmuneαSMA\\n+ Mesenchymal\\nNeoplastic Tumor0\\n1000\\n2000\\n3000\\n4000\\n5000\\nCell Phenotype\\nDensity (cells/mm2)\\n(Averaged Across Regions)\\nPrimary\\nRecurrent\\nPatient\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n0-\\n1000-\\n2000-\\n3000-\\n4000-\\n5000-350-\\n300-\\n250-\\n200-\\n150-\\n100-\\n50-\\n800-\\n600-\\n400-\\n200-\\n0-\\n300-\\n250-\\n200-\\n150-\\n100-\\n50-\\n0-0-\\n200-\\n400-\\n600-\\n800-\\n1000-\\n1200-\\n0-\\n50-\\n100-\\n150-\\n200-\\n250-\\n300-\\n350-\\n0-\\n200-\\n400-\\n600-\\n800-\\n1000-\\n1200-\\n1400-\\n1600-\\n0-\\n50-\\n100-\\n150-\\n200-\\n250-\\n300-\\n350-\\n0-\\n100-\\n200-\\n300-\\n400-\\n500-\\n600-\\n700-\\n800-\\n0-\\n250-\\n500-\\n750-\\n1000-\\n1250-\\n1500-\\n1750-\\nRPRPRPRPRPRPRPRPRPR\\nDensity (cells/mm2)\\n(Averaged Across Regions)\\nNeoplastic Tumor\\nP\\nαSMA+ MesenchymalOther Immune\\nAntigen Presenting\\nCellGranulocyteMacrophageB Cell\\nCD4+ Regulatory\\nT CellCD4+ T HelperCD8+ T Cell\\nNeoplastic Tumor -\\nOther Immune -\\nCD8+ T Cell -\\n CD4+ T Helper -\\n CD4+ Regulatory T Cell -\\nαSMA+ Mesenchymal -\\nMacrophage -\\nGranulocyte -\\nB Cell -\\nPatient ID -\\nTherapy -\\n-lleC\\ngnitneserP\\nnegitnA\\nCell Phenotype\\n- 9\\n- 1\\n- 2\\n- 5\\n- 6\\n- 8\\n- 7\\n- 3\\n- 4\\nPatient\\n- 0.8\\n- -0.4\\n- 0.4\\n- 0.0\\n- -0.8\\nBoth\\nCisplatin\\nCetuximab\\nFig. 2 Tumor cellular composition changes following therapy. aBox plot showing the average density of each cell type split by primary and\\nrecurrent status. Each dot represents the density of that cell type for one tumor, averaged across regions (n = 9 primary tumors, n = 9\\nrecurrent tumors). Boxes= quartile 1 (Q1) to quartile 3 (Q3); whiskers= smallest and largest datapoints within 1.5*interquartile range (IQR)\\n+/− Q3/Q1; solid line= median. b Line plots showing the change in cellular composition from primary to recurrent tumors for each cell type.\\nEach dot represents the density of that cell type for one tumor, averaged across regions. Lines are colored by patient.c Heatmap of average\\nchange in cellular composition from primary to recurrent tumors for each patient. Rows are individual patients that are ordered by the\\nhierarchical clustering of their change in TiME cellular composition (averaged across regions). Columns are the cell types used as clustering\\nfeatures. Compositional change was normalized [−1,1] before clustering (see“Methods” section). Leftmost two columns are color coded by\\npatient followed by therapy administered.d Box plot showing the average change in density of each cell type for each patient (n = 9) colored\\nby short-term or long-term progression free survival, splitting on median progression-free survival. Boxes= Q1 to Q3; whiskers= smallest and\\nlargest datapoints within 1.5*IQR+/− Q3/Q1; solid line= median; dotted line= mean.\\nKE Blise et al.\\n5\\nPublished in partnership with The Hormel Institute, University of Minnesota npj Precision Oncology (2022)    10 ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='cdb9b29a-7596-4fbd-bb3f-8597ea18c280', embedding=None, metadata={'page_label': '6', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='response to— or despite— increased neoplastic tumor cell density.\\nAltogether, these results indicate that shared trends in TiME\\ncomposition changes from primary to recurrent tumors speciﬁct o\\ntherapy exist, and regardless of therapy, all patients exhibited a\\ndecrease in B cells from primary to recurrent tumors.\\nApproximately half ( n = 4) of the patients in the cohort\\nexperienced an increase in CD8+ T cell density while the other\\nhalf ( n = 5) experienced a decrease in CD8 + T cell density\\nfollowing therapy. This was the only cell type that increased in\\ndensity for nearly half of the cohort and decreased for the other\\nhalf. To determine if there was a survival advantage for patients\\nthat experienced this increase, we split our cohort into short-term\\nor long-term survivor groups using median PFS and observed that\\nall patients, who experienced an increase in CD8+ T cell density\\nfrom their primary to recurrent tumors were long-term survivors\\n(Fig. 2d). Interestingly, the density of CD8+ T cells in the primary\\na\\ncb\\nd\\n1P 1R 2P 2R 3P 3R 4P 4R 5P 5R 6P 6R 7P 7R 8P 8R 9P 9R\\n0\\n0.5\\n1.0\\n1.5\\n2.0\\n2.5\\n3.0\\n3.5\\nTumor\\nMixing Score\\nCompartmentalized\\nCompartmentalized\\nMixed\\nCold\\nCompartmentalized\\nMixed\\nMixed\\nCompartmentalized\\nCompartmentalized\\nMixed\\nMixed\\nMixed\\nMixed\\ndexi\\nM\\nCompartmentalized\\nMixed\\nMixed\\nMixed\\nCompartmentalized\\nMixed\\nCold\\nOral Cavity Oropharynx\\nLarynx\\n0\\n0.5\\n1\\n1.5\\n2\\n2.5\\n3\\nAnatomic Site\\nMixing Score\\nCD8+ T Cell CD4 + T Helper B Cell\\n0\\n20\\n40\\n60\\n80\\nCell Phenotype\\n% of Cells PD-1+\\nDays\\n1400120010008006004002000\\nProgression Free Survival (%)0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0 Compartmentalized\\nMixed\\np = 0.032\\nPC1\\n-2.0 -1.0 0.0 1.0 2.0 3.0\\nPC2\\n2.5\\n2.0\\n1.0\\n0.0\\n-1.0\\n-0.5\\n1.5\\n0.5\\nCompartmentalized\\nMixed\\nCold\\nCompartmentalized\\nMixed\\ng ih\\nfe\\n0\\n1000\\n2000\\n3000\\n4000\\n5000\\n6000\\nCD8\\n+ T Cell\\nCD4\\n+ T Helper\\nCD4\\n+ Regulatory T Cell\\nB Cell MacrophageGranulocyteAntigen Presenting CellOther ImmuneαSMA\\n+ Mesenchymal\\nNeoplastic T\\numor\\nCell Phenotype\\nDensity (cells/mm2)\\nCompartmentalized\\nMixed\\nCompartmentalized\\nMixed\\n0\\n5\\n10\\n15\\nSpatial Architecture\\n% of APCs Ki-67+\\n9P0\\n100\\n200\\n300\\n400\\n500\\n7P\\n9R\\n4P\\n3R\\n7P\\n4P\\n7R\\n9P\\n9R7P\\n1R\\n6P\\n7R\\n2P\\n7R\\n9R\\n1R4R\\n5P\\n2R\\n5R\\n5R\\n1R\\n9P\\n8R\\n8R\\n3R\\n3P\\n8P\\n6P\\n3P\\n5R5P\\n4R\\n6R\\n8P\\n4R\\n2P\\n1P\\n2P\\n3P8P\\n8R\\n1P\\n1P\\n5P\\nRegion\\nαSMA+ Mesenchymal\\nCell Density (cells/mm2) Compartmentalized\\nMixed\\nCold\\np = 0.016\\np = 0.016\\np = 0.048\\np = 0.037\\np = 0.048\\np < 0.001\\np = 0.002\\np = 0.023\\np = 0.017\\np = 0.016 p = 0.032\\nMixed\\nCompartmentalized\\nImmune\\nImmune\\nNeoplastic\\nNeoplastic\\nKE Blise et al.\\n6\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='700bbd9b-69bf-4769-9368-f1d9c25f5ab1', embedding=None, metadata={'page_label': '7', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='tumor alone did not associate with PFS (p = 0.829). Prior research\\nhas revealed that increased CD8+ T cell abundance in the primary\\ntumor is associated with better outcomes in HNSCC29–33. However,\\nthese studies largely included HPV-positive HNSCCs, which is more\\noften associated with greater densities of CD8+ T cells and\\nimproved survival22–24, thus unsurprising that our results differ.\\nHowever, our results are concordant with a recent study in HNSCC\\nthat reported longer survival was associated with patients who\\nhad experienced an increase in CD8+ tumor-inﬁltrating lympho-\\ncytes from their primary to recurrent tumor34. Another study in\\nHNSCC found a similar trend between increased CD8 + T cell\\ninﬁltration, longer survival, presence of speciﬁc neoantigens, and\\nincreased cytolytic activity in recurrent tumors35. Notably, the four\\npatients in our cohort that experienced the greatest decrease in\\nCD8+ T cell density in recurrence had TNM stage 4 primary\\ntumors, while patients that experienced an increase in CD8+ T cell\\ndensity in recurrence included TNM stages 1, 2, 3, and 4.\\nQuantifying the spatial organization of neoplastic and\\nimmune cells\\nPrior studies reported the abundance of various cell types,\\nincluding CD8+ T cells33, CD4+ regulatory T cells36, and macro-\\nphages37 to be associated with survival in HNSCC. We analyzed\\nthe average density of each cell type across primary tumors for\\ntheir correlations with PFS, but found no signiﬁcant association\\nwith PFS for any single cell type (p > 0.159). Given the prognostic\\npotential of TiME cellular spatial organization as has been reported\\nin other cancer types5–13, we quantiﬁed the spatial organization of\\ncells within tumor regions and examined the association of the\\nspatial features with clinical outcome.\\nWe ﬁrst deployed a mixing score, used previously to analyze\\nimmune cell spatial compartmentalization in triple negative breast\\ncancers4. The mixing score measures the enrichment of neoplastic\\ntumor-immune cell proximity relative to immune-immune cell\\nproximity within a set distance. We quantiﬁed the number of\\nimmune and neoplastic tumor cells within 15 µm of each other,\\ndivided by the number of immune cells within 15 µm from\\nanother immune cell. Each region was labeled as mixed or\\ncompartmentalized using the median mixing score value for all\\nprimary tumors as the threshold (Fig.3a, b; see“Methods”section).\\nThis threshold classiﬁed tumor regions as mixed if at least one\\nneoplastic tumor cell was within 15 µm from an immune cell for\\napproximately every ten immune cells within 15 µm from another\\nimmune cell. Tumor regions were considered compartmentalized\\nwhen this ratio was smaller. Regions with fewer than 250 CD45+\\nimmune cells per 8002 µm2 present were labeled as cold, utilizing\\nthe same immune cell density threshold from the original study4.\\nOf the 47 total tumor regions, 25 were mixed, 20 were\\ncompartmentalized, and two were cold.\\nAfter calculating the mixing score for each region, we examined\\nthe spatial heterogeneity of our cohort. Our tumor compositional\\nheterogeneity analyses revealed that intra-tumoral and intra-\\npatient heterogeneity was less than inter-patient heterogeneity.\\nTo determine whether this observation held for spatial organiza-\\ntion heterogeneity, we compared each region’s mixing score to\\nﬁve groups of average mixing scores, which were computed from\\nthe sameﬁve groups as the analysis in Fig.1d: Intra-tumor (P or R\\nonly), Intra-patient (P and R), Inter-patient (P or R only), Inter-\\npatient (Same Anatomic Site), and Inter-patient (all). Contrary to\\nour analysis in Fig.1d, the mixing score is only one feature, not a\\ndistribution of features, thus we used the difference in mixing\\nscores, rather than the KL divergence. In addition, we calculated\\nthe absolute values of these differences as a way to normalize the\\ndata in order to capture the degree of difference in spatial\\norganization, allowing us to subsequently test for differences\\nacross the ﬁve levels of heterogeneity. We found there to be less\\nintra-tumoral heterogeneity than inter-patient heterogeneity\\n(Supplementary Fig. 2a). This result indicates that, in terms of\\nneoplastic tumor-immune cell mixing, tumor regions resemble\\nregions sampled from the same tumor more than regions sampled\\nfrom tumors of other patients.\\nWe then considered whether tumor regions sampled from the\\nsame anatomic site differed in their spatial organization and found\\nthat of the three anatomic sites, tumor regions from the oral cavity\\ncontained signiﬁcantly different average mixing scores than tumor\\nregions from the larynx ( p = 0.016, Fig. 3c). No signi ﬁcant\\ndifferences were found in average mixing scores between the\\noral cavity and the oropharynx or the larynx and the oropharynx.\\nRegions from larynx tumors did exhibit a greater range of mixing\\nscores than oral cavity or oropharynx (Fig. 3c) indicating that\\nlarynx tumors exhibit greater spatial heterogeneity in terms of\\nneoplastic tumor-immune cell proximity. We found no signiﬁcant\\ndifference in the mixing scores of primary versus recurrent tumors\\n(Wilcoxon signed rank test,p = 0.441).\\nFig. 3 Mixing score quantiﬁes the spatial organization of tumors. amIHC images of a representative mixed tumor region (top) versus a\\ncompartmentalized tumor region (bottom). Leftmost panel shows tumor regions with neoplastic tumor cells (purple) and CD45+ immune\\ncells (green); white scale bar= 200 µm. Remaining panels show zoomed in areas of mixing (top) and compartmentalization (bottom),ﬁrst with\\nboth cell populations present and then separated; white scale bar= 100 µm.b Box plot showing the mixing scores across all primary and\\nrecurrent tumors (n = 18). Each dot (n = 47) represents the mixing score for one tumor region and is colored according to its spatial\\narchitecture. The average spatial architecture designation for the overall tumor is printed above each box. Boxes= Q1 to Q3; whiskers=\\nsmallest and largest datapoints within 1.5*IQR+/− Q3/Q1; solid line= median; dotted line= mean. c Box plot showing the mixing score of\\neach tumor region split by the anatomic site of its resection. Boxes= Q1 to Q3; whiskers= smallest and largest datapoints within 1.5*IQR+/−\\nQ3/Q1; solid line= median. p-value calculated using a one-way ANOVA multi-group signiﬁcance test followed by a Tukey honestly signiﬁcant\\ndifference post-hoc test. d Kaplan– Meier curve of progression free survival for patients split by the mixing score of their primary tumors.\\nPatients were split on the median value. p-value calculated using the log-rank test. e Principal component analysis on cellular density\\nfollowing a log10 + 1 transformation. Each dot (n = 47) represents one tumor region and is colored according to the region’s spatial\\narchitecture. f Box plot showing the density of each cell type split by the tumor region’s spatial architecture. Each dot represents the density of\\nthat cell type for one region (n = 47 per cell type). Boxes= Q1 to Q3; whiskers= smallest and largest datapoints within 1.5*IQR+/− Q3/Q1;\\nsolid line = median; dotted line = mean. Statistical signiﬁcance calculated using independent one-tailed t-tests for cell types whose\\ndifferences follow a normal distribution and non-parametric one-tailedt-tests (Mann– Whitney U-test) for cell types whose differences do not\\nfollow a normal distribution.P-values were corrected using the Benjamini– Hochberg procedure. g PD-1 expression on CD8+ T cells, CD4+ T\\nhelper cells, and B cells by spatial architecture. Each dot represents the percentage of each cell type positive for PD-1 for a single tumor region\\n(n = 45, excluding cold regions). Boxes= Q1 to Q3; whiskers= smallest and largest datapoints within 1.5*IQR+/− Q3/Q1; solid line= median.\\nP-values calculated using a one-tailed Mann– Whitney U-test and corrected using the Benjamini– Hochberg procedure.h Ki-67 expression on\\nAPCs by spatial architecture. Each dot represents the percentage of cells positive for Ki-67 for a single tumor region (n = 45, excluding cold\\nregions). Boxes = Q1 to Q3; whiskers= smallest and largest datapoints within 1.5*IQR+ /− Q3/Q1; solid line= median. P-value calculated\\nusing a Mann– Whitney U-test and corrected across all cell types using the Benjamini– Hochberg procedure.i Bar chart showing the density of\\nαSMA+ mesenchymal cells present per tumor region. Bars are ordered byαSMA+ cell density and are colored according to the region’s spatial\\narchitecture.\\nKE Blise et al.\\n7\\nPublished in partnership with The Hormel Institute, University of Minnesota npj Precision Oncology (2022)    10 ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='66da7be7-9a53-4e8d-8fc9-45d9a8b69997', embedding=None, metadata={'page_label': '8', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Spatial compartmentalization associated with longer\\nprogression free survival\\nTo investigate how spatial mixing correlated with patient\\noutcome, we averaged the mixing scores across regions and\\nassigned a ﬁnal mixing score and spatial label for each tumor.\\nPatients with more compartmentalization between neoplastic\\ncells and immune cells in their primary tumors exhibited\\nsigniﬁcantly longer PFS than those with greater mixing between\\nthese cell types (p = 0.032, Fig. 3d). We then examined how the\\naverage mixing score of the tumors related to the TNM stage and\\na\\nc\\nb\\nd\\ngf\\ne\\n12345670\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nCluster\\nFraction Present\\nCD8+ T Cell\\nB Cell\\nMacrophage\\nGranulocyte\\nOther Immune\\nαSMA+ Mesenchymal\\nNeoplastic Tumor\\nAntigen Presenting Cell\\nCD4+ T Helper \\nCD4+ Regulatory T Cell\\n1 20\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0 7\\n6\\n5\\n4\\n3\\n2\\n1\\nGroup\\nFraction Present\\nGroup 1\\nGroup 2\\np = 0.047\\nProgression Free Survival (%)\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nDays\\n1400120010008006004002000\\n168342795\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nPrimary Tumor\\nFraction Present\\n7\\n6\\n5\\n4\\n3\\n2\\n1\\nPatient 1 Patient 8 Patient 4 Patient 7\\nGroup 1 Group 2\\n- 1\\nGroup -\\n- 1.50\\n- 0.75\\n- 1.25\\n- 1.00\\n- 0.50\\n- 0.25\\n12 3 456 7\\nCluster\\nPatient ID -\\n- 9\\n- 2\\n- 5\\n- 6\\n- 8\\n- 7\\n- 3\\n- 4\\nPatient\\nKE Blise et al.\\n8\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='eec2487b-a244-4b7e-82aa-ca4daa1c56cb', embedding=None, metadata={'page_label': '9', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='anatomic site of each primary tumor. Four of the ﬁve mixed\\nprimary tumors were TNM stage 4, and one was TNM stage 1. The\\nfour compartmentalized primary tumors were TNM stages 1, 2, 3,\\nand 4. All anatomic sites were present in both mixed and\\ncompartmentalized spatial architecture groups (mixed: 2 oral\\ncavity, 1 oropharynx, 2 larynx; compartmentalized: 2 oral cavity, 1\\noropharynx, 1 larynx), indicating no single anatomic site had\\npredominantly mixed or compartmentalized spatial architecture.\\nSpatial architecture associated with cellular composition\\nWe next explored how cellular composition related to spatial\\norganization, in an effort to explain the association found between\\nmixing score and PFS. By coloring the initial PCA on TiME\\ncomposition by mixing score, we found that tumor regions from\\nthe same mixing group clustered together (Fig.3e). Due to this\\nassociation between cellular composition and spatial organization,\\nwe wondered whether certain cell types would be more frequent\\nin mixed or compartmentalized tumors. Given how the mixing\\nscore is calculated, we hypothesized that compartmentalized\\ntumor regions would have greater densities of immune cells than\\nmixed regions, while mixed tumor regions would have greater\\ndensities of neoplastic tumor cells than compartmentalized\\nregions, and found this to be the case (Fig. 3f). Namely,\\ncompartmentalized tumor regions had greater densities of CD4+\\nT helper cells (p = 0.016), B cells (p = 0.048), antigen presenting\\ncells (APCs) (p = 0.037), and other CD45+ immune cells (p = 0.048)\\nthan mixed tumor regions. Given the role of CD4+ T helper cells, B\\ncells, and other MHCII+ immune cells in antigen presentation,\\nthese results could indicate enhanced antigen presentation in\\ncompartmentalized tumor regions as compared to mixed regions.\\nConversely, mixed tumor regions contained greater densities of\\nneoplastic tumor cells than compartmentalized regions (p <\\n0.001).\\nWe further examined associations between functional pheno-\\ntypes of leukocytes and mixing scores for each tumor region.\\nMixed regions contained a greater proportion of lymphocytes,\\nincluding CD8+ T cells (p = 0.023), CD4+ T helper cells (p = 0.017),\\nand B cells (p = 0.016), expressing the immunoregulatory protein\\nprogrammed death ligand (PD)-1 than compartmentalized regions\\n(Fig. 3g). PD-1 is recognized as an indicator of T cell antigen\\nexperience, whereas its expression on B cells has been reported to\\nsuppress T cell effector function38,39, indicating a suppressive and\\npotentially dysfunctional immune environment in mixed tumors.\\nOn the contrary, compartmentalized regions contained a greater\\nproportion of APCs expressing the proliferation marker Ki-67 than\\nmixed regions (p = 0.032, Fig. 3h), supporting the notion that\\nantigen presentation is a key feature of compartmentalized\\nregions. We performed a bootstrapping analysis to conﬁrm the\\nrobustness of these results and demonstrate that no one tumor\\nwas biasing the functional marker results (Supplementary Fig.\\n2b–e).\\nIn addition to identifying differences between spatial architec-\\ntures and their respective immune and neoplastic tumor cell\\ndensities and functional marker expressions, we found that tumor\\nregions with more compartmentalization between neoplastic cells\\nand immune cells also contained greater densities of αSMA+\\nmesenchymal cells, as compared to those with higher mixing (p =\\n0.002, Fig. 3f). This result was intriguing because these cells were\\nnot included when computing the mixing score, yet there is a\\nclear association betweenαSMA+ cell density and the tumor’s\\nspatial organization (Fig.3i).\\nαSMA+ mesenchymal spatial cellular neighborhoods reveal\\nspatial landscapes associated with progression free survival\\nadvantage\\nGiven the relationship between mixing score and αSMA+\\nmesenchymal cell density, we deployed a cellular neighborhood\\nclustering analysis to identify which cell types were spatially\\nproximal to αSMA+ cells across tumors in order to gain a better\\nunderstanding of whether these cells were contributing to the\\nneoplastic tumor-immune spatial compartmentalization observed.\\nThis analysisﬁrst involved calculating neighborhoods, which were\\ndeﬁned as physical groupings of cells within a set distance\\nthreshold from a seed cell (Fig.4a). Each cell within the distance\\nthreshold was deemed a neighbor of the seed cell, contributing to\\nthat seed cell ’s neighborhood’s composition. After identifying\\nneighborhoods for each seed cell present across all tumor regions,\\nneighborhoods were grouped with K-means clustering, using the\\nfraction of each cell type present in the neighborhoods as the\\nclustering features. This revealed clusters ofαSMA+ cell neighbor-\\nhoods with similar cellular makeups across all tumor regions.\\nWe applied a neighborhood clustering analysis with αSMA+\\ncells as the seed cells with a distance threshold of 30 µm, as this\\nproduced neighborhoods with an average of approximately ten\\nneighbor cells. A recent study involving cellular neighborhood\\nanalyses employed a method that selected the ten nearest spatial\\nneighbors of the seed cell, regardless of the distance between the\\nseed cell and its neighbors7. By setting a distance threshold of\\n30 µm, our method required the cells be close, if not directly\\ntouching, while still capturing enough neighbors to cluster on.\\nAny αSMA+ cell that did not have any neighbors was removed\\nfrom downstream clustering analyses. Clustering results yielded\\nseven groups, each different in their average composition of\\nαSMA+ neighborhoods (Fig. 4b). Clusters 1 and 2 contained\\nmostly other αSMA+ cells comprising the neighborhoods; in fact,\\ncluster 1 was almost exclusively made up ofαSMA+ cells. Cluster 3\\ncontained the greatest proportion of neoplastic tumor cell\\nneighbors. Clusters 4, 5, 6, and 7 were all comprised of roughly\\n75% immune cells as neighbors, although they differed in the\\nFig. 4 αSMA+ mesenchymal cellular neighborhood clustering. aCellular neighborhoods were deﬁned by drawing a circle of a speciﬁed\\nradius around each seed cell (green) of a designated phenotype. Cells whose centers were inside the circle were considered neighbors of that\\nseed cell. This ﬁgure was created using BioRender.com. b Stacked bar chart showing the average cellular composition of eachαSMA+\\nmesenchymal cell neighborhood cluster (n = 7). Bars are colored by cell type and represent the average fraction (out of 1.0) of each cell type\\npresent in the neighborhoods belonging to each cluster.c Heatmap of αSMA+ cell neighborhood clusters present averaged across primary\\ntumors. Rows are primary tumors that are ordered by the hierarchical clustering of their average ofαSMA+ neighborhood cluster presence.\\nColumns are theαSMA+ cell neighborhood clusters used as clustering features. Percent (out of 100) ofαSMA+ neighborhood clusters was\\nnormalized using a log10+ 1 transformation before clustering. Leftmost column is color coded by patient.d Stacked bar chart showing the\\naverage proportion (out of 1.0) ofαSMA+ cell neighborhood clusters present in each of the two hierarchically clustered groups of primary\\ntumors. e Panel of mIHC images containing two merged pseudo-colored images of one region from four patients visually illustrates\\nrepresentative regions from patients in group 1 and group 2. The top panel of images are merged pseudo-colored stains containing CD3, CD8,\\nCD20,αSMA, and PANCK. The bottom panel of images are the same regions as the top panel with merged pseudo-colored stains containing\\nMHCII, CD66B, Ki-67,αSMA, and PANCK. White arrows point out representativeαSMA+ cells. White scale bar= 200 µm.f Kaplan– Meier curve\\nof progression free survival for patients grouped together in the hierarchical clustering of their primary tumorαSMA+ mesenchymal cell\\nneighborhood abundance. P-value calculated using the log-rank test.g Stacked bar chart showing the average proportion (out of 1.0) of\\nαSMA+ cell neighborhood clusters present in each of the primary tumors (n = 9). Tumors are ordered by descending proportions of cluster 1.\\nKE Blise et al.\\n9\\nPublished in partnership with The Hormel Institute, University of Minnesota npj Precision Oncology (2022)    10 ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='52fb714a-ec83-4d97-9248-211fd8fd291a', embedding=None, metadata={'page_label': '10', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='types of immune cells present. Cluster 4 was deﬁned by a majority\\nof other CD45+ immune cells not explicitly deﬁned within our\\ngating strategy. To elucidate marker expression within these other\\nCD45+ cells, we performed a post-hoc t-distributed stochastic\\nneighbor embedding (t-SNE) analysis on cells classiﬁed as “other\\nimmune”, and found them to contain a large population of\\nCD163+ cells and a smaller population of mast cells (tryptase+)\\n(Supplementary Fig. 3a). Cluster 5 contained primarily CD8+ T cells\\nas the dominant immune population. Cluster 6 consisted of CD4+\\nT helper cells, B cells, and the greatest proportion of APCs of any\\ncluster. Finally, cluster 7 was deﬁned by its large proportion of\\ngranulocytes.\\nWe conﬁrmed that no single tumor region, entire tumor, or\\npatient dominated any of theαSMA+ neighborhood clusters by\\nexamining the percent contribution of each of the seven clusters\\nper 47 tumor regions, eighteen tumors, and nine patients. While\\nclusters were present in varying degrees across tumor regions, our\\nresults indicated that no region (Supplementary Fig. 3b), tumor\\n(Supplementary Fig. 3c), or patient (Supplementary Fig. 3d) was\\nsolely responsible for giving rise to any of the clusters. We also\\nconﬁrmed that all seven clusters were present in tumors collected\\nfrom each of the three anatomic sites (Supplementary Fig. 3e), as\\nwell as each of the four TNM stages (Supplementary Fig. 3f).\\nTo identify groups of patients with primary tumors of similar\\nαSMA+ cellular neighborhoods, we performed unsupervised\\nhierarchical clustering on the normalized averageαSMA+ cellular\\nneighborhood composition across the nine primary tumors. This\\nresulted in two groups of patients, differing in proportional\\ncompositions of αSMA+ cellular neighborhoods (Fig. 4c). On\\naverage, both groups had roughly 20% of their αSMA+ cells\\nassigned to cluster 2 and roughly 10% of their αSMA+ cells\\nassigned to cluster 5 (Fig.4d). However, the two groups differed in\\nthat group 1 (blue) included patients with αSMA+ cells\\npredominantly assigned to clusters 1 and 6, meaning their\\nαSMA+ cells were primarily surrounded by CD4+ T helper cells,\\nB cells, and otherαSMA+ cells. On the contrary, group 2 (orange)\\nincluded patients with more of their αSMA+ cells assigned to\\nclusters 3, 4, and 7, meaning their αSMA+ cells were mostly\\nsurrounded by neoplastic tumor cells, other immune cells, and\\ngranulocytes. Consistent with our results, visualization of tissue\\nregions illustrates primary tumors in group 1 containing more\\nαSMA+ stromal cells overall, frequently neighboring CD4 + T\\nhelper cells and B cells, with greater MHCII positivity (Fig. 4e).\\nPrimary tumors in group 2 contained less structured αSMA+\\nstromal cells, fewer neighboring immune cells, and less MHCII\\npositivity, differences that were strikingly apparent in the mIHC\\nstained tissue images (Fig.4e). Despite both groups containing\\nnearly equal proportions of CD8+ T cells neighboringαSMA+ cells,\\ngroup 1 contained higher densities of CD4+ T cells and increased\\nMHCII positivity as compared to group 2 (Fig.4e).\\nTo determine if the composition ofαSMA+ cellular neighbor-\\nhood groups was correlated with clinical outcome, we performed\\na survival analysis on the two groups of patients. We found that\\npatients in group 1 had signiﬁcantly longer PFS than patients in\\ngroup 2 (p = 0.047, Fig.4f). Patients in group 1 had primary tumors\\nannotated as TNM stages 1, 2, 3, and 4, while patients in group 2\\nhad four primary tumors annotated as TNM stage 4 and one\\nannotated as TNM stage 1. Tumors from all three anatomic sites\\nwere represented in both groups. An analysis of the proportions of\\nαSMA+ neighborhood clusters present in each of the nine primary\\ntumors revealed the varying degrees to which each of the seven\\nclusters were present in each of the tumors (Fig.4g). Notably, we\\nfound there to be a positive correlation between the presence of\\nclusters 1 and 6 (r =+ 0.69) as well as clusters 4 and 7 (r =+ 0.66).\\nWe found negative correlations between the presence of clusters\\n1 and 4 (r = −0.88), clusters 3 and 6 (r = −0.69), and clusters 6 and\\n7( r = −0.65). Finally, we found the two groups resulting from\\nhierarchical clustering to be associated with mixing status.\\nSpeciﬁcally, group 1 consisted of 75% compartmentalized tumors,\\nand group 2 consisted of 80% mixed tumors. Overall, these results\\ndescribe interesting spatial relationships between immune cells,\\nmesenchymal stroma, and neoplastic tumor cells, indicating\\nincreased antigen presentation and immune activity associated\\nwith compartmentalization and progression-free survival.\\nDISCUSSION\\nThe signiﬁcant role that TiME cellular composition plays in tumor\\nprogression and response to therapy has been accepted for over a\\ndecade40. However, recent ﬁndings powered by single-cell\\nproteomics imaging technologies have found that the spatial\\norganization of the cells present in the TiME also plays a critical\\nrole4–13. TiME spatial quantiﬁcations are just beginning to provide\\nnovel insights into tumor biology, and thus it is still unclear exactly\\nwhich spatial features are important in dictating response to\\ntherapy or clinical outcome, as well as whether these features are\\nshared across cancer types, and how they could be leveraged for\\ntherapeutic decisions and patient stratiﬁcation. Here, we lever-\\naged single-cell spatial proteomics data generated by our mIHC\\nimmunoassay-based imaging platform to quantitatively assess the\\nTiME of nine matched primary and recurrent HPV(−) HNSCCs in\\norder to demonstrate the use of spatial features in disease\\nprognosis. Our results on this cohort of nine patients provide\\ninsight into the heterogeneity and spatial landscape of HPV(−)\\nHNSCCs, and highlight possible TiME spatial landscapes that may\\nimpact clinical outcome across cancer types.\\nWe found concordance between the results from our hetero-\\ngeneity and composition analyses and those from several other\\nstudies. Our results are similar to a study that clustered HNSCC\\nbiopsies based on their neoplastic and immune gene signatures as\\ndetermined by RNA-sequencing, and reported that samples from\\nthe same patient were more similar to each other than samples\\nfrom different patients41. This trend has also been observed across\\ncancer types, including melanoma42, hepatocellular carcinoma43,\\npancreatic ductal adenocarcinoma44, and breast cancer45. More-\\nover, we found that patients who experienced an increase in CD8+\\nT cell density from their primary to recurrent tumors were\\nassociated with improved PFS, which has previously been\\nreported in head and neck cancer 34,35. Conversely, patients\\nexperiencing the greatest decrease in CD8 + T cell density in\\nrecurrence had TNM stage 4 primary tumors. This could indicate\\nthat later staged primary tumors are better equipped to evade\\nimmune attack in recurrence, and a therapy to elicit an anti-tumor\\nimmune response may be beneﬁcial.\\nDespite the fact that HPV( −) HNSCCs often contain limited\\nimmune inﬁltrates22, our analyses revealed signiﬁcant differences\\nin the immune cell spatial organization within these tumors that\\nwere associated with progression-free survival, highlighting the\\nimportance of considering the spatial context of the TiME. Most\\nstrikingly, patients whose primary tumors contained more\\ncompartmentalization between their neoplastic tumor cell and\\nimmune cell populations demonstrated longer PFS. This correla-\\ntion was also identiﬁed in a similar analysis of triple negative\\nbreast cancer patients 4, indicating that a compartmentalized\\nspatial architecture may play a favorable role in survival across\\ncancer types. Our analyses of TiME composition and functional\\nmarker expression indicate there is likely more antigen presenta-\\ntion and less immunoregulation in regions of compartmentaliza-\\ntion rather than in regions of mixing. This favorable immune\\nlandscape in compartmentalized tumors may contribute to\\nimproved survival. On the contrary, four of theﬁve of the mixed\\nprimary tumors were classi ﬁed as TNM stage 4, which may\\ncontribute to shortened survival for these patients. However,\\ngiven that compartmentalized tumors were not all classiﬁed as an\\nearly TNM stage and instead ranged from TNM stage 1 to 4, it is\\ndifﬁcult to determine the exact association between neoplastic\\nKE Blise et al.\\n10\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f4ae064e-0b08-4096-a57f-3ee3de632eb7', embedding=None, metadata={'page_label': '11', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='tumor and immune cell mixing versus TNM stage, and further\\ninvestigation is warranted to understand these correlations and\\nimpact on prognosis.\\nPrior studies have identi ﬁed a mesenchymal HNSCC sub-\\ntype46,47; our αSMA+ mesenchymal cellular neighborhood ana-\\nlyses highlight the importance of considering how these cells are\\norganized within the TiME, beyond simply considering their\\npresence in the tumor. This could reveal more precise mesench-\\nymal subtypes for improved stratiﬁcation for patient care.αSMA is\\na common marker for cancer-associated ﬁbroblasts (CAFs)20,48,\\nwhose presence in tumors, including HNSCC, tends to be\\nassociated with tumor progression, metastasis, and resistance to\\ntherapy49–54. However, despite compartmentalized tumors having\\nincreased αSMA+ cell density, patients with compartmentalized\\nprimary tumors demonstrated longer PFS, which indicates that the\\nspatial organization ofαSMA+ cells may be related to their\\nfunction.\\nWe foundαSMA+ cells neighboring immune cells in both mixed\\nand compartmentalized primary tumors, but the types of immune\\ncells nearαSMA+ cells differed. This is interesting, as emerging\\nresearch has found that CAFs, which are often deﬁned by their\\nexpression of αSMA, can modulate immune cell function within\\nthe TiME. With our mIHC platform we were able to identify the\\ndifferences in the types of immune cells neighboringαSMA+ cells\\nand relate these differences to survival, supporting recent research\\non the impact of CAFs on various immune cell populations. A\\nrecent study in melanoma found CAFs to be instrumental in aiding\\ntertiary lymphoid structure (TLS) development55; TLS are deﬁned\\nby their large concentration of B cells surrounded by T cells, and\\ntheir presence in tumors is associated with improved patient\\noutcome3,56,57. In compartmentalized tumors, we noticed the\\nαSMA+ cells to be more structured and located neighboring dense\\npockets of T cells and B cells, perhaps indicating the formation of\\nTLS. Although this antibody panels did not include a biomarker for\\nhigh endothelial venules to conﬁrm presence of TLS, our results\\nsupport the notion that CAF-lymphocyte interactions can be\\nclinically beneﬁcial.\\nConversely, patients with predominantly mixed primary tumors\\nand shorter PFS contained αSMA+ cells primarily located near\\ngranulocytes and other CD45+ immune cells, many of which were\\nlikely CD163+ myelomonocytic cells or mast cells. CD163 is a\\nmarker for scavenger receptor activity and is commonly used to\\ndemark pro-tumor type tumor-associated macrophages and\\nmonocytes. Supporting theseﬁndings, a study in oral squamous\\ncell carcinoma found the presence of CAFs to be associated with\\nincreased presence of CD163+ macrophages and worse survival58.\\nAnother study reported CAFs to be correlated with an increase in\\nmonocyte expression of CD163, which in turn suppressed T cell\\nproliferation and increased neoplastic cell proliferation in breast\\ncancer59. Mast cells mediate innate and acquired immune\\nresponse as a part of the myeloid lineage. They have been\\nreported to facilitate neo-vascularization and tumor dissemination\\nin HNSCC, and found to be correlated with increased angiogenesis\\nin advanced HNSCC60. However, interactions between mast cells\\nand CAFs is largely unknown; our results indicate this interaction\\nmay be associated with increased neoplastic density and worse\\nPFS. Of note, a recent study in melanoma and pancreatic\\nadenocarcinoma found neutrophils, a subclass of granulocytes,\\nto exert pro-tumor effects when in the presence of CAFs61, and\\nanother study found the combined presence of neutrophils and\\nCAFs to be associated with shortened survival in gastric\\nadenocarcinoma62.\\nFinally, patients with predominantly mixed primary tumors and\\nshorter PFS were found to contain fewer overall stromal cells and\\ngreater proportions of thoseαSMA+ cells neighboring neoplastic\\ntumor cells. It has been reported that neoplastic tumor cells in\\ndirect contact with CAFs move along tracks laid by the CAFs in the\\nextracellular matrix, promoting tumor growth63,64. It is possible\\nthat αSMA+ cells in mixed tumors may provide avenues for\\nneoplastic tumor cells to transit, thus leading to more advanced\\ntumor progression. Overall, spatial analyses herein deepened our\\nunderstanding of neoplastic tumor and immune cell organization\\nrelative to each other, and how this organization is related to\\nαSMA+ cells working in tandem with many cells, beyond single\\ncell–cell interactions, to impact TiME organization, and ultimately,\\nclinical outcome. The conclusions from our spatial analyses and\\ntheir relationship to clinical variables are summarized in Fig.5.\\nThe concordance found among results herein and those of the\\naforementioned studies provides orthogonal support for our\\nconclusions and indicates that single-cell heterogeneity and\\nMixed\\nMore neoplastic tumor cells\\nShorter PFS\\nLater TNM stage\\nαSMA+ cells neighboring\\nneoplastic tumor cells\\nαSMA+ cells neighboring\\ngranulocytes\\nαSMA+ cells neighboring\\nother immune cells\\nMore PD-1+ lymphocytes\\na b\\nMore immune cells involved\\nin antigen presentation\\nMore αSMA+ cells\\nLonger PFS\\nαSMA+ cells neighboring\\nαSMA+ cells\\nαSMA+ cells neighboring\\nCD4+ T helper cells and B cells\\nMore Ki-67+ APCs\\nCompartmentalized\\nFig. 5 Proposed model of primary HPV(−) HNSCC tumor-immune microenvironments. aDepiction of a tumor with a compartmentalized\\nspatial architecture. These tumors have decreased mixing between immune cells and neoplastic tumor cells and tend to contain greater\\nimmune cell density, speciﬁcally cells involved in antigen presentation, as well as increased density ofαSMA+ mesenchymal cells. TheαSMA+\\ncells present tend to be neighbored by CD4+ T helper cells and B cells, as well as otherαSMA+ cells. Compartmentalized primary tumors were\\nfound to be associated with longer progression-free survival.b Depiction of a tumor with a mixed spatial architecture. These tumors contain\\nincreased mixing between immune cells and neoplastic tumor cells and tend to contain increased neoplastic tumor cells, as well as increased\\nPD-1-positive lymphocytes. TheαSMA+ cells present tend to be neighbored by neoplastic tumor cells, other immune cells, many of which are\\nCD163+ or are mast cells, and granulocytes. Mixed primary tumors were found to be associated with shorter progression free survival and\\nlater TNM stage. Thisﬁgure was created using BioRender.com.\\nKE Blise et al.\\n11\\nPublished in partnership with The Hormel Institute, University of Minnesota npj Precision Oncology (2022)    10 ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3c124017-fc08-4dd9-9154-2136f2aaeb6c', embedding=None, metadata={'page_label': '12', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='spatial organization of tumors may share similarities across\\ndifferent types of cancer and across different molecular assays.\\nOur spatial analyses demonstrate the use of various algorithms to\\nquantify the spatial landscape of tumors using single-cell imaging\\ndata. Further, our computational methods provide a framework for\\nfuture single-cell imaging analyses, as they are applicable to any\\nmultiplex imaging assay, including mIHC, co-detection by index-\\ning technology (CODEX)7, cyclic immunoﬂuorescence18,65,66, and\\nimaging mass cytometry6. Our results highlight several spatial\\narchitectures that may help guide precision medicine approaches\\nfor HPV(−) HNSCC patients, including architectures that may help\\nstratify patients who may have shorter PFS, and thus warrant more\\naggressive therapy or clinical follow-ups. While our results provide\\nevidence that the spatial organization of HPV(−) HNSCC tumors\\ncorrelates with clinical outcome, future studies with larger cohorts\\nwill be needed to evaluate the strength and validity of our\\nobservations. Studies with greater representation of anatomic site\\nand stage are also needed to assess the prognostic value of the\\nspatial features identiﬁed in this HNSCC cohort. Despite these\\nlimitations, this study demonstrates practical analysis strategies\\nthat elucidate spatial architecture features for potential use in\\nprecision medicine.\\nMETHODS\\nMultiplex immunohistochemistry data generation\\nmIHC is an immunohistochemical-based imaging platform that evaluates\\nsequentially stained immune lineage epitope-speci ﬁc antibodies for\\nimmunodetection on FFPE tissue sections16,17. Images were stained and\\nprocessed as described in our previous report17. A table of antibodies,\\nspecies, vendor, and concentration used in staining are previously reported\\nin Table 1 of Banik et al.17. Brieﬂy, sequentially stained images were co-\\nregistered in MATLAB. AEC signal from each antibody stain was extracted\\nand normalized, and the mean intensity of each single cell for each marker\\nwas quantiﬁed in Cell Proﬁler. Watershed based nuclei segmentation on\\nhematoxylin staining was used to identify single cells in FIJI. Using a\\nhierarchical gating strategy, single cells were phenotyped using image\\ngating cytometry in FCS Express 7 Image Cytometry RUO (Supplementary\\nFig. 1a). A threshold was set on the scatterplot of mean intensity for each\\nmarker within the gating strategy, validated by visual live rendering of\\nmasked cell objects within the selected gate on extracted marker signals.\\nCartesian coordinates of each phenotyped cell were maintained relative to\\nthe tissue region.\\nWe applied the mIHC pipeline to analyze matched primary and recurrent\\nFFPE tissue specimens from nine HPV(−) HNSCC patients. Each patient\\nunderwent surgical resection of their primary tumor prior to beginning a\\nregimen of chemotherapy and radiation therapy. Upon recurrence, the\\npatient underwent another surgical resection of their recurrent tumor.\\nTissue specimens for each patient were obtained from the Oregon Health\\n& Science University Knight Biolibrary and were deidentiﬁed and coded\\nwith a unique identiﬁer prior to analysis. Patient demographic and clinical\\ndata including HPV status, tobacco and alcohol use, treatment regimens,\\nand survival outcomes were collected. All HNSCC tumors were staged\\naccording to the 8th edition AJCC/UIC TNM classiﬁcation and cohort\\ncharacteristics are shown in Table1 as reported in Banik et al17. All studies\\ninvolving human tissue were approved by institutional IRB (protocol #809\\nand #3609), and written informed consent was obtained.\\nTumor heterogeneity analyses\\nThe Kullback-Leibler divergence was calculated using the entropy function\\nfrom the Scipy Python package67. The distribution of each individual tumor\\nregion’s cellular composition was compared to the average ofﬁve different\\ncellular composition distributions: the average distribution for the region’s\\ntumor, the average distribution for the patient’s primary and recurrent\\ntumors, the average distribution across all tumors of the same timepoint in\\nthe cohort (primary or recurrent), the average distribution across all tumors\\nresected from the same anatomic site in the cohort, and the average\\ndistribution across all tumors from all patients in the cohort (primary and\\nrecurrent combined). Log base 2 was used for the calculation.\\nUnsupervised hierarchical clustering of each tumor region was performed\\nusing the log10+ 1 normalized density of each cell type present as the\\nfeatures. Euclidean distance was used to determine distances between\\nobservations, and the Ward method was used for the linkage.\\nTiME compositional change clustering analysis\\nUnsupervised hierarchical clustering of each patient was performed using\\nthe normalized change in density of each cell type as the features.\\nEuclidean distance was used to determine distances between observa-\\ntions, and the Ward method was used for the linkage. Normalization in the\\nchange in density was computing byﬁrst calculating the absolute value of\\nthe raw change in density for each cell type. These values were then\\nnormalized to a range of [0, 1]. Finally, the values that were originally\\nnegative (decreasing change), wereﬂipped to be negative values again,\\nsuch that all values ranged from [−1, 1] with zero representing no change.\\nMixing score analysis\\nKeren et al. developed a mixing score to quantify the ratio of neoplastic\\nand immune cell spatial interactions4. This score is deﬁned as the number\\nof interactions between neoplastic tumor cells and immune cells divided\\nby the number of interactions between immune cells and another immune\\ncell within a tumor region. We deﬁned there to be an interaction between\\ntwo cells if their centers [the (x, y) coordinates provided by mIHC] were\\nwithin 15 µm from one another. We used the median mixing score value\\n(0.107) for all primary tumors as the threshold to distinguish between\\nmixed and compartmentalized spatial organization groups. Tumor regions\\nwith a mixing score of greater than 0.107 were deﬁned as mixed. Tumor\\nregions with a mixing score of less than 0.107 were de ﬁned as\\ncompartmentalized. We used a density threshold of less than 250 immune\\ncells per 8002 µm2 to deﬁne tumor regions as cold. We chose this threshold\\nto match that used by Keren et al.\\nFunctional marker bootstrapping analyses\\nBootstrapping analyses involved the following steps. First, one tumor\\nregion per eighteen tumor samples was randomly selected. The regions\\nwere then split into two groups based on their mixed or compartmenta-\\nlized spatial architecture designation, and the average proportion of the\\nspeciﬁed cell population expressing the speciﬁed functional marker was\\ncalculated for each group. Finally, this process was repeated 100 times,\\nyielding 100 values representing the proportion of the speciﬁed cell\\npopulation expressing the speciﬁed functional marker for mixed tumor\\nregions and 100 values representing the proportion of the speciﬁed cell\\npopulation expressing the speciﬁed functional marker for compartmenta-\\nlized tumor regions. These values were then compared between the mixed\\nand compartmentalized groups for differences.\\nCellular neighborhood clustering analyses\\nCellular neighborhoods were deﬁned by drawing a circle of a speciﬁed\\nradius around all seed cells of a given phenotype. Cells whose centers were\\ninside the circle were considered neighbors of that seed cell and\\ncontributed to that cell ’s neighborhood. All neighborhoods across all\\ntumor regions were then clustered using scikit-learn’s MiniBatchKMeans\\nfunction68 to perform K-means clustering according to their normalized\\ncellular composition. The elbow method was used to determine the\\nnumber of clusters to form. Each resulting cluster was comprised of\\nneighborhoods with a similar cellular makeup. Unsupervised hierarchical\\nclustering of each primary tumor was performed using the log10+ 1\\nnormalized proportion (out of 100) ofαSMA+ cell neighborhood clusters\\npresent in the tumor as the features. Euclidean distance was used to\\ndetermine distances between observations, and the Ward method was\\nused for the linkage.\\nStatistics\\nIndependent t-tests were used to determine statistically signi ﬁcant\\ndifferences for independent samples whose differences followed a normal\\ndistribution. Mann–Whitney U-tests were used to determine statistically\\nsigniﬁcant differences for independent samples whose differences did not\\nfollow a normal distribution. Paired t-tests were used to determine\\nstatistically signiﬁcant differences for paired samples whose differences\\nfollowed a normal distribution. Wilcoxon signed rank tests were used to\\ndetermine statistically signiﬁcant differences for paired samples whose\\ndifferences did not follow a normal distribution. One-way ANOVA tests\\nwere used to determine statistically signiﬁcant differences for multi-group\\nKE Blise et al.\\n12\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='aace794a-0567-452d-813b-bb3ed175e53f', embedding=None, metadata={'page_label': '13', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='comparisons. If the ANOVA result was signi ﬁcant, a Tukey honestly\\nsigniﬁcant difference post-hoc test was conducted to determine which\\ngroups were signi ﬁcantly different from one another. A\\nBenjamini–Hochberg correction was used to account for multiple\\nhypothesis testing in analyses that involved systematically testing multiple\\nvariables. p-values less than 0.05 were considered statistically signiﬁcant.\\nAll statistical calculations were performed with the Scipy and statsmodels\\npackages using Python software67,69.\\nSurvival analyses\\nKaplan–Meier curves were generated and a log-rank test was performed\\nusing the lifelines package with Python software70. p-values less than 0.05\\nwere considered statistically signiﬁcant.\\nReporting summary\\nFurther information on research design is available in the Nature Research\\nReporting Summary linked to this article.\\nDATA AVAILABILITY\\nAll of the data produced by our mIHC computational image processing pipeline,\\nincluding protein abundance, cell phenotype, and cell location information saved in\\nthe form of a matrix, in addition to survival data, is available for download on Zenodo\\nathttps://doi.org/10.5281/zenodo.5540356.\\nCODE AVAILABILITY\\nAll computational analyses in this study were performed using Python software,\\nversion 3.6.5. The code created to produce the results of this study is available at\\nhttps://github.com/kblise/HNSCC_mIHC_paper.\\nReceived: 14 May 2021; Accepted: 7 January 2022;\\nREFERENCES\\n1. Binnewies, M. et al. Understanding the tumor immune microenvironment (TIME)\\nfor effective therapy.Nat. Med. 24, 541–550 (2018).\\n2. Junttila, M. R. & de Sauvage, F. J. Inﬂuence of tumour micro-environment het-\\nerogeneity on therapeutic response.Nature 501, 346–354 (2013).\\n3. Petitprez, F. et al. B cells are associated with survival and immunotherapy\\nresponse in sarcoma.Nature 577, 556–560 (2020).\\n4. Keren, L. et al. A structured tumor-immune microenvironment in triple negative\\nbreast cancer revealed by multiplexed ion beam imaging.Cell 174, 1373–1387\\n(2018).\\n5. Ali, H. R. et al. Imaging mass cytometry and multiplatform genomics deﬁne the\\nphenogenomic landscape of breast cancer.Nat. Cancer 1, 163–175 (2020).\\n6. Jackson, H. W. et al. The single-cell pathology landscape of breast cancer.Nature\\n578, 615–620 (2020).\\n7. Schurch, C. M. et al. Coordinated cellular neighborhoods orchestrate antitumoral\\nimmunity at the colorectal cancer invasive front.Cell 182, 1341–1359 (2020).\\n8. Farkkila, A. et al. Immunogenomic proﬁling determines responses to combined\\nPARP and PD-1 inhibition in ovarian cancer.Nat. Commun. 11, 1459 (2020).\\n9. Barua, S. et al. Spatial interaction of tumor cells and regulatory T cells correlates\\nwith survival in non-small cell lung cancer.Lung Cancer 117,7 3–79 (2018).\\n10. Gide, T. N. et al. Close proximity of immune and tumor cells underlies response to\\nanti-PD-1 based therapies in metastatic melanoma patients.Oncoimmunology 9,\\n1659093 (2020).\\n11. Giraldo, N. A. et al. Multidimensional, quantitative assessment of PD-1/PD-L1\\nexpression in patients with Merkel cell carcinoma and association with response\\nto pembrolizumab.J. Immunother. Cancer6, 99 (2018).\\n12. Johnson, D. B. et al. Quantitative spatial proﬁling of PD-1/PD-L1 interaction and\\nHLA-DR/IDO-1 predicts improved outcomes of anti-PD-1 therapies in metastatic\\nmelanoma. Clin. Cancer Res.24, 5250–5260 (2018).\\n13. Enﬁeld, K. S. S. et al. Hyperspectral cell sociology reveals spatial tumor-immune\\ncell interactions associated with lung cancer recurrence.J. Immunother. Cancer7,\\n13 (2019).\\n14. Galon, J. et al. Type, density, and location of immune cells within human color-\\nectal tumors predict clinical outcome.Science 313, 1960–1964 (2006).\\n15. Galon, J. et al. Cancer classiﬁcation using the Immunoscore: a worldwide task\\nforce. J. Transl. Med.10, 205 (2012).\\n16. Tsujikawa, T. et al. Quantitative multiplex immunohistochemistry reveals\\nmyeloid-inﬂamed tumor-immune complexity associated with poor orognosis.Cell\\nRep. 19, 203–217 (2017).\\n17. Banik, G. et al. High-dimensional multiplexed immunohistochemical character-\\nization of immune contexture in human cancers.Methods Enzymol. 635,1 –20\\n(2020).\\n18. Eng, J. et al. Cyclic multiplexed-immunoﬂuorescence (cmIF), a highly multiplexed\\nmethod for single-cell analysis.Methods Mol. Biol.2055, 521–562 (2020).\\n19. Brucher, B. L. & Jamall, I. S. Cell-cell communication in the tumor microenviron-\\nment, carcinogenesis, and anticancer treatment. Cell. Physiol. Biochem. 34,\\n213–243 (2014).\\n20. Alsahaﬁ, E. et al. Clinical update on head and neck cancer: molecular biology and\\nongoing challenges. Cell Death Dis.10, 540 (2019).\\n21. Cohen, E. E. W. et al. The society for immunotherapy of cancer consensus\\nstatement on immunotherapy for the treatment of squamous cell carcinoma of\\nthe head and neck (HNSCC).J. Immunother. Cancer7, 184 (2019).\\n22. Canning, M. et al. Heterogeneity of the head and neck squamous cell carcinoma\\nimmune landscape and its impact on immunotherapy.Front. Cell Dev. Biol.7,5 2\\n(2019).\\n23. Fakhry, C. et al. Improved survival of patients with human papillomavirus-positive\\nhead and neck squamous cell carcinoma in a prospective clinical trial.J. Natl\\nCancer Inst. 100, 261–269 (2008).\\n24. Ang, K. K. et al. Human papillomavirus and survival of patients with orophar-\\nyngeal cancer. N. Engl. J. Med.363,2 4–35 (2010).\\n25. Predina, J. et al. Changes in the local tumor microenvironment in recurrent\\ncancers may explain the failure of vaccines after surgery.Proc. Natl Acad. Sci. USA\\n110, E415–E424 (2013).\\n26. Zheng, B. et al. Changes in the tumor immune microenvironment in resected\\nrecurrent soft tissue sarcomas.Ann. Transl. Med.7, 387 (2019).\\n27. Walens, A. et al. Adaptation and selection shape clonal evolution of tumors\\nduring residual disease and recurrence.Nat. Commun. 11, 5017 (2020).\\n28. Watermann, C. et al. Recurrent HNSCC harbor an immunosuppressive tumor\\nimmune microenvironment suggesting successful tumor immune evasion.Clin.\\nCancer Res. 27, 632–644 (2021).\\n29. Hartman, D. J., Ahmad, F., Ferris, R. L., Rimm, D. L. & Pantanowitz, L. Utility of\\nCD8 score by automated quantitative image analysis in head and neck squamous\\ncell carcinoma. Oral. Oncol. 86, 278–287 (2018).\\n30. Balermpas, P. et al. CD8+ tumour-inﬁltrating lymphocytes in relation to HPV\\nstatus and clinical outcome in patients with head and neck cancer after post-\\noperative chemoradiotherapy: a multicentre study of the German cancer con-\\nsortium radiation oncology group (DKTK-ROG).Int. J. Cancer138, 171–181 (2016).\\n31. Nasman, A. et al. Tumor inﬁltrating CD8+ and Foxp3+ lymphocytes correlate to\\nclinical outcome and human papillomavirus (HPV) status in tonsillar cancer.PLoS\\nONE 7, e38711 (2012).\\n32. Nordfors, C. et al. CD8+ and CD4+ tumour inﬁltrating lymphocytes in relation to\\nhuman papillomavirus status and clinical outcome in tonsillar and base of tongue\\nsquamous cell carcinoma.Eur. J. Cancer49, 2522–2530 (2013).\\n33. Balermpas, P. et al. Tumour-inﬁltrating lymphocytes predict response to deﬁnitive\\nchemoradiotherapy in head and neck cancer.Br. J. Cancer110, 501–509 (2014).\\n34. So, Y. K. et al. An increase of CD8(+) T cell inﬁltration following recurrence is a\\ngood prognosticator in HNSCC.Sci. Rep. 10, 20059 (2020).\\n35. Schutt, C. R. et al. Genomic and neoantigen evolution from primary tumor toﬁrst\\nmetastases in head and neck squamous cell carcinoma.Oncotarget 12, 534–548\\n(2021).\\n36. Russell, S. et al. Immune cell inﬁltration patterns and survival in head and neck\\nsquamous cell carcinoma.Head. Neck Oncol.5, 24 (2013).\\n37. Wolf, G. T. et al. Tumor inﬁltrating lymphocytes (TIL) and prognosis in oral cavity\\nsquamous carcinoma: a preliminary study.Oral. Oncol. 51,9 0–95 (2015).\\n38. Wang, X. et al. PD-1-expressing B cells suppress CD4(+) and CD8(+) T cells via\\nPD-1/PD-L1-dependent pathway. Mol. Immunol. 109,2 0–26 (2019).\\n39. Thibult, M. L. et al. PD-1 is a novel regulator of human B-cell activation.Int.\\nImmunol. 25, 129–137 (2013).\\n40. Fridman, W. H., Pages, F., Sautes-Fridman, C. & Galon, J. The immune contexture in\\nhuman tumours: impact on clinical outcome.Nat. Rev. Cancer12,2 9 8–306 (2012).\\n41. Wood, O. et al. Head and neck squamous cell carcinomas are characterized by a\\nstable immune signature within the primary tumor over time and space.Clin.\\nCancer Res. 23, 7641–7649 (2017).\\n42. Reuben, A. et al. Genomic and immune heterogeneity are associated with dif-\\nferential responses to therapy in melanoma.NPJ Genom Med2,1 –11 (2017).\\n43. Zhang, Q. et al. Integrated multiomic analysis reveals comprehensive tumour\\nheterogeneity and novel immunophenotypic classi ﬁcation in hepatocellular\\ncarcinomas. Gut 68, 2019–2031 (2019).\\n44. Lin, W. et al. Single-cell transcriptome analysis of tumor and stromal compart-\\nments of pancreatic ductal adenocarcinoma primary tumors and metastatic\\nlesions. Genome Med. 12, 80 (2020).\\nKE Blise et al.\\n13\\nPublished in partnership with The Hormel Institute, University of Minnesota npj Precision Oncology (2022)    10 ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='412e2bba-758a-4e8f-945b-5dcf7b6a7cc4', embedding=None, metadata={'page_label': '14', 'file_name': 'blise_head_and_neck.pdf', 'file_path': '/content/data/blise_head_and_neck.pdf', 'file_type': 'application/pdf', 'file_size': 2534163, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='45. Gyanchandani, R. et al. Intratumor heterogeneity affects gene expression proﬁle\\ntest prognostic risk stratiﬁcation in early breast cancer. Clin. Cancer Res. 22,\\n5362–5369 (2016).\\n46. Chung, C. H. et al. Molecular classiﬁcation of head and neck squamous cell\\ncarcinomas using patterns of gene expression.Cancer Cell 5, 489–500 (2004).\\n47. Walter, V. et al. Molecular subtypes in head and neck cancer exhibit distinct\\npatterns of chromosomal gain and loss of canonical cancer genes.PLoS ONE 8,\\ne56823 (2013).\\n48. Parajuli, H. et al. Integrin alpha11 is overexpressed by tumour stroma of head and\\nneck squamous cell carcinoma and correlates positively with alpha smooth\\nmuscle actin expression.J. Oral. Pathol. Med.46, 267–275 (2017).\\n49. Zhang, Q. et al. Integrated analysis of single-cell RNA-seq and bulk RNA-seq\\nreveals distinct cancer-associated ﬁbroblasts in head and neck squamous cell\\ncarcinoma. Ann. Transl. Med.9, 1017 (2021).\\n50. Lim, K. P. et al. Fibroblast gene expression proﬁle reﬂects the stage of tumour\\nprogression in oral squamous cell carcinoma.J. Pathol. 223, 459–469 (2011).\\n51. Shiga, K. et al. Cancer-associatedﬁbroblasts: their characteristics and their roles in\\ntumor growth. Cancers 7, 2443–2458 (2015).\\n52. Tao, L., Huang, G., Song, H., Chen, Y. & Chen, L. Cancer associatedﬁbroblasts:\\nan essential role in the tumor microenvironment. Oncol. Lett. 14, 2611–2620\\n(2017).\\n53. Sahai, E. et al. A framework for advancing our understanding of cancer-associated\\nﬁbroblasts. Nat. Rev. Cancer20, 174–186 (2020).\\n54. Johansson, A. C. et al. Cancer-associated ﬁbroblasts induce matrix\\nmetalloproteinase-mediated cetuximab resistance in head and neck squamous\\ncell carcinoma cells.Mol. Cancer Res.10, 1158–1168 (2012).\\n55. Rodriguez, A. B. et al. Immune mechanisms orchestrate tertiary lymphoid struc-\\ntures in tumors via cancer-associatedﬁbroblasts. Cell Rep. 36, 109422 (2021).\\n56. Cabrita, R. et al. Tertiary lymphoid structures improve immunotherapy and sur-\\nvival in melanoma.Nature 577, 561–565 (2020).\\n57. Helmink, B. A. et al. B cells and tertiary lymphoid structures promote immu-\\nnotherapy response. Nature 577, 549–555 (2020).\\n58. Fujii, N. et al. Cancer-associatedﬁbroblasts and CD163-positive macrophages in\\noral squamous cell carcinoma: their clinicopathological and prognostic sig-\\nniﬁcance. J. Oral. Pathol. Med.41, 444–451 (2012).\\n59. Gok Yavuz, B. et al. Cancer associatedﬁbroblasts sculpt tumour microenviron-\\nment by recruiting monocytes and inducing immunosuppressive PD-1(+) TAMs.\\nSci. Rep. 9, 3172 (2019).\\n60. Markwell, S. M. & Weed, S. A. Tumor and stromal-based contributions to head and\\nneck squamous cell carcinoma invasion.Cancers 7, 382–406 (2015).\\n61. Munir, H. et al. Stromal-driven and Amyloid beta-dependent induction of\\nneutrophil extracellular traps modulates tumor growth. Nat. Commun. 12,\\n683 (2021).\\n62. Cong, X. et al. CD66b(+) neutrophils and alpha-SMA(+) ﬁbroblasts predict clinical\\noutcomes and bene ﬁts from postoperative chemotherapy in gastric adeno-\\ncarcinoma. Cancer Med. 9, 2761–2773 (2020).\\n63. Gaggioli, C. et al. Fibroblast-led collective invasion of carcinoma cells with dif-\\nfering roles for RhoGTPases in leading and following cells. Nat. Cell Biol. 9,\\n1392–1400 (2007).\\n64. Goetz, J. G. et al. Biomechanical remodeling of the microenvironment by\\nstromal caveolin-1 favors tumo r invasion and metastasis. Cell 146, 148–163\\n(2011).\\n65. Lin, J. R., Fallahi-Sichani, M. & Sorger, P. K. Highly multiplexed imaging of single\\ncells using a high-throughput cyclic immunoﬂuorescence method.Nat. Commun.\\n6, 8390 (2015).\\n66. Lin, J. R. et al. Highly multiplexed immunoﬂuorescence imaging of human tissues\\nand tumors using t-CyCIF and conventional optical microscopes.Elife 7, e31657\\n(2018).\\n67. Virtanen, P. et al. SciPy 1.0: fundamental algorithms for scientiﬁc computing in\\nPython. Nat. Methods 17, 261–272 (2020).\\n68. Pedregosa, F. et al. Scikit-learn: machine learning in python.JMLR 12, 2825–2830\\n(2011).\\n69. Seabold, S. & Perktold, J. statsmodels: econometric and statistical modeling with\\npython. In Proceedings of the 9th Python in Science Conference (2010). https://\\nwww.statsmodels.org/stable/index.html#citation.\\n70. Davidson-Pilon, C. lifelines: survival analysis in Python.J. Open Source Softw.4,\\n1317 (2019).\\nACKNOWLEDGEMENTS\\nThe authors thank Dr. Courtney Betts, Justin Tibbitts, Teresa Beechwood, and Meghan\\nLavoie for regulatory and technical assistance. The authors thank the OHSU Knight\\nBiostatistics Shared Resource for guidance with statistical calculations. The authors\\nalso acknowledge and thank the patients who donated tissue samples for this study.\\nK.E.B. acknowledges funding from the National Cancer Institute (NCI) of the National\\nInstitutes of Health under award number T32CA254888. L.M.C. acknowledges funding\\nfrom the NCI awards U01CA224012, R01CA223150, R01CA226909, and R21HD099367\\nand funding from the Brenden-Colson Center for Pancreatic Care at OHSU. J.G.\\nacknowledges funding under award number U24CA231877. L.M.C. and J.G.\\nacknowledge funding from the OHSU Knight Cancer Institute, NCI award number\\nU2CCA233280, and the Prospect Creek Foundation to the OHSU SMMART (Serial\\nMeasurement of Molecular and Architectural Responses to Therapy) Program.\\nAUTHOR CONTRIBUTIONS\\nK.E.B., S.S., L.M.C., and J.G. conceived the project and designed the overall study. G.L.\\nB. collected the data using the multiplex immunohistochemistry experimental assay\\nand provided clinical expertise. S.S. performed computational analyses for image\\nprocessing to produce the single-cell spatial dataset that was analyzed in the study.\\nK.E.B. designed the computational experiments, developed software for the study,\\nand analyzed all results with support from S.S. K.E.B., S.S., L.M.C., and J.G. wrote the\\noriginal and revised manuscript. L.M.C. and J.G. supervised the project. All authors\\nread, edited, and approved theﬁnal manuscript.\\nCOMPETING INTERESTS\\nL.M. Coussens is a paid consultant for Cell Signaling Technologies, AbbVie Inc., and\\nShasqi Inc., received reagent and/or research support from Plexxikon Inc.,\\nPharmacyclics, Inc., Acerta Pharma, LLC, Deciphera Pharmaceuticals, LLC, Genentech,\\nInc., Roche Glycart AG, Syndax Pharmaceuticals Inc., Innate Pharma, NanoString\\nTechnologies, and Cell Signaling Technologies, is a member of the Scientiﬁc Advisory\\nBoards of Syndax Pharmaceuticals, Carisma Therapeutics, Zymeworks, Inc, Verseau\\nTherapeutics, Cytomix Therapeutics, Inc., Hibercell., Inc., Alkermes, Inc., and Kineta Inc,\\nand is a member of the Lustgarten Therapeutics Advisory working group. No\\ncompeting interests were disclosed by the other authors.\\nADDITIONAL INFORMATION\\nSupplementary information The online version contains supplementary material\\navailable at https://doi.org/10.1038/s41698-022-00253-z.\\nCorrespondence and requests for materials should be addressed to Jeremy Goecks.\\nReprints and permission information is available at http://www.nature.com/\\nreprints\\nPublisher’s noteSpringer Nature remains neutral with regard to jurisdictional claims\\nin published maps and institutional afﬁliations.\\nOpen Access This article is licensed under a Creative Commons\\nAttribution 4.0 International License, which permits use, sharing,\\nadaptation, distribution and reproduction in any medium or format, as long as you give\\nappropriate credit to the original author(s) and the source, provide a link to the Creative\\nCommons license, and indicate if changes were made. The images or other third party\\nmaterial in this article are included in the article’s Creative Commons license, unless\\nindicated otherwise in a credit line to the material. If material is not included in the\\narticle’s Creative Commons license and your intended use is not permitted by statutory\\nregulation or exceeds the permitted use, you will need to obtain permission directly\\nfrom the copyright holder. To view a copy of this license, visithttp://creativecommons.\\norg/licenses/by/4.0/.\\n© The Author(s) 2022\\nKE Blise et al.\\n14\\nnpj Precision Oncology (2022)    10 Published in partnership with The Hormel Institute, University of Minnesota', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='bb18e853-ae89-4dd0-b8ec-b09f5771dc7f', embedding=None, metadata={'page_label': '1', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 1 \\nMachine learning links T cell function and spatial localization to neoadjuvant \\nimmunotherapy and clinical outcome in pancreatic cancer \\n \\nKatie E. Blise 1,2, Shamilene Sivagnanam 2,3, Courtney B. Betts 2,3,4, Konjit Betre 3, Nell \\nKirchberger3, Benjamin Tate2,5, Emma E. Furth 6,7, Andressa Dias Costa 8, Jonathan A. \\nNowak9, Brian M. Wolpin 8, Robert H. Vonderheide 7,10, Jeremy Goecks1,2,11,12,*, Lisa M. \\nCoussens2,3,*, Katelyn T. Byrne2,3,13,* \\n \\n1Department of Biomedical Engineering, Oregon Health & Science University, Portland, \\nOR USA \\n2The Knight Cancer Institute, Oregon Health & Science University, Portland, OR USA \\n3Department of Cell, Developmental & Cancer Biology, Oregon Health & Science \\nUniversity, Portland, OR USA \\n4Current affiliation: Akoya Biosciences, 100 Campus Drive, 6 th Floor, Marlborough, MA \\nUSA \\n5Immune Monitoring and Cancer Omics Services, Oregon Health & Science University, \\nPortland, OR USA  \\n6Department of Pathology and Laboratory Medicine, Hospital of the University of \\nPennsylvania, Philadelphia, PA USA  \\n7Abramson Cancer Center, Perelman School of Medicine, University of Pennsylvania, \\nPhiladelphia, PA USA \\n8Department of Medical Oncology, Dana-Farber Cancer Institute, Harvard Medical \\nSchool, Boston, MA USA \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e70d40ec-f194-4f36-b4d9-0df0e3342e1a', embedding=None, metadata={'page_label': '2', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 2 \\n9Department of Pathology, Brigham and Women’s Hospital and Harvard Medical School, \\nBoston, MA USA \\n10Parker Institute for Cancer Immunotherapy, Perelman School of Medicine, University of \\nPennsylvania, Philadelphia, PA USA \\n11Current affiliation: Department of Machine Learning, H. Lee Moffitt Cancer Center, \\nTampa, FL USA \\n12Current affiliation: Department of Biostatistics and Bioinformatics, H. Lee Moffitt Cancer \\nCenter, Tampa, FL USA \\n13Lead contact \\n*Co-corresponding author \\n \\n*Co-corresponding authors: jeremy.goecks@moffitt.org (J.G.), coussenl@ohsu.edu \\n(L.M.C.), byrneka@ohsu.edu (K.T.B.)  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='68518bd7-8469-4e0f-a1ed-fc008cb40485', embedding=None, metadata={'page_label': '3', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 3 \\nABSTRACT \\nTumor molecular datasets are becoming increasingly complex, making it nearly \\nimpossible for humans alone to effectively analyze them. Here, we demonstrate the \\npower of using machine learning to analyze a single-cell, spatial, and highly multiplexed \\nproteomic dataset from human pancreatic cancer and reveal underlying biological \\nmechanisms that may contribute to clinical outcome. A novel multiplex \\nimmunohistochemistry antibody panel was used to audit T cell functionality and spatial \\nlocalization in resected tumors from treatment-naive patients with localized pancreatic \\nductal adenocarcinoma (PDAC) compared to a second cohort of patients treated with \\nneoadjuvant agonistic CD40 (αCD40) monoclonal antibody therapy. In total, nearly 2.5 \\nmillion cells from 306 tissue regions collected from 29 patients across both treatment \\ncohorts were assayed, and more than 1,000 tumor microenvironment (TME) features \\nwere quantified. We then trained machine learning models to accurately predict αCD40 \\ntreatment status and disease-free survival (DFS) following αCD40 therapy based upon \\nTME features. Through downstream interpretation of the machine learning models’ \\npredictions, we found αCD40 therapy to reduce canonical aspects of T cell exhaustion \\nwithin the TME, as compared to treatment-naive TMEs. Using automated clustering \\napproaches, we found improved DFS following αCD40 therapy to correlate with the \\nincreased presence of CD44+ CD4+ Th1 cells located specifically within cellular spatial \\nneighborhoods characterized by increased T cell proliferation, antigen-experience, and \\ncytotoxicity in immune aggregates. Overall, our results demonstrate the utility of machine \\nlearning in molecular cancer immunology applications, highlight the impact of αCD40 \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a3d17565-3dce-4d89-92ec-6c8450f8a44f', embedding=None, metadata={'page_label': '4', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 4 \\ntherapy on T cells within the TME, and identify potential candidate biomarkers of DFS for \\nαCD40-treated patients with PDAC. \\n \\n \\nKEYWORDS \\nTumor microenvironment, machine learning, T cell, pancreatic ductal adenocarcinoma, \\nspatial proteomics  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='431dbb1a-1709-416e-816a-256e3623c87c', embedding=None, metadata={'page_label': '5', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 5 \\nINTRODUCTION \\nPancreatic ductal adenocarcinoma (PDAC) is one of the most aggressive treatment-\\nrefractory cancers with a median overall survival rate of just months, thus there is a critical \\nneed for new treatment strategies for this disease (1-3). Recent reports reveal that \\nimmunological responses in PDAC can be temporarily induced via approaches that \\npromote priming of T cell responses against PDAC, such as occurs following agonistic \\nCD40 (αCD40) monoclonal antibodies (4) and peptide vaccination (5) strategies. We and \\nothers have previously reported that αCD40 binds to CD40 on dendritic cells (DCs), \\nthereby licensing DCs to subsequently enhance T cell activation and bolster anti-tumor \\nimmunity (6-8). However, little is known regarding selective impact of αCD40 therapy on \\nspecific T cell states within the TME and to what degree this therapy sustains T cell \\nfunctionality in situ, or instead promotes T cell dysfunction that could in turn limit potential \\nuse of αCD40 combination therapies in the clinical setting. \\n \\nUpon antigen stimulation, T cells upregulate CD44 and exist along a spectrum of diverse \\ndifferentiation states with varying functionalities (9-11). On one end of the spectrum, T \\ncells possess stem-cell-like plasticity, accompanied by memory, proliferative and cytotoxic \\ncapabilities, and are identified by expression of T-BET, and/or TCF-1 (12, 13). On the \\nother end of the spectrum, T cells are exhausted and/or dysfunctional and canonically \\nexpress TOX1 and/or EOMES (14, 15). Along the spectrum, expressed in varying \\ncombinations, are T cells expressing immune checkpoint molecules such as PD-1, LAG-\\n3 and TIM3, with increased numbers of immune checkpoint molecules correlating with \\nmore exhausted T cells (16). These partially exhausted T cells are susceptible to \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8d87cb7b-5f5a-4b14-8399-33c0df93b888', embedding=None, metadata={'page_label': '6', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 6 \\nreinvigoration by immune checkpoint blockade (ICB) and regain the ability to proliferate \\nand produce effector cytokines after PD-1 blockade, for example (17, 18). However, \\nterminally differentiated T cells expressing TOX1 or EOMES are resistant to rescue by \\nICB and cannot proliferate or exert cytotoxic activity (19). Flow cytometric analyses of T \\ncells in preclinical tumor models or tissues following viral infection have elucidated notable \\nT cell states associated with priming, proliferation, cytotoxic capability, or functional \\nexhaustion; however, characterizations of effector versus exhausted T cell phenotypes \\nfrom tumors in patients are scant. Moreover, with the advent of single-cell sequencing \\napproaches, the diversity of T cell subset possibilities within tumors is seemingly endless \\nand in need of clarification (20, 21). We sought to clarify T cell subset characteristics \\nwithin the PDAC TME, and to identify those subsets associated with response to \\ntherapeutic αCD40 therapy. Recognizing that both the cellular composition and spatial \\norganization of cells, including T cells, is a critical metric associated with therapeutic \\nresponse and clinical outcome (22-32), we investigated and report on the impact of \\nαCD40 therapy on the complex spatial contexture of T cells within the PDAC TME and \\nassociation with survival. \\n \\nWe previously developed a multiplex immunohistochemistry (mIHC) single-cell spatial \\nproteomics imaging platform to interrogate leukocyte heterogeneity and spatial landscape \\nwithin the TMEs of various cancer types (33-35). Following a cyclical staining protocol, \\nthe mIHC platform iteratively utilizes up to 30 antibodies to stain protein biomarkers on a \\nsingle tissue specimen, thus preserving spatial context of the TME. The resulting data \\nprovide single-cell resolution maps, which can be quantified by several metrics including \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3fa8b5bf-1bc4-4b9a-a66e-5c5677a43f3a', embedding=None, metadata={'page_label': '7', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 7 \\ncellular density, cell-cell spatial interactions, and cellular neighborhoods. While it is \\npossible to quantify and correlate each of these TME features individually with clinical \\noutcome, this approach fails to combine and weight features together to capture biological \\ncomplexities of the TME. Machine learning can be used to address TME complexity, as it \\nis a form of artificial intelligence whereby computational models are trained to identify \\nwhich combinations of data features can predict a given output. Models can then be used \\nto make accurate predictions from new data and interrogated to understand which \\ncombinations of features are most important in making the predictions. Machine learning \\nis becoming widely utilized in precision oncology to decipher patterns in large data sets \\nresulting from deep interrogations (36-39). \\n \\nHere, we leveraged machine learning to elucidate the frequency of various T cell states \\nin PDAC and investigated the impact of αCD40 therapy on those states. We first designed \\na novel mIHC antibody panel to deeply audit T cell functionality  and spatial organization \\nin two PDAC cohorts, one of which was treatment-naive and the other of which received \\nneoadjuvant αCD40 therapy. Using this mIHC panel, we generated a dataset of nearly \\n2.5 million cells with spatially resolved single-cell phenotypic and functional \\nmeasurements. Interrogation of this dataset presented a unique opportunity to elucidate: \\n1) the types of T cells present at baseline in a treatment-refractory disease and 2) to what \\ndegree αCD40 therapy sustains T cell functionality in situ, or instead promotes T cell \\ndysfunction that could in turn limit potential use of αCD40 or other T cell priming therapies \\nin the clinical setting. Given the vast amount of spatially resolved data and complexity of \\nT cell function in the TME, we leveraged machine learning approaches to discern new \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ad4537d4-8fe4-48cf-9974-d62813ad8888', embedding=None, metadata={'page_label': '8', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 8 \\nbiological insights regarding T cells in the pancreatic TME and their association with \\nclinical outcome for pancreatic cancer patients. \\n \\n \\n  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='983c304e-1904-4043-8219-e38236bae8b7', embedding=None, metadata={'page_label': '9', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 9 \\nMATERIALS AND METHODS \\nTissue Acquisition \\nHuman PDAC tissue specimens from cohort 1 were obtained from patients with approval \\nfrom the Oregon Pancreas Tissue Registry under Oregon Health & Science University \\nIRB protocol #3609 and Dana Farber Harvard Cancer Center protocols #03-189 and #12-\\n013 (Fig. 1A). Specimens from cohort 2 were obtained from patients treated with \\nneoadjuvant selicrelumab with approval under the IRBs of four sites across the United \\nStates involved in an open-label phase I clinical trial (Cancer Immunotherapy Trials \\nNetwork CITN11-01; NCT02588443; Fig. 1A), including 8 patients with neoadjuvant \\nselicrelumab alone and 3 patients with neoadjuvant selicrelumab combined with \\ngemcitabine and nab-paclitaxel. For the purposes of this paper, this cohort has been \\ncombined as the αCD40-treated cohort given previous analyses of the TME (4). All \\nstudies were conducted in accordance with the Declaration of Helsinki and written \\ninformed consent was obtained. \\n \\nMultiplex Immunohistochemistry Image Acquisition and Analysis \\nFormalin-fixed, paraffin-embedded (FFPE) surgical tissue samples were sectioned and \\nassessed using hematoxylin and eosin (H&E), as well as chromogen based mIHC. Using \\npathologist annotations overlaid from the H&E-stained slides, sections were assessed \\nwith regard to histopathologic regions of interest (ROI; Fig. 1B) annotated as tumor (T), \\nimmune aggregate (IA), tumor-adjacent stroma (TAS), or normal-adjacent pancreas \\n(NAP), as defined by the pathologist (35). Multiplex staining was performed on 5 µm \\nsections, as previously described, and each stained image was scanned at 20x \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='09517446-fbca-47d6-b503-171e40973978', embedding=None, metadata={'page_label': '10', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 10 \\nmagnification on an Aperio AT2 scanner (Leica Biosystems) (35). The antibody panel was \\ndesigned to delineate epithelial cells, mesenchymal cells, B cells, myeloid cells, and 18 T \\ncell subpopulations, spanning T cell activation and exhausted states. Human tonsil and \\nspleen were included in all rounds of mIHC as staining controls. Image processing was \\nperformed using previously described methods (35). Each region was registered to the \\nfinal hematoxylin using Matlab Computer Vision Toolbox (The Mathworks, Inc ., Natick, \\nMA), color deconvolution and watershed-based nuclei segmentation was performed \\nusing ImageJ, and single cell mean intensity for each stain was quantified using Cell \\nProfiler (40). Single marker positivity thresholds were set using FCS Express Image \\nCytometry RUO (De Novo Software, Glendale, CA) to visually validate protein biomarker \\nexpression overlaid on signal extracted images. Single cell classification was performed \\nusing R Statistical Software based on filtering exclusive populations in a defined \\nhierarchy. \\n \\nMachine Learning Classifiers and Feature Importance Analyses \\nElastic net (EN) classifier models were built to predict: 1) treatment status and 2) disease-\\nfree survival (DFS), using scikit-learn’s LogisticRegression function (41). Predictions were \\nmade on a region basis to maximize sample size and model robustness. Separate models \\nwere created for each histopathologic site in order to: 1) compare performance of models \\nderived from different histopathologic sites; 2) identify where therapy was potentially \\nexerting greatest impact; and 3) mitigate broad variation in average tissue area from each \\nhistopathologic site (Fig. 1B ). A leave-one-patient-out cross-validation approach was \\nused to split the train and test sets. Thus, within each cross-validation loop, a new EN \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='697449c0-ec82-4877-85c3-97dec50cd2b6', embedding=None, metadata={'page_label': '11', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 11 \\nmodel was created and trained on all regions except those from one patient, and testing \\nwas then performed on regions from the patient withheld from training. This process was \\nrepeated until all patients were cycled through the test set. This approach prevents data \\nleakage by ensuring regions from the same patient were not in both the train and test sets \\nfor one model, thus preventing the EN models from learning patient-specific features, \\nwhich often artificially increases model accuracy. Test set predictions were aggregated \\nacross all cross-validation loops to construct one final confusion matrix, from which \\nperformance of the models was assessed by calculating accuracy, F1 score, and area \\nunder the receiver operating characteristic curve (AUC). These metrics address both \\nprecision and recall (F1 score), in addition to the true positive rate and false positive rate \\n(AUC) – these are often used to assess performance of classifier models. Model \\noverfitting was mitigated by using the same model hyperparameters across cross-\\nvalidation loops. The penalty term was set to “elasticnet,” and the “l1_ratio” \\nhyperparameter was set to 0.5, representing an equal balance of the lasso model and \\nridge model effects. All features were log10+1 normalized and scaled using a minmax \\n[0,1] scaler in order to equally compare features spanning different orders of magnitude \\nand improve model interpretability . To further prevent data leakage, in each cross-\\nvalidation loop, the scaler was fit to the train set and then applied to the train set and \\nsubsequently the test set. Test feature outliers were clipped to [0,1] following this \\nnormalization. The train set was balanced within each cross-validation loop using \\nSynthetic Minority Over-sampling Technique (SMOTE) to up-sample the minority class to \\nequal the majority class (42). Feature importance analyses were conducted by computing \\nShapley Additive exPlanations (SHAP) values for each model (43). SHAP values enable \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a812c9cd-ffd9-44f6-ae5a-2e3b0ade6102', embedding=None, metadata={'page_label': '12', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 12 \\nthe interpretation of which combinations of features contribute to the overall model \\npredictions. \\n \\nCell-Cell Spatial Interaction Quantification \\nCell-cell interactions were defined as cells separated by a distance of ≤\\t20 µm, according \\nto centroid (x,y) coordinates of any two cells, enriching for cells that were directly adjacent \\nas previously reported (44-46). Total cell-cell interactions were normalized by dividing \\nsummed densities of cell states involved in the interactions to avoid skewing by cell states \\npresent in high abundances.  \\n \\nRecurrent Cellular Neighborhood Analysis \\nRecurrent cellular neighborhoods were quantified to assess spatial organization of \\ntissues. A neighborhood was created for every cell by counting all cells within a 60 µm \\nradius of each seed cell’s center, as inferred from previous studies (47-50). Using \\nproportions of cells comprising the neighborhoods as features, neighborhoods were \\ngrouped using K-means clustering. The elbow method was used to determine the number \\nof clusters, resulting in groupings of spatial neighborhoods that were similar in cellular \\ncomposition that could be found across all regions of interest in the analysis. \\n \\nStatistics \\nMann-Whitney U tests were used to determine statistically significant differences in top \\nTME features between treatment cohorts or DFS groups. The Benjamini-Hochberg \\ncorrection was used to account for multiple hypothesis testing for each analysis. P-values \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='da5d27ab-e403-470f-87de-e17cb5f08c2c', embedding=None, metadata={'page_label': '13', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 13 \\nless than 0.05 were considered statistically significant. Statistical calculations were \\nperformed with the Scipy and statsmodels packages using Python software (51, 52). \\n \\nDATA AVAILABILITY \\nmIHC data used for this study is available for download on Zenodo at \\nhttps://doi.org/10.5281/zenodo.8357193. \\n \\nCODE AVAILABILITY \\nThe code used to generate all computational results of this research was created using \\nPython version 3.9.4 and is available at https://github.com/kblise/PDAC_mIHC_paper. \\n \\n \\n  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f90b53ee-ed66-4408-bb63-25afe1aec7d5', embedding=None, metadata={'page_label': '14', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 14 \\nRESULTS \\nDesigning a novel mIHC antibody panel to deeply phenotype T cells within the \\nPDAC TME \\n29 PDAC tumors were surgically resected from patients across two treatment cohorts \\n(Fig. 1A). Tumors from cohort 1 were treatment-naive upon resection and were previously \\nevaluated for immune contexture as a subset of a larger study (n=104 tumors) designed \\nto broadly evaluate leukocyte composition in treatment-naive PDAC (35). The 18 \\ntreatment-naive tumors evaluated in the current study were selected as a representative \\nsubset of the larger PDAC cohort based on cellular subsets not being statistically different \\nfrom the full cohort in terms of leukocyte densities or patient survival durations \\n(Supplementary Table S1). Thus, PDAC specimens from cohort 1 served as a \\nrepresentative baseline comparison to the 11 specimens from cohort 2, which reflected \\npatients who had received neoadjuvant αCD40 therapy alone (n=8) or in combination with \\ngemcitabine and nab-paclitaxel (n=3) prior to resection (4). Three to 26 ROIs per PDAC \\nresection were selected by a pathologist and quantitatively assayed by mIHC, with each \\nregion annotated as one of four histopathologic sites within the resected samples: tumor \\n(T), immune aggregate (IA), tumor-adjacent stroma (TAS), or normal-adjacent pancreas \\n(NAP) (Fig. 1B) (35). Breakdown of region types assayed per patient are shown ( Fig. \\n1C). \\n \\nIn total, nearly 2.5 million cells were assayed across 306 tissue regions by our novel 21-\\nantibody mIHC panel (Fig. 1D and E, Supplementary Table S2, Supplementary Fig. \\nS1A). Given our goal of revealing T cell states present in the PDAC TME at baseline, and \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8c41e9f2-7a54-4f06-938f-37368e22a9eb', embedding=None, metadata={'page_label': '15', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 15 \\nthe impact of αCD40 therapy on those T cells, the majority of antibodies included in the \\nmIHC panel were chosen for their ability to elucidate differentiation, exhaustion, \\nproliferative and cytotox ic status of various T cell lineage subtypes as defined by prior \\nliterature (14, 53-57). First, 423,317 T cells were identified by CD3 expression and then \\nsubsequently stratified by CD8α expression. CD8+ T cells were further classified as one \\nof six cell states (T NAIVE, T EFF, T EM, T EMRA, T EX, or T TEX) (Fig. 1F). Due to biomarker \\nselection and positional restrictions within the cyclic multiplex panel, we did not include a \\nCD4 antibody . However, the majority (72%) of CD3 + CD8- T cells were CD4 + as \\ndetermined in a testing panel using a subset of the data (Supplementary Fig. S1B); we \\ntherefore refer to CD3+ CD8- T cells as CD4+ T cells herein, although it is possible other \\nminor lineages may be represented (20). Based on this schema, CD3+ CD8- T cells were \\nfurther evaluated; CD4+ Th1 cells were defined by T-BET + expression, and further \\nclassified as one of three cell states (T EFF, T EM, or T EMRA) (Fig. 1F). We originally \\nanticipated the majority of CD8 + T cells and CD4 + Th1 cells would fall into one of the \\nclearly defined states (e.g., TNAIVE, TEFF, TEM, TEMRA, TEX, TTEX); however, only 6% of CD8+ \\nT or CD4+ Th1 cells were phenotyped as one of these states (Supplementary Table S3). \\nTo investigate the remaining 94% – labeled TOTHER (Fig. 1F) – we stratified cells based \\non expression of CD44, a canonical biomarker of prior cognate antigen experience. \\nFurthermore, there was a population of T-BET- CD4+ Th cells (non-Th1-specific T helper \\ncells), which we stratified into two cell states based upon CD44 expression only (Fig. 1F). \\nFinally, the remaining CD3+ CD8- T cells were FOXP3+ TREG cells, and classified into three \\ncell states (Naive T REG, mT REG, and T-BET+ TREG) (Fig. 1F). Despite low numbers of \\ncertain T cell states (Supplementary Table S3), we included all 18 T cell states in \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4325d0dd-b223-4dde-add9-b03f72968468', embedding=None, metadata={'page_label': '16', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 16 \\ndownstream analyses based upon calculations from previous single-cell studies (58), as \\nwell as the fact that these populations were manually gated and thus represent real \\nphenotypes of T cells present in the PDAC TME. In addition, we utilized antibodies for 10 \\nT cell functionality biomarkers to assess differentiation/exhaustion status on all T cells \\n(TOX1, TIM3, TCF-1, CD38, PD-1, EOMES, CD39, CD44, LAG-3, and T-BET), as well \\nas for proliferation (Ki-67) and cytotox icity (granzyme B, GrzB) (Fig. 1D , E). Finally, the \\nremaining CD3- (non-T) cells were defined by a hierarchical gating strategy and classified \\nas B cells, myeloid cells, mesenchymal fibroblast-like cells (also referred to as \\nmesenchymal cells), or neoplastic epithelial cells (Fig. 1F, Supplementary Fig. S1C). \\nAltogether, cells were phenotyped as one of 23 different cell lineages and states ( Fig. \\n1F). \\n \\nInterrogating cell states and spatial interactions within the PDAC TME \\nUsing single-cell spatial data collected from the mIHC assay, we used three approaches \\nto quantify treatment-naive and αCD40-treated PDAC TMEs: 1) Cell state densities, to \\nidentify types of cells present (Supplementary Table S3); 2) T cell “functionality \\nbarcodes,” to reflect functional phenotypes of T cells by unique combinations of functional \\nbiomarker expression (Supplementary Table S4); and 3) Cell-cell spatial interactions, to \\nreveal relative locations between cell types (Supplementary Table S5). Using these \\nthree metrics, we created a granular map of leukocyte infiltration, T cell functionality \\nstatus, and cellular spatial orientation in the PDAC TME (Fig. 2). \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8bd4571b-523a-4058-9950-070b29e5771f', embedding=None, metadata={'page_label': '17', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 17 \\nTo identify the types and amounts of cell states present in the PDAC TME, we first \\nquantified densities of each of the 23 cell states for each tissue region assayed by dividing \\nraw counts of cells (Supplementary Table S3) by tissue area. Varying densities of \\nleukocytes, mesenchymal fibroblast-like cells, and neoplastic epithelial cells were present \\nin regions of each histopathologic annotation and each treatment cohort (Fig. 2A, top). \\nIn T regions, neoplastic epithelial cells were dominant. In IA regions, T cells and B cells \\nwere most prevalent, while in TAS regions, which encompassed the tumor borders, there \\nwas a mix of neoplastic cells, T cells, myeloid cells, and mesenchymal cells. Finally, \\nmyeloid cells were the dominant population in distal NAP regions. T cells were further \\ndivided into 18 T cell states defined by the mIHC gating strategy (Fig. 1F, white boxes). \\nOn average, CD4 + T cells were present in increased densities, nearly two-fold, as \\ncompared to CD8+ T cells (Fig. 2A, middle, bottom rows), regardless of histopathologic \\ntype or treatment cohort. However, average densities of the CD8+ and CD4+ T cell states \\noften differed by histopathologic site and treatment cohort. These results demonstrate the \\nimportance of identifying spatial and histopathological information for interpreting how – \\nand where – αCD40 therapy alters T cells in the PDAC TME, important information that \\ncould not be captured by flow cytometric methodologies. \\n \\nTo understand how T cell functionality differed based on histopathology and prior \\ntreatment, we assigned all T cell states a “Functionality Barcode,” as defined by binary \\npositive or negative expression of unique combinations of 10 T cell functionality \\nbiomarkers (Fig. 1D, middle column). Among the 423,317 T cells present in the dataset, \\nwe identified expression of 961 unique barcodes (Supplementary Table S4). The top 15 \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3903b7c3-3aa4-463b-ba66-7acdf341a115', embedding=None, metadata={'page_label': '18', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 18 \\nmost common barcodes by average density are shown for each histopathologic type and \\ntreatment cohort (Fig. 2B). Over half of the most common barcoded T cells were present \\nin both treatment cohorts, regardless of histopathologic site, as indicated by brown bars \\n(Fig. 2B). Of the barcodes present in ‘high’ densities only within the treatment-naive \\ncohort (orange bars), the majority contained two functionality biomarkers; however, of the \\nmost abundant barcodes present in the αCD40 cohort only (blue bars, Fig. 2B), the \\nmajority contained three or more functionality biomarkers. This result supports the \\nhypothesis that αCD40 therapy shifts T cell differentiation/functionality within the PDAC \\nTME, as represented by an increase in the number of functionality biomarkers expressed. \\nHowever, further elucidation of the specific patterns of biomarkers expressed by T cells \\nis needed to identify the direction of this shift following immunotherapy. \\n \\nFinally, to address spatial organization of cells in the PDAC TME , we leveraged the fact \\nthat mIHC preserves spatial context and quantified frequency of two cell states \\ninteracting, based on their cell centers being within 20 µm from each other (44-46). We \\nidentified 268 unique pairs of cell-cell interactions present in the datasets \\n(Supplementary Table S5). On average, there were increased interactions between \\nCD4+ T cells of various subpopulation states with other CD4+ T cells in the αCD40 cohort, \\nregardless of histopathologic site (Fig. 2C, white boxes). Altogether, these results \\nsupport the hypothesis that αCD40 drives an increase in CD4 + T cell density, functional \\ncapacity, and spatial proximity in the PDAC TME, as compared to treatment-naive PDAC \\nTMEs. \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='64a428b9-3a9c-4f06-9697-899db5cad9d7', embedding=None, metadata={'page_label': '19', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 19 \\nFollowing these three TME single-cell quantification methods, all 2,428,274 cells present \\nacross the 306 regions were phenotyped as one of 23 cell states. In addition, all 423,317 \\nT cells were assigned one of 961 T cell functionality barcodes. Finally, the immediate \\nspatial neighbors of each cell were computed and binned into one of 268 types of pairwise \\ncell-cell interactions. In total, 1,252 TME features were computed across the 306 regions, \\nand each region was annotated as one of four histopathologic sites. Given the complexity \\nand large amount of data, we leveraged machine learning and feature importance \\nanalyses to identify: 1) impact of αCD40 therapy on composition, functional capacity, and \\nspatial organization of cells within the PDAC TME, and 2) the likely mechanism(s) we \\nhypothesized underlying improved clinical outcome following αCD40 therapy. To do this, \\nwe trained EN classifier models to predict treatment status and DFS within the αCD40-\\ntreated cohort from the 1,252 TME features quantified above (Fig. 2D and Methods). EN \\nmodels perform well on datasets as generated herein where there are more features than \\nsamples (tissue regions) (59). This is because EN models use mathematical \\nregularization approaches to identify and select the most informative subset of features \\nto make model predictions while accounting for feature collinearity (60). Further, our \\napproach was unbiased as we did not perform any prior feature selection and instead \\nprovided all 1,252 TME features to the models, leveraging the EN algorithm’s ability to \\nperform aggressive feature selection within model training. Finally, SHAP values were \\nused to interpret cellular biology underpinning model predictions (43). SHAP values \\ndenote the relative importance of a given feature in driving a model’s prediction – this \\nmethod has been used to explain machine learning predictions in prior cancer studies \\n(38, 61, 62). \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0dec769f-be18-4947-b734-d8b2bcc3e0db', embedding=None, metadata={'page_label': '20', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 20 \\n \\nMachine learning models classify αCD40-treated TMEs as having reduced T cell \\nexhaustion phenotypes \\nTo reveal the impact of αCD40 therapy on T cell exhaustion phenotype, we first trained \\nfour EN classifier models – one per histopathologic annotation (Fig. 1B; T, IA, TAS, and \\nNAP) – to predict the treatment status of the tissue. All models, regardless of \\nhistopathologic site, performed well, as measured by the accuracy, F1 score, and area \\nunder the receiver operating characteristic curve (AUC) for test sets of each of the models \\n(Fig. 3A and B). Across the four models, accuracy ranged from 0. 83 to 0.85, F1 score \\nranged from 0.73 to 0.89, and AUC ranged from 0.87 to 0.90. As models were trained to \\ndifferentiate treatment-naive from αCD40-treated PDAC, high performance of all four \\nmodels indicates that αCD40 modulates all types of histopathologic regions across the \\nTME evaluated herein. \\n \\nTo identify which combinations of features drove model predictions, and thus reveal how \\nαCD40 therapy impacted T cells in the PDAC TME, we calculated SHAP values for each \\nof the four models (Supplementary Fig. S2A). The top 30 weighted features out of 1,252 \\nfeatures in total  accounted for the majority of feature importance according to SHAP \\nvalues (T model: 84%; IA model: 74%; TAS model: 87%; NAP model: 92%) , while \\nremaining features exhibited limited impact on the models. Comparison of the top 15 \\nfeatures driving model predictions for each of the four histopathologic models revealed \\nthat 13 features were top contributors across multiple models (Fig. 3C ), indicating some \\namount of shared T cell densities, differentiation states, and spatial organizations across \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='094b96cc-d164-4df6-97f7-acc14d406cdf', embedding=None, metadata={'page_label': '21', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 21 \\nhistopathologic sites within a given treatment cohort. To further bolster the machine \\nlearning results, we compared the normalized values for each of the top features detected \\nby SHAP analysis between treatment-naive samples versus αCD40-treated samples \\nwithin each histopathologic site ( Supplementary Fig. S2B, C, D, and E). Following \\ncorrection for multiple hypothesis testing, all 15 features derived from the T model were \\nsignificantly different between treatment cohorts, 14 of 15 features derived from the IA \\nand TAS models were significantly different between treatment cohorts, and 9 of 15 \\nfeatures derived from the NAP models were significantly different between treatment \\ncohorts. Given that NAP sites had the fewest number of ROIs present in the data, it is \\nunsurprising that fewer features were statistically significant. \\n \\nOverall, the models identified αCD40-treated TMEs as containing increased densities of \\nmesenchymal fibroblast-like cells and several T cell states, including three CD4+ T helper \\npopulations and two antigen-experienced CD8+ T cell populations (TEX and CD44+ TOTHER) \\n(Fig. 3C, Supplementary Fig. S2B, C, D, and E), as compared to treatment-naïve TMEs, \\nwhich contained increased densities of naive CD8 + T cells (although not significantly \\ndifferent) and B cells. In addition to cell state densities, analysis of T cell functionality \\nbarcodes revealed that αCD40-treated TMEs contained increased densities of T cells \\nexpressing combinations of T-BET , CD44, CD39, TIM3, and TCF-1 (Fig. 3C, \\nSupplementary Fig. S2B, C, D, and E). On the other hand, treatment-naive TMEs \\ncontained increased densities of T cells expressing combinations of TOX1 and EOMES, \\nconcordant with mIHC stained tissue images ( Fig. 3D ). Finally, spatial interactions \\ninvolving CD4 + Th1 cells were more likely identified within  the αCD40-treated cohort, \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='daf38c86-57f3-4f9b-83b5-83f9993a1867', embedding=None, metadata={'page_label': '22', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 22 \\nwhereas interactions involving myeloid cells, naive TREGs (not significantly different), and \\nKi-67+ neoplastic epithelial cells were more likely to be associated with treatment-naive \\ntissue (Fig. 3C, Supplementary Fig. S2B, C, D, and E). Altogether, these results indicate \\nαCD40 TMEs contained increased presence of T cells in close spatial proximity to one \\nanother – in particular, CD4 + T helper cells – with reduced exhaustion profiles, as \\ncompared to treatment-naive TMEs. \\n \\nOf the top features across all four models, the majority of features were densities of T cell \\nfunctionality barcodes. As all barcodes present on any T cell (regardless of the T cell \\nstate) were provided to the machine learning models, we sought to understand the types \\nof T cells expressing each of the predictive barcodes to determine whether there were \\npatterns in the types and proportions of T cell states expressing each barcode. We \\nensured that only T cells within regions of the histopathologic site the predictive model \\nwas derived from were counted, as well as from the treatment cohort the barcoded T cell \\nwas predictive of. We then correlated barcodes based off of the types and proportions of \\nT cell states expressing each barcode (Fig. 3E). Barcodes expressed by similar \\nproportions of the same T cells were highly correlated and appear adjacent to each other \\nin the heatmap. This analysis resulted in four clusters (i. – iv.) of barcodes, each distinct \\nin types of T cells expressing them, and unrelated to histopathologic site, indicating that \\nbarcodes expressed by similar T cells exist across the TME. Barcodes belonging to \\ncluster (i) were expressed by antigen-inexperienced (as defined by a lack of CD44 \\nexpression) CD8+ T cells, antigen -inexperienced CD4+ T helper cells, naive T REGs, and \\nmTREGs. However, barcodes in cluster (i) ranged from TOX1 + and EOMES + expression \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e453d3d6-638a-4002-9f4c-7629a286c35f', embedding=None, metadata={'page_label': '23', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 23 \\nprofiles when their increased densities were predictive of treatment-naive, to CD39 + \\nTIM3+, and TCF-1+ expression profiles when their increased densities were predictive of \\nαCD40 therapy. This result supports the notion that, while the same T cell types were \\npresent regardless of therapy exposure, their functional capacity differed following αCD40 \\ntreatment. Higher densities of barcodes in clusters (ii), (iii), and (iv) were all predictive of \\nαCD40 therapy and included combinations of T-BET+, TIM3+, TCF-1+, and CD44+. These \\nbarcoded cells were expressed by distinct groups of T cell states, including multiple \\nantigen-experienced CD4+ Th1 cell states, as well as T-BET + TREGs, which have been \\nreported to be similar to CD4+ Th1 cells in their function (63). \\n \\nMachine learning model classifies long disease-free survivors as having more T \\ncell effector functionality following αCD40 therapy \\nThe clinical trial from which the αCD40-treated specimens were derived was not designed \\nto assess correlates with survival. However, we hypothesized that we could train machine \\nlearning models to accurately predict DFS for these patients, with the ultimate goal of \\nidentifying which combinations of TME features were associated with long versus short \\nDFS. In contrast to the αCD40-treated patients who all received the same adjuvant \\ntreatment regimen (αCD40 therapy and chemotherapy), the patients from the treatment-\\nnaive cohort went on to receive varying adjuvant therapies. Thus, we did not train models \\nto predict DFS for the treatment-naive cohort, as identifying the underlying biology driving \\nimproved survival for these patients would be challenging. Therefore, we trained EN \\nclassifier models to predict long or short DFS for tissue regions only from patients treated \\nwith αCD40 therapy. The median DFS timepoint (9.8 months) across all patients in the \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c32d281c-01b0-4c48-b9c3-8866901dea38', embedding=None, metadata={'page_label': '24', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 24 \\nαCD40-treated cohort was used to segregate long and short disease-free survivors. \\nAgain, separate models were built for regions annotated as each histopathologic site, \\nincluding T, IA, and TAS. There were too few NAP regions to build a model assessing \\nDFS from the αCD40-treated patients for this histopathologic site. Only the model trained \\nfrom IA regions performed well in predicting both long and short DFS, with an accuracy \\nof 0.81, F1 score of 0.88, and AUC of 0.77 (Fig. 4A and B).  \\n \\nTo identify which combinations of features drove predictions for the IA-derived model, and \\nthus hypothesize potential mechanisms contributing to improved DFS following αCD40 \\ntherapy, we followed a similar model interpretation analysis as described above. SHAP \\nvalues were calculated to highlight feature importance (Supplementary Fig. S3A, Fig. \\n4C), and the top 30 features out of 1,252 features fed into the model accounted for 78% \\nof feature importance. All of the top 15 features detected by the SHAP analysis for the IA-\\nderived model were significantly different between DFS groups (Supplementary Fig. \\nS3B), thus demonstrating how machine learning can be used to reveal potential \\ncombinations of candidate biomarkers of DFS in the PDAC TME. \\n \\nOf note, an increased density of CD44+ CD4+ Th1 cells was observed in IA regions from \\npatients with long DFS, as compared to those with from patients with short DFS (Fig. 4C, \\nSupplementary Fig. S3B). In addition, we found increased spatial interactions between \\nCD4+ Th1 TEM cells or CD8+ TEMRA cells and T-BET+ TREGs to be associated with long DFS \\nrather than short DFS (Fig. 4C , Supplementary Fig. S3B). Interestingly, of the three \\ntypes of TME features, densities of T cell functionality barcodes were the most common \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c22c6a3a-d662-4d33-b9fe-0cf2cb3d3532', embedding=None, metadata={'page_label': '25', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 25 \\nfeature predictive of DFS, accounting for 12 of the top 15 features. Eight of these barcode \\ndensities were associated with short DFS, while the remaining four were associated with \\nlong DFS (Fig. 4C , Supplementary Fig. S3B). A number of biomarkers segregated \\nbased on DFS status (Fig. 4C , D); namely, TOX1 was expressed across all eight \\nbarcodes whose increased densities were associated with short DFS, and PD-1 \\nexpression was found exclusively in five of the eight barcodes whose increased densities \\nwere associated with short DFS. In contrast, expression of CD44, CD38, CD39, TIM3, \\nand LAG-3 were unique to the four barcodes whose increased densities were associated \\nwith long DFS (Fig. 4C, D). \\n \\nTo identify whether the types and amounts of T cell states expressing each of the \\npredictive barcodes were similar, we correlated the barcodes based off of the proportions \\nof T cells expressing each of them (Fig. 4E). We found four distinct clusters of barcodes, \\nand within each cluster, the barcodes were expressed by similar types of T cell states and \\nin similar proportions. All but one of the barcodes in clusters (i) and (ii) were among the \\nfeatures whose increased densities were associated with short DFS and were expressed \\nby (i) antigen-inexperienced CD8 + T cells, antigen-inexperienced CD4 + T helper cells, \\nnaive TREGs, and mTREGs; and (ii) antigen-inexperienced CD8+ T cells and CD4+ Th1 cells, \\nas well as T-BET+ TREGs. In contrast, increased densities of all barcodes in clusters (iii) \\nand (iv) were associated with long DFS. Cluster (iii) consisted of one barcode that was \\nuniquely expressed by CD8+ TEM and TEMRA cells, CD44+ CD4+ T helper cells, as well as \\nnaive TREG cells. Finally, cluster (iv) barcodes were expressed by CD44 + CD8+ T cells, \\nCD4+ Th1 TEM and TEMRA cells, CD44 + CD4+ Th1 cells, and T-BET + TREGs. In summary, \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f4e4c0c1-f11b-4662-8b2d-3f495f0086ac', embedding=None, metadata={'page_label': '26', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 26 \\nthese findings indicate αCD40 therapy stimulates an anti -tumor T cell response, \\ncharacterized by presence of CD44 + T cells, in particular CD4 + Th1 cells, and is \\nassociated with prolonged DFS time within IAs. \\n \\nCellular neighborhood analysis identifies spatial organization of T cells to correlate \\nwith DFS following αCD40 therapy \\nThe majority of top TME feature types driving the αCD40 DFS IA model predictions were \\npresence of specific barcoded T cells, supporting the notion that these T cell subsets likely \\nplay a major role in extending DFS. However, the spatial organization of the specific \\nbarcoded T cells identified by the machine learning analyses remained unknown, as the \\nspatial interaction features did not account for T cell functionality barcodes. TME spatial \\narchitecture is associated with clinical outcomes across cancer types (24-27, 31); thus, \\nwe aimed to identify the spatial neighbors of the top barcoded T cells whose densities \\nwere associated with DFS to better understand their potential role in prolonging DFS \\nfollowing αCD40 therapy within IA regions. \\n \\nTo quantify the spatial organization of the tissue, we performed a recurrent cellular \\nneighborhood (RCN) analysis across all IA regions within the αCD40-treated cohort (26, \\n64). Here, we established a 60 µm radius (47-50) around every cell as its neighborhood \\nand identified all cells within that neighborhood (Fig. 5A). We then clustered all cells \\naccording to the cellular neighborhoods’ compositions (Methods). This resulted in seven \\nRCNs (Supplementary Fig. S4A) – each one representing spatial neighborhoods \\npresent across multiple IAs that were distinct in proportions and types of cell states \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='58f6ba9b-6b41-484f-b9d6-d9fdd85c73d1', embedding=None, metadata={'page_label': '27', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 27 \\nlocated within the neighborhood. The average cellular composition of each RCN is shown \\n(Fig. 5B). We confirmed that no single RCN dominated the IA regions analyzed \\n(Supplementary Fig. S4B and C) and that no RCN was exclusively derived from any \\nsingle IA region or patient (Supplementary Fig. S4D and E). Upon viewing the scatterplot \\nreconstructions of regions, clearly defined spatial patterns within IAs were revealed (Fig. \\n5C, Supplementary Fig. S4F). For example, cells in RCN1, whose neighborhood \\nconsisted of majority B cells, were often found to be spatially clustered together, \\npotentially representing “germinal center”-like pockets within IA regions.  \\n \\nGiven that our goal was to identify neighbors surrounding the top barcoded T cells from \\nthe αCD40 DFS model, we first identified which of the seven RCNs the barcoded T cells \\nwere assigned to. We then assessed whether there was a relationship between spatial \\norganization of the barcoded T cells and clinical outcome by comparing the RCNs of \\nbarcoded T cells from patients with long DFS versus the RCNs of barcoded T cells from \\npatients with short DFS. We performed unsupervised clustering of barcoded T cells from \\nlong and short DFS patients together based on proportions of the seven RCNs the \\nbarcoded T cells resided in (Fig. 5D). This resulted in two distinct clusters based on which \\nRCNs the barcoded T cells resided in; the clusters also separated barcoded T cells from \\npatients associated with long DFS versus short DFS. Cluster 1 (C1) included all barcoded \\nT cells from patients with long DFS, as well as CD44 + T-BET+ barcoded T cells from \\npatients with short DFS. Cluster 2 (C2) consisted of all remaining barcoded T cells from \\npatients with short DFS.  \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ad230f94-837c-4bf2-948a-52a53de6e2d1', embedding=None, metadata={'page_label': '28', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 28 \\nThe most striking difference between the two clusters was proportions of barcoded T cells \\nresiding in RCN5 ( Fig. 5E). Of all RCNs, RCN5 contained the greatest proportion of \\nCD44+ T cells, spanning both CD8+ T cells and CD4+ T helper lineages (Fig. 5B). 45% of \\nT cells in C1 (long DFS) resided in RCN5, whereas only 3% of T cells from C2 (short DFS) \\nresided in RCN5 (Fig. 5E). Given that C1 consisted of T cells from patients with long DFS \\nand C2 consisted of T cells from patients with short DFS, our results highlight a \\nrelationship between cellular spatial organization and clinical outcome. Specifically, T \\ncells correlated with long DFS were frequently found to be surrounded by CD44+ T cells, \\nthus supporting the hypothesis that αCD40 therapy promotes T cell priming and/or \\nrecruitment of primed T cells to PDAC TMEs. \\n \\nFinally, to further elucidate potential cellular mechanisms at play within various RCNs, we \\ncalculated proportions of T cells expressing Ki-67 or GrzB in each RCN and compared \\nvalues to the overall proportion of Ki-67+ or GrzB + T cells across all αCD40 IA regions. \\nGiven our prior findings that the majority of T cell barcodes whose increased density \\ncorrelated with long DFS were assigned to RCN5 (Fig. 5E), we hypothesized that T cells \\nassigned to RCN5 would possess increased proliferative and/or cytotoxic capabilities. We \\nfound a larger proportion of T cells expressing Ki-67 residing in RCN1 and RCN5 as \\ncompared to the overall T cell population (dashed line) (Fig. 5F). In addition, we found a \\nlarger proportion of T cells (excluding T REGs) expressing GrzB residing in RCN5, RCN6, \\nand RCN7, as compared to the overall non-TREG T cell population (dashed line) (Fig. 5G). \\nHowever, raw counts of GrzB + T cells in RCN6 and RCN7 were low (RCN6: n=8 cells; \\nRCN7: n=11 cells), whereas RCN5 contained the highest level of GrzB+ T cells across all \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4b497675-976f-447a-b0e6-752932b16778', embedding=None, metadata={'page_label': '29', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 29 \\nRCNs (n=168). Visualization of a representative mIHC image depicts presence of GrzB+ \\nT cells localized to RCN5 in an IA region from a patient with long DFS (Fig. 5H ). \\nCollectively, these results provide further support that T cells within RCN5 are likely \\nactivated and possess an effector phenotype capable of an anti-tumor cytotoxic response. \\n \\n \\n  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='415713d8-0e5f-426f-b8d3-0c22be25a74e', embedding=None, metadata={'page_label': '30', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 30 \\nDISCUSSION \\nIn this study, we advance spatial proteomic imaging technology and leverage machine \\nlearning approaches to understand the role of T cell phenotypes and spatial organization \\nin the complex TME of human pancreatic cancer. In contrast to previous single-cell spatial \\nproteomic studies, which often group T cells together as broad populations, such as CD8+ \\nT cells, CD4 + T cells, and T REGs (23, 25, 65-67), our novel mIHC panel was curated to \\nphenotype T cells as one of 18 distinct states along with functionality status from 10 \\ndifferent markers, all while preserving the spatial orientation of each cell in the TME. In \\nconsidering the full spectrum of T cell states in addition to their spatial organization within \\nthe TME, the data became increasingly complex, highlighting the need for quantitative \\nanalyses that consider integration of all features together, rather than one-off correlations \\nwhich fail to represent intertwining biological mechanisms at play. We leveraged machine \\nlearning, as the models detect combinations and weights of tissue features associated \\nwith a given output, such as treatment status or DFS, and thus it is an appropriate and \\npowerful tool for interrogating TME complexity. Through our machine learning approaches \\nwe identified TME features most associated with αCD40 therapy exposure or prolonged \\nDFS out of the >1000 features that we measured. This study demonstrates the value of \\nmerging single-cell spatial proteomic assays with machine learning analyses to \\ninterrogate how immunotherapy modulates the PDAC TME and drives improved survival \\nfrom a holistic perspective. \\n \\nDespite the multitude of unique T cell states identified herein, our machine learning \\nmodels were able to identify T cell subsets associated with anti-tumor characteristics and \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='35a8e27e-8500-4010-909c-c82c4f660a08', embedding=None, metadata={'page_label': '31', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 31 \\nprolonged DFS. Consistent with our preclinical studies revealing that CD4 + T cells are a \\nmajor contributor to PDAC immunity following αCD40 therapy (8), our machine learning \\nmodels revealed that αCD40-treated patient tumors contain increased densities of \\neffector memory cells specifically within the CD4+ Th1 lineage. Along the T cell spectrum, \\nTEM cells are fully functional with some self-renewal capacities (68-70), and mediate \\npotent anti -tumor immunity (71). No increase was observed in CD8 + memory T cell \\npopulations, supporting the notion that αCD40 therapy promotes immunological memory \\nfrom the CD4+ T cell lineage in both mouse and human PDAC tissue. Our models also \\nidentified antigen-experienced CD4+ Th1 helper cells as the main cell type whose density \\nassociated with prolonged DFS following αCD40 therapy. This result is supported by two \\nindependent studies, including characterization of immune cells in biopsied PDAC liver \\nmetastases following CD40 agonism (72), and the second investigating primary resected \\nPDAC TMEs after treatment with a granulocyte-macrophage colony-stimulating factor-\\nsecreting allogenic PDAC vaccine (GVAX) (rather than αCD40 therapy) (73). Both studies \\nreported that presence of CD4 + T helper cells contribute to improved survival following \\nimmunotherapy in PDAC (72, 73). Here, we further characterized expression features of \\nthe CD4+ T helper cells as CD44 + Th1 cells, which correlated with improved outcomes. \\nReports have also highlighted the direct role of CD4+ T cells in mediating anti-tumor \\nimmunity, including via cytotoxicity (74) and production of effector cytokines like \\ninterferon-γ and tumor necrosis factor-α (75, 76). As such, we hypothesize that future \\ntherapies designed to harness effector and memory functions of CD4 + T helper cells \\nfollowing administration of αCD40 therapy may be clinically beneficial. \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='213f073f-2885-4080-b36f-bdd408633408', embedding=None, metadata={'page_label': '32', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 32 \\nIn addition to investigating presence of various T cell states, our machine learning \\nanalyses also shed light on localization and spatial organization of T cells within PDAC \\nTMEs related to prolonged survival following αCD40 treatment. High performance of the \\nIA-derived DFS model, coupled with poor performance of the T and TAS-derived DFS \\nmodels, supports the notion that IAs are a major site of αCD40-induced immune response \\ncontributing to prolonged DFS. We previously reported an increase in number of IAs \\nfollowing αCD40 treatment in PDAC-bearing mice (77), and in the aforementioned GVAX \\nstudy, survival was linked to increased CD4+ T helper pathway genes specifically within \\nIAs (and not tumor regions) (73). Additionally, a recent study found the enrichment of gene \\nsignatures representing mature tertiary lymphoid structures in pre-treatment PDACs \\nassociated with improved survival in patients following treatment with varying \\nchemoimmunotherapies (78). Our RCN analyses further revealed that the key T cells \\nassociated with prolonged DFS were often surrounded by antigen-experienced CD8+ T \\ncells and CD4+ T helper cells, as well as a higher proportion of proliferating and cytotoxic \\nT cells, as compared to all T cells regardless of spatial neighborhood. Collectively, the \\nresults indicate IAs may function as sites of T cell priming or second signal, promoting T \\ncell activation and function in PDAC TMEs, contributing to prolonged DFS following \\nmultiple types of immunotherapies. \\n \\nOur interrogation of IAs also revealed T cell features correlating with short DFS following \\nαCD40 therapy, identifying T cell states that are concordant with dysfunctional tumor-\\ninfiltrating T cell phenotypes. TOX1 + T cells are at the far end of the exhausted T cell \\nspectrum (14, 79), and TOX1 expression correlates with PD-1 on T cells and impaired \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='11fe43aa-b6bb-465c-acf9-8116014bbec7', embedding=None, metadata={'page_label': '33', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 33 \\nimmunotherapy response in hepatocellular carcinoma (80). Correspondingly, we \\nobserved that expression of TOX1 and/or PD-1 on CD8+ T cells and CD4+ T helper cells \\nassociated with shorter DFS following αCD40 therapy. Further, T cells linked to short DFS \\nexpressing TOX1 and/or PD-1 were largely CD44- T cells, which may represent a \\npopulation of antigen -naïve T cells that have aberrantly upregulated these markers, or \\ncould potentially represent T cells that have become terminally exhausted and \\ndifferentiated due to repeated T cell receptor stimulation (and thus downregulated CD44 \\nexpression) (81). We also found increased presence of TOX1 + T cells within treatment-\\nnaive TMEs, indicating a baseline terminally exhausted T cell phenotype in PDAC. \\nInterestingly, despite the presence of TOX1 + exhausted T cells within treatment-naive \\nTMEs, we did not identify abundant expression of PD-1+ T cells in treatment-naive TMEs, \\nin contrast to other tumors, such as melanoma (82), where PD-1+ T cells are abundant in \\nTMEs. The paucity of PD-1+ T cells in PDAC TMEs could be a contributing factor to the \\nfailure of ICBs targeting PD-1 or PD-L1 in the vast majority of patients with PDAC (83). \\nTogether, our data support the conclusion that TOX1, but not PD-1, is a dominant feature \\nof exhausted T cells in PDAC. As such, therapies that modulate TOX1+ T cells in the TME \\n– such as αCD40 agonism in this study – may improve clinical outcomes for patients with \\nPDAC. An overview of the main TME features detected by our machine learning models \\nin this study are visually described (Fig. 6). \\n \\nNotably, the dataset we used to conduct this study was unique in several aspects – \\nincluding therapy administered, mIHC panel used, and histopathologic sites assayed – \\nmaking absolute validation of our results challenging and highlighting the need for \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='45362187-6596-425d-8b1c-f9caf3aa6804', embedding=None, metadata={'page_label': '34', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 34 \\nadditional studies on larger cohorts to address the potential of building a model capable \\nof more generalized predictions. Patient samples were collected from multiple institutions \\nper treatment cohort, yet our machine learning models were still able to accurately classify \\npatient samples according to therapy. This suggests any institutional or technical \\ndifferences in tissue processing were not driving features in model predictions. \\nImportantly, the machine learning models we trained performed comparably to or better \\nthan several other models derived from similar studies (36, 38, 84-88). Additionally, our \\nbiological conclusions on impacts of αCD40 are concordant with several prior studies (8, \\n14, 72, 73, 77, 79, 80, 83), providing further support for our methods and findings. Future \\nantibody panels may incorporate additional markers, such as chemokine receptors, to \\nfurther characterize key T cell subsets, or additional cell lineage markers, such as myeloid \\ncell markers (89), to further phenotype cell-cell interactions.  It should be noted that in our \\nstudy, treatment with αCD40 did not prolong DFS as compared to the treatment-naive \\ncohort, and similar machine learning analyses on data from clinical trials that did improve \\noutcome will be useful to validate our conclusions where the patient cohort size is better \\npowered for survival analyses . Finally, tumors were resected 12 days after αCD40 \\nadministration, so it is possible that the T cells involved in prolonged DFS did not have \\nsufficient time to transit beyond IAs and into surrounding TMEs following priming. Further \\nanalyses investigating timing of T cell trafficking into various sites within PDAC TMEs are \\nnecessary to determine if analysis of T or TAS regions sampled at later timepoints \\nfollowing treatment could be used to assess clinical outcome for these patients. \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ab5306ed-36ef-492c-9784-241cd37de40a', embedding=None, metadata={'page_label': '35', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 35 \\nThis study provides proof-of-principle for leveraging computational approaches to \\nevaluate highly multiplexed cancer datasets and supports the notion that similar analytic \\napproaches could be utilized in future studies to identify important, and otherwise \\ninconspicuous alterations in TMEs correlating with patient treatment or response. Future \\nstudies could expand upon these findings to target pathways identified via this approach \\nto benefit patients who do not respond to current treatment strategies.  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='fa2a9aeb-e7cd-4c03-b73e-9db00328e7d9', embedding=None, metadata={'page_label': '36', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 36 \\nACKNOWLEDGEMENTS \\nThe authors thank Drs. David L. Bajor, E. Gabriela Chiorean, Daniel A. Laheru, and Mark \\nH. O’Hara for efforts as site investigators for the neoadjuvant selicrelumab (CD40 agonist) \\nclinical trial. We also thank John Wherry for helpful discussion related to panel design and \\nmarker selection. This research was supported by the National Cancer Institute (NCI) of \\nthe National Institutes of Health grants T32CA254888 (K.E.B.), P50 CA127003 (J.A.N.), \\nR01 CA248857 (J.A.N.), R01 CA205406 (J.A.N.), R01 CA169141 (J.A.N.), R35 \\nCA197735 (J.A.N.), U01 CA250549 (J.A.N.), U24CA231877 (J.G.), and U2CCA233280 \\n(J.G.), the DFCI Hale Family Center for Pancreatic Cancer Research (J.A.N. and B.M.W.), \\nthe Lustgarten Foundation Dedicated Laboratory program (J.A.N., B.M.W.), the Parker \\nInstitute for Cancer Immunotherapy (R.H.V. and K.T.B.), the Brenden-Colson Center for \\nPancreatic Care (L.M.C., S.S., C.B.B., K.T.B., N.K.), the Robert L. Fine Cancer Research \\nFoundation (K.T.B.), funding from the Prospect Creek Foundation to the OHSU SMMART \\n(Serial Measurement of Molecular and Architectural Responses to Therapy) Program \\n(J.G.), and a generous startup package from the Knight Cancer Institute and the Brenden-\\nColson Center for Pancreatic Care (K.T.B.). \\n \\n \\nAUTHOR CONTRIBUTIONS \\nConceptualization: K.E.B., J.G., L.M.C., K.T.B. \\nFormal analysis: K.E.B., S.S., C.B.B., B.T. \\nFunding acquisition: J.G., L.M.C., K.T.B. \\nInvestigation: K.E.B., S.S., C.B.B., K.B., N.K., E.E.F., A.D.C., J.A.N. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c45762ce-67fa-4a8c-a967-e9cc6576a6f2', embedding=None, metadata={'page_label': '37', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 37 \\nMethodology: K.E.B., S.S., C.B.B., K.B., N.K., B.T., J.G., L.M.C., K.T.B. \\nResources: R.H.V., B.M.W., J.G., L.M.C., K.T.B. \\nSupervision: J.G., L.M.C., K.T.B. \\nVisualization: K.E.B., S.S., N.K. \\nWriting – original draft: K.E.B., S.S., J.G., L.M.C., K.T.B. \\nWriting – review and editing: all authors. \\n \\n \\nCONFLICT OF INTEREST \\nR.H.V. is an inventor on licensed patents relating to cancer cellular immunotherapy and \\ncancer vaccines, and mutant Kras specific T cell receptors; has received consulting fees \\nfrom BMS; and receives royalties from Children’s Hospital Boston for a licensed research-\\nonly monoclonal antibody and from the University of Pennsylvania for licensed research \\ncell lines. J.A.N. receives consulting fees from Leica Biosystems and research support \\nfrom Natera. B.M.W. receives research funding from AstraZeneca, Celgene/BMS, Eli Lilly, \\nNovartis, and Revolution Medicines, and consulting for Celgene, GRAIL, Ipsen, Mirati, \\nThird Rock Ventures unrelated to the current work. C.B.B. is an employee of, and holds \\nequity in, Akoya Biosciences, Inc. K.T.B. receives royalties from the University of \\nPennsylvania for licensed research cell lines and has received consulting fees from \\nGuidepoint. L.M.C. has received reagent support from Cell Signaling Technologies, \\nSyndax Pharmaceuticals, Inc., ZielBio, Inc., and Hibercell, Inc.; holds sponsored research \\nagreements with Syndax Pharmaceuticals, Hibercell, Inc., Prospect Creek Foundation, \\nLustgarten Foundation for Pancreatic Cancer Research, Susan G. Komen Foundation, \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2761035f-250a-4844-acb7-b11899cfae87', embedding=None, metadata={'page_label': '38', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 38 \\nand the National Foundation for Cancer Research; is on the Advisory Board for Carisma \\nTherapeutics, Inc., CytomX Therapeutics, Inc., Kineta, Inc., Hibercell, Inc., Cell Signaling \\nTechnologies, Inc., Alkermes, Inc., Raska Pharma, Inc., NextCure, Guardian Bio, \\nAstraZeneca Partner of Choice Network (OHSU Site Leader), Genenta Sciences, Pio \\nTherapeutics Pty Ltd., and Lustgarten Foundation for Pancreatic Cancer Research \\nTherapeutics Working Group, Inc.  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c0913b88-46a8-4bba-9f0b-bdb80a55464d', embedding=None, metadata={'page_label': '39', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 39 \\nREFERENCES \\n \\n1. Pishvaian MJ, Blais EM, Brody JR, Lyons E, DeArbeloa P, Hendifar A , et al. Overall \\nsurvival in patients with pancreatic cancer receiving matched therapies following \\nmolecular profiling: a retrospective analysis of the Know Your Tumor registry trial. \\nLancet Oncol 2020;21(4):508-18. \\n2. Siegel RL, Miller KD, Fuchs HE, Jemal A. Cancer statistics, 2022. CA Cancer J \\nClin 2022;72(1):7-33. \\n3. Hosein AN, Dougan SK, Aguirre AJ, Maitra A. Translational advances in pancreatic \\nductal adenocarcinoma therapy. Nat Cancer 2022;3(3):272-86. \\n4. Byrne KT, Betts CB, Mick R, Sivagnanam S, Bajor DL, Laheru DA , et al.  \\nNeoadjuvant Selicrelumab, an Agonist CD40 Antibody, Induces Changes in the \\nTumor Microenvironment in Patients with Resectable Pancreatic Cancer. Clin \\nCancer Res 2021;27(16):4574-86. \\n5. Rojas LA, Sethna Z, Soares KC, Olcese C, Pang N, Patterson E , et al.  \\nPersonalized RNA neoantigen vaccines stimulate T cells in pancreatic cancer. \\nNature 2023;618(7963):144-50. \\n6. Vonderheide RH. The Immune Revolution: A Case for Priming, Not Checkpoint. \\nCancer Cell 2018;33(4):563-9. \\n7. Vonderheide RH. CD40 Agonist Antibodies in Cancer Immunotherapy. Annu Rev \\nMed 2020;71:47-58. \\n8. Morrison AH, Diamond MS, Hay CA, Byrne KT, Vonderheide RH. Sufficiency of \\nCD40 activation and immune checkpoint blockade for T cell priming and tumor \\nimmunity. Proc Natl Acad Sci U S A 2020;117(14):8022-31. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='79d4b7ce-5bc2-4225-a827-1a954486b26a', embedding=None, metadata={'page_label': '40', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 40 \\n9. Baaten BJ, Tinoco R, Chen AT, Bradley LM. Regulation of Antigen-Experienced T \\nCells: Lessons from the Quintessential Memory Marker CD44. Front Immunol \\n2012;3:23. \\n10. Gattinoni L, Klebanoff CA, Restifo NP. Paths to stemness: building the ultimate \\nantitumour T cell. Nat Rev Cancer 2012;12(10):671-84. \\n11. Oliveira G, Wu CJ. Dynamics and specificities of T cells in cancer immunotherapy. \\nNat Rev Cancer 2023;23(5):295-316. \\n12. Kao C, Oestreich KJ, Paley MA, Crawford A, Angelosanto JM, Ali MA , et al.  \\nTranscription factor T-bet represses expression of the inhibitory receptor PD-1 and \\nsustains virus-specific CD8+ T cell responses during chronic infection. Nat \\nImmunol 2011;12(7):663-71. \\n13. Wang Y, Hu J, Li Y, Xiao M, Wang H, Tian Q , et al. The Transcription Factor TCF1 \\nPreserves the Effector Function of Exhausted CD8 T Cells During Chronic Viral \\nInfection. Front Immunol 2019;10:169. \\n14. Khan O, Giles JR, McDonald S, Manne S, Ngiow SF, Patel KP, et al.  TOX \\ntranscriptionally and epigenetically programs CD8(+) T cell exhaustion. Nature \\n2019;571(7764):211-8. \\n15. Wherry EJ, Ha SJ, Kaech SM, Haining WN, Sarkar S, Kalia V, et al.  Molecular \\nsignature of CD8+ T cell exhaustion during chronic viral infection. Immunity \\n2007;27(4):670-84. \\n16. Wherry EJ. T cell exhaustion. Nat Immunol 2011;12(6):492-9. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='6549efb4-738d-415f-8bef-96935f59df1c', embedding=None, metadata={'page_label': '41', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 41 \\n17. Huang AC, Orlowski RJ, Xu X, Mick R, George SM, Yan PK, et al.  A single dose of \\nneoadjuvant PD-1 blockade predicts clinical outcomes in resectable melanoma. \\nNat Med 2019;25(3):454-61. \\n18. Pauken KE, Sammons MA, Odorizzi PM, Manne S, Godec J, Khan O , et al.  \\nEpigenetic stability of exhausted T cells limits durability of reinvigoration by PD-1 \\nblockade. Science 2016;354(6316):1160-5. \\n19. Schietinger A, Philip M, Krisnawan VE, Chiu EY, Delrow JJ, Basom RS , et al. \\nTumor-Specific T Cell Dysfunction Is a Dynamic Antigen-Driven Differentiation \\nProgram Initiated Early during Tumorigenesis. Immunity 2016;45(2):389-401. \\n20. Chu Y, Dai E, Li Y, Han G, Pei G, Ingram DR, et al.  Pan-cancer T cell atlas links a \\ncellular stress response state to immunotherapy resistance. Nat Med \\n2023;29(6):1550-62. \\n21. Zheng L, Qin S, Si W, Wang A, Xing B, Gao R, et al.  Pan-cancer single-cell \\nlandscape of tumor-infiltrating T cells. Science 2021;374(6574):abe6474. \\n22. Zhang H, AbdulJabbar K, Moore DA, Akarca A, Enfield KSS, Jamal-Hanjani M , et \\nal. Spatial Positioning of Immune Hotspots Reflects the Interplay between B and T \\nCells in Lung Squamous Cell Carcinoma. Cancer Res 2023;83(9):1410-25. \\n23. Karimi E, Yu MW, Maritan SM, Perus LJM, Rezanejad M, Sorin M, et al.  Single-\\ncell spatial immune landscapes of primary and metastatic brain tumours. Nature \\n2023;614(7948):555-63. \\n24. Jackson HW, Fischer JR, Zanotelli VRT, Ali HR, Mechera R, Soysal SD, et al.  The \\nsingle-cell pathology landscape of breast cancer. Nature 2020;578(7796):615-20. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='60318b44-5ca7-4d7e-8738-e40f99375e1b', embedding=None, metadata={'page_label': '42', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 42 \\n25. Keren L, Bosse M, Marquez D, Angoshtari R, Jain S, Varma S , et al. A Structured \\nTumor-Immune Microenvironment in Triple Negative Breast Cancer Revealed by \\nMultiplexed Ion Beam Imaging. Cell 2018;174(6):1373-87 e19. \\n26. Blise KE, Sivagnanam S, Banik GL, Coussens LM, Goecks J. Single-cell spatial \\narchitectures associated with clinical outcome in head and neck squamous cell \\ncarcinoma. NPJ Precis Oncol 2022;6(1):10. \\n27. Farkkila A, Gulhan DC, Casado J, Jacobson CA, Nguyen H, Kochupurakkal B , et \\nal. Immunogenomic profiling determines responses to combined PARP and PD-1 \\ninhibition in ovarian cancer. Nat Commun 2020;11(1):1459. \\n28. Enfield KSS, Martin SD, Marshall EA, Kung SHY, Gallagher P, Milne K , et al.  \\nHyperspectral cell sociology reveals spatial tumor-immune cell interactions \\nassociated with lung cancer recurrence. J Immunother Cancer 2019;7(1):13. \\n29. Gide TN, Silva IP, Quek C, Ahmed T, Menzies AM, Carlino MS, et al. Close \\nproximity of immune and tumor cells underlies response to anti-PD-1 based \\ntherapies in metastatic melanoma patients. Oncoimmunology 2020;9(1):1659093. \\n30. Giraldo NA, Nguyen P, Engle EL, Kaunitz GJ, Cottrell TR, Berry S , et al. \\nMultidimensional, quantitative assessment of PD-1/PD-L1 expression in patients \\nwith Merkel cell carcinoma and association with response to pembrolizumab. J \\nImmunother Cancer 2018;6(1):99. \\n31. Barua S, Fang P, Sharma A, Fujimoto J, Wistuba I, Rao AUK , et al.  Spatial \\ninteraction of tumor cells and regulatory T cells correlates with survival in non-small \\ncell lung cancer. Lung Cancer 2018;117:73-9. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e77ec900-a28a-4734-b7c6-9bc4ab41ac8c', embedding=None, metadata={'page_label': '43', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 43 \\n32. Mi H, Sivagnanam S, Betts CB, Liudahl SM, Jaffee EM, Coussens LM , et al.  \\nQuantitative Spatial Profiling of Immune Populations in Pancreatic Ductal \\nAdenocarcinoma Reveals Tumor Microenvironment Heterogeneity and Prognostic \\nBiomarkers. Cancer Res 2022;82(23):4359-72. \\n33. Tsujikawa T, Kumar S, Borkar RN, Azimi V, Thibault G, Chang YH , et al. \\nQuantitative Multiplex Immunohistochemistry Reveals Myeloid-Inflamed Tumor-\\nImmune Complexity Associated with Poor Prognosis. Cell Rep 2017;19(1):203-17. \\n34. Banik G, Betts CB, Liudahl SM, Sivagnanam S, Kawashima R, Cotechini T , et al. \\nHigh-dimensional multiplexed immunohistochemical characterization of immune \\ncontexture in human cancers. Methods Enzymol 2020;635:1-20. \\n35. Liudahl SM, Betts CB, Sivagnanam S, Morales-Oyarvide V, da Silva A, Yuan C, et \\nal. Leukocyte Heterogeneity in Pancreatic Ductal Adenocarcinoma: Phenotypic \\nand Spatial Features Associated with Clinical Outcome. Cancer Discov \\n2021;11(8):2014-31. \\n36. Risom T, Glass DR, Averbukh I, Liu CC, Baranski A, Kagel A , et al. Transition to \\ninvasive breast cancer is associated with progressive changes in the structure and \\ncomposition of tumor stroma. Cell 2022;185(2):299-310 e18. \\n37. Kim DW, Lee S, Kwon S, Nam W, Cha IH, Kim HJ. Deep learning-based survival \\nprediction of oral cancer patients. Sci Rep 2019;9(1):6994. \\n38. Buk Cardoso L, Cunha Parro V, Verzinhasse Peres S, Curado MP, Fernandes GA, \\nWunsch Filho V, et al. Machine learning for predicting survival of colorectal cancer \\npatients. Sci Rep 2023;13(1):8874. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='926abe94-3094-4711-aa7e-2b70f2d638db', embedding=None, metadata={'page_label': '44', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 44 \\n39. Sorin M, Rezanejad M, Karimi E, Fiset B, Desharnais L, Perus LJM , et al. Single-\\ncell spatial landscapes of the lung tumour immune microenvironment. Nature \\n2023;614(7948):548-54. \\n40. Carpenter AE, Jones TR, Lamprecht MR, Clarke C, Kang IH, Friman O , et al.  \\nCellProfiler: image analysis software for identifying and quantifying cell \\nphenotypes. Genome Biol 2006;7(10):R100. \\n41. Pedregosa F, Varoquaux G, Gramfort A, Michel V, Thirion B, Grisel O , et al. Scikit-\\nlearn: Machine Learning in Python. JMLR 2011;12:2825-30. \\n42. Chawla NV, Bowyer KW, Hall LO, Kegelmeyer WP. SMOTE: Synthetic Minority \\nOver-sampling Technique. JAIR 2002;16:321-57. \\n43. S.M. L, S. L. A Unified Approach to Interpreting Model Predictions. NIPS \\n2017:4765-74. \\n44. Carstens JL, Correa de Sampaio P, Yang D, Barua S, Wang H, Rao A , et al. Spatial \\ncomputation of intratumoral T cells correlates with survival of patients with \\npancreatic cancer. Nat Commun 2017;8:15095. \\n45. Gartrell RD, Enzler T, Kim PS, Fullerton BT, Fazlollahi L, Chen AX , et al.  \\nNeoadjuvant chemoradiation alters the immune microenvironment in pancreatic \\nductal adenocarcinoma. Oncoimmunology 2022;11(1):2066767. \\n46. Siret C, Collignon A, Silvy F, Robert S, Cheyrol T, Andre P , et al. Deciphering the \\nCrosstalk Between Myeloid-Derived Suppressor Cells and Regulatory T Cells in \\nPancreatic Ductal Adenocarcinoma. Front Immunol 2019;10:3070. \\n47. Busse D, de la Rosa M, Hobiger K, Thurley K, Flossdorf M, Scheffold A , et al.  \\nCompeting feedback loops shape IL-2 signaling between helper and regulatory T \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9f1c5b0b-f266-4d6a-8336-33a7f7ea7f2a', embedding=None, metadata={'page_label': '45', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\" 45 \\nlymphocytes in cellular microenvironments. Proc Natl Acad Sci U S A \\n2010;107(7):3058-63. \\n48. Muller AJ, Filipe-Santos O, Eberl G, Aebischer T, Spath GF, Bousso P. CD4+ T \\ncells rely on a cytokine gradient to control intracellular pathogens beyond sites of \\nantigen presentation. Immunity 2012;37(1):147-57. \\n49. Oyler-Yaniv A, Oyler-Yaniv J, Whitlock BM, Liu Z, Germain RN, Huse M , et al. A \\nTunable Diffusion-Consumption Mechanism of Cytokine Propagation Enables \\nPlasticity in Cell-to-Cell Communication in the Immune System. Immunity \\n2017;46(4):609-20. \\n50. Thibaut R, Bost P, Milo I, Cazaux M, Lemaitre F, Garcia Z , et al. Bystander IFN-\\ngamma activity promotes widespread and sustained cytokine signaling altering the \\ntumor microenvironment. Nat Cancer 2020;1(3):302-14. \\n51. Virtanen P, Gommers R, Oliphant TE, Haberland M, Reddy T, Cournapeau D , et \\nal. SciPy 1.0: fundamental algorithms for scientific computing in Python. Nat \\nMethods 2020;17(3):261-72. \\n52. Seabold S, Perktold J. statsmodels: Econometric and statistical modeling with \\npython. Proceedings of the 9th Python in Science Conference 2010. \\n53. Blank CU, Haining WN, Held W, Hogan PG, Kallies A, Lugli E , et al.  Defining 'T \\ncell exhaustion'. Nat Rev Immunol 2019;19(11):665-74. \\n54. McLane LM, Abdel-Hakeem MS, Wherry EJ. CD8 T Cell Exhaustion During \\nChronic Viral Infection and Cancer. Annu Rev Immunol 2019;37:457-95. \\n55. Chu T, Zehn D. Charting the Roadmap of T Cell Exhaustion. Immunity \\n2020;52(5):724-6. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint \", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9c4b84c1-b1c2-4c27-9557-30cd5c5758cd', embedding=None, metadata={'page_label': '46', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 46 \\n56. Philip M, Fairchild L, Sun L, Horste EL, Camara S, Shakiba M , et al.  Chromatin \\nstates define tumour-specific T cell dysfunction and reprogramming. Nature \\n2017;545(7655):452-6. \\n57. Canale FP, Ramello MC, Nunez N, Araujo Furlan CL, Bossio SN, Gorosito Serran \\nM, et al. CD39 Expression Defines Cell Exhaustion in Tumor-Infiltrating CD8(+) T \\nCells. Cancer Res 2018;78(1):115-28. \\n58. Davis A, Gao R, Navin NE. SCOPIT: sample size calculations for single-cell \\nsequencing experiments. BMC Bioinformatics 2019;20(1):566. \\n59. Jang IS, Neto EC, Guinney J, Friend SH, Margolin AA. Systematic assessment of \\nanalytical methods for drug sensitivity prediction from cancer cell line data. Pac \\nSymp Biocomput 2014:63-74. \\n60. Zou H, Hastie T. Regularization and variable selection via the elastic net. Journal \\nof the Royal Statistical Society: Series B (Statistical Methodology) 2005;67(2):301-\\n20. \\n61. Moncada-Torres A, van Maaren MC, Hendriks MP, Siesling S, Geleijnse G. \\nExplainable machine learning can outperform Cox regression predictions and \\nprovide insights in breast cancer survival. Sci Rep 2021;11(1):6968. \\n62. Huang T, Le D, Yuan L, Xu S, Peng X. Machine learning for prediction of in-hospital \\nmortality in lung cancer patients admitted to intensive care unit. PLoS One \\n2023;18(1):e0280606. \\n63. Duhen T, Duhen R, Lanzavecchia A, Sallusto F, Campbell DJ. Functionally distinct \\nsubsets of human FOXP3+ Treg cells that phenotypically mirror effector Th cells. \\nBlood 2012;119(19):4430-40. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ffaf6b12-796c-471d-ad35-dc52a6b69e5a', embedding=None, metadata={'page_label': '47', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 47 \\n64. Schurch CM, Bhate SS, Barlow GL, Phillips DJ, Noti L, Zlobec I , et al. Coordinated \\nCellular Neighborhoods Orchestrate Antitumoral Immunity at the Colorectal \\nCancer Invasive Front. Cell 2020. \\n65. De Vargas Roditi L, Jacobs A, Rueschoff JH, Bankhead P, Chevrier S, Jackson \\nHW, et al. Single-cell proteomics defines the cellular heterogeneity of localized \\nprostate cancer. Cell Rep Med 2022;3(4):100604. \\n66. Ali HR, Jackson HW, Zanotelli VRT, Danenberg E, Fischer JR, Bardwell H , et al. \\nImaging mass cytometry and multiplatform genomics define the phenogenomic \\nlandscape of breast cancer. Nat Cancer 2020;1(2):163-75. \\n67. Xiao X, Guo Q, Cui C, Lin Y, Zhang L, Ding X , et al.  Multiplexed imaging mass \\ncytometry reveals distinct tumor-immune microenvironments linked to \\nimmunotherapy responses in melanoma. Commun Med (Lond) 2022;2:131. \\n68. Gattinoni L, Lugli E, Ji Y, Pos Z, Paulos CM, Quigley MF , et al. A human memory \\nT cell subset with stem cell-like properties. Nat Med 2011;17(10):1290-7. \\n69. Sallusto F, Lenig D, Forster R, Lipp M, Lanzavecchia A. Two subsets of memory T \\nlymphocytes with distinct homing potentials and effector functions. Nature \\n1999;401(6754):708-12. \\n70. Willinger T, Freeman T, Hasegawa H, McMichael AJ, Callan MF. Molecular \\nsignatures distinguish human central memory from effector memory CD8 T cell \\nsubsets. J Immunol 2005;175(9):5895-903. \\n71. Tay RE, Richardson EK, Toh HC. Revisiting the role of CD4(+) T cells in cancer \\nimmunotherapy-new insights into old paradigms. Cancer Gene Ther 2021;28(1-\\n2):5-17. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='18512982-88b5-4995-a23d-0dd40cd87c68', embedding=None, metadata={'page_label': '48', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\" 48 \\n72. Padron LJ, Maurer DM, O'Hara MH, O'Reilly EM, Wolff RA, Wainberg ZA, et al.  \\nSotigalimab and/or nivolumab with chemotherapy in first-line metastatic pancreatic \\ncancer: clinical and immunologic analyses from the randomized phase 2 PRINCE \\ntrial. Nat Med 2022;28(6):1167-77. \\n73. Lutz ER, Wu AA, Bigelow E, Sharma R, Mo G, Soares K, et al.  Immunotherapy \\nconverts nonimmunogenic pancreatic tumors into immunogenic foci of immune \\nregulation. Cancer Immunol Res 2014;2(7):616-31. \\n74. Sledzinska A, Vila de Mucha M, Bergerhoff K, Hotblack A, Demane DF, Ghorani E , \\net al.  Regulatory T Cells Restrain Interleukin-2- and Blimp-1-Dependent \\nAcquisition of Cytotoxic Function by CD4(+) T Cells. Immunity 2020;52(1):151-66 \\ne6. \\n75. Chen D, Varanasi SK, Hara T, Traina K, Sun M, McDonald B , et al.  CTLA-4 \\nblockade induces a microglia-Th1 cell partnership that stimulates microglia \\nphagocytosis and anti-tumor function in glioblastoma. Immunity 2023;56(9):2086-\\n104 e8. \\n76. Kennedy R, Celis E. Multiple roles for CD4+ T cells in anti-tumor immune \\nresponses. Immunol Rev 2008;222:129-44. \\n77. Byrne KT, Vonderheide RH. CD40 Stimulation Obviates Innate Sensors and Drives \\nT Cell Immunity in Cancer. Cell Rep 2016;15(12):2719-32. \\n78. Kinker GS, Vitiello GAF, Diniz AB, Cabral-Piccin MP, Pereira PHB, Carvalho MLR, \\net al. Mature tertiary lymphoid structures are key niches of tumour-specific immune \\nresponses in pancreatic ductal adenocarcinomas. Gut 2023;72(10):1927-41. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint \", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='68584c1f-7869-401c-8e86-a14903ce08b5', embedding=None, metadata={'page_label': '49', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\" 49 \\n79. Bengsch B, Ohtani T, Khan O, Setty M, Manne S, O'Brien S, et al.  Epigenomic-\\nGuided Mass Cytometry Profiling Reveals Disease-Specific Features of \\nExhausted CD8 T Cells. Immunity 2018;48(5):1029-45 e5. \\n80. Wang X, He Q, Shen H, Xia A, Tian W, Yu W , et al. TOX promotes the exhaustion \\nof antitumor CD8(+) T cells by preventing PD1 degradation in hepatocellular \\ncarcinoma. J Hepatol 2019;71(4):731-41. \\n81. Abu Eid R, Ahmad S, Lin Y, Webb M, Berrong Z, Shrimali R , et al.  Enhanced \\nTherapeutic Efficacy and Memory of Tumor-Specific CD8 T Cells by Ex Vivo PI3K-\\ndelta Inhibition. Cancer Res 2017;77(15):4135-45. \\n82. Ahmadzadeh M, Johnson LA, Heemskerk B, Wunderlich JR, Dudley ME, White \\nDE, et al. Tumor antigen–specific CD8 T cells infiltrating the tumor express high \\nlevels of PD-1 and are functionally impaired. Blood 2009;114(8):1537-44. \\n83. Morrison AH, Byrne KT, Vonderheide RH. Immunotherapy and Prevention of \\nPancreatic Cancer. Trends Cancer 2018;4(6):418-28. \\n84. Iwatate Y, Hoshino I, Yokota H, Ishige F, Itami M, Mori Y , et al. Radiogenomics for \\npredicting p53 status, PD-L1 expression, and prognosis with machine learning in \\npancreatic cancer. Br J Cancer 2020;123(8):1253-61. \\n85. Jiang C, Xiu Y, Qiao K, Yu X, Zhang S, Huang Y. Prediction of lymph node \\nmetastasis in patients with breast invasive micropapillary carcinoma based on \\nmachine learning and SHapley Additive exPlanations framework. Front Oncol \\n2022;12:981059. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint \", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2877741f-9a8e-474e-9f40-a054d17abb0d', embedding=None, metadata={'page_label': '50', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 50 \\n86. Sammut SJ, Crispin-Ortuzar M, Chin SF, Provenzano E, Bardwell HA, Ma W , et al. \\nMulti-omic machine learning predictor of breast cancer therapy response. Nature \\n2022;601(7894):623-9. \\n87. Lapuente-Santana O, van Genderen M, Hilbers PAJ, Finotello F, Eduati F. \\nInterpretable systems biomarkers predict response to immune-checkpoint \\ninhibitors. Patterns (N Y) 2021;2(8):100293. \\n88. Chowell D, Yoo SK, Valero C, Pastore A, Krishna C, Lee M , et al.  Improved \\nprediction of immune checkpoint blockade efficacy across multiple cancer types. \\nNat Biotechnol 2022;40(4):499-506. \\n89. Dias Costa A, Vayrynen SA, Chawla A, Zhang J, Vayrynen JP, Lau MC , et al. \\nNeoadjuvant Chemotherapy Is Associated with Altered Immune Cell Infiltration and \\nan Anti-Tumorigenic Microenvironment in Resected Pancreatic Cancer. Clin \\nCancer Res 2022;28(23):5167-79. \\n \\n  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5b70bb9d-ffb0-4daa-80d3-63d75e231204', embedding=None, metadata={'page_label': '51', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 51 \\nMAIN FIGURE LEGENDS \\nFigure 1. Designing a novel mIHC antibody panel to deeply phenotype T cells within \\nthe PDAC TME. \\nA. Overview of two PDAC cohorts assayed via mIHC. \\nB. Representative PDAC tissue resection stained with H&E (middle) showing four \\nhistopathologic sites annotated. Total number of regions assayed and average \\ntissue area per histopathologic site are listed. Scale bars = 100 µm. \\nC. Number of regions assayed per patient. Each box represents one tissue region \\nand is colored according to its histopathologic site. \\nD. 21-marker mIHC antibody panel used to assay tissue regions. \\nE. Representative pseudo-colored mIHC images showing T cell functionality markers \\nwith CD3 expression. Scale bars = 50 µm. \\nF. Cell phenotyping strategy by hierarchical gating of lineage markers and functional \\nmarkers. Circles indicate colors associated with each cell state in the following \\nfigures. \\n \\nFigure 2. Interrogating cell states and spatial interactions within the PDAC TME. \\nA. Stacked bar charts showing the average cell state densities for each treatment \\ncohort and histopathologic site. Top row: lineage defining cells including neoplastic \\nepithelial cells, mesenchymal fibroblast-like cells, myeloid cells, B cells, and CD3+ \\nT cells; Middle row: CD8 + T cell states; Bottom row: CD4 + T cell states. Columns \\ndenote histopathologic site, and each plot is further broken into treatment cohort. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3e00092e-8dba-40db-a9ed-1d81eb7b5681', embedding=None, metadata={'page_label': '52', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 52 \\nB. Bar charts showing average densities of barcoded T cells for each treatment cohort \\nand histopathologic site. Only the 15 most abundant barcodes are shown as \\nmeasured by average density. Rows denote histopathologic site, and columns \\ndenote treatment cohort. Brown bars denote barcoded T cells that are in the top \\n15 most abundant barcodes in both cohorts. Orange bars denote barcoded T cells \\nthat are in the top 15 most abundant barcodes in the treatment-naive cohort only. \\nBlue bars denote barcoded T cells that are in the top 15 most abundant barcodes \\nin the αCD40-treated cohort only. \\nC. Heatmaps showing average number of spatial interactions between two cell states \\nfor each treatment cohort and histopathologic site. Cell states are denoted by \\ncolors shown in Figure 1F. Interactions were normalized first by density of cells \\nparticipating in the interaction and were then log10+1 transformed. Rows denote \\ntreatment cohort and columns denote histopathologic site. \\nD. Overview schematic of analyses performed in this study. TME features were \\ncalculated for each tissue region. Two machine learning classifier models were \\nbuilt for each histopathologic site to predict treatment status and DFS. Feature \\nimportance analyses were performed to interpret biological meaning. \\n \\nFigure 3. Machine learning models classify αCD40-treated TMEs as having reduced \\nT cell exhaustion phenotypes. \\nA. Bar chart showing accuracy and F1 score for each histopathologic model that \\npredicts treatment status. \\nB. ROC curve with corresponding AUC for each histopathologic model. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='24816b67-2046-4b69-9090-480f86eea05b', embedding=None, metadata={'page_label': '53', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 53 \\nC. Bubble chart showing top 15 features whose increased presence drove each \\nhistopathologic model to predict treatment-naive (orange) or αCD40-treated (blue). \\nFeatures are grouped by TME feature type (density, barcode, interaction). Bubble \\nsize denotes relative importance of the feature for a given histopathologic model. \\nBubbles appearing multiple times in the same row indicate TME feature is a top \\nfeature across histopathologic models. \\nD. Representative pseudo-colored mIHC images showing TOX1 + and/or EOMES + \\nCD3+ T cells in treatment-naive tissue (left) and CD44+ and/or T-BET+ CD3+ T cells \\nin aCD40-treated tissue (right). \\nE. Matrix showing correlations between top barcodes from the models with each other \\nbased on types and proportions of T cell states expressing the barcodes. Stacked \\nbars at the top of correlation matrixes show proportions of T cell states expressing \\nbarcodes, with T cells color coded and listed for each group to the right of the \\nheatmap, along with corresponding barcodes in each group. Leftmost columns are \\ncolor coded according to which treatment group the presence of the barcode was \\npredicted by the model, followed by the histopathologic site the model was derived \\nfrom. \\n \\nFigure 4. Machine learning model classifies long disease-free survivors as having \\nmore T cell effector functionality following αCD40 therapy. \\nA. Bar chart showing accuracy and F1 score for each histopathologic model \\npredicting αCD40 DFS. \\nB. ROC curve with corresponding AUC for the IA histopathologic model. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0b9990cb-819d-40a3-9cb3-717f718a462b', embedding=None, metadata={'page_label': '54', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 54 \\nC. Bubble chart showing the top 15 features whose increased presence drove the IA \\nmodel to predict short DFS (purple) or long DFS (green). Features are grouped by \\nTME feature type (density, barcode, interaction). Bubble size denotes relative \\nimportance of the feature. \\nD. Representative pseudo-colored mIHC images showing TOX1+ and/or PD-1+ CD3+ \\nT cells in short DFS tissue (left) and CD44+ and/or CD39+ CD3+ T cells in long DFS \\ntissue (right). Scale bars = 100 µm. \\nE. Matrix showing correlations between the top barcodes from the IA model with each \\nother based on types and proportions of T cell states expressing the barcodes. \\nStacked bars at the top of the correlation matrix show proportions of T cell states \\nexpressing the barcodes, and T cells are color coded and listed for each group to \\nthe right of the heatmap, along with the corresponding barcodes in each group. \\nLeftmost column is color coded according to which DFS group the presence of the \\nbarcode predicted by the model. \\n \\nFigure 5. Cellular neighborhood analysis identifies spatial organization of T cells \\nto correlate with DFS following αCD40 therapy. \\nA. Schematic depicting RCN analysis. Cellular neighborhoods were defined by \\nidentifying all cells within a 60 µm radius of a given cell. Neighborhoods were \\ncalculated for all cells in αCD40-treated IA regions. Neighborhoods were then \\ngrouped using K-means clustering to identify RCNs. Created with BioRender.com. \\nB. Stacked bar chart showing average cellular composition of each of seven RCNs \\nfrom the αCD40-treated IA regions. Bars are colored by cell state and represent \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c9565d8e-78f4-4254-8851-248184834016', embedding=None, metadata={'page_label': '55', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 55 \\naverage proportions (out of 1.0) of each cell state present in neighborhoods \\nassigned to each RCN. \\nC. Representative IA tissue region as depicted by scatterplot reconstructions. Each \\ndot represents a cell present in the IA, and each cell is colored by its original cell \\nstate phenotype (left scatterplot) or RCN assignment (right scatterplot). \\nD. Heatmap showing top T cell barcodes from the αCD40 IA DFS model clustered by \\nproportion of RCNs the T cell barcodes were assigned to. Rows are barcoded T \\ncells from IA regions from patients associated with short DFS or long DFS ordered \\nby hierarchical clustering of their RCN assignment. Columns are RCNs used as \\nclustering features. Proportion of RCNs was normalized using a log10+1 \\ntransformation prior to clustering. Leftmost columns are color coded by DFS group \\nfollowed by barcode. \\nE. Stacked bar chart showing average fraction of RCNs barcoded T cells were \\nassigned to for each of two hierarchically clustered groups (C1 or C2). \\nF. Bar chart showing percentage of T cells expressing Ki-67 residing in each of seven \\nRCNs for αCD40 IA regions. Horizontal dashed line represents percentage of Ki-\\n67+ T cells across all αCD40 IA regions, regardless of RCN assignment. \\nG. Bar chart showing percentage of T cells expressing GrzB residing in each of seven \\nRCNs for αCD40 IA regions. Horizontal dashed line represents percentage of \\nGrzB+ T cells across all αCD40 IA regions, regardless of RCN assignment. T REG \\npopulations were excluded from this analysis. \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='dbd93ce9-6c0a-4d0c-ad5d-870bea932924', embedding=None, metadata={'page_label': '56', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 56 \\nH. Representative IA region from a patient with long DFS with cells colored by RCN \\nassignment in the upper left scatterplot. Remaining images show mIHC staining of \\nGrzB+ CD44+ CD3+ T cells localized within RCN5. Scale bars = 100 µm. \\n \\nFigure 6. Spatial features of T cells associated with αCD40 therapy and prolonged \\nDFS in the PDAC TME. \\nA. T cell subsets that best define resected tumor samples from treatment-naïve (left) \\nor αCD40-treated (right) patients. In the absence of therapy, T cells appear in an \\nexhausted state, while T cells present with activated and effector phenotypes after \\nCD40 agonism. Created with BioRender.com. \\nB. T cell phenotypes in IAs from αCD40-treated patients associated with long (top) or \\nshort (bottom) DFS. IAs from patients with long DFS are characterized by the \\npresence of spatial neighborhoods of effector T cells capable of proliferating and \\ncytotoxicity, while IAs from patients with short DFS have a preponderance of \\nexhausted T cell states. Created with BioRender.com. \\n \\n \\n  \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='19335722-eb5c-4065-9041-c40194ffb508', embedding=None, metadata={'page_label': '57', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 57 \\nSUPPLEMENTARY FIGURE LEGENDS \\nSupplementary Figure S1. \\nA. Representative IHC staining of each antibody used in sequence in the panel. Scale \\nbar = 50 µm. \\nB. Two representative regions stained with CD3, CD8, and CD4 antibodies. For each \\nregion, top images show gates for CD8 on CD3+ population (left) and CD4 on CD3+ \\nCD8- population (right), and bottom row shows pseudo-colored mIHC images. \\nC. Hierarchical gating template used to phenotype cells using image gating cytometry \\nin FCS Image Cytometry RUO. \\n \\nSupplementary Figure S2. \\nA. SHAP plots showing the top 30 features driving each histopathologic model. \\nFeatures are ordered on the y-axis such that those with a larger impact on model’s \\npredictions appear at the top of the SHAP plots. SHAP values are shown on the x-\\naxis, with a value of zero (center) indicating no impact on the model, and negative \\nor positive SHAP values predicting treatment-naive or αCD40-treated tissues, \\nrespectively. Red or blue dots indicate presence or absence, respectively, of the \\ncorresponding feature in the tissue.  \\nB-E.  Box plots showing feature values for each of the top 15 features for models derived \\nfrom T, IA, TAS, or NAP sites, respectively, split by treatment cohort. Each dot \\nrepresents the log10+1 normalized feature value for one tissue region, inputted \\ninto the classifier model. Boxes = quartile 1 (Q1) to quartile 3 (Q3); whiskers = \\nsmallest and largest datapoints within 1.5*interquartile range (IQR) +/- Q3/Q1; \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5fad0892-b340-4c27-b3c2-e9f75937f46b', embedding=None, metadata={'page_label': '58', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 58 \\nsolid line = median. Mann–Whitney  U-test used to determine statistical \\nsignificance. P-values corrected using the Benjamini–Hochberg procedure. *, P ≤ \\n0.05; **, P ≤ 0.01; ***, P ≤ 0.001. \\nB. T site, n= 55 treatment-naive and n = 48 αCD40-treated regions per feature. \\nC. IA site, n= 89 treatment-naive and n = 43 αCD40-treated regions per feature. \\nD. TAS site, n = 25 treatment-naive and n = 27 αCD40-treated regions per feature. \\nE. NAP site, n = 6 treatment-naive and n = 13 αCD40-treated regions per feature. \\n \\nSupplementary Figure S3. \\nA. SHAP plot showing the top 30 features driving the IA model. Features are ordered \\non the y-axis such that those with a larger impact on the model’s predictions appear \\nat the top of the SHAP plot. SHAP values are shown on the x-axis, with a value of \\nzero (center) indicating no impact on the model, and negative or positive SHAP \\nvalues predicting long DFS or short DFS, respectively. Red or blue dots indicate \\npresence or absence, respectively, of the corresponding feature in tissues.  \\nB. Box plot showing feature values for each of the top 15 features for the model \\nderived from IA regions of the αCD40 cohort split by DFS group (n = 30 regions \\nfrom short DFS patients per feature; n = 13 regions from long DFS patients per \\nfeature). Each dot represents the log10+1 normalized feature value for one tissue \\nregion, which was inputted into the classifier model. Boxes = Q1 to Q3; whiskers \\n= smallest and largest datapoints within 1.5*IQR +/- Q3/Q1; solid line = median. \\nMann–Whitney U-test used to determine statistical significance. P-values \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='df9e5b6a-1f55-4476-8f73-d6516bdaec01', embedding=None, metadata={'page_label': '59', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 59 \\ncorrected using the Benjamini–Hochberg procedure. *, P ≤ 0.05; **, P ≤ 0.01; ***, \\nP ≤ 0.001. \\n \\nSupplementary Figure S4. \\nA. Elbow plot showing optimal number of RCNs (k=7) for grouping cellular \\nneighborhoods. \\nB. Bar chart showing the number of cells assigned to each of the seven RCNs across \\nall αCD40 IA regions. \\nC. Bar chart showing the percentage (out of 100) of cells assigned to each of the \\nseven RCNs across all αCD40 IA regions. \\nD. Stacked bar chart showing fraction (out of 1.0) of RCNs present per αCD40 IA \\nregion. \\nE. Stacked bar chart showing average proportion (out of 1.0) of RCNs present in IA \\nregions for each αCD40-treated patient. \\nF. Scatterplot reconstructions for each αCD40 IA region. Each dot represents a cell \\npresent in the IA, and each cell is colored by its original cell state phenotype (top \\nscatterplot) or RCN assignment (bottom scatterplot). \\n \\n \\nSUPPLEMENTARY TABLE LEGENDS \\nSupplementary Table S1. \\nStatistical comparison between the Liudahl et al. original PDAC cohort(35) and the \\nselected subset used as Cohort 1 in this study. Mean value and standard error of the \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a16dff27-1e82-4022-b0be-89b4a42c6ddb', embedding=None, metadata={'page_label': '60', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' 60 \\nmean (SEM) shown for each variable. P-values computed using Fisher’s exact test for \\ncategorical variables, Wilcoxon rank-sum test for continuous variables, and log rank test \\nfor overall survival. \\n \\nSupplementary Table S2. \\nTable of antibodies used in mIHC panel. \\n \\nSupplementary Table S3. \\nRaw counts of cell states defined by mIHC gating strategy present in the dataset. \\n \\nSupplementary Table S4. \\nRaw counts of T cells expressing each functionality barcode present in the dataset. \\n \\nSupplementary Table S5. \\nRaw counts of cell-cell spatial interactions present in the dataset. \\n \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8cd2a44a-6285-435e-aa27-942ef58dd0d3', embedding=None, metadata={'page_label': '61', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A Blise et al. Figure 1\\nCohort 1\\nTreatment Naive\\n(n = 18 tumors; n = 175 regions)\\nCohort 2\\nNeoadjuvant αCD40\\n(n = 11 tumors; n = 131 regions)\\nmIHC staining and single-cell analysis of TME\\n(n = 29 tumors; n = 306 regions; n = 2,428,274 total cells assayed)\\nB\\n     Tumor (T)     \\nn = 103; 6.60 µm2\\n Normal-Adjacent Pancreas (NAP) \\nn = 19; 2.09 µm2\\n  Immune Aggregate (IA)  \\nn = 132; 0.29 µm2\\n Tumor-Adjacent Stroma (TAS) \\nn = 52; 3.92 µm2\\nC\\n0 -\\n5 -\\n10 -\\n15 -\\n20 -\\n25 -\\nNAP\\nTAS\\nIA\\nT\\nNumber of Regions\\nPatient\\nCohort 1\\nCohort 2\\nD\\nmIHC Antibody Panel\\nProliferation/\\nCytotoxicity\\nT Cell\\nFunctionalityLineage\\nKi-67TOX1CD3\\nGrzBTIM3CD20\\nTCF-1CD8\\nCD38FOXP3\\nPD-1CD45RA\\nEOMESCD45RO\\nCD39CD11b\\nCD44αSMA\\nLAG-3PanCK\\nT-BET\\nE\\nH&E\\ndsDNA CD3 CD38 CD39LAG-3 TIM3\\ndsDNA CD3\\ndsDNA CD3\\nTCF-1 CD44GrzB T-BET\\nPD-1 EOMES TOX1\\nF\\nCD8+ T cells\\nCD3+ CD8+\\nCD4+ Th1 cells\\nCD3+ CD8- FOXP3- T-BET+\\nCD4+ Th cells\\nCD3+ CD8- FOXP3- T-BET- at least one of CD45RA+ and/or CD45RO+\\nTREG cells\\nCD3+ CD8- FOXP3+\\nB cells\\nCD3- CD20+\\nMyeloid cells\\nCD3- CD20- CD11b+\\nMesenchymal Fibroblast-like cells\\nCD3- CD20- CD11b- αSMA+\\nNeoplastic Epithelial cells\\nCD3- CD20- CD11b- αSMA- PanCK+\\nNaive \\n(CD8+ TNAIVE)\\nTCF1+ CD44-\\nCD45RO-\\nCD45RA+\\nT Effector\\n(CD8+ TEFF)\\nPD-1+ T-BET+ \\nCD44+ CD45RO-\\nCD45RA+\\nTem\\n(CD8+ TEM)\\nTCF-1+ CD44+\\nCD45RO+ CD45RA-\\nEOMES+\\nTemra\\n(CD8+ TEMRA)\\nTCF-1+ CD44+\\nCD45RO+ CD45RA+\\nEOMES+\\nEarly\\nExhaustion (CD8+ TEX)\\nPD-1+ TCF-1+ \\nCD44+ CD45RO+\\nCD45RA-\\nTerminal\\nExhaustion (CD8+ TTEX)\\nPD-1+ TCF-1- \\nCD44+ CD45RO+\\nCD45RA-\\nCD44- CD8+ TOTHER\\nCD44-\\nCD44+ CD8+ TOTHER\\nCD44+\\nT Effector (CD4+ Th1EFF)\\nPD-1+ CD44+ CD45RO- CD45RA+\\nTem (CD4+ Th1EM)\\nCD44+ CD45RO+ CD45RA-\\nTemra (CD4+ Th1EMRA)\\nCD44+ CD45RO+ CD45RA+\\nCD44- CD4+ Th1OTHER\\nCD44-\\nCD44+ CD4+ Th1OTHER\\nCD44+\\nCD44- CD4+ Th\\nCD44-\\nCD44+ CD4+ Th\\nCD44+\\nNaive TREG\\nT-BET- CD45RA+\\nmTREG\\nT-BET- CD45RA-\\nT-BET+ TREG\\nT-BET+\\nKi-67- Neoplastic\\nKi-67-\\nKi-67+ Neoplastic\\nKi-67+\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='679eba2e-1ad5-4e65-8339-51e8a60445e4', embedding=None, metadata={'page_label': '62', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='0-\\n500-\\n1000-\\n1500-\\n2000-\\n2500-\\n3000-\\n0-\\n50-\\n100-\\n150-\\n200-\\n250-\\n300-\\n0-\\n20-\\n40-\\n60-\\n80-\\n100-\\n120-\\n140-\\n160-\\n0-\\n1000-\\n2000-\\n3000-\\n4000-\\n5000-\\n6000-\\n7000-\\n0-\\n50-\\n100-\\n150-\\n200-\\n250-\\n0-\\n50-\\n100-\\n150-\\n0-\\n500-\\n1000-\\n1500-\\n2000-\\n0-\\n500-\\n1000-\\n1500-\\n2000-\\n2500-\\n0-\\n200-\\n400-\\n600-\\n800-\\n1000-\\n1200-\\n1400-\\n1600-\\n0-\\n1000-\\n2000-\\n3000-\\n4000-\\n5000-\\n6000-\\n7000-\\nTx-Naive\\nαCD40\\n0-\\n50-\\n100-\\n150-\\n200-\\n0-\\n20-\\n40-\\n60-\\n80-\\n100-\\n120-\\n140-\\nIAT TAS NAP\\nA Blise et al. Figure 2\\nCell State Densities\\nLineage defining cells\\nDensity (cells/mm\\n2)\\nCD8\\n+ T cells\\nCD4\\n+ T cells\\nDensity (cells/mm\\n2)\\nDensity (cells/mm\\n2)\\nTx-Naive\\nαCD40\\nTx-Naive\\nαCD40\\nTx-Naive\\nαCD40\\nB\\nC\\nD\\nCell State Densities\\n(23 unique cell states)\\nT Cell Functionality\\nBarcodes\\n(961 unique barcodes)\\nCell-Cell Spatial\\nInteractions\\n(268 unique interactions)\\nT (n=103)\\nIA (n=132)\\nTAS (n=52)\\nNAP (n=19)\\nMachine learning classifier models builtTME features calculated for each region Top features identified across models Biological interpretations made\\nCD4+ T cells\\nT-BET+ TREG\\nmTREG\\nNaive TREG\\nCD44+ CD4+ Th\\nCD44- CD4+ Th\\nCD44+ CD4+ Th1OTHER\\nCD44- CD4+ Th1OTHER\\nCD4+ Th1EMRA\\nCD4+ Th1EM\\nCD4+ Th1EFF\\nCD8+ T cells\\nCD44+ CD8+ TOTHER\\nCD44- CD8+ TOTHER\\nCD8+ TTEX\\nCD8+ TEX\\nCD8+ TEMRA\\nCD8+ TEM\\nCD8+ TEFF\\nCD8+ TNAIVE\\nLineage defining cells\\nNeoplastic\\nMesenchymal\\nMyeloid cell\\nB cell\\nCD3+ T cell\\nIAT TAS NAP\\n1.2\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n1.2\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\nCD8+ T cells\\nCD4+ T Helper\\nTregs\\nB cells -Myeloid -Mesenchymal -\\nNeoplastic\\nCell-Cell Spatial Interactions\\n0.35\\n0.30\\n0.25\\n0.20\\n0.15\\n0.10\\n0.05\\n0.00\\n0.35\\n0.30\\n0.25\\n0.20\\n0.15\\n0.10\\n0.05\\n0.00\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n1.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\n0.8\\n0.6\\n0.4\\n0.2\\n0.0\\nTx-Naive\\nαCD40\\n0\\n20\\n40\\n60\\n80\\n100\\n120\\n140\\nNeg for all\\nTCF-1+\\nCD44+\\nT-BET+\\nTCF-1+ T-BET+\\nTOX1+\\nCD39+\\nCD44+ TCF-1+\\nCD44+ T-BET+\\nTOX1+ TCF-1+\\nTOX1+ T-BET+\\nTIM3+\\nEOMES+\\nTOX1+ CD39+\\nCD44+ TCF-1+ T-BET+\\n0\\n20\\n40\\n60\\n80\\n100\\n120\\n140\\nNeg for all\\nTCF-1+\\nTCF-1+ T-BET+\\nT-BET+\\nCD44+\\nCD44+ TCF-1+\\nTOX1+\\nTOX1+ TCF-1+\\nCD44+ TCF-1+ T-BET+\\nCD44+ T-BET+\\nTOX1+ TCF-1+ T-BET+\\nTOX1+ T-BET+\\nEOMES+\\nPos for all except T-BET\\nPD-1+ TOX1+ TCF-1+\\n0\\n10\\n20\\n30\\n40\\n50\\n60\\nNeg for all\\nTOX1+ CD39+\\nT-BET+\\nTOX1+\\nTOX1+ EOMES+\\nCD44+\\nTOX1+ T-BET+\\nTOX1+ LAG-3+ CD39+\\nTCF-1+ T-BET+\\nTCF-1+\\nPD-1+ TOX1+ LAG-3+ CD39+\\nCD44+ T-BET+\\nPD-1+ TOX1+ LAG-3+ CD39+ EOMES+\\nEOMES+\\nTOX1+ EOMES+ CD44+\\nTx-Naive\\nαCD40\\nIA TTASNAP\\nTCF-1+\\nNeg for all\\nTCF-1+ T-BET+\\nTOX1+ TCF-1+\\nTOX1+ TCF-1+ T-BET+\\nT-BET+\\nPD-1+ TOX1+ TCF-1+\\nCD44+ TCF-1+\\nPD-1+ TOX1+ TCF-1+ T-BET+\\nTOX1+\\nPD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nCD44+\\nTOX1+ EOMES+ TCF-1+ T-BET+\\nEOMES+ TCF-1+ T-BET+\\nCD44+ TCF-1+ T-BET+\\n0\\n200\\n400\\n600\\n800\\n1000\\nNeg for all\\nTCF-1+\\nEOMES+ TCF-1+\\nTOX1+\\nEOMES+\\nEOMES+ CD44+ TCF-1+\\nTOX1+ TCF-1+\\nCD44+ TCF-1+\\nTCF-1+ T-BET+\\nT-BET+\\nEOMES+ CD44+\\nCD44+\\nPD-1+ TOX1+\\nEOMES+ TCF-1+ T-BET+\\nTOX1+ EOMES+ TCF-1+\\n0\\n200\\n400\\n600\\n800\\n1000\\nNeg for all\\nTOX1+\\nEOMES+\\nCD44+\\nTCF-1+\\nTOX1+ T-BET+\\nT-BET+\\nTOX1+ EOMES+\\nTOX1+ TCF-1+\\nTOX1+ CD44+\\nEOMES+ TCF-1+\\nTOX1+ TIM3+\\nEOMES+ CD44+\\nCD44+ TCF-1+\\nCD38+ CD44+\\n0\\n20\\n40\\n60\\n80\\n100\\n120\\n140\\nNeg for all\\nTOX1+\\nEOMES+\\nTCF-1+\\nT-BET+\\nTOX1+ EOMES+\\nCD44+\\nTCF-1+ T-BET+\\nTOX1+ TCF-1+\\nTOX1+ T-BET+\\nEOMES+ CD44+ TCF-1+ T-BET+\\nEOMES+ TCF-1+\\nEOMES+ CD44+\\nTOX1+ TIM3+\\nCD44+ TCF-1+ T-BET+\\n0\\n20\\n40\\n60\\n80\\n100\\n120\\n140\\nNeg for all\\nTOX1+\\nTOX1+ TCF-1+\\nTOX1+ EOMES+ TCF-1+\\nTOX1+ EOMES+\\nEOMES+ TCF-1+\\nEOMES+\\nPD-1+ TOX1+ EOMES+\\nPD-1+ TOX1+ EOMES+ TCF-1+\\nTCF-1+\\nTOX1+ T-BET+\\nPD-1+ TOX1+\\nT-BET+\\nCD44+\\nPD-1+ TOX1+ TCF-1+\\n0\\n10\\n20\\n30\\n40\\n50\\n60\\nTop barcode in both treatment cohorts\\nTop barcode in αCD40 only\\nTop barcode in Tx-Naive only\\nDensity (cells/mm2) Density (cells/mm2)\\nT Cell Functionality Barcodes\\nCD8+ T cells\\nCD4+ T Helper\\nTregs\\nB cells -Myeloid -Mesenchymal -\\nNeoplastic\\nModel #1\\nTx-Naive αCD40\\nBoth cohorts αCD40\\nModel #2\\nLong DFS\\nShort DFS\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5aae9280-a47a-4a75-9bba-4ad7692f60c8', embedding=None, metadata={'page_label': '63', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Accuracy\\nF1\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nModel\\nT\\nIA\\nTAS\\nNAP\\nPerformance\\nA B\\nC D\\nIAT TAS NAP\\nTrue Positive Rate\\nFalse Positive Rate\\nAUC = 0.90\\nROC Curve on Test Dataset\\n0.0 -\\n0.2 -\\n0.4 -\\n0.6 -\\n0.8 -\\n1.0 -\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nTrue Positive Rate\\nFalse Positive Rate\\nAUC = 0.90\\nROC Curve on Test Dataset\\n0.0 -\\n0.2 -\\n0.4 -\\n0.6 -\\n0.8 -\\n1.0 -\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nTrue Positive Rate\\nFalse Positive Rate\\nAUC = 0.87\\nROC Curve on Test Dataset\\n0.0 -\\n0.2 -\\n0.4 -\\n0.6 -\\n0.8 -\\n1.0 -\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nTrue Positive Rate\\nFalse Positive Rate\\nAUC = 0.90\\nROC Curve on Test Dataset\\n0.0 -\\n0.2 -\\n0.4 -\\n0.6 -\\n0.8 -\\n1.0 -\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nNaive\\nαCD40\\nT\\nIA\\nTAS\\nNAP\\nMesenchymal\\nCD8+ TEX\\nCD8+ TNAIVE\\nB cells\\nCD44+ CD4+ Th\\nCD44- CD4+ Th1OTHER\\nCD44+CD8+ TOTHER\\nCD4+ Th1EM\\nEOMES+\\nT-BET+\\nTOX1+ EOMES+\\nCD44+ T-BET+\\nCD44+\\nCD39+ CD44+\\nTOX1+\\nCD39+\\nTCF-1+\\nTIM3+ T-BET+\\nTIM3+ TCF-1+\\nTOX1+ TCF-1+ T-BET+\\nPD-1+ TOX1+ TCF-1+ T-BET+\\nPD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nTIM3+\\nPD-1+ TOX1+ TCF-1+\\nTCF-1+ T-BET+\\nCD39+ EOMES+\\nTIM3+ TCF-1+ T-BET+\\nTOX1+ CD44+ TCF-1+\\nCD44+ TCF-1+\\nTOX1+ EOMES+ TCF-1+\\nPD-1+ TOX1+ EOMES+ TCF-1+\\nTOX1+ TCF-1+\\nMyeloid cells & Ki-67+ Neoplastic\\nB cells & Myeloid cells\\nMesenchymal & Ki-67- Neoplastic\\nCD44- CD4+ Th & Myeloid cells\\nCD44- CD4+ Th1OTHER & mTREG\\nKi-67+ Neoplastic & Ki-67+ Neoplastic\\nCD44- CD4+ Th1OTHER & CD44- CD4+ Th1OTHER\\nNaive TREG & Naive TREG\\nNaive TREG & B cells\\nNaive TREG & mTREG\\nCD8+ TNAIVE & Myeloid cells\\nFeature\\nαCD40\\nNaive\\nFeature\\nImportance\\nCell State Densitites\\nT Cell Functionality\\nBarcodes\\nCell-Cell Spatial Interactions\\nT\\nIA\\nTAS\\nNAP\\nModel\\nPrediction\\nMore\\nImportant\\nLess\\nImportant\\nTOX1\\nMerge\\nCD3\\nEOMES\\nPanCKαSMAdsDNA CD8CD3 TOX1 EOMES PanCK αSMAdsDNA CD8CD3 CD44 T-BET\\nCD3\\nT-BET Merge\\nCD44\\nTx-Naive\\nTx Cohort\\nHistopath\\nNaive_NAP: PD-1+ TOX1+ EOMES+ TCF-1+\\nNaive_T: EOMES+\\nNaive_IA: TOX1+ EOMES+\\nNaive_IA: EOMES+\\nαCD40_TAS: CD39+\\nNaive_TAS: TOX1+\\nNaive_T: TOX1+\\nNaive_NAP: TOX1+ EOMES+ TCF-1+\\nNaive_NAP: TOX1+ TCF-1+\\nNaive_TAS: TOX1+ EOMES+\\nNaive_TAS: EOMES+\\nαCD40_IA: TIM3+ TCF-1+\\nαCD40_T: CD39+\\nαCD40_T: TCF-1+\\nαCD40_TAS: TCF-1+\\nαCD40_IA: CD39+ EOMES+\\nNaive_T: TOX1+ EOMES+\\nαCD40_IA: PD-1+ TOX1+ TCF-1+\\nNaive_TAS: TOX1+ EOMES+ TCF-1+\\nαCD40_IA: TIM3+\\nαCD40_IA: TOX1+ TCF-1+ T-BET+\\nαCD40_TAS: T-BET+\\nαCD40_NAP: T-BET+\\nαCD40_IA: PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nαCD40_IA: TIM3+ T-BET+\\nαCD40_TAS: TCF-1+ T-BET+\\nαCD40_IA: PD-1+ TOX1+ TCF-1+ T-BET+\\nαCD40_T: T-BET+\\nαCD40_IA: TCF-1+ T-BET+\\nαCD40_IA: TIM3+ TCF-1+ T-BET+\\nαCD40_TAS: CD44+ TCF-1+\\nαCD40_T: CD39+ CD44+\\nαCD40_TAS: TOX1+ CD44+ TCF-1+\\nαCD40_T: CD44+\\nαCD40_NAP: CD44+ T-BET+\\nαCD40_T: CD44+ T-BET+\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\n αCD40\\nE\\nT cells\\nExpressing\\nBarcodes\\ni\\nii\\niii\\niv\\nPearson Correlation\\nT cells Expressing Barcodes\\nBarcodes Expressed by T cells\\ni.\\nii.\\niii.\\niv.\\nmTREG\\nNaive TREG\\nCD44- CD4+ Th\\nCD44- CD8+ TOTHER\\nCD8+ TNAIVE\\nT-BET+ TREG\\nCD44- CD4+ Th1OTHER\\nCD44- CD8+ TOTHER\\nCD8+ TNAIVE\\nmTREG\\nNaive TREG\\nCD44+ CD4+ Th\\nCD44+ CD8+ TOTHER\\nT-BET+ TREG\\nCD44+ CD4+ Th1OTHER\\nCD4+ Th1EMRA\\nCD4+ Th1EM\\nCD44+ CD8+ TOTHER\\ni.\\nii.\\niii.\\niv.\\nPD-1+ TOX1+ EOMES+ TCF-1+\\nEOMES+\\nTOX1+ EOMES+\\nCD39+\\nTOX1+\\nTOX1+ EOMES+ TCF-1+\\nTOX1+ TCF-1+\\nTIM3+ TCF-1+\\nTCF-1+\\nCD39+ EOMES+\\nPD-1+ TOX1+ TCF-1+\\nTOX1+ EOMES+ TCF-1+\\nTIM3+\\nTOX1+ TCF-1+ T-BET+\\nT-BET+\\nPD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nTIM3+ T-BET+\\nTCF-1+ T-BET+\\nPD-1+ TOX1+ TCF-1+ T-BET+\\nTIM3+ TCF-1+ T-BET+\\nCD44+ TCF-1+\\nCD39+ CD44+\\nTOX1+ CD44+ TCF-1+\\nCD44+\\nCD44+ T-BET+\\nBlise et al. Figure 3\\nNaive_NAP: PD-1+ TOX1+ EOMES+ TCF-1+\\nNaive_T: EOMES+\\nNaive_IA: TOX1+ EOMES+\\nNaive_IA: EOMES+\\nαCD40_TAS: CD39+\\nNaive_TAS: TOX1+\\nNaive_T: TOX1+\\nNaive_NAP: TOX1+ EOMES+ TCF-1+\\nNaive_NAP: TOX1+ TCF-1+\\nNaive_TAS: TOX1+ EOMES+\\nNaive_TAS: EOMES+\\nαCD40_IA: TIM3+ TCF-1+\\nαCD40_T: CD39+\\nαCD40_T: TCF-1+\\nαCD40_TAS: TCF-1+\\nαCD40_IA: CD39+ EOMES+\\nNaive_T: TOX1+ EOMES+\\nαCD40_IA: PD-1+ TOX1+ TCF-1+\\nNaive_TAS: TOX1+ EOMES+ TCF-1+\\nαCD40_IA: TIM3+\\nαCD40_IA: TOX1+ TCF-1+ T-BET+\\nαCD40_TAS: T-BET+\\nαCD40_NAP: T-BET+\\nαCD40_IA: PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nαCD40_IA: TIM3+ T-BET+\\nαCD40_TAS: TCF-1+ T-BET+\\nαCD40_IA: PD-1+ TOX1+ TCF-1+ T-BET+\\nαCD40_T: T-BET+\\nαCD40_IA: TCF-1+ T-BET+\\nαCD40_IA: TIM3+ TCF-1+ T-BET+\\nαCD40_TAS: CD44+ TCF-1+\\nαCD40_T: CD39+ CD44+\\nαCD40_TAS: TOX1+ CD44+ TCF-1+\\nαCD40_T: CD44+\\nαCD40_NAP: CD44+ T-BET+\\nαCD40_T: CD44+ T-BET+\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4ddb38d2-68ae-49e8-9471-0397dd39afb2', embedding=None, metadata={'page_label': '64', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Accuracy\\nF1\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nModel\\nT\\nIA\\nTAS\\nPerformance\\nA B C\\nBlise et al. Figure 4\\nCD44+ CD4+ Th1OTHER\\nTIM3+ CD44+ TCF-1+ T-BET+\\nTOX1+ TCF-1+\\nTOX1+ EOMES+ TCF-1+ T-BET+\\nTOX1+ EOMES+ TCF-1+\\nPD-1+ TOX1+ TCF-1+\\nPD-1+ TOX1+ T-BET+\\nLAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+\\nPD-1+ TOX1+ TCF-1+ T-BET+\\nPD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nTOX1+ TIM3+ CD39+ TCF-1+\\nPD-1+ TOX1+\\nCD44+ T-BET+\\nCD4+ Th1EM & T-BET+ TREG\\nCD8+ TEMRA & T-BET+ TREG\\nFeature\\nTrue Positive Rate\\nFalse Positive Rate\\nAUC = 0.77\\nROC Curve on Test Dataset\\n0.0 -\\n0.2 -\\n0.4 -\\n0.6 -\\n0.8 -\\n1.0 -\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nαCD40 DFS\\nαCD40 DFS: IA\\nαCD40 DFS: IA\\nCell State Densitites\\nT Cell Functionality\\nBarcodes\\nCell-Cell Spatial\\nInteractions\\nLong\\nShort\\nFeature\\nImportance\\nModel\\nPrediction\\nMore\\nImportant\\nLess\\nImportant\\nD\\nE\\nCD3 TOX1\\nPD-1 Merge\\nCD20dsDNA CD8CD3 TOX1PD-1\\nShort\\nDFS\\nCD20dsDNA CD8CD3 CD44 CD39\\nLong\\nDFS\\nCD3 CD44\\nCD39 Merge\\nDFS\\nShort DFS: TOX1+ EOMES+ TCF-1+\\nLong DFS: TOX1+ TIM3+ CD39+ TCF-1+\\nShort DFS: TOX1+ TCF-1+\\nShort DFS: PD-1+ TOX1+ TCF-1+\\nShort DFS: PD-1+ TOX1+\\nShort DFS: PD-1+ TOX1+ T-BET+\\nShort DFS: PD-1+ TOX1+ TCF-1+ T-BET+\\nShort DFS: TOX1+ EOMES+ TCF-1+ T-BET+\\nShort DFS: PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nLong DFS: LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+\\nLong DFS: TIM3+ CD44+ TCF-1+ T-BET+\\nLong DFS: CD44+ T-BET+\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nShort DFS\\nLong DFS\\nT cells\\nExpressing\\nBarcodes\\nPearson Correlation\\ni\\nii\\niii\\niv\\nT cells Expressing Barcodes\\nBarcodes Expressed by T cells\\ni.\\nii.\\niii.\\niv.\\nmTREG\\nNaive TREG\\nCD44- CD4+ Th\\nCD44- CD8+ TOTHER\\nCD8+ TNAIVE\\nT-BET+ TREG\\nCD44- CD4+ Th1OTHER\\nCD44- CD8+ TOTHER\\nCD8+ TNAIVE\\nT-BET+ TREG\\nCD44+ CD4+ Th1OTHER\\nCD4+ Th1EMRA\\nCD4+ Th1EM\\nCD44+ CD8+ TOTHER\\ni.\\nii.\\niii.\\niv.\\nNaive TREG\\nCD44+ CD4+ Th\\nCD8+ TEMRA\\nCD8+ TEM\\nTOX1+ EOMES+ TCF-1+\\nTOX1+ TIM3+ CD39+ TCF-1+\\nTOX1+ TCF-1+\\nPD-1+ TOX1+\\nPD-1+ TOX1+ TCF-1+\\nPD-1+ TOX1+ TCF-1+ T-BET+\\nPD-1+ TOX1+ T-BET+\\nPD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nTOX1+ EOMES+ TCF-1+ T-BET+\\nLAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+\\nTIM3+ CD44+ TCF-1+ T-BET+\\nCD44+ T-BET+\\nShort DFS: TOX1+ EOMES+ TCF-1+\\nLong DFS: TOX1+ TIM3+ CD39+ TCF-1+\\nShort DFS: TOX1+ TCF-1+\\nShort DFS: PD-1+ TOX1+ TCF-1+\\nShort DFS: PD-1+ TOX1+\\nShort DFS: PD-1+ TOX1+ T-BET+\\nShort DFS: PD-1+ TOX1+ TCF-1+ T-BET+\\nShort DFS: TOX1+ EOMES+ TCF-1+ T-BET+\\nShort DFS: PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nLong DFS: LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+\\nLong DFS: TIM3+ CD44+ TCF-1+ T-BET+\\nLong DFS: CD44+ T-BET+\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c4ed4396-9499-4c03-9258-6ad7b6160727', embedding=None, metadata={'page_label': '65', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A\\nB\\nBlise et al. Figure 5\\nαCD40: IA\\nNeighborhoods created for all t in regions of interest RCNs created by K-means clustering\\nRCN1 RCN2 ... RCNk\\n60 µm\\nDFS\\nBarcode\\nRCN6\\nRCN7\\nRCN3\\nRCN5\\nRCN4\\nRCN1\\nRCN2\\nRecurrent Cellular Neighborhood\\nLong DFS: PD-1+ TOX1+\\nLong DFS: PD-1+ TOX1+ TCF-1+ T-BET+\\nLong DFS: PD-1+ TOX1+ TCF-1+\\nLong DFS: TOX1 + TCF-1+\\nShort DFS: CD44+ T-BET+\\nLong DFS: PD-1+ TOX1+ EOMES + TCF-1+ T-BET+\\nLong DFS: TIM3+ CD44+ TCF-1+ T-BET+\\nLong DFS: TOX1 + EOMES + TCF-1+\\nLong DFS: CD44+ T-BET+\\nLong DFS: LAG-3 + CD39+ EOMES + CD38+ CD44+ TCF-1+\\nLong DFS: PD-1+ TOX1+ T-BET+\\nLong DFS: TOX1 + EOMES + TCF-1+ T-BET+\\nLong DFS: TOX1 + TIM3+ CD39+ TCF-1+\\nShort DFS: TOX1 + TIM3+ CD39+ TCF-1+\\nShort DFS: PD-1+ TOX1+\\nShort DFS: PD-1+ TOX1+ T-BET+\\nShort DFS: PD-1+ TOX1+ TCF-1+ T-BET+\\nShort DFS: TOX1 + TCF-1+\\nShort DFS: PD-1+ TOX1+ EOMES + TCF-1+ T-BET+\\nShort DFS: TOX1 + EOMES + TCF-1+ T-BET+\\nShort DFS: PD-1+ TOX1+ TCF-1+\\nShort DFS: TOX1 + EOMES + TCF-1+\\nBarcode\\nShort DFS\\nLong DFS\\nCD44 + T-BET+\\nLAG-3 + CD39+ EOMES + CD38+ CD44+ TCF-1+\\nPD-1+ TOX1+\\nPD-1+ TOX1+ EOMES + TCF-1+ T-BET+\\nPD-1+ TOX1+ T-BET+\\nPD-1+ TOX1+ TCF-1+\\nPD-1+ TOX1+ TCF-1+ T-BET+\\nTIM3 + CD44+ TCF-1+ T-BET+\\nTOX1 + EOMES + TCF-1+\\nTOX1 + EOMES + TCF-1+ T-BET+\\nTOX1 + TCF-1+\\nTOX1 + TIM3+ CD39+ TCF-1+\\n0.0\\n0.5\\n1.0\\n1.5\\n2.0\\nRCN1\\nRCN2\\nRCN3\\nRCN4\\nRCN5\\nRCN6\\nRCN7\\n0-\\n1-\\n2-\\n3-\\n4-\\n5-\\n6-\\nRecurrent Cellular Neighborhood\\nPercent T cells Ki-67+\\nRCN1\\nRCN2\\nRCN3\\nRCN4\\nRCN5\\nRCN6\\nRCN7\\n0-\\n1-\\n2-\\n3-\\n4-\\n5-\\n6-\\nRecurrent Cellular Neighborhood\\nPercent T cells GrzB+\\nRCN1\\nRCN2\\nRCN3\\nRCN4\\nRCN5\\nRCN6\\nRCN7\\n0.0 -\\n0.2 -\\n0.4 -\\n0.6 -\\n0.8 -\\n1.0 -\\nKi-67+ Neoplastic\\nKi-67- Neoplastic\\nMesenchymal\\nMyeloid cells\\nB cells\\nT-BET + TREG\\nmT REG\\nNaive TREG\\nCD44 + CD4+ Th\\nCD44 - CD4+ Th\\nCD44 + CD4+ Th1OTHER\\nCD44 - CD4+ Th1OTHER\\nCD4 + Th1EMRA\\nCD4 + Th1EM\\nCD4 + Th1EFF\\nCD44 + CD8+ TOTHER\\nCD44 - CD8+ TOTHER\\nCD8 + TTEX\\nCD8 + TEX\\nCD8 + TEMRA\\nCD8 + TEM\\nCD8 + TEFF\\nCD8 + TNAIVE\\nRecurrent Cellular Neighborhood\\nAverage Cellular Composition\\nC\\nRCN1\\nRCN2\\nRCN3\\nRNC4\\nRCN5\\nRCN6\\nRCN7\\nOriginal RCN\\nD\\nC1\\n(Long DFS)\\nC2\\n(Short DFS)\\n0-\\n20-\\n40-\\n60-\\n80-\\n100-\\nRCN7\\nRCN6\\nRCN5\\nRCN4\\nRCN3\\nRCN2\\nRCN1\\nCluster\\nAverage Fraction of RCNs\\nE\\nF G\\nH\\ndsDNA\\nCD3\\nCD20\\nCD44\\nCD3\\nGrzB\\nCD3\\n CD44\\nGrzB\\nCD3\\nCD44\\nGrzB\\nRCN1\\nRCN2\\nRCN3\\nRNC4\\nRCN5\\nRCN6\\nRCN7\\nC1\\nC2\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='609f257d-b707-4264-a5de-557ddf155c4c', embedding=None, metadata={'page_label': '66', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Blise et al. Figure 6\\nTreatment-Naive αCD40-Treated\\nαCD40-Treated Immune Aggregate: Short DFS\\nαCD40-Treated Immune Aggregate: Long DFS\\nAntigen-inexperienced T cells\\n• CD44-\\n• PD-1-\\nExhausted T cells\\n• TOX1+\\n• EOMES+\\n• CD44-\\nB cells\\nImmuno-suppressive interactions\\n• TREG\\n• Myeloid\\n• Ki-67+ Neoplastic\\nActivated/effector T cells\\n• CD44+\\n• T-BET+\\n• TCF-1+\\n• CD39+\\n• PD-1+\\n• CD4+ Th1EM\\nMesenchymal\\nImmuno-reactive interactions\\n• CD4+ Th1\\nExhausted or suppressed bystander T cells\\n• TOX1+\\n• PD-1+\\n• CD44-\\nEffector CD4+ Th1\\nImmuno-reactive interactions\\n• TEM\\n• TEMRA\\n• T-BET+ TREG\\nActivated/effector T cell neighborhoods\\n• CD44+\\n• Ki-67+\\n• GrzB+\\nαCD40\\nA B\\nGlobal TME Global TME\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9b6b191c-9343-4f1a-abc2-3b7d6e131fd8', embedding=None, metadata={'page_label': '67', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A Blise et al. Figure S1\\nB\\nCD3p\\nCD8\\nDNA\\nCD3pCD8n\\nDNA\\nCD4\\nCD3p\\nCD8\\nDNA\\nCD3pCD8n\\nDNA\\nCD4\\ndsDNA\\nCD3\\nCD4\\nCD8\\nRepresentative Region #1 Representative Region #2\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='141b150e-a8f8-4467-b64b-cb5af14ad32f', embedding=None, metadata={'page_label': '68', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='C Blise et al. Figure S1\\nTo functional gates\\nImmune gates\\nFunctional gates\\nTo immune gates\\nCellsp\\nKI67\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n1.24%97.38%\\n0.02%1.36%\\nCellsp\\nPD1\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n13.95% 0.09%\\n85.55% 0.41%\\nPANCKCells_0250ROI19.cpout\\nCellsp\\nPANCK\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n0.08%0.00%\\n27.81%72.11%\\nNo Gate\\nArea\\nDSDNA\\n10\\n0\\n10\\n1\\n10\\n2\\n10\\n3\\n10\\n-1\\n10\\n0\\n1.36%\\n11.02% 87.62%\\nCellsp_PANCKn\\nCD3\\nDSDNA\\n10\\n-5\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n3.74%25.81%\\n20.83%49.62%\\nCellsp_CD3p\\nCD8\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n26.22% 18.91%\\n36.66% 18.21%\\nCellsp_CD3p_CD8n\\nFOXP3\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n0.00%0.00%\\n18.09%81.91%\\nCellsp_CD3n\\nCD20\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n76.90% 1.36%\\n21.51% 0.23%\\nCellsp_CD3n_CD20n\\nCD11B\\nDSDNA\\n10\\n-5\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n0.24%17.59%\\n1.95%80.22%\\nCellsp\\nTOX\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n12.86%78.20%\\n1.22%\\nCellsp\\nTBET\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n3.98%95.89%\\n0.04%\\nCellsp\\nTIM3\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n0.24%99.67%\\n0.00%\\nCellsp\\nLAG3\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n0.52%96.78%\\n0.04%\\nCellsp\\nTCF17\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n3.80%56.28%\\n6.53%\\nCellsp\\nCD44\\nDSDNA\\n10\\n-5\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n14.93%85.02%\\n0.00%\\nCellsp\\nCD39\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n1.07%96.53%\\n0.19%\\nCellsp\\nCD45RO\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n15.13%53.07%\\n16.12%\\nCellsp\\nCD45RA\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n1.69%96.01%\\n0.03%\\nCellsp\\nEOMES\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n0.97%48.41%\\n1.59%\\nCellsp\\nASMA\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n3.58%94.40%\\n0.00%\\nCellsp\\nGRZB\\nDSDNA\\n10\\n-5\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n0\\n10\\n-1\\n10\\n0\\n1.16% 0.19%\\n96.79% 1.86%\\nNucleiDSDNA_0250ROI19.cpout\\nA27_DSDNA\\n0 1210 2421 3631 4841\\n0\\n586\\n1173\\n1759\\n2345\\nCellsp\\nCD38\\nDSDNA\\n10\\n-4\\n10\\n-3\\n10\\n-2\\n10\\n-1\\n10\\n-1\\n10\\n0\\n0.63%98.20%\\n0.02%\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0feb8cea-116f-442e-bbe1-ea0522e0e3b3', embedding=None, metadata={'page_label': '69', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='−0.25\\n−0.75\\n−0.75\\n−0.25\\n0.0\\n0.25\\n0.75\\n−0.4\\n−0.2\\n0.0\\n0.2\\n0.4\\n1.0\\n0.5\\n0.0\\n−0.5\\n−1.0\\nA Blise et al. Figure S2\\nIA TTASNAP\\n0.0\\n0.25\\n0.75\\nSHAP value (impact on model output)\\nInteraction: CD44- CD4+ Th1OTHER  & CD44- CD4+ Th1OTHER\\nBarcode: CD38 + T-BET+\\nInteraction: Myeloid cells & Myeloid cells\\nBarcode: CD39 + TCF-1+\\nBarcode: PD-1 + TOX1 + EOMES + CD44+ TCF-1+ T-BET+\\nDensity: CD4+ Th1EM\\nBarcode: LAG-3 + CD39 + EOMES + TCF-1+\\nBarcode: EOMES + TCF-1+\\nInteraction: CD44- CD8 + TOTHER  & CD44- CD4+ Th1OTHER\\nInteraction: CD44- CD8 + TOTHER  & CD44- CD4+ Th\\nInteraction: B cells & Mesenchymal\\nBarcode: CD39 + TCF-1+ T-BET+\\nBarcode: CD39 +\\nBarcode: CD44+ TCF-1+ T-BET+\\nBarcode: EOMES + T-BET+\\nBarcode: TIM3 + TCF-1+\\nBarcode: TIM3 + T-BET+\\nBarcode: TOX1 + TCF-1+ T-BET+\\nDensity: CD8+ TNAIV-E\\nBarcode: TIM3 + TCF-1+ T-BET+\\nBarcode: CD39 + EOMES +\\nBarcode: TCF-1 + T-BET+\\nBarcode: PD-1 + TOX1 + TCF-1+\\nBarcode: TIM3 +\\nDensity: CD8+ TEX\\nDensity: B cells\\nBarcode: TOX1 + EOMES +\\nBarcode: EOMES +\\nBarcode: PD-1 + TOX1 + EOMES + TCF-1+ T-BET+\\nBarcode: PD-1 + TOX1 + TCF-1+ T-BET+\\nLow\\nHigh\\nFeature value\\nSHAP value (impact on model output)\\nInteraction: CD8+ TEX  & CD44- CD4+ Th\\nInteraction: Mesenchymal & Mesenchymal\\nInteraction: CD4+ Th1EM  & mTREG\\nInteraction: mTREG  & B cells\\nInteraction: B cells & Ki-67+ Neoplastic\\nInteraction: CD44- CD8 + TOTHER  & NaiveTREG\\nBarcode: PD-1 + TOX1 +\\nDensity: B cells\\nDensity: Ki-67- Neoplastic\\nBarcode: EOMES + CD38 + CD44+\\nBarcode: CD44+ TCF-1+\\nInteraction: CD44- CD4+ Th1OTHER  & CD44- CD4+ Th\\nInteraction: Ki-67+ Neoplastic & Ki-67+ Neoplastic\\nBarcode: TCF-1 + T-BET+\\nInteraction: CD44- CD4+ Th & Ki-67+ Neoplastic\\nInteraction: CD44- CD4+ Th & Myeloid cells\\nBarcode: CD39 + CD44+\\nBarcode: CD44+\\nBarcode: CD44+ T-BET+\\nInteraction: Mesenchymal & Ki-67- Neoplastic\\nInteraction: B cells & Myeloid cells\\nBarcode: CD39 +\\nInteraction: CD44- CD4+ Th1OTHER   & mTREG\\nDensity: Mesenchymal\\nInteraction: Myeloid cells & Ki-67+ Neoplastic\\nBarcode: T-BET +\\nBarcode: TOX1 + EOMES +\\nBarcode: TOX1 +\\nBarcode: TCF-1 +\\nBarcode: EOMES +\\nLow\\nHigh\\nFeature value\\nSHAP value (impact on model output)\\nBarcode: PD-1 + TOX1 + TIM3+ LAG-3 + CD39 + EOMES + CD38 + CD44+ TCF-1+ T-BET+\\nBarcode: TOX1 + TIM3+\\nBarcode: PD-1 + T-BET+\\nInteraction: CD44- CD4+ Th Cells & CD44+ CD4+ Th\\nInteraction: CD44+ CD4+ Th & CD44+ CD4+ Th\\nBarcode: CD44+\\nInteraction: Myeloid cells & Ki-67+ Neoplastic\\nDensity: CD8+ TNAIVE\\nBarcode: CD39 + CD38 +\\nInteraction: Mesenchymal & Mesenchymal\\nBarcode: PD-1 +\\nBarcode: EOMES + TCF-1+\\nBarcode: PD-1 + CD44+ T-BET+\\nDensity: CD44+ CD8 + TOTHER\\nDensity: CD44- CD4+ Th1OTHER\\nInteraction: B cells & Myeloid cells\\nBarcode: TOX1 +\\nBarcode: TCF-1 + T-BET+\\nDensity: Mesenchymal\\nBarcode: TCF-1 +\\nBarcode: EOMES +\\nBarcode: TOX1 + EOMES + TCF-1+\\nDensity: CD44+ CD4+ Th\\nBarcode: CD44+ TCF-1+\\nBarcode: TOX1 + CD44+ TCF-1+\\nInteraction: Ki-67+ Neoplastic & Ki-67+ Neoplastic\\nInteraction: CD44- CD4+ Th1OTHER  & CD44- CD4+ Th1OTHER\\nBarcode: T-BET +\\nBarcode: CD39 +\\nBarcode: TOX1 + EOMES +\\nLow\\nHigh\\nFeature value\\nSHAP value (impact on model output)\\nInteraction: CD44- CD4+ Th & Mesenchymal\\nBarcode: CD44+ TCF-1+ T-BET+\\nBarcode: TOX1 + EOMES +\\nBarcode: TIM3 + TCF-1+\\nInteraction: B cells & Ki-67+ Neoplastic\\nInteraction: CD8+ TNAIVE  & Ki-67- Neoplastic\\nInteraction: CD8+ TEMRA  & B cells\\nDensity: mTREG\\nBarcode: TCF-1 + T-BET+\\nBarcode: TOX1 + CD44+ TCF-1+\\nBarcode: TOX1 + CD39 + EOMES + T-BET+\\nInteraction: CD44- CD8 + TOTHER  & CD44+ CD8 + TOTHER\\nBarcode: TOX1 + CD38 +\\nBarcode: EOMES + CD44+ T-BET+\\nInteraction: CD44- CD8 + TOTHER  & Naive TREG\\nBarcode: TOX1 + TCF-1+\\nInteraction: Naive TREG  & mTREG\\nInteraction: Naive TREG  & Naive TREG\\nDensity: CD8+ TNAIVE\\nDensity: CD4+ Th1EM\\nDensity: Other CD44+ CD4+ Th\\nDensity: Mesenchymal\\nInteraction: Naive TREG  & B cells\\nBarcode: PD-1 + TOX1 + EOMES + TCF-1+\\nBarcode: CD44+ T-BET+\\nDensity: CD44- CD4+ Th1OTHER\\nBarcode: TOX1 + EOMES + TCF-1+\\nInteraction: CD8+ TNAIVE  & Myeloid cells\\nBarcode: T-BET +\\nDensity: CD44+ CD8 + TOTHER\\nLow\\nHigh\\nFeature value\\nTx-Naive\\nαCD40\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4cb60ad9-d311-4bdc-b822-e4e0826a99c0', embedding=None, metadata={'page_label': '70', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Barcode: EOMES+\\nBarcode: TCF-1+\\nBarcode: TOX1+\\nBarcode: TOX1+ EOMES+\\nBarcode: T-BET+\\nInteraction: Myeloid cells & Ki-67+ Neoplastic\\nDensity: Mesenchymal\\nInteraction: CD44- CD4+ Th1OTHER & mTREG\\nBarcode: CD39+\\nInteraction: B cells & Myeloid cells\\nInteraction: Mesenchymal & Ki-67- Neoplastic\\nBarcode: CD44+ T-BET+\\nBarcode: CD44+\\nBarcode: CD39+ CD44+\\nInteraction: CD44- CD4+ Th & Myeloid cells\\nBarcode: PD-1\\n+ TOX1\\n+ TCF-1\\n+ T-BET\\n+\\nBarcode: PD-1\\n+ TOX1\\n+ EOMES\\n+ TCF-1\\n+ T-BET\\n+\\nBarcode: EOMES\\n+\\nBarcode: TOX1\\n+ EOMES\\n+\\nDensity: B cells\\nDensity: CD8\\n+ T\\nEX\\nBarcode: TIM3\\n+\\nBarcode: PD-1\\n+ TOX1\\n+ TCF-1\\n+\\nBarcode: TCF-1\\n+ T-BET\\n+\\nBarcode: CD39\\n+ EOMES\\n+\\nBarcode: TIM3\\n+ TCF-1\\n+ T-BET\\n+\\nDensity: CD8\\n+ T\\nNAIVE\\nBarcode: TOX1\\n+ TCF-1\\n+ T-BET\\n+\\nBarcode: TIM3\\n+ T-BET\\n+\\nBarcode: TIM3\\n+ TCF-1\\n+\\nBarcode: TOX1+ EOMES+\\nBarcode: CD39+\\nBarcode: T-BET+\\nInteraction: CD44- CD4+ Th1OTHER & CD44- CD4+ Th1OTHER\\nInteraction: Ki-67+ Neoplastic & Ki-67+ Neoplastic\\nBarcode: TOX1+ CD44+ TCF-1+\\nBarcode: CD44+ TCF-1+\\nDensity: CD44+ CD4+ Th\\nBarcode: TOX1+ EOMES+ TCF-1+\\nBarcode: EOMES+\\nBarcode: TCF-1+\\nDensity: Mesenchymal\\nBarcode: TCF-1+ T-BET+\\nBarcode: TOX1+\\nInteraction: B cells & Myeloid cells\\nDensity: CD44+ CD8+ TOTHER\\nBarcode: T-BET+\\nInteraction: CD8+ TNAIVE & Myeloid cells\\nBarcode: TOX1+ EOMES+ TCF-1+\\nDensity: CD44- CD4+ Th1OTHER\\nBarcode: CD44+ T-BET+\\nBarcode: PD-1+ TOX1+ EOMES+ TCF-1+\\nInteraction: Naive TREG & B cells\\nDensity: Mesenchymal\\nDensity: CD44+ CD4+ Th\\nDensity: CD4+ Th1EM\\nDensity: CD8+ TNAIVE\\nInteraction: Naive TREG & Naive TREG\\nInteraction: Naive TREG & mTREG\\nBarcode: TOX1+ TCF-1+\\n0.0-\\n0.5-\\n1.0-\\n1.5-\\n2.0-\\n2.5-\\n3.0-\\nLog10+1 Normalized Feature Value\\n0.0-\\n1.0-\\n2.0-\\n3.0-\\n4.0-\\nTx-Naive\\nαCD40\\nLog10+1 Normalized Feature Value\\n0.0-\\n0.5-\\n1.0-\\n1.5-\\n2.0-\\n2.5-\\n3.0-\\nLog10+1 Normalized Feature Value\\n0.0-\\n0.5-\\n1.0-\\n1.5-\\n2.0-\\n2.5-\\n3.0-\\nLog10+1 Normalized Feature Value\\nB\\nBlise et al. Figure S2\\nED\\nC\\nIAT\\nTAS NAP\\nTx-Naive\\nαCD40\\n***\\n**\\n*\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n***\\n**\\n***\\n******\\n***\\n***\\n***\\n***\\n**\\n*\\n***\\n***\\n***\\n**\\n**\\n***\\n**\\n**\\n**\\n***\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b0e5d761-e544-4f59-8e4c-c059d8a79781', embedding=None, metadata={'page_label': '71', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Density: CD44+ CD4+ Th1OTHER\\nBarcode: TIM3+ CD44+ TCF-1+ T-BET+\\nBarcode: TOX1+ TCF-1+\\nBarcode: TOX1+ EOMES+ TCF-1+ T-BET+\\nBarcode: TOX1+ EOMES+ TCF-1+\\nBarcode: PD-1+ TOX1+ TCF-1+\\nBarcode: PD-1+ TOX1+ T-BET+\\nBarcode: LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+\\nBarcode: PD-1+ TOX1+ TCF-1+ T-BET+\\nBarcode: PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nBarcode: TOX1+ TIM3+ CD39+ TCF-1+\\nBarcode: PD-1+ TOX1+\\nInteraction: CD4+ Th1EM & T-BET+ TREG\\nInteraction: CD8+ TEMRA & T-BET+ TREG\\nBarcode: CD44+ T-BET+\\nA Blise et al. Figure S3\\n0.0-\\n0.5-\\n1.0-\\n1.5-\\n2.0-\\n2.5-\\n3.0-\\nShort DFS\\nLong DFS\\nLog10+1 Normalized Feature Value\\nαCD40 DFS: IA\\nLong DFS Short DFS\\n−0.8\\n−0.6\\n−0.4\\n−0.2\\n0.0\\n0.2\\nSHAP value (impact on model output)\\nBarcode: LAG-3+ CD39+ EOMES+ TCF-1+\\nBarcode: TOX1+ TCF-1+ T-BET+\\nBarcode: CD39+ CD44+\\nInteraction: CD4+ Th1EMRA & Myeloid cells\\nBarcode: TIM3+ T-BET+\\nInteraction: CD8+ TEMRA & CD8+ TEMRA\\nBarcode: CD39+ TCF-1+\\nBarcode: EOMES+ CD44+ TCF-1+\\nBarcode: PD-1+ EOMES+ CD44+ TCF-1+ T-BET+\\nBarcode: PD-1+ TOX1+ EOMES+ TCF-1+\\nInteraction: CD8+ TEMRA & Naive TREG\\nInteraction: CD8+ TEM & CD8+ TEMRA\\nInteraction: CD8+ TEM & T-BET+ TREG\\nInteraction: CD8+ TEMRA & B cells\\nDensity: CD4+ Th1EM\\nBarcode: CD44+ T-BET+\\nInteraction: CD8+ TEMRA & T-BET+ TREG\\nInteraction: CD4+ Th1EM & T-BET+ TREG\\nBarcode: PD-1+ TOX1+\\nBarcode: TOX1+ TIM3+ CD39+ TCF-1+\\nBarcode: PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+\\nBarcode: PD-1+ TOX1+ TCF-1+ T-BET+\\nBarcode: LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+\\nBarcode: PD-1+ TOX1+ T-BET+\\nBarcode: PD-1+ TOX1+ TCF-1+\\nBarcode: TOX1+ EOMES+ TCF-1+\\nBarcode: TOX1+ EOMES+ TCF-1+ T-BET+\\nBarcode: TOX1+ TCF-1+\\nBarcode: TIM3+ CD44+ TCF-1+ T-BET+\\nDensity: CD44+ CD4+ Th1OTHER\\nLow\\nHigh\\nFeature value\\nB\\n***\\n***\\n***\\n***\\n***\\n**\\n**\\n**\\n** ****\\n*\\n**\\n**\\n**\\nαCD40 DFS: IA\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4f23270f-b23b-41d8-8913-27edbe2c9cb0', embedding=None, metadata={'page_label': '72', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A\\nBlise et al. Figure S4\\n5\\n10\\n15\\n20\\nNumber of clusters\\n4000\\n6000\\n8000\\n10000\\n12000\\n14000\\nError\\nElbow Method\\nRCN1\\nRCN2\\nRCN3\\nRCN4\\nRCN5\\nRCN6\\nRCN7\\n0\\n5k\\n10k\\n15k\\n20k\\nRecurrent Cellular Neighborhood\\nNumber of cells assigned to each RCN\\nRCN1\\nRCN2\\nRCN3\\nRCN4\\nRCN5\\nRCN6\\nRCN7\\n0\\n5\\n10\\n15\\n20\\nPercentage of cells assigned to each RCN\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nRCN7\\nRCN6\\nRCN5\\nRCN4\\nRCN3\\nRCN2\\nRCN1\\nαCD40 IA Region\\nFraction of RCNs Present\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nFraction of RCNs Present\\n0\\nk = 7\\nB\\nRecurrent Cellular Neighborhood C\\nED\\nαCD40 Patient\\nRCN7\\nRCN6\\nRCN5\\nRCN4\\nRCN3\\nRCN2\\nRCN1\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='fe315bae-0d86-4be0-bb32-3a594e80285f', embedding=None, metadata={'page_label': '73', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='F\\nBlise et al. Figure S4\\nKi-67 + Neoplastic\\nKi-67 - Neoplastic\\nMesenchymal\\nMyeloid cells\\nB cells\\nT-BET + TREG\\nmT REG\\nNaive T REG\\nCD44 + CD4 + Th\\nCD44 - CD4 + Th\\nCD44 + CD4 + Th1 OTHER\\nCD44 - CD4 + Th1 OTHER\\nCD4 + Th1 EMRA\\nCD4 + Th1 EM\\nCD4 + Th1 EFF\\nCD44 + CD8+ TOTHER\\nCD44 - CD8+ TOTHER\\nCD8 + TTEX\\nCD8 + TEX\\nCD8 + TEMRA\\nCD8 + TEM\\nCD8 + TEFF\\nCD8 + TNAIVE\\nRCN1\\nRCN2\\nRCN3\\nRNC4\\nRCN5\\nRCN6\\nRCN7\\nOriginalRCNOriginalRCNOriginalRCNOriginalRCNOriginalRCNOriginalRCN\\nRCN\\nOriginal Cell State\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5230b04b-5d26-49d9-a4a8-51baced05b08', embedding=None, metadata={'page_label': '74', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Supplementary Table 1. Statistical comparison between the Liudahl et al. original PDAC cohort and \\nthe selected subset used as Cohort 1 in this study. Mean value and standard error of the mean (SEM) \\nshown for each variable. P-values computed using Fisher’s exact test for categorical variables, \\nWilcoxon rank-sum test for continuous variables, and log rank test for overall survival. \\n  \\nOriginal \\nCohort \\nSelected \\nSubset \\nP \\nOS (months) (mean ± SEM) 27.34 ± 2.25 27.27 ± 4.75 0.90 \\n# cases OHSU/DFCI 45/59 6/12 0.61 \\nCD8+ T cell density (mean ± SEM) 144.57 ± \\n11.49 \\n152.51 ± \\n26.35 \\n0.63 \\nCD4+ T cell density (mean ± SEM) 185.95 ± \\n12.45 \\n215.15 ± \\n30.41 \\n0.33 \\nCD20+ B cell density (mean ± SEM) 98.01 ± \\n10.68 \\n97.73 ± \\n21.11 \\n0.85 \\nPlasmablast density (mean ± SEM) 12.13 ± 2.13 10.78 ± 2.86 0.89 \\nPlasma cell density (mean ± SEM) 3.31 ± 0.69 2.51 ± 1.57 0.67 \\nMast cell density (mean ± SEM) 30.00 ± 2.51 21.25 ± 3.32 0.34 \\nNeutrophil/Eosinophil density \\n(mean ± SEM) \\n155.74 ± \\n29.28 \\n220.87 ± \\n94.11 \\n0.39 \\nMature Dendritic cell density (mean \\n± SEM) \\n3.32 ± 0.32 3.46 ± 0.67 0.59 \\nImmature Dendritic cell density \\n(mean ± SEM) \\n118.52 ± \\n11.86 \\n108.79 ± \\n19.72 \\n0.89 \\nCD163- Monocyte -Macrophage \\ndensity (mean ± SEM) \\n51.68 ± 4.17 37.79 ± 5.50 0.30 \\nCD163+ Monocyte -Macrophage \\ndensity (mean ± SEM) \\n73.94 ± 7.16 62.31 ± \\n14.26 \\n0.59 \\nCD8+/CD68+ Ratio (mean ± SEM) 1.86 ± 0.24 2.41 ± 0.89 0.21 \\nTotal CD68+ cell density (mean ± \\nSEM) \\n125.61 ± \\n9.84 \\n100.09 ± \\n17.97 \\n0.35 \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0bda0af2-e57f-47df-ba21-b50e6c1bc8ff', embedding=None, metadata={'page_label': '75', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Supplementary Table 2. Table of antibodies used in the mIHC panel. \\n \\nPrimary Antibody  Clone Species Supplier  Catalog number \\nTOX1 (Tox) NAN488B Rat Abcam ab237009 \\nPD-1 (PD1) NAT105 Mouse Abcam ab52597 \\nT-BET (TBET) D6N8B Rabbit Cell Signaling Technologies 13232S \\nCD39 A1 Mouse Biolegend  328202 \\nTIM3 D5D5R Rabbit Cell Signaling Technologies 45208S \\nLAG-3 (Lag3) 17B4 Mouse Novus Biologicals NBP1-97657 \\nTCF-1 (TCF1/7) C6D39 Rabbit Cell Signaling Technologies 2203 \\nCD44 156-3C11 Mouse Cell Signaling Technologies 3570 \\nCD38 38C03 Mouse Thermofisher Scientific MA5-14413 \\nCD3 SP7 Rabbit Thermofisher Scientific RM-9107-S \\nCD45RO UCHL-1 Mouse Thermofisher Scientific MA5-11532 \\nGrzB (Granzyme B) Polyclonal Rabbit Abcam  ab4059 \\nCD8 C8/144B Mouse Thermofisher Scientific MA5-13473 \\nCD11b (CD11B) EPR1344 Rabbit Abcam ab133357 \\nCD45RA F8-11-13 Mouse Abcam ab59168 \\nFOXP3 (Foxp3) 236A/E7 Mouse eBioscience 14-4777-82 \\nEOMES Polyclonal  Rabbit EMD Millipore AB2283 \\nCD20 L26 Mouse Abcam ab9475 \\nKi-67 (KI67) SP6 Rabbit Sigma/Cell Marque 275R-14 \\nαSMA (Alpha-SMA) Polyclonal  Rabbit Abcam ab5694 \\nPanCK (Pan-Cytokeratin) AE1/AE3 Mouse Abcam ab27988 \\n \\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='38e66842-8062-4b9e-be82-bc05a8111a8c', embedding=None, metadata={'page_label': '76', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Cell State Count\\nKi-67- Neoplastic 915586\\nMyeloid cells 593571\\nMesenchymal 247778\\nCD44- CD4 Th 163809\\nCD44- CD8 TOTHER 136543\\nB cells 132728\\nKi-67+ Neoplastic 115294\\nmTREG 30836\\nCD44+ CD4 Th 24232\\nCD44+ CD8 TOTHER 21711\\nCD44- CD4 Th1OTHER 21157\\nNaive TREG 10075\\nTBET+ TREG 3674\\nCD8 TEMRA 3550\\nCD8 TNAIVE 2328\\nCD4 Th1EM 2038\\nCD4 Th1EMRA 1165\\nCD8 TEM 940\\nCD8 TEX 525\\nCD44+ CD4 Th1OTHER 449\\nCD8 TTEX 200\\nCD8 TEFF 64\\nCD4 Th1EFF 21\\nSupplementary Table S3. Raw counts of cell states defined by mIHC gating strategy present in the dataset.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b972ba59-43f5-4c0c-a596-127a660ced92', embedding=None, metadata={'page_label': '77', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='d\\x03ĐĞůů\\x03&ƵŶĐƚŝŽŶĂůŝƚǇ\\x03\\x11ĂƌĐŽĚĞ \\x12ŽƵŶƚ\\nNegative for all 144053\\n TOX1+ 51606\\n TCF-1+ 50915\\n CD44+ 19724\\n T-BET+ 15539\\n EOMES+ 15405\\n TCF-1+ T-BET+ 12775\\n TOX1+ TCF-1+ 10565\\n CD44+ TCF-1+ 7742\\n EOMES+ TCF-1+ 6118\\n TOX1+ T-BET+ 5000\\n TOX1+ EOMES+ 4876\\n TOX1+ TCF-1+ T-BET+ 4403\\n EOMES+ CD44+ 3505\\n EOMES+ CD44+ TCF-1+ 3318\\n TOX1+ CD44+ 3002\\n PD-1+ TOX1+ TCF-1+ 2684\\n PD-1+ TOX1+ 2669\\n CD44+ T-BET+ 2174\\n CD44+ TCF-1+ T-BET+ 2103\\n EOMES+ TCF-1+ T-BET+ 1862\\n CD39+ 1804\\n TOX1+ EOMES+ TCF-1+ 1629\\n TOX1+ CD44+ TCF-1+ 1591\\n TOX1+ TIM3+ 1574\\n TOX1+ CD39+ 1472\\n CD38+ CD44+ 1356\\n PD-1+ TOX1+ TCF-1+ T-BET+ 1262\\n PD-1+ 1177\\n TOX1+ EOMES+ TCF-1+ T-BET+ 1153\\n EOMES+ T-BET+ 1152\\n TIM3+ 1084\\n TOX1+ CD38+ 949\\n CD38+ 936\\n PD-1+ TOX1+ EOMES+ TCF-1+ T-BET+ 923\\n TOX1+ EOMES+ CD44+ 851\\n PD-1+ TOX1+ EOMES+ 840\\n PD-1+ TCF-1+ 833\\n LAG-3+ 819\\n PD-1+ TOX1+ EOMES+ TCF-1+ 785\\n EOMES+ CD44+ TCF-1+ T-BET+ 751\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 728\\n TOX1+ EOMES+ T-BET+ 582\\nSupplementary Table S4. Raw counts of T cells expressing each functionality barcode \\npresent in the dataset.\\nSupplementary Table S4. Raw counts of T cells expressing each functionality barcode \\npresent in the dataset.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a36b0bca-b1b6-497a-ba07-f246cf437b9f', embedding=None, metadata={'page_label': '78', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TOX1+ CD44+ TCF-1+ 553\\n TOX1+ CD38+ CD44+ 512\\n TOX1+ CD39+ EOMES+ 495\\n TOX1+ CD44+ TCF-1+ T-BET+ 483\\n CD39+ EOMES+ 478\\n LAG-3+ TCF-1+ 473\\n TOX11+ CD44+ T-BET+ 461\\n PD-1+ TOX11+ T-BET+ 457\\n CD39+ TCF-1+ 448\\n TOX11+ EOMES+ CD44+ TCF-1+ 437\\n TIM3+ CD44+ 430\\n PD-1+ T-BET+ 386\\n EOMES+ CD38+ CD44+ 351\\n CD38+ CD44+ TCF-1+ 341\\n EOMES+ CD38+ CD44+ TCF-1+ 319\\n PD-1+ TOX1+ CD39+ 310\\n TOX1+ LAG-3+ CD39+ 304\\n TOX1+ LAG-3+ 303\\n EOMES+ CD44+ T-BET+ 292\\n TIM3+ TCF-1+ 281\\n LAG-3+ T-BET+ 269\\n PD-1+ EOMES+ TCF-1+ 256\\n EOMES+ CD38+ 252\\n TIM3+ T-BET+ 249\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 247\\n PD-1+ TOX1+ CD44+ 244\\n CD38+ TCF-1+ 228\\n PD-1+ TCF-1+ T-BET+ 216\\n PD-1+ EOMES+ 216\\n TOX1+ TIM3+ T-BET+ 208\\n LAG-3+ EOMES+ TCF-1+ 198\\n LAG-3+ TCF-1+ T-BET+ 193\\n CD39+ CD44+ 190\\n TOX1+ CD38+ CD44+ TCF-1+ 178\\n PD-1+ TOX1+ LAG-3+ CD39+ 175\\n PD-1+ TOX1+ CD44+ TCF-1+ T-BET+ 173\\n LAG-3+ EOMES+ 167\\n PD-1+ CD44+ TCF-1+ 166\\n TOX1+ TIM3+ CD39+ CD44+ TCF-1+ T-BET+ 165\\n TOX1+ EOMES+ CD44+ TCF-1+ T-BET+ 162\\n CD39+ T-BET+ 161\\n TOX1+ TIM3+ CD44+ 157\\n TOX1+ TIM3+ TCF-1+ 150\\n PD-1+ CD44+ 149\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d0b44839-9ee7-40f9-bd92-452397262afd', embedding=None, metadata={'page_label': '79', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TIM3+ TCF-1+ T-BET+ 144\\n TOX1+ EOMES+ CD44+ T-BET+ 144\\n CD39+ EOMES+ TCF-1+ 144\\n CD39+ TCF-1+ T-BET+ 141\\n PD-1+ TOX1+ EOMES+ T-BET+ 139\\n TOX1+ CD39+ EOMES+ TCF-1+ 138\\n TOX1+ TIM3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 137\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ 135\\n TOX1+ EOMES+ CD38+ CD44+ TCF-1+ 133\\n TOX1+ CD39+ TCF-1+ 133\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 129\\n TOX1+ CD38+ T-BET+ 125\\n PD-1+ TOX1+ CD39+ EOMES+ 124\\n EOMES+ CD38+ TCF-1+ 124\\n TOX1+ LAG-3+ TCF-1+ 122\\n PD-1+ TOX1+ LAG-3+ 118\\n PD-1+ TOX1+ EOMES+ CD44+ TCF-1+ 116\\n TOX1+ LAG-3+ EOMES+ 115\\n PD-1+ EOMES+ TCF-1+ T-BET+ 115\\n TIM3+ EOMES+ 113\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 112\\n TOX1+ EOMES+ CD38+ CD44+ 111\\n PD-1+ TOX1+ TIM3+ 109\\n LAG-3+ CD44+ TCF-1+ 109\\n LAG-3+ CD39+ 107\\n LAG-3+ CD44+ 107\\n TOX1+ TIM3+ EOMES+ 105\\n TIM3+ CD44+ TCF-1+ 103\\n PD-1+ TOX1+ CD39+ EOMES+ TCF-1+ 101\\n CD38+ T-BET+ 100\\n TOX1+ CD39+ EOMES+ CD38+ TCF-1+ 99\\n TIM3+ CD39+ 97\\n TOX1+ TIM3+ TCF-1+ T-BET+ 97\\n PD-1+ TOX1+ EOMES+ CD44+ TCF-1+ T-BET+ 97\\n LAG-3+ EOMES+ TCF-1+ T-BET+ 95\\n CD39+ EOMES+ TCF-1+ T-BET+ 94\\n CD39+ CD38+ 94\\n TOX1+ LAG-3+ CD39+ EOMES+ 94\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 94\\n CD38+ CD44+ T-BET+ 93\\n CD39+ EOMES+ CD38+ CD44+ 93\\n TOX1+ CD39+ T-BET+ 91\\n TIM3+ CD44+ T-BET+ 91\\n CD38+ TCF-1+ T-BET+ 90\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0e6cd39f-ddc9-4c04-81d3-4f2adcecfef6', embedding=None, metadata={'page_label': '80', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' CD39+ EOMES+ CD38+ 90\\n TOX1+ EOMES+ CD38+ 88\\n TOX1+ CD38+ TCF-1+ 88\\n TOX1+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 88\\n TOX1+ LAG-3+ EOMES+ TCF-1+ 86\\n CD38+ CD44+ TCF-1+ T-BET+ 85\\n LAG-3+ EOMES+ CD44+ TCF-1+ 83\\n CD39+ CD38+ CD44+ 83\\n TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 82\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 78\\n PD-1+ TOX1+ LAG-3+ EOMES+ 75\\n TOX1+ EOMES+ CD38+ TCF-1+ 75\\n PD-1+ TOX1+ CD39+ TCF-1+ 73\\n PD-1+ CD44+ TCF-1+ T-BET+ 71\\n PD-1+ TOX1+ CD38+ 70\\n PD-1+ EOMES+ T-BET+ 69\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 67\\n PD-1+ TOX1+ EOMES+ CD44+ 67\\n PD-1+ EOMES+ CD44+ TCF-1+ T-BET+ 67\\n TOX1+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 64\\n TOX1+ CD39+ TCF-1+ T-BET+ 63\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 63\\n TIM3+ EOMES+ CD44+ 62\\n CD39+ EOMES+ T-BET+ 61\\n TOX1+ TIM3+ EOMES+ TCF-1+ 61\\n TIM3+ EOMES+ T-BET+ 60\\n TOX1+ LAG-3+ T-BET+ 60\\n PD-1+ EOMES+ CD44+ TCF-1+ 59\\n CD39+ EOMES+ CD44+ 59\\n PD-1+ TOX1+ LAG-3+ EOMES+ TCF-1+ 59\\n LAG-3+ CD39+ TCF-1+ 57\\n PD-1+ TOX1+ CD44+ T-BET+ 57\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 56\\n LAG-3+ CD44+ TCF-1+ T-BET+ 56\\n TIM3+ EOMES+ TCF-1+ 54\\n PD-1+ TOX1+ LAG-3+ T-BET+ 53\\n PD-1+ TOX1+ CD39+ EOMES+ TCF-1+ T-BET+ 53\\n CD39+ EOMES+ CD38+ CD44+ TCF-1+ 53\\n PD-1+ TOX1+ TIM3+ TCF-1+ 53\\n CD39+ CD44+ TCF-1+ 52\\n TOX1+ TIM3+ CD44+ TCF-1+ 52\\n PD-1+ TOX1+ LAG-3+ TCF-1+ 51\\n TOX1+ CD39+ EOMES+ CD38+ 50\\n CD39+ EOMES+ CD38+ TCF-1+ 49\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1be4bdc1-eb73-4a69-ba15-453454b98023', embedding=None, metadata={'page_label': '81', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ CD44+ T-BET+ 48\\n TOX1+ CD39+ EOMES+ CD38+ CD44+ 47\\n TIM3+ EOMES+ CD44+ TCF-1+ 47\\n PD-1+ TOX1+ CD39+ TCF-1+ T-BET+ 47\\n TOX1+ CD39+ CD44+ TCF-1+ T-BET+ 47\\n CD39+ EOMES+ CD44+ TCF-1+ 46\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 45\\n PD-1+ TOX1+ TIM3+ TCF-1+ T-BET+ 44\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 43\\n TOX1+ CD39+ CD44+ 43\\n CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 43\\n LAG-3+ CD38+ CD44+ TCF-1+ 43\\n TOX1+ TIM3+ CD39+ 43\\n LAG-3+ EOMES+ CD44+ 42\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ 42\\n TOX1+ CD38+ CD44+ TCF-1+ T-BET+ 42\\n EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 42\\n LAG-3+ EOMES+ T-BET+ 41\\n TIM3+ CD39+ TCF-1+ 41\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ 41\\n PD-1+ CD39+ 40\\n LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 40\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 40\\n LAG-3+ CD39+ EOMES+ 39\\n PD-1+ EOMES+ CD44+ 39\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 38\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 37\\n PD-1+ TOX1+ LAG-3+ EOMES+ T-BET+ 37\\n TOX1+ LAG-3+ TCF-1+ T-BET+ 37\\n PD-1+ TOX1+ TIM3+ T-BET+ 37\\n TOX1+ TIM3+ CD38+ CD44+ 37\\n PD-1+ TOX1+ CD38+ CD44+ 37\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ TCF-1+ 36\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 36\\n TOX1+ CD39+ EOMES+ T-BET+ 35\\n TOX1+ TIM3+ EOMES+ CD44+ 35\\n PD-1+ TOX1+ TIM3+ CD38+ CD44+ TCF-1+ T-BET+ 34\\n TIM3+ CD39+ CD44+ 34\\n LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 34\\n TOX1+ TIM3+ CD38+ 34\\n TOX1+ CD38+ TCF-1+ T-BET+ 33\\n PD-1+ LAG-3+ EOMES+ TCF-1+ 33\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 32\\n PD-1+ TOX1+ TIM3+ CD38+ CD44+ T-BET+ 32\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='395179af-4799-4d0d-a2a7-071eaa36d98d', embedding=None, metadata={'page_label': '82', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' LAG-3+ CD44+ T-BET+ 32\\n TOX1+ CD39+ EOMES+ CD44+ TCF-1+ 32\\n TIM3+ CD44+ TCF-1+ T-BET+ 32\\n TIM3+ CD39+ CD38+ 32\\n CD39+ CD44+ TCF-1+ T-BET+ 32\\n TIM3+ CD38+ CD44+ 31\\n LAG-3+ EOMES+ CD38+ CD44+ 31\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 31\\n PD-1+ LAG-3+ EOMES+ 31\\n PD-1+ TOX1+ EOMES+ CD44+ T-BET+ 31\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 31\\n LAG-3+ CD38+ TCF-1+ 31\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 30\\n TOX1+ CD39+ CD38+ 30\\n TOX1+ TIM3+ CD44+ T-BET+ 30\\n TIM3+ EOMES+ TCF-1+ T-BET+ 30\\n TOX1+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 30\\n PD-1+ LAG-3+ 30\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 30\\n TOX1+ CD39+ CD44+ TCF-1+ 29\\n PD-1+ TOX1+ CD39+ T-BET+ 29\\n TOX1+ TIM3+ CD44+ TCF-1+ T-BET+ 29\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 29\\n LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 29\\n LAG-3+ CD39+ EOMES+ CD38+ CD44+ 29\\n TIM3+ LAG-3+ 29\\n LAG-3+ CD38+ 28\\n TOX1+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 28\\n LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 28\\n TIM3+ LAG-3+ T-BET+ 28\\n TOX1+ LAG-3+ EOMES+ TCF-1+ T-BET+ 28\\n TOX1+ CD39+ EOMES+ TCF-1+ T-BET+ 28\\n TOX1+ TIM3+ EOMES+ CD44+ TCF-1+ 28\\n LAG-3+ CD39+ EOMES+ CD38+ 28\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ TCF-1+ 27\\n TIM3+ EOMES+ CD44+ TCF-1+ T-BET+ 27\\n PD-1+ TOX1+ LAG-3+ CD39+ TCF-1+ 27\\n PD-1+ LAG-3+ TCF-1+ 27\\n PD-1+ LAG-3+ CD38+ TCF-1+ T-BET+ 27\\n TOX1+ LAG-3+ CD39+ TCF-1+ 27\\n PD-1+ TOX1+ LAG-3+ CD39+ T-BET+ 27\\n TOX1+ CD38+ CD44+ T-BET+ 26\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 26\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ 26\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='27ff553a-910c-4060-a8f9-e9cc67ac6ee2', embedding=None, metadata={'page_label': '83', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' LAG-3+ EOMES+ CD38+ TCF-1+ 26\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 26\\n LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 26\\n TOX1+ TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 26\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD44+ TCF-1+ 26\\n LAG-3+ CD39+ EOMES+ TCF-1+ 26\\n PD-1+ TIM3+ 25\\n CD39+ CD38+ CD44+ TCF-1+ 25\\n PD-1+ TOX1+ LAG-3+ TCF-1+ T-BET+ 25\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 25\\n LAG-3+ CD38+ CD44+ 25\\n TOX1+ TIM3+ LAG-3+ 25\\n LAG-3+ CD39+ TCF-1+ T-BET+ 25\\n TOX1+ LAG-3+ EOMES+ T-BET+ 25\\n TOX1+ TIM3+ CD38+ CD44+ TCF-1+ T-BET+ 25\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 24\\n PD-1+ TOX1+ CD39+ EOMES+ CD38+ 24\\n CD39+ CD38+ TCF-1+ 24\\n TIM3+ EOMES+ CD38+ CD44+ TCF-1+ 24\\n TOX1+ TIM3+ EOMES+ CD38+ CD44+ TCF-1+ 23\\n PD-1+ TOX1+ TIM3+ CD44+ TCF-1+ 23\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ 23\\n PD-1+ TOX1+ CD39+ CD44+ 23\\n PD-1+ TOX1+ CD39+ EOMES+ CD38+ TCF-1+ 23\\n PD-1+ TOX1+ TIM3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 23\\n CD39+ CD44+ T-BET+ 23\\n PD-1+ TOX1+ CD39+ EOMES+ CD44+ TCF-1+ 23\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 22\\n LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 22\\n TOX1+ TIM3+ EOMES+ T-BET+ 22\\n TOX1+ CD39+ EOMES+ CD44+ 22\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 22\\n EOMES+ CD38+ TCF-1+ T-BET+ 22\\n LAG-3+ CD38+ TCF-1+ T-BET+ 22\\n TIM3+ CD39+ T-BET+ 22\\n TOX1+ LAG-3+ CD39+ T-BET+ 22\\n LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 22\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 22\\n PD-1+ LAG-3+ EOMES+ CD44+ TCF-1+ 21\\n PD-1+ TOX1+ CD39+ CD44+ TCF-1+ 21\\n PD-1+ EOMES+ CD44+ T-BET+ 21\\n TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 21\\n TIM3+ LAG-3+ EOMES+ TCF-1+ 21\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ CD44+ 21\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='41dd182e-c033-4d4a-8e6c-eeb9cd273fa3', embedding=None, metadata={'page_label': '84', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ TIM3+ EOMES+ TCF-1+ T-BET+ 21\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ T-BET+ 21\\n TOX1+ EOMES+ CD38+ T-BET+ 21\\n TOX1+ LAG-3+ EOMES+ CD38+ TCF-1+ 20\\n TIM3+ LAG-3+ TCF-1+ 20\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ 20\\n TOX1+ TIM3+ CD38+ TCF-1+ T-BET+ 20\\n TOX1+ TIM3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 20\\n PD-1+ TOX1+ TIM3+ CD44+ 20\\n TOX1+ TIM3+ CD39+ TCF-1+ 20\\n PD-1+ TOX1+ EOMES+ CD38+ 20\\n EOMES+ CD38+ CD44+ T-BET+ 20\\n PD-1+ TOX1+ CD38+ CD44+ TCF-1+ 20\\n PD-1+ TOX1+ CD39+ EOMES+ T-BET+ 19\\n TOX1+ CD39+ CD44+ T-BET+ 19\\n TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 19\\n LAG-3+ CD39+ T-BET+ 19\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ 19\\n LAG-3+ EOMES+ CD38+ 19\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 19\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 19\\n PD-1+ CD39+ EOMES+ 18\\n CD39+ CD38+ CD44+ TCF-1+ T-BET+ 18\\n TIM3+ CD39+ CD38+ CD44+ 18\\n TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 18\\n PD-1+ TOX1+ CD38+ TCF-1+ 18\\n PD-1+ TOX1+ LAG-3+ EOMES+ TCF-1+ T-BET+ 18\\n TIM3+ CD38+ 18\\n TOX1+ LAG-3+ CD44+ TCF-1+ 18\\n LAG-3+ CD39+ CD44+ TCF-1+ 17\\n PD-1+ CD39+ TCF-1+ 17\\n CD39+ CD38+ T-BET+ 17\\n PD-1+ TOX1+ TIM3+ EOMES+ TCF-1+ 17\\n TIM3+ CD38+ CD44+ TCF-1+ T-BET+ 17\\n PD-1+ TOX1+ CD38+ T-BET+ 17\\n PD-1+ TOX1+ EOMES+ CD38+ CD44+ 17\\n PD-1+ TOX1+ CD39+ EOMES+ CD38+ CD44+ 17\\n PD-1+ TOX1+ TIM3+ EOMES+ 17\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 17\\n PD-1+ TOX1+ TIM3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 17\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD44+ TCF-1+ T-BET+ 17\\n TIM3+ EOMES+ CD44+ T-BET+ 17\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 17\\n LAG-3+ CD39+ CD38+ 17\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2bf62fa4-6f01-4826-b085-f704bf06d225', embedding=None, metadata={'page_label': '85', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TOX1+ TIM3+ CD38+ CD44+ 16\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 16\\n LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 16\\n PD-1+ LAG-3+ T-BET+ 16\\n CD39+ CD38+ CD44+ T-BET+ 16\\n PD-1+ LAG-3+ TCF-1+ T-BET+ 16\\n TOX1+ LAG-3+ EOMES+ CD38+ 16\\n TOX1+ TIM3+ EOMES+ CD38+ TCF-1+ 16\\n TOX1+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 16\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD44+ 15\\n TOX1+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 15\\n PD-1+ TOX1+ LAG-3+ CD44+ 15\\n PD-1+ LAG-3+ CD38+ TCF-1+ 15\\n PD-1+ TOX1+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 14\\n PD-1+ TOX1+ EOMES+ CD38+ CD44+ TCF-1+ 14\\n LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 14\\n LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 14\\n LAG-3+ CD39+ CD38+ T-BET+ 14\\n EOMES+ CD38+ T-BET+ 14\\n TOX1+ LAG-3+ CD44+ 14\\n PD-1+ TOX1+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 14\\n TOX1+ TIM3+ EOMES+ CD44+ TCF-1+ T-BET+ 14\\n PD-1+ TOX1+ TIM3+ CD39+ CD38+ CD44+ T-BET+ 14\\n PD-1+ LAG-3+ CD39+ 14\\n PD-1+ TOX1+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 14\\n PD-1+ LAG-3+ CD39+ T-BET+ 14\\n TOX1+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 14\\n TOX1+ EOMES+ CD38+ TCF-1+ T-BET+ 14\\n TIM3+ LAG-3+ EOMES+ T-BET+ 14\\n TIM3+ CD39+ EOMES+ 14\\n PD-1+ TIM3+ T-BET+ 14\\n PD-1+ CD39+ TCF-1+ T-BET+ 14\\n PD-1+ TOX1+ LAG-3+ CD39+ TCF-1+ T-BET+ 13\\n TOX1+ LAG-3+ CD38+ TCF-1+ T-BET+ 13\\n PD-1+ TOX1+ CD38+ TCF-1+ T-BET+ 13\\n TIM3+ LAG-3+ CD38+ TCF-1+ T-BET+ 13\\n LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 13\\n LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 13\\n TOX1+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 13\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 13\\n PD-1+ TOX1+ CD39+ EOMES+ CD44+ 13\\n PD-1+ TIM3+ LAG-3+ CD38+ TCF-1+ T-BET+ 13\\n PD-1+ TOX1+ TIM3+ CD44+ TCF-1+ T-BET+ 13\\n TOX1+ CD39+ CD38+ CD44+ 13\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f0067df0-5803-4ece-9598-7ee32fe8f6e2', embedding=None, metadata={'page_label': '86', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ LAG-3+ CD39+ CD38+ 13\\n TOX1+ TIM3+ CD39+ EOMES+ 13\\n TOX1+ CD39+ CD38+ T-BET+ 13\\n PD-1+ TOX1+ TIM3+ CD38+ TCF-1+ T-BET+ 13\\n PD-1+ TOX1+ TIM3+ EOMES+ TCF-1+ T-BET+ 13\\n LAG-3+ CD39+ CD38+ CD44+ 12\\n TIM3+ CD39+ TCF-1+ T-BET+ 12\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD44+ TCF-1+ 12\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ TCF-1+ 12\\n CD39+ CD38+ TCF-1+ T-BET+ 12\\n PD-1+ TOX1+ LAG-3+ CD44+ TCF-1+ 12\\n PD-1+ TOX1+ LAG-3+ CD44+ TCF-1+ T-BET+ 12\\n PD-1+ TOX1+ CD39+ CD38+ 12\\n PD-1+ TOX1+ EOMES+ CD38+ TCF-1+ 12\\n TOX1+ LAG-3+ EOMES+ CD44+ TCF-1+ 12\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ TCF-1+ 12\\n PD-1+ TIM3+ TCF-1+ 12\\n TOX1+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 12\\n TOX1+ TIM3+ EOMES+ CD38+ TCF-1+ T-BET+ 12\\n PD-1+ LAG-3+ CD39+ EOMES+ TCF-1+ 12\\n LAG-3+ EOMES+ CD44+ T-BET+ 12\\n PD-1+ TIM3+ CD44+ 12\\n TOX1+ LAG-3+ CD38+ 12\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 12\\n TOX1+ LAG-3+ CD44+ T-BET+ 12\\n TIM3+ LAG-3+ CD39+ 12\\n TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 12\\n TOX1+ LAG-3+ EOMES+ CD38+ CD44+ 12\\n LAG-3+ CD38+ T-BET+ 12\\n CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 12\\n TIM3+ LAG-3+ CD38+ T-BET+ 12\\n PD-1+ CD39+ EOMES+ TCF-1+ T-BET+ 12\\n TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 12\\n PD-1+ CD39+ T-BET+ 12\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 12\\n PD-1+ TOX1+ EOMES+ CD38+ TCF-1+ T-BET+ 11\\n PD-1+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 11\\n PD-1+ LAG-3+ EOMES+ TCF-1+ T-BET+ 11\\n LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 11\\n TOX1+ LAG-3+ CD39+ TCF-1+ T-BET+ 11\\n TOX1+ CD39+ EOMES+ CD44+ T-BET+ 11\\n PD-1+ LAG-3+ CD44+ TCF-1+ 11\\n CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 11\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ TCF-1+ T-BET+ 11\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='935b9a29-4db3-4974-8f46-91232362c93a', embedding=None, metadata={'page_label': '87', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TIM3+ CD38+ T-BET+ 11\\n TOX1+ TIM3+ EOMES+ CD38+ CD44+ 11\\n TIM3+ LAG-3+ EOMES+ 11\\n TOX1+ TIM3+ EOMES+ CD44+ T-BET+ 11\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 11\\n PD-1+ TOX1+ LAG-3+ CD39+ CD44+ TCF-1+ 11\\n TIM3+ LAG-3+ CD44+ 11\\n LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 11\\n LAG-3+ CD39+ EOMES+ CD44+ 11\\n PD-1+ TOX1+ TIM3+ CD38+ 11\\n TOX1+ TIM3+ CD38+ T-BET+ 11\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 11\\n PD-1+ CD38+ 11\\n TOX1+ TIM3+ CD38+ TCF-1+ 11\\n TIM3+ CD38+ TCF-1+ T-BET+ 10\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 10\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 10\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 10\\n TIM3+ CD38+ CD44+ T-BET+ 10\\n TIM3+ LAG-3+ CD38+ CD44+ T-BET+ 10\\n PD-1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 10\\n TIM3+ LAG-3+ EOMES+ TCF-1+ T-BET+ 10\\n PD-1+ TOX1+ CD39+ CD38+ CD44+ 10\\n PD-1+ LAG-3+ EOMES+ T-BET+ 10\\n PD-1+ TOX1+ CD39+ EOMES+ CD44+ T-BET+ 10\\n PD-1+ TOX1+ TIM3+ CD39+ 10\\n PD-1+ TOX1+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 10\\n PD-1+ TOX1+ TIM3+ LAG-3+ 10\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ CD44+ T-BET+ 10\\n TOX1+ LAG-3+ CD39+ EOMES+ T-BET+ 10\\n LAG-3+ CD39+ CD44+ 10\\n TIM3+ CD39+ CD44+ TCF-1+ 10\\n TOX1+ TIM3+ LAG-3+ EOMES+ TCF-1+ 10\\n TIM3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 9\\n PD-1+ TIM3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 9\\n TIM3+ CD39+ CD38+ TCF-1+ 9\\n PD-1+ CD39+ CD44+ TCF-1+ 9\\n PD-1+ CD38+ TCF-1+ 9\\n PD-1+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 9\\n PD-1+ TIM3+ CD44+ TCF-1+ T-BET+ 9\\n TOX1+ EOMES+ CD38+ CD44+ T-BET+ 9\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 9\\n PD-1+ TOX1+ TIM3+ CD39+ TCF-1+ T-BET+ 9\\n LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 9\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ac1a7ad4-861a-44b6-ad09-0fe8afe73b0b', embedding=None, metadata={'page_label': '88', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TOX1+ TIM3+ CD39+ EOMES+ TCF-1+ 9\\n TOX1+ TIM3+ CD39+ EOMES+ TCF-1+ 9\\n TOX1+ LAG-3+ CD38+ T-BET+ 9\\n PD-1+ TOX1+ LAG-3+ CD38+ 9\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ 9\\n TOX1+ TIM3+ CD39+ CD38+ TCF-1+ T-BET+ 9\\n TOX1+ TIM3+ CD38+ CD44+ TCF-1+ 9\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ 9\\n TOX1+ LAG-3+ CD39+ CD44+ TCF-1+ 9\\n TIM3+ CD38+ CD44+ TCF-1+ 9\\n PD-1+ LAG-3+ EOMES+ CD38+ CD44+ 8\\n TOX1+ LAG-3+ EOMES+ CD44+ 8\\n PD-1+ TOX1+ TIM3+ CD39+ CD44+ TCF-1+ T-BET+ 8\\n PD-1+ TOX1+ TIM3+ EOMES+ CD44+ TCF-1+ 8\\n PD-1+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 8\\n PD-1+ TOX1+ TIM3+ EOMES+ CD44+ 8\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ 8\\n PD-1+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 8\\n PD-1+ TOX1+ TIM3+ CD44+ T-BET+ 8\\n LAG-3+ EOMES+ CD38+ T-BET+ 8\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ CD44+ 8\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ 8\\n CD39+ EOMES+ CD38+ CD44+ T-BET+ 8\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ 8\\n CD39+ EOMES+ CD38+ T-BET+ 8\\n CD39+ EOMES+ CD44+ T-BET+ 8\\n PD-1+ CD39+ EOMES+ CD44+ TCF-1+ 8\\n TOX1+ CD39+ CD38+ TCF-1+ 8\\n PD-1+ CD39+ EOMES+ CD44+ 8\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ 8\\n TOX1+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 8\\n PD-1+ CD39+ EOMES+ TCF-1+ 8\\n TOX1+ TIM3+ LAG-3+ TCF-1+ 8\\n PD-1+ TOX1+ CD39+ CD44+ T-BET+ 8\\n PD-1+ TOX1+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 8\\n TOX1+ LAG-3+ CD39+ EOMES+ CD44+ 8\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD44+ T-BET+ 8\\n PD-1+ TOX1+ CD39+ CD44+ TCF-1+ T-BET+ 8\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 8\\n PD-1+ TOX1+ CD38+ CD44+ TCF-1+ T-BET+ 8\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 8\\n TIM3+ EOMES+ CD38+ CD44+ 8\\n TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 8\\n PD-1+ TIM3+ EOMES+ 8\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9da95a63-01fa-457d-b811-9f8609b687b8', embedding=None, metadata={'page_label': '89', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ TIM3+ LAG-3+ CD39+ 8\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 8\\n PD-1+ TOX1+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 8\\n PD-1+ LAG-3+ CD38+ 7\\n TIM3+ EOMES+ CD38+ 7\\n PD-1+ TOX1+ CD38+ CD44+ T-BET+ 7\\n PD-1+ TOX1+ TIM3+ CD38+ T-BET+ 7\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ 7\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ 7\\n PD-1+ TOX1+ LAG-3+ CD44+ T-BET+ 7\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 7\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ 7\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 7\\n LAG-3+ CD39+ CD38+ TCF-1+ 7\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD44+ 7\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 7\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 7\\n PD-1+ TOX1+ EOMES+ CD38+ T-BET+ 7\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ T-BET+ 7\\n TOX1+ LAG-3+ CD39+ CD38+ T-BET+ 7\\n TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ 7\\n TOX1+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 7\\n TOX1+ LAG-3+ CD39+ CD44+ 7\\n TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 7\\n TOX1+ TIM3+ LAG-3+ CD38+ TCF-1+ T-BET+ 7\\n TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ 7\\n TOX1+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 7\\n PD-1+ CD38+ CD44+ TCF-1+ 7\\n PD-1+ CD38+ CD44+ 7\\n TOX1+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 7\\n TOX1+ TIM3+ LAG-3+ EOMES+ TCF-1+ T-BET+ 7\\n TOX1+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 7\\n TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 7\\n TIM3+ LAG-3+ TCF-1+ T-BET+ 7\\n TOX1+ TIM3+ CD39+ TCF-1+ T-BET+ 7\\n TOX1+ TIM3+ CD39+ T-BET+ 7\\n TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ 7\\n LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 6\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 6\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 6\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 6\\n LAG-3+ CD39+ EOMES+ T-BET+ 6\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ TCF-1+ T-BET+ 6\\n TIM3+ LAG-3+ CD39+ EOMES+ 6\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='927f9b9e-e6a7-4299-bc68-4da038d15426', embedding=None, metadata={'page_label': '90', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TOX1+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 6\\n PD-1+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 6\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 6\\n TOX1+ TIM3+ LAG-3+ CD44+ TCF-1+ T-BET+ 6\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ 6\\n TOX1+ LAG-3+ CD44+ TCF-1+ T-BET+ 6\\n LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 6\\n PD-1+ TOX1+ TIM3+ LAG-3+ TCF-1+ T-BET+ 6\\n TIM3+ LAG-3+ CD39+ CD38+ 6\\n TOX1+ TIM3+ CD39+ CD38+ CD44+ 6\\n PD-1+ TIM3+ CD44+ TCF-1+ 6\\n TOX1+ LAG-3+ CD39+ CD44+ T-BET+ 6\\n PD-1+ CD39+ CD38+ T-BET+ 6\\n TOX1+ CD39+ CD38+ CD44+ TCF-1+ 6\\n TIM3+ CD39+ CD44+ TCF-1+ T-BET+ 6\\n TIM3+ EOMES+ CD38+ TCF-1+ T-BET+ 6\\n PD-1+ TOX1+ TIM3+ CD39+ T-BET+ 6\\n TOX1+ LAG-3+ EOMES+ CD38+ T-BET+ 6\\n PD-1+ TOX1+ TIM3+ CD39+ TCF-1+ 6\\n PD-1+ LAG-3+ CD44+ 6\\n PD-1+ LAG-3+ CD44+ T-BET+ 6\\n LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 6\\n PD-1+ TIM3+ EOMES+ CD44+ TCF-1+ 6\\n TOX1+ TIM3+ CD39+ CD44+ 6\\n TIM3+ CD39+ CD38+ T-BET+ 6\\n TOX1+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 6\\n PD-1+ TOX1+ LAG-3+ CD38+ TCF-1+ T-BET+ 6\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ TCF-1+ T-BET+ 6\\n TIM3+ CD39+ CD38+ TCF-1+ T-BET+ 6\\n PD-1+ CD38+ TCF-1+ T-BET+ 6\\n TOX1+ CD39+ EOMES+ CD38+ T-BET+ 6\\n PD-1+ LAG-3+ CD39+ EOMES+ T-BET+ 6\\n TIM3+ LAG-3+ CD39+ CD38+ T-BET+ 6\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ T-BET+ 5\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ 5\\n TOX1+ TIM3+ CD39+ CD38+ CD44+ T-BET+ 5\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ T-BET+ 5\\n PD-1+ TOX1+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 5\\n TOX1+ TIM3+ CD39+ CD38+ T-BET+ 5\\n TOX1+ TIM3+ CD39+ CD38+ 5\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ 5\\n PD-1+ TIM3+ EOMES+ TCF-1+ T-BET+ 5\\n PD-1+ TIM3+ EOMES+ CD44+ TCF-1+ T-BET+ 5\\n PD-1+ TOX1+ TIM3+ CD38+ CD44+ TCF-1+ 5\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='bc21569d-1ba0-4552-be1b-c91de52123c4', embedding=None, metadata={'page_label': '91', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ LAG-3+ CD39+ CD38+ TCF-1+ 5\\n TOX1+ LAG-3+ CD39+ EOMES+ TCF-1+ 5\\n PD-1+ TOX1+ LAG-3+ CD39+ CD44+ 5\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ T-BET+ 5\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 5\\n TOX1+ TIM3+ CD39+ CD44+ TCF-1+ 5\\n TOX1+ LAG-3+ CD38+ CD44+ 5\\n TOX1+ TIM3+ LAG-3+ CD38+ TCF-1+ 5\\n PD-1+ LAG-3+ CD39+ TCF-1+ T-BET+ 5\\n TOX1+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 5\\n PD-1+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 5\\n PD-1+ LAG-3+ EOMES+ CD44+ 5\\n PD-1+ EOMES+ CD38+ TCF-1+ 5\\n TIM3+ LAG-3+ CD38+ TCF-1+ 5\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD44+ 5\\n TOX1+ TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ 5\\n TOX1+ TIM3+ LAG-3+ EOMES+ 5\\n TIM3+ CD39+ EOMES+ TCF-1+ T-BET+ 5\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 5\\n TIM3+ EOMES+ CD38+ CD44+ T-BET+ 5\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ 5\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 5\\n PD-1+ CD39+ CD38+ TCF-1+ 5\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 5\\n PD-1+ LAG-3+ CD39+ TCF-1+ 5\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 5\\n PD-1+ LAG-3+ CD39+ EOMES+ 5\\n PD-1+ TOX1+ LAG-3+ CD38+ CD44+ 5\\n TOX1+ TIM3+ CD39+ EOMES+ CD44+ TCF-1+ 5\\n TOX1+ TIM3+ LAG-3+ TCF-1+ T-BET+ 5\\n PD-1+ TOX1+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 5\\n PD-1+ TIM3+ LAG-3+ CD39+ CD38+ T-BET+ 5\\n PD-1+ TOX1+ LAG-3+ CD38+ T-BET+ 5\\n TIM3+ LAG-3+ CD38+ 5\\n PD-1+ LAG-3+ CD39+ CD44+ TCF-1+ 5\\n TOX1+ LAG-3+ CD38+ TCF-1+ 5\\n TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 5\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 4\\n PD-1+ TIM3+ LAG-3+ T-BET+ 4\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ 4\\n TIM3+ EOMES+ CD38+ TCF-1+ 4\\n TOX1+ TIM3+ CD39+ EOMES+ TCF-1+ T-BET+ 4\\n TIM3+ EOMES+ CD38+ T-BET+ 4\\n TOX1+ TIM3+ CD39+ CD38+ CD44+ TCF-1+ 4\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='44a1a2df-7829-4c38-b499-9317ceb845bb', embedding=None, metadata={'page_label': '92', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 4\\n TOX1+ TIM3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 4\\n PD-1+ TOX1+ CD39+ CD38+ T-BET+ 4\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ 4\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 4\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 4\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 4\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD38+ TCF-1+ T-BET+ 4\\n PD-1+ TIM3+ EOMES+ TCF-1+ 4\\n TIM3+ LAG-3+ EOMES+ CD44+ T-BET+ 4\\n PD-1+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ CD39+ CD38+ CD44+ 4\\n TOX1+ LAG-3+ CD38+ CD44+ TCF-1+ 4\\n TOX1+ LAG-3+ CD38+ CD44+ T-BET+ 4\\n PD-1+ CD39+ CD44+ 4\\n PD-1+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 4\\n TIM3+ CD39+ EOMES+ CD44+ TCF-1+ 4\\n PD-1+ LAG-3+ CD44+ TCF-1+ T-BET+ 4\\n PD-1+ TIM3+ TCF-1+ T-BET+ 4\\n PD-1+ EOMES+ CD38+ CD44+ TCF-1+ 4\\n TIM3+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 4\\n PD-1+ LAG-3+ CD38+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD44+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ 4\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ TCF-1+ 4\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ TCF-1+ T-BET+ 4\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 4\\n PD-1+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ EOMES+ T-BET+ 4\\n TOX1+ LAG-3+ CD39+ CD38+ CD44+ 4\\n PD-1+ TIM3+ CD44+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ EOMES+ CD44+ TCF-1+ T-BET+ 4\\n TOX1+ TIM3+ CD38+ CD44+ T-BET+ 4\\n TOX1+ LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ T-BET+ 4\\n PD-1+ TOX1+ TIM3+ EOMES+ CD44+ T-BET+ 4\\n PD-1+ TIM3+ CD39+ TCF-1+ 4\\n PD-1+ TOX1+ TIM3+ EOMES+ CD38+ CD44+ TCF-1+ 4\\n PD-1+ TIM3+ EOMES+ CD44+ T-BET+ 4\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ 3\\n TOX1+ CD39+ CD38+ TCF-1+ T-BET+ 3\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 3\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='31ead680-765c-45ef-bb95-7b13f7905250', embedding=None, metadata={'page_label': '93', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ TIM3+ LAG-3+ CD38+ CD44+ T-BET+ 3\\n TOX1+ TIM3+ LAG-3+ CD39+ TCF-1+ T-BET+ 3\\n PD-1+ TOX1+ EOMES+ CD38+ CD44+ T-BET+ 3\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ 3\\n TIM3+ CD38+ TCF-1+ 3\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ 3\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ 3\\n TIM3+ LAG-3+ CD39+ EOMES+ CD44+ T-BET+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ T-BET+ 3\\n TOX1+ TIM3+ LAG-3+ CD38+ CD44+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ 3\\n TOX1+ TIM3+ LAG-3+ T-BET+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 3\\n TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 3\\n TIM3+ LAG-3+ EOMES+ CD38+ T-BET+ 3\\n TIM3+ LAG-3+ CD39+ T-BET+ 3\\n TIM3+ LAG-3+ EOMES+ CD44+ 3\\n TIM3+ LAG-3+ CD39+ TCF-1+ 3\\n TIM3+ LAG-3+ CD39+ TCF-1+ T-BET+ 3\\n TOX1+ TIM3+ CD39+ CD44+ T-BET+ 3\\n TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 3\\n TOX1+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TOX1+ TIM3+ CD39+ CD44+ TCF-1+ 3\\n TIM3+ LAG-3+ CD39+ CD38+ CD44+ 3\\n TOX1+ LAG-3+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TOX1+ LAG-3+ CD38+ CD44+ TCF-1+ 3\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 3\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ 3\\n LAG-3+ CD38+ CD44+ T-BET+ 3\\n TOX1+ TIM3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 3\\n TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 3\\n TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ T-BET+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD44+ TCF-1+ 3\\n LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 3\\n TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 3\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ T-BET+ 3\\n TOX1+ TIM3+ LAG-3+ CD39+ CD44+ T-BET+ 3\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 3\\n TIM3+ LAG-3+ CD44+ TCF-1+ 3\\n PD-1+ TIM3+ EOMES+ CD38+ TCF-1+ 3\\n TIM3+ CD39+ CD44+ T-BET+ 3\\n PD-1+ LAG-3+ CD39+ EOMES+ CD44+ 3\\n TIM3+ LAG-3+ CD44+ TCF-1+ T-BET+ 3\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5bda45dd-be30-4f45-8219-37d2b75769c1', embedding=None, metadata={'page_label': '94', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ EOMES+ CD38+ 3\\n PD-1+ TIM3+ EOMES+ CD44+ 3\\n TIM3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ LAG-3+ EOMES+ CD38+ 3\\n PD-1+ LAG-3+ EOMES+ CD38+ TCF-1+ 3\\n PD-1+ TIM3+ EOMES+ T-BET+ 3\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 3\\n PD-1+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ 3\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ 3\\n PD-1+ LAG-3+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ CD38+ T-BET+ 3\\n TIM3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TIM3+ LAG-3+ EOMES+ TCF-1+ T-BET+ 3\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 3\\n PD-1+ TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 3\\n TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 3\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ T-BET+ 3\\n TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TIM3+ LAG-3+ CD38+ TCF-1+ 3\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 3\\n PD-1+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TIM3+ CD39+ EOMES+ CD44+ TCF-1+ 3\\n PD-1+ CD39+ CD38+ 3\\n PD-1+ CD39+ EOMES+ T-BET+ 3\\n PD-1+ CD39+ EOMES+ CD38+ TCF-1+ 3\\n PD-1+ CD39+ CD38+ TCF-1+ T-BET+ 3\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ T-BET+ 3\\n PD-1+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 3\\n PD-1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TIM3+ CD39+ 3\\n PD-1+ TIM3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 3\\n PD-1+ TIM3+ EOMES+ CD38+ CD44+ TCF-1+ 3\\n PD-1+ CD39+ CD44+ T-BET+ 3\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ 2\\n PD-1+ TOX1+ TIM3+ CD39+ CD38+ CD44+ TCF-1+ 2\\n TOX1+ LAG-3+ EOMES+ CD44+ T-BET+ 2\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ EOMES+ CD38+ CD44+ 2\\n PD-1+ TIM3+ CD39+ CD38+ T-BET+ 2\\n TIM3+ LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ EOMES+ CD38+ CD44+ 2\\n PD-1+ TIM3+ EOMES+ CD38+ CD44+ T-BET+ 2\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='caa0fa68-05b4-49d1-b382-90569535362b', embedding=None, metadata={'page_label': '95', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TIM3+ CD39+ CD44+ T-BET+ 2\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 2\\n TIM3+ LAG-3+ CD44+ T-BET+ 2\\n TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ 2\\n TOX1+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ CD38+ CD44+ TCF-1+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD44+ 2\\n PD-1+ LAG-3+ CD39+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ TCF-1+ 2\\n PD-1+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 2\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ T-BET+ 2\\n PD-1+ CD39+ EOMES+ CD38+ T-BET+ 2\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ 2\\n PD-1+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ 2\\n TIM3+ CD39+ EOMES+ CD44+ 2\\n TIM3+ CD39+ EOMES+ TCF-1+ 2\\n PD-1+ LAG-3+ CD38+ CD44+ TCF-1+ 2\\n TIM3+ LAG-3+ CD39+ EOMES+ CD44+ 2\\n TIM3+ CD39+ EOMES+ T-BET+ 2\\n PD-1+ EOMES+ CD38+ TCF-1+ T-BET+ 2\\n TOX1+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ 2\\n TIM3+ LAG-3+ CD39+ EOMES+ T-BET+ 2\\n TIM3+ CD39+ EOMES+ CD38+ T-BET+ 2\\n PD-1+ LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ CD39+ EOMES+ TCF-1+ T-BET+ 2\\n TIM3+ CD39+ EOMES+ CD38+ CD44+ 2\\n PD-1+ TOX1+ TIM3+ LAG-3+ TCF-1+ 2\\n PD-1+ LAG-3+ CD39+ CD38+ 2\\n PD-1+ LAG-3+ CD39+ CD38+ T-BET+ 2\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ LAG-3+ CD39+ EOMES+ CD44+ T-BET+ 2\\n PD-1+ CD39+ CD38+ CD44+ 2\\n TIM3+ LAG-3+ EOMES+ CD38+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ 2\\n PD-1+ CD39+ CD38+ CD44+ T-BET+ 2\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='efa7496c-346c-4f1b-ae3b-3c40d7030d42', embedding=None, metadata={'page_label': '96', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ TCF-1+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ 2\\n PD-1+ TOX1+ CD39+ CD38+ TCF-1+ 2\\n PD-1+ TOX1+ CD39+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD44+ TCF-1+ 2\\n TOX1+ TIM3+ CD39+ EOMES+ T-BET+ 2\\n PD-1+ TOX1+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ CD39+ T-BET+ 2\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ 2\\n TOX1+ TIM3+ LAG-3+ EOMES+ T-BET+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 2\\n TIM3+ CD39+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ CD39+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ LAG-3+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ LAG-3+ CD38+ TCF-1+ 2\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 2\\n TOX1+ TIM3+ LAG-3+ CD44+ TCF-1+ 2\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ EOMES+ 2\\n PD-1+ TIM3+ LAG-3+ EOMES+ T-BET+ 2\\n LAG-3+ CD39+ CD38+ CD44+ T-BET+ 2\\n TOX1+ TIM3+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ CD39+ EOMES+ CD44+ 2\\n TOX1+ TIM3+ EOMES+ CD38+ 2\\n TOX1+ TIM3+ EOMES+ CD38+ T-BET+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ 2\\n PD-1+ TIM3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 2\\n TIM3+ LAG-3+ CD38+ CD44+ 2\\n PD-1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 2\\n PD-1+ TOX1+ LAG-3+ CD39+ CD38+ CD44+ 2\\n TOX1+ TIM3+ LAG-3+ CD44+ 2\\n PD-1+ TIM3+ LAG-3+ TCF-1+ T-BET+ 2\\n PD-1+ TOX1+ LAG-3+ CD39+ CD44+ T-BET+ 2\\n PD-1+ TIM3+ LAG-3+ CD38+ T-BET+ 2\\n TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ CD44+ T-BET+ 2\\n TOX1+ TIM3+ LAG-3+ CD39+ CD44+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ 1\\n PD-1+ CD39+ EOMES+ CD38+ CD44+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ 1\\n PD-1+ CD39+ EOMES+ CD38+ 1\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='29e96d40-1043-4c64-80ef-1bd721923ab9', embedding=None, metadata={'page_label': '97', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' TOX1+ TIM3+ LAG-3+ EOMES+ CD44+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ 1\\n PD-1+ LAG-3+ EOMES+ CD38+ T-BET+ 1\\n TIM3+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 1\\n TOX1+ TIM3+ LAG-3+ CD38+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ TCF-1+ T-BET+ 1\\n PD-1+ CD39+ CD38+ CD44+ TCF-1+ 1\\n TOX1+ TIM3+ LAG-3+ CD44+ T-BET+ 1\\n TOX1+ TIM3+ LAG-3+ CD38+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ LAG-3+ EOMES+ CD38+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD38+ TCF-1+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ T-BET+ 1\\n PD-1+ TOX1+ CD39+ CD38+ TCF-1+ T-BET+ 1\\n PD-1+ EOMES+ CD38+ CD44+ T-BET+ 1\\n PD-1+ TOX1+ CD39+ CD38+ CD44+ TCF-1+ 1\\n PD-1+ LAG-3+ CD38+ CD44+ 1\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ 1\\n PD-1+ LAG-3+ CD38+ CD44+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ T-BET+ 1\\n PD-1+ CD39+ EOMES+ CD44+ T-BET+ 1\\n PD-1+ LAG-3+ EOMES+ CD44+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD44+ T-BET+ 1\\n PD-1+ EOMES+ CD38+ CD44+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ T-BET+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ EOMES+ CD38+ 1\\n PD-1+ EOMES+ CD38+ T-BET+ 1\\n TOX1+ CD39+ CD38+ CD44+ T-BET+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ EOMES+ 1\\n TIM3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 1\\n TOX1+ TIM3+ CD39+ CD38+ TCF-1+ 1\\n PD-1+ TOX1+ TIM3+ EOMES+ CD38+ TCF-1+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ EOMES+ CD38+ TCF-1+ 1\\n PD-1+ TIM3+ CD38+ 1\\n PD-1+ TIM3+ CD38+ T-BET+ 1\\n PD-1+ TIM3+ CD38+ TCF-1+ 1\\n PD-1+ TIM3+ CD38+ TCF-1+ T-BET+ 1\\n PD-1+ TIM3+ CD38+ CD44+ 1\\n PD-1+ TIM3+ LAG-3+ CD38+ CD44+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ CD38+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ LAG-3+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ LAG-3+ CD44+ 1\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='16884d57-ea75-411b-a216-5006205afd64', embedding=None, metadata={'page_label': '98', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' PD-1+ TIM3+ LAG-3+ TCF-1+ 1\\n PD-1+ TIM3+ CD39+ T-BET+ 1\\n TIM3+ LAG-3+ CD39+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ CD39+ TCF-1+ T-BET+ 1\\n TIM3+ LAG-3+ CD39+ CD44+ 1\\n TOX1+ LAG-3+ CD39+ EOMES+ CD38+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ CD39+ CD44+ 1\\n LAG-3+ CD39+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ CD39+ CD38+ 1\\n PD-1+ TIM3+ CD39+ CD38+ CD44+ T-BET+ 1\\n TIM3+ LAG-3+ EOMES+ CD38+ CD44+ 1\\n PD-1+ TOX1+ TIM3+ EOMES+ CD38+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ LAG-3+ EOMES+ TCF-1+ 1\\n PD-1+ CD38+ CD44+ T-BET+ 1\\n TOX1+ TIM3+ CD39+ EOMES+ CD38+ 1\\n TOX1+ TIM3+ CD39+ EOMES+ CD38+ CD44+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ T-BET+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ T-BET+ 1\\n PD-1+ LAG-3+ CD39+ CD44+ 1\\n PD-1+ LAG-3+ CD39+ CD44+ T-BET+ 1\\n PD-1+ TIM3+ CD39+ EOMES+ TCF-1+ 1\\n PD-1+ LAG-3+ CD39+ CD38+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ CD39+ TCF-1+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ CD39+ EOMES+ CD38+ 1\\n PD-1+ LAG-3+ CD39+ CD38+ CD44+ 1\\n PD-1+ LAG-3+ CD39+ CD38+ CD44+ TCF-1+ 1\\n PD-1+ TIM3+ LAG-3+ EOMES+ CD44+ 1\\n LAG-3+ CD39+ EOMES+ CD44+ T-BET+ 1\\n TIM3+ CD39+ CD38+ CD44+ TCF-1+ 1\\n TOX1+ TIM3+ CD39+ EOMES+ CD44+ T-BET+ 1\\n TIM3+ LAG-3+ CD39+ CD38+ TCF-1+ 1\\n PD-1+ LAG-3+ CD39+ EOMES+ CD44+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ CD39+ CD38+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ CD39+ CD38+ 1\\n PD-1+ TOX1+ TIM3+ CD39+ CD44+ T-BET+ 1\\n PD-1+ TOX1+ TIM3+ CD39+ CD44+ 1\\n PD-1+ LAG-3+ CD39+ EOMES+ CD38+ TCF-1+ T-BET+ 1\\n TOX1+ TIM3+ LAG-3+ CD39+ CD44+ TCF-1+ T-BET+ 1\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e2490f78-e68a-4449-9c7a-ef4cb0001aae', embedding=None, metadata={'page_label': '99', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='\\x12ĞůůͲ\\x12Ğůů\\x03^ƉĂƚŝĂů\\x03/ŶƚĞƌĂĐƚŝŽŶ \\x12ŽƵŶƚ\\nKi-67- Neoplastic & Ki-67- Neoplastic 2876770\\nMyeloid cells & Myeloid cells 1480053\\nMyeloid cells & Ki-67- Neoplastic 465886\\nB cells & B cells 450530\\nKi-67- Neoplastic & Ki-67+ Neoplastic 437638\\nCD44- CD4 Th & CD44- CD4 Th 226250\\nMesenchymal & Mesenchymal 220227\\nCD44- CD4 Th & B cells 219910\\nCD44- CD8 T Other & CD44- CD4 Th 206383\\nMesenchymal & Ki-67- Neoplastic 202107\\nKi-67+ Neoplastic & Ki-67+ Neoplastic 165702\\nCD44- CD8 T Other & CD44- CD8 T Other 158210\\nCD44- CD8 T Other & B cells 111164\\nCD44- CD4 Th & Myeloid cells 107677\\nMyeloid cells & Mesenchymal 90234\\nCD44- CD8 T Other & Myeloid cells 89976\\nB cells & Myeloid cells 62552\\nMyeloid cells & Ki-67+ Neoplastic 58793\\nCD44- CD8 T Other & Ki-67- Neoplastic 49618\\nB cells & Ki-67- Neoplastic 49336\\nCD44- CD4 Th & Ki-67- Neoplastic 48406\\nCD44- CD4 Th1 Other & CD44- CD4 Th 42570\\nCD44- CD8 T Other & CD44- CD4 Th1 Other 34844\\nCD44- CD4 Th & mTREG 32692\\nCD44- CD4 Th & CD44+ CD4 Th 31233\\nCD44- CD4 Th & Mesenchymal 29024\\nCD44- CD8 T Other & mTREG 27872\\nCD44+ CD4 Th & B cells 27606\\nCD44- CD8 T Other & Mesenchymal 26494\\nCD44+ CD4 Th & CD44+ CD4 Th 24718\\nCD44+ CD4 Th & Myeloid cells 24454\\nCD44- CD4 Th1 Other & B cells 24437\\nCD44+ CD8 T Other & Myeloid cells 23784\\nmTREG & Myeloid cells 23570\\nCD44- CD8 T Other & CD44+ CD8 T Other 23249\\nNaive TREG & B cells 23177\\nCD44- CD4 Th & Naive TREG 22405\\nCD44+ CD8 T Other & CD44+ CD8 T Other 22351\\nMesenchymal & Ki-67+ Neoplastic 20910\\nCD44+ CD8 T Other & CD44+ CD4 Th 20823\\nCD44+ CD8 T Other & CD44- CD4 Th 20030\\nCD44+ CD8 T Other & B cells 19305\\nCD44- CD8 T Other & Naive TREG 16461\\nSupplementary Table S5. Raw counts of cell-cell spatial interactions present in the dataset.Supplementary Table S5. Raw counts of cell-cell spatial interactions present in the dataset.\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='6f3222ba-bdc2-4f3d-b98f-436e0cd03e73', embedding=None, metadata={'page_label': '100', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='B cells & Mesenchymal 15010\\nmTREG & B cells 14656\\nmTREG & Ki-67- Neoplastic 14362\\nCD44- CD8 T Other & CD44+ CD4 Th 12734\\nCD44- CD4 Th1 Other & CD44- CD4 Th1 Other 11730\\nmTREG & Mesenchymal 10419\\nCD44- CD4 Th1 Other & Myeloid cells 9616\\nmTREG & mTREG 9087\\nCD8 T NAIVE & CD44- CD8 T Other 8303\\nNaive TREG & Naive TREG 7839\\nCD8 T NAIVE & CD44- CD4 Th 7103\\nB cells & Ki-67+ Neoplastic 6512\\nCD44- CD4 Th & Ki-67+ Neoplastic 6452\\nCD44- CD8 T Other & Ki-67+ Neoplastic 6392\\nCD44- CD4 Th & T-BET+ TREG 6102\\nNaive TREG & Myeloid cells 6035\\nCD44+ CD4 Th & mTREG 5918\\nCD44- CD4 Th1 Other & Mesenchymal 5711\\nCD44- CD4 Th1 Other & Ki-67- Neoplastic 5640\\nCD8 TEMRA & CD44+ CD4 Th 5367\\nCD44- CD8 T Other & T-BET+ TREG 5300\\nCD44+ CD8 T Other & mTREG 5186\\nNaive TREG & Ki-67- Neoplastic 5185\\nCD8 T NAIVE & B cells 5149\\nT-BET+ TREG & B cells 5086\\nNaive TREG & mTREG 4921\\nCD44+ CD8 T Other & Ki-67- Neoplastic 4863\\nCD8 TEMRA & CD44+ CD8 T Other 4772\\nCD44+ CD4 Th & Mesenchymal 4593\\nCD44- CD4 Th1 Other & mTREG 4305\\nCD44+ CD4 Th & Naive TREG 4166\\nCD8 TEMRA & B cells 3878\\nCD44+ CD8 T Other & Mesenchymal 3536\\nCD44+ CD4 Th & Ki-67- Neoplastic 3476\\nCD8 TEMRA & CD44- CD8 T Other 3454\\nCD8 TEMRA & CD44- CD4 Th 3399\\nCD44- CD4 Th1 Other & T-BET+ TREG 3382\\nCD8 TEMRA & CD8 TEMRA 3337\\nCD8 TEMRA & Myeloid cells 3226\\nCD44- CD4 Th1 Other & CD44+ CD4 Th 3112\\nCD44- CD4 Th1 Other & Naive TREG 3052\\nCD44+ CD8 T Other & CD44- CD4 Th1 Other 2766\\nCD44+ CD8 T Other & Naive TREG 2300\\nCD4 Th1EMRA & CD44- CD4 Th 1997\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='bdbd91f6-597c-4039-8b98-38ef4faaf6cf', embedding=None, metadata={'page_label': '101', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CD4 Th1EM & CD44- CD4 Th 1949\\nmTREG & Ki-67+ Neoplastic 1924\\nT-BET+ TREG & Myeloid cells 1857\\nNaive TREG & Mesenchymal 1852\\nCD4 Th1EMRA & B cells 1839\\nNaive TREG & T-BET+ TREG 1681\\nCD4 Th1EM & Myeloid cells 1667\\nCD4 Th1EMRA & CD44+ CD4 Th 1607\\nCD4 Th1EM & CD44+ CD4 Th 1542\\nCD44- CD8 T Other & CD4 Th1EM 1486\\nmTREG & T-BET+ TREG 1458\\nCD8 TEM & CD44- CD8 T Other 1322\\nCD44+ CD8 T Other & CD4 Th1EM 1311\\nCD8 TEM & CD44+ CD8 T Other 1285\\nCD8 TEM & CD44+ CD4 Th 1283\\nCD8 T NAIVE & CD44- CD4 Th1 Other 1238\\nCD44- CD8 T Other & CD4 Th1EMRA 1162\\nCD8 TEM & CD44- CD4 Th 1140\\nT-BET+ TREG & T-BET+ TREG 1119\\nCD4 Th1EM & CD44- CD4 Th1 Other 1112\\nCD44+ CD4 Th & T-BET+ TREG 1092\\nCD8 TEX & CD44+ CD8 T Other 1071\\nCD44+ CD8 T Other & T-BET+ TREG 1020\\nT-BET+ TREG & Ki-67- Neoplastic 1013\\nT-BET+ TREG & Mesenchymal 969\\nCD44+ CD8 T Other & CD4 Th1EMRA 969\\nCD4 Th1EMRA & CD44- CD4 Th1 Other 934\\nCD4 Th1EMRA & Myeloid cells 932\\nCD8 TEMRA & Ki-67- Neoplastic 878\\nCD8 TEM & Myeloid cells 847\\nCD8 T NAIVE & CD44+ CD8 T Other 837\\nCD8 TEM & B cells 808\\nCD4 Th1EM & Mesenchymal 790\\nCD8 TEMRA & Naive TREG 769\\nCD8 TEX & CD44- CD8 T Other 754\\nCD8 T NAIVE & Myeloid cells 740\\nCD4 Th1EMRA & CD4 Th1EMRA 739\\nCD44- CD4 Th1 Other & Ki-67+ Neoplastic 727\\nCD44+ CD8 T Other & Ki-67+ Neoplastic 710\\nCD4 Th1EM & mTREG 685\\nCD8 T NAIVE & CD8 T NAIVE 678\\nCD8 TEM & CD8 TEM 628\\nCD8 T NAIVE & Naive TREG 618\\nCD4 Th1EM & B cells 609\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f913b125-faf0-4693-825a-8e351b351018', embedding=None, metadata={'page_label': '102', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CD44+ CD4 Th1 Other & Ki-67- Neoplastic 580\\nCD8 TEMRA & CD4 Th1EMRA 576\\nCD8 T NAIVE & mTREG 574\\nCD44+ CD4 Th1 Other & Myeloid cells 550\\nCD8 TEX & CD44+ CD4 Th 522\\nCD8 TEX & CD44- CD4 Th 514\\nCD4 Th1EM & Ki-67- Neoplastic 510\\nCD4 Th1EMRA & Naive TREG 507\\nCD44+ CD4 Th & Ki-67+ Neoplastic 495\\nCD8 TEM & CD8 TEMRA 478\\nCD44- CD4 Th1 Other & CD44+ CD4 Th1 Other 477\\nCD4 Th1EMRA & T-BET+ TREG 476\\nCD8 TEMRA & CD44- CD4 Th1 Other 467\\nCD8 TEMRA & mTREG 459\\nCD8 T NAIVE & CD44+ CD4 Th 457\\nNaive TREG & Ki-67+ Neoplastic 451\\nCD8 TEX & B cells 442\\nCD4 Th1EM & CD4 Th1EM 431\\nCD8 TEMRA & Mesenchymal 383\\nCD8 TEX & CD8 TEX 368\\nCD44+ CD4 Th1 Other & B cells 353\\nCD8 TEMRA & T-BET+ TREG 350\\nCD8 TEM & CD4 Th1EM 340\\nCD44+ CD4 Th1 Other & CD44+ CD4 Th1 Other 338\\nCD8 TEX & mTREG 316\\nCD8 TEX & Myeloid cells 316\\nCD8 TEM & mTREG 309\\nCD44+ CD4 Th1 Other & CD44- CD4 Th 303\\nCD4 Th1EMRA & CD44+ CD4 Th1 Other 290\\nCD8 TTEX & CD44+ CD8 T Other 285\\nCD4 Th1EM & T-BET+ TREG 281\\nCD44+ CD4 Th1 Other & Mesenchymal 278\\nCD8 TEMRA & CD8 TEX 250\\nCD4 Th1EMRA & Ki-67- Neoplastic 247\\nCD4 Th1EMRA & mTREG 239\\nCD8 TTEX & Myeloid cells 221\\nCD44- CD8 T Other & CD44+ CD4 Th1 Other 215\\nCD8 T NAIVE & T-BET+ TREG 213\\nCD44+ CD4 Th1 Other & CD44+ CD4 Th 206\\nCD8 TEM & CD44- CD4 Th1 Other 199\\nCD8 TEX & Ki-67- Neoplastic 199\\nCD8 TTEX & CD44- CD8 T Other 194\\nCD8 T NAIVE & Mesenchymal 193\\nCD44+ CD8 T Other & CD44+ CD4 Th1 Other 186\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2226b0f5-5f3f-4460-a889-777c4364c4a1', embedding=None, metadata={'page_label': '103', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CD8 TTEX & CD44+ CD4 Th 176\\nCD8 TEMRA & CD4 Th1EM 174\\nCD4 Th1EMRA & Mesenchymal 172\\nCD8 TTEX & CD44- CD4 Th 168\\nCD8 TEM & T-BET+ TREG 159\\nT-BET+ TREG & Ki-67+ Neoplastic 150\\nCD8 T NAIVE & CD8 TEMRA 150\\nCD4 Th1EM & CD4 Th1EMRA 146\\nCD4 Th1EM & Naive TREG 137\\nCD8 TEM & CD4 Th1EMRA 131\\nCD8 T NAIVE & Ki-67- Neoplastic 116\\nCD8 TTEX & B cells 111\\nCD8 TEFF & CD44+ CD8 T Other 108\\nCD8 TEX & Naive TREG 101\\nCD8 TEX & CD44- CD4 Th1 Other 96\\nCD8 TEM & CD8 TEX 91\\nCD8 TEFF & B cells 88\\nCD8 TEM & Mesenchymal 85\\nCD8 TEX & T-BET+ TREG 83\\nCD8 TEX & CD8 TTEX 83\\nCD8 TEM & Naive TREG 82\\nCD8 TEMRA & Ki-67+ Neoplastic 81\\nCD8 TTEX & mTREG 80\\nCD8 TTEX & Ki-67- Neoplastic 77\\nCD8 TEX & Mesenchymal 75\\nCD8 TEX & CD4 Th1EM 72\\nCD8 TEM & Ki-67- Neoplastic 68\\nCD44+ CD4 Th1 Other & Ki-67+ Neoplastic 65\\nCD8 TEFF & CD44- CD8 T Other 64\\nCD44+ CD4 Th1 Other & T-BET+ TREG 61\\nCD8 TEFF & CD8 TEMRA 56\\nCD4 Th1EFF & B cells 56\\nCD44+ CD4 Th1 Other & mTREG 54\\nCD8 TEFF & Myeloid cells 50\\nCD8 TEFF & CD44- CD4 Th 49\\nCD4 Th1EFF & Myeloid cells 47\\nCD4 Th1EM & Ki-67+ Neoplastic 45\\nCD8 TEFF & CD8 TEFF 42\\nCD4 Th1EM & CD44+ CD4 Th1 Other 41\\nCD8 T NAIVE & CD4 Th1EMRA 35\\nCD8 TEMRA & CD8 TTEX 34\\nCD44+ CD4 Th1 Other & Naive TREG 34\\nCD8 TEMRA & CD44+ CD4 Th1 Other 31\\nCD8 TTEX & Mesenchymal 30\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='41e1300f-f6d1-4cef-bc17-014b987a5fe0', embedding=None, metadata={'page_label': '104', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CD8 TEX & CD4 Th1EMRA 29\\nCD8 T NAIVE & CD4 Th1EM 28\\nCD4 Th1EFF & CD44+ CD4 Th 28\\nCD8 TEFF & CD44- CD4 Th1 Other 26\\nCD8 TEFF & Mesenchymal 25\\nCD8 TTEX & Naive TREG 25\\nCD8 TEFF & Ki-67- Neoplastic 24\\nCD4 Th1EMRA & Ki-67+ Neoplastic 23\\nCD8 T NAIVE & CD8 TEM 22\\nCD8 TEX & Ki-67+ Neoplastic 22\\nCD4 Th1EFF & CD44+ CD4 Th1 Other 21\\nCD8 TTEX & CD44- CD4 Th1 Other 21\\nCD8 T NAIVE & CD8 TEX 20\\nCD8 T NAIVE & CD44+ CD4 Th1 Other 20\\nCD4 Th1EFF & Ki-67- Neoplastic 19\\nCD4 Th1EFF & CD44- CD4 Th1 Other 18\\nCD8 TTEX & CD4 Th1EM 18\\nCD8 TTEX & CD8 TTEX 17\\nCD8 TTEX & Ki-67+ Neoplastic 15\\nCD4 Th1EFF & CD4 Th1EMRA 15\\nCD8 TEFF & CD44+ CD4 Th 13\\nCD8 TEM & CD8 TTEX 11\\nCD8 TEMRA & CD4 Th1EFF 11\\nCD8 TEX & CD44+ CD4 Th1 Other 10\\nCD44- CD8 T Other & CD4 Th1EFF 8\\nCD8 T NAIVE & Ki-67+ Neoplastic 8\\nCD4 Th1EFF & Ki-67+ Neoplastic 8\\nCD44+ CD8 T Other & CD4 Th1EFF 7\\nCD4 Th1EFF & CD44- CD4 Th 7\\nCD8 TTEX & CD4 Th1EMRA 7\\nCD8 TTEX & T-BET+ TREG 7\\nCD8 TEM & Ki-67+ Neoplastic 5\\nCD4 Th1EFF & Naive TREG 5\\nCD8 TEFF & CD8 TTEX 5\\nCD8 TEFF & T-BET+ TREG 4\\nCD8 TEFF & CD44+ CD4 Th1 Other 4\\nCD8 TEM & CD44+ CD4 Th1 Other 4\\nCD4 Th1EFF & CD4 Th1EM 4\\nCD8 TEX & CD4 Th1EFF 4\\nCD4 Th1EFF & Mesenchymal 4\\nCD8 TEFF & Naive TREG 4\\nCD8 T NAIVE & CD8 TEFF 3\\nCD4 Th1EFF & CD4 Th1EFF 3\\nCD4 Th1EFF & T-BET+ TREG 3\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a864972e-3fc0-41e9-99cc-d79ae406cb82', embedding=None, metadata={'page_label': '105', 'file_name': 'blise_pancreatic.pdf', 'file_path': '/content/data/blise_pancreatic.pdf', 'file_type': 'application/pdf', 'file_size': 13948693, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CD4 Th1EFF & mTREG 2\\nCD8 T NAIVE & CD8 TTEX 2\\nCD8 TEM & CD4 Th1EFF 2\\nCD8 TEFF & mTREG 1\\nCD8 TEFF & CD8 TEM 1\\n.CC-BY 4.0 International licenseavailable under a\\n(which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprintthis version posted October 23, 2023. ; https://doi.org/10.1101/2023.10.20.563335doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a3974b8d-a466-4827-a7f8-9b761709c67e', embedding=None, metadata={'page_label': '1', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='TITLE: Robust biomarker discovery through multiplatform multiplex image analysis of \\nbreast cancer clinical cohorts \\nJennifer Eng1,2, Elmar Bucher2, Zhi Hu2, Melinda Sanders3,4, Bapsi Chakravarthy3, Paula \\nGonzalez3, Jennifer A. Pietenpol3,5, Summer L. Gibbs2, Rosalie C. Sears1, Koei Chin2,6. \\n \\nAFFILIATIONS \\n1. Department of Molecular and Medical Genetics, Oregon Health and Science University, \\nPortland, OR, 97239, USA. \\n2. Department of Biomedical Engineering, Oregon Health and Science University, Portland, OR, \\n97239, USA. \\n3. Vanderbilt-Ingram Cancer Center, Vanderbilt University Medical Center, Nashville, TN, \\n37232, USA, USA.   \\n4. Department of Pathology, Microbiology and Immunology, Vanderbilt University Medical \\nCenter, Nashville, TN, 37232, USA. \\n5. Department of Biochemistry, Vanderbilt University, Nashville, TN, 37232, USA. \\n6. Center for Early Detection Advanced Research, Oregon Health and Science University, \\nPortland, OR, 97239, USA. \\n \\nRunning Title: Robust biomarker discovery in breast cancer multiplex images.  \\n \\nKEYWORDS: Breast tumor microenvironment, spatial biomarker discovery, multiplex image \\nanalysis tools  \\n \\nDisclosures:  \\nThe authors have no relevant financial relationships to disclose.  \\n \\n \\nCorresponding author(s): \\n1. Jennifer Eng, Department of Molecular and Medical Genetics, 2730 SW Moody Ave \\nCL4RS, Portland, OR, 97239, USA. Tel: (503) 494 – 3021. Fax: (503) 418 – 9311. \\nengje@ohsu.edu. \\n2. Koei Chin, Center for Early Detection Advanced Research, Oregon Health and Science \\nUniversity, Portland, OR, 97239, USA. chinkoei@ohsu.edu. \\n \\n  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='080df0a3-4769-4a7e-98ec-cf02b3608643', embedding=None, metadata={'page_label': '2', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ABSTRACT \\nSpatial profiling of tissues promises to elucidate tumor-microenvironment interactions and enable \\ndevelopment of spatial biomarkers to predict patient response to immunotherapy and other \\ntherapeutics.  However, spatial biomarker discovery is often carried out on a single patient cohort \\nor imaging technology, limiting statistical power and increasing the likelihood of technical \\nartifacts.  In order to analyze multiple patient cohorts profiled on different platforms, we developed \\nmethods for comparative data analysis from three disparate multiplex imaging technologies: 1) \\ncyclic immunofluorescence data we generated from 102 breast cancer patients with clinical follow-\\nup, in addition to publicly available 2) imaging mass cytometry and 3) multiplex ion-beam imaging \\ndata. We demonstrate similar single-cell phenotyping results across breast cancer patient cohorts \\nimaged with these three technologies and identify cellular abundance and proximity -based \\nbiomarkers with  prognostic value across platforms. In multiple platforms, w e identified \\nlymphocyte infiltration as independently associated with longer survival in triple negative and \\nhigh-proliferation breast tumors. Then, a comparison of nine spatial analysis methods revealed \\nrobust spatial biomarkers . In estrogen receptor-positive disease, quiescent stromal cells close to \\ntumor were more abundant in good prognosis tumors while tumor neighborhoods of mixed \\nfibroblast phenotypes were enriched in poor prognosis tumors. In triple-negative breast cancer \\n(TNBC), macrophage proximity to tumor and B cell proximity to T cells were greater in good \\nprognosis tumors, while tumor neighborhoods of vimentin -positive fibroblasts were enriched in \\npoor prognosis tumors . We also tested previously published spatial biomarkers in our ensemble \\ncohort, reproducing the positive prognostic value of isolate d lymphocytes and lymphocyte \\noccupancy and failing to reproduce the prognostic value of tumor-immune mixing score in TNBC. \\nIn conclusion, we demonstrate assembly of larger clinical cohorts from diverse platforms to aid in \\nprognostic spatial biomarker identification and validation.    \\n  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4085955b-5910-45c0-8f19-6173aff6c601', embedding=None, metadata={'page_label': '3', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='STATEMENT OF SIGNIFICANCE \\nOur single-cell spatial analysis of multiple clinical cohorts uncovered novel biomarkers of patient \\noutcome in breast cancer. Additionally, our data, software, and methods will help advance spatial \\ncharacterization of the tumor microenvironment. \\n \\n \\n  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ceae1376-51d4-4bab-ada0-7763cb262726', embedding=None, metadata={'page_label': '4', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='INTRODUCTION \\nRecent advances in the treatment landscape of breast cancer  have motivated the characterization \\nof the breast tumor microenvironment for deeper understanding of immune -tumor interactions. \\nIdentification of biomarkers predicting breast cancer i mmunotherapy response is still an urgent \\nclinical need1. For instance, in metastatic TNBC, only a quarter of PD-L1 positive patients respond \\nto single-agent immune checkpoint blockade1. In contrast, in early stage TNBC, response rates to \\nneoadjuvant immune checkpoint plus chemotherapy were similar in PD-L1 positive and negative \\ngroups2. This highlights the need for biomarker development for better patient stratification across \\ndisease stages. Furthermore, trials evaluating novel immune t argeted therapies in breast cancer \\nshould be accompanied by biomarker development for maximum therapeutic efficacy3.  \\n \\nHighly multiplex imaging methods enable quantification of dozens of biomarkers in a single tissue \\nsection with at sub-cellular resolution while retaining spatial context 4–9. Tissue structures such as \\ntertiary lymphoid structures, identified with multiplex imaging, are predictive biomarkers of \\nimmunotherapy response in melanoma10,11. Spatial proximity between tumor, immune and stromal \\ncell types is associated with response to neoadjuvant therapy in HER2+ breast cancer12. In several \\nbreast cancer multiplex imaging studies, s ingle-cell spatial context has prognostic relevance and \\nshows correlations with transcriptomic and genomic features of tumors 13–16. However, spatial \\nbiomarkers can be difficult to reproduce due to limited numbers of patients used to develop them \\nand difficulties in comparing data from different imaging platforms. Furthermore, overfitting is an \\nissue in biomedical imaging data  due to the number of steps in the processing pipeline and the \\nnumber of variables and parameters involved. Overfitting can be addressed through use of a \\ndiscovery cohort to tune analytical methods, which are then fixed and subsequently applied to a \\nvalidation cohort17. In theory, validation cohorts can be readily obtained through incorporation of \\npublicly available data from disparate imaging platforms into biomarker studies.  In practice, \\nintegrated analysis of such data remains a challenge. Furthermore,  documentation of metadata, \\nanalysis protocols and code is essential for reuse of data and reproducibility of findings, preferably \\nusing open-source software tools17. We developed an open-source python software, mplexable18, \\nfor multiplex image processing and analysis, which we use herein to process and analyze three \\nmultiplex imaging breast cancer cohorts: a cyclic immunofluorescence (CyCIF) dataset which we \\ngenerated, and publicly available imaging mass cytometry (IMC) and multiplex ion-beam imaging \\n(MIBI) datasets13,15,19. In this proof -of-concept study, we identify prognostic single -cell spatial \\nbiomarkers common across the imaging platforms. As such imaging datasets become more widely \\navailable, tools such as ours can facilitate biomarker discovery with high accuracy, reliability, and \\nefficiency.      \\nMETHODS \\nPatient samples \\nTwo breast cancer tissue microarrays were graciously provided by Dr. Jennifer Pietenpol (JP). All \\nsamples were collected at time of surgical resection (mastectomy or breast conserving surgery) at \\nVanderbilt University Medical Center with the same fixation protocol. JP-TMA1 had 131 cores of \\napproximately 1.2 µm diameter, with duplicate cores from 19 TNBC, 8 HER2+ and 36 ER+ \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='6d65b9a1-862c-4104-8b37-ffd5724053e0', embedding=None, metadata={'page_label': '5', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='patients. It also included one control core each of inflamed appendix, colon cancer, muscle, \\npancreas and normal breast. Four of the TMA1 TNB C patients received neoadjuvant \\nchemotherapy. JP-TMA2 contained a single, slightly larger (~1.4µm diameter) core from 39 triple-\\nnegative tumors and 1 ER+/HER2+ control core. Thirteen of the patients in TMA2 received \\nneoadjuvant therapy. Clinical outcome an d clinicopathological information are available for \\nTMA1 and TMA2. \\nImaging data generation and sources \\nCyCIF staining of tumor tissue was completed on JP -TMA1 and TMA2. The whole tissue core \\nwas imaged, as described previously18. MIBI imaging data was previously published by Keren et \\nal13 and the images were downloaded from https://mibi-share.ionpath.com/tracker/imageset under \\nthe name “Keren et al., Triple Negative Breast Cancer.” Survival and recurrence data were \\nobtained from a second publication by the same group 20, and were downloaded from \\nhttps://github.com/aalokpatwa/rasp-mibi. IMC imaging data were previously published by \\nJackson et al .15 and images and clinical data were downloaded from \\nhttps://doi.org/10.5281/zenodo.3518284.  \\nImage Processing \\nCyCIF tiff images were registered, segmented and single-cell intensity as well as nuclear size and \\nshape features were extracted as  previously described18. The CyCIF pipeline is available at \\nhttps://gitlab.com/engje/mplexable/-/tree/master/jupyter. Nuclear and cell segmentation were run \\nusing the Cellpose algorithm 21, which showed visually superior performance on CyCIF data \\ncompared to a watershed algorithm ( S1a). Nuclear and cell segmentation masks were matched \\nusing mplexable, enabling subtraction of nuclear mask from cell mask to ob tain segmentation of \\nthe cytoplasm (S1b). \\n \\nMIBI and IMC processing pipelines documenting the steps described here are available as Jupyter \\nnotebooks at https://github.com/engjen/cycIF_TMAs. MIBI and IMC images were downloaded as \\nmultipage OMEtiffs. Hot pix els14 were detected by identifying pixels that were 10 standard \\ndeviations above a median filtered image with a 2x2 pixel kernel size. Hot pixels were set to the \\nmedian filter values and resulting images were saved as tiff s for downstream feature extract ion. \\nDNA channels were processed for nuclear segmentation as follows. DNA images were rescaled \\nbetween the 3rd and one and a half times  the 99.999 quantile. The gamma value was adjusted by \\n0.6 in MIBI data and 0.4 in IMC data to enhance dimly stained nuclei. A two-channel nuclear plus \\ncytoplasm image was generated for cell segmentation. A mixture of markers was used to generate \\na composite image which stained the cytoplasm of the majority of cells. For MIBI data, the β-\\ncatenin, vimentin, CD45 and CD31 channels were combined into a maximum intensity projection \\ncytoplasm image and the gamma value was adjusted by 0.6. For IMC data, E -cadherin, vimentin, \\nCD44 and CD45 were combined into a maximum intensity projection cytoplasm image and \\ngamma adjusted by 0.4. Chamboelle total  variation denoising, implemented in scipy 22, was used \\nto smooth out pixelated nuclear and cytoplasmic projection image s (weight=0.1, except \\nweight=0.05 for IMC cytoplasm). All parameters were selected by testing segmentation results at \\nhttps://www.deepcell.org/predict and cellpose.org (S1c-d). Skipping either nuclear or cytoplasmic \\nChamboelle total-variation de-noising resulted in failure of deep learning-based algorithms on the \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2c75016b-8d00-45d2-a5ea-dedf84618e59', embedding=None, metadata={'page_label': '6', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='IMC data (S1e). Mesmer segmentation 23 performed better than Cellpose 24 in IMC data due to \\nimproved detection of dim nuclei, likely due to the incorporation of cytoplasmic staining in the \\nnuclear segmentation model ( S1d). This issue with Cellpose was not detected in CyCIF images, \\nwhich had brighter DNA staining. For IMC and MIBI data, nuclear and cellular segmentation were \\nperformed on processed nuclear and nuclear + cytoplasm images using Mesmer23. Matching of cell \\nIDs in the nuclear and cell masks was done with mplexable 18, with cell masks relabeled to match \\nthe ID of the nucleus which they had most overlap with. Cytoplasm masks were calculated by \\nsubtracting the nuclear mask from the matching cell mask. Nuclear and cytoplasmic  mean \\nintensity, nuclear size and shape feature s, and nuclear centroid coordinates were extracted with \\nmplexable18.  \\n \\nMesmer, a deep-learning-based approach for nuclear and whole -cell segmentation of tissue data, \\nwas run on IMC data and compared to the watershed-based segmentation originally published by \\nJackson et al15. The cell counts across the two methods had a Pearson correlation of 0.98 ( S2a). \\nVisual examination of ROIs with discordant cell numbers  revealed that Mesmer segmentation \\nperformed better in tissues with necrosis and high background noise in the DNA channel (S2 b-d). \\nImage QC \\nIn IMC data, artifacts include non -specific background staining, necrotic regions, and bright \\nantibody aggregates. IMC data were collected from small ROIs (~600 µm) within TMA cores and \\nsome samples annotated as estrogen-receptor (ER)+ tissues did not show any ER+ staining in the \\nROI. Therefore, quality control (QC) was performed on ER -stained images, a marker which has \\nbeen noted to exhibit non-specific background staining on the IMC platform25. QC images of ER \\nstaining were generated and sorted in a blinded fashion into negative and positive for nuclear -\\nspecific-staining (S3a). ROIs from clinically annotated ER+ patients that were classified a s ER \\npositive during QC or ROIs that came from ER negative patients and classified as ER negative \\nwere used for analysis. Samples that passed ER QC did not have significantly different grade, PR \\nstatus, TMA block, age of specimen, age of patient or tumor size (S3c-d). There were no significant \\nsurvival differences between QC passed versus failed  tumors from ER+, TNBC or ER+HER2+ \\npatients (S3e). In the IMC dataset, additional QC steps included: necrotic regions were manually \\ncircled using the napari 26 image viewer and excluded and bright aggregates in the CD3 channel \\nwere excluded by removing cells above a  threshold set at the value of CD3+ cells showing an \\nappropriate membranous staining pattern. \\n \\nIn the CyCIF data, imaging artifacts included autofluorescence (AF ), non-specific background, \\nfloating tissue and tissue loss. Background AF images were obtained half-way through CyCIF data \\ncollection and these images were scaled by exposure time and subtracted from the AF488, AF55 \\nand AF647 channels  using mplexable 18. Feature extraction was performed on AF subtracted \\nimages. Areas of floating tissue, air bubbles or necrotic regions were manually circled using the \\nnapari26 image viewer and excluded. Non-specific background staining was removed by setting \\nmanual thresholds for selected markers and subtracting those values from extracted data. The PD1 \\nantibody had bright aggregates that were excluded with an upper threshold. Tissue loss was \\ndetected by cells that lacked DAPI staining in the last round of imaging, and the se cells were \\nexcluded.  \\n \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9bd3611a-492c-4758-b4aa-ef924e798e7c', embedding=None, metadata={'page_label': '7', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='In all three platforms, additional artifacts caused by floating tissue or imaging problems (e.g., dark \\nor bright bands across IMC and MIBI images perhaps caused by problems with the rastering  \\nprocess) were detected through unsupervised clustering and visual inspection of clusters on the \\nimages. C lusters comprised of artifacts showed atypical very bright or dim staining in many \\nchannels, formed distinct artifact clusters and were removed.  \\nSingle cell phenotyping \\nCell types were defined in two ways: manual gating and unsupervised clustering. Unsupervised \\nclustering was conducted using the scanpy 27 software. Single -cell mean intensity values were \\nselected from either the nucleus or cytoplasm ma sks for each marker, depending on expected \\nintracellular distribution.  β-catenin, which can localize on the membrane, in the cytoplasm or in \\nthe nucleus, was selected from both nuclear and cytoplasmic masks for separate analysis. Since \\nthe CyCIF and IMC platforms had more marker and breast cancer subtype overlap than the MIBI \\npanel ( Figure 1, S4) 20 matching markers were selected for clustering in these datasets, plus \\nselected markers for immune, epithelial and fibroblast subsets ( collagen I [ ColI], CD4, CD8 in \\nCyCIF, and fibronectin [ FN], pan-cytokeratin [ panCK] in IMC). For MIBI data, all available \\nmarkers were used for clustering. Additionally, the nuclear area feature was used for clustering. \\nEach marker was divided by its standard deviation, without zero-centering, and clipped above 20 \\nstandard deviations. A Uniform Manifold Approximation and Projection (UMAP) embedding was \\ngenerated using 30 k -nearest neighbors and the embedding was clustered using the Leiden \\ncommunity detection algorithm 28. The Leiden resolution parameter was selected that resulted in \\n20 - 25 clusters. Each cluster was annotated and categorized as epithelial, endothelial, fibroblast, \\nimmune or stromal. Some clusters were comprised of multiple cell types, and these were manually \\nsplit, for example, the CD44+ cluster was split into CD44+ tumor and CD44+ stroma based on \\nmanual gating results (described below).   \\n \\nWe then performed manual gating to verify our annotated -cluster cell type. A threshold was set \\nfor each gating marke r based on the expected morphology of positive staining in images. \\nFibroblasts were defined as positive for one or more of vimentin, fibronectin (FN) or collagen I \\n(ColI). Epithelial cells were defined as positive for one or more of Ecad, cytokeratins, or β-catenin. \\nEndothelial cells were defined as CD31+. Immune cells were defined as CD45+. Stromal cells \\nwere defined as all non-fibroblast, non-endothelial, non-epithelial, non-immune segmented nuclei. \\nJupyter notebooks recording single cell phenotyping pipe lines are available here:  \\nhttps://github.com/engjen/cycIF_TMAs \\nSingle Cell Based Patient Subtyping  \\nEpithelial and stromal subtypes were determined by unsupervised clustering of patients based on \\nthe fraction of epithelial or stromal cell types within each compartment, respectively. Cell types \\nrepresenting greater than 2 -4% of the total cell population in the respective tissue compartment \\nwere used for clustering. T his cutoff was chosen to ignore rare, platform specific cell types that \\nmay represent method-specific artifacts. Unsupervised clustering of patients was performed using \\nthe Leiden algoritm implemented in scanpy27. For epithelial subtypes, the resolution of clustering \\nwas selected to minimize differences between the platforms (Figure 2). For stromal subtypes, we \\nselected the minimum number of clusters need to separate T cells from other clusters (k=6).  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='83ea7a9e-7c10-44d6-8d0c-98e357a97d6c', embedding=None, metadata={'page_label': '8', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Survival Analysis \\nFor each cell type, fraction of cells of that type over all cells from combined ROIs/cores from each \\npatient was calculated and binarized into high and low abundance based on tertile or median values \\nwithin each subtype. The CyCIF TMA1 dataset was used as a discovery dataset to determine the \\nquantile separating high and low abundance of each cell type  or spatial metric  that was most \\npredictive of survival.  Three quantiles were tested: 0.33 (e.g. split patients into 1/3 low and 2/3 \\nhigh), 0.5 and 0.66. The most prognostic cutoff value was selected for each cell type and for cell \\ntypes having prognostic value (alpha<0.05) these cutoffs were applied in the validation dataset \\nwhich included CyCIF TMA2, MIBI and IMC samples. Since overall cell type fractions diff ered \\nbetween platforms and subtypes (Figure 1), high and low values were determined relative to other \\nsamples from the same platform and subtype , using the predetermined cutoffs . In the validation \\ncohort, the log -rank test p-values were corrected for multi ple testing using the Benjamini–\\nHochberg False Discovery Rate (FDR) . For biomarkers with FDR < 0.1, m ultivariate Cox \\nproportional hazards (CPH) modelling was used to combine imaging biomarkers with patient age, \\ntumor size and clinical stage to test if they were independently prognostic. Collectively, 89 TNBC \\nand 160 ER+ patients had these additional clinical parameters (additional clinical parameters were \\nnot available in the MIBI dataset). \\nSpatial Analysis \\nSpatial distributions of cells were calculated in multiple ways. For analysis of cellular neighbors12 \\nand homotypic/heterotypic interactions 14 each cell’s neighbors within a 40 µm radius were \\ncounted. For tumor-immune mixing score a 25 µm radius was selected to replicate Keren et al.13 \\nFor lymphocyte clusters, a 20 µm radius was used and for lymphocyte occupancy a 50 µm grid \\nsquare was used, both to replicate Wortman et al.16 Ripley’s L (a density-normalized measure of \\nclustering) and the multitype K function (Kcross; a density-normalized measure of two cell types \\nco-localization) and G function (Gcross , a measure of two cell types co -localization) were \\ncalculated using spatsat29 with a radius of 50 µm. Spatial latent Dirichlet allocation (LDA) analysis \\nwas done using spatialLDA30, using the default radius of 100 µm and 6, 8 or 10 topics for topic \\nmodelling. Shorter distances of ~25 µm may be interpreted as cells nearly or directly touching, \\nwhile 100 µm represents a distance at which oxygen, nutrients and potentially other molecules \\ndiffuse in tissues31. Survival analysis was done as described above, using the CyCIF TMA1 dataset \\nas a discovery cohort and the other datasets as the validation cohort. For previously published \\nbiomarkers, the validation cohort included patients not used in development of that biomarkers. \\nSpecifically, tumor -immune mixing score, which was developed using the MIBI cohort, was \\nvalidated with the CyCIF and  IMC cohorts. Lymphocyte clustering, lymphocyte occupancy and \\nheterotypic neighbor biomarkers were originally developed in external cohorts, so all samples were \\nincluded in the validation cohort (and no discovery cohort was used). Spatial and survival analysis \\ncode, and links to data are available here: https://github.com/engjen/cycIF_TMAs.  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='7bd6a622-1c8c-4529-9d10-b175c7a08559', embedding=None, metadata={'page_label': '9', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='RESULTS \\nAnalysis of different multiplex imaging datasets produces similar single-cell phenotypes \\nWe generated CyCIF data from two tissue microarrays (TMAs) containing surgical breast cancer \\nspecimens and analyzed publicly available breast cancer data from IMC and MIBI platforms \\ngenerated with similar antigen targets (Figure 1). The CyCIF data was comprised of 47 biomarkers \\nimaged at a resolution of 0.325 µm per pixel in entire cores with 1.2 to 1.4 mm diameters, with 1-\\n2 full cores imaged per patient (Figure 1c top, S4). The IMC data included 35 biomarkers imaged \\nin the largest square area contained within the TMA cores (from 0.6 - 0.8 mm diameter), at a \\nresolution of 1 µm per pixel ( Figure 1c middle, S4)15. The MIBI data included 36 biomarkers \\nimaged in 0.8 x 0.8 mm square ROIs at a resolution of 0.5 µm per pixel (Figure 1c bottom, S4)13.        \\n \\nWe used the same methods to generate single-cell phenotypes via unsupervised clustering in each \\nplatform. In the CyCIF dataset, we generated a 30 k-nearest neighbors (k-NN) graph using a subset \\nof 23 markers common between the CycIF and IMC datasets, plus nuclear area. Markers were \\nscaled by the standard devia tion (SD), clipped at 20 SDs, and nearest neighbors were calculated \\nusing scanpy27.  We visualized the k-NN using UMAP embedding, confirming good separation of \\nlineage specific markers, CD31, endothelial, E -cadherin (Ecad), epithelial, Collagen I (ColI),  \\nextracellular matrix, vimentin, mesenchymal cells including activated fibroblasts, and CD45, \\nimmune infiltrate ( S5a). The UMAP visualization showed good mixing of cells from different \\nTMA sources and separation of tumor cells from different breast cancer subtypes (S5b). We used \\nthe Leiden algorithm27 to cluster the k-NN graph, resulting in 23 cell type clusters (S5c). The mean \\nexpression of each biomarker in each cluster was used to annotate cell types (e.g., CD8 T cell, \\nCD4 T cell, luminal ER+ tumor) and lineages (i.e., endothelial, epithelial, fibroblast, immune and \\nother stromal, S5d). The most common cell types included luminal and luminal ER+ tumor, CD4 \\nT cells, vimentin+ fibroblasts and quiescent stroma ( S5d). To confirm our cell typing, we \\nperformed manual gating on lineage specific markers ( S5e). Gating and clustering -based cell \\nlineages localized to similar areas of the UMAP and had 73% accuracy on a single cell level, as \\ncalculated using metrics.accuracy_score in scikit-learn32 (S5e, f).       \\n \\nIn the IMC dataset, we used the same method as in the CyCIF dataset to visualize marker \\nexpression, TMA, and subtype and cluster cell types (S6f). We used 21 markers plus nuclear area \\nfor clustering, resulting in 24 cell types ( S6d). Upon annotation, we found the most common cell \\ntypes were similar to those in the CyCIF samples, namely luminal, luminal ER+, and ER+ HER2+ \\ntumor, vimentin or fibronectin (FN)+ fibroblasts, quiescent stroma, and T cells ( S6d). Clustering \\nand gating-based cell lineages localized to similar areas of the UMAP and had 77% accuracy (S6e, \\nf).  \\n \\nThe MIBI panel included more immune-specific markers than the other panels and had 15 markers \\nshared with  our Cy CIF panel  (S4). To generate a dataset in which to audit deeper i mmune \\ncontexture, we clustered on all 33 markers plus nuclear area and eccentricity. Again, we saw good \\nseparation of lineage specific markers and Leiden clustering generated 22 cell types of which \\nluminal tumor, fibroblasts, T cells and quiescent stroma w ere the most common ( S7a-c). Gating-\\nbased cell types showed good overlap with clustering cell types with 72% accuracy (S7d-e).  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='51ce3edf-5204-4758-b1fb-f7e27c45c981', embedding=None, metadata={'page_label': '10', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' \\nCell type fractions are similar across platforms \\nAs our most common cell types were similar across platforms, we were encouraged that we could \\nsuccessfully combine the data for cross -platform analysis. We visually validated individual \\nclusters, as well as cell lineages by inspecting the images. In all three platforms, cell lineage \\nidentity and spatial distribution matched the underlying imaging data (Figure 1 c,d).  \\n \\nFor further validation, we sought to quantitatively compare the fractions of the five main cell \\nlineages. We calculated the fraction of cells in each lineage for each platform and breast cancer \\nsubtype. Both gating and clustering cell types showed high correlation (Pearson R= 0.97 gating \\nand 0.96 clustering) across platforms for ER+ (n=30 CyCIF, 170 IMC) and TNBC (n=59 CyCIF, \\n50 IMC, 41 MIBI) while HER2+, which had a smaller number of samples (n= 8 CyCIF, 22 IMC) \\nhad more variability between platforms (Figure 1 e, f). We did note some platform specific bias; \\nfor example, IMC showed a smaller fraction of immune cells defined by clustering in all three \\nsubtypes ( Figure 1 e, f ). Therefore, when settin g high/low cutoffs for cell abundances, we \\ncalculated high and low relative to each platform and subtype, as opposed to the whole dataset. \\nSince different antibody clones, probes and imaging systems with were used, resulting in different \\nsignal-to-background ratios between platforms, even for the same target, we believe this is a \\nnecessary step to account for technical variability.        \\nUnsupervised clustering defines prognostic tumor subtypes consistent across platforms \\nFor subtyping based on single -cell phenotypes, we calculated the fraction of epithelial or non -\\nepithelial stromal cells in the total epithelial or stromal cells in that sample, respectively. \\nNormalized f ractions of epithelial cell types representing greate r than 4% of the epithelial \\ncompartment were used to cluster ER+ and TNBC patients from all platforms (Figure 2 a, S8). \\nThe resulting seven subtypes included tissues enriched for luminal, basal, luminal ER+, \\nmyoepithelial, cytokeratin-low, and proliferating tumor cells, as well as a heterogeneous group not \\ndominated by one phenotype ( Figure 2 a). Each subtype contained a mixture of patients from \\nmultiple platforms, with no significant enrichment for any platform (Chi-squared p=0.2 ER+, \\np=0.52 TNBC, Figure 2 b, c). The epithelial subtypes present in the ER+ patients were prognostic \\n(log-rank overall-survival (OS) p=0.04, n=162 patients, Figure 2 d). Cox proportional hazards \\nmodelling (CPH) showed that the heterogeneous subtype 6 had a log hazard ratio > 1, indicating \\npoor prognosis (p=0.079, Figure 2 d). The CPH hazard ratios (HRs) were similarly ordered across \\nthe IMC and CyCIF cohorts, with heterogeneous subtype 6 having a HR > 1 in both platforms (S9 \\na, b). Investigation of the poor prognosis subtype 6 tissues revealed expression of CD44 and EGFR \\nin tumor cells (S9g). Quantification of associated stromal celltypes  significant enrichment of \\nCD31+ endothelial cells (Tukey HSD p<0.05) and analysis of epithelial marker expression showed \\nincreased CD44 and EGFR expression in subtype 6  (FDR<0.05, Figure 2 g-h, S9 h ). In TNBC \\npatients, the epithelial subtypes were not significantly associated with prognosis (S9c-e).  \\nMicroenvironment subtypes associate with clinical subtype \\nWe then clustered patients based on the stromal cell type fraction in each tissue in a similar manner \\nto the epithelial subtyping above, selecting k=6 for the number of clusters (S10). Since the stromal \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='29d8e973-09ee-41c4-b58d-a7c74ad0f8cb', embedding=None, metadata={'page_label': '11', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='phenotypes differed across platforms due to different markers, we clustered patients from each \\nplatform separately, using t he fractions of stromal cell types representing greater than 2% of the \\nstromal compartment. The stromal subtypes were not prognostic, with the exception of the MIBI \\nplatform (log -rank=0.003, CPH p=0.056, n=39 patients S10b-g). However, we observed \\nsignificant correlation between stromal subtypes and clinical subtypes ER+ and TNBC. In the \\nCyCIF cohort, ER+ patients had significantly more of the vimentin+ stromal subtype 0 and less T \\ncell-rich stroma (Chi-squared p=0.098, Bonferroni p-adj<0.05 for subtype 0, n=89 patients, S10e). \\nSimilarly, in the IMC cohort, ER+ patients had more Vim+/FN+ fibroblast stromal subtype 0 and \\nsignificantly less T cell stromal subtype 4 (Chi squared p=0.002, Bonferroni p -adj<0.05 for \\nsubtype 4, n=220 patients S10h). Our characterization of stromal subtypes supports the observation \\nthat ER+ breast cancer is immune -poor33 and shows significant enrichment for vimentin and \\nfibronectin+ fibroblasts relative to TNBC.    \\nT cells are an independent prognostic factor in TNBC and high proliferation ER+ tumors \\nNext, we investigated the prognostic value of multiplex imaging-defined cell types within each \\nclinical subtype. We used the CyCIF TMA1 dataset as a training cohort to identify cell types whose \\nfraction of all cells in the tissue were significantly associated with overall survival and cut-offs for \\nhigh or low abundance. We tested the 0.33, 0.5 and 0.66 quantile to binarize tissues into low and \\nhigh cell abundance ( Figure 3 a, c, S11a-c). For any cell type showing prognostic significance \\n(log-rank p<0.05), w e selected the cut -off with the lowest p -value for validation in the other \\ncohorts (Figure 3 a, c). Using this methodology, we found that a h igh abundance of T and B cells \\nwere associated with longer OS in TNBC in both the discovery and validation cohorts (validation \\nFDR<0.05, Figure 3, a-d).  \\n \\nBiomarkers significant as a single variable were  combined with the clinical variables of stage, \\npatient age and tumor size  in a multivariate Cox-proportional hazards ( CPH) model. In TNBC \\nsamples with clinical data, high CD3 T abundance remained significantly associated with longer \\nOS in the multivariate model (CPH p=0.038, n=89, Figure 3 d). High CD20 B cell abundance was \\nborderline significantly associated with longer OS in the multivariate model (CPH p=0.073, n=89, \\nFigure 3 b).  \\n \\nCD3 T cells were not associated with OS  in the ER+ breast cancer patients (FDR=0.12, n=162), \\nhowever, in tumors with proliferation above the median, increased CD3 T cells were associated \\nwith longer OS (FDR=0.0028, n=74) (Figure 3 e). Multivariate CPH modelling revealed that high \\nCD3 T cells were independently prognostic for longer OS in high proliferation but not low \\nproliferation ER+ tumors (CPH p=0.028 high proliferation, p=0.99 low proliferation, Figure 3 f). \\nSeparation of tumors by high/low proliferation and high/low T cell abundance showed that in ER+ \\ndisease high proliferation, high T cell tumors had the best prognosis, while high proliferation low \\nT cell had the worse prognosis ( Figure 3 g). In TNBC, high- and low-proliferation, high-T cell \\ntumors had similarly good survival, while high - and low -proliferation, low -T cell tumors had \\nsimilarly poor survival (Figure 3 f). Multivariate CPH modelling revealed that high CD3 T cells \\nwere borderline independently prognostic for longer OS  in high proliferation but not low \\nproliferation TNBC tumors (CPH p=0.067 high proliferation, p=0.31 low proliferation, Figure 3 \\nh). Independent analysis i n each platform revealed significant outcome stratif ication by T cell \\nabundance and proliferation in ER+ patients from the IMC cohort but not the CyCIF cohort ( log-\\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b86f499f-390c-44c2-aad5-a529a0c50715', embedding=None, metadata={'page_label': '12', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='rank p=0.028 and 0.4 S12 b), and in TNBC patients from the IMC and CyCIF cohorts but not MIBI \\ncohort (log-rank p=0.056, 0.012 and 0.49 S12 c).  \\n \\nT cells in patients deriving a survival benefit from infiltration have distinct functional states \\n \\nIn order to identify T cell functional states present in patients deriving a survival benefit from T \\ncell infiltration, we compared T cell spatial localization and marker expression in T cell infiltrated \\ngroups with different survival outcomes. There was no significant difference in T cell abundance \\nor T cell to macrophage or endothelial ratio in high versus low proliferation ER+ tumors (S10 e). \\nInterestingly, high-proliferation ER+ tumors in the IMC cohort, which gained survival benefit \\nfrom CD3 T cells,  showed more clustering of CD3 T cells than low-proliferation ER+ tumors, \\nquantified by the number of CD3 T cell neighbors of each CD3 T cell (Tukey HSD p=0.05, Figure \\n3 f). High proliferation ER+ tumors in the CyCIF cohort, which did not gain survival benefit from \\nCD3 T cells, did not show increased T cell clustering  (Kruskal-Wallis p=0.61, S12 d). Similarly, \\nCD3 T cells in high -proliferation ER+ tumors from IMC cohort had higher levels of the \\nproliferation marker Ki67 and the memory/effector marker CD44 than in low proliferation ER+ \\ntumors, indicating a more activated functional state (Tukey HSD p=0.006 and 0.01, Figure 3 i-k). \\nIn the CyCIF cohort, similar differences in Ki67 and CD44 expression were observed between \\nER+ and TNBC subtypes, consistent with an activated T cell state correlating with a survival \\nbenefit derived from increased T cell infiltration ( S12 f). High-proliferation TNBC, in which T \\ncells independently predicted overall survival in the multivariate CPH model (Figure 3 j, k, also \\nshowed increased levels of PD -1, FoxP3, IDO and Lag3 expression in T cells, consistent with \\nupregulation of negative feedback checkpoints following immune activation ( S12 f-g). Epithelial \\ncells in high -proliferation TNBC had increased expression of the antigen presentation molecule \\nHLA-Class-1 and immune checkpoint PD-L1 (S10 h). \\nAnalysis of tumor-immune interactions reveals conserved spatial biomarkers  \\nIntrigued by the finding that T -cells in high-proliferation tumors had increased T cell neighbors \\nand were associated with a survival benefit (Figure 3 i), we leveraged our discovery and validation \\ncohort analysis to systematically investigate spatial tumor-immune interactions as biomarkers in \\nbreast cancer. We calculated the number of immune and tumor neighbors within proximity of each \\nother to  derived previously publish ed biomarkers including mean neighbor counts 12,14, tumor-\\nimmune mixing score 13, lymphocyte clustering and lymphocyte occupancy 16. We also used \\ncommon statistical methods for quantification of spatial correlation (Ripley’s L, Kcross and Gcross \\nfunctions29). Analysis of cell lineage proximity revealed that stromal (non-fibroblast, non-immune, \\nnon-endothelial) neighbors of epithelial cells predicted longer RFS in the discovery (log-rank \\np=0.018) and validation ER+ cohorts (log-rank FDR=0.028), and independently predicted RFS in \\na multivariate CPH model ( CPH p=0.02, Figure 4 a-c). Immune neighbors of immune were \\nassociated with longer OS in both TNBC cohorts (validation log-rank FDR=0.019), but not in the \\nmultivariable model (CPH p=0. 14, Figure 4 c-d). Analysis of cell type neighbors showed \\nincreased m acrophage neighbors of tumor and increased B cell neighbors of T cells predicted \\nlonger RFS in TNBC in both  cohorts (validation FDR=0.05)  and r emained significant  the \\nmultivariate model ( CPH p=0.047 Macrophage -tumor, p=0.017 B cell -T cell , Figure 4 e-h). \\nAdditional spatial metrics, including Ripley’s L, Kcross and Gcross, did not yield any significant \\nvalidated biomarkers (S13a, g).  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='767548be-cbed-4c8f-a690-b412f6860779', embedding=None, metadata={'page_label': '13', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' \\nPreviously, Ali et al. 14 showed that heterotypic neighborhoods of myofibroblasts , fibroblasts , \\ncytokeratin low tumor cells, and vimentin+ Slug-macrophages were associated with poor outcome \\nand homotypic neighbors of fibroblasts and myofibroblasts were as sociate with good outcome in \\nall breast cancer subtypes. We tested the prognostic value of heterotypic and homotypic neighbors \\nof fibroblast subsets, CK low tumor and macrophages but did not find the significant association \\nwith survival (log-rank FDR>0.3, S13 b, c). Previously, Keren et al.13 showed that a high tumor-\\nimmune mixing score was associated with poor survival  in TNBC. Encouragingly, we were able \\nto reproduce the prognostic value of the mixing score in the MIBI cohort, where it was developed \\n(log-rank p=.027, Figure 4 i). However, in a validation cohort containing CyCIF and IMC patients, \\nthe mixing score was not prognostic (log-rank p=.26, Figure 4 i), nor was it independently \\nprognostic in samples with clinical outcome (CPH p=0.4, S13 e). Previously, Wortman et al.16 \\nshowed that TNBC  with longer RFS  had more spatially dispersed lymphocytes, as defined by \\nisolated lymphocytes  at <5 per 20 um radius, and an increased tissue area occupied by \\nlymphocytes. Our analysis supported the findings of Wortman et al., with greater numbers of \\nisolated T cand B cel ls and higher T and B cell area occupancy in tissue associated with longer \\nRFS (log-rank FDR <0.05, Figure 4 j, k). Isolated lymphocytes and lymphocyte occupancy also \\nremained prognostic in the multivariable model in TNBC ( CPH p<0.05, except CD20 B cell \\noccupancy p=0.078, Figure 4 l). \\n \\nTo identify spatial metrics which provided additional information beyond abundance, we \\ncalculated the Pearson correlation between all our spatial metrics and cell type abundance within \\neach patient’s tissue (S14). Stromal neighbors of epithelial (good prognosis in ER+) correlated \\nwith quiescent stroma abundance and macrophage neighbors of tumor (good prognosis in TNBC) \\ncorrelated with macrophage abundance (S14) . CD20 B cell neighbors of T cells, isolated \\nlymphocytes and lymphocyte occupancy  (good prognosis in TNBC)  correlated with each other \\nand T and B cell abundance (S14). The tumor-immune mixing score was positively correlated with \\ntumor abundance and negatively correlated with immune abundance (S14). Many of the  Kcross \\nand Ripley’s L function results correlated with each other and were not as strongly correlated with \\nabundance (S14).   \\n \\nNeighborhood analysis reveals multicellular spatial biomarkers \\n \\nFinally, we analyzed multicellular spatial neighborhoods  by considering all stromal cells within \\n100 µm radius of each tumor cell. We used spatial latent Dirichlet allocation (LDA) to model the \\nneighborhood around each tumor cell as a com bination of topics, utilizing a spatial parameter to \\nincrease the likelihood that adjacent cells share the same topics30. In LDA analysis, each topic can \\ncontain multiple cell types and each cell type can be present in multiple topics. In our CyCIF data, \\nfor example, topic -0 in TNBC tissues was enriched in macrophages, vimentin+ fibroblasts and \\nCD4 T cells, while CD4 T cells are found in topic-0, 4, 5 and 6 (Figure 5 a). After topic modelling, \\nK-means clustering was run on the single -cell topic matrix to define “tumor neighborhood ” \\nclusters which contained one or more topics (Figure 5 b). Clustering the topic matrix rather than \\nthe neighbor count matrix is believed to be less sensitive to noise 34  and allows for smooth \\ntransitions between neighborhoods as opposed to arbitrary cutoffs 30. W e did observe \\ntransitioning/mixed neighborhoods within both TNBC and ER+ cluster results (Figure 6 b, d, S15). \\nThe spatial LDA neighborhood clusters were annotated based on their topics and examination of \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d36302a4-45ec-4743-9def-1afbe9b7005a', embedding=None, metadata={'page_label': '14', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='the images showed neighborhoods reflected the spatial distribution of the markers in the tissue \\n(S15a-d). A TNBC tissue in our CyCIF cohort, for example,  showed tumor cell neighborhoods \\nwith more  T ce lls (blue) on the tumor margin, with  adjacent macrophage-rich tumor \\nneighborhoods (purple) ( S15 a -b). These neighborhoods transition into a mixed neighborhood \\n(brown), and finally a vimentin+ FB neighborhood (green) distant from the infiltrating T cells (S15 \\na-b). An ER+ tissue showed quiescent stroma tumor cell neighborhoods in the tumor core, \\nvimentin+ FB neighborhoods on the tumor margin and T cell neighborhoods in tumor nests \\nisolated in the stroma (S15 c -d). Similar neighborhoods were identified in the IMC TNBC and \\nER+ cohorts (S15 e-h) but the MIBI cohort was excluded from this analysis due to lower stromal \\ncell type overlap (Figure 1). For survival analysis, we used the CyCIF TMA1 as a discovery cohort \\nand CyCIF TMA2 plus IMC patients as a validation cohort. Due to fewer numbers of patients, we \\nused an alpha of 0.1 rather than 0.05 to report findings. Increased V imentin+ FB neighborhoods \\naround tumor cells were associated with shorter OS in both TNBC cohorts (validation log-rank \\np=0.07, Figure 5 e). Increased vimentin+ fibroblast neighborhoods were associated with shorter \\nOS and RFS in the multivariate model ( CPH p=0.049 OS, 0.0 53 RFS Figure 7 f, S14b). \\nInterestingly, vimentin+ fibroblast abundance alone was not prognostic , although neighborhood \\nabundance was correlated with fibroblast abundance  (S16). In ER+ tumors, increased mixed \\nfibroblast neighborhoods containing vimentin-positive and negative fibroblasts around tumor cells \\nwere associated with shorter OS (validation log -rank p=0.088), and trended significant in the \\nmultivariable model (CPH p=0.087 OS, p=0.046 RFS, Figure 5 g-h). Finally, similar to other \\ngroups34, we found that directly clustering the neighborhood counts using Kmeans (rather than \\nrunning LDA and clustering the topics) did not result in robust prediction of prognosis (S17). \\n \\nTumor phenotypes correlate with stromal cell abundance and spatial neighborhoods \\nWe hypothesized that there would be significant correlation between tumor cell types and the \\nsurrounding stromal cell neighborhoods, correlations that could  shed light on biologically and \\nclinically relevant tumor -stroma crosstalk. First, we visualized a matrix of pairwise correlation \\nbetween epithelial and stromal cell fractions and spatial LDA neighborhoods across subtypes \\n(S16a, c). Epithelial cell types were inversely correlated with each other, indicating most tumors \\nhad just one main epithelial cell type (S16a, c). The exception was luminal tumor, which correlated \\nwith cytokeratin low tumor in ER+ breast cancer, indicating mixing of these phenotypes within \\nthe same tissues (S16c).  \\n \\nImmune cells exhibited distinct tissue-level correlations in the different subtypes. In TNBC, T \\ncells correlated with proliferating tumor, and macrophages correlated with CD4 T cells (S16a). In \\nER+ breast cancer, T cells correlated with B cells while proliferating tumor, macrop hages and \\nendothelial cells were correlated with each other but not T cells (S16c). Vimentin+, FN+, and ColI+ \\nfibroblasts, as well as quiescent stroma were inversely correlated with immune cells (S16a, c). In \\nboth subtypes, spatial LDA neighborhoods correl ated strongly with the abundance of the ir \\nrespective stromal cell types; however, neighborhoods showed unique correlations to other cell \\ntypes present. For example, proliferating and luminal ER+ tumor cell abundances did not correlate \\nwith total T cell abundance in ER+ breast cancer, but they did correlate with the fraction of T cell \\nneighborhoods (S16c). In TNBC, vimentin+ and fibronectin+ fibroblast abundances were n ot \\ncorrelated, but their respective neighborhoods were inversely correlated, suggesting tissue \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f1e253ab-9551-47c1-8cec-4d748453bdf9', embedding=None, metadata={'page_label': '15', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='exclusivity for a single fibroblast phenotype near tumor cells (S16a, b). Therefore, although spatial \\nneighborhoods tend to correlate with cell abundance, they do reveal unique features of tumor -\\nstromal organization in tissues.  \\n \\nDISCUSSION \\nOur approach of combined analysis across multiple platforms shows the power of our methods for \\nbiomarker discovery. We were able to incorporate analysis of two publicly available imaging \\ndatasets with our own CyCIF data for efficient discovery of robust biomarkers.       \\n \\nWe utilized our validated method for CyCIF staining and image processing18 to generate multiplex \\nimaging data of 42 markers in a single tissue section from two TMAs with clinical follow-up. Our \\ndataset alone represents a valuable new clinical cohort that provides improved plex, resolution, \\nand ROI size compared to previously published datasets 13,15. We then developed an analysis \\npipeline (https://github.com/engjen/cycIF_TMAs) to generate single cell phenotyping data from \\nour CyCIF dataset and two publicly available datasets13,15. The advantage of using our pipeline for \\nimage processing is the development of smoothing algorithms so that pixelated image data can be \\nsegmented with deep -learning models trained on higher resolution images  and an algorithm to  \\nmatch nuclear and cell segmentation results from separate deep -learning segmentation models to \\nextract features from subcellular compartments such as the nucleus and cytoplasm ( S1b). Using \\nour methods, we  generated single cell data that produced a high correlation between  cell type \\nfractions across cohorts from the same breast cancer subtype and different platforms (Figure 1).  \\n \\nAdditionally, we were able  identify similar epithelial phenotypes across platforms  and cluster  \\npatient data from all platforms to separate seven epithelial subtypes that did not have a platform-\\nspecific bias ( Figure 2 ). Our subtypes were similar to the intrinsic breast cancer subtypes 35, \\nincluding a luminal ER+ luminal A -like group with good prognosis  and a cytokeratin low group \\npreviously shown to share features with luminal B tumors 14. Triple-negative breast cancers also \\nfell into categories similar to those defined by gene expression profiling 36, including a highly \\nproliferative, basal-like 1 (BL1) -like group, a luminal androgen receptor (LAR) -like group with \\nluminal epithelial phenotypes, and a group with a  basal/myoepithelial phenotype reminiscent of \\nthe basal-like 2 (BL2) group. We also identified a heterogeneous subtype with low cytokeratin and \\nelevated CD44 expression that may represent tumors with mesenchymal features.  Jackson et al., \\nidentified a similar single cell pathology cluster of hormone-receptor positive mixed tumors with \\npoor prognosis15. In our analysis, the ER+ tumors in the heterogenous subtype had poor prognosis \\nand showed increased angiogenesis. An EMT-program in breast cancer cells is linked to increased \\nvascular endothelial growth factor A expression, increasing angiogenesis and the capacity for \\ntumor initiation37, a mechanism that could explain these correlated tumor and stromal phenotypes \\nand their association with poor outcome.  \\n \\nTumor infiltrating lymphocytes have been linked to good prognosis in TNBC38, and we confirmed \\nthat T and B cells are independent ly prognostic in TNBC  in the multiplex imaging datasets \\nanalyzed herein. Previous gene expression profiling studies link productive anti-tumor immunity \\nand tumor proliferation. Nagalla et al. found that immune signatures were prognostic solely in \\nbreast cancer patients with the highest proliferation gene expression 39. Subsequently, the same \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='065d6e3b-2b28-4543-9e55-e9d9079a1f78', embedding=None, metadata={'page_label': '16', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='group showed that immune gene signatures were prognostic in highly proliferative basal -like, \\nHER2-enriched and luminal B subtypes, but not those with low proliferation40. Similar to this \\nprevious work, we showed that CD3 T cells were independently prognostic specifically in high-\\nproliferation ER+ and TNBC tumors.  \\n \\nOur analysis of immune functional states showed increased T cell proliferation, activation and \\ncheckpoint molecules and epithelial antigen presentation in high -proliferation tumors, consistent \\nwith IFNγ pathway activation. Consistent with our analysis, g ene network analysis previously \\nshowed activation of TNFα/IFNγ signaling pathways in tumors with productive anti -tumor \\nimmunity and TGF -β, an immunosuppressive cytokine , in tumors with unproductive anti -tumor \\nimmunity40. TGF-β also has anti-proliferative effects and is associated with good outcome in ER+ \\nbreast cancer cohorts41, suggesting that it could mechanistically link lower proliferation rates with \\nimmunosuppression and represent a rational drug combination with immune checkpoint \\ntargeting42.  \\n \\nMore recently, the Thomas et al. showed that immune gene signatures were prognostic exclusively \\nin tumor-mutation burden (TMB)-high breast cancer tumors43. Thirty-seven percent of basal-like \\ntumors had high TMB, while only 11.5% of luminal A tumors did 43, explaining the poor \\nimmunogenicity of the latter subtype. Together, these data point to a model of high TMB \\ncorrelating with high-proliferation and both linked to productive anti-tumor immunity. It had been \\nhypothesized that oncogenes driving sustained proliferation also induce DNA r eplication stress, \\nwhich generates genomic instability and would increase TMB 44. In summary, TMB provides a \\nmechanistic link between proliferation and anti -tumor immunity and should be investigated in \\nfuture studies. Furthermore, our analysis shows enrichment of potential immune checkpoint targets \\nin high-proliferation breast cancer, including PD1, Lag3, IDO and PD-L1 elevation (S12). \\n \\nOne of the main goals of  this study was to provide methods and a framework for robust \\nidentification of spatial biomarkers. Our use of external cohorts to validate biomarkers discovered \\nin our CyCIF data increases our confidence in biomarker identification. In ER+ breast cancer, we \\nfound that increased stromal neighbors of tumor correlated with better progn osis, similar to \\nprevious studies showing survival benefit for high stroma in ER+ tumors 45. We found that \\nmacrophage proximity to tumor was associated with good prognosis in TNBC, which is surprising \\ngiven previous publications . Specifically, t umor associated macrophages  (TAMs)46 were \\nassociated with shorter overall survival in a cohort of ER+ and ER - patients, but the prognostic \\nvalue of macrophages specifically in TNBC was not investigated. Furthermore, Medrek et al.47 \\nfound that CD68+ macrophages  in close proximity to tumor cells were not associated with poor \\nsurvival, but those out in the stroma were, suggesting that it may be difficult to compare our metric \\nof macrophage-tumor neighbors in a 40 µm radius with previous studies and further investigation \\nis warranted. Finally, numerous  immune-related spatial biomarkers, including immune -immune \\nproximity, B cell-T cell proximity, isolated lymphocyte abundance and lymphocyte occupancy \\nwere associated with good prognosis in  TNBC, supporting a model of productive anti -tumor \\nimmunity in the triple-negative subtype. Encouragingly, our results for the prognostic value of \\nlymphocyte spatial metrics were similar to those Wortman et al.16 discovered in a different TN \\ncohort.   \\n \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4eff7a0f-3d0a-45ba-9003-09c940f2df22', embedding=None, metadata={'page_label': '17', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='We utilized spatial LDA modelling to analyze multicellular neighborhoods of stromal cells \\nsurrounding tumor cells. We identified a neighborhood enriched for vimentin+ fibroblasts that was \\nindependently associated with shorter survival in TNBC. Given the high levels of vimentin and \\nlow levels of alpha-SMA, these cells may have an inflammatory phenotype similar to CAFs that \\ndifferentiate under TNFα + IL-1β stimulation48. Interestingly, TNFα + IL-1β have been shown to \\nstimulate pro -metastatic chemokine expression (CXCL8, CCL2 and CCL5) and aggressive \\ncharacteristics in TNBC cell lines, mediated in part by direct CAF -tumor cell contact in co -\\ncultures49, consistent with close proximity between poor-prognosis CAFs and tumor cells in spatial \\nLDA neighborhoods.  \\n   \\nThe limitations of our study include dif ferent antibody probes and imaging systems resulting in \\ndifferent signal -to-background ratios for biomarkers across platforms. Therefore, we relied on \\nmatching hand -annotated clusters across platforms for our integrated analysis. Since our \\nannotations may not correspond to the same cell types in each platform, this introduces uncertainty. \\nSome well -defined phenotypes, such as T cells and proliferating tumor, are relatively \\nstraightforward. However, variable performance of antibodies, such as anti-ER, for example, could \\nlead to variability in the classification of phenotypes such as luminal ER+ versus luminal tumor \\nacross platforms. To correct for platform -specific bias in cell types, we binarized patients into \\nhigh/low expression within each subtype and plat form for survival analysis . However, such \\nbinarization may not reflect underlying heterogeneity in quantitative biomarker abundance.  \\n \\nAnother limitation of our study is the use of 1 -2 TMA cores per patient for analysis. It has been \\nshown that a limited number of TMA cores (≤3) are needed to binarize patients in to high and low \\nTILs, although a larger number of cores (≥11) are needed to a ccurately estimate the mean TIL \\nabundance50. Our survival analysis relied on binarizing patients; therefore, the use of TMAs may \\nbe appropriate in this context. Lacking large tissue sections, we had no way to quantify sensitivity \\nof spatial metrics to sa mpling bias introduced by analyzing small ROIs, but our cross-cohort \\nanalysis revealed a lack of generalization of spatial biomarkers in several cases  (S13a). \\nPerformance of spatial biomarkers could potentially be improved by sampling larger tissue areas, \\nalthough increased heterogeneity in large section s could introduce noise . The optimal balance \\nbetween the area analyzed in each tissue and number of patients included  for estimation of \\nprognostic tumor microenvironment composition and spatial  architecture remains an open and \\nimportant question in the field.     \\n \\nOverall, our spatial analysis supports the utility of spatial information in uncovering novel \\nbiomarkers of patient outcome in breast cancer. The tools developed in this study can be utilized \\nto analyze additional cohorts for further characterization of biomarkers in breast cancer and other \\ntumor types.   \\n  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c55ede0d-285b-41e1-b4ed-3b44126492e5', embedding=None, metadata={'page_label': '18', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='ACKNOWLEDGEMENTS \\nWe would like to thank Dr. Joe Gray for invaluable advice on analysis and feedback on the \\nmanuscript. We would also like to thank Drs. Sandra Rugonyi, Sara Courtneidge, Young Hwan \\nChang, and Pepper Schedin for feedback on earlier versions of this work. We would like to thank \\nDamir Sudar for technical support related to image storage and Elliot Gray for sharing his neighbor \\ncounter script. We sincerely appreciate Drs. Clare Yu, Peter Lee and Julie Wortman for sharing \\ntheir lymphocyte occupancy code. We appreciate the sample scanning assistance of both Drs. \\nStefanie Kaech Petrie and Crystal Chaw at the OHSU Advanced Light Microscopy Core. This \\nwork was supported by funding from NIH/NCI U54 CA209988, the Prospect Creek Foundation, \\nand OHSU Foundation. We received the services from the Knight Cancer institute Histopathology \\nShared Resource and Advanced Light Microscopy Core, supported by the Cancer Center Support \\nGrant (NIH/NCI P30CA69533).  Additionally, J.A.P acknowledges support from NIH P50 \\nCA098131 and Komen SAB210301. \\nAUTHOR CONTRIBUTIONS \\nJ.E., S.L.G., and K.C. conceived of the project. M.S., B.C., P.G., and J.A.P. collected patient \\ntissues, clinical outcom e data, and constructed the TMAs. Z.H. performed staining and imaging \\nexperiments. J.E. and E.B. did single -cell image processing. J.E. ran the analysis, drafted the \\nmanuscript, and prepared the figures. S.L.G., K.C., E.B., B.C. and J.A.P. edited the manuscript.  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='796dd61b-bcff-498c-b94f-bd453d634251', embedding=None, metadata={'page_label': '19', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='REFERENCES \\n1. Rizzo, A. & Ricci, A. D. Biomarkers for breast cancer immunotherapy: PD-L1, TILs, and beyond. \\nExpert Opin Investig Drugs 1–7 (2021) doi:10.1080/13543784.2022.2008354. \\n2. Schmid, P. et al. Pembrolizumab for Early Triple-Negative Breast Cancer. New England Journal of \\nMedicine 382, 810–821 (2020). \\n3. Gatti-Mays, M. E. et al. If we build it they will come: targeting the immune response to breast \\ncancer. NPJ Breast Cancer 5, (2019). \\n4. Tsujikawa, T. et al. Quantitative Multiplex Immunohistochemistry Reveals Myeloid-Inflamed \\nTumor-Immune Complexity Associated with Poor Prognosis. Cell Rep 19, 203–217 (2017). \\n5. Gut, G., Herrmann, M. D. & Pelkmans, L. Multiplexed protein maps link subcellular organization \\nto cellular states. Science (1979) 361, (2018). \\n6. Lin, J. R., Fallahi-Sichani, M. & Sorger, P. K. Highly multiplexed imaging of single cells using a high-\\nthroughput cyclic immunofluorescence method. Nat Commun 6, 1–7 (2015). \\n7. Goltsev, Y. et al. Deep Profiling of Mouse Splenic Architecture with CODEX Multiplexed Imaging. \\nCell 174, 968-981.e15 (2018). \\n8. Giesen, C. et al. Highly multiplexed imaging of tumor tissues with subcellular resolution by mass \\ncytometry. Nat Methods 11, 417–422 (2014). \\n9. Eng, J. et al. Cyclic Multiplexed-Immunofluorescence (cmIF), a Highly Multiplexed Method for \\nSingle-Cell Analysis. in Biomarkers for Immunotherapy of Cancer. Methods in Molecular Biology \\n(eds. Thurin, M., Cesano, A. & Marincola, F. M.) vol. 2055 521–562 (Humana, 2020). \\n10. Cabrita, R. et al. Tertiary lymphoid structures improve immunotherapy and survival in melanoma. \\nNature 577, 561–565 (2020). \\n11. Helmink, B. A. et al. B cells and tertiary lymphoid structures promote immunotherapy response. \\nNature 577, 549–555 (2020). \\n12. Janiszewska, M. et al. The impact of tumor epithelial and microenvironmental heterogeneity on \\ntreatment responses in HER2+ breast cancer. JCI Insight 6, (2021). \\n13. Keren, L. et al. A Structure Tumor-Immune Microenvironment in Triple Negative Breast Cancer \\nRevealed by Multiplexed Ion Beam imaging. Cell 174, 1373-1387.e19 (2018). \\n14. Ali, H. R. et al. Imaging mass cytometry and multiplatform genomics define the phenogenomic \\nlandscape of breast cancer. Nat Cancer 1, 163–175 (2020). \\n15. Jackson, H. W. et al. The single-cell pathology landscape of breast cancer. Nature (2019) \\ndoi:10.1038/s41586-019-1876-x. \\n16. Wortman, J. C. et al. Spatial distribution of B cells and lymphocyte clusters as a predictor of triple-\\nnegative breast cancer outcome. NPJ Breast Cancer 7, (2021). \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='eacefed1-2788-448d-bb3f-f645fe9536c6', embedding=None, metadata={'page_label': '20', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='17. Parmar, C., Barry, J. D., Hosny, A., Quackenbush, J. & Aerts, H. J. W. L. Data Analysis Strategies in \\nMedical Imaging. Clinical Cancer Research 24, 3492–3499 (2018). \\n18. Eng, J. et al. A framework for multiplex imaging optimization and reproducible analysis. \\nCommunications Biology 2022 5:1 5, 1–11 (2022). \\n19. Eng, J. et al. Cyclic Multiplexed-Immunofluorescence (cmIF), a Highly Multiplexed Method for \\nSingle-Cell Analysis. Methods in Molecular Biology vol. 2055 (2020). \\n20. Patwa, A. et al. Multiplexed imaging analysis of the tumor-immune microenvironment reveals \\npredictors of outcome in triple-negative breast cancer. Commun Biol 4, 1–14 (2021). \\n21. Stringer, C., Wang, T., Michaelos, M. & Pachitariu, M. Cellpose: a generalist algorithm for cellular \\nsegmentation. Nat Methods 18, 100–106 (2021). \\n22. Virtanen, P. et al. SciPy 1.0: fundamental algorithms for scientific computing in Python. Nat \\nMethods 17, 261–272 (2020). \\n23. Greenwald, N. F. et al. Whole-cell segmentation of tissue images with human-level performance \\nusing large-scale data annotation and deep learning. Nat Biotechnol 40, (2021). \\n24. Stringer, C., Wang, T., Michaelos, M. & Pachitariu, M. Cellpose: a generalist algorithm for cellular \\nsegmentation. 1–19 (2020) doi:10.1101/2020.02.02.931238. \\n25. Danenberg, E. et al. Breast tumor microenvironment structures are associated with genomic \\nfeatures and clinical outcome. Nat Genet 18, (2022). \\n26. napari contributors. napari: a multi-dimensional image viewer for python. Preprint at (2019). \\n27. Wolf, A., Angerer, P. & Theis, F. SCANPY: large-scale single-cell gene expression data analysis. \\nGenome Biol 19, (2018). \\n28. Traag, V. A., Waltman, L. & van Eck, N. J. From Louvain to Leiden: guaranteeing well-connected \\ncommunities. Sci Rep 9, 1–12 (2019). \\n29. Baddeley, A. R. E. T. R. Spatial Point Patterns: Methodology and Applications with R. (Chapman \\nand Hall/CRC Press, 2015). \\n30. Chen, Z., Soifer, I., Hilton, H., Keren, L. & Jojic, V. Modeling Multiplexed Images with Spatial-LDA \\nReveals Novel Tissue Microenvironments. Journal of Computational Biology 27, 1204–1218 \\n(2020). \\n31. Weinberg, D. H. and R. A. The Hallmarks of Cancer. Cell 107, 251–262 (2000). \\n32. Pedregosa, F. et al. Scikit-learn: Machine Learning in Python Fabian. Journal of Machine Learning \\nResearch 12, 2825–2830 (2011). \\n33. Goldberg, J. et al. The Immunology of Hormone Receptor Positive Breast Cancer. Front Immunol \\n12, 1–22 (2021). \\n34. Nirmal, A. J. et al. The Spatial Landscape of Progression and Immunoediting in Primary Melanoma \\nat Single-Cell Resolution. Cancer Discov 12, 1518–1541 (2022). \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='eff76ee8-e319-49cc-b469-1999d4057850', embedding=None, metadata={'page_label': '21', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='35. Perou, C. M. et al. Molecular portraits of human breast tumours. Nature 406, 747–752 (2000). \\n36. Lehmann, B. D. et al. Refinement of triple-negative breast cancer molecular subtypes: \\nImplications for neoadjuvant chemotherapy selection. PLoS One 11, (2016). \\n37. Fantozzi, A. et al. VEGF-mediated angiogenesis links EMT-induced cancer stemness to tumor \\ninitiation. Cancer Res 74, 1566–1575 (2014). \\n38. Denkert, C. et al. Tumour-infiltrating lymphocytes and prognosis in different subtypes of breast \\ncancer: a pooled analysis of 3771 patients treated with neoadjuvant therapy. Lancet Oncol 19, \\n40–50 (2018). \\n39. Nagalla, S. et al. Interactions between immunity, proliferation and molecular subtype in breast \\ncancer prognosis. Genome Biol 14, (2013). \\n40. Miller, L. D. et al. Immunogenic Subtypes of Breast Cancer Delineated by Gene Classifiers of \\nImmune Responsiveness. Cancer Immunol Res 4, 600–611 (2016). \\n41. Sato, M. et al. An integrated genomic approach identifies persistent tumor suppressive effects of \\ntransforming growth factor-β in human breast cancer. Breast Cancer Research 16, 1–23 (2014). \\n42. Li, S. et al. Cancer immunotherapy via targeted TGF-β signalling blockade in TH cells. Nature 587, \\n121–125 (2020). \\n43. Thomas, A. et al. Tumor mutational burden is a determinant of immune-mediated survival in \\nbreast cancer. Oncoimmunology 7, 1–12 (2018). \\n44. Macheret, M. & Halazonetis, T. D. DNA replication stress as a hallmark of cancer. Annual Review \\nof Pathology: Mechanisms of Disease 10, 425–448 (2015). \\n45. Downey, C. L. et al. The prognostic significance of tumour-stroma ratio in oestrogen receptor-\\npositive breast cancer. Br J Cancer 110, 1744–1747 (2014). \\n46. Leek, R. D. et al. Association of macrophage infiltration with angiogenesis and prognosis in \\ninvasive breast carcinoma. Cancer Res 56, 4625–4629 (1996). \\n47. Medrek, C., Pontén, F., Jirström, K. & Leandersson, K. The presence of tumor associated \\nmacrophages in tumor stroma as a prognostic marker for breast cancer patients. BMC Cancer 12, \\n1–9 (2012). \\n48. Rubinstein-Achiasaf, L. et al. Persistent Inflammatory Stimulation Drives the Conversion of MSCs \\nto Inflammatory CAFs That Promote Pro-Metastatic Characteristics in Breast Cancer Cells. \\nCancers 2021, Vol. 13, Page 1472 13, 1472 (2021). \\n49. Liubomirski, Y. et al. Tumor-stroma-inflammation networks promote pro-metastatic chemokines \\nand aggressiveness characteristics in triple-negative breast cancer. Front Immunol 10, 757 (2019). \\n50. Lee, A. T. J. et al. The adequacy of tissue microarrays in the assessment of inter- and intra-\\ntumoural heterogeneity of infiltrating lymphocyte burden in leiomyosarcoma. Sci Rep 9, 14602 \\n(2019). \\n \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='72d57ded-d3cf-4ea4-a2f2-6591907be560', embedding=None, metadata={'page_label': '22', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='FIGURES \\n \\nFigure 1. Analysis of breast cancer multiplex imaging datasets from different platforms. \\na. Three multiplex imaging (MXI) datasets from breast cancer tissue microarrays were processed through single cell \\nsegmentation and feature extraction using the mplexable pipeline. The single cell datasets were separately clustered \\nusing the unsupervised L eiden algorithm resulting in cell types which were annotated with similar names across \\nplatforms. Finally, the data were used for discovery and validation of prognostic cell abundance and spatial \\nbiomarkers. cycIF, cyclic immunofluorescence; IMC, imaging mass cytometry; MIBI, multiplex ion beam imaging. \\nb. Overlap of markers (top) and annotated cell types (bottom) in each multiplex imaging dataset. c. Representative \\nimages from the three multiplex imaging platforms showing epithelial (orange), immune (red) and fibroblast (green) \\nmarkers. d. Gated cell types showing cell location and lineages: epithelial (orange), immune (red), fibroblast (green), \\nendothelial (blue) and other stromal (purple). e-f. Cell lineage fraction of total cells per subtype, per platfor m using \\neither manual gating (e) or unsupervised clustering and annotation (f) to determine lineage. Pearson’s correlation \\nbetween platforms (r=0.xx) and number of patients for each subtype and platform shown in figure title. \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0f219848-468f-4944-b586-179f7df20754', embedding=None, metadata={'page_label': '23', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' \\n \\nFigure 2. Prognostic breast cancer subtypes in combined multiplex imaging data. \\na. All ER+ and TNBC patients were clustered based on the fraction in each patient’s tissue of the six most common \\nepithelial cell types, resulting in seven epithelial (Epithel.) subtypes. b. UMAP embedding of patients by fraction of \\nepithelial cell types in all tumor cells, colored by epithelial subtype (top) and platform (bottom). c. Chi-squared \\nanalysis of epithelial subtypes versus platform, p-values in figure title. d. Kaplan-Meier curves (p-value from log-\\nrank test) comparing overall survival (OS) in the seven epithelial subtypes present in ER+ tumors. N number of \\npatients given in figure title. e. Cox proportional hazard (CPH) model estimating hazard ratios for epithelial \\nsubtypes of ER+ tumors. f. Mean abundance of epithelial cell types per subtype. g. Fraction of endothelial cells in \\ntissue stromal cells of each epithelial subtype. Kruskal-Wallis H-test P-value given in figure title. Post-hoc Tukey \\nHSD used for pairwise comparisons between groups. h. CD44 intensity in epithelial cells from poor prognosis \\nepithelial subtype 6 compared to other ER+ patients. P-value FDR corrected t-test test given in figure title.     \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='91fcec08-8226-4865-937e-fe5d0e2821e3', embedding=None, metadata={'page_label': '24', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' \\nFigure 3. T cell infiltrate has prognostic value in TN and high proliferation ER+ breast \\ncancer.  \\na. Kaplan-Meier analysis of abundance of CD20 B cells versus OS in TNBC discovery (left) and validation cohort \\n(right) b. Multivariate CPH modeling adding patient age, tumor size and stage to CD20 B cell high variable defined \\nin (a). c. Kaplan-Meier analysis of abundance of CD3 T cells versus OS in TNBC discovery (left) and validation \\ncohort (right) d. Multivariate CPH modeling adding patient age, tumor size and stage to CD3 T cell high variable \\ndefined in (c). e. Kaplan-Meier analysis of abundance of CD3 T cell versus OS in all ER+ patients (left) and ER+ \\npatients with high (above the median) tumor proliferation (right). f. CPH modeling of CD3 T cell abundance plus \\nclinical variables in high and low proliferation ER+ tumors. g. Kaplan-Meier analysis of all ER+ and TNBC patients \\nstratified into four groups by tumor proliferation and T cell abundance. h. CPH modeling of CD3 T cell abundance \\nplus clinical variables in high and low proliferation TNBC tumors.  a-h. All Kaplan-Meier p-values obtained from \\nthe log-rank test, validation cohort multiple testing corrected with Benjamini-Hochberg method. CPH modelling p-\\nvalues for cell type variable given in figure titles. n number of patients given in figure titles. i. Mean number of T \\ncell neighbors (within 25 µm) of T cells in tissues from high and low proliferation ER+ or TNBC tumors in IMC \\ncohort. j. Ki67 intensity indicating proliferation levels of T cells in tissues from high and low proliferation ER+ or \\nTNBC tumors in IMC cohort. k. CD44 intensity in T cells, indicating memory/effector phenotypes in IMC tissues. i-\\nk. Kruskal-Wallis H-test P-value given in figure title. Post-hoc Tukey HSD used for pairwise comparisons between \\ngroups. \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='82e34edf-35f8-479b-bf36-7daba70b2710', embedding=None, metadata={'page_label': '25', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' \\nFigure 4. Prognostic tumor-immune spatial correlations in breast cancer cohorts \\na. Spatial locations of cells in a TMA core with each cell colored by number of CD3 T cell neighbors in a 40 μm \\nradius. Tumor cells shown in orange. b. Recurrence free survival (RFS) Kaplan-Meier (K-M) analysis of ER+ patients \\nwith high versus low stromal nei ghbors of epithelial cells in a 40 μm radius in the discovery (left) and validation \\n(right) cohorts. c. Left - Multivariate CPH modeling of ER+ patient RFS adding patient age, tumor size and stage to \\nthe stromal neighbors of epithelial variable defined in (b). Right - multivariate CPH modeling of TNBC patient OS \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='963cc0c7-aef6-4f37-b67c-d154461bf7fe', embedding=None, metadata={'page_label': '26', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='versus immune neighbors of immune cells. d. Overall survival (OS) K-M analysis of TNBC patients with high versus \\nlow immune neighbors of immune cells in a 40 μm radius in the discovery (left) and validation (right) cohorts. e & g. \\nRFS K-M analysis of TNBC patients with high versus low macrophage neighbors of tumor (e) or B cell neighbors of \\nT cells (g) in a 40 μm radius in the discovery (left) and validation (right) cohorts. f & h. Multivariate CPH modeling \\nof TNBC patient RFS versus macrophage neighbors of tumor (f) or CD20 B cell neighbors of CD3 T cells (h). i. K -\\nM analysis of TNBC OS stratified by tumor immune mixing score developed by Keren et al. in the MIBI cohort (top) \\nand validation cohort (i.e. CyCIF and IMC; bottom). j. K-M analysis of TNBC OS stratified by isolated T (top) or B \\nlymphocyte (bottom) metric developed by Wortman et al. k. K -M analysis of TNBC OS stratified by T (top) or B \\nlymphocyte (bottom) occupancy metric developed by Wortman et al. l. CPH modeling of Wortman et al. metrics from \\nj and k. a-l. All Kaplan-Meier p-values obtained from the log-rank test. CPH modelling p-values for spatial variable \\ngiven in figure titles. N number of patients given in figure titles.  \\n.  \\n  \\n  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e93a4b6a-164f-4dcb-876b-a247dd5bdbf3', embedding=None, metadata={'page_label': '27', 'file_name': 'eng_breast.pdf', 'file_path': '/content/data/eng_breast.pdf', 'file_type': 'application/pdf', 'file_size': 4708099, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=' \\nFigure 5. Prognostic multicellular neighborhoods surrounding tumor cells modeled \\nwith spatial latent Dirichlet allocation.  \\na. Heatmap of stromal cell enrichment in spatial latent Dirichlet allocation (LDA) topic models of 100 μm TNBC \\ntumor neighborhoods in the CyCIF cohort. b. Heatmap of fraction of each topic in each neighborhood cluster \\nresulting from K-means clustering (k=8) of spatial LDA topics from (a). c-d. heatmaps as defined in a and b, for \\nER+ tumors from the CyCIF cohort.e. Kaplan-Meier (K-M) analysis of overall survival (OS) versus high or low \\nvimentin+ fibroblast neighborhoods in TNBC tissues in discovery (left) and validation cohorts (right). p-values \\nobtained from the log-rank test; n given in figure titles. f. CPH modelling of OS and recurrence-free survival (RFS) \\nwith clinical variables plus spatial LDA neighborhoods from (e). p-values and n number of patients given in figure \\ntitle. g. K-M analysis of OS versus high and low mixed fibroblast neighborhoods in ER+ tissues in the discovery \\n(left) and validation cohorts (right). h. CPH modeling of OS and RFS for mixed fibroblast neighborhoods in ER+ \\ntumors.  \\n  \\n.CC-BY-ND 4.0 International licenseavailable under a\\nwas not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made \\nThe copyright holder for this preprint (whichthis version posted May 15, 2023. ; https://doi.org/10.1101/2023.01.31.525753doi: bioRxiv preprint ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4f1dd5e1-fe0b-4654-81c6-62557509c02c', embedding=None, metadata={'page_label': '572', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='572 | Nature | Vol 619 | 20 July 2023\\nArticle\\nOrganization of the human intestine at \\nsingle-cell resolution\\n     \\nJohn W. Hickey1,10, Winston R. Becker2,10, Stephanie A. Nevins2,10, Aaron Horning2, \\nAlmudena Espin Perez3, Chenchen Zhu2, Bokai Zhu1, Bei Wei2, Roxanne Chiu2, Derek C. Chen2, \\nDaniel L. Cotter2, Edward D. Esplin2, Annika K. Weimer2, Chiara Caraccio1, \\nVishal Venkataraaman1, Christian M. Schürch1,4, Sarah Black1, Maria Brbić5,6, Kaidi Cao5, \\nShuxiao Chen7, Weiruo Zhang3, Emma Monte2, Nancy R. Zhang7, Zongming Ma7, \\nJure Leskovec5, Zhengyan Zhang8, Shin Lin9, Teri Longacre1, Sylvia K. Plevritis3, Yiing Lin8, \\nGarry P . Nolan1\\u2009✉, William J. Greenleaf2\\u2009✉ & Michael Snyder2\\u2009✉\\nThe intestine is a complex organ that promotes digestion, extracts nutrients, \\nparticipates in immune surveillance, maintains critical symbiotic relationships with \\nmicrobiota and affects overall health1. The intesting has a length of over nine metres, \\nalong which there are differences in structure and function2. The localization  \\nof individual cell types, cell type development trajectories and detailed cell \\ntranscriptional programs probably drive these differences in function. Here, to better \\nunderstand these differences, we evaluated the organization of single cells using \\nmultiplexed imaging and single-nucleus RNA and open chromatin assays across eight \\ndifferent intestinal sites from nine donors. Through systematic analyses, we find cell \\ncompositions that differ substantially across regions of the intestine and\\xa0demonstrate \\nthe complexity of epithelial subtypes, and find that the same cell types are organized \\ninto distinct neighbourhoods and communities, highlighting distinct immunological \\nniches that are present in the intestine. We also map gene regulatory differences in \\nthese cells that are suggestive of a regulatory differentiation cascade, and associate \\nintestinal disease heritability with specific cell types. These results describe the \\ncomplexity of the cell composition, regulation and organization for this organ, and \\nserve as an important reference map for understanding human biology and disease.\\nThe human adult intestinal system is a complex organ that consists of \\napproximately 7\\u2009m of small intestine and 2\\u2009\\u2009m of large intestine. This \\nsystem completes the digestive process that begins in the oral cavity \\nand stomach, first absorbing water and small-molecule nutrients (such \\nas sugars, monovalent ions and amino acids) in the small intestine, \\nthen accumulating larger molecules such as fibre in the large intestine, \\nwhich serves as an anaerobic fermentation chamber enabling the \\nbreakdown and absorption of by-products and the synthesis, often \\nthrough alimentary gut microbiota, and absorption of other nutrients \\nsuch as vitamins1.\\nThe small intestine is phenotypically heterogeneous, comprising \\nthree morphologically distinct regions—the duodenum, jejunum \\nand ileum2. The large intestine can be partitioned into the ascend -\\ning, transverse, descending and sigmoid regions. Each of these ana-\\ntomical regions contains an immense diversity of phenotypically and \\nmorphologically distinct cell types. Epithelial, stromal and immune \\ncells, each comprising multiple cell types, reside throughout the intes-\\ntine. Immune cells are of particular interest, as they interact with the \\nmicrobiome and foreign material present in the gut3. Although these \\nbroad cell types are common to all portions of the intestinal system, spe-\\ncific cell types are known to display locational preferences. For example, \\nPaneth cells populate the small intestine, and enteroendocrine L cells \\nare found primarily in the ileum and large intestine4,5. Moreover, these \\ncell types are spatially organized into different ‘neighbourhoods’ across \\nthese intestinal regions, and both the composition of these neighbour-\\nhoods and the molecular phenotypes of the\\xa0underlying cellular types \\nvary in relatively unknown ways across these anatomical regions. These \\ndifferences in both the composition of functional neighbourhoods and \\nthe molecular identity of the cell states that comprise these neighbour-\\nhoods define the composition and function of the human intestine.\\nHere, we map many portions of the intestine at the single-cell resolu-\\ntion using single-nucleus RNA, open chromatin and spatial proteomic \\nimaging technologies. Previous studies have mapped cell types using \\nsingle-cell RNA-sequencing (scRNA-seq) and have catalogued cell types \\nacross the intestine6. We extend this research by spatially mapping \\ncells and proteins using co-detection by indexing (CODEX)7–10 as well \\nhttps://doi.org/10.1038/s41586-023-05915-x\\nReceived: 29 November 2021\\nAccepted: 2 March 2023\\nPublished online: 19 July 2023\\nOpen access\\n Check for updates\\n1Department of Pathology, Stanford School of Medicine, Stanford, CA, USA. 2Department of Genetics, Stanford School of Medicine, Stanford, CA, USA. 3Department of Biomedical Data Science, \\nStanford School of Medicine, Stanford, CA, USA. 4Department of Pathology and Neuropathology, University Hospital and Comprehensive Cancer Center Tübingen, Tübingen, Germany. \\n5Department of Computer Science, Stanford University, Stanford, CA, USA. 6School of Computer and Communication Sciences, École Polytechnique Fédérale de Lausanne, Lausanne, \\nSwitzerland. 7Department of Statistics and Data Science, University of Pennsylvania, Pennsylvania, PA, USA. 8Department of Surgery, Washington University, St Louis, MO, USA. 9Department  \\nof Medicine, University of Washington, Seattle, WA, USA. 10These authors contributed equally: John W. Hickey, Winston R. Becker, Stephanie A. Nevins. ✉e-mail: gnolan@stanford.edu;  \\nwjg@stanford.edu; mpsnyder@stanford.edu', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e1cb01b6-4bb4-469b-8e8f-99f0d0f27bfb', embedding=None, metadata={'page_label': '573', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 573\\nas mapping gene regulatory information using single-cell assay of \\nopen chromatin using single-nucleus assay for transposase-accessible \\nchromatin with sequencing (snATAC–seq)11. We define the relative abun-\\ndance of distinct cell types across the intestine, including the enormous \\ncomplexity of epithelial cells across different intestinal regions, and the \\norganization of cells into different multicellular structural niches. We \\nalso map gene regulatory differences in these cells that are suggestive \\nof a regulatory differentiation cascade. These results provide important \\ninsights into cell function, regulation and organization for this complex \\norgan and serve as an important reference for understanding human \\nbiology and disease.\\nMapping the human intestine\\nWe mapped the cell composition, regulatory information and spatial \\ndistribution of single cells across the intestines of multiple donors \\nusing single-nucleus RNA-seq (snRNA-seq), which measures nuclear \\nRNA transcripts in individual nuclei; snATAC-seq, which measures \\nopen chromatin in single cells; and CODEX, which stains the same  \\ntissue section with up to 54 antibody probes against different targets \\n(usually proteins). We analysed eight sections from nine individuals: \\nseven individuals of European ancestry (five male and two female) \\nand two African American individuals (one male and one female). Age \\nranges were from 24 to 78\\u2009years. The eight regions (in order of trajectory \\nfrom the stomach) were as follows: the duodenum, proximal jejunum, \\nmid-jejunum and ileum from the small intestine, and the ascending, \\ntransverse, descending and sigmoid regions of the large intestine.\\nMultiplexed imaging of the intestine\\nT o create a spatial map of the intestine across the eight regions, we \\nused CODEX multiplexed imaging, which enables insights into cellular \\ninteractions, composition of multicellular tissue units and spatial \\nrelationships to the overall function of the intestine 9,10. We first vali-\\ndated and optimized CODEX staining, imaging and image process -\\ning for 16-mm 2 sections of fresh-frozen samples on one participant \\n(B001) (Supplementary Figs.\\xa01–3). For the other eight donors, we \\nexpanded our CODEX antibody panel by adding and validating 17 \\nintestine-specific markers (Supplementary Information\\xa01 and Sup -\\nplementary Fig.\\xa04) for a total of 54 antibodies that enabled the spatial \\nidentification of 25 cell types 12 (Extended Data Fig.\\xa01c and Supple -\\nmentary Figs.\\xa05 and 6).\\nWe used the resultant dataset (a total of 2.7\\u2009million cells) to compare \\nthe cellular composition and organization across the different tissue \\nregions, normalizing to overall cell grouping (Fig.\\xa01a–c). Within the \\nstromal compartment, moving from the small intestine to the colon, \\nwe observed a decrease in endothelial cells and an increase in smooth \\nmuscle cells (Fig.\\xa01a and Extended Data Fig.\\xa01d). T o verify that this was \\nnot an artefact of capturing more muscularis externa within samples of \\nthe colon compared with the small intestine, we calculated the percent-\\nages of all cell types within the four different pathological compart-\\nments of the intestine: the mucosa, muscularis mucosa, submucosa \\nand muscularis externa. Indeed, even when comparing all of the cell \\ntypes found within the muscularis externa, there was still a significant \\ndecrease in endothelial cells and an increase in smooth muscle cells \\n(Extended Data Fig.\\xa01e). Thus, not only is there less vasculature more \\nbroadly in the colon, but there is less within the muscularis externa and \\na higher density of smooth muscle cells.\\nIn the immune compartment, we observed a decrease in CD8+ T\\xa0cells \\nfrom the small intestine to the colon (Fig.\\xa01b, Extended Data Fig.\\xa01f and \\nSupplementary Fig.\\xa07a–c), consistent with previous observations 13. \\nConversely, we observed an increase in the percentage of dendritic \\ncells within the colon compared with in the small intestine that is also \\nseen when examining total cell percentages within the mucosa (Fig.\\xa01b \\nand Extended Data Fig.\\xa01f).\\nIn the epithelial compartment, we observed a decrease in entero-\\ncytes, an increase in secretory enterocytes (goblet cells) and CD66+ \\nenterocytes and an absence of Paneth cells when moving from the small \\nintestine to the colon (Fig.\\xa01c, Extended Data Fig.\\xa01g and Supplemen-\\ntary Fig.\\xa07d). We also detected a rare population of CD57+ enterocytes \\nthat is enriched within the duodenum compared with in other areas \\nof the intestine (Extended Data Fig.\\xa01h and Supplementary Fig.\\xa07e). \\nThese gastric-like cells are enriched in areas of the duodenum within \\nsubmucosal glands (Extended Data Fig.\\xa01i and Supplementary Fig.\\xa07f).\\nCell type associations with clinical data\\nWe also evaluated cell type changes with donor metadata. M1 macro-\\nphage levels had the highest correlation with body mass index (BMI) \\n(Fig.\\xa01d) and were restricted to the mucosa (Fig.\\xa01e). M1 macrophages \\nare pro-inflammatory and have been implicated in chronic inflamma-\\ntory disease, autoimmunity and problems with wound healing in the \\nintestine14–16. Similarly, obesity increases the risk of gastrointestinal \\ndisorders17. Although the donors did not have histories of gastrointes-\\ntinal disorders, we found that individuals with a BMI characterized as \\noverweight (25–29.9) have a fivefold increase in M1 macrophages and \\nindividuals with obesity (BMI\\u2009>\\u200930) have an eightfold increase compared \\nwith individuals who are normal weight (18.5–25). We also observe \\ndecreases in endothelial cells (from 25% to 20%) and CD8+ T\\xa0cells (from \\n42% to 25%) in donors with a history of hypertension (Fig.\\xa01f ). High \\npressure due to a lower ratio of total vasculature is expected, but a sub-\\nstantial decrease in CD8+ T\\xa0cells is surprising (Supplementary Fig.\\xa07a).\\nSpatial restriction of immune cells\\nIn addition to cell type composition, cellular density can highlight \\nwhether a cell has broad functions over large regions, is spatially \\nrestricted for specialized functions or has the need for specific cell–\\ncell interactions. We quantified the local cell density for all cell types \\n(Fig.\\xa0 1g and Extended Data Fig.\\xa01j,k). Visual inspection suggested \\nthat plasma cells (~0.2) with the highest same-cell type density were \\nrestricted to specific mucosal areas, followed by CD8+ T\\xa0cells (~0.37), \\nthen M2 macrophages (~0.5), which were diffuse throughout all areas \\nof the intestine (Extended Data Fig.\\xa01l). Indeed, M1 macrophage den-\\nsity (~0.39) was lower than its M2 counterpart. Quantification of the \\ndistribution of each macrophage subset within the different intesti-\\nnal tissue units indicates a spatial restriction of macrophage subsets \\n(Fig.\\xa01h), suggesting important functional roles in these regions and \\nthat other macrophage subtypes among M2 macrophages exist that \\nmay also be spatially restricted. In summary, these results suggest an \\nimportant role for spatial restriction of immune cell subtypes along \\nthe length of the intestine.\\nStromal multicellular neighbourhoods\\nT o provide a global view of intercellular interactions, cellular densities \\nand overall multicellular structures of the intestine, we performed cel-\\nlular neighbourhood analysis10 (Methods and Extended Data Fig.\\xa01a). \\nThis revealed 18 significant multicellular structures with major epi-\\nthelial, stromal and immune-based neighbourhoods (Fig.\\xa02a,b and \\nExtended Data Fig.\\xa02b–d). Eight neighbourhoods were classified as \\nstromal neighbourhoods and identified major structures within the \\nintestine: micro- and macrovasculature, innervated stroma and smooth \\nmuscle, and innate immune hubs within the stroma and smooth muscle \\nareas (Fig.\\xa02a). Only the Smooth Muscle neighbourhood increased \\nmoving from the small intestine to the colon, whereas the innervated \\nand innate immune smooth muscle neighbourhoods did not (Extended \\nData Fig.\\xa02e and Supplementary Fig.\\xa08a). This further suggests that \\nthese dense compartmentalized smooth muscle cell areas (Extended \\nData Fig.\\xa02f) increase within the colon.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8b3634b2-1a1a-451f-a087-74c4e9949eeb', embedding=None, metadata={'page_label': '574', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='574 | Nature | Vol 619 | 20 July 2023\\nArticle\\nImmune multicellular neighbourhoods\\nCongruent with our observation of high plasma cell density (Fig.\\xa01g), we \\nobserved a Plasma-Cell-Enriched neighbourhood driven by increased \\ndensity of plasma cells (Fig.\\xa02a). This Plasma-Cell-Enriched neighbour-\\nhood also exhibits co-enrichment of CD4+ T\\xa0cells and antigen-presenting \\ncells such as dendritic cells and macrophages (Fig.\\xa02a) and is localized \\nwithin the mucosa lamina propria (Extended Data Fig.\\xa02g–i). These \\nobservations are consistent with observations suggesting that secre-\\ntion or ligand engagement of plasma cells from antigen-presenting cells \\nwithin the bone marrow can maintain long-term survival in plasma- \\nspecific niches18–20.\\nNotably, despite their relatively low density in the intestine, CD8+ \\nT\\xa0cells were enriched in two major neighbourhoods (Fig.\\xa02a). One neigh-\\nbourhood (CD8+ T\\xa0Cell-Enriched IEL\\xa0(intraepithelial lymphocyte)) \\nexhibits enrichment of both epithelial cell types and CD8+ T\\xa0cells. Thus, \\nthe neighbourhood analysis was able to separate the CD8+ T\\xa0cells that \\nare intraepithelial lymphocytes, which are critical for rapid immuno-\\nlogical responses against infection21 and maintenance of epithelial \\nintegrity22. This CD8+ T\\xa0Cell-Enriched IEL neighbourhood was one of \\nthe neighbourhoods of which the prevalence changed from the small \\nintestine (~30%) to the colon (~3%) (Extended Data Figs.\\xa02c and 3a and \\nSupplementary Fig.\\xa08a). This is the spatial compartment from which we \\nobserved a global decrease in CD8+ T\\xa0cell percentage (Fig.\\xa01b) and also \\nsignificantly decreases with a history of hypertension (Extended Data \\nFig.\\xa03b). We suggest that the ability of CD8+ T\\xa0cells to survive or locate \\nwithin intraepithelial spaces is negatively affected by hypertension.\\nThe decrease in the CD8+ T\\xa0Cell-Enriched IEL neighbourhood is met \\nwith an increase in the Plasma-Cell-Enriched neighbourhood within the \\ncolon (Extended Data Figs.\\xa02c and 3a). In particular, there is a significant \\ndecrease in Plasma-Cell-Enriched neighbourhoods within the ileum as \\ncompared to the colon (Supplementary Fig.\\xa07c–d). By contrast, there \\nis also an increase in the Adaptive-Immune-Enriched neighbourhood \\nwithin the ileum as compared to the colon (Supplementary Fig.\\xa07e–f). \\nNotably, although proximal to the colon, the ileum has the most distinct \\nimmune microenvironment from the colon.\\nCD4+ T\\xa0cells contributed to five diverse multicellular neighbour -\\nhoods (Fig.\\xa02a). This broad neighbourhood membership is fitting, given \\nthat CD4+ T\\xa0cells coordinate innate and adaptive immune responses. \\nCD4+ T\\xa0cell, B cell and dendritic cell membership defined two different \\nfollicle-based structures. The first of these structures, which exists \\nin outer regions of the follicle, exhibited higher enrichment of CD4+ \\nT\\xa0cells, whereas inner regions of the follicle were enriched for B cells \\n(Fig.\\xa02a). The presence of the Inner Follicle (that is, the germinal centre)  \\nneighbourhood was dependent on a fully mature lymphoid follicle \\nlike a Peyer’s patch within the image (Fig.\\xa0 2b). However, the Outer \\nPlasma\\nB\\nCD4\\n+  T cell\\nDC\\nCD8\\n+  T\\nNeutrophil\\nM1 macrophag\\ne\\nM1 macrophag\\ne\\nCD7\\n+  immune\\nNK\\nM2 macrophage M2 macrophage\\n0.1\\n0.2\\n0.3\\n0.4\\n0.5\\n0.6\\n0.7\\n0.8\\n05 10\\nM1 macrophage\\npercentage of immune cells \\n20.0\\n22.5\\n25.0\\n27.5\\n30.0\\n32.5\\n35.0\\n37.5BMI\\nEnterocyte\\nCycling TA\\nGoblet\\nCD66+ enterocyte \\nTA\\nMUC1+ enterocyte \\nPaneth \\nNeuroendocrine \\nCD57+ enterocyte\\n0\\n20\\n40\\n60\\n80\\n100\\nCD8+ T\\nPlasma\\nM2 macrophage \\nCD4+ T cell\\nB\\nDC\\nM1 macrophage \\nNeutrophil\\nCD7+ immune\\nNK\\n0\\n20\\n40\\n60\\n80\\n100\\n0\\n20\\n40\\n60\\n80\\n100\\nSmooth muscle \\nStroma \\nEndothelial \\nNerve \\nLymphatic \\nICC\\ng\\nIncreasing same-celld ensity\\nAvg. same cell–cell distance\\nMax. avg. same cell–cell distance\\nImmune cell types\\na\\nSB CL\\n*\\n*\\n*\\nStromal\\nImmune\\nEpithelial\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\nEpithelialc ells (%) Immune cells (%) Stromalc ells (%)\\nb\\nc\\nHistoryo f hypertension\\nCD\\n8\\n+ Tcell\\nPercentage of cells in subtype\\n** **\\nfh\\nPercentageb yl ocationi ni ntestine\\nde\\nr = 0.86 HighB MI\\nEpithelial\\nM1 macrophage\\nStromal\\nLow BMI\\nMid-jejunum\\n10\\n20\\n30\\n40\\nDuodenum\\nProximal jejunum\\nIleum\\nMid-jejunu\\nm\\nAscendingT ransvers\\ne\\nDescending\\nDescending–sigmoi\\nd\\n0\\n20\\n40\\n60\\n80\\n100\\nMucosa\\nMuscularis mucosa\\nSubmucosa \\nMuscularis externa\\nNo\\nYes\\nEn\\ndo\\nthelia\\nl\\nFig. 1 | CODEX multiplexed imaging of eight regions from the small \\nintestine and colon to create a single-cell map of the healthy human \\nintestine. a–c, Cell type percentages from CODEX data averaged across eight \\ndonors. Cell types are normalized to the stromal ( a), immune (b) and epithelial \\n(c) compartments. Statistical analysis was performed using two-sided t-tests \\ncomparing the difference in cell type percentage between the small bowel (SB) \\nand the colon (CL); *P\\u2009<\\u20090.05. ICC, interstitial cells of Cajal; NK, natural killer \\ncells; TA, transit-amplifying cells. d , The percentage of M1 macrophages within \\nthe small bowel and colon for all donors plotted versus donor BMI (Pearson \\ncorrelation r\\u2009=\\u20090.86). e, Cell type maps of the mid-jejunum from representative \\nindividuals ( n\\u2009=\\u20098 donors) with high or low BMI with M1 macrophages (black) \\nhighlighted among stromal (light grey) and epithelial (grey) cell types also \\nshown. Scale bar, 250\\u2009µm. f, Cell type percentages for endothelial and CD8 + \\nT\\xa0cells compared for donors with or without a history of hypertension. \\nStatistical analysis was performed using two-sided t-tests; *P\\u2009=\\u20090.038, \\n***P\\u2009=\\u20090.00013. n\\u2009=\\u20093–5 donors. g, Quantification of the same-cell density \\nmeasured as an average distance of its five nearest same-cell neighbours \\ndivided by the maximal possible same-cell distance within the tissue. n\\u2009=\\u200964 \\ntissue sections. h , The percentage of macrophage subsets across major \\nintestinal compartments.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='158f37ce-cd7b-4f82-bea2-473c6bd891f3', embedding=None, metadata={'page_label': '575', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 575\\nFollicle neighbourhood was found across the intestine irrespective \\nof the presence of a fully mature follicle. Out of the 64 tissues that we \\nimaged, 11 had mature follicles that we segmented for comparison \\n(Supplementary Fig.\\xa09a–b). The cell type and neighbourhood compo-\\nsitions differed between each follicle irrespective of location, driven \\nprimarily by percentage of dense B cells (Supplementary Fig.\\xa09a), found \\nwithin the Inner Follicle neighbourhood (Supplementary Fig.\\xa09b). The \\nvariation in the ratio of Inner Follicle neighbourhood to Outer Follicle \\nneighbourhood across the intestine suggests a continuum of lymphoid \\ntissues within the intestine23–26.\\nVariation in the multicellular composition of cellular neighbour -\\nhoods may indicate a core functionality as well as a need for composi-\\ntional flexibility based on anatomical location. We compared the cell \\ntype compositions in the small intestine versus the colon for all neigh-\\nbourhoods and found that both inner and outer follicle structures are \\nless conserved, whereas stromal neighbourhoods are more conserved \\n(Extended Data Fig.\\xa03c). We observed differences in neighbourhood \\ncomposition in Paneth-Cell-Enriched (significantly less abundant in \\nthe colon\\xa0than the small bowel) and Transit-Amplifying-Zone (not dif-\\nferent\\xa0in terms of abundance) neighbourhoods that are both enriched \\nfor early epithelial progenitor cells, potentially indicating different \\ncrypt microenvironments across the intestine (Extended Data Fig.\\xa02d \\nand Supplementary Fig.\\xa09c,d).\\nIntestinal crypt neighbourhoods\\nAs we observed differences in neighbourhood cell type conservation, \\nwe performed neighbourhood analysis on each individual region \\nof the intestine separately and then concatenated the results. All \\naggregated neighbourhoods (Fig.\\xa02a) were identified (Extended \\nData Fig.\\xa03d). We also identified two unique neighbourhoods: \\nNeuroendocrine-Enriched, which was found only in the colon, \\nand Neutrophil-Enriched, which was found throughout the intes -\\ntine but enriched in the colon (Fig.\\xa02c and Extended Data Fig.\\xa03e,f).  \\nPaneth cell\\nCD4/CD8\\nICCs\\na\\nImmuneStromal Epithelial\\nNeighbourhood\\nNeighbourhood map\\nPaneth cell\\nneighbourhood window\\nNeuroendocrine\\nenriched neighbourhoods\\nNeuroendocrine Neuroendocrine\\nCell type\\nc\\nf\\nNeighbourhood\\nCLSB\\nPercentage of all\\nneighbourhoods\\nNeighbourhood\\nLumen\\nMucosa\\nSubmucosa\\nAvg. same cell–cell distance\\nMax. avg. same cell–cell distance\\nAvg. same cell–cell distance\\nMax. avg. same cell–cell distance\\nMuscularis mucosa\\nHoechst\\nDEFA5\\nCytokeratin\\n/uni03B1-SMA\\nCD4\\nCD8\\nMuscularis mucosa\\ngh i\\nCell type\\nlog2[fold\\nenrichment] \\nNumber of nearest neighbours\\nj\\n5 nearest neighbours\\n10\\n20\\nlog2[fold\\nenrichment] \\nb\\nCLSB\\nd e\\n–2.5\\n0\\n2.5\\nPaneth\\nMUC1+ enterocyte\\nCD57+ enterocyte\\nCD8+ T\\nCD7+ immune\\nCD66+ enterocyte\\n Cycling TA\\nTA\\nEnterocyte\\nGoblet\\nNeuroendocrine\\nB\\nPlasma\\nSmooth muscle\\nNerve\\nNK\\nDC\\nStroma\\nCD4+ T cell\\nM2 macrophage\\nLymphatic\\nICC\\nEndothelial\\nM1 macrophage\\nNeutrophil\\nGlandular epithelial\\nPaneth enriched\\nCD66+ mature epithelial\\nSecretory epithelial\\nTransit-amplifying zone\\nCD8+ T enriched IEL\\nMature epithelial\\nInner follicle\\nInnate immune enriched\\nPlasma cell enriched\\nAdaptive immune enriched\\nOuter follicle\\nSmooth muscle\\nStroma\\nStroma and innate immune\\nMicrovasculature\\nMacrovasculature\\nInnervated smooth muscle\\nInnervated stroma\\nSmooth muscle and innate immune\\n0\\n2.5\\n5.0\\n7.5\\n10.0\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nNeuroendocrine enriched\\nNeuroendocrine enriched\\nTransit-amplifying zone\\nSecretory epithelial\\nSecretory epithelial\\nSmooth muscle\\nTransit-amplifying zone\\nCD8+ T enriched IEL\\nMature epithelial\\nMature epithelial\\nCD66+ mature  epithelial\\nCD66+ mature epithelial\\nCD8+ mature epithelial\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nPaneth\\nCD66+ enterocyte\\nNeuroendocrine\\nCD57+ enterocyte\\nB\\nCD7+ immune\\nLymphatic\\nM2 macrophage\\nPlasma\\nEndothelial\\nSmooth muscle\\nNeutrophil\\nStroma\\nCD4+ T cell\\nMUC1+ enterocyte\\nM1 macrophage\\nDC\\nNerve\\nTA\\nGoblet\\nEnterocyte\\nNK\\nCycling TA\\nCD8+ T\\nICC\\n300\\n100 \\n50\\n10 \\n30 \\n5\\n–2.5\\n0\\n2.5\\nFig. 2 | Multicellular neighbourhood analysis of the intestine. a, Twenty \\nunique intestinal multicellular neighbourhoods were defined by enriched cell \\ntypes as compared to the overall percentage of cell types in the samples. b , An \\nexample in which neighbourhoods mapped back to the tissue show overall \\ntissue structures. Scale bar, 0.5\\u2009mm. c , The percentage of Neuroendocrine-  \\nEnriched neighbourhood of all of the neighbourhoods as determined by \\nindividually characterizing cellular neighbourhoods by region. n\\u2009=\\u20098 donors. \\nThe box plots show the median (centre line), 25th to 75th percentile (box limits), \\nminimum and maximum values (whiskers), and outliers (points outside  \\n1.5× the interquartile range). d,e, Quantification of the same-cell density for \\nneuroendocrine cells compared across the small bowel and colon ( n\\u2009=\\u200932 tissue \\nsections) (d) or the epithelial neighbourhoods as determined by individually \\ncharacterizing cellular neighbourhoods by region ( n\\u2009=\\u200964 tissue sections) ( e). \\nAvg., average; max., maximum. f, A subset of epithelial neighbourhoods mapped \\nback to a representative magnified region ( n\\u2009=\\u20098 donors) of the mucosa of a \\ntransverse colon section. Scale bar, 250\\u2009µm. g ,h, The approach to calculate \\nconcentric increasing neighbourhoods around a Paneth cell ( g) to generate \\ncellular neighbourhoods for Paneth cells at increasing radii ( h). i,j, Schematic \\n(i) and CODEX fluorescence data illustrating a representative (1 of 32 sections \\nfrom 8 donors) magnified portion of the proximal jejunum depicting \\ncolocalization of Paneth cells (DEFA5, green) and CD8+ T\\xa0cells (CD8, cyan) and \\nCD4+ T\\xa0cells (CD4, yellow) in the intestinal crypt environment (j). Scale bar, 50\\u2009µm.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='187a72e7-f30d-4a10-ab8e-57aa6fb9d994', embedding=None, metadata={'page_label': '576', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='576 | Nature | Vol 619 | 20 July 2023\\nArticle\\nThe Neutrophil-Enriched neighbourhood was characterized by a \\nhigh density of neutrophils associated with vasculature and innate \\nimmune cells, found often within stromal and smooth muscle areas \\n(Extended Data Fig.\\xa03d). The Neuroendocrine-Enriched neighbour -\\nhood had a mixture of epithelial and immune cell types enriched \\n(Extended Data Fig.\\xa03d).\\nThe identification of the Neuroendocrine-Enriched neighbourhood \\nsuggested differential organization of neuroendocrine cells in the small \\nintestine compared with in the colon. Indeed, neuroendocrine cells \\nwere found to be denser within the colon compared with in the small \\nintestine (Fig.\\xa02d). Furthermore, neuroendocrine cells are most dense \\nwithin the Neuroendocrine-Enriched neighbourhood as compared to \\nneuroendocrine cells in other epithelial neighbourhoods (Fig.\\xa02e), and \\nthe density decreases with the maturity of the epithelial cell types that \\ndefine these epithelial neighbourhoods (Fig.\\xa02e). This suggests that \\nthe Neuroendocrine-Enriched neighbourhood represents the colon \\ncrypt neighbourhood, which is confirmed by its localization near the \\nmuscularis mucosa (Fig.\\xa02f).\\nIdentification of this crypt environment only within the colon and not \\nin the small intestine is consistent with our finding of variation in early \\nepithelial environments in our first neighbourhood analysis (Extended \\nData Fig.\\xa03c), and Paneth-Cell-Enriched neighbourhoods are observed \\nonly\\xa0within the small intestine (Extended Data Fig.\\xa03e). Indeed, Paneth \\ncells are known to be restricted to the small intestine and also to be \\nenriched within the intestinal crypt. As a consequence, to understand \\nwhether the Neuroendocrine-Enriched neighbourhood was similar to \\nthe small intestine crypt environment, we examined the neighbours \\nsurrounding Paneth cells with increasing window size (Fig.\\xa02g). This \\nanalysis revealed that there was a high enrichment of neuroendocrine \\ncells, but it also underscored that interstitial cells of Cajal, CD4+ T\\xa0cells \\nand CD8+ T\\xa0cells were enriched within the local microenvironment \\n(Fig.\\xa02h–j). This agrees with enrichment of CD4+ T\\xa0cells and interstitial \\ncells of Cajal found within the Neuroendocrine-Enriched neighbour-\\nhood (Extended Data Fig.\\xa03d).\\nHierarchical structural organization\\nMulticellular neighbourhood analysis revealed key differences in the \\nstructural composition across the intestine as well as in the composi-\\ntion of these neighbourhoods, particularly with relation to the adaptive \\nimmune system and the intestinal crypt. However, how these multi-\\ncellular neighbourhoods interact with one another, and how they \\nare spatially structured in the tissue is unclear. Understanding how  \\nmulticellular groups are related is key to both defining the hierarchy of  \\ntissue organization, as well as defining key functional tissue interfaces.\\nWe investigated higher-order structural organization using several \\nmethods. First, we clustered windows of neighbourhood compositions \\nin a manner similar to how we clustered windows of cell types (Fig.\\xa03a) \\nto define neighbourhoods (Fig.\\xa03b). This generated communities of \\nneighbourhoods (Fig.\\xa03c and Extended Data Fig.\\xa04a–c), which we then \\nleveraged to identify major tissue units such as the muscularis mucosa \\n(Fig.\\xa03d and Supplementary Fig.\\xa010). The Paneth-Cell-Enriched neigh-\\nbourhood was enriched within the Adaptive-Immune-Enriched com-\\nmunity (Extended Data Fig.\\xa04a). Indeed, concentric neighbourhood \\nenvironments surrounding the Paneth-Cell-Enriched neighbourhood \\nalso showed colocalization with Outer Follicle and Adaptive-Immune- \\nEnriched neighbourhoods (Extended Data Fig.\\xa04d). These results sup-\\nport the idea that the adaptive immune system forms a conserved niche \\nwith the intestinal crypt.\\nT o relate the various levels of spatial organization, we created a hier-\\narchical structure network graph (Fig.\\xa03e and Supplementary Fig.\\xa011). \\nEach level of this graph is connected to the next by its major contribu-\\ntors. Using this intuitive formalism, we can observe many levels of intes-\\ntinal cell and tissue-structure organization. For example, we observed \\ncrosstalk between stromal and smooth muscle cell types and structures, \\nwhich are in turn isolated from epithelial and immune components that \\nare more entwined with one another (Fig.\\xa03e (red bracket)).\\nUsing this graph structure, we can also observe multilevel relation-\\nships between the structures. For example, Paneth cells (Fig.\\xa03e (green \\ncircle)) are a rare cell subset (size) and are primarily enriched within \\nPaneth-Cell-Enriched neighbourhoods (Fig.\\xa03e (light blue square)), \\nwhich are enriched in the Adaptive-Immune-Enriched community \\n(Fig.\\xa03e (orange triangle)) that, in turn, is enriched within the muscularis \\nmucosa tissue unit (Fig.\\xa03e (red diamond)). This relationship can be \\nseen within the tissue, where we see the Adaptive-Immune-Enriched \\ncommunity localized to the bottom of the colon crypt (Fig.\\xa03f).\\nVisualizing the communities also revealed spatial layering of the \\nintestine, moving from the smooth muscle, stroma and particularly \\nwithin epithelial areas (Fig.\\xa03f ). T o formalize these observations of \\nintercommunity-level spatial interactions, we created27 a spatial con-\\ntext map27 (Methods) that revealed major structural relationships \\nbetween communities within the colon (Fig.\\xa03g). For example, moving \\nfrom left to right in the spatial context map parallels tissue organiza-\\ntion in a cross-section of the intestine moving from the muscularis \\nexterna to the top of the mucosa (Fig.\\xa03f,g). In brief, the Smooth Muscle \\ncommunity (red triangle) is often found alone (size of circle), indi -\\ncating its compartmentalization from other communities (Fig.\\xa03g). \\nHowever, it is found next to the Stromal community (grey triangle), \\nwith which it forms an interface (grey and red triangle combination) \\n(Fig.\\xa03g (yellow highlighted edge)). This then forms a trio interface \\nwith the Adaptive-Immune-Enriched community (orange triangle). \\nMoving to the right, we observe another trio that involves the Smooth \\nMuscle, Adaptive-Immune-Enriched and Secretory Epithelial com -\\nmunities (Fig.\\xa0 3g (green box)). This pattern continues across the \\nPlasma-Cell-Enriched community (yellow triangle) and then the Mature \\nand CD66+ Epithelial community (teal triangle) (Fig.\\xa03g).\\nWe confirmed these community–community interactions using \\nan established method for identifying two-combination community  \\nmotifs27 (Methods). Significant associations with just the Adaptive- \\nImmune-Enriched community shared across the intestine demon -\\nstrate that there are connections to the\\xa0Plasma-Cell-Enriched, Smooth  \\nMuscle, Secretory Epithelial, Stromal and Follicle communities but not \\nto the Mature Epithelial community (Extended Data Fig.\\xa05b and Sup-\\nplementary Fig.\\xa012), which aligns with our previous analyses (Fig.\\xa03f,g). \\nSimilarly, analysis of other shared significant motifs shows that the \\nPlasma-Cell-Enriched community intersects with both the Secretory \\nEpithelial and Mature Epithelial communities (Extended Data Fig.\\xa05c). \\nWe created a spatial context map for just the cells found in the mucosa \\n(Extended Data Fig.\\xa05d–f) and, again, observed a high-frequency inter-\\nsection between the Plasma-Cell-Enriched, Secretory Epithelial and \\nTransit-Amplifying Zone\\xa0neighbourhoods (Extended Data Fig.\\xa05f (red \\nbox)) with many connections to the Plasma-Cell-Enriched neighbour-\\nhood, implicating an important role in overall intestinal tissue structure.\\nIn summary, these results indicate that immune-cell-enriched \\nneighbourhoods have important roles in intestinal tissue organiza-\\ntion. Indeed, we found enrichment of cell types moving from the \\nsmooth muscle community to the lumen (Fig.\\xa03h and Extended Data \\nFig.\\xa05g,h), which shows an increase in adaptive immune cells in the \\nbase of the crypt, restricted zones of plasma cells, and an increase in \\ninnate immune cells and CD8+ T IELs towards the top of the intestine \\n(Fig.\\xa03h). In conclusion, our hierarchical mapping data further confirm \\ncompositional differences in multicellular structures between the \\nsmall intestine and colon but also highlight conserved multicellular \\nstructure interactions and an important distribution of distinct cell \\ntypes in subregions of the intestine.\\nSingle-nucleus RNA and chromatin atlas\\nThe CODEX experiments revealed distinct cellular arrangements across \\nintestinal regions, but included only 54 probes, potentially limiting ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0e323a4f-4d01-42b5-a5c6-3c132a992270', embedding=None, metadata={'page_label': '577', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 577\\nboth the number and complexity of cell types identified. T o overcome \\nthese limitations, we performed 10x multiome sequencing analysis \\nof intestinal regions from six donors and separate 10x snRNA-seq and \\nsnATAC-seq analysis of intestine regions from three additional donors \\n(Fig.\\xa04a, Methods and Extended Data Fig.\\xa06a, b).\\nWe first annotated all snRNA cells from both the snRNA and multi-\\nome experiments together, by dividing cells into immune, stromal and \\nepithelial compartments, revealing a total of 10 immune, 16 stromal \\nand 16 epithelial cell types (Fig.\\xa04b–d), and annotated the remaining \\nscATAC cells from three donors separately. Cell types were anno -\\ntated by examining gene expression levels and gene activity scores of \\nknown marker genes as well as by labelling the datasets with previously  \\npublished scRNA-seq data28 (Methods).\\nThe majority of immune cell types were identified in both the single- \\nnucleus and CODEX data (Supplementary Table\\xa02). T o determine which \\ncell types were significantly differentially abundant between locations, \\nwe applied two methods that control for the compositional nature of \\nsingle-cell data29,30. We found that CD8+ T\\xa0cells were more abundant in \\nthe small intestine, whereas B cells were more abundant in the colon \\n(Fig.\\xa04f and Extended Data Fig.\\xa07), consistent with the CODEX results \\n(Fig.\\xa01b).\\nWithin the stromal compartment, we annotated eight fibroblast \\nsubtypes—interstitial cells of Cajal (KIT, ANO), glial cells (SOX10, CDH19, \\nPLP1), neurons (SYP, SYT1, RBFOX1), pericytes (NOTCH3, MCAM1, RGS5), \\nadipocytes (PLIN1, LPL) and three endothelial cell clusters (Fig.\\xa04b and \\nExtended Data Fig.\\xa06f). Cells with high expression of MYH11 and ACTA2 \\nwere classified as smooth muscle/myofibroblasts. One of these clus-\\nters had high expression of DES, which we labelled DES high and may \\nrepresent smooth muscle. We also identified fibroblasts with high \\nlevels of WNT agonists, such as RSPO3, and the BMP antagonist GREM1, \\nwhich are probably present at the crypts, and fibroblasts with high \\nexpression of WNT5B and BMP transcripts thought to be present at the \\nvilli28 (Extended Data Fig.\\xa06f). Elsewhere, the ADAMDEC1high popula-\\ntion has been referred to as S1 fibroblasts, the WNT5B and BMP signal-\\nling fibroblasts as S2 fibroblasts, and the KCNN3high population as S3 \\nfibroblasts6,31. Similar to the immune cells, we observed changes in cell \\ntype abundance along the intestine (Fig.\\xa04f and Extended Data Fig. 7). \\nFor example, two smooth muscle/myofibroblast clusters were less \\nabundant in the small intestine than in the colon. Conversely, villus \\nfibroblasts and endothelial cells were most abundant in the duodenum \\nand jejunum, less abundant in the Ileum and least abundant in the colon.\\nAs epithelial cells initially clustered on the basis of location (Extended \\nData Fig.\\xa06c), we subclustered and annotated epithelial cells from different \\nprimary locations—duodenum, jejunum, ileum and colon—separately.  \\nIn each intestinal region, we observed a differentiation trajectory from \\nstem cells to mature absorptive cells (enterocytes). We divided this \\na SB\\nCell typeNeighbourhoodCommunity\\nCL\\nTissue units\\nCommunities\\nTissue unit\\nCell types\\nb\\nc\\nd\\nCommunity\\nTissueu nit\\nPercentage of\\ncategory\\nPercentage of\\nnext level\\nTissuel evel\\nCell type\\nNeighbourhood\\ne\\nMuscularism ucosa\\nPaneth enriched\\nAdaptive immune enriched\\nPaneth\\nCommunities\\nNeighbourhoods\\nCell types\\nTissue units\\nf g\\nPercentage of combination\\nCL community\\nspatial context map\\nCL community\\nCloser to lumen\\nNeighbourhoods\\nh\\nImmune cell\\ntypes \\nOrdered communities\\nPercentage of cell type in community\\n02 04 06 0\\nCD8+ T\\nCD4+ T cell \\nM2 macrophage \\nPlasma\\nB\\nDC\\nM1 macrophage \\nNeutrophil\\nCD7+ immune\\nNK\\nB\\nCD4+ T cell \\nCD57+ enterocyte \\nCD66+ enterocyte \\nCD7+ immune\\nCD8+ T\\nCycling TA\\nDC\\nEndothelial \\nEnterocyte\\nGoblet\\nICC\\nLymphatic\\nM1 macrophage\\nM2 macrophage\\nMUC1+ enterocyte \\nNerve \\nNK\\nNeuroendocrine \\nNeutrophil\\nPlasma\\nPaneth\\nSmooth muscle \\nStroma\\nTA\\nAdaptive immune enriched \\nCD66+ mature epithelial\\nCD8+ T enriched IEL\\nGlandular epithelial\\nInnate immune enriched\\nInner follicle\\nInnervated smooth muscle \\nInnervated stroma\\nMacrovasculature\\nMicrovasculature\\nMature epithelial \\nOuter follicle\\nPaneth enriched\\nPlasma cell enriched\\nSecretory epithelial\\nSmooth muscle\\nSmooth muscle and innate immune \\nStroma\\nStroma and innate immune \\nTransit-amplifying zone\\nAdaptive immune enriched \\nCD66+ mature epithelial\\nCD8+ T enriched IEL\\nInnate immune enriched \\nFollicle\\nMature epithelial\\nPlasma cell enriched \\nSecretory epithelial\\nSmooth muscle\\nStroma\\nMucosa \\nMuscularis externa \\nMuscularis mucosa\\nSubmucosa\\nFig. 3 | Multilevel hierarchical structural description of the small intestine \\nand colon. a–d, Representation of multiple levels of hierarchical description: \\ncell type (a), multicellular neighbourhood ( b), community (based on clustering \\nwindows of cell neighbourhoods) ( c) and tissue units (based on clustering \\ncommunities) ( d) comparing the small bowel with the colon for two \\nrepresentative tissue sections (from a total of 64 sections from 8 donors).  \\nScale bar, 1\\u2009mm. e, Tissue hierarchy graph of the multilevel network of the tissue \\ncomprised of the different structures. Shapes correspond to structural level \\n(cell type, neighbourhoods, communities, tissue units); colours represent \\nindividual categories as indicated in a–d; the size of shapes represents the \\npercentage of tissue; and the size of connected lines represents the overall \\ncontribution to the next level of the structure when moving down the graph in \\nincreasing tissue structural hierarchy. The black rectangles highlight a single \\ntrajectory highlighted within this Article. The red bracket indicates separation \\nof stromal tissue units from the mucosal tissue units. f, Magnified mucosal area \\nof a colon community map shown within c . Scale bar, 100\\u2009µm. g, The spatial- \\ncontext maps of the colon highlighting relationships of communities across \\nthe entire sample. This structure is defined by the number of unique \\ncommunities required to make up at least 85% in a given window. The circles \\nrepresent the number of cells represented by a given structure. The green \\nrectangle highlights a structure discussed in this Article and maps this \\nstructure back to g. The colours are as indicated in c. h, The cell type percentage \\nof immune cells shown for each community ordered in relative order of general \\nincreasing proximity to the lumen on the basis of community spatial-context \\nanalysis.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0a45f453-602f-4560-a0f6-493d0d514f72', embedding=None, metadata={'page_label': '578', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='578 | Nature | Vol 619 | 20 July 2023\\nArticle\\ndifferentiation trajectory into five cell types using similar annotations to \\nother studies (stem\\u2009>\\u2009TA2\\u2009>\\u2009TA1\\u2009>\\u2009immature enterocyte\\u2009>\\u2009enterocyte)28. \\nHowever, these cells exist along a continuum and the exact number of \\ncell types and locations of the divisions between cell types is therefore \\narbitrary, and changing the resolution during clustering results in more \\nor fewer clusters along this trajectory. In addition to absorptive cells, \\nwe also observed goblet cells, tuft cells and enteroendocrine cells in all \\nregions of the intestine, while Paneth cells were present only in regions of \\nthe small intestine. Consistent with recent reports of BEST4+ enterocytes \\nin the small and large intestine6,32, we observed populations of BEST4+ \\nenterocytes in all eight intestinal regions assayed.\\nT o further examine the diversity of enteroendocrine cells along the \\nintestine, we subclustered enteroendocrine cells from all intestinal \\nregions and annotated the clusters based on expression of enteroen-\\ndocrine marker genes (Extended Data Fig.\\xa06g,h). We identified many \\nknown subtypes of enteroendocrine cells, including D cells (SSThigh),  \\nI cells (CCKhigh), K cells (GIPhigh), Mo cells (MLNhigh), S cells (SCThigh), L cells  \\n(GCGhigh and PYY high) and enterochromaffin cells (TPH1high). Two clus-\\nters of enteroendocrine cells that did not express any of these specific \\nmarkers were labelled as enteroendocrine Un1 and enteroendocrine \\nUn2 in the subclustered dataset. L cells can be further divided on the \\nbasis of expression of INSL5+, which is primarily expressed by L cells in \\nthe colon. T o determine which gut cell types the INSL5+ L cells may be \\nsignalling, we examined the expression of RXFP4—the cognate recep-\\ntor for INSL533,34. We found that RXFP4 is primarily expressed by colon \\nenterochromaffin cells and, to a lesser extent, I cells, suggesting that \\nthese are the most likely cell types that the INSL5+ L cells are signalling \\n(Fig.\\xa04e). Enteroendocrine and enterochromaffin cells were more abun-\\ndant in the duodenum and jejunum compared with in the colon, and \\nwe observed shifts in the fraction of each subtype of enteroendocrine \\ncells along the intestine (Extended Data Fig.\\xa07). For example, D cells \\nwere most abundant in the duodenum35. These results provide detail \\non how populations of enteroendocrine and enterochromaffin cells \\nchange along the length of the intestine.\\nFinally, the duodenum also contained an additional cluster of cells \\nthat expressed secretory markers, but did not cluster with goblet, \\ntuft, enteroendocrine or Paneth cells. We subclustered these cells \\n(Fig.\\xa04g) and identified one cluster with high expression of MUC5B , \\none cluster with high expression of MUC6  and TFF2, and one clus -\\nter with high expression of the exocrine markers CELA3B and CPB1  \\n(Fig.\\xa04h). The MUC5B+ and MUC6+ cells are probably different types of \\nmucin-producing cells, with the MUC6+ cells probably representing the \\nsnRNAsnRNA\\nDuodenum Duodenum\\nIleum\\nIleum\\nMid-jejunum\\nJejunum\\nAscending\\nTransverse\\nColon\\nDescending\\nSigmoid\\nProximal jejunum\\nUMAP dimension 2 (snRNA)\\n8 sectionsa d\\ng\\ni\\nh\\nef\\nUMAP dimension 1U MAP dimension 1UMAP dimension 1\\nCrypt /f_ib. 2\\nCrypt /f_ib. 3\\nEndothelial\\nSM/Myo/f_ib. 1\\nSM/Myo/f_ib. 2\\nSM/Myo/f_ib. 3\\nPericytes\\nbc\\nBEST4+ enterocytes\\nCycling TA1\\nCycling TA2\\nEnterocytes\\nEnteroendocrine\\nEnterocytes 2\\nGoblet\\nImmature enterocytes\\nPaneth\\nSpecialized secretory\\nStem\\nTA1\\nTA2\\nTuft\\nImmature goblet\\nsnATAC\\nsnRNA\\nIntegration\\n(CCA)3 donors\\n6 donors Multiome\\n(snATAC and snRNA)\\nUnknown\\nAverage expression Percentage expressed\\nMUC6+\\nExocrine\\nExocrine\\nMUC5B+\\nMUC6TFF2\\nMUC5BCELA3B\\nCPB1\\nPercentage\\nexpressed\\n25\\n50\\n75 –0.5\\n0\\n0.5\\n1.0\\nAverage\\nexpression\\nB cells\\nCD4\\nCD8\\nCycling\\nimmuneDC\\nILC\\nMonocytes/\\nmacrophages Mast\\nNK\\nPlasma\\nT cells\\nCI\\nRXFP3\\nRXFP4\\nINSL5\\n10 20\\n012\\nDJC ID JC ID JC ID JC ID JC ID J\\nMUC6+\\nMUC5B+\\nNeurons\\nGlia\\nICC\\nLymph. endo. cells\\nEndothelial, CD36+\\nEndothelial, venules\\nPericytes\\nMyo/f_ib./SM DEShigh\\nMyo/f_ib./SM 1\\nMyo/f_ib./SM 2\\nMyo/f_ib./SM 3\\nVillus /f_ib. WNT5B+\\nCrypt /f_ib. 1 WNT2B+\\nCrypt /f_ib. 2\\nCrypt /f_ib. 3 RSPO3+\\nMixed\\nCD4\\nCD8\\nT cells\\nNK\\nILC\\nB cells\\nPlasma\\nCycling immune\\nMast\\nDC\\nMono/macrophages\\nMixed\\n–5.0 –2.5 0 2.5\\nlog[fold change] \\nEnriched in\\ncolon \\nEnriched in\\nsmall intestine\\n–4 –2 02 46 j\\nHoechst\\nMUC6\\nMUC1\\nCytokeratin /uni03B1-SMA\\nVimentin\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\n1.2\\nPercentage of\\nMUC6+ enterocytes\\nDuodenum\\nIleum\\nMid-jejunum\\nAscending\\nTransverse\\nDescending\\nSigmoid\\nProximal jejunum\\nEnterochrom\\n.\\nEnteroendo.\\nUn1 \\nEnteroendo.\\nUn2 I cells L cells\\nNEUROG3\\nhigh\\nFig. 4 | Single-cell atlas of gene expression and chromatin accessibility  \\nin the human intestine.  a, Sections of the intestine analysed by separate \\nsnRNA-seq and snATAC-seq experiments or multiome experiments. b ,c, UMAP \\nrepresentation of all snRNA stromal ( b) and immune (c) cells coloured by cell \\ntype. DC, dendritic cells; fib., fibroblasts; ILC, innate lymphoid cells; myofib., \\nmyofibroblasts; SM, smooth muscle cells. d, UMAP representation of snRNA \\nepithelial cells in the four primary regions of the intestine. Jejunum includes \\nboth proximal- and mid-jejunum samples. Colon includes samples from the \\nascending, transverse, descending and sigmoid colon. e, Expression of INSL5 \\nand the INSL5 receptor, RXFP4, in different cell types in different regions of the \\nintestine. RXFP4 was expressed in less than 2.5% of all epithelial cell types that \\nwere not included in the dot plot. C, colon; D, duodenum; I, ileum; J, jejunum; \\nenterochrom., enterochromaffin cells; enteroendo., enteroendocrine cells.  \\nf, Beeswarm plot showing the log-transformed fold change between the small \\nintestine and colon for groups of nearest-neighbour cells from different cell \\ntype clusters. Significant changes are indicated in red and blue. Lymph. endo., \\nlymphatic endothelial cells. g, Subclustering of specialized secretory cells in  \\nd coloured by cell type. h , The expression of secretory genes in specialized \\nsecretory cells defined in g . i, The percentage of MUC6 + enterocytes among all \\ncell types for the four donors imaged using MUC6 antibodies. j , Representative \\nCODEX fluorescence image of the duodenum (6 out of 57 markers overlaid) \\n(left). Hoechst (nuclei), MUC6, MUC1 (also found in gland areas), cytokeratin \\n(pan-epithelial), α-SMA (muscle) and vimentin (stromal) staining is shown. \\nRight, the magnified area highlights the gland just below the mucosa in the \\nsubmucosa. This experiment was independently repeated four times.  \\nScale bars, 500\\u2009µm (left) and 50\\u2009µm (right).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c62cadf3-5f73-496d-bcba-401ee1d3d0d8', embedding=None, metadata={'page_label': '579', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 579\\ncells of the Brunner’s glands36. Notably, the MUC5B+ cells and exocrine \\ncells were primarily present in only one sample in our dataset, making \\nit possible that they are contaminating cells from a different tissue,  \\nin which expression of these markers is more common, or a rare cell \\ntype in the duodenum (Extended Data Fig.\\xa06i). The MUC6+ cells were \\npresent in the majority of the samples, so we further validated their \\npresence by labelling cells in CODEX experiments with MUC6 (Fig.\\xa04i,j).\\nWithin the CODEX data, we made the distinction between CD66 + \\nenterocytes and CD57+ enterocytes. When we examined the expres-\\nsion of these markers in the snRNA data, we found that CD66+ entero-\\ncytes typically represent mature cell types in the colon (Extended Data \\nFig.\\xa08a,b). CD57+ cells are much less abundant and CD57 expression \\nwas primarily observed only at low levels in the cluster of MUC6+ cells, \\nconsistent with the presence of CD57+ cells in glands in the duodenum \\nin the CODEX data (Extended Data Fig.\\xa08a,b).\\nMolecular interactions in nearby cells\\nThe CODEX data enable the assignment of neighbouring cell types and \\nthe snRNA-seq data provide RNA expression levels in the corresponding \\ncell types. By combining both data types, we can nominate potential \\nligand–receptor pairs that may facilitate cell-to-cell interactions. We \\nidentified significant pairwise cell type colocalizations and focused \\non those that were significantly different between the small bowel and \\ncolon (Extended Data Fig.\\xa08c and Supplementary Table\\xa02). Cell type pairs \\ninvolving plasma cells are more colocalized in the descending-sigmoid \\ncolon tissue section than in other sections (Extended Data Fig.\\xa08e).\\nUsing snRNA-seq data from these six cell types, we performed dif-\\nferential expression analyses of ligands and receptors (Methods) and \\nidentified 48 pairs of ligands and receptors (across 41 cell–cell pairs) \\nthat are more expressed in the colon than in the small bowel (Supple-\\nmentary Table\\xa03). For example, we found that the ligand SEMA4D and \\nreceptor MET were upregulated in plasma cells and TA2, respectively, \\nin colon tissue (Extended Data Fig.\\xa08d), which was not observed in \\nthe small intestine (Supplementary Table\\xa03). SEMA4D signalling has \\nbeen associated with B cell aggregation and long-term survival37. The \\nimplication of the MET receptor on TA2 cells in the colon, compared \\nwith TA2 cells in the small intestine, is further evidenced by RAS and \\nMAPK signalling (Extended Data Fig.\\xa09b, Supplementary Table\\xa04) and \\nupregulation of plexins (Extended Data Fig.\\xa09b), consistent with the \\nCODEX imaging data (Fig.\\xa03f,g and Extended Data Figs.\\xa05a–f and 9c). \\nUsing spatial transcriptomics analysis based on Molecular Cartogra-\\nphy38–40 (Methods), we validated that TA cells positive for MET and \\nplasma cells positive for SEMA4D were more colocalized in the colon \\n(colocalization quotient (CLQ)\\u2009=\\u20091.57). We also validated nine other \\nreceptor–ligand interactions involving plasma cells that were more \\ncolocalized in the colon (Supplementary Table\\xa05). Consequently, this \\nindicates a potential differential survival signal that maintains the \\nPlasma-Cell-Enriched neighbourhood in the colon.\\nIn addition to colocalization of receptor–ligand pairs, we probed our \\nspatial transcriptomics dataset and evaluated whether other receptor– \\nligand pairs were more expressed by target cells in the colon than in \\nthe small bowel. We validated 15.3% of our predictions (9 out of 58; \\nP\\u2009<\\u20090.002; Methods) that have a matching cell type with snRNA-seq \\n(Methods and Supplementary Table\\xa06). One other example that we \\nexamined was the expression of ligand FN1 (fibronectin) in myofibro-\\nblasts and its receptor PLAUR (urokinase receptor) in enterocytes \\n(Extended Data Fig.\\xa09d). Overall, these results nominate potential \\nligand–receptor interactions that mediate specific cell type interac-\\ntions in distinct regions of the intestine.\\nIntegration of CODEX and snRNA data\\nOur multiplexed imaging analysis also enables examination of the cell–\\ncell pairs that are enriched within each neighbourhood or community \\nregardless of tissue location. We used the same approach to calcu-\\nlate cell–cell colocalization in the Follicle community. This resulted \\nin 57 colocalized cell pairs, such as the CD8+ T\\xa0cell–CD4+ T\\xa0cell pair, \\neach pair containing a large list of potential receptor–ligand inter -\\nactions (Extended Data Fig.\\xa09e). T o more comprehensively describe \\npotential interactions and expressions within neighbourhoods, we \\nintegrated CODEX and snRNA-seq at the single-cell level using Max-\\nFuse41, a method that we specifically designed for these challenging \\nintegration tasks: linear assignment coupled with graph smoothing \\nand meta cell construction. The overall matching accuracy was 92.1% \\nand cell types segregated within the co-embedding space, and the \\nsame cell types from snRNA-seq and CODEX were well blended while \\nshowing concordant RNA/protein expression patterns, indicating \\nrobust integration performance (Fig.\\xa05a). Using this integration, we \\ncalculated the differentially expressed genes (DEGs) among the pre-\\nviously CODEX-defined cellular neighbourhoods with the pair-linked \\ntranscriptome information (Extended Data Fig.\\xa09f). In particular, we \\nexamined the expression levels of DEGs involved with the follicle neigh-\\nbourhoods (Fig.\\xa05a (bottom UMAP)) and identified gene pathways that \\nare enriched in these spatial organizations, including B-cell-receptor \\nsignalling, T-cell-polarity regulation and tolerance induction of self \\nantigen pathways (Extended Data Fig.\\xa09g). Thus, integrating CODEX \\nand snRNA-seq data revealed specific expression patterns associated \\nwith distinct cellular neighbourhoods.\\nRegulatory TFs in cells of the intestine\\nT o obtain insights into the factors that control intestinal differentia-\\ntion, we next investigated potential transcription factors (TFs) that \\nregulate gene expression in intestinal cell types. We first computed \\nChromVar deviation scores42 for each cell in our dataset, identifying \\nTF motifs that are associated with chromatin accessibility in different \\ncell types. As many TFs share similar motifs, examining TF expression \\nin conjunction with motif activity helps to identify the specific TFs \\nthat are functional in different intestinal cell types. We next identi-\\nfied the TFs with the highest correlation between their gene expres-\\nsion and the chromatin-accessibility activity level of their putative \\nDNA-binding motifs43 to nominate TFs that directly drive accessibility \\nchanges. Across the intestine, this analysis revealed 61 TFs with motif \\nactivity that was strongly correlated with expression (r\\u2009>\\u20090.5; Fig.\\xa05b). \\nBroadly, we observed TFs that are active in the secretory lineage and \\nTFs that are active in the absorptive lineage with few TFs that partici-\\npate in both (for example, KLF4 in colon enterocytes and goblet cells \\nthroughout the intestine; Fig.\\xa05b).\\nThis analysis highlights many TFs that are known to have impor -\\ntant roles in the intestine. For example, ASCL2, a master regulator \\nof intestinal stem cells44, exhibited high expression and motif acces-\\nsibility in stem cells. Other TFs with high expression and motif activ -\\nity in stem cells include NFIX, NFIC and HNF1B. Within the secretory \\nlineage, POU2F3, a regulator necessary for the development of tuft \\ncells in mice 45, was highly expressed and had high motif activity in \\ntuft cells throughout the human intestine. When comparing the TF \\nfootprinting signal around POU2F3 motifs in tuft cells compared with \\nenterocytes, we observed greater accessibility flanking POU2F3 motifs \\nin tuft cells, suggesting that POU2F3 is more likely to be bound to \\nits motif in tuft cells (Extended Data Fig.\\xa010d). Along with POU2F3, \\nRUNX1 and RUNX2 also exhibited high expression and accessibility \\nin tuft cells throughout the intestine, indicating that they may also \\nhave a role in tuft cell differentiation or maintenance. Among goblet \\ncells, KLF4, which is required for terminal differentiation into colonic \\ngoblet cells in mice46, exhibited high gene expression and motif activ-\\nity. Notably, the expression and motif activity of KLF4 was also high in \\ndifferentiated absorptive epithelial cells (immature enterocytes and \\nenterocytes) in the colon, but not in other regions of the intestine, \\nindicating location-specific regulation.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='278398a2-06c4-424f-85aa-a6b0330dfadf', embedding=None, metadata={'page_label': '580', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='580 | Nature | Vol 619 | 20 July 2023\\nArticle\\nSeveral regulatory TFs were nominated in goblet cells, including \\nATOH1 and FOXA3. ATOH1 is necessary for secretory lineage commit-\\nment in the mouse intestine47,48, and here we provide data supporting \\na similar function in human goblet cells. T o provide further evidence \\nthat ATOH1 drives accessibility in goblet cells, we compared ATOH1 \\nfootprints in goblet cells and enterocytes (Extended Data Fig.\\xa010e). \\nIndeed, we found greater flanking accessibility around ATOH1 motifs \\nin goblet cells, consistent with greater ATOH1 binding in these cells. For \\nFOXA3, there is evidence that FOXA3 leads to goblet cell metaplasia in \\nthe lungs49, and our findings indicate that it probably is an important \\nregulatory TF in the colon. We noted above that goblet cells are more \\nabundant in the colon compared with in the small intestine, but it is \\nunclear what biases cells to differentiate into goblet cells with greater \\nfrequency in the colon. Possible causes include differences in signalling \\nin the crypts versus differences in stem cells between the regions. We \\nidentified motifs that are enriched in differential peaks between small \\nintestine and colon stem cells and found that FOX motifs are enriched \\nin peaks that are more accessible in colon stem cells (Extended Data \\nFig.\\xa010a–c). Similarly, FOX TFs have greater motif deviation scores in \\ncolon stem cells compared with in small intestine stem cells (Extended \\nData Fig.\\xa010a–c). As we nominate several FOX TFs as potential regula-\\ntors of goblet cells, increased FOX activity may partially explain why \\ngoblet cells are more abundant in the colon.\\nWithin enteroendocrine/enterochromaffin cells, RFX2, RFX3 and \\nRFX6 exhibited high expression and accessibility. Of these, RFX6 is a \\nproposed regulator of enteroendocrine cell differentiation, and loss of \\nRFX6 impairs enteroendocrine cell differentiation in mice50. T ogether, \\nthese results support previous findings and nominate additional TFs \\nthat may be important regulators of distinct intestinal cell types that \\ncan vary across the different regions of the intestine (for example, \\nKLF4).\\nT o help to validate these findings, we completed this analysis in the \\ntwo separate cohorts: the multiome samples and the samples with \\nseparate snRNA and snATAC data. T o examine TF expression in the \\nlatter group, we integrated the snRNA and snATAC data using canoni-\\ncal correlation analysis to align the datasets and assign snRNA data to \\neach snATAC cell51,52 (Methods). This analysis reproduced many of the \\nfindings in the multiome analysis, with 48 out of the 61 TFs originally \\nidentified in the multiome analysis reaching the same significance \\ncriteria (Extended Data Fig.\\xa010f).\\nT o determine which TFs may drive cell function in stromal and \\nimmune cells in the intestine, we performed the same analysis for cells \\nin each of these compartments. Within the immune compartment, this \\nanalysis highlighted many TFs that are known to be important for cell \\ntype differentiation and maintenance, including GATA2 in mast cells53 \\nand PAX5 in B cells54 (Extended Data Fig.\\xa010g). Among stromal cells, we \\nidentified TFs associated within specific lineages, including EBF1 in \\npericytes—which was recently suggested to contribute to pericyte cell \\ncommitment55—PPARG in adipocytes56 and SOX10 in glia57 (Extended \\nData Fig.\\xa010h). We also nominate potential regulatory TFs in interstitial \\ncells of Cajal, including HAND1, HOXD11 and MEIS1, as well as potential \\nregulatory TFs for different classes of intestinal fibroblasts.\\n–2 2\\nAbsorptive cellsSecretory cells\\n–2 2TF motif activity row z-score: RNA expression row z-score:\\nEE Tuft\\nAbsorptive cellsSecretory cells\\nGobletPrimarily abs.Primarily abs. StemGoblet\\n61 TFs\\nNFIC\\nNFIX\\nHNF1A\\nHNF1B\\nASCL2\\nE2F3\\nFOXP2\\nFOXK1\\nETS1\\nRUNX3\\nPOU2F1\\nPOU2F3\\nSOX9\\nRUNX2\\nRUNX1\\nFOXA1\\nFOXD2\\nFOXA2\\nFOXP1\\nETS2\\nMYB\\nEHF\\nHOXA10\\nHOXB13\\nHOXA11\\nHOXA13\\nATOH1\\nFOXA3\\nXBP1\\nSPDEF\\nNEUROD1\\nRFX2\\nRFX3\\nRFX6\\nONECUT2\\nPDX1\\nNR2F6\\nNR1H3\\nRXRA\\nESRRA\\nPPARG\\nHNF4A\\nVDR\\nGATA6\\nGATA5\\nHNF4G\\nGATA4\\nMEF2D\\nMEF2A\\nKLF4\\nJUN\\nATF3\\nJUNB\\nFOS\\nJUND\\nBACH1\\nJDP2\\nPPARA\\nNR1I2\\nMAF\\nNR1H4\\nDJ IC DJ IC DJ IDJ CD JI CDJ ICDJ ICDJ ICDJ ICDJ IC CDJ ICDDJ ID JI CD JI C DJ IDJ CD JI CDJ ICDJ ICDJ ICDJ ICDJ IC CDJ ICDDJ I\\nba\\nSmall bowelColon\\nsnRNA-seqCODEXCell type\\nMUC2 (RNA)MUC2 (protein)\\nB cells\\nEndothelial\\nEnteroendocrine\\nGoblet\\nMono/macrophages\\nPlasma\\nSmooth muscle\\nStroma\\nT cells\\nCN:inner \\nfollicle DEGs\\nMin. Max.Min. Max.Min. Max.\\nGoblet\\nImmature goblet\\nEnterochromaf/f_in\\nEnteroendocrine U n\\nEnteroendocrine Un 1D cells I cells\\nL cells\\nMo cells\\nNEUROG3 high\\nS cells\\nSec. Spec. MUC6 +\\nPaneth\\nTuft\\nStem\\nCycling TA1\\nCycling TA2\\nTA2\\nTA1\\nImmature ent.\\nEnt.\\nBEST4 +  ent.\\nEnterocytes 2\\nK cells\\nGoblet\\nImmature goblet\\nEnterochromaf/f_in\\nEnteroendocrine Un\\nEnteroendocrine Un 1\\nD cells I cells\\nL cells\\nMo cells\\nNEUROG3 high\\nS cells\\nSec. Spec. MUC6 +\\nPaneth\\nTuft\\nStem\\nCycling TA1\\nCycling TA2\\nTA2\\nTA1\\nImmature ent.\\nEnt.\\nBEST4 +  ent.\\nEnterocytes 2\\nK cells\\nFig. 5 | Integration of CODEX multiplexed imaging, snRNA-seq and snATAC–  \\nseq reveals genes associated with cellular neighbourhoods and regulatory \\nTFs in the human intestine. a, snRNA-seq and CODEX integration using \\nMaxFuse for both the small bowel and colon. Integrated cells are coloured by \\nmodality of origin, cell types or RNA/protein expression levels. MUC2 expression \\n(protein and RNA) is shown on the integrated plots, as well as the top 10 DEGs of \\nthe Inner Follicle neighbourhood as determined by CODEX. Min., minimum.\\xa0 \\nCN, cellular neighbourhood. b, TFs of which the integrated gene expression \\ncorrelated with their motif activity in one region of the intestine. Row z-scores \\nof ChromVar deviation scores are shown on the left and row z- scores of \\nintegrated TF expression are shown on the right. Ent., enterocytes.\\xa0abs., \\nabsorptive.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='cc9d6b4d-5e0a-4aa5-b444-495d93d45b94', embedding=None, metadata={'page_label': '581', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 581\\nAbsorptive differentiation trajectories\\nIntestinal stem cells differentiate into mature enterocytes, goblet cells \\nand specialized cell types such as enteroendocrine, tuft and Paneth \\ncells, renewing the epithelial lining approximately every three to seven \\ndays58. T o map the regulatory and gene expression changes that accom-\\npany stem cell differentiation into mature enterocytes, we defined dif-\\nferentiation trajectories along this pathway in the single-nucleus data \\nfrom the duodenum, jejunum, ileum and colon (Fig.\\xa06a and Methods). \\nWe next identified regions of variable chromatin accessibility (peaks) \\nand variable gene expression across these four differentiation trajec-\\ntories. An example of the variable genes is TMPRSS15, which encodes \\nthe protein that converts trypsinogen to trypsin in the duodenum; \\nits expression gradually increases in more differentiated cells in the \\nduodenum (Extended Data Fig.\\xa010m–p). We clustered these variable \\npeaks and genes to identify sets with shared behaviour (Fig.\\xa06b,c). The \\nresulting clusters include peaks and genes that are open and expressed \\nearly in the differentiation pseudotime (for example, in stem cells) \\nin all regions of the intestine, which we denote as early. This cluster \\nincludes general markers of intestinal stem cells, including RGMB, \\nSOX9, SMOC2 and LGR5. Other clusters of genes and peaks include \\nthose that are predominantly found in differentiated small intestine \\ncells (for example, MTTP, APOA4, APOC3, MME), in undifferentiated \\ncolon (for example, GPX2), in ileum (such as PLB1, CUBN) and in dif-\\nferentiated colon (such as SCNN1B59, SLC26A3, CLCA4).\\nT o identify both chromatin drivers of cluster-specific regulation\\xa0and \\nrelevant cluster-specific gene expression programs, we computed \\nTF-motif enrichments in each cluster of peaks (Extended Data Fig.\\xa010k) \\nand KEGG pathway enrichment in each cluster of genes (Extended Data \\nFig.\\xa010l). Groups of peaks accessible late in the differentiation trajecto-\\nries were enriched for HNF4 and JUN/FOS motifs. As expected, genes \\nprimarily expressed late in the differentiation trajectory in the small \\nintestine (late duodenum jejunum cluster) were enriched for multiple \\nmetabolic KEGG pathways including fat digestion and absorption and \\nvitamin digestion and absorption.\\nWe next identified TFs of which the gene expression is correlated \\nwith the activity of their motifs in each of the four differentiation tra-\\njectories. We found 67 TFs with expression correlated with their motif \\nactivity (r\\u2009>\\u20090.5) and plotted the integrated gene expression of these TFs \\n(Fig.\\xa06d). Many TFs display similar activity along all four differentiation \\ntrajectories. For example, ASCL2, a master regulator in intestinal stem \\ncells, is highly expressed at the beginning of all four trajectories. Other \\nTFs, such as ETV6, exhibit different behaviours in different regions of \\nthe intestine. ETV6 is a TF with decreased expression in colorectal can-\\ncer compared with in the normal colon, and genetic variation in ETV6 \\nmay confer colorectal cancer susceptibility60. We found that ETV6 is \\n40,311 peaks: accessibility \\n–2 2\\n5,462 genes: expression\\n–1.75 1.75\\nCIJDC IJD\\nba e\\nf\\nd\\nc\\nLate colon Late colon\\nEarly\\nEarly colon\\nEarly\\ncolon\\nEarly\\nLate\\nIleum\\nLate small\\nbowel\\nLate small\\nintestine\\nEarly\\nduodenum,\\njejunum\\nLate\\nduodenum,\\njejunum\\nUMAP dimension 2 (snATAC)\\nUMAP dimension 1\\nEnterocytes\\nEnterocytes\\nEnterocytes\\nEnterocytes\\nGoblet\\nGoblet\\nStem\\nStem\\nStem\\nStem\\nRow z-scoreRow z-score\\nDifferentiation\\npseudotimeCell types\\n0\\nMax.\\nImputed gene expression\\nAccessibility (each peak norm. to 1) \\n3\\n4\\n4\\n5\\n4\\n6\\nETV6\\n8\\n12\\nExonic 1\\nPromoter 1\\nIntronic 1\\nIntronic 2\\nIntronic 3\\nIntronic 4\\n67 TFs\\n–2\\n2 Row z-score\\nPseudotime Pseudotime\\nLinked peaksColon Ileum Jejunum Duodenum0\\n1\\n0\\n1\\n0\\n1\\n0\\n1\\n–log10[P]\\nLGR5\\nSMOC2\\nAPOA4\\nAPOC3\\nMME\\nMTTP\\nS OX9\\nRGMB\\nGPX2\\nCUBN\\nSLC26A3\\nCLCA4\\nPLB1\\nFOXP2\\nTFCP2\\nE2F3\\nNFIX\\nSOX9\\nTFDP1\\nRFX3\\nE2F8\\nNRF1\\nE2F7\\nFOXM1\\nFOXK1\\nPOU5F1B\\nNFIC\\nHNF1B\\nFOXP1\\nMYBL2\\nFOXK2\\nMYB\\nHNF1A\\nGATA6\\nETS2\\nASCL2\\nFOXA1\\nERG\\nEHF\\nRBPJ\\nLEF1 NFIA\\nRREB1\\nVDR\\nETV6\\nHNF4A\\nNFATC2\\nRARB\\nCEBPD\\nHNF4G\\nNFATC3\\nNR1H4\\nTFEC\\nTHRB\\nNRL\\nMAF\\nNR3C1\\nPPARG\\nSREBF1\\nJUNB\\nNR1D2\\nBACH1\\nMEF2A\\nJUND\\nATF3\\nCEBPG\\nRXRA\\nJUN\\nNR3C2\\nKLF4\\nJDP2\\nNR1D1\\nPPARA\\nKLF9\\nUSF1\\nNR1I2\\nNR2F6\\nFOS\\nSMAD3\\nMEF2D\\n*\\n*\\n*\\n*\\n*\\n*\\n*\\nEntero endocrine\\nSec. spec. MUC6 +\\nMast\\nICC\\nTu ft\\nGlia\\nPaneth\\nImmatur e goblet\\nCD4\\nStem\\nPlasma\\nNK\\nEntero cytes 2\\nGoblet\\nBEST4 +  entero cytes\\nCD8\\nPericytes\\nMyo/f_ibr oblasts/SM\\nB cells\\nAbsorptive epithelial\\nCycling immune\\nEndothelial\\nCycling TA  \\nMono/macr ophages\\nVillus /f_ib. WNT5B +\\nCrypt /f_ibr oblasts\\nBMI\\nUC\\nCD\\nCoeliac\\nHTN\\nDental caries\\n1\\n2\\n3\\n4\\nC\\nI\\nJ\\nD\\nFig. 6 | Regulation of differentiation in the human intestine. a, UMAP \\nprojections depicting the cells in the four primary regions of the intestine \\n(duodenum, jejunum, ileum and colon), labelled by cell type (left) and \\ndifferentiation pseudotime (right). b,c, Variable peaks ( b) and genes  \\n(c) identified along the absorptive differentiation trajectories. The rows \\nrepresent the z -scores of accessibility for each peak or expression for each \\ngene. The columns represent the position in pseudotime from the start to the \\nend for each section of the intestine. Peaks and genes were k-means-clustered \\nand the clusters were labelled on the basis of the dominant time and location \\nwhere they are most accessible/expressed. d, Integrated gene expression of \\nTFs of which the expression is correlated with ChromVar motif activity along \\nthe differentiation trajectory. e , Accessibility at peaks correlated with the \\nexpression of ETV6  along the differentiation trajectory in each region (left). \\nEach peak is normalized to the maximum accessibility along any of the \\ntrajectories. Right, integrated gene expression of ETV6 along the differentiation \\ntrajectory in each region is plotted on the right. Norm., normalized. f , Linkage-  \\ndisequilibrium score regression to identify the enrichment of GWAS SNPs in \\ncell-type-specific marker peaks. Unadjusted coefficient P values computed \\nfrom linkage-disequilibrium score regression are plotted in the heat map. \\nSignificance is indicated by an asterisk, as determined by a Bonferroni- \\ncorrected coefficient P  value of <0.05. P  values for determining significance \\nwere adjusted for the number of cell classes tested.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ba59dc6e-c9b2-4b6d-9ed5-9234e62b76fb', embedding=None, metadata={'page_label': '582', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='582 | Nature | Vol 619 | 20 July 2023\\nArticle\\nmore highly expressed in the colon compared with in the small intestine \\nand, in contrast to in the small intestine, ETV6 expression increases in \\nmore differentiated cells in the colon.\\nAs an example of examining how genes with unique expression \\npatterns may be controlled, we next investigated which regulatory \\nelements may be responsible for the variable expression of ETV6 in \\ndifferent regions of the intestine. We identified accessibility peaks \\ncorrelated with ETV6 expression along the differentiation pseudo-\\ntime in any region of the intestine (Fig.\\xa06e). The correlated peaks most \\naccessible in the small intestine (exonic 1 and promoter 1) became less \\naccessible along the differentiation trajectory, consistent with the \\ndecreased expression along the differentiation pseudotime in these \\nregions. Similar behaviour was also observed in the colon, in which \\nthese peaks became less accessible along the differentiation trajec-\\ntory. However, multiple other peaks exhibited increasing accessibility \\nin more differentiated cells only in the colon, and we speculate that \\nthese regulatory elements may drive the increased expression of ETV6 \\nin differentiated colon cells. The same logic can be applied to iden -\\ntify regulatory elements that may drive changes in gene expression \\nthroughout the intestine. For example, we identified three peaks that \\nare highly correlated with expression of TMPRSS15 and may drive its \\nincreased expression in the duodenum (Extended Data Fig.\\xa010p). Taken \\ntogether, this analysis provides a reference for the regulation of stem \\ncell to enterocyte differentiation across the intestine.\\nWe next tested whether disease heritability is enriched in cell-type- \\nspecific marker peaks in intestine cell types using linkage-disequilibrium \\nscore regression (Fig.\\xa06f and Methods). We identified a significant \\nincrease in heritability for Crohn’s disease and coeliac disease in T\\xa0cell \\nmarker peaks, consistent with the importance of T\\xa0cells in their patho-\\ngenesis61. We observed the most significant enrichment of heritability \\nfor BMI in enteroendocrine cells, suggesting that genetic variation may \\nhave an effect on enteroendocrine cells leading to effects on BMI. As a \\ncontrol, we also tested whether heritability of GWAS SNPs was enriched \\nin an unrelated trait (dental caries) and found no cell-type-specific \\nenrichment. These results map important disease traits to specific \\ncell types in the intestine.\\nDiscussion\\nHere we show extensive cellular complexity of the intestine including \\nconsiderable epithelial heterogeneity and new secretory cell subtypes, \\nthat the different regions of the intestine have different cell composi-\\ntions, and that cells are organized into different neighbourhoods that \\nalso form communities defined by both distinct areas of epithelial and \\nimmune cells. Additionally, the open chromatin regulatory program \\ndefined key regulators and differentiation pathways used in the dif-\\nferent regions of the intestine. Our study greatly extends previous \\nsingle-cell studies by combining both spatial proteomic datasets and \\nsingle-cell RNA and ATAC technologies, and the resulting datasets serve \\nas a useful resource for the scientific community.\\nOur multiplexed imaging is the first in-depth spatial study of single\\xa0 \\ncells in the intestine, to our knowledge. Many of our spatial analyses \\nrevealed important immune cell type organization within the intestine \\n(Supplementary Discussion). We observed a correlation of M1 mac-\\nrophages with BMI and spatial restriction of macrophage subtypes17. \\nAlthough our donors did not have histories of gastrointestinal diseases, \\nan increase in M1 macrophages may potentially indicate an early stage \\nof gastrointestinal disease progression. We also observed that CD8+ \\nT\\xa0cells decreased from the small intestine to the colon primarily within \\nthe CD8+ T\\xa0Cell-Enriched IEL neighbourhood. Notably, we also found \\nthat CD8+ T\\xa0cells decrease in donors with a history of hypertension and \\nwithin the CD8+ T\\xa0Cell-Enriched IEL neighbourhood. Further investiga-\\ntion of T\\xa0cell levels within the gut in patients with hypertension may \\nprovide cell type mechanisms for associations of hypertension with \\nother diseases such as colon cancer62,63.\\nWe identified intestinal crypt multicellular neighbourhoods that \\nwere co-enriched with adaptive immune cells such as CD4+ and CD8+ \\nT\\xa0cells in the small bowel and colon. In the small bowel, stem cell crypt \\nareas were identified by Paneth-Cell-Enriched neighbourhoods, \\nwhereas, in the colon, the stem cell crypt was identified with increas-\\ning neuroendocrine cell density towards the bottom of the crypt. CD4+ \\nT\\xa0cells were also the most enriched cell type in the Outer Follicle multi-\\ncellular neighbourhood, which was present at all sites of the intestine \\nregardless of the presence of fully developed Inner Follicle structures \\nacross neighbourhoods. This observation suggests that the immune \\nsystem appears structurally poised along the intestine to generate \\ngerminal-centre-focused immune responses locally as needed. Inte-\\ngration of CODEX and snRNA-seq single-cell data revealed differential \\ngene expression across these cellular neighbourhoods and highlighted \\nimportant gene modules for the immune response in Inner Follicle \\nneighbourhood structures.\\nWe confirmed the adaptive immune and crypt association with hier-\\narchical spatial analysis and further observed discrete immune cell com-\\nposition zones across the intestine from an adaptive-immune-enriched \\narea at the base of the crypt, then a plasma-cell-enriched area in the \\nmiddle of the mucosa, with an innate-immune-enriched zone at \\nthe top. Indeed, plasma cells had the highest same-cell density and  \\nwere also found to co-localize with antigen-presenting cells in a  \\nmulticellular neighbourhood found in all areas of the intestine. Merg-\\ning the snRNA-seq data and CODEX data showed that plasma cells and \\ntransit-amplifying epithelial cells co-localized more in the colon than \\nin the small intestine, and this was validated by spatial transcriptom-\\nics. Restriction and layering of the intestine primarily has focused \\non epithelial subtypes previously, but we highlight that immune \\ncells share a spatial restriction and zonation. This hierarchical view \\nof multiplexed spatial data can serve as a template for other spatial  \\natlas efforts.\\nThis study extends the work of several scRNA gut atlases6,32. These \\nprevious scRNA datasets examined the diversity of cell types in the \\nhuman intestine, whereas the integrated snRNA and snATAC dataset \\nprovides a detailed single-cell regulatory map of the intestine. We used \\nthis dataset to identify TFs in different cell types of which the gene \\nexpression is highly correlated with the accessibility of the motif to \\nwhich they bind. This identified TFs that are known to be master regu-\\nlators of different cell types, including POU2F3 in tuft cells, ASCL2 in \\nstem cells and RFX6 in enteroendocrine cells, while also nominating \\nmany additional TFs that are probably important regulators in their \\nrespective cell types. This includes RUNX1 and RUNX2 in Tuft cells \\nand FOXA3 and ATOH1 in goblet cells. Finally, with the inclusion of two \\ndonors from under-represented backgrounds, our study also extends \\nsingle-cell analyses to include samples from such groups.\\nWithin all regions of the intestine, intestinal stem cells differentiate \\ninto mature absorptive enterocytes. By integrating gene expression \\nand chromatin accessibility data along the differentiation trajecto-\\nries in different regions of the intestine, we nominate TFs that exhibit \\nconsistent behaviour across absorptive differentiation in all regions \\nof the intestine. Among these are known intestinal stem cell regulators \\nsuch as SOX9 and ASCL2, which are highly expressed and have high \\nchromatin activity of their binding motifs in cells at the beginning of \\nthe differentiation trajectory in all regions of the intestine. We also \\nobserved differences in TF dynamics between the trajectories in differ-\\nent regions, such as ETV6, which is highly expressed in differentiated \\nabsorptive cells in the colon, but not other regions of the intestine. \\nNotably, ETV6 expression is decreased in colorectal cancer and genetic \\nvariation in ETV6 may confer colorectal cancer risk60. We speculate that \\nETV6 is important for normal colon differentiation and its loss may \\nprevent differentiation of colon stem cells. Examining these data also \\nenables us to link specific regulatory elements with the expression of \\nTFs along the trajectory. For the case of ETV6, this analysis identifies \\none distal and three intronic regulatory elements with similar activity ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a26d94a2-56af-4cce-9a83-c9ff408850a8', embedding=None, metadata={'page_label': '583', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 583\\nthat are probably responsible for driving expression of ETV6 along the \\nabsorptive differentiation trajectory in the colon.\\nT ogether, these data provide a detailed atlas that can serve as a \\nvaluable reference for future studies of the intestine. This includes \\ncontextualizing GWAS studies of intestinal disease, similar to the \\nanalysis performed above showing heritability of BMI is enriched in \\nenteroendocrine cells, and severing as a reference when comparing \\ndisease state such as cancer to the normal colon64.\\nThere are several limitations to our study. First, for each patient, we \\ntypically analysed a single sample from each intestinal region. Second, \\nCODEX is limited to 54 markers and does not capture the entire breadth \\nof known cell types within the intestine. Third, it is also important to \\nnote that, although there was variation of ages, all individuals were \\nabove the age of 24, limiting our analysis of the development of the \\nintestine in children. Also, six out of the nine adults analysed were male. \\nAlthough our study represents two ethnicities, the patterns across a \\nwide range of ethnicities remain to be elucidated. We are also under-\\npowered to ascertain sex differences, which will probably be important \\ngiven differences in disease risk for male and female individuals65. These \\nlimitations can be addressed in the future with the investigation of \\nmore samples.\\nIn summary, we present a detailed map of the human intestine, and \\nthe first multiplexed imaging reference for healthy small intestine and \\ncolon. In addition to biological insights, this can serve as an important \\nreference for intestinal diseases (such as inflammatory bowel disease) \\nas well as comparisons with other organisms.\\nOnline content\\nAny methods, additional references, Nature Portfolio reporting summa-\\nries, source data, extended data, supplementary information, acknowl-\\nedgements, peer review information; details of author contributions \\nand competing interests; and statements of data and code availability \\nare available at https://doi.org/10.1038/s41586-023-05915-x.\\n1. Boland, M. Human digestion—a processing perspective. J. Sci. Food Agric. 96, 2275–2283 \\n(2016).\\n2. Cronin, C. G., Delappe, E., Lohan, D. G., Roche, C. & Murphy, J. M. Normal small bowel wall \\ncharacteristics on MR enterography. Eur. J. Radiol. 75, 207–211 (2010).\\n3. Kabat, A. M., Srinivasan, N. & Maloy, K. J. Modulation of immune development and \\nfunction by intestinal microbiota. Trends Immunol. 35, 507–517 (2014).\\n4. Bowcutt, R. et\\xa0al. Heterogeneity across the murine small and large intestine. World J. \\nGastroenterol. 20, 15216–15232 (2014).\\n5. Gunawardene, A. R., Corfe, B. M. & Staton, C. A. Classification and functions of \\nenteroendocrine cells of the lower gastrointestinal tract. Int. J. Exp. Pathol. 92, 219–231 \\n(2011).\\n6. Elmentaite, R. et\\xa0al. Cells of the human intestinal tract mapped across space and time. \\nNature 597, 250–255 (2021).\\n7. Kennedy-Darling, J. et\\xa0al. Highly multiplexed tissue imaging using repeated oligonucleotide \\nexchange reaction. Eur. J. Immunol. 51, 1262–1277 (2021).\\n8. Black, S. et\\xa0al. CODEX multiplexed tissue imaging with DNA-conjugated antibodies.  \\nNat. Protoc. 16, 3802–3835 (2021).\\n9. Goltsev, Y. et\\xa0al. Deep profiling of mouse splenic architecture with CODEX multiplexed \\nimaging. Cell 174, 968–981 (2018).\\n10. Schürch, C. M. et\\xa0al. Coordinated cellular neighborhoods orchestrate antitumoral immunity \\nat the colorectal cancer invasive front. Cell 182, 1341–1359 (2020).\\n11. Buenrostro, J. D. et\\xa0al. Single-cell chromatin accessibility reveals principles of regulatory \\nvariation. Nature 523, 486–490 (2015).\\n12. Brbić, M. et\\xa0al. Annotation of spatially resolved single-cell data with STELLAR. Nat. Methods \\n19, 1411–1418 (2022).\\n13. Mayassi, T. & Jabri, B. Human intraepithelial lymphocytes. Mucosal Immunol. 11, 1281–1289 \\n(2018).\\n14. Wang, J., Chen, W.-D. & Wang, Y.-D. The relationship between gut microbiota and \\ninflammatory diseases: the role of macrophages. Front. Microbiol. 11, 1065 (2020).\\n15. Murray, P. J. & Wynn, T. A. Protective and pathogenic functions of macrophage subsets. \\nNat. Rev. Immunol. 11, 723–737 (2011).\\n16. Na, Y. R., Stakenborg, M., Seok, S. H. & Matteoli, G. Macrophages in intestinal inflammation \\nand resolution: a potential therapeutic target in IBD. Nat. Rev. Gastroenterol. Hepatol. 16, \\n531–543 (2019).\\n17. Emerenziani, S. et\\xa0al. Role of overweight and obesity in gastrointestinal disease. Nutrients \\n12, 111 (2019).\\n18. Rozanski, C. H. et\\xa0al. Sustained antibody responses depend on CD28 function in bone \\nmarrow-resident plasma cells. J. Exp. Med. 208, 1435–1446 (2011).\\n19. Tezuka, H. & Ohteki, T. Regulation of IgA production by intestinal dendritic cells and \\nrelated cells. Front. Immunol. 10, 1891 (2019).\\n20. Kawakami, T. et\\xa0al. Abundant a proliferation-inducing ligand (APRIL)-producing \\nmacrophages contribute to plasma cell accumulation in immunoglobulin G4-related \\ndisease. Nephrol. Dial. Transplant. 34, 960–969 (2019).\\n21. Lehmann, M. et\\xa0al. Human small intestinal infection by SARS-CoV-2 is characterized by  \\na mucosal infiltration with activated CD8 T\\xa0cells. Mucosal Immunol. 14, 1381–1392  \\n(2021).\\n22. Lutter, L., Hoytema van Konijnenburg, D. P., Brand, E. C., Oldenburg, B. & van Wijk, F.  \\nThe elusive case of human intraepithelial T\\xa0cells in gut homeostasis and inflammation. \\nNat. Rev. Gastroenterol. Hepatol. 15, 637–649 (2018).\\n23. Glaysher, B. R. & Mabbott, N. A. Isolated lymphoid follicle maturation induces the \\ndevelopment of follicular dendritic cells. Immunology 120, 336–344 (2007).\\n24. Eberl, G. & Lochner, M. The development of intestinal lymphoid tissues at the interface of \\nself and microbiota. Mucosal Immunol. 2, 478–485 (2009).\\n25. Knoop, K. A. & Newberry, R. D. Isolated lymphoid follicles are dynamic reservoirs for the \\ninduction of intestinal IgA. Front. Immunol. 3, 84 (2012).\\n26. Sipos, F. & Muzes, G. Isolated lymphoid follicles in colon: switch points between \\ninflammation and colorectal cancer? World J. Gastroenterol. 17, 1666–1673 (2011).\\n27. Bhate, S. S., Barlow, G. L., Schürch, C. M. & Nolan, G. P. Tissue schematics map the \\nspecialization of immune tissue motifs and their appropriation by tumors. Cell Syst. \\nhttps://doi.org/10.1016/j.cels.2021.09.012 (2021).\\n28. Smillie, C. S. et\\xa0al. Intra- and inter-cellular rewiring of the human colon during ulcerative \\ncolitis. Cell 178, 714–730 (2019).\\n29. Büttner, M., Ostner, J., Müller, C. L., Theis, F. J. & Schubert, B. scCODA is a Bayesian model \\nfor compositional single-cell data analysis. Nat. Commun. 12, 6876 (2021).\\n30. Dann, E., Henderson, N. C., Teichmann, S. A., Morgan, M. D. & Marioni, J. C. Differential \\nabundance testing on single-cell data using k-nearest neighbor graphs. Nat. Biotechnol. \\n40, 245–253 (2022).\\n31. Kinchen, J. et\\xa0al. Structural remodeling of the human colonic mesenchyme in inflammatory \\nbowel disease. Cell 175, 372–386 (2018).\\n32. Burclaff, J. et\\xa0al. A proximal-to-distal survey of healthy adult human small intestine and \\ncolon epithelium by single-cell transcriptomics. Cell Mol. Gastroenterol. Hepatol. 13, \\n1554–1589 (2022).\\n33. Hu, M.-J. et\\xa0al. Mechanism for insulin-like peptide 5 distinguishing the homologous \\nrelaxin family peptide receptor 3 and 4. Sci. Rep. 6, 29648 (2016).\\n34. Liu, C. & Lovenberg, T. W. Relaxin-3, INSL5, and their receptors. Results Probl. Cell Differ. \\n46, 213–237 (2008).\\n35. Gribble, F. M. & Reimann, F. Enteroendocrine cells: chemosensors in the intestinal \\nepithelium. Annu. Rev. Physiol. 78, 277–299 (2016).\\n36. Bartman, A. E. et\\xa0al. The MUC6 secretory mucin gene is expressed in a wide variety of \\nepithelial tissues. J. Pathol. 186, 398–405 (1998).\\n37. Hall, K. T. et\\xa0al. Human CD100, a novel leukocyte semaphorin that promotes B-cell \\naggregation and differentiation. Proc. Natl Acad. Sci. USA 93, 11780–11785 (1996).\\n38. Groiss, S. et\\xa0al. Highly resolved spatial transcriptomics for detection of rare events in \\ncells. Preprint at bioRxiv https://doi.org/10.1101/2021.10.11.463936 (2021).\\n39. Guilliams, M. et\\xa0al. Spatial proteogenomics reveals distinct and evolutionarily conserved \\nhepatic macrophage niches. Cell 185, 379–396 (2022).\\n40. Hu, S. et\\xa0al. Single-cell spatial transcriptomics reveals a dynamic control of metabolic \\nzonation and liver regeneration by endothelial cell Wnt2 and Wnt9b. Cell Rep. Med. 3, \\n100754 (2022).\\n41. Chen, S. et\\xa0al. Integration of spatial and single-cell data across modalities with weak \\nlinkage. Preprint at bioRxiv https://doi.org/10.1101/2023.01.12.523851 (2023).\\n42. Schep, A. N., Wu, B., Buenrostro, J. D. & Greenleaf, W. J. chromVAR: inferring transcription- \\nfactor-associated accessibility from single-cell epigenomic data. Nat. Methods 14,  \\n975–978 (2017).\\n43. Vierstra, J. et\\xa0al. Global reference mapping of human transcription factor footprints. Nature \\n583, 729–736 (2020).\\n44. van\\xa0der Flier, L. G. et\\xa0al. Transcription factor achaete scute-like 2 controls intestinal stem \\ncell fate. Cell 136, 903–912 (2009).\\n45. Gerbe, F. et\\xa0al. Intestinal epithelial tuft cells initiate type 2 mucosal immunity to helminth \\nparasites. Nature 529, 226–230 (2016).\\n46. Katz, J. P. et\\xa0al. The zinc-finger transcription factor Klf4 is required for terminal \\ndifferentiation of goblet cells in the colon. Development 129, 2619–2628 (2002).\\n47. Yang, Q., Bermingham, N. A., Finegold, M. J. & Zoghbi, H. Y. Requirement of Math1 for \\nsecretory cell lineage commitment in the mouse intestine. Science 294, 2155–2158 \\n(2001).\\n48. VanDussen, K. L. & Samuelson, L. C. Mouse atonal homolog 1 directs intestinal \\nprogenitors to secretory cell rather than absorptive cell fate. Dev. Biol. 346, 215–223 \\n(2010).\\n49. Chen, G. et\\xa0al. Foxa3 induces goblet cell metaplasia and inhibits innate antiviral immunity. \\nAm. J. Respir. Crit. Care Med. https://doi.org/10.1164/rccm.201306-1181OC (2014).\\n50. Piccand, J. et\\xa0al. Rfx6 promotes the differentiation of peptide-secreting enteroendocrine \\ncells while repressing genetic programs controlling serotonin production. Mol. Metab. \\n29, 24–39 (2019).\\n51. Granja, J. M. et\\xa0al. Single-cell multiomic analysis identifies regulatory programs in \\nmixed-phenotype acute leukemia. Nat. Biotechnol. 37, 1458–1465 (2019).\\n52. Granja, J. M. et\\xa0al. ArchR is a scalable software package for integrative single-cell \\nchromatin accessibility analysis. Nat. Genet. 53, 403–411 (2021).\\n53. Tsai, F.-Y. & Orkin, S. H. Transcription factor GATA-2 is required for proliferation/survival of \\nearly hematopoietic cells and mast cell formation, but not for erythroid and myeloid \\nterminal differentiation. Blood 89, 3636–3643 (1997).\\n54. Cobaleda, C., Schebesta, A., Delogu, A. & Busslinger, M. Pax5: the guardian of B cell identity \\nand function. Nat. Immunol. 8, 463–470 (2007).\\n55. Pagani, F., Tratta, E., Dell’Era, P., Cominelli, M. & Poliani, P. L. EBF1 is expressed in \\npericytes and contributes to pericyte cell commitment. Histochem. Cell Biol. 156,  \\n333–347 (2021).\\n56. Rosen, E. D. et\\xa0al. PPARγ is required for the differentiation of adipose tissue in\\xa0vivo and \\nin\\xa0vitro. Mol. Cell 4, 611–617 (1999).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='fe71ffa1-c701-4483-8356-aecc293912bb', embedding=None, metadata={'page_label': '584', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='584 | Nature | Vol 619 | 20 July 2023\\nArticle\\n57. Britsch, S. et\\xa0al. The transcription factor Sox10 is a key regulator of peripheral glial \\ndevelopment. Genes Dev. 15, 66–78 (2001).\\n58. Barker, N. Adult intestinal stem cells: critical drivers of epithelial homeostasis and \\nregeneration. Nat. Rev. Mol. Cell Biol. 15, 19–33 (2014).\\n59. Duc, C., Farman, N., Canessa, C. M., Bonvalet, J. P. & Rossier, B. C. Cell-specific expression \\nof epithelial sodium channel alpha, beta, and gamma subunits in aldosterone-responsive \\nepithelia from the rat: localization by in\\xa0situ hybridization and immunocytochemistry.  \\nJ. Cell Biol. 127, 1907–1921 (1994).\\n60. Wang, M. et\\xa0al. Common genetic variation in ETV6 is associated with colorectal cancer \\nsusceptibility. Nat. Commun. 7, 11478 (2016).\\n61. Jabri, B. & Sollid, L. M. T\\xa0cells in celiac disease. J. Immunol. 198, 3005–3014 (2017).\\n62. Xuan, K. et\\xa0al. The association between hypertension and colorectal cancer: a meta- \\nanalysis of observational studies. Eur. J. Cancer Prev. 30, 84–96 (2021).\\n63. Grossman, E., Messerli, F. H., Boyko, V. & Goldbourt, U. Is there an association between \\nhypertension and cancer mortality? Am. J. Med. 112, 479–486 (2002).\\n64. Becker, W. R. et\\xa0al. Single-cell analyses define a continuum of cell state and composition \\nchanges in the malignant transformation of polyps to colorectal cancer. Nat. Genet. 54, \\n985–995 (2022).\\n65. Kassam, I., Wu, Y., Yang, J., Visscher, P. M. & McRae, A. F. Tissue-specific sex differences in \\nhuman gene expression. Hum. Mol. Genet. 28, 2976–2986 (2019).\\nPublisher’s note Springer Nature remains neutral with regard to jurisdictional claims in \\npublished maps and institutional affiliations.\\nOpen Access This article is licensed under a Creative Commons Attribution \\n4.0 International License, which permits use, sharing, adaptation, distribution \\nand reproduction in any medium or format, as long as you give appropriate \\ncredit to the original author(s) and the source, provide a link to the Creative Commons licence, \\nand indicate if changes were made. The images or other third party material in this article are \\nincluded in the article’s Creative Commons licence, unless indicated otherwise in a credit line \\nto the material. If material is not included in the article’s Creative Commons licence and your \\nintended use is not permitted by statutory regulation or exceeds the permitted use, you will \\nneed to obtain permission directly from the copyright holder. To view a copy of this licence, \\nvisit http://creativecommons.org/licenses/by/4.0/.\\n© The Author(s) 2023', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='bbeaca81-0cf9-4846-bf3a-d29fc5ec470f', embedding=None, metadata={'page_label': '585', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Methods\\nTissue collection and processing\\nThis study complies with all relevant ethical regulations and was \\napproved by the Washington University Institutional Review Board \\nand the Stanford University Institutional Review Board. Human bowel \\ntissues were procured from deceased organ donors. Written informed \\nconsent was obtained from the next-of-kin for all donor participants.\\nOrgans were preserved according to surgical protocols to prepare \\nthe bowel for transplantation. In brief, a large-bore cannula was placed \\ninto the infrarenal aorta before circulatory arrest. Immediately after \\ncirculatory cessation, the abdominal viscera was rapidly cooled by \\nflushing ice-cold HTK preservation solution through the aortic can-\\nnula and by packing the abdominal cavity with ice. The bowel was kept \\ncold throughout the transport and dissection process until samples \\ncould be frozen for long-term storage. Tissue samples collected from \\nthe designated bowel sites were preserved by snap freezing in liquid \\nnitrogen for snRNA-seq/scATAC, and embedded and frozen in optimal \\ncutting temperature compound (OCT) for CODEX.\\nArray creation\\nImaging data were collected from four human donors, each of whom \\nconstitutes a dataset. Each dataset includes two arrays of tissues that \\nwere imaged together on the same coverslip with four tissues per array: \\ncolon (sigmoid, descending, transverse and ascending), and small intes-\\ntine (ileum, mid-jejunum, proximal jejunum and duodenum). Tissues \\nwere individually frozen in OCT moulds and then cut and assembled into \\narrays of four tissues with known directionality such that a cross-section \\nof each tissue would be achieved cutting the block at once. Arrays were \\nconstructed on the cryostat and sectioned at a width of 7\\u2009µm.\\nRegistration of samples with HuBMAP common coordinate \\nframework\\nWe have also registered these blocks within HuBMAP’s tissue registra-\\ntion in a common coordinate framework66. In brief, male and female \\n3D reference objects for 11 organs including the small bowel and colon \\nwere created using Visible Human Project datasets. Using standard \\nsurgical anatomical landmarks used to collect the eight bowel sites, the \\ntissue blocks were registered to the reference objects. The anatomical \\nlandmarks used for small bowel segments were as follows: (1) descend-\\ning duodenum to the right of the pancreas head; (2) 5\\u2009cm beyond the \\nligament of Treitz in the jejunum; (3) 200\\u2009cm of the jejunum beyond \\nthe ligament of Treitz in the mid-bowel; and (4) 5\\u2009cm proximal to the \\nileocecal valve for terminal ileum. Five centimetres of bowel at each \\nsite was collected, representing approximately 20\\u2009g of tissue at each \\nsite. For the colon, the following landmarks used: (1) the right colon \\nmidway between the ileocecal valve and the hepatic flexure; (2) the \\ntransverse colon midway between the hepatic and splenic flexures;  \\n(3) the left colon midway between the splenic flexure to the appearance \\nof the sigmoid mesentery; and (4) the sigmoid colon midway to the \\nrectosigmoid junction where the taenia coli ceased.\\nCODEX antibody conjugation and panel creation\\nCODEX multiplexed imaging was performed according to the CODEX \\nstaining and imaging protocol previously described8. Antibody panels \\nwere chosen to include targets that identify subtypes of intestinal epi-\\nthelium and stromal cells, and cells of the innate and adaptive immune \\nsystem. Detailed panel information is provided in Supplementary \\nTable\\xa07. Each antibody was conjugated to a unique oligonucleotide bar-\\ncode, after which the tissues were stained with the antibody–oligonu-\\ncleotide conjugates and we validated that the staining patterns matched \\nthe expected patterns already established for immunohistochemistry \\nwithin positive control tissues of the intestine or tonsil. Similarly, hae-\\nmatoxylin and eosin morphology staining was used to confirm the \\nlocation of marker staining. First, antibody–oligonucleotide conjugates \\nwere tested in low-plex fluorescence assays and the signal-to-noise ratio \\nwas also evaluated at this step, then they were tested all together in a \\nsingle CODEX multicycle.\\nCODEX multiplexed imaging\\nThe tissue arrays were then stained with the complete validated panel \\nof CODEX antibodies and imaged8. In brief, this entails cyclic strip-\\nping, annealing and imaging of fluorescently labelled oligonucleotides \\ncomplementary to the oligonucleotide on the conjugate. After valida-\\ntion of the antibody–oligonucleotide conjugate panel, a test CODEX \\nmultiplexed assay was run, during which the signal-to-noise ratio was \\nagain evaluated, and the optimal dilution, exposure time and appropri-\\nate imaging cycle was evaluated for each conjugate (Supplementary \\nTable\\xa07). Finally, each array underwent CODEX multiplexed imaging. \\nMetadata from each CODEX run are provided in Supplementary Table\\xa08.\\nCODEX data processing\\nRaw imaging data were then processed using the CODEX Uploader for \\nimage stitching, drift compensation, deconvolution and cycle con-\\ncatenation. Processed data were then segmented using the CODEX \\nSegmenter or CellVisionSegmenter, a watershed-based single-cell \\nsegmentation algorithm and a neural network R-CNN-based single-cell \\nsegmentation algorithm, respectively. The donor sample from indi-\\nvidual B001 was segmented using the CODEX Segmenter (with param-\\neters tuned as described previously8), whereas all of the other donor \\nsamples were segmented using CellVisionSegmenter. CellVisionSeg-\\nmenter has been shown to work well with segmenting both dense and \\ndiffuse cellular tissues with CODEX data67. CellVisionSegmenter is an \\nopen-source, pretrained nucleus segmentation and signal quantifica-\\ntion software based on the Mask region-convolutional neural network \\n(R-CNN) architecture. Indeed, it was designed and trained on manu-\\nally annotated images from CODEX multiplexed imaging data within \\nour own group. Consequently, the only parameter that was altered \\nwas the growth pixels of the nuclear mask, which we found experi -\\nmentally to work best at a value of 3. Despite this, no segmentation \\nalgorithm does a perfect job of segmentation in cases in which the \\nboundaries identified may capture portions of neighbouring cells, \\nand nuclear segmentation can limit quantified signal that whole-cell \\nsegmentation might be able to capture (although also imperfect from \\nlack of consistent cell membrane stains), which has been discussed in \\nmore detail in reviews and primary sources of segmentation67–74. For \\nthis reason, we performed an in-depth analysis of the different data \\nnormalization techniques and unsupervised clustering methods for \\nrobust identification of cell types in CODEX intestine data. This analysis \\nrevealed that there is some segmentation noise that could affect cell \\ntype identification if using manual gating, but using z-normalization, \\nVortex or Leiden-based unsupervised clustering, over-clustering the \\ndata and manually overlaying resultant cell type clusters to the image \\nresults in cell type identification at a much higher fidelity75.\\nBoth the CODEX Uploader and Segmenter software can be down-\\nloaded from GitHub (https://github.com/nolanlab/CODEX), and the \\nCellVisionSegmenter software is available at GitHub (https://github.\\ncom/bmyury/CellVisionSegmenter or https://github.com/michael-\\nlee1/CellSeg). After the upload, the images were again evaluated for \\nspecific signal: any markers that produced an untenable pattern or \\na low signal-to-noise ratio were excluded from the ensuing analysis. \\nUploaded images were visualized in ImageJ (https://imagej.nih.gov/ij/).\\nCell type analysis\\nB001 and B004 cell type identification was performed according \\nto methods developed previously75. In brief, nucleated cells were \\nselected by gating DRAQ5, Hoechst double-positive cells, followed \\nby z-normalization of protein markers used for clustering (some phe-\\nnotypic markers were not used in the unsupervised clustering). Cells \\npositive (z\\u2009>\\u20091) for greater than 35 fluorescent markers were removed ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='857ea411-ee40-40c0-8f2e-47502692f445', embedding=None, metadata={'page_label': '586', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nfrom the data. Then the data were overclustered with X-shift (https://\\ngithub.com/nolanlab/vortex) or Leiden-based clustering with the \\nscanpy Python package (v.1.9.1). These processing steps were per-\\nformed based on an in-depth analysis of normalization techniques \\nand unsupervised clustering algorithms used for CODEX multiplexed \\nimaging data of the intestine75. These are not new approaches and many \\npackages have emerged for integrating these clustering algorithms \\ninto libraries such as Squidpy75. Clusters were assigned a cell type on \\nthe basis of average cluster protein expression and the location within \\nimage. Impure clusters were split or reclustered after mapping back to \\nthe original fluorescence images.\\nCell type annotation using STELLAR\\nCODEX cell type labels were transferred to other donors using the \\nSTELLAR framework for annotating spatially resolved single-cell data, \\nas we described previously12. In brief, STELLAR is a geometric deep \\nlearning method that uses the spatial and molecular cell information \\nto transfer cell type labels across different datasets. While SpaGCN76 \\nand Spage2vec77 can leverage spatial and molecular data to annotate \\ncells, these methods are unsupervised clustering methods. As such, \\nthey require manual effort to assign cell clusters to corresponding cell \\ntypes, and can also require additional manual effort for multiple itera-\\ntive cluster refinements. On the other hand, STELLAR automatically \\nidentifies existing cell types and discovers cell types without requiring \\nmanual effort, so we used STELLAR as the preferred method. T o apply \\nSTELLAR, we used B004 donor data as the labelled training dataset as \\ndefined in STELLAR. This dataset was curated and annotated by clus-\\ntering, merging, reclustering, subclustering and assigning cells to cell \\ntypes based on average marker expression. Each cluster’s purity and \\naccuracy were confirmed by location of the cell within CODEX images \\nwith corresponding fluorescence images and also H&E staining. We \\nused B004, B005 and B006 donor dataset to train STELLAR and transfer \\nannotations to all other donor datasets that were treated as unanno-\\ntated test datasets in the STELLAR framework. We did not expect to find \\nany new cell types across different donors so we used STELLAR to iden-\\ntify existing cell types across donors. All datasets were z-normalized \\nas suggested previously75. We then manually confirmed the quality of \\nSTELLAR’s cell type assignments in all donor datasets by looking at aver-\\nage marker expression profiles of predicted cell types. We found that \\nprotein marker distributions match expert hand-annotated profiles12.\\nCODEX cell type percentage normalization\\nWe normalized the cell type percentage to the overall cell category \\nfor three reasons. First, we captured a cross-section of each individual \\nintestinal piece; however, a few of the intestinal pieces (4 out of 64) were \\nnot representative cross-sections. Second, with a fixed CODEX imaging \\nwindow, we attempted to capture the full mucosal area, as this contains \\nthe majority of cell types identified by antibodies in our panel, which \\nled to variable amounts of the muscularis externa captured. Third, it \\nis useful to normalize by overall cell type to understand how cell type \\ncompositions change across the three compartments.\\nCODEX same-cell density analysis\\nCODEX same-cell density was analysed by taking the average distance \\nof the five nearest neighbours of the same cell type for each individual \\ncell in each imaged region. This average distance was divided by the \\nmost diffuse distance for same cell types. The most diffuse same cell \\ndistance was calculated by taking the total number of cells of a given \\ncell type divided by the total area of the region. Thus, numbers that \\nare closer to 1 are least dense and numbers closer to 0 are more dense \\nwith cells of the same cell type.\\nNeighbourhood identification analysis\\nNeighbourhood analysis was performed as described previously10,27. In \\nbrief, this analysis involved (1) taking windows of cells across the entire \\ncell type map of a tissue with each cell as the centre of a window; (2) \\ncalculating the number of each cell type within this window; (3) clus-\\ntering these vectors; and (4) assigning overall structure on the basis of \\nthe average composition of the cluster (Extended Data Fig.\\xa02a). In brief, \\ndetermining window size cut-offs for cellular neighbourhood analysis is \\nan important metric to be chosen. In general, smaller window sizes will \\nidentify more local or microstructures, whereas larger window sizes will \\nlead to the identification of similarly composed structures that require \\na larger window size. For our neighbourhood analysis here, we chose \\nto have window size cut-offs by selecting the ten nearest neighbours \\naround a given cell. This number has worked well to identify conserved \\ncompositions representing a cell’s immediate microenvironment or \\nlocal neighbours in other tissues10,12,27,75,78,79. We chose this strategically \\nto look at the microstructures at the neighbourhood level because \\nwe also were curious to understand how these microstructures work \\ntogether and come together to form macrostructures of the intestine \\nat a multihierarchical scale. In general, the size of the structure does \\nnot directly relate to the window size choice, but instead relates to \\nhow compartmentalized the conserved cell types are within a given \\nstructure. Neighbourhoods were overclustered to 30 clusters. These \\nclusters were mapped back to the tissue and evaluated for cell type \\nenrichments to determine overall structure and merged down into \\n20 unique structures.\\nNeighbourhood conservation analysis\\nT o determine neighbourhood compositional (cell type) conservation \\nacross the small bowel and colon, neighbourhood enrichment scores \\nwere found separately for both the small bowel and colon samples \\nacross all donors. This enrichment score is the average cell type per-\\ncentage within the average of the neighbourhood cluster divided by \\nthe average cell type percentage for all cells in the tissue. The colon \\nenrichment scores for each cell type were subtracted from the small \\nbowel scores to provide the heat map that was ordered both in terms of \\nthe greatest absolute sum of differences for both neighbourhood and \\ncell type in conservation. Thus, differences in cell type enrichment for a \\ngiven neighbourhood indicate that this cell type is not compositionally \\nconserved across the small and large intestines for this neighbourhood \\nstructure.\\nCommunity and tissue unit identification analysis\\nCommunities were determined similarly to how multicellular neigh-\\nbourhoods were determined with some small differences. In brief, \\nthe cells in the neighbourhood tissue maps were taken with a larger \\nwindow size of 100 nearest neighbours. These windows were taken \\nacross the entirety of the tissue and the vectors then clustered with \\nk-means clustering and overclustering with 20 total clusters. These \\nclusters were mapped back to the tissue and evaluated for neighbour-\\nhood composition and enrichment to determine overall community \\ntype. This same approach was applied to communities with a window \\nsize of 300 nearest neighbours.\\nHierarchical intestine structural graphs\\nEach hierarchical level was connected to the next by either what it con-\\ntributed to the largest in the next level, or what made up at least 15% \\nof this next hierarchy. The percentage of each feature in the level was \\nrepresented by size of the shape. The shape and colour combination \\ncorrespond to the level and feature respectively. The size of the con-\\nnecting line represents the amount that a feature contributes to the \\nnext feature.\\nSpatial context maps\\nSpatial context maps were created as previously described27. In brief, \\nthe spatial context analysis of neighbourhood–neighbourhood or \\ncommunity–community associations has some similarities with our \\nmethod to identify multicellular neighbourhoods but also contains a ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b15986bc-7f74-4c6f-846a-79e4db7f3c83', embedding=None, metadata={'page_label': '587', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='few key differences. First, windows of neighbourhoods or communities \\nwere calculated with either 100 or 300 nearest neighbours respectively. \\nFor the neighbourhood spatial context maps, only the cells classified \\nin the mucosa tissue unit were included in the analysis, whereas the \\ncommunity spatial context maps included all cells from all tissues. \\nOnce windows were calculated (number of each cell type within the \\nwindow), then the combinations representing more than 85% of the \\nneighbourhoods within that window were selected as a combination. \\nThis combination informs about prominent associations of neighbour-\\nhoods or communities in the window, a feature that we termed spatial \\ncontext. Combinations were then counted and represented in size \\nby the size of the black circle underneath the square neighbourhood \\ncombination and only combinations that had a greater frequency than \\n0.1% of all combinations were plotted for visualization purposes. Each \\ncombination was then connected to each combination containing it \\nwith another combination in sublayers of the graph. For example, if \\nthe Secretory Epithelial community was found to represent 85% of a \\nwindow, then this would be its own combination (purple triangle). If it \\nis found as a combination with the Adaptive-Immune-Enriched com-\\nmunity, then it is connected with this in a combination (purple triangle \\nand orange triangle). Similarly, if it is found as a combination with the \\nPlasma-Cell-Enriched community, then it is connected with this in a \\ncombination (purple triangle and yellow triangle). The single combina-\\ntion is therefore connected (through black edges) to these combina-\\ntions (nodes) in the spatial context graph. Similarly, combinations of \\nthe Secretory Epithelial and Adaptive-Immune-Enriched communi-\\nties\\xa0(purple triangle and orange triangle) derive from this combination \\n(for example, Secretory Epithelial, Adaptive-Immune-Enriched and \\nFollicle (purple, orange and blue triangles)).\\nTissue motif identification\\nWe used a previously developed method to identify significantly \\nassociated cellular neighbourhood-neighbourhood motifs 27.  \\nIn brief, motif identification uses segmented areas of the tissue \\nwhere multiple cells of the same community are co-located, instead \\nof individual cells (Extended Data Fig.\\xa05a). The tissue network graphs \\ntherefore represent shared edges between instances of communi -\\nties. T o create a tissue graph for each treatment group, we took the \\nunion of the tissue graphs of each unique imaging region. We then \\ncreated a null-set as the graph of the set of cell neighbourhood or \\ncommunity assignments by a sequence of valid transpositions of cell \\nneighbourhood or community assignments. Permuting neighbour-\\nhood or community assignments and fixing the number of vertices \\ncreated the maximum entropy null distribution. Only two chains \\nwith at least five instances were considered. T o identify significant \\nchains, P values were Bonferroni corrected by multiplying by twice \\nthe number of tests conducted in each comparison group (small bowel  \\nand CL).\\nTissue dissociation and nucleus isolation for single-nucleus \\nexperiments\\nNuclei were isolated using the OmniATAC protocol80. Isolation of nuclei \\nwas carried out on wet ice. A total of 40–60\\u2009mg of flash-frozen tissue \\nwas gently triturated and thawed in 1\\u2009ml HB (lysis) buffer (1.0341× HB \\nstable solution, 1\\u2009M DTT, 500\\u2009mM spermidine, 150\\u2009mM spermine, 10% \\nNP40, cOmplete Protease Inhibitor, Ribolock) for 5\\u2009min. Tissue was \\nthen dounced 10 times with pestle A and 20 times with pestle B, or \\nuntil there was no resistance from either pestle. The samples were \\nthen filtered through a 40\\u2009µm cell strainer (Falcon, 352340) and the \\nresulting homogenate was transferred to a prechilled 2\\u2009ml LoBind \\ntube. The samples were centrifuged in a 4\\u2009°C fixed-angle centrifuge \\nfor 5\\u2009min at 350\\u2009rcf to pellet the nuclei. After centrifugation, all but \\n50\\u2009µl of supernatant was removed. Then, 350\\u2009µl HB was added to the \\nnucleus pellet for a total volume of 400\\u2009µl and the nuclei were gently \\nresuspended using a wide bore pipet. One volume of 50% iodixanol \\n(60% OptiPrep (Sigma Aldrich; D1556), diluent buffer (2\\u2009M KCl, 1\\u2009M \\nMgCl2, 0.75\\u2009M Tricine-KOH pH\\u20097.8), water) was added and the resulting \\nsolution was gently triturated. Next, 600\\u2009µl of 30% iodixanol was care-\\nfully layered under the 25% mixture. Finally, 600\\u2009µl of 40% iodixanol \\nwas layered under the 30% mixture. The sample was then centrifuged \\nfor 20\\u2009min at 3,000\\u2009rcf in a 4\\u2009°C swinging-bucket centrifuge, result-\\ning in a visible band of nuclei. The supernatant was aspirated down to \\nwithin 200–300\\u2009µl of the nucleus band. The nucleus band was then \\ncollected at 200\\u2009µl and transferred to a fresh 1.5\\u2009ml tube. The sample \\nwas diluted with one volume (200\\u2009µl) of resuspension buffer (1× PBS, 1% \\nBSA, 0.2\\u2009U\\u2009µl−1 Ribolock). The nucleus concentration was determined \\nusing the Countess II FL Automated Cell Counter (Thermo Fisher Sci-\\nentific; AMQAF1000).\\nsnA T AC–seq\\nsnATAC-seq targeting 9,000 nuclei per sample was performed using \\nthe Chromium Next GEM Single Cell ATAC Library & Gel Bead Kit \\nv1.1 (10x Genomics, 1000175) and the Chromium Next GEM Chip H  \\n(10x Genomics, 1000161) or Chromium Single Cell ATAC Library & Gel Bead \\nKit (10x Genomics, 1000110). Libraries were sequenced on the Illumina \\nNovaSeq 6000 system (1.4\\u2009pM loading concentration, 50\\u2009×\\u20098\\u2009×\\u200916\\u2009×\\u200949 \\nbp read configuration) targeting an average of 25,000 reads per  \\nnucleus.\\nSingle-nucleus transcriptome sequencing using snRNA-seq\\nsnRNA-seq targeting 9,000 nuclei per sample was performed using \\nChromium Next GEM Single Cell 3’ Reagent Kits v3.1 (10x Genomics, \\n1000121) and the Chromium Next GEM Chip G Single Cell Kit (10x \\nGenomics, 1000120). Libraries were pooled and sequenced on the \\nIllumina NovaSeq 6000 system (read\\u20091\\u2009=\\u200928\\u2009bp, i7 index\\u2009=\\u20098\\u2009bp, i5 \\nindex\\u2009=\\u20090\\u2009bp, read 2\\u2009=\\u200991\\u2009bp read configuration) targeting an average \\nof 20,000 reads per nucleus.\\nSingle-nucleus multiome experiments\\nsnMultiome experiments targeting 9,000 nuclei per sample were per-\\nformed using Chromium Chromium Next GEM Single Cell Multiome \\nATAC + Gene Expression (10x Genomics, 1000283). ATAC (read 1\\u2009=\\u200950\\u2009bp, \\ni7 index\\u2009=\\u20098\\u2009bp, i5 index\\u2009=\\u200924\\u2009bp, read 2\\u2009=\\u200949\\u2009bp read configuration) and \\nRNA (read 1\\u2009=\\u200928\\u2009bp, i7 index\\u2009=\\u200910\\u2009bp, i5 index\\u2009=\\u200910\\u2009bp, read 2\\u2009=\\u200990\\u2009bp \\nread configuration) libraries were sequenced separately on the Illumina \\nNovaSeq 6000 system.\\nInitial processing of single-nucleus data\\nInitial processing of scATAC-seq data was performed using the Cell \\nRanger ATAC Pipeline (https://support.10xgenomics.com/single-cell- \\natac/software/pipelines/latest/what-is-cell-ranger-atac ) by first  \\nrunning cellranger-atac mkfastq to demultiplex the bcl files and \\nthen running cellranger-atac count to generate scATAC fragments  \\nfiles. Initial processing of snRNA-seq data was performed using the  \\nCell Ranger Pipeline (https://support.10xgenomics.com/single-cell- \\ngene-expression/software/pipelines/latest/what-is-cell-ranger ) by \\nfirst running cellranger mkfastq to demultiplex the bcl files and then \\nrunning cellranger count. As nuclear RNA was sequenced, data were \\naligned to a pre-mRNA reference. Initial processing of the mutiome \\ndata, including alignment and generation of fragments files and expres-\\nsion matrices, was performed using the Cell Ranger ARC Pipeline.\\nColocalization analyses\\nThe CODEX data were used to compute and compare the CLQ between \\nall cell-type pairs in the small bowel versus the colon. The colocalization \\nquotient between cell type A and cell type B was calculated using the \\nexpression CLQ=\\nCN\\nNNA/uni2192B\\n/\\n/( −1 )\\nA/uni2192BA\\nB\\n (ref. 81), where CA→B is the number of cells \\nof cell type A among the defined nearest neighbours of cell type B.  \\nN is the total number of cells and NA and NB are the numbers of cells for \\ncell type A and cell type B. Student’s t -tests, adjusted for multiple-  ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='aff0de95-f4c6-4c18-9787-97f1a82beb87', embedding=None, metadata={'page_label': '588', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nhypothesis testing, were used identity statistically significant different \\nCLQs between the small bowel and colon.\\nLigand and receptor analyses\\nThe FANTOM5 database82 and 12 more literature-supported experimen-\\ntally validated ligand and receptor pairs were used to obtain the final list \\nof validated ligand receptor pairs83. Non-parametric Wilcoxon rank-sum \\ntests were used to identify differentially expressed ligands and receptors \\nin the small bowel versus the colon (adjusted P-value cut-off\\u2009=\\u20090.05).\\nQuality control, dimensionality reduction and clustering of \\nsnA T AC–seq data\\nThe snATAC fragments files were loaded into R (v.4.1.2) using the cre-\\nateArrowFiles function in ArchR52. Quality-control metrics were com-\\nputed for each cell and any cells with TSS enrichments less than 5 were \\nremoved. Cells were also filtered on the basis of the number of unique \\nfragments sequenced using a unique fragment cut-off that was defined \\nfor each individual sample. The sample-specific cut-offs enabled us \\nto account for differences in sequencing depth between samples and \\nranged from 1,000 to 10,000, with the most common cut-off being \\n3,000 fragments per cell. After quality control and filtering, doublet \\nscores for all multiome cells and all non-multiome snATAC cells were \\ncomputed using the ArchR function addDoubletScores with k\\u2009=\\u200910, \\nknnMethod = “UMAP” and LSIMethod = 1. An ArchR project was then \\ncreated and doublets were filtered with filterDoublets with a filter ratio \\nof 1.2. A small number of snATAC samples did not separate into distinct \\nclusters of expected cell types and were removed from downstream \\nanalysis. For the non-multiome scATAC cells, an IterativeLSI dimension-\\nality reduction was generated using addIterativeLSI, with iterations\\u2009=\\u20093, \\nsampleCellsPre\\u2009=\\u200925000, dimsT oUse\\u2009=\\u20091:25 and varFeatures\\u2009=\\u200915000. \\nNext, clusters were added with addClusters with resolution\\u2009=\\u20091.5, nOut-\\nlier\\u2009=\\u200920, seed\\u2009=\\u20091, sampleCells\\u2009=\\u200940000 and maxClusters\\u2009=\\u200940, and the \\nresulting clusters were divided into groups on the basis of whether the \\ncells exhibited high gene activity scores52, a measure of accessibility \\nwithin and around a gene body, for known immune, stromal or epithelial \\nmarker genes.\\nQuality control, dimensionality reduction and clustering of \\nsnRNA-seq data\\nAfter running Cell Ranger, the raw_feature_bc_matrix produced by Cell \\nRanger was read into R with the Seurat84 function Read10X. The data \\nwere filtered to remove nuclei with fewer than 400 unique genes per \\nnucleus, greater than or equal to 10,000 genes per nucleus, greater than \\nor equal to 20,000 counts per nucleus, or greater than or equal to 5% \\nmitochondrial RNA per nuclei. T o limit the contributions of ambient \\nRNA, we also filtered out nuclei that did not have at least three times \\nas many counts as the median number of counts in all droplets, which \\nshould reflect the median number of counts in an empty droplet, as the \\nlarge majority of droplets are empty. This limits the fraction of RNA that \\ncan come from ambient RNA in droplets that are included in the dataset. \\nDoubletFinder85 was run for each non-multiome snRNA sample using \\nprincipal components 1–20. nExp was set to 0.076\\u2009×\\u2009nCells2/10,000, pN \\nto 0.25 and pK was determined using paramSweep_v3, and cells that \\nwere classified as doublets were removed before downstream analysis. \\nsnRNA data for both multiome and non-multiome cells was corrected \\nfor possible ambient RNA correction using DecontX86.\\nThe remaining cells from all samples were merged into a single \\nseurat object. The data was then processed using Seurat’s standard \\npipeline84. First, NormalizeData was run using the method LogNormal-\\nize and scale.factor of 10,000. Variable features were identified with \\nSeurat’s findVariableFeatures using the vst method and 2,000 features. \\nScaleData was then run on all genes and principal components were \\ncomputed with RunPCA. T o account for batch effects between differ-\\nent donors in clustering, the RunHarmony87 function was run with the \\ndata being grouped by donor. RunUMAP was then used to generate a \\nUMAP dimensionality reduction from the harmony reduction and the \\ncells were clustered by first using FindNeighbors with reduction\\u2009=\\u2009“har-\\nmony” and dims\\u2009=\\u20091:20 and then FindClusters with a resolution of 1. \\nExpression of marker genes in the resulting clusters was then used to \\nlabel clusters as epithelial, stromal or immune for downstream analysis.\\nAnnotation of single-nucleus data\\nThe snATAC and snRNA data were annotated in the following groups: \\nepithelial duodenum, epithelial jejunum, epithelial ileum, epithelial \\ncolon, stromal and immune. The snATAC data were further divided into \\nseparate projects for multiome and nonmultiome cells and the RNA \\nannotations were used for the multiome cells while the snATAC-only \\ncells were annotated separately. For the ATAC data, the cells in each of \\nthese compartments were subset into a new ArchR project. addItera-\\ntiveLSI was then run for each compartment. addHarmony was then run \\nusing the LSI dimensions as input. After computation of the harmony \\ndimensions, the cells were clustered using addClusters and a UMAP was \\ncomputed on the basis of the harmony coordinates using addUMAP. \\nClusters were annotated by examining gene activity scores of known \\nmarker genes. Marker genes were used for initial annotation of cell types \\nincluding BEST4+ enterocytes (BEST4, OTOP2), goblet cells (MUC2, \\nTFF1, SYTL2), immature goblet cells (KLK1, RETNLB, CLCA1), stem cells \\n(RGMB, SMOC2, LGR5, ASCL2), tuft cells (SH2D6, TRPM5, BMX, LRMP, \\nHCK), enteroendocrine cells (SCGN, FEV, CHGA, PYY , GCG), cycling \\ntransit-amplifying cells (TICRR, CDC25C) and Paneth cells (LYZ, DEFA5).\\nWithin the immune compartment, we identified clusters of CD4+ \\nand CD8+ T\\xa0cells (CD2, CD3E, IL7R, CD4, CD8), B cells (PAX5, MS4A1, \\nCD19), plasma cells (IGLL5, AMPD1), natural killer cells (SH2D1B), mac-\\nrophages/monocytes (CD14) and mast cells (HDC, GATA2, TPSAB1) \\n(Fig.\\xa04c and Extended Data Fig.\\xa06d). Natural killer cells and T\\xa0cells from \\none donor clustered separately from the other T\\xa0cells in the snRNA data \\n(Extended Data Fig.\\xa06e), possibly because this donor was much younger \\n(24\\u2009years) than the others in this study (Supplementary Table\\xa01). Smooth \\nmuscle/myofibroblast clusters exhibited high expression of ACTA2, \\nMTH11 and TAGLN. Villus fibroblasts exhibited high expression of \\nWNT5B and some crypt fibroblasts exhibited high expression of RSPO3.\\nFor the snRNA cells, cells were divided into six Seurat objects from the \\ncompartments listed above. Cells from the immune and stromal compart-\\nments were run through a pipeline analogous\\xa0to the pipeline\\xa0listed above \\nfor\\xa0initial clustering (NormalizeData, ScaleData, RunHarmony, Find-\\nNeighbors and FindClusters). For cells in the four epithelial groups, we \\nintegrated the data using a different approach, first running SCTransform \\non the epithelial cells from each sample with assay\\u2009=\\u2009“decontXcounts” , \\nmethod = “glmGamPoi” , and vars.to.regress = c(“percent.mt” , “percent.\\nribo”). We then ran the Seurat functions SelectIntegrationFeatures with \\nfeatures\\u2009=\\u20093000, PrepSCTIntegration, FindIntegrationAnchors using \\nreference-based integration and dims 1–30, and finally IntegrateData. \\nWe then ran RunPCA on the integrated data followed by FindNeighbors \\nand FindClusters to cluster the resulting integrated data.\\nIn addition to removing probable doublets based on simulating dou-\\nblets as described above, we also identified clusters with expression \\nof markers from multiple lineages (for example, stromal and immune) \\nduring downstream clustering and annotation. For example, some \\ncells that initially clustered with immune cells expressed higher levels \\nof stromal genes than would be expected. For these cases, we took the \\nfollowing approach: we first clustered all of the cells initially classified as \\nimmune cells and identified marker genes for each cluster. We next com-\\npared the marker genes to a previously published list of colon marker \\ngenes28 to nominate clusters that may not contain singlet immune \\ncells. Next, we moved these cells to the stromal or epithelial compart-\\nments, in which we clustered them with all of the cells that were initially \\nclassified as epithelial or stromal cells. In this case, if the cells had high \\nexpression of immune marker genes when compared to stromal cells, \\nwe reasoned that they were most likely immune/stromal doublets and \\nremoved the cluster of cells before downstream analysis. After initial ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='52eb5bf6-d8fd-4a94-bd32-f14fa7a62cdc', embedding=None, metadata={'page_label': '589', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='annotation of epithelial cells, enteroendocrine cells in the snRNA data \\nfrom all samples were integrated according to the SCTransform-based \\nintegration approach by running SCTranform on all epithelial cells from \\nall samples and integrating all epithelial cells using reference-based \\nintegration as described above. After integration of all cells, we subset \\nonly the enteroendocrine cells and then computed the principal com-\\nponents using RunPCA and identifyied clusters using FindNeighbors \\nand FindClusters. Known subtypes of enteroendocrine cells were then \\nannotated based on expression of marker genes (Fig.\\xa04 and Extended \\nData Fig. 6h). T o annotate the MUC6+, MUC5B+ and exocrine cells, we \\nsubset the integrated duodenum data and computed the principal com-\\nponents using RunPCA and identified clusters using FindNeighbors and \\nFindClusters. Exocrine cells were annotated on the basis of expression \\nof CELA3B and CPB1, and MUC5B+ and MUC6+ cells were annotated on \\nthe basis of expression of MUC5B and MUC6 (Fig.\\xa04h).\\xa0Comparisons in \\ncell-type abundance between regions of the intestine were done using \\nthe packages scCoda29 and Milo30.\\nPeak calling for single-nucleus data\\nPeak calling was performed with MACS2 using ArchR52. Peak sets were \\ndefined independently for epithelial cells, stromal cells and immune \\ncells. For epithelial cells, we wanted to generate a union peak set that \\ncaptured both cell-type-specific and location-specific peaks in the \\nepithelial compartment. T o accomplish this, we divided the cells into \\ngroups, generated pseudobulk replicates for each group, called peaks \\non the pseudobulk replicates, generated a reproducible peak set for \\neach group using the peaks called for the pseduobulk replicates, and \\nthen iteratively merged the peak sets for each group into a union peak \\nset using the approach implemented in ArchR. Groups used for epithe-\\nlial peak calling were the cells from each epithelial cell type—with all \\nenteroendocrine subtypes combined, and MUC5B+, MUC6+, exocrine \\nand unknown cells combined—in the four main regions of the intestine. \\nTuft cells from the small intestine were merged into a single group \\nowing to the low number of tuft cells in the dataset. For defining the \\nimmune and stromal peak sets, cells were divided by cell type, but \\nnot location, as there were fewer cells in these compartments. Finally, \\nan additional peak set for all cell types in the immune, stromal and \\nepithelial cell types was defined for determining marker peaks for \\nlinkage-disequilibrium score regression.\\nIntegration of snRNA and snA T AC data\\nT o assign snRNA profiles to the non-multiome snATAC samples, the \\nsnRNA and snATAC datasets from the four primary regions of the intes-\\ntine (duodenum, jejunum, ileum and colon) were integrated separately \\nusing the ArchR function addGeneIntegrationMatrix with reduced-\\nDims\\u2009=\\u2009“Harmony” and useMatrix\\u2009=\\u2009“GeneScoreMatrix” .\\nNomination of regulatory TFs in single-nucleus data\\nWe next identified TF regulators according to the ArchR manual for \\nidentifying TF regulators for each region, with a correlation cut-off of \\n0.5. TFs that met the criteria for regulators in any of the four primary \\nregions of the intestine are plotted in Fig.\\xa05b. This process was per-\\nformed separately for the multiome datasets and the integrated sin-\\ngleome datasets. Cell types with few cells in each region of the intestine \\n(for example, L cells) were combined into a single group regardless of \\nlocation of origin, leading to the final cell type groupings on the x-axis \\nof Fig.\\xa05b. Regulators were identified separately for cells in the immune \\nand stromal compartment with the final cell type groupings indicated \\non the x-axis of Extended Data Fig.\\xa010g,h. TF footprints were computed \\nusing the ArchR functions getFootprints and plotFootprints.\\nAnalysis of absorptive differentiation trajectories in single- \\nnucleus data\\nAbsorptive differentiation trajectories for each main section of the \\ncolon (duodenum, jejunum, ileum and colon) were inferred by running \\nthe ArchR function addTrajectory with the trajectory set as harmony \\nclusters moving from Stem to TA2 to TA1 to immature enterocytes \\nto enterocytes and reducedDims set to the harmony dimensions. T o \\nidentify variable peaks along the trajectory, a matrix of accessibility \\nin all peaks along the trajectory was first generated with getTrajec-\\ntory with useMatrix\\u2009=\\u2009“PeakMatrix” and log2Norm\\u2009=\\u2009TRUE. Peaks with \\nvariance\\u2009>\\u20090.9 in any of the four regions were then identified with the \\nfunction plotTrajectoryHeatmap with varCutOff\\u2009=\\u20090.9, returnMa -\\ntrix\\u2009=\\u2009TRUE, scaleRows\\u2009=\\u2009FALSE and maxFeatures\\u2009=\\u2009100000. The four \\nmatrices returned by getTrajectory were then concatenated into a \\nsingle matrix and the matrix was subset to include only peaks that met \\nthe variance criteria of 0.9 in at least one of the four regions and had an \\nabsolute difference in magnitude of at least 0.2. Row z-scores for the \\nresulting matrix were computed using the ArchR function .rowZscore. \\nThe resulting row z-scores were k-means clustered using the function \\nkmeans with the number of clusters set to 7 and iter.max\\u2009=\\u2009500. Two \\nclusters of peaks did not show a characteristic pattern and were not \\nincluded in Fig.\\xa06. Hypergeometric enrichment of motifs in marker \\npeaks was computed with peakAnnoEnrichment and the resulting P \\nvalues are plotted in Extended Data Fig.\\xa010k. Variable genes along the \\ntrajectory were identified with an analogous method, using GeneEx-\\npressionMatrix for the multiome data or GeneIntegrationMatrix (for \\nthe separate snATAC and snRNA data) instead of PeakMatrix when run-\\nning getTrajectory and plotTrajectoryHeatmap. For gene expression, \\nlog2Norm was set to TRUE when running getTrajectory and genes were \\nfiltered to include only those with an absolute difference in magnitude \\nof at least 0.5. Row z-scores of the resulting matrices were again k-means \\nclustered using the function kmeans with the number of clusters set to \\n7 and iter.max\\u2009=\\u2009500. Enrichment of KEGG pathways in these clusters \\nof genes was determined using the limma function kegga88, and the \\nresulting unadjusted P values are plotted in Extended Data Fig.\\xa010l. Plots \\nof gene expression versus pseudotime were generated using the ArchR \\nfunction plotTrajectory with the default parameters, including using \\nimputeWeights added by addImputeWeights. TFs with correlated motif \\nactivity and RNA expression were identified with correlateTrajectories \\nas outlined in the ArchR manual. The row z-scores of the smoothed \\nexpression of TFs along the pseudotime trajectories is plotted in Fig.\\xa06d. \\nTFs that were correlated with expression in any of the four trajectories \\nwere included in the heat map in Fig.\\xa06d. Peaks correlated with gene \\nexpression were identified with addPeak2GeneLinks. In Fig.\\xa06e, the set \\nof peaks that were correlated with ETV6 expression with a correlation \\nof at least 0.4 in one of the four main intestinal regions was determined. \\nFor TMPRSS15, a correlation cut-off of 0.55 was used to show the most \\ncorrelated peaks in the figure. The smoothed trajectory peak accessibil-\\nity for each of these peaks was then plotted along the differentiation \\ntrajectory. This process was performed separately for the multiome \\ndatasets and the integrated singleome datasets.\\nCell-type-specific linkage-disequilibrium score regression\\nLinkage-disequilibrium score regression is a method that aims to \\ndistinguish heritability from confounding factors such as popula -\\ntion stratification and cryptic relatedness. T o run cell-type-specific \\nlinkage-disequilibrium score regression, we first computed marker \\npeaks for coarse cell types in our dataset. T o do this, we added cell \\ntype annotations to the full ArchR project with all cells and then \\ndefined a peak set for this object by running addGroupCoverages \\nwith groupBy\\u2009=\\u2009“CellType” followed by addReproduciblePeakSet and \\naddPeakMatrix. We next defined less granular cell types by merging all \\nmyofibroblast clusters and pericytes into a single group, all non-villus \\nfibroblast clusters into a single group, all non-stem absorptive epi-\\nthelial cells into a single group, cycling TA cells into a single group, \\nall enteroendocrine cells except for EnteroendocrineUn into a single \\ngroup, and lymphatic endothelial and endothelial cells into a single \\ngroup. We determined marker peaks for the resulting groups of cells \\nwith getMarkerFeatures and ten selected peaks with getMarkers with ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2a30173c-d355-4682-a397-afe75ca87615', embedding=None, metadata={'page_label': '590', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\ncutOff\\u2009=\\u2009“FDR <= 0.1 & Log2FC >= 0.5” . Marker peaks from cell types with \\nvery few cells were not included (T\\xa0cells, unknown, exocrine, secretory \\nspecialized MUC5B+, DC, ILC, adipocytes, neurons, Enteroendocri-\\nneUn) The resulting peaks were then lifted over to hg19 from hg38. \\nWe then followed the linkage-disequilibrium score regression tutorial \\n(https://github.com/bulik/ldsc/wiki) for cell-type-specific analysis89–91. \\nWe used summary statistics from a number of UKBB traits (https://\\nnealelab.github.io/UKBB_ldsc/downloads.html) related to the intes-\\ntine, including non-cancer illness code;self-reported: crohns disease, \\nnon-cancer illness code;self-reported: ulcerative colitis, non-cancer \\nillness code;self-reported: malabsorption/coeliac disease, BMI as well \\nas traits less clearly related to the intestine including non-cancer illness \\ncode;self-reported: hypertension and diagnoses—main ICD10: K02 \\ndental caries. Coefficient P values from ldsc are plotted in the heat map \\nin Fig.\\xa06. Significance was determined by correcting the coefficient  \\nP values for the number of cell types tested with Bonferroni correction \\nwith the R function p.adjust.\\nProcessing of single-nucleus data for ligand–receptor analysis \\nand snRNA-CODEX integration\\nThe ligand–receptor analysis and scRNA-seq CODEX integration \\nwere performed with an initial dataset consisting of samples from \\ndonors B001, B004, B005 and B006 before collection of the remain-\\ning data. This dataset was annotated similarly to the full dataset with \\nthe following differences. First, all analysis was carried out in R v.4.0.2. \\nSecond, the quality-control cut-offs used were slightly different, with \\nthe requirement to have three times as many RNA counts as an empty \\ndroplet not implemented in the initial dataset. Third, when running \\ndoubletFinder, pK was set as 0.09 instead of running paramSweep_v3. \\nFourth, scTransform was not used in the subclustering analysis for \\nthe epithelial compartments. Instead, Seurat’s standard normalize \\nand scale pipeline followed by Harmony was run to compute an inte-\\ngrated dimensionality reduction that was then clustered using Seurat’s \\nfindNeighbors and FindClusters for the immune and stromal cells as \\nwell as the epithelial cells from the four main regions of the intestine. \\nAs the ligand–receptor analysis involved making predictions that we \\nlater attempted to validate using Molecular Cartography, this analysis \\ncould not be redone with the remaining dataset.\\nMolecular Cartography validation of ligand–receptor pairs\\nSmall intestine (duodenum) and colon (sigmoid/descending) sam-\\nples from donors B004, B008 and B009 were analysed for spatial \\ntranscriptomics. The cryosections (thickness, 10\\u2009µm) from the same \\nOCT-embedded tissue arrays were placed onto the glass slides provided \\nby Resolve Biosciences for Molecular Cartography assay. The Molec-\\nular Cartography assay was performed by the Resolve Biosciences \\nteam with their optimized protocol ‘human colon v1.3’ and targeting \\na panel of total 100 transcripts of interest, including 63 genes from our \\nligand–receptor predictions (Supplementary Table\\xa03) and 37 genes \\nfor cell type annotation (Supplementary Table\\xa09). Segmentation was \\nperformed using DAPI signal with cellPose (https://github.com/Mou-\\nseLand/cellpose; v.2.0.5) and followed by Baysor (https://github.com/\\nkharchenkolab/Baysor; v.0.5.1). Gene counts were quantified per cell \\nand cells were removed by area (<50 or >8,000 pixels) and total gene \\ncounts (<2). Manual cell type annotation was performed on the basis \\nof the marker gene expression after Leiden clustering (Scanpy). As only \\n37 genes were used as cell type markers and the sample number (n\\u2009=\\u20096) \\nis limited, only 20 cell types were identified in this dataset. T o validate \\nthe ligand–receptor repair (Supplementary Table\\xa03), we first match the \\ncell type with snRNA-seq annotation. Only 58 out of 152 predictions have \\nmatching cell types. We compared ligand and receptor expression (log \\ntransformed) between the colon and small intestine in their predicated \\ncell types (one sided Wilcoxon rank-sum test). P values were corrected \\nfor multiple testing using the Benjamin–Hochberg procedure. In total, \\n15 pairs have consistent higher expression in the colon compared with \\nin the small intestine (both adjusted P\\u2009<\\u20090.05; Supplementary Table\\xa06). \\nPermutation tests were used to assess whether the success rate (15.5%, \\n9 out of 58) of our validation is higher than random. Gene labels were \\nswapped and the same DEG procedure was repeated 10,000 times.\\nsnRNA-seq/CODEX single-cell matching and integrational \\nanalysis using MaxFuse\\nsnRNA-seq cells and CODEX cells were matched and downstream inte-\\ngrative analysis was performed using MaxFuse, of which the meth-\\nodology details were described previously41. In brief, MaxFuse is an \\nalgorithm that matches cells across different single-cell modalities by \\nlinear assignment, using both shared (when available) and unshared \\nfeatures, and implements signal boosting steps (for example, graph \\nsmoothing and meta cell construction) to enable matching cells across \\nweakly correlated modalities (for example, RNA to protein). Although \\nvarious methods are available for integration tasks on modalities with \\nrobust sharing information (such as scRNA/scATAC)84,92,93, when such \\ntasks involve integration between protein and sequencing modali-\\nties, with much weaker shared features available (<60 versus thou-\\nsands), a specialized method is needed94. We applied MaxFuse to match \\nsnRNA-seq cells to CODEX cells. Cells that were previously annotated \\nas B, T, monocyte, macrophage, plasma, goblet, endothelial, enter-\\noendocrine, smooth muscle and stromal cells were used during this \\nintegration process, whereas other cell types were not used owing to \\nlimited sharing information across modalities. Subsequently, a shared \\nco-space was calculated to embed both modalities, with visualiza-\\ntion of the embedding (first 20 MaxFuse-components) using UMAP. \\nT o evaluate single-cell matching performance across RNA to protein \\nmodality, we used cell type annotation accuracy (for example, a single \\nCODEX plasma cell matched to a snRNA-seq plasma cell) as a proxy, and \\nboth CL and small bowel matching achieved >90% accuracy. Using the \\nsingle-cell-level pairing information, we transferred the transcriptome \\nexpression profile to each individual CODEX cell, and subsequently \\nperformed analysis of the DEGs across various CODEX cellular neigh-\\nbourhoods. The DEGs were selected with the function FindAllMarkers \\nin the R package Seurat, with the parameters only.pos = TRUE, min.pct = \\n0.3, logfc.threshold = 0.25. The genes with adjusted P\\u2009<\\u20090.05 and shared \\nacross the CL and small bowel datasets were shown on the heat map. \\nGene Ontology enrichment analysis was performed for individual CNs \\nwith the DEGs, by using the PANTHER database. We have uploaded the \\ncode that we used to perform the MaxFuse matching for the snRNA-seq \\nand CODEX datasets within the paper here (https://github.com/shuxi-\\naoc/maxfuse/tree/main/Archive/hubmap_nature). This includes both \\nthe dataset preparation and analysis features of the code.\\nReporting summary\\nFurther information on research design is available in the\\xa0Nature Port-\\nfolio Reporting Summary linked to this article.\\nData availability\\nAll of the datasets (snRNA-seq, snATAC–seq and CODEX) can be visual-\\nized and accessed through a website portal (https://portal.hubmap-\\nconsortium.org/). Our landing page links all of the raw dataset IDs \\nand the HuBMAP ID for this Collection is HBM692.JRZB.356 and the \\nDOI is https://doi.org/10.35079/HBM692.JRZB.356. Supplementary \\nTable\\xa010 also lists all of the dataset IDs within the HuBMAP portal where \\nall raw datasets are stored that can be downloaded and also viewed in a \\nprocessed state. We provide the processed fluorescence CODEX mul-\\ntiplexed image stacks (https://doi.org/10.5061/dryad.76hdr7t1p and \\nhttps://doi.org/10.5061/dryad.gmsbcc2sq). We also provide processed, \\nquantified and annotated single-cell CODEX datasets with labelled \\ncell types, neighbourhoods, communities, tissue units and also pro-\\ntein expression at Dryad (https://doi.org/10.5061/dryad.pk0p2ngrf). \\nProcessed snRNA-seq and snATAC-seq datasets are available at\\xa0https://', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f42c7179-9807-46fc-a3ad-82aa94bc6472', embedding=None, metadata={'page_label': '591', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='doi.org/10.5061/dryad.8pk0p2ns8\\xa0and\\xa0 https://doi.org/10.5061/\\ndryad.0zpc8672f. Unprocessed\\xa0snRNA-seq and snATAC–seq datasets \\nare available at dbGaP (phs002272.v1.p1).\\xa0Source data are provided \\nwith this paper.\\nCode availability\\nCode for analysis of the snATAC and snRNA data is available at GitHub \\n(https://github.com/winstonbecker/scColonHuBMAP/), and code for \\nprocessing CODEX multiplexed imaging data (https://github.com/\\nnolanlab/CODEX)8, for clustering (https://github.com/nolanlab/vor-\\ntex), for transferring cell type labels with STELLAR (https://github.com/\\nsnap-stanford/stellar), for neighbourhood analysis (https://github.\\ncom/nolanlab/NeighborhoodCoordination), for tissue schematics \\n(https://github.com/nolanlab/TissueSchematics) and for MaxFuse \\n(https://github.com/shuxiaoc/maxfuse) is available at GitHub.\\n \\n66. Börner, K. et\\xa0al. Anatomical structures, cell types and biomarkers of the Human Reference \\nAtlas. Nat. Cell Biol. 23, 1117–1128 (2021).\\n67. Lee, M. Y. et\\xa0al. CellSeg: a robust, pre-trained nucleus segmentation and pixel quantification \\nsoftware for highly multiplexed fluorescence images. BMC Bioinform. 23, 46 (2022).\\n68. Han, W., Cheung, A. M., Yaffe, M. J. & Martel, A. L. Cell segmentation for \\nimmunofluorescence multiplexed images using two-stage domain adaptation and weakly \\nlabeled data for pre-training. Sci. Rep. 12, 4399 (2022).\\n69. Durkee, M. S., Abraham, R., Clark, M. R. & Giger, M. L. Artificial intelligence and cellular \\nsegmentation in tissue microscopy images. Am. J. Pathol. 191, 1693–1701 (2021).\\n70. Meijering, E. Cell segmentation: 50 years down the road. IEEE Signal Process. Mag. 29, \\n140–145 (2012).\\n71. Kulikov, V. et\\xa0al. DoGNet: a deep architecture for synapse detection in multiplexed \\nfluorescence images. PLoS Comput. Biol. 15, e1007012 (2019).\\n72. Van Valen, D. A. et\\xa0al. Deep learning automates the quantitative analysis of individual cells \\nin live-cell imaging experiments. PLoS Comput. Biol. 12, e1005177 (2016).\\n73. Moen, E. et\\xa0al. Deep learning for cellular image analysis. Nat. Methods 16, 1233–1246 \\n(2019).\\n74. Moshkov, N., Mathe, B., Kertesz-Farkas, A., Hollandi, R. & Horvath, P. Test-time augmentation \\nfor deep learning-based cell segmentation on microscopy images. Sci. Rep. 10, 5068 \\n(2020).\\n75. Hickey, J. W., Tan, Y., Nolan, G. P. & Goltsev, Y. Strategies for accurate cell type identification \\nin CODEX multiplexed imaging data. Front. Immunol. 12, 727626 (2021).\\n76. Hu, J. et\\xa0al. SpaGCN: Integrating gene expression, spatial location and histology to \\nidentify spatial domains and spatially variable genes by graph convolutional network. \\nNat. Methods 18, 1342–1351 (2021).\\n77. Partel, G. & Wählby, C. Spage2vec: unsupervised representation of localized spatial gene \\nexpression signatures. FEBS J. 288, 1859–1870 (2021).\\n78. Jiang, S. et\\xa0al. Combined protein and nucleic acid imaging reveals virus-dependent  \\nB cell and macrophage immunosuppression of tissue microenvironments. Immunity 55, \\n1118–1134 (2022).\\n79. Phillips, D. et\\xa0al. Immune cell topography predicts response to PD-1 blockade in cutaneous \\nT\\xa0cell lymphoma. Nat. Commun. 12, 6726 (2021).\\n80. Corces, M. R. et\\xa0al. The chromatin accessibility landscape of primary human cancers. \\nScience 362, eaav1898 (2018).\\n81. Leslie, T. F. & Kronenfeld, B. J. The colocation quotient: a new measure of spatial \\nassociation between categorical subsets of points. Geogr. Anal. 43, 306–326 (2011).\\n82. Ramilowski, J. A. et\\xa0al. A draft network of ligand-receptor-mediated multicellular \\nsignalling in human. Nat. Commun. 6, 7866 (2015).\\n83. Yu, A. et\\xa0al. Reconstructing codependent cellular cross-talk in lung adenocarcinoma \\nusing REMI. Sci. Adv. 8, eabi4757 (2022).\\n84. Stuart, T. et\\xa0al. Comprehensive integration of single-cell data. Cell 177, 1888–1902 (2019).\\n85. McGinnis, C. S., Murrow, L. M. & Gartner, Z. J. DoubletFinder: doublet detection in \\nsingle-cell RNA sequencing data using artificial nearest neighbors. Cell Syst. 8, 329–337 \\n(2019).\\n86. Yang, S. et\\xa0al. Decontamination of ambient RNA in single-cell RNA-seq with DecontX. \\nGenome Biol. 21, 57 (2020).\\n87. Korsunsky, I. et\\xa0al. Fast, sensitive and accurate integration of single-cell data with \\nHarmony. Nat. Methods 16, 1289–1296 (2019).\\n88. Ritchie, M. E. et\\xa0al. limma powers differential expression analyses for RNA-sequencing \\nand microarray studies. Nucleic Acids Res. 43, e47 (2015).\\n89. Bulik-Sullivan, B. K. et\\xa0al. LD Score regression distinguishes confounding from \\npolygenicity in genome-wide association studies. Nat. Genet. 47, 291–295 (2015).\\n90. Gazal, S. et\\xa0al. Linkage disequilibrium-dependent architecture of human complex traits \\nshows action of negative selection. Nat. Genet. 49, 1421–1427 (2017).\\n91. Finucane, H. K. et\\xa0al. Partitioning heritability by functional annotation using genome-wide \\nassociation summary statistics. Nat. Genet. 47, 1228–1235 (2015).\\n92. Welch, J. D. et\\xa0al. Single-cell multi-omic integration compares and contrasts features of \\nbrain cell identity. Cell 177, 1873–1887 (2019).\\n93. Cao, Z.-J. & Gao, G. Multi-omics single-cell data integration and regulatory inference with \\ngraph-linked embedding. Nat. Biotechnol. https://doi.org/10.1038/s41587-022-01284-4 \\n(2022).\\n94. Zhu, B. et\\xa0al. Robust single-cell matching and multimodal analysis using shared and \\ndistinct features. Nat. Methods 20, 304–315 (2023).\\nAcknowledgements We thank the patients for donating their tissues. This work was supported \\nby the US National Institutes of Health (3U54HG010426; P50HG007735;\\xa0U19AI057266;\\xa0 \\nU01HG011762); Cancer Research UK (C27165/A29073); the Parker Institute for Cancer \\nImmunotherapy (PICI0025); Hope Realized Medical Foundation (209477). J.W.H. was supported \\nby an NIH T32 Fellowship (T32CA196585) and an American Cancer Society—Roaring Fork Valley \\nPostdoctoral Fellowship (PF-20-032-01-CSM). S.A.N. was supported in part by the Stanford \\nGraduate Fellowship and the NHGRI Stanford Genome Training Program 5 T32 000044. C.M.S. \\nwas supported by the Swiss National Science Foundation (P300PB_171189, P400PM_183915). \\nFigs.\\xa02g,i and 4a and Extended Data Figs.\\xa01a, 2a and 6a were created using BioRender. We thank \\nthe members of the HuBMAP consortium for their many contributions to this project.\\nAuthor contributions S.B. and C.M.S. developed the original CODEX panel and acquired the \\nCODEX data from B001. J.W.H., S.B. and C.C. developed gut-specific CODEX antibodies and \\nran the remaining donor samples. V.V. helped to establish CODEX data-processing pipelines \\nand integration with HuBMAP data transfers. M.B., K.C., J.W.H. and J.L. created STELLAR,  \\nwhich was used for cell type label transfer of the CODEX data. J.W.H. processed and analysed \\nCODEX multiplexed imaging data. R.C., D.C., S.A.N., A.H. and W.R.B. established experimental \\nprotocols and collected the snRNA-seq and snATAC–seq data. A.E.P. and W.Z. analysed  \\nCODEX and snRNA-seq data for neighbourhood crosstalk. C.Z. and J.W.H. analysed the spatial \\ntranscriptomics dataset based on Molecular Cartography. B.Z., S.C., N.Z. and Z.M. integrated \\nCODEX and snRNA datasets using MaxFuse. W.R.B. analysed snATAC–seq and snRNA-seq data. \\nW.R.B., J.W.H., W.J.G. and M.S. wrote the manuscript and created figures. D.C. handled data \\nmanagement and submission to repositories. A.H., A.K.W. and E.M. were involved in project \\ncoordination. T.L. assessed morphological staining of samples. E.D.E. assisted in metadata \\nannotation and biological interpretation. Y.L., Z.Z. and S.L. procured samples. Y.L., S.K.P., \\nG.P.N., W.J.G. and M.S. supervised the study. All of the authors revised the manuscript and \\naccepted its final version.\\nCompeting interests C.M.S. is a scientific advisor to, has stock options in and has received \\nresearch funding from Enable Medicine. 10X Genomics holds the license to patents in which \\nW.J.G. is listed as an inventor. W.J.G. is an equity holder of 10X Genomics, and a co-founder of \\nProtillion Biosciences. W.J.G. consults for Guardant Health, Quantapore, Protillion Biosciences, \\nUltima Genomics, Lamar Health, and Erdio Biosciences.\\xa0M.P.S. is a cofounder and an advisory \\nboard member of Personalis, Qbio, January AI, Mirvie, Filtricine, Fodsel, Protos, RTHM, Marble \\nTherapeutics and Crosshair Therapeutics. G.P.N. received research grants from Pfizer, Vaxart, \\nCelgene and Juno Therapeutics; and has equity in and is a scientific advisory board member of \\nAkoya Biosciences. The other authors declare no competing interests.\\nAdditional information\\nSupplementary information The online version contains supplementary material available at \\nhttps://doi.org/10.1038/s41586-023-05915-x.\\nCorrespondence and requests for materials should be addressed to Garry P. Nolan,  \\nWilliam J. Greenleaf or Michael Snyder.\\nPeer review information Nature thanks Shalev Itzkovitz, Caroline Wählby and the other, \\nanonymous, reviewer(s) for their contribution to the peer review of this work. Peer reviewer \\nreports are available.\\nReprints and permissions information is available at http://www.nature.com/reprints.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a53dc82d-f3ca-4ec3-818d-112e540e8613', embedding=None, metadata={'page_label': '592', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 1 | CODEX mulitplexed imaging across the healthy \\nhuman intestine reveals changes in cell composition and organization.   \\nA) Schematic for how CODEX multiplexed imaging was performed on arrays of \\n4 different sections of either colon and small intestine from the same donor \\nsimultaneously. Image processing steps done to extract single-cell spatial data. \\nB) An example CODEX fluorescent image of one region of the small bowel (SB) \\n(1 of 64 tissue sections) for CODEX with 6/54 markers shown for one donor \\n(scale bar = 1 mm and magnified insert = 100 µm) with C) accompanying cell \\ntype map following cell segmentation and unsupervised clustering. D-E) Stromal \\ncell type percentages either as a percent of D) All stromal cells, or E) all cells \\nrestricted to the Muscularis Externa tissue unit. F) Immune cell type or G) \\nepithelial cell type percentages either as a percent of all cells restricted to  \\nthe Mucosa  tissue unit. H) Percentage of CD57+ Enterocyte cells of all cell  \\ntypes across different areas samples from small intestine to colon. (for D-H:  \\n* p value< 0.05, ** p value< 0.01, *** p value < 0.001 by two-sided T test,  \\nn=8 donors). (All boxplots in figures are plotted as minimum, 25 percentile, \\nmedian, 75 percentile, maximum, and outliers as points outside 1.5 the \\ninterquartile range). I) Cell map of a representative section (one of 8 donors)  \\nof the Duodenum that shows CD57+ Enterocyte presence in glands in the \\nSubmucosa  where Enterocytes and TA cells are shown in dark grey and Smooth \\nmuscle cells in light grey (other cell types not shown) (scale bar = 500 µm).  \\nJ-K) Quantification of the same-cell density that is measured as an average \\ndistance of its 5 nearest same-cell neighbours normalized by the maximal \\npossible same-cell distance within the tissue (n\\u2009=\\u200964 tissue sections) for J) \\nstromal and K) epithelial cell types. L) A representative cell type map (one of  \\n64 tissue sections from 8 donors) with only plasma cells, CD8+ T\\xa0cells, and  \\nM2 Macrophages shown (scale bar = 500 µm).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d33451ef-7fd0-4471-b169-dbd90f0d0bc5', embedding=None, metadata={'page_label': '593', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 2 | Multicellular neighbourhood analysis of CODEX \\nimaging data indicates conserved cellular structures across the intestine.  \\nA) Neighbourhood analysis was done by taking a window across cell type  \\nmaps and vectorizing the number of cell types in each window, clustering \\nwindows, and assigning clusters as cellular neighbourhoods of the intestine. \\nB-D) Neighbourhood percentages from CODEX data averaged normalized by  \\nB) stromal, C) immune, and D) epithelial compartments. Asterix indicates \\np-value less than 0.05 difference in cell type percentage from the small bowel \\n(SB) to the colon (CL) by two-sided T test. E) Stromal multicellular neighbourhood \\npercentages either as a percent of all neighbourhoods restricted to the \\nMuscularis Externa tissue unit (* p value< 0.05, *** p value < 0.001, n\\u2009=\\u20098 donors, \\nby two-sided T test,). F) Quantification of the same-cell density for just smooth \\nmuscle cells within different smooth muscle multicellular neighbourhoods  \\n(x axis) (n\\u2009=\\u200932 tissue sections). G-H) Cell type maps for a region of the small \\nintestine (one of 64 tissue sections imaged from 8 donors) with G) all cell types \\nplotted for the whole tissue (scale bar = 500 µm), H) cells contained within  \\nthe plasma cell neighbourhood (scale bar = 500 µm), and a magnified area of \\ndenoted by rectangle showing subset of cell types (scale bar = 50 µm).  \\nI) CODEX fluorescent imaging with subset of fluorescent markers overlaid for \\nthe same tissue as G (Hoechst=Blue, CD4=Green, CD68=magenta, CD38=yellow, \\nCD206=cyan, CD138=grey), (scale bar = 500 µm with magnified insert  \\nscale bar = 100 µm).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='db505664-67af-40f1-8756-5b6e3ca79729', embedding=None, metadata={'page_label': '594', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 3 | Change in multicellular neighbourhoods across the \\nintestine.  A) Immune neighbourhood percentages as a percent of immune \\nneighbourhoods. B) Neighbourhood percentages for Microvasculature  and \\nCD8+ T\\xa0cell IEL neighbourhoods compared for donors with or without a history \\nof hypertension (Microvasculature p value\\u2009=\\u20090.0065<, CD8+ T Enriched IEL  \\np value = 0.0017 by two-sided T test, n\\u2009=\\u20093-5 donors). C) Difference in composition \\nin neighbourhood by cell type for all neighbourhoods based on subtracting  \\nthe log2 fold enrichment of each cell type found within that neighbourhood \\ncompared to average percentages in the tissue in SB from CL. Neighbourhoods \\nand cell types are ordered by summing the absolute value of all rows and columns \\nto denote conservation of a neighbourhood. D) 22 unique intestinal multicellular \\nneighbourhoods (y axis of heatmap) were defined by enriched cell types  \\n(x axis of heatmap) as compared to overall percentage of cell types in the \\nsamples with 2 unique neighbourhoods not identified with overall neighbourhood \\nanalysis. E) Epithelial neighbourhood percentages as a percent of epithelial \\nneighbourhoods from multicellular analysis performed on each individual \\nregion of the intestine separately. F) Immune neighbourhood percentages  \\nas a percent of immune neighbourhoods from multicellular analysis performed \\non each individual region of the intestine separately. (* p value< 0.05,  \\n** p value< 0.01, *** p value < 0.001 by two-sided T test, n\\u2009=\\u20098 donors).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='7ff9ffbe-a985-4049-978e-d1ef2bebcebf', embedding=None, metadata={'page_label': '595', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 4 | Multi-neighbourhood community analysis of  \\nthe intestine.  A) Community analysis was done by taking a window across \\nneighbourhood maps and vectorizing the number of each neighbourhood \\ntype in each window, clustering windows, and assigning clusters as multi- \\nneighbourhood communities of the intestine. 10 unique intestinal multi- \\nneighbourhood communities (y axis of heatmap) were defined by enriched \\nneighbourhood types (x axis of heatmap) as compared to overall percentage of \\nneighbourhood types in the samples. B) Quantification of neighbourhood \\ntypes across each section of the intestine (colour legend within panel A).  \\nC) Community percentages as a percent of all communities for small intestine \\nand colon (* p value< 0.05, ** p value< 0.01, *** p value < 0.001, n\\u2009=\\u20098 donors).  \\nD) Concentric multi-neighbourhood analysis surrounding only neighbourhoods \\nlabelled as Paneth Cell Enriched  neighbourhoods, with number of nearest \\nneighbours for a given row in the heatmap.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5e416937-ab79-4437-81cb-fd111acc0ce7', embedding=None, metadata={'page_label': '596', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 5 | Multicellular neighbourhood and community \\ninteractions across the intestine.  A) Schematic of a representative community \\nmap and corresponding tissue community network graph (one of 64 tissues). \\nB-C) Community-community motifs that are significantly enriched in both the \\nsmall intestine and colon as compared to a null distribution of motif instances \\ncreated from random permutation of tissue graph labels, where B) shows only \\nthose motifs that interact with the Adaptive Immune Enriched community and \\nC) shows all other shared motifs between the SB and CL. Motifs indicated by \\nshape and colour indicate those motifs that have significant p value versus those \\nthat are indicated with an x in the graph; p values were Bonferroni corrected by \\nmultiplying by twice the number of tests conducted in each comparison group. \\nColour legend is also the same as panel B. D) Representative neighbourhood \\nmap (one of 64 tissue sections from 8 donors) with (scale bar = 500 µm)  \\nE) Region magnified as in the main figure of the mucosal area of a colon \\ncommunity map, but this time with the multicellular neighbourhoods coloured \\n(see panel D for legend) (scale bar = 100 µm). F) Spatial context maps of the CL \\nhighlighting relationships of multicellular neighbourhoods across just the \\nneighbourhoods found within the tissue unit Mucosa . This structure is defined \\nby the number of unique neighbourhoods required to make up at least 85% in a \\ngiven window. Circles represent the number of cells represented by a given \\nstructure. Red rectangle highlights a structure discussed in the manuscript \\nand maps this structure back to panel K. Colour legend is also the same as panel \\nD. G-H) Cell type percentage of G) epithelial and H) stromal cells shown for each \\ncommunity ordered in relative order of general increasing proximity to the \\nlumen based on community spatial context analysis.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a690d580-f626-4c46-855f-fa49aa09cce3', embedding=None, metadata={'page_label': '597', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='a\\nb\\nc\\nf\\ned\\nTSSEnrichment\\nRNA Counts/Cell\\nDuodenum\\nProx-jejunum\\nIleum\\nAscending\\nTransverse\\nDescending\\nSigmoid\\nMid-Jejunum\\nB001\\nB004\\nB005\\nB006\\nB008\\nB009\\nB010\\nB011\\nB012\\nAdipocytes\\nNeurons\\nGlia\\nICC\\nLymphatic Endothelial Cells\\nEndothelial−CD36+ Microvascular\\nEndothelial−Venules\\nPericytes\\nMyofibroblasts/SM DES High\\nMyofibroblasts/SM 1\\nMyofibroblasts/SM 2\\nMyofibroblasts/SM 3\\nVillus Fibroblasts WNT5B+\\nCrypt Fibroblasts 1 WNT2B+\\nCrypt Fibroblasts 2\\nCrypt Fibroblasts 3 RSPO3+\\nPLIN\\n1\\nLP\\nL\\nSYPSYT1\\nMIR124−2HG\\nRBFO\\nX1\\nSOX10CDH19PLP1S100B\\nKIT\\nANO1PR\\nOX1RELNCD36\\nPECAM1\\nSELP\\nMADCAM\\n1\\nNO\\nTCH3MCAMRGS5TAGLNACTA2\\nMYH11\\nDES\\nCOL6A1COL6A2WNT5BBMP5BMP4BMP2\\nPDGFRA\\nCCL8CCL13CCL19\\nAD\\nAMDEC1CCL1\\n1\\nKCNN\\n3\\nWNT2BRSPO3GREM1SEMA3E\\nC3\\n−1\\n0\\n1\\n2\\nAverage\\nExpression\\nPercent\\nExpressed\\nAverage\\nExpression\\nPercent\\nExpressed\\n25\\n50\\n75\\nCD4\\nCD8\\nT Cells\\nNK\\nILC\\nB Cells\\nPlasma\\nCycling\\nImmune\\nMast\\nDC\\nMono/\\nMacrophages\\nCD3DCD3ECD4CD8BNKG7NCAM1TNFSF11\\nIL23\\nR\\nIL4I\\n1\\nLST1PAX5IGLL5AMPD1TOP2AMKI67CDK1HDCGA\\nTA2\\nITGA\\nX\\nL Y75CD14\\n0\\n1\\n2\\n20\\n40\\n60 B005 T Cells\\nUMAP Dimension 2\\nUMAP Dimension 1\\nUMAP Dimension 2\\nUMAP Dimension 1\\nsnRNA Immune Cells\\nsnRNA Immune Marker Genes – All Samples\\nsnRNA Stromal Marker Genes – All Samples\\n1000\\n3000\\n10000\\nB001−A−001 (n = 5651)B001−A−006 (n = 4156)B001−A−101 (n = 4163\\n)\\nB001−A−201 (n = 4682\\n)\\nB001−A−3\\n01 (n = 3523)\\nB001−A−401 (n = 2037)B001−A−406 (n = 2328)B001−A−\\n501 (n = 2543\\n)\\nB004−A−004 (n = 3276)B004−A−008 (n = 4759)B004−A−104 (n = 2906)\\nB004−A−104−R2 (n = 4791)\\nB004−A−204 (n = 4504)B004−A−304 (n = 2014\\n)\\nB0\\n04−A−404 (n\\n = 6153)\\nB004−A−404−R2 (n = 4852\\n)\\nB004−A−40\\n8 (n = 4548)\\nB004−A−408−R2 (n \\n= 4291)\\nB004−A−504 (n = 5187)\\nB00\\n4−A−504−R2 (n = \\n4935)\\nB005−A−001 (n \\n= 401)\\nB005−A−002 (n = 55\\n74)\\nB005−A−101 (n = 2476\\n)\\nB005−A−201 (n = 1546)B005−A−301 (n = 7121\\n)\\nB005−A−401 (n = 3190\\n)\\nB005−A−40\\n2 (n = 3541)\\nB005−A−501 (n = 5411)\\nB005−A−501−R2 (n = 5406\\n)\\nB006−\\nA−001 (n = 1020)\\nB006−A−002 (n = 1659\\n)\\nB006−A−101 (n = 529)\\nB006−A−201 (n = 352\\n)\\nB006−A−201−R2 (n = 16\\n01)\\nB006−A−301 (n = 2271)B006−A−401 (n = 1414\\n)\\nB006−A−\\n402 (n = 832)\\nB006−A−501 (n = 3604\\n)\\nB0\\n08−A−001\\n (n = 5090)\\nB008−A−002 (n = 575\\n3)\\nB008−A−101 (n \\n= 318)\\nB008−A−20\\n1 (n = 4\\n021)\\nB008−A−\\n301 (\\nn = 6285\\n)\\nB008−A−\\n401 (n = 5180)\\nB00\\n8−A−402\\n (n = 6077\\n)\\nB008\\n−A\\n−501 (n = 406\\n1)\\nB009\\n−A−\\n001 (n \\n= 138\\n7)\\nB009−A−\\n101 (n\\n = 3834\\n)\\nB009−A−\\n301 (n\\n = 3396)\\nB00\\n9−A\\n−401\\n (n = 29\\n44)\\nB009\\n−A−\\n405 (n = 376\\n2)\\nB01\\n0−\\nA−\\n001 (n\\n = 392\\n4)\\nB0\\n10\\n−A−00\\n2 (n = 46\\n37)\\nB0\\n10−A−\\n101\\n (n = \\n4493)\\nB010\\n−A\\n−201 (n = 883\\n6)\\nB010\\n−A−\\n301 (n = 388\\n8)\\nB01\\n0−A−401\\n (n = 3891)\\nB01\\n0−A−\\n405 (n\\n = 51\\n15)\\nB01\\n0−\\nA−\\n501 (\\nn = 71\\n24)\\nB011\\n−A−001 (n\\n = 2078\\n)\\nB011\\n−A−002 \\n(n = 2\\n900)\\nB011\\n−A−101 (n \\n= 2\\n745)\\nB011−A−\\n201 (n = 4504\\n)\\nB011−A−\\n30\\n1 (n = \\n1871)\\nB0\\n11−A−401\\n (n = \\n20\\n66\\n)\\nB011−A−\\n405 (n = 1954\\n)\\nB01\\n1−\\nA−501 \\n(n = \\n61\\n0)\\nB012−A−\\n001 \\n(n = 5386)\\nB01\\n2−\\nA−002\\n (n = 4912)\\nB012−A−\\n101 (n = 5764)\\nB012−A−\\n201 (\\nn = 3333\\n)\\nB01\\n2−A\\n−301\\n (n = 26\\n58)\\nB012\\n−A−\\n401 (n\\n = 530\\n2)\\nB01\\n2−\\nA−\\n405\\n (n = 48\\n65)\\nB012−A−\\n50\\n1 (n = \\n5900)\\n8\\n12\\n16\\nColon Epi.\\nIleum Epi.\\nDuodenum/Jejunum Epi.\\nAll snRNA Cells - Harmony Reduction\\nB001−A−501 (n=3147)B001−A−406 (n=3560)B001−A−401 (n=6226)B001−A−302 (n=\\n6041)\\nB001−A−301 (n=5591)B001−A−201 (n=3741)B001−A−101 (n=1841)B001−A−006 (n=4732)B001−A−001 (n=2406)B004−A−504 (n=3752\\n)\\nB004−A−408 (n=3306)B004−A−404 (n=1750)\\nB004−A−404−R2 (n=1053)\\nB004−A−304 (n=1662)B004−A−2\\n04 (n=2521\\n)\\nB004−A−008 (n=336\\n6)\\nB004−A−004 (n=529)\\nB004−A−004−R2 (n=4173)\\nB005−A−5\\n01 (n=7307\\n)\\nB005−A−402 (n=3160\\n)\\nB005−A−401 (n=4666)B005−A−301 (n=4734)B005−A−201 (n=841)B005−A−101 (n=1450\\n)\\nB005−A−002 \\n(n=2761\\n)\\nB005−A−001 (n=3348\\n)\\nB006−A−501 (n=3604\\n)\\nB006−A−402 (n=832\\n)\\nB006−A−401 (n=1414)B006−A−301 (n=2271)B006−A−2\\n01 (n=352\\n)\\nB006−A−201−R2 (n=1\\n601)\\nB00\\n6−A−101 (n=529\\n)\\nB006−A−002 (n=165\\n9)\\nB006−A−001\\n (n=1020)\\nB008−A−501 (n=4061\\n)\\nB008−A−402 (n\\n=60\\n77\\n)\\nB008−A−401\\n (n=518\\n0)\\nB008−A\\n−3\\n01 (n=6285)\\nB0\\n08−\\nA−201\\n (n=4021)\\nB008−A−101 (n\\n=318\\n)\\nB008−\\nA−\\n002 (n\\n=5753)\\nB008−\\nA−001 (n=5090)\\nB009−A−405 (n\\n=3762\\n)\\nB009−\\nA−101 (n\\n=3834\\n)\\nB010−\\nA−5\\n01 (n=7124)\\nB010−A−405 (n\\n=5115)\\nB010\\n−A−401 (n\\n=3891)\\nB010−\\nA−\\n301 (n\\n=3888\\n)\\nB010−A−201 (n\\n=8836)\\nB010−A−101 (n=4493)B01\\n0−A−002 (n=\\n4637\\n)\\nB010−A−001 (n=3924\\n)\\nB011\\n−A−501 (n=610)\\nB0\\n11−\\nA−405 (n=1954)\\nB0\\n11−\\nA−401 \\n(n=2066)\\nB0\\n11−\\nA−3\\n01 (n=1871)\\nB011−\\nA−20\\n1 (n=\\n45\\n04)\\nB011−A−101 (n=2745\\n)\\nB011−A−002 (n\\n=2900)\\nB011\\n−A−001 \\n(n=\\n2078)\\nB0\\n12−\\nA−501 \\n(n=5900)\\nB012−\\nA−405 (n\\n=4865)\\nB012−A−401 (n\\n=5302\\n)\\nB012−A−301 (n\\n=2658)\\nB012−\\nA−2\\n01 (n\\n=3333)\\nB012−A−101 (n\\n=5764)\\nB0\\n12−\\nA−002 \\n(n=491\\n2)\\nB012\\n−A−001 (n\\n=5386)\\nhi\\nUMAP Dimension 1\\ng\\nL Cells\\nL Cells NEUROG3high\\nNEUROG3high\\nS Cells\\nS Cells\\nMo CellsMo\\nK Cells\\nK Cells\\nI Cells\\nI Cells\\nD Cells\\nD Cells\\nEnterochromaffin\\nEnterochromaffin\\nEnteroendo. Un1\\nEnteroendo. Un\\nEnteroendo. Un1\\nEnteroendo. Un\\nSCGNCHGB\\nNEUR\\nOD1CHGATPH1SSTCCK\\nONECUT3\\nGIP\\nMLNGHRLSCT\\nPR\\nOX1\\nNEUR\\nOG3\\nSMOC2\\nGCGPYYINSL5\\n−2\\n−1\\n0\\n1\\n2\\nAverage \\nExpression\\nPercent \\nExpressed\\n25\\n50\\n75\\nB001−A−201\\nB004−A−504\\nB004−A−504−2\\nB005−A−501\\nB005−A−501−2\\nB006−A−501\\nB008−A−501\\nB010−A−501\\nB011−A−501\\nB012−A−501\\nUMAP Dimension 2\\nExtended Data Fig. 6 | Quality control and clustering of single-nucleus data.  \\n(a,b) Violin plots of TSS enrichment (a) and RNA counts/cell (b) for different \\nsamples included in the study. Samples are coloured by the location from which \\nthey were obtained. (c) UMAP projection of all snRNA cells coloured by location. \\n(d) Dotplot representation of expression of immune marker genes by immune \\ncell types. (e) UMAP projection of scRNA immune cells coloured by donor.  \\n(f) Dotplot representation of expression of stromal marker genes by stromal cell \\ntypes. g) Sub-clustering of enteroendocrine cells from all regions of the intestine. \\nh) Dotplot representation of the expression of subtype specific enteroendocrine \\nand enterochromaffin marker genes in different enteroendocrine cell types in \\nour datasets. I) Sub-clustering of specialized secretory cells coloured by sample.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='04fbe130-30d5-4d9d-9c7b-c57541f08928', embedding=None, metadata={'page_label': '598', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nCD4\\nCD8\\nT Cells\\nNK\\nILC\\nB Cells\\nPlasma\\nCyclingImmune\\nMast\\nDC\\nMonocytes/Macrophages\\nMixed\\n−5 05 −5.0 −2.50 2.5 −5.0 −2.5 02 .5 −6 −3 03 6\\nNeurons\\nGlia\\nICC\\nLymphatic Endothelial\\nEndothelial−CD36+\\nEndothelial−Venules\\nPericytes\\nMyofib./SM DES High\\nMyofib./SM 1\\nMyofib./SM 2\\nMyofib./SM 3\\nVillus Fib. WNT5B+\\nCrypt Fib. 1 WNT2B+\\nCrypt Fib. 2\\nCrypt Fib. 3 RSPO3+\\nMixed\\n−2.50 2.5 5.0 −2.50 2.5 5.0 −2.5 02 .5 5.07 .5 −2.5 02 .5 5.0\\nEnriched in \\nColon\\nEnriched in\\nDuodenum\\nEnriched in \\nColon\\nEnriched in\\nProximal-Jejunum\\nEnriched in \\nColon\\nEnriched in\\nMid-Jejunum\\nEnriched in \\nColon\\nEnriched in\\nIleum\\nLog Fold Change\\na\\nb\\nc\\nCyclingTA 1\\nStem\\nTA2\\nTA1\\nImmature Enterocytes\\nEnterocytes\\nEnterocytes 2\\nBest4+ Enterocytes\\nImmature Goblet\\nGoblet\\nEnteroendocrine\\nTuft\\nSec. Spec.\\nMixed\\n−3 03 −5.0 −2.5 0.0 2.55 .0 −3 03 6 −5.0 −2.5 0.02 .5 5.0\\nAdipocytes\\nEnteroendocrineUn 1\\nExocrine\\nI Cells\\nMo Cells\\nILC\\nD Cells\\nK Cells\\nNEUROG3high\\nL Cells\\nDC\\nS Cells\\nUnknown\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1\\nColon (n=34)\\nIleum (n=9)\\nJejunum (n=18)\\nDuodenum (n=8)\\nGlia\\nMyofib./SM 3\\nNeurons\\nVillus Fib. WNT5B+\\nLymphatic Endothelial\\nPericytes\\nSec. Specialized MUC5B+\\nMyofib./SM 2\\nICC\\nNK\\nCyclingImmune\\nCyclingTA 2\\nMast\\nEnterochromaffin\\n0\\n1\\n2\\n3\\n4\\n5\\nPaneth\\nMyofib./SM DES High\\nT Cells\\nCD8\\nCrypt Fib. 2\\nPlasma\\nCrypt Fib. 3 RSPO3+\\nSec. Specialized MUC6+\\nMonocytes/Macrophages\\nCD4\\nCrypt Fib. 1 WNT2B+\\nBest4+ Enterocytes\\nTuft\\nEnteroendocrineUn\\nStem\\nEndothelial\\n0\\n5\\n10\\n15\\n20\\nAbsorptive\\nMyofib./SM 1\\nB Cells\\nEnterocytes 2\\nCyclingTA 1\\nGoblet\\n0\\n20\\n40\\n60\\n80\\nSI vs Colon\\nDuodenum vs Colon\\nJejunum vs Colon\\nIleum vs Colon\\nDuodenum vs Ileum\\nJejunum vs Ileum\\nStem\\nCyclingTA 1\\nCyclingTA 2\\nAbsorptive\\nEnterocytes 2\\nBest4+ Enterocytes\\nGoblet\\nTuft\\nPaneth\\nUnknown\\nSec. Specialized MUC6+\\nExocrine\\nSec. Specialized MUC5B+\\nEnteroendocrineUn\\nEnteroendocrineUn 1\\nEnterochromaffin\\nNEUROG3high\\nL Cells\\nI Cells\\nMo Cells\\nD Cells\\nK Cells\\nS Cells\\nAdipocytes\\nNeurons\\nGlia\\nICC\\nLymphatic Endothelial\\nEndothelial\\nPericytes\\nMyofib./SM DES High\\nMyofib./SM 1\\nMyofib./SM 2\\nMyofib./SM 3\\nVillus Fib. WNT5B+\\nCrypt Fib. 1 WNT2B+\\nCrypt Fib. 2\\nCrypt Fib. 3 RSPO3+\\nCD4\\nCD8\\nNK\\nILC\\nB.Cells\\nPlasma\\nCyclingImmune\\nMast\\nDC\\nMonocytes/Macrophages\\n−4\\n−2\\n0\\n2\\n4Duodenum vs Jejunum\\n−4\\n−2\\n0\\n2\\n4\\nLog2 Fold-Change\\nd\\ne\\nExtended Data Fig. 7 | Differential cell type abundance.  a–c) Beeswarm  \\nplot showing the log-fold change between the three main regions of the small \\nintestine and colon for groups of nearest neighbour cells from different cell \\ntype clusters in the stromal (a), immune (b), and epithelial (c) compartments \\ncomputed with Milo. Significant changes are indicated in colour. d) Boxplots \\ncomparing the fraction of all cells in each sample composed of each cell type \\nfor samples from the colon, ileum, jejunum, and duodenum. e) Log2FC in \\nabundance of each cell type between the regions listed on the y-axis as \\nestimated with scCODA. Only significant results at an FDR of 0.05 are shown, \\nwith all nonsignificant differences plotted as white.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4c11b843-06b1-438b-bdc2-45fcee9e4e79', embedding=None, metadata={'page_label': '599', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='dc\\n dEnterocytes Plasma-\\nEnterocytesPlasma-\\nTA Plasma-\\nTAPlasma -\\nEnterocytesSmooth Muscle -\\nEnterocytes Smooth Muscle-\\nEnterocytes DC-\\nEnterocytesDC -\\n–log10(p)\\nlog2FC\\n02–2\\n0\\n1\\n2\\nMore co-localized \\nin CL than in SB \\nSEMA4D\\n0\\n1\\n2\\n3\\n4\\nColon Small\\nBowel\\nGene Expression\\nMET\\n0\\n1\\n2\\n3\\n4\\n5\\nSmall\\nBowel\\nColon\\np<2.22e-16p=1.1e-07\\nB3GAT1/CD57\\nUMAP 1\\nCEACAM1/CD66a\\n CEACAM6/CD66c\\n CEACAM3/CD66d\\n CEACAM5/CD66e\\nB3GAT1\\nCEACAM1\\nCEACAM6\\nCEACAM3\\nCEACAM5\\nCyclin\\ngTA 1−Colon\\nCyclingT\\nA 2−Colon\\nCyclingT\\nA 1−Duodenu\\nm\\nCyclingT\\nA 1−Ileum\\nCycling\\nTA 1−Jeju\\nnum\\nStem−Colo\\nn\\nStem−Duodenu\\nm\\nStem−Ileum\\nStem−Jejunu\\nm\\nTA\\n2−Colon\\nTA2−Duode\\nnum\\nTA2−Ileum\\nTA2−Jejun\\num\\nTA1−Colon\\nTA1−Duode\\nnum\\nTA\\n1−Ileum\\nTA1−\\nJej\\nunum\\nImmature Enterocytes−Colon\\nImmature Enterocytes−Duodenum\\nImmature Enterocytes−Ileum\\nImmature Enterocytes−Jej\\nunum\\nEnterocytes−Colo\\nn\\nEnterocytes−Duode\\nnum\\nEnterocytes−Ileum\\nEnterocytes−Jejun\\num\\nEnterocytes 2−Duoden\\num\\nEnterocytes 2−Ileum\\nEnterocytes 2−Jejunu\\nm\\nBest4+ Enterocytes−Colon\\nBest4+ Enterocytes−Duode\\nnum\\nBest4+ Enterocytes−Ileu\\nm\\nBest4+ Enterocytes−Jejun\\num\\nImmature Go\\nblet−Colon\\nImmature Gob\\nlet−Duode\\nnum\\nImmature Go\\nblet−Ileum\\nImmature Go\\nblet−Jejun\\num\\nGo\\nblet−Colo\\nn\\nGob\\nlet−Duodenu\\nm\\nGo\\nblet−Ileum\\nGob\\nlet−Jejun\\num\\nPaneth−Duode\\nnum\\nPaneth−Ileum\\nPaneth−Jeju\\nnum\\nTuft−Colon\\nTuft−Duoden\\num\\nTuft−Ileu\\nm\\nTuft−Jejunum\\nEnteroendo\\ncrine−Colon\\nEnteroendoc\\nrine−Duoden\\num\\nEnteroendoc\\nrine−Ileum\\nEnteroendoc\\nrine−Jej\\nunum\\nSecretor\\ny Speciali\\nzed MUC6+−Duod\\nenum\\nUnkno\\nwn−Duodenu\\nm\\nPercent\\nExpressed\\n20\\n40\\n60\\n80\\n−0.5\\n0.0\\n0.5\\n1.0\\n1.5\\n2.0\\nAverage\\nExpression\\nUMAP 2\\nDuodenumJejunumIleumColon\\n0M ax\\nGene\\nExpression\\na\\nb\\ne\\nDC-EnterocyteEnterocyte-DC\\nEnterocyte-Plasm\\na\\nPlasma-Enterocyte\\nPlasma-T\\nA\\nTA-Plasma\\nEnterocyte-Smoth muscl\\ne\\nSmoth muscle-Enterocyte\\n0\\n5\\n10\\n15CLQ\\nDuodenum\\nProx-jejunum\\nIleum\\nAscending\\nTransverse\\nDescending\\nSigmoid\\nMid-Jejunum\\nLocation\\nExtended Data Fig. 8 | Pairing of CODEX multiplexed imaging and snRNA-seq \\nfor cell-cell colocalization analysis. a) Expression of B3GAT1 and four CEACAM \\ntranscripts plotted on the UMAP manifold of epithelial cells from the duodenum, \\njejunum, ileum, and colon. b) Dotplot representation of the expression of B3GAT1 \\nand four CEACAM transcripts by different epithelial cell types in different regions \\nof the intestine. c) Large colon (CL) and small bowel (SB) show differences in cell- \\ncell co-localization patterns; annotated cell-pairs are more colocalized in the \\ncolon compared to the small bowel (Student’s T test, two-sided, corrected  \\nfor multiple hypothesis testing with the Benjamini Hochberg procedure).  \\nd) SEMA4D ligand expression in plasma cells and MET receptor gene expression \\nin TA2 cells, showing higher expression in colon than small bowel (one-sided \\nWilcoxon Rank-Sum Test). e) Differences in pairwise cell-type colocalization \\npatterns across tissue locations (n\\u2009=\\u20093 samples for each location).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a79f0cb0-b1e2-41cb-8548-dbeafcd6a44f', embedding=None, metadata={'page_label': '600', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nSmall\\nBowel\\nSmall\\nBowel\\nColon Colon\\nExpression\\nExpression\\n0.0\\n0.5\\n0.1\\n0.2\\n0.3 1.0\\ne\\ng\\nac d\\nb\\n0\\n1\\n2\\nColon\\n(n=8,239)\\nSmall\\nBowel\\n(n=6,208)\\nColon\\n(n=3,493)\\nSmall\\nBowel\\n(n=8,804)\\nFN1 Myofibroblast\\nlog10(RNA expression)\\n0\\n1\\nPLAUR EnterocyteTransverseC olon\\nCODEX Neighborhood Enriched GenesAdaptive Immune Enriched\\nCD8+ T Enriched IEL\\nInnate Immune Enriched\\nInner Follicle\\nInnervated Smooth Muscle\\nMacrovasculature\\nMature Epithelial\\nMicrovasculature\\nOuter Follicle\\nPaneth Enriched\\nPlasma Cell Enriched\\nSecretory Epithelial\\nSmooth Muscle\\nSmooth Muscle & Innate Immune\\nStroma\\nStroma & Innate Immune\\nTransit Amplifying Zone\\nInnervated Stroma\\nSmall Bowel\\nColon\\nDifferentially Expressed Genes\\nNormalized RNA\\nExpression\\nDifferentially Expressed Genes\\nInner Folliclle GO Enrichment Outer Folliclle GO Enrichment\\nTolerance induction to self antigen\\nNegative regulation of mast cell proliferation\\nPositive regulation of dendritic cell apoptotic process\\nNegative regulation of B cell proliferation\\nRegulation of B cell receptor signaling pathway\\nNegative regulation of B cell activation\\nRegulation of B cell proliferation\\nNegative regulation of lymphocyte proliferation\\nNegative regulation of mononuclear cell proliferation\\nB cell receptor signaling pathway\\nRegulation of B cell differentiation\\nRegulation of erythrocyte differentiation\\nRegulation of antigen receptor-mediated signaling pathway\\nNegative regulation of leukocyte proliferation\\nNegative regulation of lymphocyte activation\\nB cell activation\\nNegative regulation of leukocyte activation\\nRegulation of mononuclear cell proliferation\\nRegulation of leukocyte proliferation\\nB cell differentiation\\nNegative regulation of cell activation\\nRegulation of lymphocyte proliferation\\nAntigen receptor-mediated signaling pathway\\nRegulation of B cell activation\\nRegulation of lymphocyte differentiation\\nMononuclear cell differentiation\\nImmune response-activating cell surface receptor signaling pathway\\nImmune response-activating signal transduction\\nLymphocyte differentiation\\nLeukocyte differentiation\\nPositive regulation of B cell receptor signaling pathway\\nRegulation of establishment of T cell polarity\\nPositive regulation of antigen receptor-mediated signaling pathway\\nNegative regulation of T cell apoptotic process\\nPositive T cell selection\\nT cell selection\\nNegative regulation of lymphocyte apoptotic process\\nT cell proliferation\\nPositive regulation of B cell proliferation\\nRegulation of antigen receptor-mediated signaling pathway\\nRegulation of B cell differentiation\\nRegulation of B cell proliferation\\nRegulation of T cell apoptotic process\\nT cell differentitation of thymus\\nB cell proliferation\\nRegulation of T cell migration\\nLeukocyte cell-cell adhesion\\nNegative regulation of leukocyte apoptotic process\\nLymphocyte proliferation\\nAlpha-beta T cell activation\\nMononuclear cell proliferation\\nRegulation of lymphocyte apoptotic process\\nT cell receptor signaling pathway\\nB cell differentiation\\nRegulation of lymphocyte migration\\nRegulation of insulin receptor signaling pathway\\nT cell activation\\nB cell activation\\nLeukocyte proliferation\\nT cell differentiation\\n25 50 75 10025 50 75 100 - Log10(P value)- Log10(P value)\\nAverage CLQ\\nFollicle Community Interactions\\n575746 9\\nGene Expression\\nSmall\\nBowel\\nColonSmall\\nBowel\\nColonSmall\\nBowel\\nColon\\np<2.22e-16p<2.22e-16 p<2.22e-16\\n0\\n1\\n2\\n3\\n4\\n5\\n0\\n1\\n2\\n3\\n4\\n5\\n0\\n2\\n4\\nRASA1PLXNA2 MAP2K5\\nPlasma\\nTA\\nOtherC ells\\nf\\nExtended Data Fig. 9 | Integration of snRNA-seq and CODEX datasets.  \\na) Average MET expression for all TA cells from a given donor and average \\nSEMA4D expression for all plasma cells from a given donor. b) PLXNA2, RASA1, \\nand MAP2K5 expression in TA2 cells in large colon (CL) and small bowel (SB) \\n(one-sided Wilcoxon Rank-Sum Test). c) Representative image of a donor’s \\ntransverse colon with plasma cells (red), TA cells (blue), and other cell types \\n(grey) highlighted with also a magnified area indicated with rectangle.  \\n(left scale bar = 500 µm; right scale bar = 100 µm) d) Example of receptor \\nligands expressed at higher levels in the colon than the small intestine (SI) for \\nthe FN1 by myofibroblast and PLAUR by enterocyte. e) Colocalization quotient \\n(CLQ) for all cell types found within the follicle community. f) Heatmap of \\ndifferentially expressed genes (from MaxFuse matched snRNA-seq cells) \\namong individual CODEX cells, grouped based on previously determined \\ncellular neighbourhoods from CODEX analysis. g) Gene Ontology Enrichment \\nanalysis for cellular neighbourhoods “inner folliclle” and “outer folliclle”,  \\nbased on the differentially expressed genes shown in f.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='04117ef8-67a8-4159-a4b2-2f5764a8e678', embedding=None, metadata={'page_label': '601', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 10 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='deff9f09-0704-4c47-b31a-391bc7878d2b', embedding=None, metadata={'page_label': '602', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 10 | Regulatory TFs in the intestine. a) Hypergeometric \\np-values of selected TF motifs in differential peaks between colon stem cells \\nand stem cells from other regions of the intestine. Colour represents the log10 \\nadjusted p-value computed with ArchR. b) Violin plots of motif deviation scores \\nfor FOXA3 for goblet cells, immature goblet cells, stem cells, and enterocytes in \\ndifferent regions of the intestine. c) Dotplot representation of expression of \\ndifferent transcription factors in goblet cells, immature goblet cells, stem cells, \\nand enterocytes in different regions of the intestine. d) TF motif footprints for \\nPOU2F3 in proximal jejunum tuft cells and enterocytes. e) TF motif footprints \\nfor ATOH1 in duodenum goblet cells and enterocytes. Error bands in d and e \\nrepresent the standard deviation. f) Overlap of epithelial regulators identified \\nwith the multiome data and separate snRNA and snATAC datasets.(g, h) Heatmap \\nrepresentation of transcription factors whose integrated gene expression was \\ncorrelated with their motif activity in one region of the intestine for immune  \\n(g) and stromal (h) cell types. Row z-scores of ChromVar deviation scores are \\nshown on the left and row z-scores of integrated TF expression are shown on \\nthe right. i, j) Overlap of immune (i) and stromal ( j) regulators identified with \\nthe multiome data and separate snRNA and snATAC datasets. k) Hypergeometric \\np-values of TF motifs enriched in the clusters of peaks identified in b. l) Enrichment \\nof KEGG pathways in the clusters of genes identified in c. Uncorrected p-values \\nas determined by kegga are plotted. (m–o) Integrated gene expression of  \\nMTTP (m) and SCNN1B (n) and LGR5 (o) along the differentiation trajectory.  \\n(p) Accessibility at peaks correlated with the expression of TMPRSS15 along the \\ndifferentiation trajectory in each region is plotted on the left. Each peak is \\nnormalized to the maximum accessibility along any of the trajectories. Integrated \\ngene expression of TMPRSS15 along the differentiation trajectory in each \\nregion is plotted on the right.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8624e5b8-c453-4c2e-a31d-bd25da1bb43e', embedding=None, metadata={'page_label': '603', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ec568f35-2b9f-48fc-887b-d68da890ae58', embedding=None, metadata={'page_label': '604', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d37129f5-99d3-4a65-8990-bc48237dd7f3', embedding=None, metadata={'page_label': '605', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a8340b22-48dc-475f-aa2b-e6a386382583', embedding=None, metadata={'page_label': '606', 'file_name': 'hickey_intestine.pdf', 'file_path': '/content/data/hickey_intestine.pdf', 'file_type': 'application/pdf', 'file_size': 51173777, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f1b23e8e-c542-47c3-9890-7e186d22845d', embedding=None, metadata={'page_label': '1', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nT cell-mediated curation and restructuring of tumor\\ntissue coordinates an effective immune response\\nGraphical abstract\\nHighlights\\nd CODEX multiplexed imaging of human and mouse,\\nimmunotherapy-treated melanomas\\nd Tumor-speciﬁc T cells cause inﬂamed tumor phenotype and\\nunique cell neighborhoods\\nd T cell phenotype inﬂuences capacity to make effective tumor-\\nkilling neighborhoods\\nd Potent therapy creates larger, linked areas of inﬂamed tumor\\nand immune inﬁltrate\\nAuthors\\nJohn W. Hickey, Maximillian Haist,\\nNina Horowitz, ..., John B. Sunwoo,\\nMarkus Covert, Garry P. Nolan\\nCorrespondence\\ngnolan@stanford.edu\\nIn brief\\nHickey et al. discovered that tumor-\\nspeciﬁc T cells induce tumor\\ninﬂammation and establish progressive\\ninﬂammatory environments. Effective\\ntherapies exhibit compartmentalized\\nimmune inﬁltration, co-located with T cell\\nand inﬂamed tumor zones, emphasizing\\nthe need for therapeutic T cells to be\\ndesigned to curate local and distal\\nmicroenvironments.\\nHickey et al., 2023, Cell Reports42, 113494\\nDecember 26, 2023ª 2023 The Authors.\\nhttps://doi.org/10.1016/j.celrep.2023.113494 ll\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='61e7cf32-3d79-4f01-b7fc-e1cef012edd0', embedding=None, metadata={'page_label': '2', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nT cell-mediated curation and restructuring\\nof tumor tissue coordinates\\nan effective immune response\\nJohn W. Hickey, 1,2 Maximillian Haist, 1,2 Nina Horowitz, 3 Chiara Caraccio,1,2 Yuqi Tan,1,2 Andrew J. Rech, 4\\nMarc-Andrea Baertsch,1,2 Xavier Rovira-Clave´ ,1,2 Bokai Zhu,1,2 Gustavo Vazquez, 1,2 Graham Barlow,1,2 Eran Agmon, 3,5\\nYury Goltsev,1,2 John B. Sunwoo, 6 Markus Covert,3 and Garry P. Nolan 2,7,*\\n1Department of Microbiology & Immunology, Stanford University School of Medicine, Stanford, CA 94305, USA\\n2Department of Pathology, Stanford University School of Medicine, Stanford, CA 94305, USA\\n3Department of Bioengineering, Stanford University, Stanford, CA 94305, USA\\n4Department of Pathology and Laboratory Medicine, Perelman School of Medicine, University of Pennsylvania, Philadelphia, PA 19104, USA\\n5Center for Cell Analysis and Modeling, University of Connecticut Health, Farmington, CT 06032, USA\\n6Department of Otolaryngology, Head and Neck Surgery, Stanford Cancer Institute, Institute for Stem Cell Biology and Regenerative\\nMedicine, Stanford University School of Medicine, Stanford, CA 94305, USA\\n7Lead contact\\n*Correspondence: gnolan@stanford.edu\\nhttps://doi.org/10.1016/j.celrep.2023.113494\\nSUMMARY\\nAntigen-speciﬁc T cells trafﬁc to, are inﬂuenced by, and create unique cellular microenvironments. Here we\\ncharacterize these microenvironments over time with multiplexed imaging in a melanoma model of adoptive\\nT cell therapy and human patients with melanoma treated with checkpoint inhibitor therapy. Multicellular\\nneighborhood analysis reveals dynamic immune cell inﬁltration and inﬂamed tumor cell neighborhoods\\nassociated with CD8+ T cells. T cell-focused analysis indicates T cells are found along a continuum of neigh-\\nborhoods that reﬂect the progressive steps coordinating the anti-tumor immune response. More effective\\nanti-tumor immune responses are characterized by inﬂamed tumor-T cell neighborhoods, ﬂanked by dense\\nimmune inﬁltration neighborhoods. Conversely, ineffective T cell therapies express anti-inﬂammatory cyto-\\nkines, resulting in regulatory neighborhoods, spatially disrupting productive T cell-immune and -tumor inter-\\nactions. Our study providesin situmechanistic insights into temporal tumor microenvironment changes, cell\\ninteractions critical for response, and spatial correlates of immunotherapy outcomes, informing cellular ther-\\napy evaluation and engineering.\\nINTRODUCTION\\nT cells mediate antigen-speciﬁc responses that are essential for\\nefﬁcient immune surveillance in the human body. Consequently,\\nthey encounter and must be able to effectively respond within a\\nwide number of tissue microenvironments. An outstanding ques-\\ntion is how antigen-speciﬁc T cells must manipulate the local tis-\\nsue microenvironment to accomplish effector function coupled\\nto the extent to which we can alter T cell biology to overcome\\nmechanisms of tumor immune evasion to accomplish positive\\ntherapeutic outcomes.\\nCompared with small molecule and protein-based drugs,\\nT cell-based immunotherapies offer substantial beneﬁts in terms\\nof antigen speciﬁcity, proliferation, and memory. Particularly,\\nimmunotherapy using checkpoint inhibitors (CPIs) has had\\nincredible clinical impacts in the last decade and is now used\\nas single agents or combination therapies in about 50 cancer\\ntypes.1 Despite the remarkable clinical efﬁcacy that was\\nobserved in some patients, the majority of patients do not\\nrespond to CPI therapy or develop secondary acquired resis-\\ntance.2 Similarly, T cell therapies have resulted in impressive\\nclinical outcomes3–6; yet, lack of efﬁcacy and systemic toxicities\\nhave limited broader application of these therapies to solid\\ntumors.7,8\\nAdvances in synthetic biology8,9 or modulation of T cell pheno-\\ntype10–14 promise to hold solutions to challenges of toxicity, tu-\\nmor inﬁltration, and modulation of the tumor microenvironment\\nfacing immune cell therapies. Analyses using single-cell technol-\\nogies have shown that improved function results from enhanced\\ncapacity for self-renewal, killing, and resistance to cancer cell-\\nmediated immunosuppression.15–20 However, our understand-\\ning remains limited of how T cells enter tumor microtissue and\\nhow such cellular structures react to perturbations initiated by\\nT cell therapies.\\nIndeed, the challenges and outstanding questions in targeting\\ncertain classes of solid cancers are spatially related: trafﬁcking,\\ntumor antigen expression, and spatial co-enrichment with\\nstimulating or inhibiting immune cells will likely drive different\\nCell Reports42, 113494, December 26, 2023ª 2023 The Authors. 1\\nThis is an open access article under the CC BY license (http://creativecommons.org/licenses/by/4.0/).\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='40b95a37-fb3a-4ce4-a035-5e154d819229', embedding=None, metadata={'page_label': '3', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 1. CODEX multiplexed imaging of T cell therapy shows dynamic induction of immune cell inﬁltration and tumor cell inﬂammation\\n(A) Experimental layout for T cell therapy model and CODEX multiplexed imaging.\\n(B) Forty-two-antibody panel used for CODEX multiplexed imaging.\\n(C) Representative CODEX images of B16-F10 tumors from mice treated on day 10 with activated PMEL therapeutic T cells and harvested 0, 1, 3, 5, or 12 days\\nfollowing treatment. Scale bars, 250mm, and the zoomed box is 100mm long.\\n(D) Normalized expression of 10 proteins in four of 23 identiﬁed cell types.\\n(legend continued on next page)\\n2 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2cbc018b-bd2e-4f47-b098-52861da2a8f3', embedding=None, metadata={'page_label': '4', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='outcomes to therapy. Determining how these spatial relation-\\nships and multicellular neighborhoods are altered based on\\nT cell phenotype will further reveal the molecular and cellular\\nmechanisms driving therapeutic efﬁcacy. Ultimately, a dataset\\nthat links spatial correlates of therapeutic efﬁcacy such as cell-\\ncell interactions and cellular neighborhoods, and the dynamic\\nchanges of tissue microenvironments following therapeutic\\nintervention, to T cell biology will improve our capacity to better\\nengineer cell therapies.\\nTo evaluate how therapeutic T cells inﬂuence the dynamic,\\ncellular anti-tumor response and how perturbing T cell pheno-\\ntype impacts the tumor microenvironment, we applied\\nrecently developed multiplex ed imaging technologies that\\nenable single-cell analysis of more than 50 targets at once\\nin tissue sections.21,22 We applied these to analyze tumors\\nin a syngeneic, murine B16F10 model of antigen-speciﬁc\\nTc e l lt h e r a p y .5,11,16,23–25 We both study tumors across time\\npoints and also with differential phenotypes. Similarly, we\\ninvestigated the dynamic changes of the tumor microenviron-\\nments of responding and non-responding metastatic mela-\\nnoma patients before and after initiation of CPI therapy. In\\nsummary, we show distinct developmental pathways for the\\nformation of active, higher-order spatial biology that is corre-\\nlated with success of T cell-based immunotherapies. Such\\nﬁndings can guide development of strategies for screening\\nof T cells prior to transfer to ensure optimal efﬁcacy and for\\nengineering of T cell therapies to yield more productive anti-\\ntumor immune responses.\\nRESULTS\\nCODEX multiplexed imaging shows T cell therapy\\ninduces dynamic immune cell inﬁltration and tumor\\ninﬂammation\\nTo evaluate the coordination of efﬁcient immune responses\\nalong with antigen-speciﬁc T cell and tumor interactions, we\\nchose an established, antigen-speciﬁc, syngeneic murine mela-\\nnoma model of adoptive T cell therapy.5,16,23,26 Activated PMEL\\nCD8+ T cells were transferred intravenously into mice with estab-\\nlished, palpable B16-F10 tumors. Tumors were imaged with CO-\\nDetection by indEXing (CODEX) multiplexed imaging27–29 with a\\n42-plex antibody panel at 0-, 1-, 3-, 5-, and 12-days post-treat-\\nment (Figures 1A and 1B, n = 3–7).\\nOn days 0 and 1, a large proportion of tumor cells were prolif-\\nerating, and there was little immune inﬁltration (Figure 1C). By\\nday 3, there were multiple areas of the tumor with upregulated\\nmajor histocompatibility complex class I (MHCI), colocalized in\\nareas with increased immune cell inﬁltration. MHCI upregulation\\nfurther increased on day 5 as were levels of PDL1 and immune\\ncells, such as CD8+ T cells. By day 12, there was still evidence\\nof inﬂammation but had become increasingly compartmental-\\nized (Figure 1C).\\nThese trends are mirrored by cell-type percentages\\n(Figures 1D, S1A, andS1B) that varied more across conditions\\nthan multiple depths (Figure S1C), indicating that the tumor im-\\nmune response is largely conserved throughout an individual tu-\\nmor. Within the stromal compartment of tumors from T cell-\\ntreated mice, the percentage of endothelial cells was decreased\\non day 5, whereas the lymphatic cell percentage increased\\n(Figures 1E andS2), supporting an important dynamic balance\\nof vasculature and lymphatics in tumor growth rates.30–32\\nPercentages of adaptive immune cells increased over the ﬁrst\\n5 days (Figure 1F). Particularly CD8+ T cells increased from\\n/C240.5% to nearly 5% of all cells at day 3 and 10% at day 5. In\\ncontrast, CD4+ T cell subsets remained at approximately 1%\\nof all cells, though CD4+ regulatory T cells (Tregs) increased be-\\ntween day 3 and day 12. Interestingly, by day 12 we observed a\\ndecrease in percentages across all T cell subtypes, reﬂecting the\\naggressive growth kinetics of B16F10 melanoma.\\nInnate immune cells made up the majority of immune cell types\\non day 1, with nearly a 10 to 1 ratio to adaptive immune cells,\\nthough this decreased to about 3 to 1 on days 3, 5, and 12 (Fig-\\nure 1G). This indicates that therapeutic T cells either initially\\nhome to secondary lymphoid organs or require additional time\\n(/C243 days) for expansion and stimulation from a small starting\\npercentage within the tumor.\\nChanges in tumor cell phenotype match our qualitative obser-\\nvations with increasing inﬂammation over the 12-day period\\n(Figures 1H and S2D). The percentage of tumor cells that ex-\\npress PDL1 and MHCI increased from less than 10% on day 1\\nto nearly 70% by day 5 post therapy (Figure 1I). Indeed, PDL1+\\nMHCI+ tumor cell percentages were negatively correlated with\\nKi67+ tumor cells across all sections and on each day, indicating\\na tumor phenotype conversion (Figure 1J, r =/C00.81 Pearson cor-\\nrelation). On the other hand, percentages of PDL1+ MHCI+ tumor\\ncells were positively correlated with percentages of CD8+ T cells\\n(Figure 1K, all values r > 0.8). Thus, adoptive T cell therapy in-\\nduces an anti-tumor immune response that induces a tumor in-\\nﬂammatory phenotype conversion and broad intratumoral im-\\nmune inﬁltration.\\nCD8+ T cells are associated with inﬂammatory tumor\\ncell neighborhoods\\nBecause we qualitatively observed a non-uniform spatial distri-\\nbution of cells (Figure 1C), we posited that spatial analysis of\\nour deﬁned cell types could classify conserved cellular microen-\\nvironments in the tumor. We used a method to extract each cell’s\\nlocal microenvironment by cataloging its distinct cellular neigh-\\nborhoods (CNs) (Figure 2A). Co-correlation indicated that CD8+\\nT cells tend to spatially co-cluster with dendritic cells (DCs)\\nand also the tumor PDL1+ MHCI+ cells (Figure 2B). Since we\\nobserved multiple modules, we clustered into distinct CNs with\\nunsupervised clustering,33 resulting in nine major CNs that could\\nbe visualized on the tissue (Figure 2C).\\n(E–H) Percentages of indicated cells types out of total cells in (E) stromal, (F) adaptive immune, (G) innate immune, and (H) tumor populations over time (plots show\\naverages from three to seven replicates, * indicates p value < 0.05, one-way ANOVA test).\\n(I) Percentages of tumor cell subtypes normalized to total number of tumor cells for each sample (n = 3–7). The boxplots show the median (center line), 25th to 75th\\npercentile (box limits), minimum and maximum values (whiskers), and outliers (points outside 1.53 the interquartile range).\\n(J) Percentage of Ki67+ tumor cells and (K) percentage of CD8+ PD1+ T versus percentage of PDL1+ H2Kb+ tumor cells per sample.\\nCell Reports42, 113494, December 26, 20233\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='fa01cd59-9908-4884-8444-55e0bf0c7120', embedding=None, metadata={'page_label': '5', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Four of the nine CNs are enriched with tumor cell types.\\nMany of the tumor neighborhoods are more homogeneous\\nand enriched only for a few cell types. For example, CN6\\n(Proliferating Tumor) is enriched for marker negative tumor cells\\nand for Ki67+ tumor cells and depleted for all other cell types\\n(Figure 2D). Similarly, CN7 ( Resting Tumor) is enriched for\\nmarker negative tumor cells and is often found localized with\\nCN6, interspersed occasionally by CN9 (Vasculature)( Fig-\\nure 2C), and this ﬁnding agrees with our correlation modules\\n(Figure 2B). These CNs decreased from days 1–5 but gained\\ngreater percentages by day 12 (Figure 2E). Thus, areas where\\nthe tumor is proliferating are compositionally devoid of immune\\ncells and span large areas of immune deserts across the tumor\\n(Figure 2C).\\nIn contrast, PDL1+ MHCI+ tumor cells are enriched in CN8 and\\nCN2 where CN2 contains an enrichment of the majority of im-\\nmune cells including CD8+ T cells (Figure 2D). Areas of an in-\\nﬂamed tumor phenotype, indicated by the CN8 and CN2 are\\noften found together, alongside other inﬂammatory CNs (1, 3,\\n4, 5) (Figure 2C) that are distinct organizations of immune and\\nFigure 2. CD8+ T cells are associated with inﬂammatory tumor cell neighborhoods\\n(A) Schematic of multicellular neighborhood analysis.\\n(B) Correlation heatmap of cell-type abundance within cell neighbor windows.\\n(C) Representative images of neighborhoods mapped to tumor tissues from days 0, 1, 3, 5, and 12 post-treatment.\\n(D) Heatmap of fold enrichment of each cell type as compared with tissue averages in each of the neighborhoods identiﬁed.\\n(E) Quantiﬁcation of neighborhood percentage over time (plots show averages from three to seven replicates).\\n4 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f7c90228-359e-46a5-aca7-b1ebe401ead1', embedding=None, metadata={'page_label': '6', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='stromal cell populations not enriched for tumor cells (Figure 2D).\\nThese inﬂammatory CNs increased from days 1–5 but shrank in\\noverall percentage by day 12, suggesting cooling of the tumor\\nresponse by day 12 (Figure 2E). The spatial association of\\nPDL1+ MHCI+ tumor cells and CD8+ T cells (Figure 2D) and the\\ncorrelation between their frequencies (Figure 1K) indicates a\\nrole of the T cells in converting the tumor cell phenotype. In a\\nseparate study, we demonstrated the importance of tumor\\nphenotype conversion for tumor control in multiscale modeling\\nsimulations of adoptive T cell therapies.34\\nT cell-focused spatial analysis shows a time-linked\\ncontinuum of cell neighborhoods\\nSince there was a time-dependent change in neighborhood\\ncomposition of the tumor tissue after treatment with T cells (Fig-\\nure 2E) but spatial patterning of neighborhoods remained rela-\\ntively constant (Figure 2C), we hypothesized that a more granular\\nlook into the single-cell data could give clues to transition states\\nof neighborhoods. Particularly, views into the ways CD8+ T cell\\nenvironments change over time may guide the design of more\\neffective T cell therapies.\\nTo only look at the cellular environments surrounding CD8+\\nT cells, we developed a new version of CN analysis that is cell-\\ntype centric, with each point in the UMAP representing an indi-\\nvidual CD8+ T cell found within the tumor microenvironment (Fig-\\nure 3A). We identiﬁed cells that are co-enriched as neighbors\\nwith T cells from different timepoints to create a timeline of\\nT cell neighborhood evolution (Figure 3B). On days 0 and 1, the\\nfew CD8+ T cells observed were mostly associated with marker\\nnegative and Ki67+ tumor cells (Figure 3C, panels 1 and 2). On\\nday 3, T cells were also observed next to marker negative and\\nKi67+ tumor cells and endothelial cells, indicating initial T cell\\ninﬁltration (Figure 3C, panels 1 and 2;Figure S3A). On day 3,\\nT cells were also associated with lymphatic cells and other im-\\nmune cells such as DCs, indicative of immune cell recruitment\\n(Figure 3C, panel 3;Figure S3A). By day 5, T cells disengage\\nwith DCs and are enriched with higher levels of macrophages\\nand found associated with PDL1+ MHCI+ tumor cells (Figure 3C,\\npanels 4 and 5), an association that increased on day 12 (Fig-\\nure 3C, panel 5).\\nWe clustered these CD8+ T cell neighborhood windows and\\nused cell-type enrichments within each cluster to deﬁne seven\\nmajor T cell-speciﬁc neighborhoods (TCNs) (Figures 3D and\\nS3B). Interestingly, all TCNs are found across all tumors that\\nwere taken at different time points post-treatment. However,\\neach day is enriched in different proportions of TCNs, conﬁrming\\nchanges in T cell tumor microenvironments over time. On days\\n0 and 1, the T cells were often associated with tumor (TCN A)\\nand both vasculature (TCN B) (Figure 3E), indicating tumor inﬁl-\\ntration processes (Figure 3Fi). On day 3, T cells were found\\nmore in TCN C that was enriched with DCs and CD4+ T cells (Fig-\\nure 3E), indicating local antigen presentation and activation (Fig-\\nure 3Fii). In contrast, on day 5, T cells were not often observed\\nwithin vasculature neighborhood (TCN B), which is supported\\nby increased average distances between CD8+ T cells and endo-\\nthelial cells over time (Figure S3C). Instead CD8+ T cells were\\nfound within areas of tumor cells characterized by an inﬂamed\\nphenotype (TCN D, F) and with other immune cells (TCN E, G)\\n(Figure 3E), indicating active tumor killing (Figure 3Fiii). At day\\n12, the composition of TCNs largely mimics the compositions\\nof days 0 and 1 (Figure 3E), indicating tumor growth that T cell\\nresponses are not able to keep up with (Figure 3Fiv). Taken\\ntogether, these data indicate that T cells could be found along\\na time-dependent continuum of neighborhoods, particularly\\nwith changes in antigen-presenting cell compositions.\\nTo understand if there were any differences in the tumor\\nmicroenvironment speciﬁcally for adoptively transferred cells,\\nwe stained T cells in CODEX multiplexed imaging experiments\\nalso with Thy1.1, which marks PMEL-derived CD8+ T cells (Fig-\\nure 3G). At day 12, about 75% of the CD8+ T cells within the tu-\\nmor were from adoptively transferred cells (Figure 3H). Surpris-\\ningly, adoptively transferred cells (Thy1.1+) had a decreased\\nproportion of PD1+ cells than endogenous CD8+ T cells in the\\ntumor microenvironment ( Figure 3I). Additionally, there was\\na higher percentage of transferred cells that expressed higher\\nlevels of CD25, indicative of activation, recent antigen\\nencounter, effector function, and proliferation (Figure 3 J).\\nAligned with this phenotypic shift, we see adoptively transferred\\ncells are more likely to be found in areas engaged with inﬂamed\\ntumor cells (TCN F, D, and CN2) (Figures 3K andS4), indicating a\\npreferential active anti-tumor response by transferred cells as\\ncompared with host-derived T cells.\\nMetabolically treated T cells induce productive T cell\\nand tumor neighborhoods ﬂanked by neighborhoods of\\nimmune inﬁltrate\\nHaving identiﬁed critical steps within the dynamic endogenous\\nimmune response to an antigen-speciﬁc T cell therapy for our\\nB16F10 melanoma model, we next asked how manipulation of\\nthe T cell phenotype might modulate the tumor microenviron-\\nment.11,18–20,26,35,36 One study using this model showed that\\nacetyl-CoA inhibited T cells were more effective at eradicating tu-\\nmors than were cells activated only with gp100 and interleukin\\n(IL)-2.11 We therefore activated the T cells with 2-hydroxycitrate\\n(2HC), a metabolic inhibitor of acetyl-CoA production11 (further\\nreferred to as2HC T cells). We evaluated the T cell phenotype\\nfollowing culture, and2HC T cellswere enriched for memory phe-\\nnotypes, whereas Tc e l l swere enriched for effector and ex-\\nhausted phenotypes (Figures S5A–S5E), in agreement with previ-\\nous results.11 We treated tumor-bearing mice with activated2HC\\nTc e l l sor Tc e l l sand both T cell treatments provided control of the\\ntumor, with 2HC providing slight increase in efﬁcacy over conven-\\ntionally activated T cells (Figures S5F and S5G). As opposed to\\nVodnala et al., we did not detect a complete reduction for the\\n2HC-treated T cell condition in the B16F10 model, which could\\nbe attributed to the aggressive nature of the B16F10 model.11\\nThree days post-treatment, CODEX imaging of the tumors\\nindicated immune inﬁltration (CD45), tumor inﬂammation\\n(H2Kb, PDL1), and downregulation of tumor proliferation (Ki67)\\nfrom both T cell-treated tumors compared to the control tumor\\nnot treated with T cells (referred to asNo T cells)( Figures 4A\\nand 4B, n = 4–7). Indeed, inﬂamed tumor cells (PDL1+ MHCI+),\\nCD8+ T cells, and Ki67+ tumor cells were identiﬁed as the three\\nmost important cell-type percentages that could predict treat-\\nment group (Figure 4C, AUC = 0.95). In tumors that received\\neither T cell treatment, there was an increase in PDL1+ MHCI+\\nCell Reports42, 113494, December 26, 20235\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8ad40b37-eadc-49e0-861a-ba5a42e9ed6a', embedding=None, metadata={'page_label': '7', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='tumor cells, whereas there were less than 2% inﬂamed tumor\\ncells in the control tumors, associated with higher levels of\\nCD8+ T cells (Figures 4D andS6). Moreover, tumors from mice\\ntreated with2HC T cellshad a lower percentage of proliferating\\ntumor cells than did either of the other conditions (Figure S6A).\\nThis further conﬁrms that T cells can mediate tumor cell prolifer-\\nation inhibition and induce tumor cell inﬂammation.\\nCharacterization of the CNs for all three conditions revealed four\\ntumor CNs and six immune-tumor CNs (Figure S7A) with similar-\\nities to those found in our dynamic analysis inFigure 2. One major\\ndifference was two CNs involving CD8+ T cells and inﬂamed tumor\\ncells. One of these CNs was enriched for MHCI+ tumor cells, CD4+\\nTc e l l s ,K i 6 7+ CD4+ T cells, and NK cells (termed hereinProductive\\nT cell & TumorCN). In contrast, the second neighborhood lacks\\nthese features and is enriched for Tregs and MHCI/C0tumors\\n(termed as the Unproductive T cell & TumorCN). The tumors\\ntreated with Tc e l l shad a higher proportion of Unproductive\\nT cell & TumorCN compared with the tumors treated with2HC\\nAB\\nC\\nDE F\\nGH I J K\\nFigure 3. Newly developed T cell-focused spatial analysis showing that CD8+ T cells can be found along a continuum of cell neighborhoods\\nrepresenting steps of inﬂammation and tumor attack\\n(A) T cell centric spatial UMAP encodes how similar the local microenvironment surrounding individual CD8+ T cells is to other CD8+ T cell microenvironments.\\n(B and C) Spatial UMAP plots of T cell neighborhood composition vectors across all experiments with T cells from tumors harvested (B) each day highlighted in\\nred. (C) Overlays of cell-type enrichment of each T cell microenvironment.\\n(D and E) Seven T cell neighborhoods (TCNs) were found and (D) plotted on UMAP coordinates and (E) percentages of T cell neighborhoods present at each time\\npoint (plots show averages from three to seven replicates).\\n(F) Illustration of dynamic response of the tumor microenvironment to T cell therapy.\\n(G) Representative CODEX images of B16–F10 tumors harvested on day 12 after being treated with activated PMEL therapeutic T cells. Scale bar, 100mm long.\\n(H) Percent of total CD8+ T cells that are Thy1.1+//C0from tumors harvested 12 days after adoptive T cell transfer treatment (n = 7).\\n(I) Percent of CD8+ T cells that are positive for PD1 staining for either Thy1.1+//C0cells from tumors harvested 12 days after adoptive T cell transfer treatment (n = 7).\\n(J) Scatterplot of the normalized expression of CD25 and CD27 by either Thy1.1+ (blue) or Thy1.1/C0(pink) CD8+ T cells and gate indicating positivity for CD25.\\n(K) Percent of CD8+ T cells that are positive for either TCN F or D and CN2 (n = 7, p value = 0.01, 0.01, and 0.008 calculated by two-sided t test). For H, I, and K the\\nboxplots show the median (center line), 25th to 75th percentile (box limits), minimum and maximum values (whiskers), and outliers (points outside 1.53 the\\ninterquartile range).\\n6 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='82ae34d2-e59f-4d3e-ae82-8d2c0cac361e', embedding=None, metadata={'page_label': '8', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='A\\nB\\nCD\\nEF\\nG\\nH\\nI\\nJK\\nFigure 4. Metabolically treated T cells induce productive T cell and tumor neighborhoods ﬂanked by neighborhoods of immune inﬁltrate\\n(A) Representative CODEX images and (B) Cell-type maps of the treated and untreated tumors. Scale bar, 250mm.\\n(C) Minimal depth of cell-type percentages that differentiate treatment groups.\\n(D) Tumor cell phenotypes measured by CODEX multiplexed imaging of tumors from controls (no T cells) and mice treated with T cells or 2HC T cells (n = 4–7\\nindividual tumors from four independent experiments). The boxplots show the median (center line), 25th to 75th percentile (box limits), minimum andmaximum\\nvalues (whiskers), and outliers (points outside 1.53 the interquartile range).\\n(legend continued on next page)\\nCell Reports42, 113494, December 26, 20237\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d5c7d0b4-5966-4fc5-bbea-74a88002ac8e', embedding=None, metadata={'page_label': '9', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Tc e l l s(Figures 4Ea n dS7B). Cell-cell interaction analysis also\\nindicated that CD8+ T cells are closer to both NK cells and\\nCD4+ T cells in tumors treated with2HC T cells, also having closer\\nCD4+ Tc e l l - C D 4+ Tc e l la s s o c i a t i o n s(Figures 4F, S7C, andS7D).\\nSimilarly, we observed that CD4+ Tregs were closer to stromal cell\\ntypes, which helps explain spatial segregation of the CD4+ Tregs\\naway fromProductive T cell & TumorCNs in2HC T cell-treated tu-\\nmors. Finally, CN analysis indicatesNo T cellshad uniform spatial\\npatterning of mostly tumor CNs (Figure 4G). In both tumors treated\\nwith T cells, there was increased regional density of immune-\\nbased CNs, and there were clear borders around CNs (Figure 4G).\\nIn tumors treated with2HC T cells,t h eProductive T cell & Tumor\\nCN (blue) was often bordered by theImmune InﬁltrateCN (light\\nblue) (Figure 4G).\\nMulticellular neighborhood interfaces may indicate coordina-\\ntion of critical functions within the tumor.21 To quantify and\\nformalize general observations about CN associations, we\\nused spatial context analysis developed previously (Figure\\n4H).37 The tumors treated with2HC T cellshad a greater propor-\\ntion of compartmentalization than the other treatment groups\\n(Figure 4I, row 1). Here, CNs such asProductive T cell & Tumor\\n(blue) have high degrees of compartmentalization. This indicates\\nthat these T cells transform large areas of the tumor into regions\\nwith conserved cell-type composition of immune inﬁltrate and\\nT cell-mediated tumor cell killing areas. In contrast,No T cells\\ncondition has a lack of coordinated structure with the majority\\nin combinations of three or more CNs (Figures 4I andS8A).\\nThe cases where a combination of two CNs make up at least\\n85% of a window are indicative of important interfaces and co-\\nordination of structures potentially necessary for function. In tu-\\nmors treated with2HC T cells, there were higher proportions of\\nthe Productive T cell & Tumor(blue) andImmune Inﬁltrate(light\\nblue) CN combination than in the other two groups (Figures 4I\\nand S8B–S8D). In contrast, theProductive T cell & Tumorand\\nUnproductive T cell & TumorCN combination was prevalent in\\ntumors treated withT cells.\\nWe next evaluated CN interactions37 and in tumors treated\\nwith T cells. Here, we observed a signiﬁcant association of the\\nProductive T cell & Tumorneighborhood (blue) with theUnpro-\\nductive T cell & Tumor(red) CN; whereas this motif was not de-\\ntected in the tumors treated with2HC T cells (Figure 4J). In\\ncontrast, within the tumors treated with 2HC T cells ,w e\\nobserved combinations of theProductive T cell & Tumorneigh-\\nborhood (blue) with tumor-based CNs. Consequently, we\\nobserve this indicates theProductive T cell & TumorCN plays\\nan important role in the tumor response and is impacted by\\nwhich CNs border it.\\nCanonical correlation analysis 33 also revealed that fre-\\nquencies of CD8+ T cells are differentially correlated along\\nconnections of the Immune Inﬁltrate and Productive T cell &\\nTumor CNs (Figure 4K). This demonstrates that spatial rela-\\ntionships between inﬂamed tumor CNs and theImmune Inﬁl-\\ntrate neighborhood are also critical for CD8+ T cell function\\nin the tumor. Overall, these data suggest that pairing ofUn-\\nproductive and Productive T cell & TumorCNs is undesirable,\\nand that proximity ofImmune Inﬁltrate CNs is critical for CD8+\\nT cell function.\\nAcetyl-CoA pathway modulated T cells have a distinct\\ncapacity to shape coordinated immune inﬁltration\\nWe added the fractions ofImmune inﬁltrate and Productive\\nT cell & TumorCNs and subtracted the fraction of theUnproduc-\\ntive T cell & TumorCN for each condition. This value was higher\\nfor tumors treated with2HC T cellsthan for mice treated withT\\ncells (Figure 5A, p < 0.05). This suggests that2HC T cellshave\\ngreater propensity to create productive T cell tumor killing areas\\nthan doT cells.\\nWe surmised that compositional differences when these CNs\\nare found in the same area may be more informative than bulk av-\\nerages. To quantify this, we measured the relative percentages of\\nthese three CNs in windows where they made up at least 85% of\\nthe CNs (Figure 5B)—or all possible combinations with these CNs\\nfrom our spatial context map (Figure 4I). For tumors given2HC\\nTc e l l s, there was a high proportion of cells that fall in the interface\\nof theImmune Inﬁltrateand Productive T cell & TumorCNs (Fig-\\nure 5C, black rectangle). This localized combination of CNs\\n(blue, light blue) can be seen when CN labels are mapped to a tu-\\nmor treated with2HC T cells(Figure 5D). Focusing in on a region\\ncontaining theImmune Inﬁltrateand Productive T cell & Tumor\\nCNs, the ﬂuorescence data conﬁrmed that this is an area of tumor\\ninﬂammation (PDL1+,r e d ;H 2 K b+, yellow) and immune inﬁltration\\n(CD45+,g r e e n )(Figure 5E). At the single-cell level, inProductive\\nT cell & TumorCNs (blue, Figure 5F) ﬂuorescent images show\\nCD8+ T cells (white membranes and white arrows,Figure 5G)\\nare in contact with tumor cells. Also, these are abutted by CD8+\\nT cells that are next to stromal and immune cells (CD45+,g r e e n ,\\nFigure 5G) inImmune InﬁltrateCNs (light blue,Figure 5F).\\nIn tumors given canonically activatedT cells, there was a shift\\nto a higher proportion of theUnproductive T cell & TumorCN\\n(Figure 5H). Particularly the edge along theImmune Inﬁltrate\\nand Productive T cell & TumorCNs represents an important\\ninterface devoid ofUnproductive T cell TumorCNs. There was\\nnearly a 4-fold increase in frequency of cells in this category in\\ntumors given2HC T cellsrelative to those givenT cells(Figure 5I,\\np < 0.05). This indicates that theUnproductive T cell & TumorCN\\ndisrupts the Immune Inﬁltrate and Productive T cell & Tumor\\ninterface in tumors treated with T cells than tumors treated\\nwith 2HC T cells.\\n(E) Volcano plot of neighborhood percentage from CODEX imaging data for2HC T cellsor T cellstreated tumors (p value calculated from likelihood ratio test).\\n(F) Selected signiﬁcant cell-cell interaction pairs and compared log2 fold distance calculations compared with null distribution average.\\n(G) Neighborhoods mapped back to tissue coordinates.\\n(H) Schematic of spatial context map generation.\\n(I) Spatial context maps where row shows the number of neighborhoods in combinations.\\n(J) Signiﬁcant two-combination neighborhood motifs enriched or depleted compared with null distribution of permuted samples.\\n(K) Canonical correlation of CD8+ T cell frequencies across all neighborhoods. Linkages plotted indicate correlations found above permuted controls between\\nsets of neighborhoods with color as shown in the key in (G).\\n8 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5ea6647c-6c08-4e41-8b26-a2c5503b82a3', embedding=None, metadata={'page_label': '10', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='AB C\\nD\\nE\\nF G\\nHI\\nJ\\nFigure 5. The phenotype of the therapeutic T cell inﬂuences its capacity to shape coordinated immune inﬁltration\\n(A) Productive neighborhood fraction per sample (n = 6–7 individual tumors from four independent experiments). The boxplots show the median (centerline), 25th\\nto 75th percentile (box limits), minimum and maximum values (whiskers), and outliers (points outside 1.53 the interquartile range).\\n(B) Graphic of generation of barycentric plots of window combinations.\\n(C) Barycentric plot for tumors from mice treated with 2HC-activated T cells (cells from n = 6 individual tumors from four independent experiments).\\n(D) Neighborhood map for a tumor from a mouse given 2HC-treated T cells.\\n(E) Magniﬁcation of region in orange rectangle in (D) with six of the 42 CODEX ﬂuorescent markers shown. Scale bar, 250mm.\\n(legend continued on next page)\\nCell Reports42, 113494, December 26, 20239\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='05c1ea88-bac2-4f87-88c9-e485218c0b3d', embedding=None, metadata={'page_label': '11', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Furthermore, the intersection between theImmune Inﬁltrate\\n(light blue) andProductive T cell & Tumor(blue) CNs was greater\\nin tumors given2HC T cellsthan in those givenT cells(Figure 5J).\\nIn contrast, the intersection of theUnproductive T cell & Tumor\\nwith bothImmune Inﬁltrateand Productive T cell & TumorCNs\\nwas greatest in the tumors treated with T cells (Figure 5J).\\nThus, the phenotype of the therapeutic T cell inﬂuences its ca-\\npacity to shape coordinated immune inﬁltration that support pro-\\nductive T cell tumor-killing zones.\\nA subset of therapeutic T cells express anti-\\ninﬂammatory cytokines that are associated with greater\\nlevels of regulatory environments\\nNext, we looked into why 2HC T cells may cause more compart-\\nmentalized productive tumor interfaces. Interestingly, T cells\\nactivated with or without 2HC kill tumor cells with the same efﬁ-\\nciency (Figure 6A). Furthermore, similar levels of effector mole-\\ncules and similar levels of polyfunctionality38,39 (i.e., percentages\\nof cells able to produce multiple effector molecules at once) were\\nobserved for T cells activated with and without 2HC (Figures 6B\\nand S9A).\\nRNA-sequencing (RNA-seq) data indicated that the expres-\\nsion of genes enriched in2HC T cellsover T cellswas also found\\nto be associated with favorable disease outcomes (represented\\nby lowerZ scores) relative to randomly permuted gene sets of\\nthe same size within human skin cutaneous melanoma samples\\nfrom The Cancer Genome Atlas (TCGA) (Figures 6C and 6D).\\nFurther analysis with gene ontology enrichment analysis showed\\nenrichment for pathways relating to immune cell differentiation,\\ninterferon signaling, and cytokines ( Figure S9 B, Extended\\ndata). Consequently, we compared the two T cell treatment con-\\nditions with each other highlighting the greatest log-fold change\\nexpression of genes falling into these major pathways: pheno-\\ntypic, interferon (IFN)g signaling, migration, and cytokine-related\\ngenes (Figure 6C).\\nDifferences in phenotypic genes agreed with the CyTOF\\nphenotypic characterization (Figure S5and 6C). Moreover, in a\\nseparate study, we showed that IFNg plays a role in transforma-\\ntion of tumor phenotype. 34 Particularly interesting to tumor\\ncellular organization, CXCL10 was enriched within2HC T cells\\n(Figure 6C), known to be a T cell recruitment factor.40 In the\\nCODEX data, we observed higher CD8+ T cell density in tumors\\nof both treated groups that was most apparent in tumors treated\\nwith2HC T cells (Figure S9C). LTB is also enriched within\\n2HC T cells(Figure 6C) and is involved in lymphangiogenesis,\\nwhich aligns with our observation of 2-fold increased levels\\nof lymphatic cells in tumors from mice given 2HC T cells\\n(Figure S5H).\\nInterestingly, T cells also had higher levels of expression of\\nmRNAs encoding anti-inﬂammatory cytokines IL5, IL4, IL10,\\nand IL13 (Figure 6C). The expression of anti-inﬂammatory cyto-\\nkines may explain the higher levels of the regulatory neighbor-\\nhood Unproductive T cell & Tumorwithin tumors givenT cells.\\nHowever, RNA-seq data provide a population-level average of\\nexpression, and we observed diverse populations in the single-\\ncell CyTOF data (Figures 3 A and 3B). Consequently, we\\nconﬁrmed with CyTOF that a greater percentage of T cells\\nsecrete IL-10 than do2HC T cells (/C2419% versus 1.5%) (Fig-\\nure 6E, p < 0.001). Interestingly, the cell types that expressed\\nIL-10 also expressed other effector molecules and PD1 (Fig-\\nure 6F). We conﬁrmed this using spatial transcriptomics, where\\nwe observe 12.5% of the GZMB+ spots were positive for IL-10\\n(Figure 6G, panels 2 and 3). This supports the idea that a fraction\\nof T cells that secrete anti-inﬂammatory molecules may coun-\\nteract development of effective T cell tumor-killing microenviron-\\nments that are supported by immune cells. Taken together,\\nthese in vivo and in vitro data indicate that T cell phenotypes,\\nnot related to tumor-killing efﬁcacy or traditional effector mole-\\ncules, inﬂuence the ability of T cells to curate productive tu-\\nmor-killing microenvironments.\\nMoreover, TCNs of2HC T cellsindicated theimmune inﬁltrate\\nneighborhood also shares proximity in the UMAP space with the\\ninﬂamed tumorTCN (Figures 6H and 6I). This ﬁnding agrees with\\nour analyses that indicate the importance of the border of a\\ndense immune inﬁltration TCN and T cell killing zones with in-\\nﬂamed tumor cells (Figures 4and 5). Accordingly, this indicates\\nthat 2HC T cells curate the tumor microenvironment and enable\\nbroad immune cell inﬁltration that supports an effective anti-tu-\\nmor T cell killing zone.\\nCoordinated interactions of CNs orchestrate anti-tumor\\nimmune responses in patients with metastatic\\nmelanoma receiving immune-checkpoint inhibitors\\nTo see if there is similar coordinated CN behavior within effective\\nimmune responses to solid tumors of human patients, we\\ncollected tumor samples from patients with metastatic stage IV\\nmelanoma before and after treatment with immune-checkpoint\\ninhibitors (CPIs) (Figure 7A). We then performed CODEX multi-\\nplexed imaging on the 12 whole-slide sections with a panel of\\n58 antibodies targeting major immune, stromal, and tumor com-\\npartments. Even from gross qualitative examination of the\\nCODEX imaging we observed substantially higher degree of\\ninﬂammation within responder (RS) tumors, particularly post\\nCPI (Figure 7B).\\nWhile we observed no signiﬁcant differences in the abundance\\nof major cell types between RS and NR at baseline (Figures S10A\\nand S10B), our results show that greater abundance of PD-1+\\nCD8 T cells and TCF1/7+ CD8 T cells was associated with pa-\\ntients responding to treatment (Figures 7C, S10C, and S10D).\\nIndeed, post-CPI therapy, responders showed a strong increase\\nin the fraction of immune cells at the tumor (Figure 7D). This ma-\\njor increase in the immune cell compartment was particularly\\n(F and G) Magniﬁcation of region in magenta rectangle in (E) with both the (F) neighborhood and (G) CODEX ﬂuorescent imaging overlay shown. White arrows\\nindicate CD8+ T cells. Scale bar, 100mm.\\n(H) Barycentric plot of data from conventional-activated T cell treatment group (cells from n = 7 individual tumors from four independent experiments).\\n(I) Quantiﬁcation of percentage of windows that have 95% or greater ofProductive T cell & Tumorand Immune Inﬁltrateneighborhoods (n = 5–6 individual tumors\\nfrom four independent experiments, p < 0.05 from two-sided t test).\\n(J) Intersection of the neighborhood instances quantiﬁed for deﬁned neighborhood combinations.\\n10 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b5a14133-210c-4839-889c-d5c32945e0c1', embedding=None, metadata={'page_label': '12', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 6. A fraction of T cells activated under standard conditions express anti-inﬂammatory cytokines at levels greater than metabolically\\ntreated T cells\\n(A) Percent cytotoxicity in anin vitrokilling assay (n = 3 replicates, representative of two independent experiments).\\n(B) Pie charts of polyfunctionality of T cells (averages n = 4 replicates).\\n(C) log2 fold differences in gene expression.\\n(D) A density plot showing the distribution of multivariate Cox model gene expressionZ scores for the 2HC T cell-enriched gene set.\\n(E) Proportion of IL-10+ cells as determined by CyTOF analysis (n = 3 replicates, p < 0.05 two-sided t test). The boxplots show the median (center line), 25th to 75th\\npercentile (box limits), minimum and maximum values (whiskers), and outliers (points outside 1.53 the interquartile range).\\n(legend continued on next page)\\nCell Reports42, 113494, December 26, 202311\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='06fdf0d4-a60e-4bff-967d-0c54be5314a4', embedding=None, metadata={'page_label': '13', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='driven by a strong inﬂux of CD8+ T cells, NK cells, DCs, and mac-\\nrophages, and a decrease in tumor cells when comparing pre-\\ntreatment with post-treatment samples of RS as compared\\nwith NR tumors (Figures 7C andS10D).\\nWe found 16 unique human melanoma cellular neighborhoods\\n(hCNs) (Figures 7E andS11A), of which many were composition-\\nally similar to the neighborhoods we previously observed in our\\nmouse melanoma tumor samples, though we also observed\\nsome new tissue structures (Figure S11A). Two of the hCNs\\nthat were compositionally similar to murine tumor CNs were\\nthe Immune Inﬁltrate hCN and the Productive T cell & Tumor\\nhCN (Figure S11A). Akin to what we observed in the murine tu-\\nmors, we see shared proximity of theImmune Inﬁltrate hCN\\nwith the Productive T cell & TumorhCN at the tumor interface\\n(Figure 7E–H).\\nPre-CPI therapy, the Productive T cell & Tumor hCN was\\nlargely associated independent or with major tumor hCNs (Fig-\\nure S11B). However, post CPI, theProductive T cell & Tumor\\nhCN was almost exclusively found co-associated with theIm-\\nmune InﬁltratehCN (Figure S11C). Permutation testing of hCN in-\\nstances also identiﬁed this shift in organization of theProductive\\nT cell & Tumor hCN within responders after CPI therapy,\\nincluding highly enriched colocalization with theImmune Inﬁl-\\ntrate hCN (Figure 7I). This hCN-hCN spatial colocalization was\\nnot observed pre-CPI (Figure S11D). This contrasts with thePro-\\nductive T cell & TumorhCN in post-CPI samples for NR, where\\nthere was no spatial co-enrichment with theImmune Inﬁltrate\\nhCN at either timepoint (Figures 7J andS11E).\\nThese results corroborate our earlier ﬁndings within murine\\nmelanoma tumors and suggest that an active and robust anti-tu-\\nmor CD8+ T cell response can cause and is dependent on neigh-\\nboring immune-dense microenvironments. This is also evident\\nfrom the fact that theProductive T cell & TumorhCN does not in-\\ncrease in frequency pre to post CPI, though theImmune Inﬁltrate\\nhCN does (Figure S11F), indicating recruitment upon a more\\nrobust CD8+ T cell response. Similarly, the overall tissue size of\\nImmune InﬁltratehCN instances (hCNs of the same type spatially\\nconnected) increased post CPI in responders. This result, in\\naddition to the fact that we observe globally larger hCN instances\\nfrom responders compared with non-responders post CPI\\n(Figures S12A and S12B), indicate a less fragmented immune or-\\nganization within the tumor microenvironment. Taken together,\\nthese data suggest that organized immune inﬁltration areas\\nabutting robust anti-tumor T cell zones is necessary for pro-\\nlonged and effective anti-tumor efﬁcacy.\\nDISCUSSION\\nIn this study, we characterized the spatial cellular microenviron-\\nments that anti-cancer T cells are found in over time and across\\nT cell phenotypes. Critically, the studies demonstrate the ability,\\nby simple empiric data extraction, to derive an inference history\\nof immune cell inﬁltration of a tumor along with mechanistic in-\\nsights into cell-cell interactions that are critical for T cell-medi-\\nated progression toward tumor eradication.\\nWe leveraged multiplexed imaging techniques to explore the\\ncellular microenvironments within B16F10 tumors treated with\\nactivated cancer-speciﬁc T cells. Multiplexed imaging demon-\\nstrated that T cell therapies drive tumor restructuring and lead\\nto changes in tumor cell phenotypes. Higher levels of tumor cells\\nwith the inﬂamed phenotype were correlated with increased\\nlevels and proximities of CD8+ T cells (Figures 7K and S12C).\\nIn parallel work, multiscale modeling also indicated that thera-\\npeutic T cells convert tumor cells to an inﬂamed, anti-prolifera-\\ntive phenotype at scales larger than direct cell-cell interaction\\ncritical for tumor control.34 The application of T cell therapies\\nthat allow for the targeted remodeling of tumor microenviron-\\nments might therefore help to overcome primary or acquired\\nresistance to existing treatments for advanced melanoma,\\nsuch as checkpoint-blocking antibodies.\\nIn addition to changing the tumor phenotype, we demonstrate\\nthat T cells can mediate the formation of either productive or un-\\nproductive tumor T cell neighborhoods that are co-enriched with\\ninﬂamed tumor cells. Metabolic manipulation of T cells,11 co-\\ninduced broad immune inﬁltration that shared borders with pro-\\nductive T cell and tumor neighborhoods. Further, the immune\\ninﬁltrate and productive T cell and tumor neighborhoods were\\nnot broken up by regulatory neighborhoods as they were in tu-\\nmors of mice treated with T cells activated using the standard\\nprotocol. We similarly observed only in patients responding to\\nCPI therapy a pronounced interaction between the productive\\nT cell and tumor cellular neighborhood and the immune inﬁltrate\\nneighborhood in samples post-CPI start. Particularly, these CNs,\\nsimilar to those found in murine tumors, also had greater\\ncompartmentalization and were characterized by a higher abun-\\ndance of T cells with a memory phenotype as opposed to non-\\nresponding patients. This further afﬁrms previous ﬁndings that\\ndense immune inﬁltrates as opposed to diffuse or unorganized\\nimmune inﬁltrates are beneﬁcial for tumor outcomes33,41,42 and\\nadds to howin situcellular relationships provide spatial mecha-\\nnisms of homeostasis and disease.21,43–53\\nNext, our ﬁndings that the T cell phenotype inﬂuences struc-\\ntural reprogramming of the tumor microenvironment adds to\\nthe list of factors known to be critical for determining therapeutic\\nT cell success, such as effector molecule expression and T cell\\nlongevity. Here we saw that canonically activated T cells have\\nsubset populations that secrete anti-inﬂammatory cytokines in\\naddition to inﬂammatory molecules. We found that therapeutic\\nT cells express IL-10. IL-10 is known to be expressed by Tregs,54\\nregulate CD8+ T cell responses,55 inhibit activity of myeloid\\ncells,56 and be implicated in relapse of CAR T therapy in acute\\nlymphoblastic leukemia.57 The presence of these T cell pheno-\\ntypes may explain why regulatory microenvironments disrupt\\nproductive T cell and tumor responses more in tumors treated\\nwith these T cells than with metabolically activated T cells. Our\\nstudy supports the idea that T cell therapies should be designed\\n(F) Individual cell phenotypes characterized by CyTOF phenotyping (n = 3 replicates).\\n(G) Spatial transcriptomics expression of GZMB, IL-10, or both in tumor from mouse treated with T cells (without 2HC) and harvested on day 3.\\n(H) Spatial UMAP plots of T cell neighborhood composition vectors across all experiments with T cells from tumors treated with 2HC-modiﬁed T cells.\\n(I) T cell neighborhood cell-type composition per individual T cell neighborhoods.\\n12 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a98b57bf-57a1-4134-a04f-b695542e985e', embedding=None, metadata={'page_label': '14', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Figure 7. CODEX analysis of pretreatment and post-treatment samples from metastatic melanoma patients who received immune-\\ncheckpoint inhibitor therapy reveals distinct changes in T cell inﬁltration characterizing responding patients similar to changes observed for\\nmurine B16F10 experiments\\n(A) Schematic representation of the study design.\\n(B) Representative CODEX multiplexed imaging overlays (6/58 markers shown) for responders and non-responders, pre and post therapy (scale bar, 200mm).\\n(C) Volcano plot of the log2 of the fold change in cell-type percentage of non-responders (NR) to responders (RS).\\n(legend continued on next page)\\nCell Reports42, 113494, December 26, 202313\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='16dcab1e-8c08-498f-892d-5b9bd13491fe', embedding=None, metadata={'page_label': '15', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='with cytokine and chemokine expression in mind. This will\\nrequire new methods of evaluatingex vivoT cell phenotypes to\\ninclude molecules that are involved in tissue restructuring and,\\nnotably, anti-inﬂammatory processes.\\nOne new spatial analysis we developed was a method of\\ncollapsing a cell’s microenvironment into lower dimensional\\nspace. We used this to evaluate similarity of composition of\\nCNs, similarity of neighborhoods across different conditions,\\nand connections back to temporal sequences. Particularly, we\\nleveraged this new technique to deﬁne T cell-speciﬁc CNs within\\nthe tumor across time and treatment. The T cell-speciﬁc neigh-\\nborhoods represented major changes necessary within the\\nmicroenvironment for the anti-tumor response: tumor inﬁltration\\nfrom vasculature, immune cell recruitment (particularly changes\\nin antigen-presenting cells with inﬂux of dendritic cells and mac-\\nrophages), and induction of tumor death and inﬂammation. Here\\nwe were able to characterize differences in the microenviron-\\nments of adoptively transferred T cells compared with endoge-\\nnously recruited T cells.\\nIn summary, our results demonstrate that therapeutic T cells\\nshape the tumor cellular microenvironment and depend on the\\ninput of the therapeutic T cell phenotype. Moreover, our results\\nare the ﬁrst to our knowledge to investigate the temporal dy-\\nnamics of immune responses in the course of tumor progression\\nand relationships of higher-order spatial biology that are corre-\\nlated with success of the therapy. In this regard, our results\\nstress the importance of compartmentalized immune inﬁltration\\nbordering the edge of T cell attack of tumor cells. Thus, this data-\\nset is an archetype for understanding and evaluating cell-cell\\ninteraction in situfor cellular therapies. It calls for transformative\\nchanges to how T cell therapies should be screened prior to\\ntransfer and suggests additional avenues for engineering of\\nT cell therapies to prioritize cellular microenvironment manipula-\\ntion to yield more productive anti-tumor immune responses.\\nLimitations of the study\\nHere we used an established, syngeneic murine model of mela-\\nnoma that allowed analysis of the native immune response, ther-\\napeutic perturbations, and dynamic restructuring. This work and\\nthe ﬁndings in this model can be followed up in additional mouse\\ntumor models of both CAR T and endogenous T cells, along with\\nadditional perturbations. Our multiplexed imaging panel only\\nincluded 42 antibodies for mouse and 58 antibodies for human,\\nwhich may not fully represent all cell types or states that are\\nmeaningful in the tumor, and our panels could be expanded by\\ndevelopments in multiplexed imaging technologies such as sin-\\ngle-cell spatial transcriptomics.44,58,59 Here, we characterized\\nvery large areas of the tumor from a limited set of human mela-\\nnoma patients (six total from two timepoints), and future studies\\nincorporating more patients through smaller tissue microarrays\\ncould be studied using multiplexed imaging.\\nSTAR+METHODS\\nDetailed methods are provided in the online version of this paper\\nand include the following:\\nd KEY RESOURCES TABLE\\nd RESOURCE AVAILABILITY\\nB Lead contact\\nB Materials availability\\nB Data and code availability\\nd EXPERIMENTAL MODEL AND STUDY PARTICIPANT DE-\\nTAILS\\nB Cell-lines\\nB Mice\\nB Human samples\\nd METHOD DETAILS\\nB T cell culture and stimulation\\nB In vitroT cell killing assay\\nB CyTOF phenotyping\\nB Mouse tumor model\\nB CODEX multiplexed imaging\\nB RNA-seq\\nB Spatial transcriptomics\\nd QUANTIFICATION AND STATISTICAL ANALYSIS\\nB CODEX multiplexed imaging data analysis\\nB RNA-seq data analysis\\nSUPPLEMENTAL INFORMATION\\nSupplemental information can be found online athttps://doi.org/10.1016/j.\\ncelrep.2023.113494.\\nACKNOWLEDGMENTS\\nThis work was supported by the US National Institutes of Health (P01HL108797,\\nU01AI101984, 5U54CA209971, 5U01AI140498, U54HG010426, U19AI100627,\\nR01HL120724, R01HL128173, 5P01AI131374, UH3DK114937, U19AI135976,\\nU2CCA233238, U2CCA233195); the US Food and Drug Administration\\n(HHSF223201610018C, DSTL/AGR/00980/01); Cancer Research UK (C27165/\\nA29073); the Bill and Melinda Gates Foundation (OPP1113682); the Cancer\\nResearch Institute; the Parker Institute for Cancer Immunotherapy (PICI0025);\\nHope Realized Medical Foundation (209477); the Kenneth Rainin Foundation\\n(2020-1463); the Silicon Valley Community Foundation (2017-175329 and\\n2017-177799-5022); the Beckman Center for Molecular and Genetic Medicine;\\n(D) Donut charts representing overall major cell-type compartments from CODEX multiplexed imaging (inside donut), and coarse grain cell-type percentages\\n(outside donut, with legend) for all four groups shown.\\n(E) Human cellular neighborhood (hCN) map from one of the patient’s tumor samples that was imaged by CODEX multiplexed imaging taken post-checkpoint\\ninhibitor (CPI) therapy (black scale bar, 1 mm).\\n(F) Magniﬁed portion of the human cellular neighborhood (hCN) map indicating a location where theImmune InﬁltratehCN (sky blue) is adjacent to theProductive\\nT cell & TumorhCN (blue).\\n(G) The same area magniﬁed but showing a seven-color overlay (out of 58 total markers) highlighting immune cell inﬁltration (scale bar, 100mm) next to areas\\nwhere (H) CD8+ T cells are engaging with tumor cells with an inﬂamed phenotype (orange scale bar, 100mm).\\n(I and J) Signiﬁcant two-combination neighborhood motifs enriched or depleted that are found associated with theProductive T cell & TumorhCN compared with\\nnull distribution of permuted samples for (I) tumors from responders and (J) post therapy. Color represents shared neighborhood found across all samples and\\ncorresponding colors can be found in (E).\\n14 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a4b56d58-fc06-492f-9a47-72a6e03863b8', embedding=None, metadata={'page_label': '16', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Celgene (133826, 134073); Vaxart, Inc. (202627); and the Rachford & Carlotta A.\\nHarris Endowed Chair to G.P.N. J.W.H. was supported by an NIH T32 Fellowship\\n(T32CA196585) and an American Cancer Society—Roaring Fork Valley Post-\\ndoctoral Fellowship (PF-20-032-01-CSM). Sequencing was done through Stan-\\nford Genomics supported with NIH S10OD018220 and 1S10OD021763. Some\\nﬁgures were created with BioRender.com. We thank Stanford Functional Geno-\\nmics Core for performing the spatial transcriptomics.\\nAUTHOR CONTRIBUTIONS\\nJ.W.H., M.H., M.B., and N.H. completed in vivo based experiments with\\nT cells. N.H. completedin vitro T cell killing assays. J.W.H., M.H., and C.C.\\ncompleted CODEX multiplexed imaging. J.W.H. processed and analyzed\\nCODEX multiplexed imaging. J.W.H., G.V., and Y.G. developed and validated\\nthe CODEX antibody panel. J.W.H. performed CyTOF. J.W.H. and X.R.C.\\nanalyzed CyTOF data. J.W.H. and B.Z. performed and analyzed RNA-seq ex-\\nperiments. A.J.R. performed LRT, random forest, and TCGA analysis. J.W.H.\\nand G.B. performed neighborhood and spatial context analysis. Y.T. analyzed\\nspatial transcriptomics experiments. J.W.H. wrote the manuscript, and all au-\\nthors edited and approved the ﬁnal version. J.S., M.C., and G.P.N. supervised\\nthe project, provided resources and feedback, and helped in writing the\\nmanuscript.\\nDECLARATION OF INTERESTS\\nG.P.N. has equity in and is a scientiﬁc advisory board member of Akoya Bio-\\nsciences, Inc. A.J.R. is an advisor for Afﬁni-T Therapeutics. G.P.N. and\\nJ.W.H. share a patent related to this work. The other authors declare no\\ncompeting interests.\\nReceived: January 10, 2023\\nRevised: September 6, 2023\\nAccepted: November 10, 2023\\nPublished: December 11, 2023\\nREFERENCES\\n1. Robert, C. (2020). A decade of immune-checkpoint inhibitors in cancer\\ntherapy. Nat. Commun.11, 3801.\\n2. Gide, T.N., Wilmott, J.S., Scolyer, R.A., and Long, G.V. (2018). Primary and\\nacquired resistance to immune checkpoint inhibitors in metastatic mela-\\nnoma. Clin. Cancer Res.24, 1260–1270.\\n3. Yee, C. (2010). Adoptive therapy using antigen-speciﬁc T-cell clones.\\nCancer J.16, 367–373.\\n4. Yee, C. (2014). The use of endogenous T cells for adoptive transfer. Immu-\\nnol. Rev.257, 250–263.\\n5. Restifo, N.P., Dudley, M.E., and Rosenberg, S.A. (2012). Adoptive immu-\\nnotherapy for cancer: harnessing the T cell response. Nat. Rev. Immunol.\\n12, 269–281.\\n6. Park, J.H., Rivie` re, I., Gonen, M., Wang, X., Se´ ne´ chal, B., Curran, K.J.,\\nSauter, C., Wang, Y., Santomasso, B., Mead, E., et al. (2018). Long-term\\nfollow-up of CD19 CAR therapy in acute lymphoblastic leukemia.\\nN. Engl. J. Med.378, 449–459.\\n7. Johnson, L.A., and June, C.H. (2017). Driving gene-engineered T cell\\nimmunotherapy of cancer. Cell Res.27, 38–58.\\n8. Raﬁq, S., Hackett, C.S., and Brentjens, R.J. (2020). Engineering strategies\\nto overcome the current roadblocks in CAR T cell therapy. Nat. Rev. Clin.\\nOncol.17, 147–167.\\n9. Finck, A., Gill, S.I., and June, C.H. (2020). Cancer immunotherapy comes\\nof age and looks for maturity. Nat. Commun.11, 3325–3334.\\n10. Gattinoni, L., Zhong, X.-S., Palmer, D.C., Ji, Y., Hinrichs, C.S., Yu, Z.,\\nWrzesinski, C., Boni, A., Cassard, L., Garvin, L.M., et al. (2009). Wnt\\nsignaling arrests effector T cell differentiation and generates CD8+ mem-\\nory stem cells. Nat. Med.15, 808–813.\\n11. Vodnala, S.K., Eil, R., Kishton, R.J., Sukumar, M., Yamamoto, T.N., Ha, N.-\\nH., Lee, P.-H., Shin, M., Patel, S.J., Yu, Z., et al. (2019). T cell stemness and\\ndysfunction in tumors are triggered by a common mechanism. Science\\n363. eaau0135.\\n12. Kosmides, A.K., Necochea, K., Hickey, J.W., and Schneck, J.P. (2018).\\nSeparating T Cell Targeting Components onto Magnetically Clustered\\nNanoparticles Boosts Activation. Nano Lett.18, 1916–1924.\\n13. Giuffrida, L., Sek, K., Henderson, M.A., Lai, J., Chen, A.X.Y., Meyran, D.,\\nTodd, K.L., Petley, E.V., Mardiana, S., Mølck, C., et al. (2021). CRISPR/\\nCas9 mediated deletion of the adenosine A2A receptor enhances CAR\\nT cell efﬁcacy. Nat. Commun.12, 3236–3318.\\n14. Hickey, J.W., Dong, Y., Chung, J.W., Salathe, S.F., Pruitt, H.C., Li, X.,\\nChang, C., Fraser, A.K., Bessell, C.A., Ewald, A.J., et al. (2019). Engineer-\\ning an Artiﬁcial T-Cell Stimulating Matrix for Immunotherapy. Adv. Mater.\\n31, 1807359.\\n15. Van der Leun, A.M., Thommen, D.S., and Schumacher, T.N. (2020). CD8+\\nT cell states in human cancer: insights from single-cell analysis. Nat. Rev.\\nCancer1–15.\\n16. Henning, A.N., Roychoudhuri, R., and Restifo, N.P. (2018). Epigenetic con-\\ntrol of CD8+ T cell differentiation. Nat. Rev. Immunol.18, 340–356.\\n17. Blank, C.U., Haining, W.N., Held, W., Hogan, P.G., Kallies, A., Lugli, E.,\\nLynn, R.C., Philip, M., Rao, A., Restifo, N.P., et al. (2019). Deﬁning ‘T cell\\nexhaustion. Nat. Rev. Immunol.19, 665–674.\\n18. Gattinoni, L., Speiser, D.E., Lichterfeld, M., and Bonini, C. (2017). T mem-\\nory stem cells in health and disease. Nat. Med.23, 18–27.\\n19. Speiser, D.E., Utzschneider, D.T., Oberle, S.G., M€unz, C., Romero, P., and\\nZehn, D. (2014). T cell differentiation in chronic infection and cancer: func-\\ntional adaptation or exhaustion? Nat. Rev. Immunol.14, 768–774.\\n20. Speiser, D.E., Ho, P.-C., and Verdeil, G. (2016). Regulatory circuits of T cell\\nfunction in cancer. Nat. Rev. Immunol.16, 599–611.\\n21. Baertsch, M.A., Nolan, G.P., and Hickey, J.W. (2022). Multicellular mod-\\nules as clinical diagnostic and therapeutic targets. Trends in cancer8,\\n164–173.\\n22. Hickey, J.W., Neumann, E.K., Radtke, A.J., Camarillo, J.M., Beuschel,\\nR.T., Albanese, A., McDonough, E., Hatler, J., Wiblin, A.E., Fisher, J.,\\net al. (2021). Spatial mapping of protein composition and tissue organiza-\\ntion: a primer for multiplexed antibody-based imaging. Nat. Methods19,\\n284–295.\\n23. Hinrichs, C.S., Borman, Z. a, Gattinoni, L., Yu, Z., Burns, W.R., Huang, J.,\\nKlebanoff, C.A., Johnson, L.A., Kerkar, S.P., Yang, S., et al. (2011). Human\\neffector CD8 + T cells derived from naive rather than memory subsets\\npossess superior traits for adoptive immunotherapy. Blood117, 808–814.\\n24. Krishna, S., Lowery, F.J., Copeland, A.R., Bahadiroglu, E., Mukherjee, R.,\\nJia, L., Anibal, J.T., Sachs, A., Adebola, S.O., Gurusamy, D., et al. (2020).\\nStem-like CD8 T cells mediate response of adoptive cell immunotherapy\\nagainst human cancer. Science370, 1328–1334.\\n25. Klebanoff, C.A., Gattinoni, L., Torabi-Parizi, P., Kerstann, K., Cardones,\\nA.R., Finkelstein, S.E., Palmer, D.C., Antony, P.A., Hwang, S.T., Rosen-\\nberg, S.A., et al. (2005). Central memory self/tumor-reactive CD8+\\nT cells confer superior antitumor immunity compared with effector mem-\\nory T cells. Proc. Natl. Acad. Sci. USA102, 9571–9576.\\n26. Parisi, G., Saco, J.D., Salazar, F.B., Tsoi, J., Krystoﬁnski, P., Puig-Saus,\\nC., Zhang, R., Zhou, J., Cheung-Lau, G.C., Garcia, A.J., et al. (2020).\\nPersistence of adoptively transferred T cells with a kinetically engineered\\nIL-2 receptor agonist. Nat. Commun.11, 660–712.\\n27. Goltsev, Y., Samusik, N., Kennedy-Darling, J., Bhate, S., Hale, M., Vaz-\\nquez, G., Black, S., and Nolan, G.P. (2018). Deep proﬁling of mouse\\nsplenic architecture with CODEX multiplexed imaging. Cell174, 968–\\n981.e15.\\n28. Kennedy-Darling, J., Bhate, S.S., Hickey, J.W., Black, S., Barlow, G.L.,\\nVazquez, G., Venkataraaman, V.G., Samusik, N., Goltsev, Y., Sch€urch,\\nC.M., and Nolan, G.P. (2021). Highly multiplexed tissue imaging using\\nCell Reports42, 113494, December 26, 202315\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b498697a-ddde-412e-b803-a99f6ed1fbc0', embedding=None, metadata={'page_label': '17', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='repeated oligonucleotide exchange reaction. Eur. J. Immunol. 51,\\n1262–1277.\\n29. Black, S., Phillips, D., Hickey, J.W., Kennedy-Darling, J., Venkataraaman,\\nV.G., Samusik, N., Goltsev, Y., Sch€urch, C.M., and Nolan, G.P. (2021).\\nCODEX multiplexed tissue imaging with DNA-conjugated antibodies.\\nNat. Protoc.16, 3802–3835.\\n30. Ellis, L.M., and Hicklin, D.J. (2008). VEGF-targeted therapy: mechanisms\\nof anti-tumour activity. Nat. Rev. Cancer8, 579–591.\\n31. Turley, S.J., Cremasco, V., and Astarita, J.L. (2015). Immunological hall-\\nmarks of stromal cells in the tumour microenvironment. Nat. Rev. Immu-\\nnol.15, 669–682.\\n32. Fukumura, D., Kloepper, J., Amoozgar, Z., Duda, D.G., and Jain, R.K.\\n(2018). Enhancing cancer immunotherapy using antiangiogenics: oppor-\\ntunities and challenges. Nat. Rev. Clin. Oncol.15, 325–340.\\n33. Sch€urch, C.M., Bhate, S.S., Barlow, G.L., Phillips, D.J., Noti, L., Zlobec, I.,\\nChu, P., Black, S., Demeter, J., McIlwain, D.R., et al. (2020). Coordinated\\ncellular neighborhoods orchestrate antitumoral immunity at the colorectal\\ncancer invasive front. Cell182, 1341–1359.e19.\\n34. John W Hickey, Eran Agmon, Nina Horowitz, Matthew Lamore, John B\\nSunwoo, Markus Covert, Garry P Nolan. (2023). Integrating Multiplexed\\nImaging and Multiscale Modeling Identiﬁes Tumor Phenotype Transforma-\\ntion as a Critical Component of Therapeutic T Cell Efﬁcacy. 12–08https://\\ndoi.org/10.1101/2023.12.06.570168.\\n35. Chakraborty, P., Vaena, S.G., Thyagarajan, K., Chatterjee, S., Al-Khami,\\nA., Selvam, S.P., Nguyen, H., Kang, I., Wyatt, M.W., Baliga, U., et al.\\n(2019). Pro-survival lipid sphingosine-1-phosphate metabolically pro-\\ngrams T cells to limit anti-tumor activity. Cell Rep.28, 1879–1893.e7.\\n36. Hajaj, E., Eisenberg, G., Klein, S., Frankenburg, S., Merims, S., Ben David,\\nI., Eisenhaure, T., Henrickson, S.E., Villani, A.C., Hacohen, N., et al. (2020).\\nSLAMF6 deﬁciency augments tumor killing and skews toward an effector\\nphenotype revealing it as a novel T cell checkpoint. Elife9, e52539.\\n37. Bhate, S.S., Barlow, G.L., Sch€urch, C.M., and Nolan, G.P. (2022). Tissue\\nschematics map the specialization of immune tissue motifs and their\\nappropriation by tumors. Cell Systems13, 109–130.\\n38. Ma, L., Dichwalkar, T., Chang, J.Y.H., Cossette, B., Garafola, D., Zhang,\\nA.Q., Fichter, M., Wang, C., Liang, S., Silva, M., et al. (2019). Enhanced\\nCAR–T cell activity against solid tumors by vaccine boosting through the\\nchimeric receptor. Science365, 162–168.\\n39. Chiu, Y.-L., Shan, L., Huang, H., Haupt, C., Bessell, C., Canaday, D.H.,\\nZhang, H., Ho, Y.-C., Powell, J.D., Oelke, M., et al. (2014). Sprouty-2 reg-\\nulates HIV-speciﬁc T cell polyfunctionality. J. Clin. Invest.124, 198–208.\\n40. Dufour, J.H., Dziejman, M., Liu, M.T., Leung, J.H., Lane, T.E., and Luster,\\nA.D. (2002). IFN-g-inducible protein 10 (IP-10; CXCL10)-deﬁcient mice\\nreveal a role for IP-10 in effector T cell generation and trafﬁcking.\\nJ. Immunol.168, 3195–3204.\\n41. Keren, L., Bosse, M., Marquez, D., Angoshtari, R., Jain, S., Varma, S.,\\nYang, S.-R., Kurian, A., Van Valen, D., West, R., et al. (2018). A structured\\ntumor-immune microenvironment in triple negative breast cancer revealed\\nby multiplexed ion beam imaging. Cell174, 1373–1387.e19.\\n42. Fridman, W.H., Meylan, M., Petitprez, F., Sun, C.-M., Italiano, A., and\\nSaute` s-Fridman, C. (2022). B cells and tertiary lymphoid structures as de-\\nterminants of tumour immune contexture and clinical outcome. Nat. Rev.\\nClin. Oncol.1–17.\\n43. Moldoveanu, D., Ramsay, L., Lajoie, M., Anderson-Trocme, L., Lingrand,\\nM., Berry, D., Perus, L.J.M., Wei, Y., Moraes, C., Alkallas, R., et al.\\n(2022). Spatially mapping the immune landscape of melanoma using im-\\naging mass cytometry. Sci. Immunol.7, eabi5072.\\n44. Hoch, T., Schulz, D., Eling, N., Go´ mez, J.M., Levesque, M.P., and Boden-\\nmiller, B. (2022). Multiplexed imaging mass cytometry of the chemokine\\nmilieus in melanoma characterizes features of the response to immuno-\\ntherapy. Sci. Immunol.7, eabk1692.\\n45. Phillips, D., Matusiak, M., Gutierrez, B.R., Bhate, S.S., Barlow, G.L., Jiang,\\nS., Demeter, J., Smythe, K.S., Pierce, R.H., Fling, S.P., et al. (2021). Im-\\nmune cell topography predicts response to PD-1 blockade in cutaneous\\nT cell lymphoma. Nat. Commun.12, 6726–6818.\\n46. Berry, S., Giraldo, N.A., Green, B.F., Cottrell, T.R., Stein, J.E., Engle, E.L.,\\nXu, H., Ogurtsova, A., Roberts, C., Wang, D., et al. (2021). Analysis of mul-\\ntispectral imaging with the AstroPath platform informs efﬁcacy of PD-1\\nblockade. Science372, eaba2609.\\n47. McCaffrey, E.F., 2020. Multiplexed imaging of human tuberculosis granu-\\nlomas uncovers immunoregulatory features conserved across tissue and\\nblood. bioRxiv Published online June 9, 2020. Jun, 9(10.1101),\\npp.2020-06.\\n48. Neumann, E.K., Rivera, E.S., Patterson, N.H., Allen, J.L., Brewer, M., deC-\\naestecker, M.P., Fogo, A.B., Caprioli, R.M., and Spraggins, J.M. (2020).\\nHighly multiplexed immunoﬂuorescence of the human kidney using co-\\ndetection by indexing (codex). biorxiv. 2020–12.\\n49. Patwa, A., Yamashita, R., Long, J., Risom, T., Angelo, M., Keren, L., and\\nRubin, D.L. (2021). Multiplexed imaging analysis of the tumor-immune\\nmicroenvironment reveals predictors of outcome in triple-negative breast\\ncancer. Commun. Biol.4, 852–914.\\n50. Jain, S., Pei, L., Spraggins, J.M., Angelo, M., Carson, J.P., Gehlenborg, N.,\\nGinty, F., Gonc¸ alves, J.P., Hagood, J.S., and Hickey, J.W. (2023). Ad-\\nvances and prospects for the Human BioMolecular Atlas Program\\n(HuBMAP). Nat. Cell Biol., 1–12.\\n51. Strasser, M., Gibbs, D.L., Gascard, P., Bons, J., Hickey, J.W., Sch€urch,\\nC.M., Tan, Y., Black, S., Chu, P., and Ozkan, A. (2023). Concerted epithe-\\nlial and stromal changes during progression of Barrett’s Esophagus to\\ninvasive adenocarcinoma exposed by multi-scale, multi-omics analysis.\\nPreprint at bioRxiv.\\n52. Hickey, J.W., Becker, W.R., Nevins, S.A., Horning, A., Perez, A.E., Zhu, C.,\\nZhu, B., Wei, B., Chiu, R., Chen, D.C., et al. (2023). Organization of the hu-\\nman intestine at single-cell resolution. Nature619, 572–584.\\n53. Matusiak, M., Hickey, J.W., Luca, B., Lu, G., Kidzinski, L., Shu, S., Col-\\nburg, D.R.C., Phillips, D.J., Brubaker, S.W., and Charville, G.W. (2022). A\\nspatial map of human macrophage niches links tissue location with func-\\ntion. Preprint at bioRxiv.\\n54. Ng, T.H.S., Britton, G.J., Hill, E.V., Verhagen, J., Burton, B.R., and Wraith,\\nD.C. (2013). Regulation of adaptive immunity; the role of interleukin-10.\\nFront. Immunol.4, 129.\\n55. Smith, L.K., Boukhaled, G.M., Condotta, S.A., Mazouz, S., Guthmiller, J.J.,\\nVijay, R., Butler, N.S., Bruneau, J., Shoukry, N.H., Krawczyk, C.M., and\\nRicher, M.J. (2018). Interleukin-10 directly inhibits CD8+ T cell function\\nby enhancing N-glycan branching to decrease antigen sensitivity. Immu-\\nnity48, 299–312.e5.\\n56. Saraiva, M., Vieira, P., and O’garra, A. (2020). Biology and therapeutic po-\\ntential of interleukin-10. J. Exp. Med.217, e20190418.\\n57. Bai, Z., Woodhouse, S., Zhao, Z., Arya, R., Govek, K., Kim, D., Lundh, S.,\\nBaysoy, A., Sun, H., Deng, Y., et al. (2022). Single-cell antigen-speciﬁc\\nlandscape of CAR T infusion product identiﬁes determinants of CD19-pos-\\nitive relapse in patients with ALL. Sci. Adv.8, eabj2820.\\n58. Patwa, A., Yamashita, R., Long, J., Risom, T., Angelo, M., Keren, L., and\\nRubin, D.L. (2021). Multiplexed imaging analysis of the tumor-immune\\nmicroenvironment reveals predictors of outcome in triple-negative breast\\ncancer. Communications Biology4, 852.\\n59. Jiang, S., Chan, C.N., Rovira-Clave, X., Chen, H., Bai, Y., Zhu, B., McCaf-\\nfrey, E., Greenwald, N.F., Liu, C., and Barlow, G.L. (2021). Virus-depen-\\ndent Immune Conditioning of Tissue Microenvironments.\\n60. Wrzesinski, C., Paulos, C.M., Kaiser, A., Muranski, P., Palmer, D.C., Gat-\\ntinoni, L., Yu, Z., Rosenberg, S.A., and Restifo, N.P. (2010). Increased in-\\ntensity lymphodepletion enhances tumor treatment efﬁcacy of adoptively\\ntransferred tumor-speciﬁc T cells. J. Immunother.33, 1–7. 1997.\\n61. Lee, M.Y., Bedia, J.S., Bhate, S.S., Barlow, G.L., Phillips, D., Fantl, W.J.,\\nNolan, G.P., and Sch€urch, C.M. (2022). CellSeg: a robust, pre-trained nu-\\ncleus segmentation and pixel quantiﬁcation software for highly multi-\\nplexed ﬂuorescence images. BMC Bioinf.23. 46-17.\\n16 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e00080f9-14fd-494b-9614-9a017728c0f4', embedding=None, metadata={'page_label': '18', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='62. Jiang, S., Chan, C.N., Rovira-Clave´ , X., Chen, H., Bai, Y., Zhu, B., McCaf-\\nfrey, E., Greenwald, N.F., Liu, C., Barlow, G.L., et al. (2022). Combined pro-\\ntein and nucleic acid imaging reveals virus-dependent B cell and macro-\\nphage immunosuppression of tissue microenvironments. Immunity55,\\n1118–1134.e8.\\n63. Twyman-Saint Victor, C., Rech, A.J., Maity, A., Rengan, R., Pauken, K.E.,\\nStelekati, E., Benci, J.L., Xu, B., Dada, H., Odorizzi, P.M., et al. (2015). Ra-\\ndiation and dual checkpoint blockade activate non-redundant immune\\nmechanisms in cancer. Nature520, 373–377.\\n64. Rech, A.J., Dada, H., Kotzin, J.J., Henao-Mejia, J., Minn, A.J., Twyman-\\nSaint Victor, C., and Vonderheide, R.H. (2018). Radiotherapy and CD40\\nActivation Separately Augment Immunity to Checkpoint Blockade in\\nCancerRadiotherapy and aCD40 Augment Immunity in Refractory PDA.\\nCancer Res.78, 4282–4291.\\n65. Narayan, V., Barber-Rotenberg, J.S., Jung, I.-Y., Lacey, S.F., Rech, A.J.,\\nDavis, M.M., Hwang, W.-T., Lal, P., Carpenter, E.L., Maude, S.L., et al.\\n(2022). PSMA-targeting TGFb-insensitive armored CAR T cells in metasta-\\ntic castration-resistant prostate cancer: a phase 1 trial. Nat. Med.28,\\n724–734.\\n66. Smith, J.C., and Sheltzer, J.M. (2022). Genome-wide identiﬁcation and\\nanalysis of prognostic features in human cancers. Cell Rep.38, 110569.\\n67. Bergenstra˚hle, J., Larsson, L., and Lundeberg, J. (2020). Seamless inte-\\ngration of image and molecular analysis for spatial transcriptomics work-\\nﬂows. BMC Genom.21, 1–7.\\nCell Reports42, 113494, December 26, 202317\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='7bd4925d-7219-489a-933a-0e9fe1cdd626', embedding=None, metadata={'page_label': '19', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='STAR+METHODS\\nKEY RESOURCES TABLE\\nREAGENT or RESOURCE SOURCE IDENTIFIER\\nAntibodies\\nCustom-conjugated CyTOF antibodies Table S2.xlsx All information on clones, companies,\\nRRIDs, etc. are included within these\\ndetailed tables\\nCustom-conjugated human CODEX\\nantibodies\\nTable S2.xlsx\\nCustom-conjugated murine CODEX\\nantibodies\\nTable S2.xlsx\\nanti-CD3 (clone 145-2C11) Bioxcell Catalog #BE0001-1; RRID: AB_1817016\\nanti-CD28 (clone 37.51) Bioxcell Catalog #BE0015-1; RRID: AB_1817016\\nTruStain FcXTM (anti-mouse CD16/32)\\nAntibody\\nBiolegend Catalog #101319; RRID: AB_2783137\\nBiological samples\\nHuman FFPE melanoma blocks University Medical Center Mainz See Table S1for details\\nChemicals, peptides, and recombinant proteins\\nPotassium hydroxycitrate tribasic\\nmonohydrate\\nSigma-Aldrich 59847-1G\\ngp100 - KVPRNQDWL AnaSpec AS-62589\\nRecombinant Human IL-2 (carrier-free) Biolegend 589106\\nProtein Transport Inhibitor (Containing\\nMonensin)\\nBD Biosciences 554724\\nProtein Transport Inhibitor (Containing\\nBrefeldin A)\\nBD Biosciences 555029\\nRecombinant Mouse IFN-g (carrier-free) Biolegend 575302\\ncis-Diammineplatinum(II) dichloride\\n(cisplatin)\\nSigma-Aldrich P4394-25MG\\nEMS 16% Paraformaldehyde aqueous Fisher 50-980-487\\nCritical commercial assays\\nxCELLigence Real-Time Cell Analysis Agilent Single Plate\\nPureLink RNA Mini Kit ThermoFisher 12183018A\\nNEBNext Poly(A) mRNA magnetic Isolation\\nModule\\nNew England Biolabs E7490\\nNEBNext Ultra II Directional RNA Library\\nPrep Kit for Illumina\\nNew England Biolabs E7760\\nNEBNext Multiplex Oligos for Illumina New England Biolabs E7500\\nVisium Tissue Optimization slides 10x Genomics 3000394\\nVisium Spatial Gene Expression slides 10x Genomics 2000233\\nDeposited data\\nAll single cell deposited data Dryad https://doi.org/10.5061/dryad.k0p2ngfcc\\nExperimental models: Cell lines\\nB16-F10 tumor cell line ATCC RRID:CVCL_0159\\nExperimental models: Organisms/strains\\nB6 - C57BL/6J Jackson Laboratories RRID:IMSR_JAX:000664\\nPMEL - B6.Cg-Thy1a/Cy Tg(TcraTcrb)\\n8Rest/J\\nJackson Laboratories RRID:IMSR_JAX:005023\\nSoftware and algorithms\\nCODEX Processor https://github.com/nolanlab/CODEX\\n(Continued on next page)\\n18 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5b47c42b-d1a6-4195-a11f-7e75d609032d', embedding=None, metadata={'page_label': '20', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='RESOURCE AVAILABILITY\\nLead contact\\nFurther information and requests for resources and reagents should be directed to and will be fulﬁlled by the lead contact, Garry No-\\nlan (gnolan@stanford.edu).\\nMaterials availability\\nNo shareable materials were created in this manuscript.\\nData and code availability\\nd We also provide processed, quantiﬁed and annotated single-cell CODEX datasets with labeled cell types, neighborhoods, and\\nalso protein expression at Dryad for the time course datasets, different T cell phenotype studies, and the human data. It is avail-\\nable for download from dryad:https://doi.org/10.5061/dryad.k0p2ngfcc.\\nd This paper does not report original code\\nd Any additional information required to reanalyze the data reported in this work paper is available from thelead contactupon\\nrequest\\nEXPERIMENTAL MODEL AND STUDY PARTICIPANT DETAILS\\nCell-lines\\nB16-F10 melanoma cells were obtained from ATCC catalog CRL-6475 and maintained at low passage number.\\nMice\\nB6 and PMEL transgenic mice were maintained as per guidelines approved by Stanford University’s Laboratory Animal Care (APLAC)\\nInstitutional Review Board (Protocol number 33502). All experiments conform to the relevant regulatory standards. C57BL/6J and\\nPMEL mice were purchased from Jackson Laboratories.\\nHuman samples\\nA cohort of 50 patients who were diagnosed with metastatic stage IV melanoma and who received ﬁrst-line immune-checkpoint in-\\nhibitor therapy with either nivolumab, pembrolizumab or ipilimumab plus nivolumab between 2015 and 2019 at the University Medical\\nCenter Mainz, Germany were identiﬁed for this study. Among this cohort, 6 patients with survival follow-up data until April 2023 were\\nretrospectively identiﬁed according to the following selection criteria (see alsoFigure 7A): Histopathological conﬁrmed diagnosis of\\nmelanoma, sufﬁcient pre and post-checkpoint inhibitor therapy start tissue available, >30% uncompromised tumor tissue, complete\\nfollow-up with documentation of treatment outcomes including best overall response according to RECIST criteria, progression-free\\nsurvival and overall survival (for details seeTable S1). Details on clinical-pathological parameters at initial diagnosis were collected\\nfrom the patient’s medical records.\\nThe studies involving human participants were reviewed and approved by the local ethics committee of the University Medical\\nCenter Mainz (Ethik-Komission der Landesa¨ rztekammer Rheinland-Pfalz, No: 2020–14822). The patients provided written informed\\nContinued\\nREAGENT or RESOURCE SOURCE IDENTIFIER\\nSegmenter https://michaellee1.github.io/CellSegSite/\\nindex.html\\nNeighborhood analysis https://github.com/nolanlab/\\nNeighborhoodCoordination\\nTissue Schematic Analysis https://github.com/nolanlab/\\nTissueSchematics\\nScanpy https://scanpy.readthedocs.io/en/stable/\\nGalaxy https://usegalaxy.org/\\nImageJ https://imagej.net/software/ﬁji/\\nSTutility https://ludvigla.github.io/\\nSTUtility_web_site/\\nOther\\nFisherbrandTM SuperfrostTM Plus\\nMicroscope Slides\\nThermo Fisher Scientiﬁc 12-550-15\\nCell Reports42, 113494, December 26, 202319\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='117b83e5-932e-4a0f-a34c-6579ef7f4b43', embedding=None, metadata={'page_label': '21', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='consent to participate in the study. The analysis of the anonymized retrospective data was conducted according to the guidelines of\\nthe Declaration of Helsinki.\\nMETHOD DETAILS\\nT cell culture and stimulation\\nImmune cell isolation\\nMurine cells were obtained from adult mouse lymph nodes and spleens. Obtained cells were treated with ACK lysis buffer to lyse red\\nblood cells, and lysates were ﬁltered through cell strainers to isolate splenocytes.\\nT cell media\\nThe T cell media was RPMI 1640 supplemented with glutamine, 1x non-essential amino acids, 1 mM sodium pyruvate, 0.4x vitamin\\nsolution, 92mM 2-mercaptoethanol, 10mM ciproﬂoxacin, and 10% fetal bovine serum.\\nCD3-coated plate preparation\\nTo each well of a 96-well, U-bottomed plate we added 50mL of a solution of 5mg/mL anti-CD3 (Bioxcell, clone 145-2C11) in PBS. After\\nincubation at 4/C14C overnight, liquid was decanted.\\nT cell stimulation\\nIsolated murine immune cells (all splenocytes harvested) were stimulated by incubation with 1 mM cognate peptide gp100\\n(KVPRNQDWL) and 50 IU/mL IL-2. Cells were seeded at a density of 2-53106 cells/mL. Cells were fed with additional IL-2 in\\nT cell media every other day. On day 5, cells were added to CD3-coated plates in culture media containing 2mg/mL anti-CD28 (Bio-\\nxcell, clone 37.51). On day 8 cells were removed from plates and plated on uncoated plates and fed with IL-2-containing media until\\nday 10.\\n2HC T cell stimulation\\nFor stimulation in the presence of 2HC, we followed the protocol for T cell stimulation as described above. The only exception was\\nthat 2HC was prepared sterile within T cell media and then added to culture media at a concentration of 5 mM and maintained within\\nculture every time cells were fed or exchanged with media on days 3, 5, 8.\\nIn vitroT cell killing assay\\nT cell killing and tumor cell growth rates were determined using the xCELLigence Real-Time Cell Analysis platform. Wells of xCEL-\\nLigence E-plates were coated with gold nanoparticles, and electrical potential was passed across the plate every 15 min. Monitoring\\nof changes in the electrical impedance enabled quantiﬁcation of adherent cells over time. For these assays, T cells were expanded.\\nB16-F10 melanoma cells were split and then left as control cells or treated with 10 ng/mL IFNg for 24 h prior to plating. After the pre-\\ntreatment, 10,000 B16-F10 cells were plated in each well of the xCELLigence E-plate and allowed to adhere for 12 h. Next, T cells\\nwere added at 8:1 effector to target ratio. The growth of tumor cells and killing by T cells was monitored for up to 24 h. Killing was\\ncalculated by normalizing the cell index of each well to the time point just before addition of the T cells and then quantifying the dif-\\nferences between T cell and control wells without T cells over time.\\nCyTOF phenotyping\\nAntibodies\\nPrimary antibody transition metal-conjugates were prepared in-house using 100-mg antibody lots and the MaxPAR antibody conju-\\ngation kit (DVS Sciences) according to the manufacturer’s recommended protocol. Following conjugation, antibodies were diluted in\\nCandor PBS Antibody Stabilization solution and stored at 4/C14C. Information on CyTOF staining panels (phenotyping, ICS, and IL-10\\npanels) is given inTable S2.\\nSurface staining\\nFollowing T cell stimulation on day 10, 23106 cells were stained by addition of 1 mL of 25mM cisplatin in PBS. After 1 min at 4/C14C, the\\nreaction was quenched with 1 mL of fetal bovine serum, and cells were washed with cell staining medium (CSM; PBS with 0.5%\\nbovine serum albumin and 0.02% sodium azide). Cells were blocked with FcBlock (0.25mg/13106 cells) for 15 min at room temper-\\nature, then the surface antibody cocktail was added and incubated 1 h at room temperature on a shaker at 100 rpm. Cells were\\nwashed with CSM and then with PBS. Cells were ﬁxed and stained with intercalators overnight at 4/C14C in a solution of 1.6% parafor-\\nmaldehyde in PBS. The next day, the cells were washed once with CSM and twice with doubly distilled water, resuspended in doubly\\ndistilled water, and analyzed using CyTOF.\\nICS/IL10 staining\\nFollowing T cell stimulation on day 10, 23106 cells were isolated from each condition and separated into restimulation and no stim-\\nulation groups in 100mL T cell media at 500,000 cells per well in a 96 u-bottom plate. Samples were added to previously coated anti-\\nCD3 plates, and anti-CD28 was added to the culture media at 2mg/mL. To inhibit protein transport, 1:350 BD GolgiStop Protein\\nTransport Inhibitor (BD Biosciences), and 1:350 BD GolgiPlug Protein Transport Inhibitor (BD Biosciences) in T cell media was added\\nto the samples. Both groups were incubated at 37/C14C for 4–6 h. Cells were washed with PBS then stained with cisplatin at 25mMi n\\nPBS in 1 mL for 1 min at 4/C14C, quenched with 1 mL of fetal bovine serum, and washed with CSM. Cells were blocked with FcBlock\\n(0.25 mg/13106 cells) for 15 min at room temperature, then the surface antibody cocktail was added and incubated 1 h at room\\n20 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ad106cfe-e3b8-48d0-8e1c-8e229c473f06', embedding=None, metadata={'page_label': '22', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='temperature on a shaker at 100 rpm. Cells were washed with CSM and then with PBS. Then cells were then ﬁxed and permeabilized\\nwith 500mL BD Cytoﬁx/Cytoperm Fixation and Permeabilization Solution (BD Biosciences) 20 min at room temperature or 4/C14C over-\\nnight. Cells were then washed with 13 BD PERM/Wash buffer with 2% bovine serum albumin twice. Cells were stained with intra-\\ncellular cytokine antibodies in PERM/Wash buffer with 2% bovine serum albumin at room temperature for 45 min, washed again in\\nPERM/Wash buffer with 2% bovine serum albumin, and then washed with PBS. Cells were then ﬁxed and stained with intercalators\\novernight at 4/C14C or 20 min at room temperature in a solution of 1.6% paraformaldehyde in PBS. The next day, the cells were washed\\nonce with CSM and twice with doubly distilled water, resuspended in doubly distilled water, and analyzed using CyTOF.\\nMouse tumor model\\nOn day 0, B6 mice were injected with 23 105 B16-F10 melanoma tumor cells. On day 0, immune cells were isolated from a PMEL\\nmouse and stimulated as described above with or without 2HC for 10 days to produce stimulated T cells for adoptive transfer. On day\\n9, mice were given a central dose of 500 cGy, which induces transient lymphopenia.60 On day 10, T cells culturedex vivowere har-\\nvested and adoptively transferred intravenously in volumes of 100mL with 13106 of conventionally stimulated or 2HC-activated\\nT cells per mouse. Tumors were harvested 1, 3, 5, or 12 days after adoptive transfer.\\nCODEX multiplexed imaging\\nArray creation\\nImaging data was collected from multiple mice from multiple experiments. We included up to six tumors into a single array, which\\nwere subsequently cut onto the same coverslip. Arrays were constructed on the cryostat and sectioned at a width of 7mm. Available\\nhuman tumor tissue was cut with a 3mm section thickness and mounted onto Superfrost PLUS slides for further processing. For hu-\\nman tumors, we investigated a total of 12 tissues (each 6 pre-treatment and 6 post-treatment) from 6 metastatic melanoma patients\\nthat were treated with immune-checkpoint inhibitors all placed on individual slides.\\nCODEX antibody conjugation and panel creation\\nCODEX multiplexed imaging was executed according to the CODEX staining and imaging protocol.29 CODEX imaging involves iter-\\natively annealing and stripping of ﬂuorophore-labeled oligonucleotide barcodes complimentary to the barcodes attached to 40+ an-\\ntibodies used to stain the tissue. Antibody panels were chosen to include targets that identify subtypes of tumor, stromal, innate, and\\nadaptive immune cells. Detailed panel information can be found inTable S2. Each antibody was conjugated to a unique oligonucle-\\notide barcode, after which the tissues were stained with the antibody-oligonucleotide conjugates. We validated that staining patterns\\nmatched patterns observed by immunohistochemical analysis within positive control tissues of tumor or mouse spleen. Antibody-\\noligonucleotide conjugates were ﬁrst tested and titrated in low-plex ﬂuorescence assays, and signal-to-noise ratio was evaluated,\\nthen antibody-oligonucleotide conjugates were tested together in a single CODEX multicycle. Signal-to-noise ratio was again eval-\\nuated, and the optimal dilution, exposure time, and appropriate imagine cycle was determined for each conjugate (Table S2).\\nCODEX multiplexed imaging\\nThe tissue arrays were stained with the validated panels of CODEX antibodies and imaged.29 Brieﬂy, this entailed cyclic stripping,\\nannealing, and imaging of ﬂuorescently labeled oligonucleotides complementary to the oligonucleotide conjugated to the antibody.\\nEach array underwent CODEX multiplexed imaging; metadata from each CODEX run can be found inTable S2.\\nRNA-seq\\nLibrary preparation\\nRNA isolation was performed with PureLink RNA Mini Kit (ThermoFisher, USA) following the protocol for monolayer cells. The\\nNEBNext Poly(A) mRNA magnetic Isolation Module (E7490, New England Biolabs) was used for mRNA puriﬁcation. For the library\\npreparation, the NEBNext Ultra II Directional RNA Library Prep Kit for Illumina (E7760) and NEBNext Multiplex Oligos for Illumina\\n(E7500) were used. The quality of the library was checked using an Agilent Bioanalyzer. RNA-seq libraries were sequenced by the\\nStanford Genomics Core service, The sequencing data was generated with instrumentation purchased with NIH funds.\\nSpatial transcriptomics\\nVisium spatial gene expression acquisition\\nB16-F10 tumors were embedded in OCT (TissueTek Sakura) and cryo sectioned at 10-mm thickness at/C020/C14C. Tissue sections were\\nplaced on Visium Tissue Optimization slides (3000394, 10x Genomics) and Visium Spatial Gene Expression slides (2000233, 10x Ge-\\nnomics). The tissue optimization sample slide and spatial gene expression slide were processed according to the manufacturer’s\\nprotocols.\\nQUANTIFICATION AND STATISTICAL ANALYSIS\\nCODEX multiplexed imaging data analysis\\nCODEX data processing\\nRaw imaging data were processed using the CODEX Uploader for image stitching, drift compensation, deconvolution, and cycle\\nconcatenation. CODEX enables single-cell resolution protein quantiﬁcation that can be used for evaluating cell type deﬁnition, state,\\nCell Reports42, 113494, December 26, 202321\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ad18c951-6353-426f-8bf0-c5c7fdc2237a', embedding=None, metadata={'page_label': '23', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='and location. To obtain quantitative single cell information, we processed the multiplexed imaging data, segmented individual cells,\\nand extracted single-cell protein expression. Processed data were then segmented using the CellVisionSegmenter, a neural network\\nR-CNN-based single-cell segmentation algorithm.61 Both the CODEX Uploader and Segmenter software can be downloaded from\\nour GitHub site (https://github.com/nolanlab/CODEX), and the CellVisionSegmenter software can be downloaded athttps://github.\\ncom/michaellee1/CellSeg. After the upload, images were evaluated for speciﬁc signal: Any markers that produced an untenable\\npattern or a low signal-to-noise ratio were excluded from the ensuing analysis. Uploaded images were visualized in ImageJ\\n(https://imagej.nih.gov/ij/).\\nCell-type analysis\\nA total of 1,052,125 cells were identiﬁed and classiﬁed into 23 cell types and states based on marker expression for murine studies.\\nWe segmented 5,019,159 individual cells from the 12 CODEX human images and identiﬁed 39 major cell types based on their expres-\\nsion proﬁles. Cell type identiﬁcation were done following the strategies we have developed.29,52 Brieﬂy, nucleated cells were selected\\nby gating DRAQ5, Hoechst double-positive cells, followed by z-normalization of protein markers used for clustering (some pheno-\\ntypic markers were not used in the unsupervised clustering). The data were overclustered with Leiden-based clustering with the\\nscanpy Python package. Clusters were assigned a cell type based on average cluster protein expression and location within image.\\nImpure clusters were split or reclustered following mapping back to original ﬂuorescent images.\\nCell type across multiple tissue slices\\nWe imaged with CODEX multiplexed imaging 2-3 sections separated by/C241 mm for the same tumor from each mouse. More variation\\nis observed between mouse-to-mouse replicates within an experiment, and the most variation is observed between experiment repli-\\ncate to experiment replicate.\\nNeighborhood identiﬁcation analysis\\nNeighborhood analysis was performed based on code and settings we have developed.33 Brieﬂy a window size of 10 nearest neigh-\\nbors was taken across the tissue cell-type maps. Cells of each type were quantiﬁed for each of these windows. We correlated the\\nabundance of the neighboring cell types to each other for all cell types and identiﬁed distinct modules of cells that co-exist within the\\ntumor microenvironment. Also, these vectors were clustered into commonly composed neighborhoods. We both employed k-means\\nelbow plot and overclustering to inform our choice for the number of clusters of cellular neighborhoods. For overclustering, we would\\ngenerate 20–30 clusters and map the clusters back to the tissue and evaluate for cell-type enrichments to determine overall structure.\\nThen with this evaluation we would merge down into the uniquely identiﬁed neighborhoods. CNs (e.g.,Figure 2D, rows) were deﬁned\\nbased on enrichment or depletion of certain cell types (e.g.,Figure 2B, columns) as compared to average tissue percent composition.\\nCell-type speciﬁc neighborhood identiﬁcation analysis\\nFor T cell-speciﬁc neighborhoods, we used the cell neighbor composition vectors by taking the 10-nearest neighbors surrounding the\\nCD8+ T cells. We then clustered cell neighborhoods with k-means clustering. We also used these vectors for UMAP plots prepared\\nusing the scanpy package in Python. Thus, each point in the UMAP is a CD8+ T cell, and each relationship of points in the UMAP\\nrepresents the similarity of one CD8+ T cell’s neighbor composition to another CD8+ T cell neighbor composition.\\nCell interaction analysis\\nCell interaction analysis was carried out using a method we have developed.62 Brieﬂy, the Delaunay triangulation of cells were iden-\\ntiﬁed within each ﬁeld of view using the x, y position. Cell to cell interactions within 35mm from one another were identiﬁed and base-\\nline distributions were established by performing the calculation for each iteration the cell label was randomly assigned existing x, y\\npositions and compared to observed distances with a Wilcoxon Test. We used 1000 permutations for the distribution. The fold\\nenrichment of distances between the observed data over the mean distances from the permutation test. The log fold of the distances\\nfor each cell type interaction where p values less than 0.05 were plotted for each group using ggplot2 in R. Selected pairs were cho-\\nsen based on distance averages closer than expected for2HC T cellsand difference in distance betweenT cells, except for the CD8+\\nT cell and Tumor PDL1+ MHCI+ combination.\\nSpeciﬁcally, pairs were chosen that had a log fold of the distance less than/C00.15 for2HC T cellcondition and also a difference of\\n0.5 between the2HC T cellsand T cellsconditions.\\nRandom forest analysis\\nThis was conducted on cell type percentages using the randomforestSRC R package (version 2.3.0)63,64. During bootstrapping, a\\nrandomly chosen two-thirds of samples were used to train each tree and remaining (out-of-bag, OOB) samples were used for\\ncross-validation and forest-related estimates. No training was conducted on OOB samples and no iterative parameter optimization\\nwas performed. Variables were then ranked by minimal depth (MD), a dimensionless statistic that measures variable predictiveness in\\ntree-based models. MD is deﬁned as the shortest distance between the root node of a tree and the parent node of a maximal subtree,\\nwhich is the largest subtree whose root node splits on the variable. Smaller MD values indicate greater predictiveness and a tree-\\naveraged threshold MD was used to classify variables as predictive.\\nLikelihood ratio test\\nDifferential cell type or neighborhood percentage was done using a linear mixed-effects model via the R package lme4 1.1–27.1 to\\naccount for experimental replicate.65 Brieﬂy, for each cell type or neighborhood type, the percentage was regressed with experi-\\nmental replicate as a ﬁxed effect. p values were estimated using a likelihood ratio test via the R package lmtest 0.9–38. Signiﬁcance\\nat a false discovery rate of 5% was estimated using the Bioconductor package qvalue (https://www.bioconductor.org/packages/\\nrelease/bioc/html/qvalue.html), after conﬁrmation that null p values followed an approximately uniform (0,1) distribution.\\n22 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='86eee457-418f-48d5-8941-48ebf851f5d6', embedding=None, metadata={'page_label': '24', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Spatial context maps\\nSpatial context maps were created using our code and settings we developed.37 Brieﬂy, the spatial context analysis of CN-CN as-\\nsociations has some similarities with our method to identify CNs but also contains a few key differences. First, we used a larger win-\\ndow size (100 nearest neighbors) for each cell across the tissue. Second, we used CN labels instead of cell type labels to create the\\ncomposition windows (Figure 4H). Third, instead of clustering the windows, we determined the combination that included the fewest\\nCNs to make up more than 85% of the neighborhoods within that window (Figure 4H). This combination informs about prominent\\nassociations of CNs in the window, a feature we termed spatial context. Fourth, we counted each combination and connected\\nthe most prevalent combinations together into a spatial context map (Figure 4H).\\nTo construct the graphs, the combination of the fewest neighborhoods that make up more than 85% of total percentage within that\\nwindow are assigned to a speciﬁc combination. Next, the spatial contexts are quantiﬁed and plotted in a network graph structure with\\ncombination color representing the neighborhood combination and the size of the black circle indicating the relative proportion of the\\ncombination across the tissue. The row show the number of neighborhoods in combinations: row 1, a single neighborhood accounts\\nfor at least 85% of the neighborhoods surrounding the window; row 2, two neighborhoods make up more than 85% of the neighbor-\\nhoods in the window; and row 3 and beyond, multiple neighborhoods are present in the window.\\nThe number of CNs forming a combination provides insight into structural biological mechanisms. In some windows, a single CN\\nmade up more than 85% of all CNs. This indicates structural compartmentalization where there is similar cell type composition, and\\nthus CN composition, compartmentalized across large areas of the tissue. In contrast to compartmentalization, spatial contexts with\\nthree or more CNs in a combination are indicative of CN mixing. The cases where a combination of two CNs make up at least 85% of a\\nwindow are indicative of important interfaces and coordination of structures potentially necessary for function.\\nSpatial context analysis on all samples using shared CNs was used forFigure S9that match the analysis done forFigure 4I from an\\nindividual experiment.\\nBarycentric plots for spatial context maps\\nFirst, all windows with any combination ofImmune Inﬁltrate(light blue), Productive T cell & Tumor(dark blue), andUnproductive\\nT cell & Tumor(red) neighborhoods that is greater than 85% were selected. Second, the relative frequencies of these neighborhoods\\nwere computed and then plotted in the barycentric coordinate system. To quantify this shift in T cell-tumor CN organization, we\\nselected those cells with at least 95 of 100 neighboring cells inProductive T cell & Tumorand Immune InﬁltrateCNs (Figure 5C,\\nrectangle).\\nInstance segmentation\\nWe measured overlap between neighborhood contacts, by constructing a tissue graph of the data.37 Brieﬂy, a 10-nearest neighbor\\ngraph for the cells of each imaging region was constructed using the sklearn.nearest_neighbors module in Python. The connected\\ncomponents of this graph were identiﬁed using the scipy.sparse.csgraph module in Python. Instances that consisted of fewer than 5\\ncells were discarded. We calculated the total intersection of CN instances post identiﬁcation, representing the shared border be-\\ntween each pair of the three CNs for all tumors.\\nNeighborhood-neighborhood motif identiﬁcation\\nWe used a similar method to instance segmentation to identify signiﬁcantly associated cellular neighborhood-neighborhood motifs.37\\nBrieﬂy, motif identiﬁcation leverages computational identiﬁcation of CN instances, or segmented areas of the tissue where multiple\\ncells of the same CN are co-located instead of using individual cells. Then tissue network graphs are created that represent shared\\nedges between instances of CNs and this is used to compute whether pairs of CNs are signiﬁcantly associated more than a permuted\\nnull distribution.\\nTo create a tissue graph for each treatment group, we took the union of the tissue graphs of each unique imaging region. We then\\ncreated a null-set as the graph of the set of cell neighborhood assignments by a sequence of valid transpositions of cell neighborhood\\nassignments. Permuting neighborhood assignments and ﬁxing the number of vertices created the maximum entropy null distribution.\\nOnly two chains with at least ﬁve instances were considered. To identify signiﬁcant chains, p values were Bonferroni corrected by\\nmultiplying by twice the number of tests conducted in each treatment group.\\nCanonical correlation analysis\\nCorrelation analyses were performed using cell type compositions of neighborhoods.33 Brieﬂy, for each cell neighborhood, the log\\ncell neighborhood-speciﬁc cell type frequency of CD8+ T cells and CD8+ PD1+ T cells was computed. For each pair of cell neighbor-\\nhoods, estimated canonical directions for the frequencies of these cells in each cell neighborhood was estimated using the scikit-\\nlearn Python package. For each pair of cell neighborhoods, replicates with no cells assigned to either neighborhood were not\\nincluded in the analysis. The correlation of the projections along these canonical directions was compared to a permutation distri-\\nbution, corresponding to 5000 random permutations of the data. The permutation p value (i.e., the percentage of permutations\\nwith an estimated canonical correlation the exceeded the observed one) was interpreted as the strength of communication.\\nRNA-seq data analysis\\nPrimary data analysis\\nRNA-seq data was processed using Galaxy (https://usegalaxy.org/). First,FASTQC was run followed by theTrim_galore function that\\nwas done with a paired trim using other default settings. NextRNA STARwas run on the trimmed data using default settings with\\nmouse reference m10 to align and eliminate unmapped reads/unannotated in bam ﬁles. Next the functionRun Feature Counts\\nCell Reports42, 113494, December 26, 202323\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a9016361-d99c-4674-81d7-c1fd452f3697', embedding=None, metadata={'page_label': '25', 'file_name': 'hickey_melanoma.pdf', 'file_path': '/content/data/hickey_melanoma.pdf', 'file_type': 'application/pdf', 'file_size': 13596253, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='was used and ran on paired runs.Make data tabularand annotateMyIDs was run using Entrez ID to generate read count per gene\\ntables. Reads were normalized to count per million. Finally, genes were selected with expression of greater than 15 reads in either\\ncondition, and log2 fold change greater than 2.5 for gene ontology. Gene ontology analysis was done with the differentially enriched\\ngenes of 2HC T cells compared to T cells withhttp://geneontology.org/.\\nTCGA comparison analysis\\nTo evaluate the prognostic signiﬁcance of the gene set enriched in 2HC T cells, we used previously reported multivariate Cox pro-\\nportional hazards model Z scores to assess the association of gene expression with survival.66 Brieﬂy, this approach permits the use\\nof gene expression measurements without dichotomization, uses right-censored survival data, and is adjusted for factors such as\\nage and disease stage. We compared the Z scores for the gene set of interest to randomly permuted gene sets of the same size\\nto determine p values.\\nVisium spatial data preprocessing\\nThe standard preprocessing of the Visium spatial data was performed using the STutility package.67 Cells with more than 2000 reads\\nand less than 10% mitochondrial gene expression were kept for downstream analysis. We harvested the tumors for these experi-\\nments 3 days after treatment. Since all T cells expressed granzyme B (Figure 6F), we used this marker to identify areas of T cells\\nin the tumor (Figure 6G, panel 1).\\n24 Cell Reports42, 113494, December 26, 2023\\nArticle\\nll\\nOPEN ACCESS', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='580f045c-7241-4419-9cd5-dc3da9c414da', embedding=None, metadata={'page_label': '1252', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Original Article\\nHigh-multiplex single-cell imaging analysis\\nreveals tumor immune contexture associated\\nwith clinical outcomes after CAR T cell therapy\\nJin Jin,1,2,3,10 Li Lin,1,2,10 Jiao Meng,1,4,10 Lijun Jiang,1,2,10 Man Zhang,5 Yuekun Fang,1,2 Wanying Liu,1,2\\nXiangke Xin,1,2 Xiaolu Long,1,2 Dong Kuang,6 Xilai Ding,7 Miao Zheng,1,2 Yicheng Zhang,1,2,8 Yi Xiao,1,2\\nand Liting Chen1,2,9\\n1Department of Hematology, Tongji Hospital, Tongji Medical College, Huazhong University of Science and Technology, Wuhan 430030, China;2Immunotherapy Research\\nCenter for Hematologic Diseases of Hubei Province, Wuhan 430030, China;3Department of Hematology, Renmin Hospital of Wuhan University, Wuhan 430060, China;\\n4Department of Hematology, The First Afﬁliated Hospital, Harbin Medical University, Harbin 150010, China;5Department of Hematology, Harbin Medical University\\nCancer Hospital, Harbin Medical University, Harbin 150081, China;6Department of Pathology, Tongji Hospital, Tongji Medical College, Huazhong University of\\nScience and Technology, Wuhan 430030, China; 7Biomedical Research Core Facilities, Westlake University, Hangzhou 310024, China;8Key Laboratory of Organ\\nTransplantation, Ministry of Education, Wuhan 430030, China;9Research Institute of Huazhong University of Science and Technology in Shenzhen, Shenzhen 518000,\\nChina\\nChimeric antigen receptor (CAR) T cell therapy has made great\\nprogress in treating lymphoma, yet patient outcomes still vary\\ngreatly. The lymphoma microenvironment may be an impor-\\ntant factor in the efﬁcacy of CAR T therapy. In this study, we\\ndesigned a highly multiplexed imaging mass cytometry (IMC)\\npanel to simultaneously quantify 31 biomarkers from 13 pa-\\ntients with relapsed/refractory diffuse large B cell lymphoma\\n(DLBCL) who received CAR19/22 T cell therapy. A total of\\n20 sections were sampled before CAR T cell infusion or after\\ninfusion when relapse occurred. A total of 35 cell clusters\\nwere identiﬁed, annotated, and subsequently redeﬁned into\\n10 metaclusters. The CD4+ T cell fraction was positively associ-\\nated with remission duration. Signiﬁcantly higher Ki67, CD57,\\nand TIM3 levels and lower CD69 levels in T cells, especially the\\nCD8+/CD4+ Tem and Te cell subsets, were seen in patients with\\npoor outcomes. Cellular neighborhood containing more im-\\nmune cells was associated with longer remission. Fibroblasts\\nand vascular endothelial cells resided much closer to tumor\\ncells in patients with poor response and short remission after\\nCAR T therapy. Our work comprehensively and systematically\\ndissects the relationship between cell composition, state, and\\nspatial arrangement in the DLBCL microenvironment and\\nthe outcomes of CAR T cell therapy, which is beneﬁcial to pre-\\ndict CAR T therapy efﬁcacy.\\nINTRODUCTION\\nChimeric antigen receptor (CAR) T cell immunotherapy is a novel\\nstrategy involving the genetic engineering of a patient’s own T cells\\nto kill tumor cells. It is giving rise to breakthroughs and unprece-\\ndented success in the treatment of hematologic malignancies, espe-\\ncially B cell acute lymphoblastic leukemia and B cell lym-\\nphomas.1– 6 However, growing evidence has revealed that a\\nsubstantial number of patients still experience remission.7 Antigen\\nloss or modulation and poor CAR T cell persistence contribute to\\ndisease relapse after CAR T cell therapy.8– 12 Moreover, the immu-\\nnosuppressive tumor microenvironment (TME), which may render\\nCAR T cells inactive, will be particularly relevant to ineffective\\nCAR T cell therapy for lymphoma.13– 15 The TME is a dynamic\\nmilieu that is assembled by neoplastic cells together with extracel-\\nlular matrix (ECM), tumor vasculature, and a variety of immune\\ncells.16 The ﬁtness advantages afforded by the TME play an impor-\\ntant role in the successful outgrowth of tumors and eventual\\nmetastasis. A positive response to immunotherapy, however, usu-\\nally relies on dynamic interactions between tumor cells and immu-\\nnomodulators inside the TME.17\\nDiffuse large B cell lymphoma (DLBCL) represents 30%– 40% of all\\nnon-Hodgkin lymphomas (NHLs).18 It is a disease with aggressive\\nbehavior, and approximately one-third of all DLBCL patients will\\nbe refractory or resistant to standard therapy.19 Against this back-\\nground of therapeutic futility, CAR T cell therapy represents a\\nReceived 7 October 2023; accepted 15 March 2024;\\nhttps://doi.org/10.1016/j.ymthe.2024.03.023.\\n10These authors contributed equally\\nCorrespondence: Yicheng Zhang, Department of Hematology, Tongji Hospital,\\nTongji Medical College, Huazhong University of Science and Technology,\\nWuhan 430030, China.\\nE-mail:yczhang@tjh.tjmu.edu.cn\\nCorrespondence: Yi Xiao, Department of Hematology, Tongji Hospital, Tongji\\nMedical College, Huazhong University of Science and Technology, Wuhan\\n430030, China.\\nE-mail:yixiao@tjh.tjmu.edu.cn\\nCorrespondence: Liting Chen, Department of Hematology, Tongji Hospital,\\nTongji Medical College, Huazhong University of Science and Technology,\\nWuhan 430030, China.\\nE-mail: ltchen@tjh.tjmu.edu.cn\\n1252 Molecular Therapy Vol. 32 No 5 May 2024 ª 2024 The American Society of Gene and Cell Therapy.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2d2773e6-0d9d-4621-8087-ee0dbdf1664d', embedding=None, metadata={'page_label': '1253', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='major advance in treatment, with an estimated 40% of patients re-\\nmaining in remission at 1 year and possibly being cured.20,21 Bio-\\nlogical heterogeneity in DLBCL is partly driven by cell-of-origin\\nsubtypes and associated genomic lesions, but also by diverse\\ncell types and cell states in the TME.22,23 In contrast to other\\ntypes of cancers, such as carcinomas, where the TME has been\\nwidely investigated, its role in DLBCL pathogenesis and patient\\nsurvival has not been well illustrated. How the cellular composi-\\ntion, status, and spatial arrangement of DLBCL microenviron-\\nments affect the prognosis of CAR T cell therapy remains poorly\\nunderstood.24,25\\nIdentiﬁcation and enumeration of the elements of the microenvi-\\nronment has mostly relied on morphology, immunohistochemistry\\n(IHC), and, to a lesser extent,ﬂow cytometry, which provide a rela-\\ntively low-resolution survey of the cellular subtypes that are present.\\nThe growing demands on understanding subsets within broadly\\ndeﬁned cellular categories and their diverse and often opposing\\nfunctions require a higher-resolution assay of the TME. Spatial tran-\\nscriptomics is a spatially resolved, high-dimensional assessment of\\ngene transcription, which demands much of specimens due to the\\ninstability of RNA.26 Imaging mass cytometry (IMC) is a novel tech-\\nnique to comprehensively quantify the cellular heterogeneity and\\nspatial organization of tumor tissue.27 IMC combines IHC staining\\nusing metal isotope-labeled antibodies with laser ablation and mass\\nspectrometry-based detection to produce high-dimensional images,\\ngenerating images that are comparable to those produced by immu-\\nnoﬂuorescence or IHC but that have the capacity for highly multi-\\nplexed staining.28– 30\\nIn the present study, we designed a panel of 31 antibodies speciﬁct o\\nDLBCL histology. Using this panel, section samples from 13 patients\\nbefore and/or after CAR T cell therapy were imaged. By applying to-\\npological single-cell network analysis to high-dimensional mass cy-\\ntometry images, we uncovered the cellular composition, status, and\\nspatial organization within DLBCL tissue that are relevant to the suc-\\ncess of CAR T cell therapy.\\nRESULTS\\nImmunoproﬁling panel for DLBCL microenvironment\\nassessment\\nTo identify multiple complex cellular clusters within the context of\\nthe DLBCL microenvironment, we performed highly multiplexed\\nimaging and comprehensively quantiﬁed the cellular composition\\nand spatial organization of lymphoma tissue (Figure 1A). As previ-\\nously reported, the TME of B cell lymphomas contains highly var-\\niable numbers of tumor/nontumor cells and ECM. We designed an\\nIMC panel speciﬁcally for DLBCL histology with 31 isotope-labeled\\nantibodies for simultaneously quantifying markers of different cell\\ntypes and cellular status (Table 1). CD19 and CD20 were used for\\nstaining B lymphoma cells. CD3, CD4, and CD8 were T cell\\nmarkers. CD56, CD68, CD163, CD66b, CD11b, CD14, CD15,\\nFOXP3, T-bet, and GATA3 were stained for natural killer (NK)\\ncells, macrophages, tumor-associated macrophages (TAMs),\\nmyeloid-derived suppressor cells, myeloid cells, monocytes, neutro-\\nphiles, T regulatory cells (Tregs), and T helper 1 (Th1) and Th2 cells\\nseparately, which were reported as the main immune cell types\\nwithin the DLBCL TME.23 Fibroblasts, vascular endothelial, and\\ncollagen were labeled by a-smooth muscle actin (a-SMA), CD31\\nand collagen I antibodies separately. These cells encompass almost\\nall of the cell types in the TME. In addition, proliferation marker\\nKi67, cytotoxic marker interferon-g (IFN-g), and granzyme B\\nwere also included. Sixteen of the 31 antibodies were metal tagged\\nwhen purchased, whereas the other 14 were carrier-free immuno-\\nglobulin G (IgG) antibodies conjugated to puriﬁed lanthanide\\nmetals. Antibody speciﬁcities were assessed by chromogenic IHC,\\nand their postconjugation concentrations are listed in Table S1.\\nStained sections were laser ablated at subcellular resolution, and\\nliberated isotopes were detected with a mass cytometer to yield im-\\nages revealing the abundance and location of the 31 proteins of in-\\nterest simultaneously (Figure 1A).\\nImmune microenvironment landscape in DLBCL patients before\\nand after CAR T cell therapy\\nTo proﬁle the immune microenvironment, 13 patients with relapsed/\\nrefractory DLBCL who received CAR19/22 T cell“cocktail” treatment\\nwere enrolled in the present study (Table 1). Each patient’s response\\nand sample collection time are visualized as a bar chart inFigure 1B.\\nOf these 13 patients, section samples were available from 10 before\\nCAR T cell infusion. In 5 of these 10 patients, sections were also avail-\\nable after infusion. In addition, for the other 3 patients, sections were\\navailable only after CAR T cell infusion. A total of 20 section samples\\nwere subjected to multiplex immunostaining with the designed panel.\\nFor each section, 3 regions of interest (ROIs) were randomly chosen\\nand processed for imaging, the distances of which were calculated\\n(Figure S1). The H&E staining image is shown for each ROI\\n(Figure S2).\\nImages were then segmented into single cells using CellProﬁler. A to-\\ntal of 493,715 cells were identiﬁed, and the marker gene expression of\\neach cell as well as the spatial features of the cells were quantiﬁed\\n(Figures 1A and S3). Consistent with previous reports,28– 30 the im-\\nages generated from IMC were qualitatively comparable to those pro-\\nduced by immunoﬂuorescence or IHC (Figure S4).\\nUnsupervised clustering with PhenoGraph identiﬁed 35 cell clus-\\nters that were annotated and assigned based on their differential\\nexpression of markers described in previous studies and known ca-\\nnonical markers.31 These included 8 B cell clusters, 10 cytotoxic\\nCD8+ T cell clusters, 7 CD4+ T cell clusters, 1 NK cell cluster, 3\\nmacrophage clusters, 1 granulocyte cluster, 1 myeloid-derived sup-\\npressor cell (MDSC) cluster, 2ﬁbroblast clusters, and 2 vascular\\nclusters, as illustrated in Figures 1C and 1D. Thereafter, dimen-\\nsionality reduction was carried out using t-stochastic neighbor\\nembedding (t-SNE) as well as uniform manifold approximation\\nand projection, which separated these cells into similar cellular\\nmetaclusters (Figures 1 Ca n d S5A). These 35 cell clusters\\nwere then redeﬁned into 10 metaclusters according to their major\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1253', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e9f5889e-7b3d-448d-9261-30a3f3bcf8a0', embedding=None, metadata={'page_label': '1254', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='cell-type markers. We deﬁned B cells (clusters 21– 28) with high\\nexpression of CD19 and/or CD20 and T cells (clusters 1– 10 and\\n11– 17) with high expression of CD3 together with CD4 or CD8.\\nIn addition, NK (CD56, CD11b) (cluster 30), macrophage M1\\n(high CD68, CD14, CD11b) (clusters 19 and 20), macrophage\\nM2 (high CD68, CD14, CD11b, CD163) (cluster 18), granulocyte\\n(high CD15) (cluster 31), MDSC (high CD11b, CD66b) (cluster\\n29), ﬁbroblast (high collagen I,a-SMA) (clusters 33 and 35), and\\nvascular endothelium (high CD31) (clusters 32 and 34) were as-\\nsigned based on their expression signatures. In addition, the spatial\\ndistribution results of expression for the key markers for cell typing\\nare shown inFigures S5C‒S5F.\\nThese results indicate that IMC analysis with the 31-marker panel\\npossesses the capacity for highly multiplexed staining of DLBCL tis-\\nsues and revealing multiple complex cellular clusters, as well as areas\\nwith distinct tissue architectures within the context of the DLBCL\\nmicroenvironment.\\nFractions of T cells, NK cells, and ﬁbroblasts within the DLBCL\\nmicroenvironment vary between patients with different CAR T\\nprognoses\\nTo reveal how the lymphoma TME inﬂuences the outcome of CAR\\nT cell therapy, the immunoproﬁl e sb e f o r ea n da f t e rC A RTi n f u -\\nsion were resolved (Figure 2A). We ﬁrst assessed the prognostic\\nvalue of the DLBCL microenvironment in the CAR T cell treat-\\nment response by comparing the metaclusters in tumor sections\\nbefore CAR T cell infusion between 5 patients who achieved com-\\nplete remission (CR) and 5 patients with progressive disease (PD).\\nThree biopsies were performed after disease relapsed for patient 12.\\nOnly theﬁrst biopsy was brought into outcomes comparison anal-\\nysis. As shown inFigures 2A (left panels) and 2B, higher fractions\\nof NK cells and ﬁbroblasts were associated with poor response,\\nwhereas no signiﬁcant difference was found with other cell meta-\\nclusters. To investigate the role of the TME in remission duration,\\nwe compared the metaclusters in sections during relapse after CAR\\nT cell infusion between 3 long-term remission (LR) patients and 5\\nFigure 1. High-dimensional single-cell immunophenotyping in the diffuse large B cell lymphoma microenvironment\\n(A) Schematic workﬂow of IMC. Tumor biopsy sections from patients were simultaneously labeled with 31 metal-conjugated antibodies, and serial multiplexed images were\\nobtained through laser ablation. Raw data were processed by cell segmentation, followed by quantitative and spatial analysis related to the prognosis of patients. (B) The\\ntherapy responses of 13 enrolled patients with DLBCL receiving CAR19/22 T cell therapy. Twenty tumor biopsies, including lymph nodes, testes, brain, skin, stomach,\\nadrenal glands, and base of the tongue, were collected and subjected to the IMC technique. Responses were evaluated according to the criteria described in thematerials\\nand methods. Arrows indicate ongoing remission. HSCT, hematopoietic stem cell transplantation; NA, not available; SD, stable disease. (C) Map using t-SNE of 493,715\\nsingle cells from IMC images of tumor sections, colored by cell-type metaclusters. (D) Heatmap showing the annotated phenotype expression proﬁle depicted using the mean\\nnormalized intensity (Z score). The absolute cell counts of each cluster are displayed as a bar plot (right).\\nMolecular Therapy\\n1254 Molecular Therapy Vol. 32 No 5 May 2024', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='23c3d3ca-abf2-4be4-afc9-3b0bbfce8d83', embedding=None, metadata={'page_label': '1255', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='short-term remission patients. Their differentially enriched meta-\\nc l u s t e rs i g n a t u r e sw e r ea sf o l l o w s .As u b t y p eo fC D 1 9+ tumor cells\\nas well as ﬁbroblasts were signiﬁcantly less abundant in patients\\nwith a longer duration of remission. In contrast, higher inﬁltration\\nof CD4+ T cells in the TME was accompanied by a longer duration\\nof remission after CAR T infusion ( Figures 2 A, right panels,\\nand 2B).\\nLess inﬁltration and impaired function of T cells within the TME\\nare associated with poor outcomes of CAR T cell therapy\\nTo shed light on distances between cell types and the frequency of\\ntheir interactions, we performed a neighborhood analysis to look\\nat local enrichment and depletion of cells in the direct surround-\\nings of tumor cells with imcRtools. The distance of each cell to the\\nnearest tumor cell was calculated using the Pythagorean theorem,\\nyielding the spectrum of proximities between cell types and works\\nacross the whole image. As expected, fewer CD4+ T cells and more\\nﬁbroblasts came into close contact with the tumor cells in the TME\\nof patients with poor response or shorter remission duration.\\nInterestingly, less inﬁltration of CD8+ T cells appeared in patients\\nwith shorter remission durations, but no signiﬁcant difference was\\nfound between patients with different responses (Figures 3A– 3D).\\nGiven that immune checkpoint receptors play an inhibitory role in\\nregulating T cell responses, we evaluated their expression levels.\\nAlthough both CD4+ and CD8+ T cell fractions showed no signiﬁcant\\ndifference in the sections of patients with different responses before\\nCAR T cell infusion, checkpoint TIM3 expression on CD8+ T cells\\nTable 1. IMC panel\\nCompartment Marker Main target cells Clone Label Catalog\\nLymphoma CD19 CD19 + B cell lymphoma 6OMP31 142Nd 3142014D\\nCD20 CD20 + B cell lymphoma H1 161Dy 3161029D\\nImmune cell CD68 Macrophages KP1 159Tb 3159035D\\nCD163 TAMs EDHu-1 147Sm 3147021D\\nCD66b MDSCs Polyclonal 153Eu ab218740\\nCD11b Myeloid cells EPR1344 175Lu ab209970\\nCD14 Monocytes EPR3653 144Nd 3144025D\\nCD15 Neutrophils W6D3 149Sm 3149026D\\nCD3 CD3 + T cell Polyclonal 170Er 3170019D\\nCD8 CD8 + T cell D8A8Y 162Dy 3162035D\\nCD4 CD4 + T cell EPR6855 156Gd 3156033D\\nFOXP3 Treg 236A/E7 164Dy 14-4777-82\\nT-BET Th1 D6N8B 145Nd 3145015D\\nGATA3 Th2 L50-823 155Gd 558686\\nCD56 NK cell RNL-1 171Yb ab9018\\nStructural a-SMA Fibroblasts 1A4 141Pr 3141017D\\nCD31 Endothelial EPR3094 151Eu 3151025D\\nCollagen I Collagen I Polyclonal 169Tm 3169023D\\nCell status Ki67 Proliferating B56 168Er 3168022D\\nIFN-g Interferon gamma Polyclonal 150Nd ab9657\\nGranzymeB Granzyme B EPR20129-217 167Er 3167021D\\nCD69 Early T cell activation EPR21814 148Nd ab234512\\nCD45RA Effector T cells HI100 166Er 3166031D\\nCD45RO Memory T cells UCHL1 173Yb 3173016D\\nCCR7 Naive T cell Y59 160Gd ab272938\\nCD103 Exhaustion phenotype of T cells EPR22590-27 172Yb ab254201\\nPD-1 Exhaustion phenotype of T cells EPR4877(2) 165Ho 3165039D\\nCTLA-4 Exhaustion phenotype of T cells CAL49 158Gd ab237712\\nCD366/TIM-3 Exhaustion phenotype of T cells EPR22241 154Sm ab242080\\nEOMES Exhaustion phenotype of T cells WD1928 143Nd 14-4877-82\\nCD57 T cell senescence NK-1 174Yb ab212405\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1255', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='290004df-2707-406c-9bf3-a80c14c24e91', embedding=None, metadata={'page_label': '1256', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='was markedly increased in patients with poor responses (Figures 3E\\nand 3F). On both CD4+ and CD8+ T cells, Ki67 expression was signif-\\nicantly higher in PD patients than in CR patients. Moreover, both\\nCD4+ and CD8+ T cells in patients with longer remission expressed\\nlower levels of the immunosenescence epitope CD57 and higher levels\\nof the activation marker CD69, which suggested better functioning of\\nthese T cells. CD4+ T cells also expressed higher levels of TIM3 when\\nrelapse occurred within 3 months after CAR T cell infusion\\n(Figures 3E and 3F).\\nOverall, these results demonstrate that the lymphoma TME in pa-\\ntients with a poor response or shorter remission duration after\\nCAR T cell therapy was inﬁltrated with fewer and worse-functioning\\nT cells.\\nSubpopulation analysis of T cells associated with outcomes of\\nCAR T cell therapy\\nTo further investigate the subpopulation of T cells within the TME,\\nT cells were divided into subgroups by unsupervised clustering and\\nwere characterized by known expression signatures of speciﬁc T cell\\nsubtypes (Figure 4A). Eleven canonical T cell subsets were identiﬁed:\\nCD4+ naive (Tn) (CD3+CD4+CD45RA+CD45RO/C0CCR7+), CD4+\\ncentral memory (Tcm) (CD3 +CD4+CD45RA/C0CD45RO+CCR7+),\\nCD4+ effector memory (Tem) (CD3 +CD4+CD45RA/C0CD45RO+\\nCCR7/C0), CD4 + effector (Te) (CD3 +CD4+CD45RA+CD45RO/C0\\nCCR7/C0), CD8+ Tn (CD3+CD8+CD45RA+CD45RO/C0CCR7+), CD8+\\nTcm (CD3+CD8+CD45RA/C0CD45RO+CCR7+), CD8+ Tem (CD3+\\nCD8+CD45RA/C0CD45RO+CCR7/C0), CD8+ Te (CD3+CD8+CD45RA+\\nCD45RO/C0CCR7/C0), Treg (CD3 +CD4+FoxP3+), Th1 (CD3 +CD4+\\nFigure 2. Metacluster proportions within the TME associated with CAR T cell therapy prognosis\\n(A) Overlay of unsupervised clustering by Rphenograph on optimized t-SNE (optSNE), separated longitudinally. Each color designates a cluster ID. (B) Frequency of depicted\\ncell subsets in patients with different CAR T therapy responses (CR vs. PD) and different remission durations (long-term vs. short-term). A 2-tailedunpaired t test was\\nconducted for statistical analysis. ns, not signiﬁcant; *p < 0.05, **p < 0.01, ***p < 0.001, ****p < 0.0001. Before, sections sampled before CAR T cell infusion; After, sections\\nsampled after CAR T cell infusion.\\nMolecular Therapy\\n1256 Molecular Therapy Vol. 32 No 5 May 2024', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='fa14406b-c989-4534-906f-3de9f20573c9', embedding=None, metadata={'page_label': '1257', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='CD45RO+T-bet+), and Th2 (CD3+CD4+CD45RO+GATA3+). CD8+\\nTem made up nearly half of the T cells (48.02%), followed by CD8+\\nTe (17.99%) and CD4+ Tem (11.19%) (Figure 4B).\\nIn sections before CAR T therapy, fractions of the T cell subpopula-\\ntions were not signiﬁcantly different between CR and PD patients\\n(red and blue dots) (Figures 4C– 4H andS6, left panels). Considering\\nthat the fractions of total CD4+ and CD8+ T cells were similar (Fig-\\nure 2B), we concluded that there was no signiﬁcant difference in the\\ncell numbers of T cell subpopulations between CR and PD patients\\nbefore CAR T infusion. However, CD8+ Tem, CD8+ Te, and CD4+\\nTem cells more highly expressed the proliferation marker Ki67 in\\nPD patients (Figures 4E– 4G, right panel), which also reﬂects a lower\\nsurvival rate in those subgroups of T cells since the cell numbers did\\nnot change. Moreover, the level of the checkpoint protein TIM3 was\\nsigniﬁcantly higher in CD8+ Tem and CD8+ Te cells of PD patients\\nthan in those of CR patients, but this was not true of other subpop-\\nulations (Figures 4E– 4G and S6). Interestingly, checkpoint TIM3\\nexpression on CD8+ T cells was markedly increased in patients\\nwith poor response, whereas its expression on CD4+ T cells was asso-\\nciated with quick relapse. Other inhibitory receptors did not have\\nthis expression pattern (Figures 4E– 4H andS7).\\nFigure 3. Inﬁltration analysis of CD4+ and CD8+ cells within the DLBCL TME\\n(A and B) Histograms plotting the normalized mean metacluster fractions for CD8+ T cells, CD4+ T cells, NK cells, macrophages, granulocytes, MDSCs, ﬁbroblasts, and\\nvascular endothelium at speciﬁc distances from B cells. (A) Before; (B) after. (C and D) Comparison of metacluster fractions for CD8+ T cells, CD4+ T cells and ﬁbroblasts at\\nspeciﬁc distances (5–10mm [C] and 10–15mm [D]) from B cells between patients with different CAR T therapy responses and remission durations. (E and F) Comparison of the\\nlevels of Ki67, CD57, CD69, IFN-g, and TIM3 on CD8+ (E) and CD4+ (F) T cells between patients with different CAR T therapy responses and remission durations. A 2-tailed\\nunpaired t test was conducted for statistical analysis. *p < 0.05, **p < 0.01, ***p < 0.001, ****p < 0.0001. Before, sections sampled before CAR T cell infusion; After, sections\\nsampled after CAR T cell infusion. Error bars represent 95% conﬁdence interval.\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1257', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c8ce84e5-c082-4b0a-80e3-faf713e4c79c', embedding=None, metadata={'page_label': '1258', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='In sections after CAR T therapy, fractions of CD8+ Tem, CD8+ Te,\\nCD4+ Tem, and CD4+ Te were signiﬁcantly lower in patients who\\nrelapsed 3 months after CAR T infusion than in patients who had\\nLR (purple and green dots) (Figures 4C– 4H, left panels). These 4\\nT cell subpopulations made up 82.27% of the T cells (Figure 4B).\\nMoreover, these cell subgroups in LR patients expressed higher levels\\nof CD69 and IFN-g and lower levels of CD57 and TIM3 (Figures 4C–\\n4H, right panel). IFN-g is a cytokine secreted by cytotoxic lympho-\\ncytes that can directly enhance their motility and cytotoxicity. Higher\\nCD69 and lower CD57 reﬂected a more active status. In summary,\\nthese results showed that impaired cytotoxic function and active sta-\\ntus of T cells, especially CD8+ Tem cells, were associated with shorter\\nremission and that patients were more likely to relapse within\\n3 months after CAR T infusion. Furthermore, TIM3 could be used\\nas a marker for poor response and short remission after CAR T\\ntherapy.\\nCellular neighborhood (CN) within the DLBCL microenvironment\\ncorrelates with CAR T cell therapy outcomes\\nCNs are functional immune units within the TME architecture.32\\nTo illustrate the spatial distribution and neighboring cell relation-\\nships of different types of cells in speciﬁc TMEs, we detected the\\ncorrelation of CN with clinical outcomes after CAR T therapy.\\nCells were grouped based on information contained in their direct\\nFigure 4. Cell fractions and status analysis of canonical T cell subsets\\n(A) Overlay of unsupervised clustering by Rphenograph on optSNE embedding, separated longitudinally. Each color designates a T cell subset. (B) Heatmap showing the\\nphenotype expression and absolute cell counts of 11 canonical T cell subsets. (C‒H) Comparison of the fractions and levels of Ki67, CD57, CD69, IFN-g, and TIM3 on CD8+\\nTcms (C), CD4+ Tcms (D), CD8+ Tem cells (E), CD4+ Tem cells (F), CD8+ Te cells (G), and CD4+ cells (H) between patients with different CAR T therapy responses and\\nremission durations. A 2-tailed unpaired t test was conducted for statistical analysis. *p < 0.05, **p < 0.01, ***p < 0.001, ****p < 0.0001. Before, sections sampled before CAR\\nT cell infusion; After, sections sampled after CAR T cell infusion. Error bars represent 95% conﬁdence interval.\\nMolecular Therapy\\n1258 Molecular Therapy Vol. 32 No 5 May 2024', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9621c6d1-b090-4241-98ae-ebc9fc0fdbcf', embedding=None, metadata={'page_label': '1259', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='neighborhood. The direct CN was deﬁned by using a 20 nearest-\\nneighbor graph. All of the neighbors of each cell were aggregated\\nby computing the fraction of cells of a certain cell type among\\nits neighbors. Based on these measures, cells could be clustered\\ninto CNs. In this study, we computed the fraction of the different\\ncell types among the 20 nearest neighbors and used K-means clus-\\ntering to group cells. We found 2 different CN functional units,\\nCN1 and CN2.\\nIn CN1, immune cells made up the majority of the microenviron-\\nment. CD8+ T cells had the highest proportion, exceeding 25% of\\nCN1, so we called it the CD8+ T cell-enriched CN. Conversely, tumor\\ncells exceeded 60% in CN2, so we called it the tumor cell-enriched CN\\n(Figure 5A). Two typical ROIs dominated by CN1 and CN2, respec-\\ntively, are shown inFigure 5B. A topological map of each section sam-\\nple was generated, and the CN function unit frequency was calculated.\\nThe results suggested that the frequencies of CN units were not asso-\\nciated with CAR T therapy response (red and blue dots). Of note, the\\nCD8+ T cell-enriched CN1 and the tumor cell-enriched CN2 showed\\nopposite correlations with remission duration. A higher frequency of\\nCN1 was linked with a longer remission, whereas a higher frequency\\nof CN2 was associated with fast recurrence (within 3 months) after\\nCAR T therapy (green and purple dots) (Figure 5C). In addition,\\nspatial proximity of clusters was analyzed using Squidpy Python\\npackage. If spots belonging to 2 different clusters are often close to\\neach other, then they will have a high score and can be deﬁned as be-\\ning enriched. We found that higher proximity scores occurred be-\\ntween CD4+ T cell and macrophages as well as MDSCs in PD patients\\ncompared to CR patients (Figure 5D). These results suggest that topo-\\nlogical functional units in the DLBCL TME could be novel bio-\\nmarkers for post-CAR T cell prognosis.\\nFibroblasts and vascular endothelium were enriched near tumor\\ncells in patients with poor outcomes\\nThe tumor cells and tumor-inﬁltrating cells (TICs) within the micro-\\nenvironment are heterogeneously distributed, and their proximity has\\ngreat clinical signiﬁcance.33 We next sought to evaluate the clinical\\nsigniﬁcance of the proximity between tumor cells and TICs. To\\nfurther study these localization patterns, we precisely deﬁned the po-\\nsitions of individual tumor cells and TICs and determined the nu-\\ncleus-to-nucleus distances between any 2 cell types. The cell prox-\\nimity and number were further incorporated by using an“effective\\nscore” parameter (Figures 5E and 5F). This score was calculated by\\ndividing the number of nontumor/tumor cell pairs by the total num-\\nber of nontumor cells across the whole slide to represent the spatial\\nvariation. A higher effective score indicated that there was a higher\\ndensity of tumor cells within a certain distance around the nontumor\\ncells. Interestingly,ﬁbroblasts and vascular endothelial cells were\\ndistributed much closer to tumor cells in patients with poor response\\nand short remission duration after CAR T therapy (Figures 5E and\\n5F). These results indicated that the proximity ofﬁbroblasts and\\nvascular endothelium to tumor cells is an adverse prognostic factor\\nof CAR T cell therapy. Proximity between tumor cells and other\\nTICs was not associated with outcomes.\\nTME reﬂects the consolidation therapeutic efﬁcacy of immune\\ncheckpoint inhibitors after CAR T therapy\\nPatient 12 was diagnosed with relapsing/remitting (r/r) DLBCL and\\nwas admitted to the hospital to pursue CAR19/22 T cell therapy. He\\nobtained CR at 3 months after CAR T cell infusion by positron\\nemission tomography/computed tomography assessment. Anti-pro-\\ngrammed cell death 1 (PD1) monoclonal antibody was given to him\\n6 months after CAR T infusion for consolidation. This patient re-\\nmained in CR until 2 years after CAR T infusion, when the lymphoma\\nrecurred locally in the chest wall. Disease progressed thereafter and 3\\nbiopsies were performed during this procedure (Figure 6A). During\\nthe DLBCL progression, PD1 monoclonal antibody was continuously\\nused. The cellular metaclusters frequencies in the 3 biopsy samples dur-\\ning disease progression are shown inFigure S8.T h e r ew a sn ot r e n di n\\nthe cell composition of the three biopsies over time. However, the\\nexpression level of PD1 was continuously decreased (p < 0.05) in\\nboth CD4+ and CD8+ T cells (Figure 6B). Based on these results, we hy-\\npothesize that when the expression level of PD1 on the T cells surface is\\nhigh, PD1 antibody could speciﬁcally bind to PD1, thereby lifting the\\nimmunosuppressive regulation of T lymphocytes, and the tumor is\\nlimited to local due to the immune pressure from T cells. However,\\nwhen T cells are impaired by reasons other than checkpoints such as\\nPD1, PD1 monoclonal antibodies cannot maintain immune balance,\\nand eventually the tumor progresses.\\nDISCUSSION\\nCAR T cell therapy has been touted as a breakthrough in immuno-\\ntherapy that has the potential of ushering in a new era in cancer treat-\\nment. The lack of robust predictive biomarkers for patient stratiﬁca-\\ntion and monitoring remains a major challenge in theﬁeld of cellular\\nimmunotherapy. The importance of the TME to CAR T cell therapy\\nsuccess has been implied by several studies. Yan et al. evaluated how\\nthe TME inﬂuenced the effectiveness of anti-CD19 CAR T cells in re-\\nfractory B cell NHL.13 By gene expression signature analysis, they\\nfound that higher TAM inﬁltration was negatively associated with\\nremission status. Rodriguez-Garcia et al. demonstrated that CAR\\nT cell-mediated selective elimination of folate receptorb+ TAMs in\\nthe TME delayed tumor progression and prolonged survival by en-\\nriching the population of CD8+ T cells in syngeneic tumor mouse\\nmodels.34 Cancer-associated ﬁbroblasts inhibit CAR T cell functions\\nin preclinical models by limiting T cell inﬁltration through a trans-\\nforming growth factor-b-dependent mechanism. In addition, an in-\\ncrease in monocytic MDSCs after CAR T cell infusion was reported\\nto correlate with poor survival.35 To a degree, the composition of\\nthe TME represents remnants of the normal lymphoid tissue, effaced\\nor inﬁltrated by malignant cells, and the host inﬂammatory response.\\nNo study has performed high-dimensional and single-cell-resolution\\nquantiﬁcation of the cellular heterogeneity and spatial organization to\\nuncover the relationship between the DLBCL TME and CAR T suc-\\ncess. In this study, we applied highly multiplexed IMC to discover the\\nkey TME immune features that are associated with CAR T cell ther-\\napy clinical outcomes. Thus, tumor and nontumor architectures that\\nare deﬁned by single-cell spatial analysis could inform the prognosis\\nbeyond the current clinical classiﬁcations.\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1259', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='09384655-6e3b-48eb-939a-9345a9706700', embedding=None, metadata={'page_label': '1260', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Fibroblasts have multiple functions, including secreting collagens and\\nother ﬁbrous macromolecules, that shape the ECM. Tumor-associ-\\nated ECM andﬁbroblasts in the TME have immunomodulatory ef-\\nfects by extensive reciprocal signaling interactions with tumor cells\\nand crosstalk with inﬁltrating leukocytes. They contribute to promot-\\ning cancer progression and metastasis and lead to resistance to\\nFigure 5. CN and heterogeneous distribution analysis within the DLBCL TME\\n(A) CN composition. (B) Two typical ROIs dominated by CN1 and CN2, respectively. Scale bar: 100mm. (C) Correlation between the frequency of CNs and patient clinical\\noutcomes after CAR T therapy. (D) Spatial neighborhood enrichment results analyzed by Squidpy Python package. High score was deﬁned as 2 different clusters often close\\nto each other. (E) Illustration of the distance analysis involving tumor and nontumor cells. Red dots: tumor cells; green dots: nontumor cells. Effective score, number of paired\\nnontumor cells and tumor cells/number of nontumor cells. (F) Comparison of effective score between patients with different CAR T therapy responses and remission du-\\nrations. Before, sections sampled before CAR T cell infusion; After, sections sampled after CAR T cell infusion. Scale bar, 200mm. *p < 0.05, **p < 0.01, ***p < 0.001.\\nMolecular Therapy\\n1260 Molecular Therapy Vol. 32 No 5 May 2024', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='511cd8e8-6673-4fa3-be7b-de992d1b7fad', embedding=None, metadata={'page_label': '1261', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='therapy and disease recurrence.16,36 Sakemura et al. reported bone\\nmarrow-derived cancer-associatedﬁbroblasts (BM-CAFs) could pro-\\nmote tumor growth bothin vitroand in vivo.37 In addition, BM-CAFs\\ncould also inhibit anti-B cell maturation antigen CAR T cells’ prolif-\\neration and CD107a expression through the secretion of multiple\\nsuppressive cytokines. Sakemura et al. developed multiple myeloma\\ncells and CAFs dual-targeting CAR T cell strategy in mouse models,\\nwhich signiﬁcantly improved the effector functions of CAR T cells.37\\nHowever, the association between CAFs and clinical outcomes of\\nCAR T cell therapy has not been well studied. Here, we found thatﬁ-\\nbroblasts were enriched near tumor cells in patients with poor\\nresponse and short remission after CAR T therapy. These results sug-\\ngest that ﬁbroblasts could be a target for enhancing efﬁciency of\\nimmunotherapy in lymphoma.\\nIn recent years, many studies have reported that NK cells are not only\\n“killers” but also regulate innate and adaptive immunity, especially in\\nthe early stage, by secreting cytokines and cell-to-cell contact to main-\\ntain immune homeostasis.38,39 The negative regulatory effects of NK\\ncells specialized as a regulatory NK cell subset that express CD56bright\\nplay important roles in normal and pathogenic immune responses.40\\nIt is reported that CD56bright NK cells inhibited CD4+ and CD8+ acti-\\nvated T cell survival by a contact-dependent mechanism during\\ninterleukin-2 receptor a (IL-2Ra)-targeted therapy (daclizumab).\\nThis immunoregulation has potential importance for the treatment\\nof autoimmune diseases and transplant rejection and toward the\\nmodiﬁcation of tumor immunity.41 In our study, IMC results of tissue\\nsections from CAR T-treated patients suggested that higher fractions\\nof CD56bright NK cells were associated with poor CAR T therapy\\nresponse. The mechanisms regarding how the NK cells regulate the\\nactivity of CAR T cells within the lymphoma microenvironment\\nneed further investigation. The NK regulator cells would be a target\\nfor enhancing CAR T cell activity.\\nThe deep immunoproﬁling strategies of IMC allow the quantiﬁcation\\nof a remarkable variety of T cell subsets and the range of activation\\nstates. Here, the expression of multiple inhibitory receptors, including\\nPD-1, CTLA-4, TIM3, EOMES, and CD103, was evaluated. We found\\nthat checkpoint TIM3 expression on CD8+ T cells was markedly\\nincreased in patients with poor response, whereas its expression in\\nCD4+ T cells was associated with quick relapse. No other inhibitory\\nreceptors had this expression pattern. Our results show that the\\nTIM3 pathway has nonredundant effects that synergize with PD-1\\nsignaling to dampen antitumor responses, as previously implicated.42\\nThe TIM gene family has 3 genes in humans (TIM1, TIM3, and\\nTIM4), and they encode proteins in the Ig superfamily. Knowledge\\nFigure 6. The expression levels of immune checkpoints in the TME after anti-PD1 consolidation\\n(A) The therapy responses of patient 12 after receiving CAR19/22 T cell therapy. Responses were evaluated according to the criteria described in thematerials and methods.\\nArrows indicate ongoing remission. CR, complete response. (B) The expression levels of status markers as well as checkpoints in both CD4+ and CD8+ T cells during disease\\nprogression after anti-PD1 consolidation. *p < 0.05, **p < 0.01, ***p < 0.001. Error bars represent 95% conﬁdence interval.\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1261', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9a6ff0fd-3a53-4a2e-9bb4-bfd499e247ae', embedding=None, metadata={'page_label': '1262', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='about the intracellular signaling cascade activated by TIM3 during the\\ninduction of exhaustion and senescence is incomplete. Because high\\nexpression of TIM3 is a prominent event in replicative senescence,\\nTIM3 has been targeted in some cellular immunotherapy studies. It\\nwas reported that the blockade of PD-1 in combination with other im-\\nmune checkpoint receptors, including TIM3 and LAG-3, has strong\\nsynergistic effects and boosts the effector functions of CAR\\nT cells.43,44 In another study, antibody-based blocking of galectin-9,\\na putative TIM3 ligand, reduced the exhaustion of CAR T cells and\\nsigniﬁcantly increased their cytotoxicity against previously resistant\\ntumor cells.\\nIn the present study, CD8+ Tem, CD8+ Te, and CD4+ Tem cells ex-\\npressed higher levels of Ki67 in patients with poor outcomes after\\nCAR T therapy. Ki67 is a proliferation marker, but the numbers of\\nthese Ki67+ cells were comparable to or even lower than those of\\nKi67/C0T cells. Recent studies reported that the high-Ki67 subpopula-\\ntion displayed a higher expression of burnout markers such as PD-1,\\nLAG-3, TIM3, and FAS, which was consistent with our results29\\n(Figure 4).\\nIMC increases the number of detected proteins on the same section\\ncompared to traditional IHC. However, there are still some limita-\\ntions in IMC. To apply the IMC assay, target proteins need to be\\ndetermined before detection, which are usually previously reported\\nproteins or those of interest to the investigators. Thus, IMC could\\nnot be applied unbiasedly. Spatial transcriptomics, which resolves\\nsection samples at the genome-wide transcription level, can make\\nup for this bias shortcoming of IMC. However, due to the instability\\nof RNA, the spatial transcriptome is more demanding for speci-\\nmens.45 Future studies combining IMC and spatial transcriptomics,\\nwhich correlates speciﬁc protein markers with gene expression pat-\\nterns to elucidate the lymphoma microenvironment, are worth\\nconducting.\\nIn conclusion, we used IMC to simultaneously quantify 31 bio-\\nmarkers, resulting in 20 high-dimensional pathological images of tu-\\nmor tissue from 13 patients with r/r DLBCL who had received CAR\\nT cell therapy. Our work comprehensively and systematically dissects\\nthe relationship between the cell composition, state, and spatial\\narrangement of the DLBCL microenvironment and the outcomes of\\nCAR T cell therapy. CAFs, NK regulator cells, and inhibitory receptor\\nTIM3 on T cells are suggested as targets for remolding the lymphoma\\nmicroenvironment to enhance CAR T cell efﬁciency.\\nMATERIALS AND METHODS\\nPatients and samples\\nTwenty tissue samples were obtained from 13 patients who were diag-\\nnosed with diffuse large B cell lymphoma receiving CAR19/22 T cell\\n“cocktail” therapy at Tongji Hospital, Huazhong University of Sci-\\nence & Technology, Wuhan, China, from July 2018 to November\\n2021. The efﬁcacy of CAR T therapy was evaluated monthly for\\n6 months and then every 3 months according to the National\\nComprehensive Cancer Network guidelines and Lugano Treatment\\nResponse Criteria.46 All of the patients were followed up until trans-\\nplantation, death, loss to follow-up, or the cutoff date for data collec-\\ntion (August 30, 2022). Twenty biopsy tissues were taken from\\ndifferent sites: 9 from lymph nodes, 2 from the brain, 2 from the\\nskin and soft tissue, 2 from the mammary glands, 1 from the stomach,\\n1 from the testes, 1 from the adrenal glands, 1 from the mediastinal\\ndiaphragm and 1 from the base of the tongue. All of the samples\\nwere veriﬁed by pathological examinations. We performed H&E\\nstaining on every tissue sample, and the resulting stained samples\\nwere examined with an optical microscope (Nikon Eclipse Ci) by a\\npathology expert to select the ROIs. We chose at least 3 ROIs for\\neach tissue to be scanned by IMC. Our study was conducted after\\napproval by the institutional review board, and informed consent\\nwas obtained from individuals in strict accordance with the principles\\nstipulated in the Declaration of Helsinki.\\nIMC panel design\\nAn antibody panel was designed to target the DLBCL TME and T/NK\\ncell state. The markers distinguishing the DLBCL TME consist mainly\\nof lymphoma cells, multiple immune cells, and stromal cells, and the\\nmarkers of T/NK cell status mainly consist of functional and deple-\\ntion markers (Table 2).\\nAntibodies and metal conjugation\\nBefore metal conjugation, antibody speciﬁcity was assessed by IHC. An-\\ntibodies were conjugated to metals using the MaxPar X8 Multimetal La-\\nbeling Kit (Fluidigm) according to the manufacturer’s protocol. Car-\\nrier-free IgG antibodies (concentrations between 0.5 and 1 mg/mL)\\nwere conjugated to puriﬁed lanthanide metals (Fluidigm) (Table S1).\\nAfter conjugation, the stock solution of each antibody was diluted to\\na concentration of 0.5 mg/mL in antibody stabilizer supplemented\\nwith 0.05% sodium azide (Table S1.). Antibodies were stored at 4/C14C\\nand remained stable for at least 6 months. To exclude the possibility\\nthat the labeling process substantially affected the performance of the\\nantibodies, they were tested by IMC.\\nROI selection\\nTwenty formalin-ﬁxed parafﬁn-embedded (FFPE) tissue sections\\nwere selected for H&E staining to select the ROI. They were imaged\\nwith a slide scanner (3DHISTECH CaseViewer, Pannoramic SCAN)\\nat 20/C2magniﬁcation. ROIs were manually selected based on the\\nmorphology of the lymphoma cells. Three ROIs were selected per\\nslice, each 1 mm2 (Figure S2B).\\nIMC section staining\\nTwenty FFPE tissue samples were stained with our full antibody panel\\n(Table 2). The FFPE tissue samples were heated at 62/C14C for 1 h. They\\nwere dewaxed by incubating the sections in xylene at 68/C14C for 10 min\\ntwice. The sections were then rehydrated for 5 min each in 100%,\\n90%, and 70% ethanol at room temperature (RT), then washed 3\\ntimes with double-distilled H2O (ddH2O) for 2 min each time.\\nHeat-mediated antigen retrieval was performed for 30 min at 96/C14C\\nin sodium citrate solution. After cooling naturally to RT, the sections\\nwere washed twice with ddH2O for 5 min and then twice with PBS\\nMolecular Therapy\\n1262 Molecular Therapy Vol. 32 No 5 May 2024', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='28a9d898-1bc9-48e5-9803-ec1e0acae086', embedding=None, metadata={'page_label': '1263', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='containing 0.5% Tween 20 and 1% bovine serum albumin (PBS-Tris\\nbuffer [TB]) for 5 min. The section was then blocked with 3% MACS\\nBSA stock solution for 45 min at RT. After washing 3 times in PBS-\\nTB, the sections were incubated overnight at 4/C14C with an antibody\\ncocktail. Four additional PBS-TB washes were performed for 8 min\\neach. To label cell nuclei, the sections were incubated with an interca-\\nlator-Ir (201192B, Fluidigm) solution in PBS-TB (1.25mM) for\\n30 min at RT, followed by washing twice with ddH2O. They were\\ndried at RT for 20 min.\\nIMC image acquisition\\nData acquisition was performed on a Helios time-of-ﬂight mass cy-\\ntometer coupled to a Hyperion Imaging System (Fluidigm). Before\\nlaser ablation, optical images of slides were acquired using Hyperion\\nsoftware, and the areas to ablate were selected as described above.\\nLaser ablation was performed at a resolution of/C241 mm and a fre-\\nquency of 200 Hz, and the ablation of 1 area took 2 h. The data\\nwere exported as MCD formatﬁles and were visualized in the Fluid-\\nigm MCD viewer.\\nIMC analysis\\nAfter acquiring the IMC data (MCD and textﬁle format) from the\\nHyperion Imaging System, Fluidigm MCD viewer (version\\n1.0.560.2) was used to visualize the marker expression of each ROI\\non the tissue sections. The raw MCD formatﬁles were converted to\\nstacked and multichannel TIFF (tagged imageﬁle format)ﬁles. Hot\\npixels were removed and areas were randomly cropped into\\n500/C2500 pixels by CellProﬁler.47 The ilastik machine learning algo-\\nrithm was trained on manual annotations to distinguish between\\nnuclei, cytoplasm, and background regions.48 This step of supervised\\nlearning generated probability maps, which were exported for each\\nsample and batch applied to the complete images. These probability\\nmaps were then returned to CellProﬁler to perform the masks. Nuclei\\nwere identiﬁed as primary objects with a diameter between 5 and 25\\npixels.49 Finally, we extracted single-cell features from these\\nimages using HistoCAT version 1.7.6 software.50 For single-cell pro-\\ncessing, we used Harmony version 0.1.051 to eliminate batch effects.\\nRphenograph version 0.99.1 was used for clustering (https://github.\\ncom/JinmiaoChenLab/Rphenograph). A heatmap was generated to\\ndemonstrate the medianZ score (0– 1) marker expression of cells in\\neach cluster with ComplexHeatmap version 2.10.0. For dimension-\\nality reduction, we used visualization by t-SNE to determine the\\nphenotypic diversity of cell populations. Cytomapper version 1.6.0\\nwas applied to support the handling and visualization of multiple\\nmultichannel images and segmentation masks. Spatial analysis and\\nneighborhood analysis were performed by imcRtools version 1.1.7.\\nStatistical analysis\\nFor expression differentiation analysis, the expression matrix was\\nformed with log10 (counts+1). When analyzing intergroup differ-\\nences, the median expression value of each sample was subtracted\\nfrom the expression level of the sample. Differences were assessed\\nin R version 4.1.3 using the 2-tailed Student’s t test, with p values\\n<0.05 considered signiﬁcant. Statistical analysis was performed by\\nTable 2. Clinical characteristics of patients\\nDLBCL (n = 13)\\nAge\\nYears, median (range) 50 (25 – 61)\\nSex\\nFemale 5 (38.5)\\nTumor burden\\nLDH baseline 275.5 (162 – 741)\\nLarge tumor (>10 cm) 4 (30.8)\\nTP53 mutation 3 (23.1)\\nTumor stage\\nI– II 0 (0.0)\\nII– IV 13 (100.0)\\n“Double” and “triple” hit\\nYes 3 (23.1)\\nNo 7 (53.8)\\nUndetermined 3 (23.1)\\nExtranodal invasion\\nLymph node 9 (45.0)\\nSkin 2 (10.0)\\nTestis 1 (5.0)\\nCNS 2 (10.0)\\nLung, stomach, adrenal glands, tongue, etc. 6 (30.0)\\nPrior HSCT\\nAutologous 2(15.4)\\nAllogeneic 1 (7.7)\\nPrior treatment\\nFirst line 3 (23.1)\\nSecond line 7 (53.8)\\nRThird line 3 (23.1)\\nRefractory or relapsed\\nRefractory 3 (23.1)\\nFirst relapse 7 (53.8)\\nRSecond relapse 3 (23.1)\\nCAR T cell dose,/C2106 cells/kg, median (range)\\nCAR19 T cells 3.66 (11.6 – 4.08)\\nCAR22 T cells 4.00 (2.00 – 4.86)\\nCRS grade\\n0– 2 13 (100.0)\\n3– 5 0 (0.0)\\nOutcomes\\nResponse 7 (53.8)\\nNo response 6 (46.2)\\nFollow-up time\\nDays, median (range) 725 (56 – 1677)\\nCRS, cytokine release syndrome; LDH,lactate dehydrogenase.\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1263', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0949fd1c-e04a-4b1f-9767-7b05af97d061', embedding=None, metadata={'page_label': '1264', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='using IBM SPSS Statistics software (version 19). Adobe Illustrator\\n2020 and GraphPad Prism 8 were used for ﬁgure editing. The\\nHolm-Bonferroni method was used to counteract the problem of\\nmultiple comparisons. All of the adjusted p values are provided in\\nTable S2.\\nDATA AND CODE AVAILABILITY\\nThe data used or analyzed during the present study are available on\\nrequest from the corresponding author, upon reasonable request.\\nSUPPLEMENTAL INFORMATION\\nSupplemental information can be found online athttps://doi.org/10.\\n1016/j.ymthe.2024.03.023.\\nACKNOWLEDGMENTS\\nThis work was supported in part by the National Natural Science\\nFoundation of China (no. 82270238 to L.C.); the National High Tech-\\nnology Research and Development Program of China (No.\\n2021YFA1101504 to Y.Z.); and the National Natural Science Founda-\\ntion of China (nos. 82070213 and 82370196 to Y.X.). This study was\\napproved by the medical ethics committee of Tongji Hospital, Tongji\\nMedical College, Huazhong University of Science and Technology\\n(TJ-IRB202303138).\\nAUTHOR CONTRIBUTIONS\\nJ.J., J.M., L.L., and L.J. designed and performed the experiments,\\ncurated the data, and applied the formal analysis. Y.F., W.L., X.X.,\\nX.L., D.K., M.Z., and X.D. assisted in the experimental work. M.Z. im-\\nplemented the computer code and supporting algorithms. L.C., Y.Z.,\\nY.X., and J.J. visualized the data and interpreted the outcomes. L.C.\\nconceptualized the study, supervised the experiments, and wrote\\nthe manuscript. L.C., Y.Z., and Y.X. obtained the funding. We appre-\\nciate Mengtao Zhan from Nanjing Aoyin Biotechnology for imple-\\nmenting the computer code and supporting algorithms.\\nDECLARATION OF INTERESTS\\nThe authors declare no competing interests.\\nREFERENCES\\n1. Sadelain, M., Brentjens, R., and Rivière, I. (2013). The basic principles of chimeric an-\\ntigen receptor design. Cancer Discov.3, 388– 398.\\n2. June, C.H., O’Connor, R.S., Kawalekar, O.U., Ghassemi, S., and Milone, M.C. (2018).\\nCAR T cell immunotherapy for human cancer. Science359, 1361– 1365.\\n3. Neelapu, S.S., Locke, F.L., Bartlett, N.L., Lekakis, L.J., Miklos, D.B., Jacobson, C.A.,\\nBraunschweig, I., Oluwole, O.O., Siddiqi, T., Lin, Y., et al. (2017). Axicabtagene\\nCiloleucel CAR T-Cell Therapy in Refractory Large B-Cell Lymphoma. N. Engl. J.\\nMed. 377, 2531– 2544.\\n4. Maude, S.L., Laetsch, T.W., Buechner, J., Rives, S., Boyer, M., Bittencourt, H., Bader,\\nP., Verneris, M.R., Stefanski, H.E., Myers, G.D., et al. (2018). Tisagenlecleucel in\\nChildren and Young Adults with B-Cell Lymphoblastic Leukemia. N. Engl. J. Med.\\n378, 439– 448.\\n5. Frey, N.V. (2022). Approval of brexucabtagene autoleucel for adults with relapsed\\nand refractory acute lymphocytic leukemia. Blood140,1 1– 15.\\n6. Abramson, J.S., Palomba, M.L., Gordon, L.I., Lunning, M.A., Wang, M., Arnason, J.,\\nMehta, A., Purev, E., Maloney, D.G., Andreadis, C., et al. (2020). Lisocabtagene mar-\\naleucel for patients with relapsed or refractory large B-cell lymphomas\\n(TRANSCEND NHL 001): a multicentre seamless design study. Lancet396, 839– 852.\\n7. Hegde, P.S., and Chen, D.S. (2020). Top 10 Challenges in Cancer Immunotherapy.\\nImmunity 52,1 7– 35.\\n8. Majzner, R.G., and Mackall, C.L. (2018). Tumor Antigen Escape from CAR T-cell\\nTherapy. Cancer Discov.8, 1219– 1226.\\n9. Xu, X., Sun, Q., Liang, X., Chen, Z., Zhang, X., Zhou, X., Li, M., Tu, H., Liu, Y., Tu, S.,\\nand Li, Y. (2019). Mechanisms of Relapse After CD19 CAR T-Cell Therapy for Acute\\nLymphoblastic Leukemia and Its Prevention and Treatment Strategies. Front.\\nImmunol. 10, 2664.\\n10. Cheng, J., Zhao, L., Zhang, Y., Qin, Y., Guan, Y., Zhang, T., Liu, C., and Zhou, J.\\n(2019). Understanding the Mechanisms of Resistance to CAR T-Cell Therapy in\\nMalignancies. Front. Oncol.9, 1237.\\n11. Srivastava, S., and Riddell, S.R. (2018). Chimeric Antigen Receptor T Cell Therapy:\\nChallenges to Bench-to-Bedside Efﬁcacy. J. Immunol.200, 459– 468.\\n12. Sterner, R.C., and Sterner, R.M. (2021). CAR-T cell therapy: current limitations and\\npotential strategies. Blood Cancer J.11,6 9.\\n13. Yan, Z.X., Li, L., Wang, W., OuYang, B.S., Cheng, S., Wang, L., Wu, W., Xu, P.P.,\\nMuftuoglu, M., Hao, M., et al. (2019). Clinical Ef ﬁcacy and Tumor\\nMicroenvironment Inﬂuence in a Dose-Escalation Study of Anti-CD19 Chimeric\\nAntigen Receptor T Cells in Refractory B-Cell Non-Hodgkin’s Lymphoma. Clin.\\nCancer Res.25, 6995– 7003.\\n14. Junttila, M.R., and de Sauvage, F.J. (2013). Inﬂuence of tumour micro-environment\\nheterogeneity on therapeutic response. Nature501, 346– 354.\\n15. Scholler, N., Perbost, R., Locke, F.L., Jain, M.D., Turcan, S., Danan, C., Chang, E.C.,\\nNeelapu, S.S., Miklos, D.B., Jacobson, C.A., et al. (2022). Tumor immune contexture\\nis a determinant of anti-CD19 CAR T cell efﬁcacy in large B cell lymphoma. Nat.\\nMed. 28, 1872– 1882.\\n16. Gajewski, T.F., Schreiber, H., and Fu, Y.X. (2013). Innate and adaptive immune cells\\nin the tumor microenvironment. Nat. Immunol.14, 1014– 1022.\\n17. Tang, H., Qiao, J., and Fu, Y.X. (2016). Immunotherapy and tumor microenviron-\\nment. Cancer Lett.370,8 5– 90.\\n18. Li, S., Young, K.H., and Medeiros, L.J. (2018). Diffuse large B-cell lymphoma.\\nPathology 50,7 4– 87.\\n19. Goldﬁnger, M., and Cooper, D.L. (2022). Refractory DLBCL: Challenges and\\nTreatment. Clin. Lymphoma Myeloma Leuk.22, 140– 148.\\n20. Spiegel, J.Y., Dahiya, S., Jain, M.D., Tamaresis, J., Nastoupil, L.J., Jacobs, M.T.,\\nGhobadi, A., Lin, Y., Lunning, M., Lekakis, L., et al. (2021). Outcomes of patients\\nwith large B-cell lymphoma progressing after axicabtagene ciloleucel therapy.\\nBlood 137, 1832– 1835.\\n21. Schuster, S.J., Bishop, M.R., Tam, C.S., Waller, E.K., Borchmann, P., McGuirk, J.P.,\\nJäger, U., Jaglowski, S., Andreadis, C., Westin, J.R., et al. (2019). Tisagenlecleucel in\\nAdult Relapsed or Refractory Diffuse Large B-Cell Lymphoma. N. Engl. J. Med.\\n380,4 5– 56.\\n22. Steen, C.B., Luca, B.A., Esfahani, M.S., Azizi, A., Sworder, B.J., Nabet, B.Y., Kurtz,\\nD.M., Liu, C.L., Khameneh, F., Advani, R.H., et al. (2021). The landscape of tumor\\ncell states and ecosystems in diffuse large B cell lymphoma. Cancer Cell39, 1422–\\n1437.e10.\\n23. Scott, D.W., and Gascoyne, R.D. (2014). The tumour microenvironment in B cell\\nlymphomas. Nat. Rev. Cancer14, 517– 534.\\n24. Seferbekova, Z., Lomakin, A., Yates, L.R., and Gerstung, M. (2023). Spatial biology of\\ncancer evolution. Nat. Rev. Genet.24, 295– 313.\\n25. Giraldo, N.A., Sanchez-Salas, R., Peske, J.D., Vano, Y., Becht, E., Petitprez, F.,\\nValidire, P., Ingels, A., Cathelineau, X., Fridman, W.H., and Sautès-Fridman, C.\\n(2019). The clinical role of the TME in solid cancer. Br. J. Cancer120,4 5– 53.\\n26. Rao, A., Barkley, D., França, G.S., and Yanai, I. (2021). Exploring tissue architecture\\nusing spatial transcriptomics. Nature596, 211– 220.\\n27. Giesen, C., Wang, H.A.O., Schapiro, D., Zivanovic, N., Jacobs, A., Hattendorf, B.,\\nSchüfﬂer, P.J., Grolimund, D., Buhmann, J.M., Brandt, S., et al. (2014). Highly multi-\\nplexed imaging of tumor tissues with subcellular resolution by mass cytometry. Nat.\\nMethods11, 417– 422.\\nMolecular Therapy\\n1264 Molecular Therapy Vol. 32 No 5 May 2024', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='addc1add-fc6b-431d-a0aa-13e9aa9a79f3', embedding=None, metadata={'page_label': '1265', 'file_name': 'jin_lymphoma.pdf', 'file_path': '/content/data/jin_lymphoma.pdf', 'file_type': 'application/pdf', 'file_size': 5963122, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='28. Jackson, H.W., Fischer, J.R., Zanotelli, V.R.T., Ali, H.R., Mechera, R., Soysal, S.D.,\\nMoch, H., Muenst, S., Varga, Z., Weber, W.P., and Bodenmiller, B. (2020). The sin-\\ngle-cell pathology landscape of breast cancer. Nature578, 615– 620.\\n29. Sanmamed, M.F., Nie, X., Desai, S.S., Villaroel-Espindola, F., Badri, T., Zhao, D., Kim,\\nA.W., Ji, L., Zhang, T., Quinlan, E., et al. (2021). A Burned-Out CD8+ T-cell Subset\\nExpands in the Tumor Microenvironment and Curbs Cancer Immunotherapy.\\nCancer Discov.11, 1700– 1715.\\n30. Damond, N., Engler, S., Zanotelli, V.R.T., Schapiro, D., Wasserfall, C.H.,\\nKusmartseva, I., Nick, H.S., Thorel, F., Herrera, P.L., Atkinson, M.A., and\\nBodenmiller, B. (2019). A Map of Human Type 1 Diabetes Progression by Imaging\\nMass Cytometry. Cell Metab.29, 755– 768.e5.\\n31. Mitsialis, V., Wall, S., Liu, P., Ordovas-Montanes, J., Parmet, T., Vukovic, M.,\\nSpencer, D., Field, M., McCourt, C., Toothaker, J., et al. (2020). Single-Cell\\nAnalyses of Colon and Blood Reveal Distinct Immune Cell Signatures of Ulcerative\\nColitis and Crohn’s Disease. Gastroenterology159, 591– 608.e10.\\n32. Schürch, C.M., Bhate, S.S., Barlow, G.L., Phillips, D.J., Noti, L., Zlobec, I., Chu, P.,\\nBlack, S., Demeter, J., McIlwain, D.R., et al. (2020). Coordinated Cellular\\nNeighborhoods Orchestrate Antitumoral Immunity at the Colorectal Cancer\\nInvasive Front. Cell183, 838.\\n33. Chen, Y., Jia, K., Sun, Y., Zhang, C., Li, Y., Zhang, L., Chen, Z., Zhang, J., Hu, Y., Yuan,\\nJ., et al. (2022). Predicting response to immunotherapy in gastric cancer via multi-\\ndimensional analyses of the tumour immune microenvironment. Nat. Commun.\\n13, 4851.\\n34. Rodriguez-Garcia, A., Lynn, R.C., Poussin, M., Eiva, M.A., Shaw, L.C., O’Connor,\\nR.S., Minutolo, N.G., Casado-Medrano, V., Lopez, G., Matsuyama, T., and Powell,\\nD.J., Jr. (2021). CAR-T cell-mediated depletion of immunosuppressive tumor-associ-\\nated macrophages promotes endogenous antitumor immunity and augments adop-\\ntive immunotherapy. Nat. Commun.12, 877.\\n35. Gunderson, A.J., Yamazaki, T., McCarty, K., Fox, N., Phillips, M., Alice, A., Blair, T.,\\nWhiteford, M., O’Brien, D., Ahmad, R., et al. (2020). TGFb suppresses CD8+ T cell\\nexpression of CXCR3 and tumor trafﬁcking. Nat. Commun.11, 1749.\\n36. Enblad, G., Karlsson, H., Gammelgård, G., Wenthe, J., Lövgren, T., Amini, R.M.,\\nWikstrom, K.I., Essand, M., Savoldo, B., Hallböök, H., et al. (2018). A Phase I/IIa\\nTrial Using CD19-Targeted Third-Generation CAR T Cells for Lymphoma and\\nLeukemia. Clin. Cancer Res.24, 6185– 6194.\\n37. Sakemura, R., Hefazi, M., Siegler, E.L., Cox, M.J., Larson, D.P., Hansen, M.J.,\\nManriquez Roman, C., Schick, K.J., Can, I., Tapper, E.E., et al. (2022). Targeting can-\\ncer-associatedﬁbroblasts in the bone marrow prevents resistance to CART-cell ther-\\napy in multiple myeloma. Blood139, 3708– 3721.\\n38. Collins, P.L., Cella, M., Porter, S.I., Li, S., Gurewitz, G.L., Hong, H.S., Johnson, R.P.,\\nOltz, E.M., and Colonna, M. (2019). Gene Regulatory Programs Conferring\\nPhenotypic Identities to Human NK Cells. Cell176, 348– 360.e12.\\n39. Dubois, S., Feigenbaum, L., Waldmann, T.A., and Müller, J.R. (2020). NK cells pre-\\nvent T cell lymphoma development in T cell receptor-transgenic mice. Cell.\\nImmunol.352, 104081.\\n40. Zwirner, N.W., Domaica, C.I., and Fuertes, M.B. (2021). Regulatory functions of NK\\ncells during infections and cancer. J. Leukoc. Biol.109, 185– 194.\\n41. Bielekova, B., Catalfamo, M., Reichert-Scrivner, S., Packer, A., Cerna, M., Waldmann,\\nT.A., McFarland, H., Henkart, P.A., and Martin, R. (2006). Regulatory CD56(bright)\\nnatural killer cells mediate immunomodulatory effects of IL-2Ralpha-targeted ther-\\napy (daclizumab) in multiple sclerosis. Proc. Natl. Acad. Sci. USA103, 5941– 5946.\\n42. Zeidan, A.M., Komrokji, R.S., and Brunner, A.M. (2021). TIM-3 pathway dysregula-\\ntion and targeting in cancer. Expert Rev. Anticancer Ther.21, 523– 534.\\n43. Takata, A., Nakashima, M., Saitsu, H., Mizuguchi, T., Mitsuhashi, S., Takahashi, Y.,\\nOkamoto, N., Osaka, H., Nakamura, K., Tohyama, J., et al. (2019). Comprehensive\\nanalysis of coding variants highlights genetic complexity in developmental and\\nepileptic encephalopathy. Nat. Commun.10, 2506.\\n44. Anderson, A.C., Joller, N., and Kuchroo, V.K. (2016). Lag-3, Tim-3, and TIGIT: Co-\\ninhibitory Receptors with Specialized Functions in Immune Regulation. Immunity\\n44, 989– 1004.\\n45. Longo, S.K., Guo, M.G., Ji, A.L., and Khavari, P.A. (2021). Integrating single-cell and\\nspatial transcriptomics to elucidate intercellular tissue dynamics. Nat. Rev. Genet.22,\\n627– 644.\\n46. Cheson, B.D., Fisher, R.I., Barrington, S.F., Cavalli, F., Schwartz, L.H., Zucca, E.,\\nLister, T.A., et al.; Alliance, Australasian Leukaemia and Lymphoma Group,\\nEastern Cooperative Oncology Group, European Mantle Cell Lymphoma\\nConsortium (2014). Recommendations for initial evaluation, staging, and response\\nassessment of Hodgkin and non-Hodgkin lymphoma: the Lugano classiﬁcation.\\nJ. Clin. Oncol.32, 3059– 3068.\\n47. Carpenter, A.E., Jones, T.R., Lamprecht, M.R., Clarke, C., Kang, I.H., Friman, O.,\\nGuertin, D.A., Chang, J.H., Lindquist, R.A., Moffat, J., et al. (2006). CellProﬁler: image\\nanalysis software for identifying and quantifying cell phenotypes. Genome Biol.\\n7, R100.\\n48. Berg, S., Kutra, D., Kroeger, T., Straehle, C.N., Kausler, B.X., Haubold, C., Schiegg, M.,\\nAles, J., Beier, T., Rudy, M., et al. (2019). ilastik: Interactive machine learning for (bio)\\nimage analysis. Nat. Methods16, 1226– 1232.\\n49. Damond, N., Engler, S., Zanotelli, V.R.T., Schapiro, D., Wasserfall, C.H.,\\nKusmartseva, I., Nick, H.S., Thorel, F., Herrera, P.L., Atkinson, M.A., and\\nBodenmiller, B. (2019). A map of human type 1 diabetes progression by imaging\\nmass cytometry. Cell Metab.29, 755– 768.e5.\\n50. Schapiro, D., Jackson, H.W., Raghuraman, S., Fischer, J.R., Zanotelli, V.R.T., Schulz,\\nD., Giesen, C., Catena, R., Varga, Z., and Bodenmiller, B. (2017). histoCAT: analysis\\nof cell phenotypes and interactions in multiplex image cytometry data. Nat. Methods\\n14, 873– 876.\\n51. Korsunsky, I., Millard, N., Fan, J., Slowikowski, K., Zhang, F., Wei, K., Baglaenko, Y.,\\nBrenner, M., Loh, P.R., and Raychaudhuri, S. (2019). Fast, sensitive and accurate inte-\\ngration of single-cell data with Harmony. Nat. Methods16, 1289– 1296.\\nwww.moleculartherapy.org\\nMolecular Therapy Vol. 32 No 5 May 2024 1265', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='66144c9d-1d56-4f8a-b1ce-0bf2bbc7bde2', embedding=None, metadata={'page_label': '555', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 614 | 16 February 2023 | 555\\nArticle\\nSingle-cell spatial immune landscapes of \\nprimary and metastatic brain tumours\\nElham Karimi1,20, Miranda W. Yu1,2,20, Sarah M. Maritan1,3,20, Lucas J. M. Perus1,2,20, \\nMorteza Rezanejad4, Mark Sorin1,5, Matthew Dankner1,3, Parvaneh Fallah6, Samuel Doré1,5, \\nDongmei Zuo1, Benoit Fiset1, Daan J. Kloosterman7, LeeAnn Ramsay1, Yuhong Wei1, \\nStephanie Lam8, Roa Alsajjan9,10, Ian R. Watson1,11, Gloria Roldan Urgoiti12,13, Morag Park1,6,11, \\nDieta Brandsma14, Donna L. Senger6,15, Jennifer A. Chan13,16, Leila Akkari7, Kevin Petrecca10,17, \\nMarie-Christine Guiot1,10,17,18, Peter M. Siegel1,3,11,19\\u2009✉, Daniela F. Quail1,2,3\\u2009✉ & Logan A. Walsh1,5\\u2009✉\\nSingle-cell technologies have enabled the characterization of the tumour \\nmicroenvironment at unprecedented depth and have revealed vast cellular diversity \\namong tumour cells and their niche. Anti-tumour immunity relies on cell–cell \\nrelationships within the tumour microenvironment1,2, yet many single-cell studies \\nlack spatial context and rely\\xa0on dissociated tissues3. Here we applied imaging mass \\ncytometry to characterize the immunological landscape of 139 high-grade glioma  \\nand 46 brain metastasis tumours from patients. Single-cell analysis of more than 1.1 \\nmillion cells across 389 high-dimensional histopathology images enabled the spatial \\nresolution of immune lineages and activation states, revealing differences in immune \\nlandscapes between primary tumours and brain metastases from diverse solid \\ncancers. These analyses revealed cellular neighbourhoods associated with survival  \\nin patients with glioblastoma, which we\\xa0leveraged to identify a unique population of \\nmyeloperoxidase (MPO)-positive macrophages associated with long-term survival. \\nOur findings provide insight into the biology of primary and metastatic brain tumours, \\nreinforcing the value of integrating spatial resolution to single-cell datasets to dissect \\nthe microenvironmental contexture of cancer.\\nBrain tumours comprise a diverse repertoire of malignancies that arise \\neither from within the brain or from cancer cells that have spread from \\nother primary sites. The most common types of cancer representing \\nthese two classes include glioblastoma (around 50% of all primary brain \\nmalignancies in adults4) and brain metastasis (BrM) (about 90% of all \\nbrain malignancies), with BrM most frequently arising from melanoma, \\nlung or breast tumours5. Besides surgery, cytotoxic therapies that target \\ntumour cells—such as stereotactic radiotherapy—are often the first \\nline of treatment, but they yield minimal benefit, with survival\\xa0beyond \\n2 years being rare6,7. The tumour microenvironment (TME) is a major \\nregulator of cancer progression, whose therapeutic value has grown \\nwith the advent of immune checkpoint blockade1,2. Compared with \\nother tissues, the brain TME has a distinct composition, dominated by \\nfunctionally diverse astrocytes and pro-tumorigenic macrophages that \\nare ontogenically distinct, with the exclusion of infiltrating lympho-\\ncytes8. Many promising therapeutic targets within the TME of other \\ncancers have been revealed by single-cell profiling technologies; for \\nexample, multiplex imaging has enabled the discovery of several new \\nbiomarkers that are predictive of outcomes and therapeutic efficacy in \\nbreast9–12, colorectal13 and pancreatic cancer14. However, comprehen-\\nsive profiling of the brain TME has seen fewer (albeit important15–19) \\nadvances compared with other malignancies, and so far none have \\nincluded the spatial characterization of individual cells within their \\nniche using highly multiplexed histology. Here we use imaging mass \\ncytometry (IMC) on patient samples\\xa0to characterize the brain TME of \\nglioblastoma and BrM, and explore how spatially resolved features \\nrelate to clinical outcomes.\\nMapping the brain TME with IMC\\nT o comprehensively profile the cellular composition and spatial organi-\\nzation of the brain TME, we optimized a highly multiplexed antibody \\nhttps://doi.org/10.1038/s41586-022-05680-3\\nReceived: 24 March 2022\\nAccepted: 22 December 2022\\nPublished online: 1 February 2023\\nOpen access\\n Check for updates\\n1Rosalind and Morris Goodman Cancer Institute, McGill University, Montreal, Quebec, Canada. 2Department of Physiology, Faculty of Medicine, McGill University, Montreal, Quebec, Canada. \\n3Department of Medicine, Division of Experimental Medicine, McGill University, Montreal, Quebec, Canada. 4Departments of Psychology and Computer Science, University of Toronto, Toronto, \\nOntario, Canada. 5Department of Human Genetics, Faculty of Medicine, McGill University, Montreal, Quebec, Canada. 6Gerald Bronfman Department of Oncology, McGill University, Montreal, \\nQuebec, Canada. 7Division of Tumor Biology and Immunology, Oncode Institute, Netherlands Cancer Institute, Amsterdam, Netherlands. 8Department of Diagnostic Radiology, Faculty of Medicine, \\nMcGill University, Montreal, Quebec, Canada. 9Department of Medicine, Division of Neurology, King Saud University College of Medicine, Riyadh, Saudi Arabia. 10Department of Neurology and \\nNeurosurgery, Faculty of Medicine, McGill University, Montreal, Quebec, Canada. 11Department of Biochemistry, Faculty of Medicine, McGill University, Montreal, Quebec, Canada. 12Department of \\nOncology, University of Calgary, Calgary, Alberta, Canada. 13Arnie Charbonneau Cancer Institute, University of Calgary, Calgary, Alberta, Canada. 14Department of Neuro-Oncology, Netherlands \\nCancer Institute, Antoni van Leeuwenhoek Hospital, Amsterdam, Netherlands. 15Lady Davis Institute for Medical Research, Jewish General Hospital, Montreal, Quebec, Canada. 16Department of \\nPathology and Laboratory Medicine, University of Calgary, Calgary, Alberta, Canada. 17Montreal Neurological Institute-Hospital, McGill University Health Centre, Montreal, Quebec, Canada. \\n18Department of Pathology, Faculty of Medicine, McGill University, Montreal, Quebec, Canada. 19Department of Anatomy and Cell Biology, Faculty of Medicine, McGill University, Montreal, Quebec, \\nCanada. 20These authors contributed equally: Elham Karimi, Miranda W. Yu, Sarah M. Maritan, Lucas J. M. Perus. ✉e-mail: peter.siegel@mcgill.ca; daniela.quail@mcgill.ca; logan.walsh@mcgill.ca', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='73789feb-dc94-44f5-b5de-907f8bbb2cf6', embedding=None, metadata={'page_label': '556', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='556 | Nature | Vol 614 | 16 February 2023\\nArticle\\npanel and IMC pipeline (Extended Data Figs.\\xa01–3, Supplementary \\nFigs.\\xa01–3 and Supplementary Table\\xa01). Antibodies were validated in \\nnormal and malignant tissues on the basis of their expected staining \\npattern (Extended Data Fig.\\xa03a and Supplementary Figs.\\xa01 and 2). We \\nacquired 389 high-dimensional histopathology images representing \\n139 high-grade glioma and 46 BrM patient tumours (Fig.\\xa0 1a,b and \\nExtended Data Fig.\\xa01a). Gliomas comprised resected tissues obtained \\nduring surgery (270 images in total, 192 from primary glioblastoma), \\nincluding a subset from long-term survivors (78 images). BrM images \\nwere derived from multiple primary malignancies, including lung  \\na\\nMetal-conjugated\\nantibodies\\nStain UV laser\\nCyTOF\\nn = 36\\nAcquire Process\\nSegment\\nLineage ID\\nAnalysis\\nb d MyeloidL ymphoid Other\\nPercent\\n0\\n0 100 0 100\\n0\\n05 0\\nAstrocytes\\nEndothelial\\nUnde/f_ined CD45 –\\nMG\\nMonocyte\\nNeutrophil\\nDCs\\nMast cells\\nMDM\\nOther T cells\\nB cells\\nNK cells\\nTc cells\\nTH cells Unde/f_ined CD45 +\\nTreg cells\\nGliomaBrM-coreBrM-margin\\nc\\n–5\\n0\\n5\\ne\\nGlioblastoma BrM-core BrM-margin\\nRaw image\\nCell lineage ID\\n(nuclei)\\nTH cells Cl Mo Unde/f_inedNeutrophils M1-like MGB cellsAstrocytes\\nNK cells Treg cells Non-Cl MoM2-like MGM1-like MDMEndothelial\\nTc cells Other T cells DCs Int MoM2-like MDMMast cellsCancer cells\\nKi67\\nGFAP\\nCD31\\nCD68\\nMPO\\nCD3\\nCancer100 /uni03BCm\\n100 /uni03BCm\\n100 /uni03BCm\\n100 /uni03BCm\\n100 /uni03BCm\\n100 /uni03BCm\\nUnknown IDH status\\nProgression from grade II/III\\nTissue type\\nRecurrent\\nGlioblastoma\\nResidual disease\\nIDH mut grade IV astrocytoma\\nAdjacent normal\\nBrM-core\\nBrM-margin\\nTissue\\n0%\\n100%\\n50%\\nGlioma Brain metastasis\\nTissue\\n0%\\n100%\\n50%\\nGlioma\\nPanCK\\nOLIG2\\nSOX2\\nSOX9\\nPMEL\\nMelanA\\nCD31\\nCD20\\nMPO\\nCD117\\nGFAP\\nCD94\\nHLA-DR\\nCD11c\\nCD8a\\nCD4\\nFOXP3\\nCD3\\nCD163\\nP2Y12\\nCD14\\nCD16CD68\\nKi67\\npSTAT3\\nClaudin-5\\nCD39\\nCD40\\nHIF1/uni03B1\\nMMP9\\npERK\\nCIRBP\\nCC3\\nOX40L\\nCD206\\nCD45\\nDNA\\nCancer cells\\nEndothelial cells\\nB cells\\nNeutrophils\\nMast cells\\nAstrocytes\\nNK cells\\nDendritic cells\\nTc cells\\nTH cells\\nTreg cells\\nOther T cells\\nM1-like MDMs\\nM2-like MDMs\\nM1-like MG\\nM2-like MG\\nCl Mo\\nNon-Cl Mo\\nInt Mo\\nUnde/f_ined\\nMarker\\nIntensity\\n(z-score) \\nAdj norm\\nIDH WT\\nIDH mut\\nRecur\\nLung\\nBreast\\nMel\\nOther\\nLung\\nBreast\\nMel\\nOther\\nFig. 1 | IMC reveals cell dynamics within the brain TME. a, Schematic of the IMC \\npipeline applied to glioma and BrM tissue microarrays. Samples were subject to \\nmultiplex staining and data were acquired using cytometry by time-of-flight \\n(CyTOF). Cell segmentation and lineage assignment was performed prior to \\nspatial analysis. Created with BioRender.com. b, IMC images from glioblastoma, \\nBrM-core and BrM-margin samples (top) and corresponding lineage assignment \\n(bottom), with magnified regions to the right of each image. The colour codes  \\nfor IMC markers (top right) and lineage assignment (bottom) are provided \\n(representative of n\\u2009=\\u2009389 images). Scale bars, 100\\u2009μm. c, Heat map showing \\nrelative average expression of all markers across cell populations identified using \\nIMC (n\\u2009=\\u2009389 images). A subset of markers was specific to the glioma IMC antibody \\npanel (SOX2, SOX9, OLIG2, CD40, CD206; n\\u2009=\\u2009270 images) and a second subset  \\nto the BrM IMC antibody panel (pan-cytokeratin, PMEL, MelanA, pERK, CIRBP; \\nn\\u2009=\\u2009119 images). d, Stacked bar graph of the indicated\\xa0cell types as a percentage of \\nall\\xa0cells\\xa0within the TME according to clinical subgroups. Glioma: adjacent normal \\n(adj norm), n\\u2009=\\u200918; primary isocitrate dehydrogenase (IDH) wild type (WT), n\\u2009=\\u2009192; \\nprimary IDH mutant (mut), n\\u2009=\\u200919; recurrent (recur), n\\u2009=\\u200922. BrM-core: lung, n\\u2009=\\u200929; \\nbreast, n\\u2009=\\u200917; melanoma (mel), n\\u2009=\\u200913; other, n\\u2009=\\u200913. BrM-margin: lung, n\\u2009=\\u200922; \\nbreast, n\\u2009=\\u200912; melanoma, n\\u2009=\\u20096; other, n\\u2009=\\u20097. Data are mean values; n refers to \\nnumber of images. e, The distribution of cell populations as a percentage of all \\ncells in the TME, sorted by tissue type. Cell frequencies for each image (n\\u2009=\\u2009389 \\nimages) are displayed as vertical bars (colours correspond to cell lineages in b) \\nand the associated tissue type is indicated in the horizontal panels below (colours \\nindicated in the legend, right). Cl Mo, classical monocyte; DC, dendritic cells; Int \\nMo, intermediate monocyte; MG, microglia; Non-Cl Mo, non-classical monocyte; \\nNK cells, natural killer cells; panCK, pan-cytokeratin;\\xa0Tc, cytotoxic T\\xa0cell; TH, T \\nhelper; Treg, T regulatory cell; other T\\xa0cells, CD8−CD4−\\xa0double-negative\\xa0T cells.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='fb9d5e75-ed17-4a71-90c7-5221c7ff1b60', embedding=None, metadata={'page_label': '557', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 614 | 16 February 2023 | 557\\n(51 images), breast (29 images), melanoma (19 images) and other \\nprimary sources (20 images), with patient-matched samples from \\nthe centre of the metastatic lesion (BrM-core) and tissue interface \\n(BrM-margin). Images were segmented into 1,163,362 total cells and a \\nsupervised lineage assignment approach was used to classify tumour \\ncells, astrocytes, blood vessels and more than 16 immune cell popu-\\nlations using canonical identity markers (Fig.\\xa01c and Extended Data \\nFigs.\\xa01b and 2). As expected, within the stromal compartment across all \\ntissues, major cell populations included GFAP+ astrocytes and CD68+  \\nmacrophages, whereas lymphocytes were relatively infrequent \\n(Fig.\\xa01d,e).\\nMacrophages stimulated in\\xa0vitro can be defined along a continuum \\nof activation states; although usually considered M2-like, tumour-  \\nassociated macrophage activation is much more complex in\\xa0situ and \\ndoes not necessarily conform to the M1/M2 paradigm19,20. With this \\nlimitation in mind, we subdivided macrophages by ontogeny and activa-\\ntion state; expression of the purinergic receptor P2Y12 distinguished \\ntissue-resident microglia from monocyte-derived macrophages 19 \\n(MDMs), and CD163 expression distinguished putative ‘M2-like’ \\nfrom ‘M1-like’ cells21,22 (Fig.\\xa01c and Supplementary Figs.\\xa03, 4 and 5a).  \\nAdditional pro-tumorigenic markers, including CD206 and CD39, were \\nalso enriched in the CD163+ M2-like macrophage fraction (Supplemen-\\ntary Fig.\\xa05b,c). Consistent with previous reports15,16, MDMs and microglia \\nwere the dominant immune populations across all samples, comprising  \\napproximately 30.5% and 9.2% of the TME, respectively (Fig.\\xa01d).\\nSingle-cell interaction networks\\nWe quantified\\xa0the frequency of each\\xa0cell type\\xa0as a\\xa0percentage of the\\xa0total \\nnumber of\\xa0cells within each image,\\xa0and compared\\xa0clinically relevant \\nsubgroups of patients (Fig.\\xa02a,b and Supplementary Figs.\\xa06 and 7). \\nCell density within each image area was similarly assessed (Extended \\nData Fig.\\xa04a and Supplementary Fig.\\xa08). As expected, there was an \\nincrease in the frequency of most cell types in adjacent normal tissue \\ncompared with glioblastoma (Fig.\\xa02a), whereas this trend was reversed \\nwhen examining cell density (Extended Data Fig.\\xa04a), reflecting the \\nsparse cellular landscape of the normal brain niche. In IDH wild-type \\n(glioblastoma) versus IDH mutant (grade IV astrocytoma) tumours23, \\nNK cells were reduced in frequency (Fig.\\xa02a). Moreover, there was a \\nlower proportion of CD16+ (cytotoxic) than CD16− (immature) NK cell \\nsubsets in glioblastoma tumours (Supplementary Fig.\\xa07b,c), consistent  \\nwith previous findings15. We also found a higher frequency and den-\\nsity of recruited MDMs (but not microglia) in IDH wild-type versus \\nmutant tumours (Fig.\\xa0 2a and Extended Data Fig.\\xa04a), suggestive of \\nenhanced peripheral recruitment of macrophages with greater disease \\nseverity24. Parsing glioblastoma samples by MGMT methylation status  \\n(a prognostic indicator) unveiled minimal immunological differences; \\nhowever, subdivision by survival time revealed a higher endothelial \\nfrequency in tumours from long-term survivors (L TSs) (overall survival \\nmore than three years) compared with those from short-term survivors \\n(STSs) (overall survival less than one year) (Fig.\\xa02a). This was unex-\\npected, and may be related to efficiencies in chemotherapy delivery.  \\nAlternatively, features of the vascular niche may confer a survival \\nbenefit in some patients. Compared with STS tumours, L TS tumours \\nalso had a higher frequency of CD8−CD4− T\\xa0cells (potentially including \\nγδ T\\xa0cells, which are associated with increased survival25) and M1-like \\nmacrophage accumulation, with no difference in M2-like macrophages \\n(Fig.\\xa02a). Across glioblastoma clinical subgroups, very few lympho-\\ncytes were observed\\xa0based on\\xa0frequency (Fig.\\xa02b) as well as density \\n(Supplementary Fig.\\xa08), supporting observations that glioblastomas \\nare T\\xa0cell deserts that exhibit poor responses to immune checkpoint \\nblockade26. However, a subset of glioblastoma images exhibited unusu-\\nally high T\\xa0cell frequencies (more than 5% of cells in the TME). These \\nsamples were enriched for CD8+, CD4+ and CD8−CD4− T\\xa0cell subsets, \\nbut not immunosuppressive Treg cells, and exhibited a 62% increase in \\nmean survival time compared with samples with low numbers of\\xa0T\\xa0cells \\n(less than 5%\\xa0of cells in the TME) (Extended Data Fig.\\xa04b,c). Finally, we \\nobserved\\xa0higher frequencies of peripherally derived monocytes in \\ntumours from male patients than in those from female patients, coincid-\\ning with higher frequencies of endothelial cells (Fig.\\xa02a), highlighting \\nputative sexual dimorphism in immune responses to cancer.\\nWe next examined immune cell dynamics across BrM samples. We \\nfound an increase\\xa0in the frequency and density of NK cells (notably, those \\nthat were CD16+15), neutrophils, macrophages, classical monocytes and \\nT\\xa0cells (including Treg cells15) compared with glioblastoma, and a decrease \\nin dendritic cells and non-classical monocytes (Fig.\\xa02a, Extended Data \\nFig.\\xa04a and Supplementary Fig.\\xa07b,c). When examining immunologi-\\ncal changes associated with BrM progression in BrM-cores, we found \\nincreases in monocytes and microglia in patients without leptome-\\nningeal disease (Fig.\\xa02a) or local recurrence (after more than 724 days)  \\n(Supplementary Fig.\\xa09a), suggesting a putative protective role for \\nthese cells in this context. In comparing BrMs arising from distinct \\nprimary tumour sites, the degree of BrM-associated inflammation \\n(that is, the frequency of immune cell types analysed) was\\xa0generally \\nlowest in tumours originating from breast tumours, highest in those \\nfrom melanoma, with an intermediate level in tumours originating \\nfrom lung. Melanoma BrM were enriched for monocytes and micro-\\nglia compared with lung and breast BrM, and exhibited a pronounced \\naccumulation of CD8+ T\\xa0cells in the tumour margin\\xa0as a percentage of \\ntotal cells (Fig.\\xa02a and Supplementary Fig.\\xa09b). This is consistent with \\nobservations that, unlike glioblastoma, BrM displays some vulnerability\\xa0 \\nto immune checkpoint blockade—particularly melanoma BrM 27,28. \\nSimilar to glioblastoma, we observed sexual dimorphism in BrM-cores, \\nwith more CD8+ T\\xa0cells in those from male patients compared with \\nthose from female patients.\\xa0We additionally found more neutrophils, \\ndendritic cells and CD4+ T\\xa0cells in BrM-cores from younger patients \\ncompared with those from older patients (Fig.\\xa02a).\\nT o characterize the patterns of communication between individual \\ncells, we interrogated the positional architecture of brain tumours using \\npermutation tests to quantify cell–cell co-localization and identify \\ninteraction or avoidance behaviours between cell pairs11 (Fig.\\xa02c). This \\napproach revealed that cancer cells in BrM were more likely to avoid \\nmost non-cancer lineages within the TME compared with glioblastoma \\n(Fig.\\xa02c and Extended Data Fig.\\xa05a), indicating that the topography of \\ncancer cells in BrM is more compact relative to the dispersed nature \\nof glioblastoma, consistent with the observed pattern of homotypic  \\ncellular interactions (Fig.\\xa0 2c). Global heterotypic cellular interac -\\ntions were increased in BrM compared with glioblastoma (Fig.\\xa02c and \\nExtended Data Fig.\\xa05b), suggesting that the ways in which glioblastoma \\nand BrM interface with the surrounding brain parenchyma are funda-\\nmentally different, despite sharing a common tissue niche.\\nThe vasculature\\xa0is a key component of the brain TME in both glioblas-\\ntoma and BrM; for example, the perivascular niche maintains the glioma- \\ninitiating cell pool29, and the blood-brain barrier (BBB) regulates BrM \\ndissemination. Given our observation of endothelial enrichment in L TS \\ntumours\\xa0(Fig.\\xa02a), we examined cellular dynamics within the vascular \\nniche of glioblastoma. As expected, endothelial cells\\xa0 \\xa0exhibited a strong \\nlikelihood of interacting with astrocytes in glioblastoma, essential for \\nBBB function (Fig.\\xa02c). Endothelial cells also showed strong interac-\\ntions with cancer cells (Fig.\\xa02c); in particular, perivascular cancer cells \\nexhibited reduced Ki67:CC3 ratios compared with those that avoided \\nblood vessels (Fig.\\xa02d and Extended Data Fig.\\xa06a), suggesting that direct \\ncontact with the vasculature may impede cancer cell expansion30— \\na finding that may provide insight into L TS tumour biology. Consistent \\nwith previous reports, we observed that MDMs, but not monocytes or \\nmicroglia, also displayed a strong tendency to interact with endothelial \\ncells in glioblastoma15,16 (Fig.\\xa02c), despite a weak correlation between the \\nfrequencies of MDMs and endothelial cells (Extended Data Fig.\\xa06b). This \\nhighlights that these interactions are spatially coordinated rather than \\nsimply resulting from associations in abundance. Mirroring tumour ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8d24d32b-f7c1-4d63-bf4e-772a9a97511c', embedding=None, metadata={'page_label': '558', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='558 | Nature | Vol 614 | 16 February 2023\\nArticle\\ncells, MDMs displayed lower Ki67:CC3 ratios when engaged with the \\nendothelium compared with those that avoided direct\\xa0endothelial \\ninteractions (Fig.\\xa02e and Extended Data Fig.\\xa06a), recapitulating the rela-\\ntionship between proliferating brain macrophages and glioma progres-\\nsion31. Finally, despite their low prevalence, there was a modest tendency \\nfor T\\xa0cells to interact with endothelial cells or MDMs (Fig.\\xa02c), prompting \\nus to further dissect these relationships. Specifically, we found more \\nCD40+ MDMs interacting with TH cells\\xa0than CD40− MDMs (Extended \\nData Fig.\\xa06c), CD40 being a co-stimulatory protein implicated in T\\xa0cell \\nrecruitment in glioma32. When examining vessel proximity,\\xa0perivas-\\ncular M1-like MDMs exhibited higher CD40 expression than those  \\nfurther away from blood vessels (Extended Data Fig.\\xa06d) and similarly, \\nperivascular M2-like MDMs expressed high levels of OX40L (another \\nco-stimulatory molecule) (Extended Data Fig.\\xa06e). T ogether, these data \\nallude to the existence of vascular microniches—where macrophages \\nmay provide beneficial signalling cues to T\\xa0cells and cancer cell expan-\\nsion is kept in check—and support a role for blood vessels in shaping \\nthe brain TME contexture.\\n Adj norm (n = 18)\\n  IDH WT (n = 192)\\n  Recur (n = 22)\\n  IDH mut (n = 19)\\nBrM-cores (n = 72)\\nBrM-margins (n = 47)\\nSurvival <1 yr (n = 63)\\nSurvival >3 yr (n = 55)\\nMale (n = 102)\\nFemale (n = 81)\\n>Mean age (n = 94)\\n<Mean age (n = 89)\\nMGMT meth (n = 104)\\nMGMT unmeth (n = 84)\\nPartial resect (n = 73)\\nTotal resect (n = 106)\\nEarly recur (n = 9)\\nLate recur (n = 5)\\nNo recur (n = 19)\\nMale (n = 22)\\nFemale (n = 50)\\n>Mean age (n = 27)\\n<Mean age (n = 33)\\nBreast (n = 17)\\nLung (n = 29)\\nMelanoma (n = 13)\\nOther (n = 13)\\nWith LMD (n = 43)\\nWithout LMD (n = 22)\\nEarly recur (n = 6)\\nLate recur (n = 4)\\nNo recur (n = 10)\\nMale (n = 14)\\nFemale (n = 33)\\n>Mean age (n = 21)\\n<Mean age (n = 19)\\nBreast (n = 12)\\nLung (n = 22)\\nMelanoma (n = 6)\\nOther (n = 7)\\nWith LMD (n = 30)\\nWithout LMD (n = 12)\\n0\\n8\\n16\\n40\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nEndothelial\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells\\nOther T cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nNon-Cl Mo\\nInt Mo\\nUnde/f_ined\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells\\nEndothelial\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nOther T cells\\nNon-Cl Mo\\nInt Mo\\nUnde/f_ined\\n+1 (interact)–1 (avoid)\\n0 Glioblastoma\\nBrM-margins\\nGrid layout\\nBrM-cores\\n T cells (% of total)\\nAll patients Glioblastoma BrM-cores BrM-margins\\n0\\n40\\n80\\n300\\n600\\nKi67:CC3\\n(cancer cells)\\nGlioblastoma Glioblastoma BrM-cores\\nCancer–endothelial\\ninteractions\\nMDM–endothelial\\ninteractions\\nEndothelial–cancer\\ninteractions\\n5\\n0\\n10\\n15\\n50\\n150\\nBrM-margins\\nEndothelial–cancer\\ninteractions\\nEndothelial–Tc\\ninteractions\\nP = 0.0498\\n0\\n3\\n6\\n20\\n50\\nKi67:CC3\\n(MDM)\\n5\\n0\\n10\\n15\\n50\\n100\\nKi67+CD31+ cells\\n(% of total CD31+ cells)\\nKi67+CD31+ cells\\n(% of total CD31+ cells)\\nBrM-cores\\nP = 0.0080\\n0\\n10\\n20\\n100\\n200\\nKi67+CD31+ cells\\n(% of total CD31+ cells)\\nInteract\\nAvoid\\nUnpairedPaired\\nTotal mono\\nTotal T cells\\nM1-like MDM\\nUnde/f_ined\\nTotal MG\\nM2-like MG\\nTotal macs\\nCl Mo\\nTotal MDM\\nM1-like MG\\nEndothelial\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells\\nOther T cells\\nM2-like MDM\\nNon-Cl Mo\\nInt Mo\\nIDH WT\\nIDH WT\\nIDH WT\\nIDH WT\\nIDH WT\\nMGMT meth\\nSurvival <1 yr\\nMale\\n<Mean age\\nBrM-cores\\nBreast\\nBreast\\nBrM-cores\\nLung\\nNo recur\\nEarly recur\\nNo recur\\nWithout LMD\\n<Mean age\\nBreast\\nMale\\nBreast\\nNo recur\\nNo recur\\nLung\\nEarly recur\\nMale\\n<Mean age\\nWithout LMD\\nAdj norm\\nIDH mut\\nRecur\\nMGMT unmeth \\nSurvival >3 yr \\nFemale\\n>Mean age\\nBrM-cores\\nBrM-margins\\nBrM-margins\\nBrM-margins\\nLung\\nMelanoma\\nMelanoma\\nEarly recur\\nLate recur\\nLate recur\\nWith LMD\\nFemale\\n>Mean age\\nLung\\nMelanoma\\nMelanoma\\nEarly recur\\nLate recur\\nLate recur\\nWith LMD\\nFemale\\n>Mean age\\nA term vs        B term\\nP ≤ 0.001\\nP = 0.01\\nP = 0.05\\nGliomaBrM-margins BrM-cores\\nP = 0.0083 P = 0.0377 P = 0.0067\\na c\\nb de f\\ngh\\nFig. 2 | Single-cell interaction networks in high-dimensional histopathology \\nimages represent clinical subgroups of patients with brain tumours.   \\na, Cell frequency comparisons between clinical subgroups of patients, \\ncorresponding to data in b and Supplementary Fig.\\xa06. Within each row, the \\nbubble colour indicates the clinical subgroup with the higher cell type \\nrepresentation (A term\\xa0versus B term, right),\\xa0and the bubble size indicates  \\nthe P-value. Two-sided Student’s t-test, unpaired unless indicated otherwise; \\npaired analyses are from patient-matched samples. LMD, leptomeningeal \\ndisease;\\xa0Meth, methylated; unmeth, unmethylated; recur, recurrence. b, T\\xa0cell \\nfrequencies as a percentage of total cells across clinical subgroups. Data are \\nmean\\u2009±\\u2009s.e.m.; all data points overlaid; n refers to the number of images.  \\nResect, resection. c , Heat map of pairwise interaction–avoidance scores for \\nglioblastoma (top rows, n\\u2009=\\u2009192 images), BrM-cores (middle rows, n\\u2009=\\u200959 images) \\nand BrM-margins (bottom rows, n\\u2009=\\u200940 images). Associations should be read \\nrow-to-column. d, Ki67:CC3 ratio in cancer cells interacting with (red; n\\u2009=\\u2009107 \\ncells across 6 images) or avoiding (blue; n\\u2009=\\u200911,163 cells across 67 images) \\nendothelial cells in glioblastoma. Data are median\\u2009±\\u2009interquartile range; \\ntwo-sided Mann–Whitney test. e, Ki67:CC3 ratio in MDMs interacting with  \\n(red; n\\u2009=\\u2009270 cells across 51 images) or avoiding (blue; n\\u2009=\\u20092,808 cells across  \\n94 images) endothelial cells in glioblastoma. Data are mean\\u2009±\\u2009s.e.m.; two-sided \\nStudent’s t-test. f, Ki67 expression in endothelial cells interacting with (red) or \\navoiding (blue) cancer cells in BrM-cores. Data are mean\\u2009±\\u2009s.e.m.; n\\u2009=\\u2009347 cells \\nacross 61 images per group; two-sided Student’s t-test. g, Ki67 expression in \\nendothelial cells interacting with (red) or avoiding (blue) cancer cells in \\nBrM-margins. Data are mean\\u2009±\\u2009s.e.m.; n\\u2009=\\u2009156 cells across 45 images per group; \\ntwo-sided Student’s t -test. h, Ki67 expression in endothelial cells interacting \\nwith (red) or avoiding (blue) Tc cells in BrM-cores. Data are mean\\u2009±\\u2009s.e.m.; n\\u2009=\\u2009235 \\ncells across 41 images per group; two-sided Student’s t-test. In d–h, images with \\nzero cells of interest or lacking pairwise interactions of interest were excluded \\nfrom analysis.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9a2f8f31-f336-463a-9b86-e1d64da50429', embedding=None, metadata={'page_label': '559', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 614 | 16 February 2023 | 559\\nWe next explored vascular interactions in BrM. Similar to glioblas-\\ntoma, endothelial cells had a high tendency to interact with cancer cells \\nin BrM (Fig.\\xa02c), which is\\xa0essential for metastatic colonization following \\nextravasation33,34. Within both BrM-cores and BrM-margins, endothe-\\nlial cells that were associated with cancer cells displayed increased \\nKi67 expression, reminiscent of microvascular proliferation (Fig.\\xa02f,g \\nand Extended Data Fig.\\xa06a). This was of interest given the relationship \\nbetween microvascular proliferation and high-grade glioma35, poten-\\ntially suggesting localized cellular niches of more aggressive tumour \\nfeatures within metastases. Endothelial cell proliferation appeared \\nto be suppressed via interactions with CD8+ T\\xa0cells—an effect that was \\nspecific to BrM-cores (Fig.\\xa02h and Extended Data Fig.\\xa06f). Notably, the \\nexpression of claudin-5—a tight junction protein in the BBB implicated \\nin vascular permeability in BrM36,37—was spatially regulated. Within \\nBrM-cores (but not BrM-margins), cancer-adjacent endothelial cells \\nexhibited lower claudin-5 expression compared with cancer-avoiding \\nendothelial cells (Extended Data Fig.\\xa06g). Moreover, the frequency of \\nclaudin-5+ cancer-adjacent endothelial cells was lower in BrM-cores \\ncompared with BrM-margins (Extended Data Fig.\\xa06g), supporting a \\nmodel of vascular co-option during BrM colonization that is initiated \\nin regions of weakened endothelial junctions34,38,39. As downregulation of \\nclaudin-5 is associated with peritumoural brain oedema40, we subdivided \\nBrM-cores by the degree of oedema as assessed by pre-operative MRI. \\nEndothelial cells that were associated with cancer cells displayed reduced \\nclaudin-5 in BrMs with a moderate-high degree of peritumoural oedema \\n(oedema score 2–3); this relationship was absent in BrMs with absent or \\nlow peritumoural oedema (oedema score 0–1.5) (Extended Data Fig.\\xa06h). \\nThese data highlight a spatially resolved link between BBB integrity and \\nmetastasis, and how it relates to vascular proliferation and oedema.\\nSpatial cellular neighbourhoods\\nWe next explored whether multicellular structures within tumours, \\nrather than pairwise interactions, would provide meaningful insights \\ninto the organization and prognostic value of brain TME dynamics. \\nTwo variables affect cellular neighbourhood assessment: the num-\\nber of interacting cells within a neighbourhood (N) and the number \\nof total cellular neighbourhoods\\xa0(CNs). T o gain insight into how the \\nsize and complexity of neighbourhoods relate to survival, we first \\nused our glioblastoma dataset as a model, and altered the number of \\nnearest spatial neighbours for each individual cell (N\\u2009=\\u20093,5,10,20,30) \\nwhile maintaining a constant number of neighbourhoods (CN\\xa0=\\xa09, as \\nin previous work13). In most cases, CNs\\xa0enriched in M1-like MDMs were \\nassociated with increased survival, regardless of the number of nearest \\nspatial neighbours (Extended Data Fig.\\xa07). Notably, the frequency of \\nM1-like MDMs was not associated with overall survival (Supplemen-\\ntary Fig.\\xa07a), highlighting the value of spatial relationships rather \\nthan abundance alone. T o resolve specific cellular interactions that \\nunderlie this survival advantage, we forced the number of CNs to 30 \\n(rather than 9) while maintaining N\\u2009=\\u200910 nearest neighbours. Using this \\napproach, we resolved six CNs that were enriched for M1-like MDMs \\n(Extended Data Fig.\\xa08a). Of these, only two maintained their relation-\\nship with prolonged survival; these CNs were both primarily composed \\nof M1-like MDMs, neutrophils and M1-like microglia. In the remaining \\nM1-like MDM-enriched CNs, the survival relationship was lost if either \\nneutrophils or M1-like microglia were reduced (Extended Data Fig.\\xa08a). \\nImportantly, we saw no correlation in the prevalence of M1-like MDMs, \\nneutrophils and M1-like microglia (Extended Data Fig.\\xa08b), suggesting \\nthat spatial interactions between these cells are purposeful and not a \\nproduct of their coordinated abundance.\\nWe next compared multicellular interactions between glioblastoma \\nand BrM. Using N\\u2009=\\u200910 nearest neighbours (the mid-point of our model \\nand similar to other studies13), we identified 9 CNs across glioblastoma \\nand BrM\\xa0images (Fig.\\xa03a,b and Extended Data Fig.\\xa09a,b). The cellular \\ncomposition of CNs recapitulated known tissue features, including \\nthe tumour boundary (CN1) or\\xa0tumour compartment (CN8), two \\npan-immune hotspots with either high levels of all immune popula-\\ntions (CN2) or deficiencies in select subsets (CN9), high (CN3) or low \\n(CN4) astrocytes, vascular niche (CN6), macrophage-enriched (CN7), \\nand a neighbourhood largely represented by cells undefined by our \\npanel (CN5) (Fig.\\xa03b). As expected, glioblastoma was dominated by \\nCN3 and CN4 (astrocyte-enriched) whereas BrM-cores were enriched \\nfor CN8 (tumour compartment), reflecting the infiltrative nature of \\ngliomas compared with metastatic tumours (Fig.\\xa03c).\\nMany patients with BrM exhibit metastatic involvement outside \\nthe brain, confounding survival analysis; we therefore correlated CN \\nenrichment with local recurrence. The strongest trend was an associa-\\ntion between high CN5 (undefined) and reduced time to local recur-\\nrence (Extended Data Fig.\\xa09c). We confirmed that 96% of undefined \\ncells in our dataset were CD45− (non-immune) (Extended Data Fig.\\xa09d). \\nWe next investigated the distributions of CNs between BrM-margins \\nand BrM-cores from distinct primary sites. Neighbourhood simi -\\nlarities among BrMs was dictated regionally, rather than by primary \\ntumour type, with BrM-margins being more similar to glioblastoma \\nthan BrM-cores (Fig.\\xa03c and Extended Data Fig.\\xa09e). For example, lung \\nBrM-cores were more similar to melanoma or breast BrM-cores than \\nthey were to lung BrM-margins (Extended Data Fig.\\xa09e). Despite a high \\ndegree of variability in the cell frequencies between BrM from distinct \\nprimary tumours (Supplementary Fig.\\xa06), CN distribution was relatively \\nconstant, highlighting an ability to identify unifying features in brain \\ntumours that may be therapeutically informative.\\nWe next examined a balanced glioblastoma\\xa0cohort of STS and L TS \\nby excluding patients with the greatest confounding clinical variable \\n(partial resection) and controlled for other variables that affect survival, \\nsuch as MGMT methylation status (Extended Data Fig.\\xa09f and Supple-\\nmentary Table\\xa02). We compared the proportion of cells representing \\neach CN within a given tumour sample, and found that L TS tumours \\nhad significantly higher representation of macrophage-enriched CN7 \\nthan STS tumours (Fig.\\xa03d). Moreover, using this refined cohort, we \\nconfirmed the association between CN7 and improved survival (Fig.\\xa03e \\nand Extended Data Fig.\\xa09g). This aligned with our neighbourhood analy-\\nsis using variable numbers of interacting cells, where\\xa0CNs enriched \\nin M1-like MDMs were associated with prolonged survival (Extended \\nData Fig.\\xa07). Notably, CN2 and CN9 (pan-immune hotspots) were also \\nassociated with improved survival (Fig.\\xa03e); analysis of cell dynamics \\nrevealed elevated numbers of CD4+ T\\xa0cells in L TS tumours compared \\nwith STS tumours (Supplementary Fig.\\xa010), which were enriched in both \\nCN2 and CN9 (Fig.\\xa03b). These data suggest a potential beneficial role \\nfor T\\xa0cell neighbourhoods in glioblastoma, despite their low frequency.\\nWe focused on the prognostic relevance of CN7 in glioblastoma, \\nsince it was most consistently associated with survival compared \\nwith other CNs. Moreover, targeting macrophages in brain tumours \\nin the clinical setting is receiving increasing interest in light of prom-\\nising preclinical studies41–43. T o define macrophage identities across \\nneighbourhoods, we extracted all macrophage and monocyte subsets \\nfrom glioblastoma samples and performed t -stochastic neighbour \\nembedding (t-SNE) dimensionality reduction and spectral clustering \\nbased on phenotypic markers in our panel (Fig.\\xa03f). Cellular clusters \\n(CL)1–3 were significantly enriched in L TS tumours compared with \\nSTS tumours (Fig.\\xa03g and Supplementary Fig.\\xa011), the majority of which \\nwere CD163−P2Y12−, suggestive of an M1-like MDM phenotype (Fig.\\xa03h). \\nHowever, they co-expressed CD206, indicating these cells do not fol-\\nlow the M1/M2 paradigm, unlike other clusters with high CD206 that \\nwere enriched for CD163+ cells (for example CL11–15; Fig.\\xa03h). In com-\\nparing the relative representation of CL1–3 across each CN (Fig.\\xa03b), \\nwe observed an enrichment in CN7 specifically in patients with L TS \\ntumours (Fig.\\xa03i). T ogether, these data indicate that macrophage spatial \\nrelationships may contain critical prognostic information, as we have \\nidentified a unique macrophage-enriched neighbourhood associated \\nwith long-term survival—a rarity in this disease.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a232829b-c3d9-4075-8019-efa2cb8fa9fc', embedding=None, metadata={'page_label': '560', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='560 | Nature | Vol 614 | 16 February 2023\\nArticle\\nMPO+ macrophages are associated with survival\\nGiven the relationship between CL1–3 macrophages and L TS \\ntumours, we explored their putative function. CL1–3 macrophages \\nexpressed high levels of MPO (Fig.\\xa04a), and more than 80% of MPO+ \\nmacrophages were CD163−P2Y12− (Extended Data Fig.\\xa010a,b), sug -\\ngesting a pro-inflammatory phenotype dominated by peripher -\\nally\\xa0derived MDMs. MPO is often used as a marker for neutrophils, \\nwhere it mediates production of reactive oxygen species and oxida-\\ntive burst. Although it is likely\\xa0that MPO staining\\xa0within macrophages  \\npartially\\xa0reflects\\xa0enhanced\\xa0phagocytosis of neutrophils, MPO\\xa0transcript \\nis also\\xa0detectable\\xa0in brain tumour MDMs (Extended Data Fig.\\xa010c) and \\nperipheral monocytes (Extended Data Fig.\\xa010d) at comparable levels \\nto neutrophils. Indeed,\\xa0neutrophil-like monocytes and macrophages \\nhave been identified in several immunopathologic contexts, including  \\natherosclerosis 44, neuroinflammation 45,46 and lung cancer 47. It is  \\npossible they arise either from a shift in monocyte developmental \\ntrajectories to favour granulocyte-monocyte progenitor (GMP)-derived \\nlineages48 or through MPO induction within the tissue niche. We con-\\nfirmed the presence of MPO +IBA1+ macrophages in glioblastoma \\ntumours\\xa0using immunofluorescence (Fig.\\xa0 4b; IBA1 was used as an \\nalternative macrophage marker to CD68 for validation purposes). \\nUsing single-cell RNA-sequencing datasets from patients with glioblas-\\ntoma, we identified genes enriched in MPO+ macrophages versus MPO− \\nmacrophages (Extended Data Fig.\\xa010e). Among the top differentially \\nexpressed genes were S100A8 and S100A9, markers of GMP-derived \\nlineages48,49. We observed signatures associated with reactive oxygen \\nspecies biosynthesis and phagosome formation (indicative of cytotox-\\nicity), and HIF1α signalling (suggesting distance from blood vessels) \\n(Fig.\\xa04c). Consistently, MPO+ macrophages were less likely to interact \\nwith endothelial cells compared with MPO− macrophages\\xa0(Extended \\nData Fig.\\xa010f), coinciding with enriched HIF1α in CL1–3 (Fig.\\xa04d). We also \\nsaw reduced LXR–RXR signalling (Fig.\\xa04c), suggesting altered fatty acid \\nmetabolism within these cells that is consistent with their avoidance of \\nab\\nCN9\\nCN8\\nCN6\\nAssign each cell to a \\ncellular neighbourhood\\nDe/f_ine neighbour\\nwindows\\n10 nearest \\nneighbours\\nNK cells\\nMast cells\\nTreg cells\\nUnde/f_ined\\nB cells\\nOther T cells\\nNeutrophil\\nM1-like MDM\\nM2-like MDM\\nCl Mo\\nM1-like MG\\nNon-Cl Mo\\nTc cells\\nTH cells\\nEndothelial\\nM1-like MG\\nDendritic cells\\nInt Mo\\nCancer cells\\nAstrocytes\\nTumour boundary\\nPan-immune hotspot 1\\nAstrocyte high\\nAstrocyte low\\nUnde/f_ined\\nVascular niche\\nMacrophage enriched\\nTumour compartment\\nPan-immune hotspot 2\\nCN1\\nCN2\\nCN3\\nCN4\\nCN5\\nCN6\\nCN7\\nCN8\\nCN9\\n3 \\n0 \\n–3 \\nCN enrichment \\nscore\\nCN1\\nCN2\\nCN3\\nCN4\\nCN5\\nCN6\\nCN7\\nCN8\\nCN9\\nGlioblastoma\\nBrM-cores\\nc\\nCN1 CN2 CN3 CN4 CN5 CN6 CN7 CN8 CN9\\n0\\n20\\n40\\n60\\n80\\n100\\nP = 0.0283\\nPercentage of cells in each CN\\n(average across all patients)\\nSTS\\nLTS\\nd\\ne\\nCN high (n = 16) CN low (n = 16)\\ng\\nh\\nCN2 CN7 CN9\\nSurvival (%)\\nTime (years)\\nCD163+ MDM\\nCD163– MDM\\nCD163– MG\\nCD163+ MG\\nCD14+CD16– mono\\nCD14+CD16+ mono\\nCD14–CD16+ mono\\n1\\n2\\n3\\n4\\n5\\n6\\n7\\n8\\n9\\n10\\n11\\n12\\n13\\n14\\n15\\n0 5,000 10,000\\nTotal no. of cells\\nCN1\\nCN2\\nCN3\\nCN4\\nCN5\\nCN6\\nCN7\\nCN8\\nCN9\\nNo. of CL1-3 cells in each CN\\n1,5000 3,000\\ni\\nSTS\\nLTS\\nf\\nt-SNE2\\nt-SNE1\\nSTS LTS\\nMarker\\nintensity \\n1\\n0.5\\n0.25\\n0 123\\n0\\n25\\n50\\n75\\n100\\nP = 0.0223\\n0 123\\nP = 0.0021\\n0 123\\nP = 0.0161\\nt-SNE2\\nt-SNE1\\nCC3\\nCD40\\nCD206\\nHIF1/uni03B1\\nKi67\\nMMP9\\nMPO\\nOX40L\\npSTAT3\\nFig. 3 | Spatial cellular neighbourhoods relate to survival in glioblastoma. \\na, Schematic of cellular neighbourhood (CN) assignments. CNs are projected  \\nas a Voronoi diagram (right). b, Heat map of cell types represented across 9 CNs \\ndiscovered in glioblastoma and BrM-cores ( n\\u2009=\\u2009251 images; N\\u2009=\\u200910 nearest \\nneighbours, CN =\\xa09 neighbourhoods). c, The distribution of CNs across \\nglioblastoma ( n\\u2009=\\u2009192 images) and BrM-cores (n\\u2009=\\u200959 images). For each image, \\nthe percentage of cells from each CN was determined and then averaged for \\neach disease type. d–i, Analysis of a controlled glioblastoma\\xa0cohort of LTS \\n(overall survival\\u2009>3 years) and STS (overall survival\\u2009<1 year) (see Extended Data \\nFig.\\xa09f). d, The distribution of CNs\\xa0in the LTS and STS glioblastoma cohort. CN \\nfrequencies were averaged where there were multiple samples from the same \\npatient. Data are median\\u2009±\\u2009interquartile range; n\\u2009=\\u200916 patients per group; \\ntwo-sided Mann–Whitney test. e , Kaplan–Meier analysis of the LTS and STS \\nglioblastoma cohort based on the median CN frequency. CN frequencies were \\naveraged where there were multiple samples from the same patient. Log-rank \\n(Mantel–Cox) test; n\\u2009=\\u200916 patients per group. f, t-SNE unsupervised clustering \\nof macrophages and monocytes from all glioblastoma images ( n\\u2009=\\u200993,513 cells \\nacross 192 images). g, t-SNE projection of monocytes and macrophages from \\npatients with glioblastoma, with cells in clusters CL1–3 outlined in red.  \\nLTS, n\\u2009=\\u200917,752 cells across 32 images; STS, n\\u2009=\\u200910,456 cells across 28 images.  \\nh, Relative expression of functional markers (left) and the distribution of cell \\ntypes (right) across 15 monocyte and macrophage clusters (CL1–15). i , The \\nnumber of cells from CL1–3 found within each cellular neighbourhood in the \\nLTS and STS glioblastoma cohort.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b43d0e70-1a87-41b3-b2d8-4f704c478a51', embedding=None, metadata={'page_label': '561', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 614 | 16 February 2023 | 561\\nfatty streaks in atherosclerosis44. These data suggest that, although rare \\nin the TME, MPO+ macrophages may have anti-tumorigenic properties.\\nWe next categorized tissues on the basis of the median density of \\nMPO+ M1-like MDM: none, low (1–5 cells per mm2) and high (6 or more \\ncells per mm2). Higher numbers of MPO+ macrophages corresponded \\nwith an increase in total neutrophils, M1-like MDMs and classical \\nmonocytes (Fig.\\xa04e), supporting the notion that these tumours may \\nbe primed for strong innate effector responses. We also compared \\nCN prevalence and found an enrichment in cells associated with \\nCN7 (macrophage-enriched, as expected) and CN2 (pan-immune  \\nhotspot) (Fig.\\xa04f), both of which provided a significant survival benefit \\nin patients (Fig.\\xa03e). T o gain insight into the tumour architecture, we \\ncompared spatial interactions between major immune lineages across \\nMPO categories. Cancer cells displayed a greater tendency to avoid \\nneutrophils and M1-like MDMs as the density of MPO+ macrophages \\nincreased (Fig.\\xa04g). Indeed, most myeloid cell populations avoided \\nneutrophils as MPO+ macrophage density increased, with the excep-\\ntion of M1-like MDMs (Fig.\\xa04g). The relationship between neutrophils \\nand MPO+ macrophages was confirmed by immunohistofluorescence, \\nas samples\\xa0with\\xa0high MPO+IBA1− neutrophils also had high MPO+IBA1+ \\nmacrophages (Extended Data Fig.\\xa010g). Similarly, interaction analysis \\nof MPO+ M1-like MDMs revealed\\xa0interactions with both neutrophils and \\nM1-like microglia (Extended Data Fig.\\xa010h), echoing our CN\\xa0findings\\xa0 \\n(Extended Data Fig.\\xa08a). Higher densities of MPO+ macrophages were \\nalso associated with increased interactions between endothelial cells \\nand M1-like MDMs with T\\xa0cells, concomitant with reduced interac -\\ntions between M2-like MDMs and CD8+ T\\xa0cells (Fig.\\xa04g), which can be \\nimmunoregulatory in cancer. Within the macrophage compartment, \\nActin cytoskeleton signalling\\nPhagosome formation\\nRef 51 Ref 52 Ref 53\\nc\\nTREM1 signalling \\nHIF1/uni03B1 signalling\\nProduction of NO and ROS \\nin macrophages\\nLXR–RXR activation\\nEndothelial cells\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nT c  cells\\nT H  cells\\nT reg  cells\\nOther T cells\\nM1-like MG\\nM2-like MG\\nCl Mo\\nInt Mo\\nUnde/f_ined\\nM1-like MDM\\nM2-like MDM\\nNon-Cl Mo\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nT c  cells\\nT H  cells\\nT reg  cells\\nT c  cells\\nT H  cells\\nT reg  cells\\nT c  cells\\nT H  cells\\nT reg  cells\\nEndothelial cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nOther T cells\\nNon-Cl Mo\\nInt Mo\\nUnde/f_ined\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nEndothelial cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nOther T cells\\nM2-like MG\\nCl Mo\\nNon-Cl Mo\\nInt Mo\\nUnde/f_ined\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nEndothelial cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nOther T cells\\nNon-Cl Mo\\nInt Mo\\nUnde/f_ined\\nNo MPO Low MPO\\n1\\n10\\n100\\n1,000\\nf CN7\\nP = 0.0465\\nP = 0.0206\\nP = 0.0494\\nLow\\nNone\\nHigh\\n1\\n10\\n100\\n1,000\\n1\\n10\\n100\\n1,000\\nCN2\\nP = 0.0027\\nP = 0.0385\\nHigh MPO\\n0\\n25\\n50\\n75\\n100\\nActivity z-score\\n05–5\\nLow (n  = 102)\\nHigh (n = 33)\\n0123\\nTime (years)\\nInteract\\n+1\\nP = 0.0279\\nAvoid\\n0\\n–1\\nLow\\nNone\\nHigh\\nSurvival (%)\\nh\\nMPO IBA1 DAPI\\n0\\n0.5\\nMarker\\nintensity\\nMarker\\nintensity\\n1.0d\\ng\\nHIF1/uni03B1 expression\\nt-SNE2\\nt-SNE1\\ne\\n1\\n10\\n100\\n1,000\\nNo. of cells \\nper mm 2  core\\nNo. of cells \\nper mm 2  core\\nP = 0.0028\\nP = 0.0006\\n1\\n10\\n100\\n1,000\\nP = 0.0069\\nP = 0.0067\\n1.0\\n0\\n0.5\\nb\\n50 /uni03BCm \\na MPO expression\\nCl MoM1-like MDMsNeutrophils\\nFig. 4 | MPO+ macrophages are enriched in LTS tumours and are associated \\nwith enhanced cytotoxic functions. a, Heat map of MPO expression projected \\nonto a t-SNE map (Fig.\\xa03f) of monocytes and macrophages from patients  \\nwith glioblastoma (n\\u2009=\\u200993,513 cells across 192 images). b, Representative \\nimmunohistofluorescence (IHF) images showing MPO and IBA1 (a macrophage \\nmarker) co-localization in glioblastoma tumours. Expanded regions show \\nexamples of MPO+ macrophages (n\\u2009=\\u20095 images). c, Ingenuity pathway analysis of \\nenriched pathways in MPO+ versus MPO− macrophages from three publicly \\navailable datasets51–53. d, HIF1α expression projected onto a t-SNE map of \\nmonocytes and macrophages from patients with glioblastoma (n\\u2009=\\u200993,513 cells \\nacross 192 images). e, The number of cells per 1\\xa0mm2 core in glioblastoma samples \\nwith zero (n\\u2009=\\u200932 images), low (n\\u2009=\\u200979 images) or high (n\\u2009=\\u200981 images) MPO+ M1-like \\nMDMs. The graph shows mean values (black horizontal line) and all data points; \\none-way ANOVA; data are presented in log scale, so\\xa0images with 0 cells were \\nassigned a value of 1. f, The raw number of cells in each cellular neighbourhood \\nper 1\\u2009mm2 core from patients with glioblastoma with zero (n\\u2009=\\u200932 images), low \\n(n\\u2009=\\u200979 images) or high (n\\u2009=\\u200981 images) numbers of MPO+ M1-like MDMs. The graph \\nshows mean values (black horizontal line) and all data points; one-way ANOVA; \\nimages with 0 cells were assigned a value of 1. g, Pairwise interactions across \\ntwo-sided permutation tests on individual images (1,000 permutations each) for \\npatients with zero, low or high numbers of MPO+ M1-like MDMs. Red, interactions \\n(interact); blue, avoidances (avoid). h, Kaplan–Meier analysis based on MPO+IBA1+ \\ncell frequency as determined by IHF staining in 135 tumours from patients with \\nglioblastoma (z-score). Cell frequencies were averaged when multiple samples \\ncorresponded to the same individual. Log-rank (Mantel–Cox) test.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='dc8e27c0-1f4e-4eb0-b651-725e93727e8c', embedding=None, metadata={'page_label': '562', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='562 | Nature | Vol 614 | 16 February 2023\\nArticle\\nassociations between MDMs increased, whereas they were relatively \\nless pronounced for microglia (Fig.\\xa04g)—suggesting that the dynamics \\nof tissue-resident versus monocyte-derived macrophages may have \\ndistinct effects on tumour biology, as has been suggested in preclini-\\ncal models41. These data illuminate gradual shifts in TME composition \\nwith changes in the density of MPO+ macrophages.\\nFinally, to further probe clinical relevance, we confirmed that \\nincreased levels of MPO +CD163−P2Y12−CD68+ macrophages were \\nassociated with prolonged survival using our balanced glioblastoma \\ncohort (Extended Data Fig.\\xa010i). Consistently, recurrent glioblastomas \\ncontained fewer MPO+CD163−P2Y12−CD68+ macrophages compared \\nwith either STS\\xa0or\\xa0L TS tumours\\xa0(Extended Data Fig.\\xa010j). T o test the prac-\\nticality of our findings using a lower-plex technology, we performed \\nimmunohistofluorescence co-staining for MPO and IBA1 to evaluate \\nsurvival outcomes in a cohort of 135 patients with glioblastoma. As the \\nvast majority (83.96%) of MPO+ macrophages are CD163− and P2Y12− \\n(Extended Data Fig.\\xa010a), the combination of MPO with IBA1 was suf-\\nficient to confirm significantly prolonged survival in patients with high \\nfrequencies of double-positive cells (Fig.\\xa04h). Building on previous work \\nshowing that macrophage accumulation in brain tumours is associated \\nwith advanced disease24, our findings highlight an MPO+ subpopula-\\ntion associated with an unexpected survival benefit. This suggests that \\ntherapies that broadly target macrophages in glioblastoma may deplete \\na beneficial macrophage subset, adding insight to clinical trials with \\nCSF-1R inhibitors that have been\\xa0largely unsuccessful50, and deepening \\nour understanding of macrophage complexity in this disease.\\nDiscussion\\nWe have provided a high-dimensional spatial map of the brain TME \\nusing IMC. We performed a comprehensive analysis of cellular dynam-\\nics, interactions and neighbourhoods in glioblastoma and BrM, and \\ncorrelated spatial features of glioblastoma with\\xa0patient survival. We \\ndefined multicellular structures that are common across disease \\nstates and regions, which are superior for predicting patient survival \\ncompared with cell frequencies alone. We identified a unique subset \\nof neutrophil-like macrophages that stain positively for\\xa0MPO, which \\nappear to be beneficial for the survival of patients with glioblastoma. We \\nspeculate that host immunity favouring GMP-derived neutrophil-like \\nstates may be advantageous for patient outcomes48. Alternatively, \\nprogramming macrophages to adopt neutrophil-like characteristics\\xa0or \\nenhanced phagocytosis may occur within specific tumour niches. Our \\ndataset adds to the growing evidence that the spatial organization of \\ntumours at the cellular level is critical for understanding immunological \\nmechanisms governing cancer. Given the limited therapeutic options \\nfor patients with brain tumours and dismal prognosis, there is untapped \\ntranslational value in understanding how the spatial architecture of the \\nbrain TME relates to tumour biology, and whether specific immune cell \\nsubsets can be harnessed to improve outcomes.\\nOnline content\\nAny methods, additional references, Nature Portfolio reporting summa-\\nries, source data, extended data, supplementary information, acknowl-\\nedgements, peer review information; details of author contributions \\nand competing interests; and statements of data and code availability \\nare available at https://doi.org/10.1038/s41586-022-05680-3.\\n1. Quail, D. F. & Joyce, J. A. Microenvironmental regulation of tumor progression and \\nmetastasis. Nat. Med. 19, 1423–1437 (2013).\\n2. Binnewies, M. et\\xa0al. Understanding the tumor immune microenvironment (TIME) for effective \\ntherapy. Nat. Med. 24, 541–550 (2018).\\n3. Baslan, T. & Hicks, J. Unravelling biology and shifting paradigms in cancer with single-cell \\nsequencing. Nat. Rev. Cancer 17, 557–569 (2017).\\n4. Ostrom, Q. T. et\\xa0al. CBTRUS statistical report: primary brain and other central nervous \\nsystem tumors diagnosed in the United States in 2010–2014. Neuro Oncol. 19, v1–v88 \\n(2017).\\n5. Barnholtz-Sloan, J. S. et\\xa0al. Incidence proportions of brain metastases in patients diagnosed \\n(1973 to 2001) in the Metropolitan Detroit Cancer Surveillance System. J. Clin. Oncol. 22, \\n2865–2872 (2004).\\n6. Stupp, R. et\\xa0al. Radiotherapy plus concomitant and adjuvant temozolomide for \\nglioblastoma. N. Engl. J. Med. 352, 987–996 (2005).\\n7. Cagney, D. N. et\\xa0al. Incidence and prognosis of patients with brain metastases at \\ndiagnosis of systemic malignancy: a population-based study. Neuro Oncol. 19, 1511–1521 \\n(2017).\\n8. Quail, D. F. & Joyce, J. A. The microenvironmental landscape of brain tumors. Cancer Cell \\n31, 326–341 (2017).\\n9. Carvajal-Hausdorf, D. E. et\\xa0al. Multiplexed (18-Plex) measurement of signaling targets \\nand cytotoxic T\\xa0cells in trastuzumab-treated patients using imaging mass cytometry. \\nClin. Cancer Res. 25, 3054–3062 (2019).\\n10. Ali, H. R. et\\xa0al. Imaging mass cytometry and multiplatform genomics define the \\nphenogenomic landscape of breast cancer. Nat. Cancer 1, 163–175 (2020).\\n11. Jackson, H. W. et\\xa0al. The single-cell pathology landscape of breast cancer. Nature 578, \\n615–620 (2020).\\n12. Keren, L. et\\xa0al. A structured tumor–immune microenvironment in triple negative breast \\ncancer revealed by multiplexed ion beam imaging. Cell 174, 1373–1387.e1319 (2018).\\n13. Schürch, C. M. et\\xa0al. Coordinated cellular neighborhoods orchestrate antitumoral \\nimmunity at the colorectal cancer invasive front. Cell 182, 1341–1359 (2020).\\n14. Liudahl, S. M. et\\xa0al. Leukocyte heterogeneity in pancreatic ductal adenocarcinoma: \\nphenotypic and spatial features associated with clinical outcome. Cancer Discov. 11, \\n2014–2031 (2021).\\n15. Friebel, E. et\\xa0al. Single-cell mapping of human brain cancer reveals tumor-specific \\ninstruction of tissue-invading leukocytes. Cell 181, 1626–1642.e1620 (2020).\\n16. Klemm, F. et\\xa0al. Interrogation of the microenvironmental landscape in brain tumors \\nreveals disease-specific alterations of immune cells. Cell 181, 1643–1660.e1617 (2020).\\n17. Schaettler, M. O. et\\xa0al. Characterization of the genomic and immunologic diversity  \\nof malignant brain tumors through multisector analysis. Cancer Discov. 12, 154–171  \\n(2022).\\n18. Puchalski, R. B. et\\xa0al. An anatomic transcriptional atlas of human glioblastoma. Science \\n360, 660–663 (2018).\\n19. Muller, S. et\\xa0al. Single-cell profiling of human gliomas reveals macrophage ontogeny as a \\nbasis for regional differences in macrophage activation in the tumor microenvironment. \\nGenome Biol. 18, 234 (2017).\\n20. Ginhoux, F., Schultze, J. L., Murray, P. J., Ochando, J. & Biswas, S. K. New insights into the \\nmultidimensional concept of macrophage ontogeny, activation and function. Nat. Immunol. \\n17, 34–40 (2016).\\n21. Komohara, Y., Ohnishi, K., Kuratsu, J. & Takeya, M. Possible involvement of the M2 \\nanti-inflammatory macrophage phenotype in growth of human gliomas. J. Pathol. 216, \\n15–24 (2008).\\n22. Lu-Emerson, C. et\\xa0al. Increase in tumor-associated macrophages after antiangiogenic \\ntherapy is associated with poor survival among patients with recurrent glioblastoma. \\nNeuro Oncol. 15, 1079–1087 (2013).\\n23. Louis, D. N. et\\xa0al. The 2021 WHO Classification of Tumors of the Central Nervous System: \\na summary. Neuro Oncol. 23, 1231–1251 (2021).\\n24. Prosniak, M. et\\xa0al. Glioma grade is associated with the accumulation and activity of cells \\nbearing M2 monocyte markers. Clin. Cancer Res. 19, 3776–3786 (2013).\\n25. Park, J. H. et\\xa0al. Tumor hypoxia represses γδ T\\xa0cell-mediated antitumor immunity against \\nbrain tumors. Nat. Immunol. 22, 336–346 (2021).\\n26. Reardon, D. A. et\\xa0al. Effect of nivolumab vs bevacizumab in patients with recurrent \\nglioblastoma: the CheckMate 143 Phase 3 Randomized Clinical Trial. JAMA Oncol. 6, \\n1003–1010 (2020).\\n27. Tawbi, H. A. et\\xa0al. Combined nivolumab and ipilimumab in melanoma metastatic to the \\nbrain. N. Engl. J. Med. 379, 722–730 (2018).\\n28. Long, G. V. et\\xa0al. Combination nivolumab and ipilimumab or nivolumab alone in \\nmelanoma brain metastases: a multicentre randomised phase 2 study. Lancet Oncol. \\n19, 672–681 (2018).\\n29. Calabrese, C. et\\xa0al. A perivascular niche for brain tumor stem cells. Cancer Cell 11, 69–82 \\n(2007).\\n30. Ghajar, C. M. et\\xa0al. The perivascular niche regulates breast tumour dormancy. Nat. Cell \\nBiol. 15, 807–817 (2013).\\n31. Liu, H. et\\xa0al. Pro-inflammatory and proliferative microglia drive progression of \\nglioblastoma. Cell Rep. 36, 109718 (2021).\\n32. van Hooren, L. et\\xa0al. Agonistic CD40 therapy induces tertiary lymphoid structures but \\nimpairs responses to checkpoint blockade in glioma. Nat. Commun. 12, 4127 (2021).\\n33. Kienast, Y. et\\xa0al. Real-time imaging reveals the single steps of brain metastasis formation. \\nNat. Med. 16, 116–122 (2010).\\n34. Valiente, M. et\\xa0al. Serpins promote cancer cell survival and vascular co-option in brain \\nmetastasis. Cell 156, 1002–1016 (2014).\\n35. Gi, T. et\\xa0al. Microvascular proliferation of brain metastases mimics glioblastomas in squash \\ncytology. Cytopathology 28, 228–234 (2017).\\n36. Vanlandewijck, M. et\\xa0al. A molecular atlas of cell types and zonation in the brain vasculature. \\nNature 554, 475–480 (2018).\\n37. Arvanitis, C. D., Ferraro, G. B. & Jain, R. K. The blood–brain barrier and blood–tumour \\nbarrier in brain tumours and metastases. Nat. Rev. Cancer 20, 26–41 (2020).\\n38. Avraham, H. K. et\\xa0al. Angiopoietin-2 mediates blood-brain barrier impairment and \\ncolonization of triple-negative breast cancer cells in brain. J. Pathol. 232, 369–381 (2014).\\n39. Carbonell, W. S., Ansorge, O., Sibson, N. & Muschel, R. The vascular basement membrane \\nas “soil” in brain metastasis. PLoS ONE 4, e5857 (2009).\\n40. Liebner, S. et\\xa0al. Claudin-1 and claudin-5 expression and tight junction morphology are \\naltered in blood vessels of human glioblastoma multiforme. Acta Neuropathol. 100,  \\n323–331 (2000).\\n41. Akkari, L. et\\xa0al. Dynamic changes in glioma macrophage populations after radiotherapy \\nreveal CSF-1R inhibition as a strategy to overcome resistance. Sci. Transl. Med. 12, \\neaaw7843 (2020).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c1948760-dc9b-4c99-a803-f41e885b94d8', embedding=None, metadata={'page_label': '563', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 614 | 16 February 2023 | 563\\n42. Pyonteck, S. M. et\\xa0al. CSF-1R inhibition alters macrophage polarization and blocks glioma \\nprogression. Nat. Med. 19, 1264–1272 (2013).\\n43. Quail, D. F. et\\xa0al. The tumor microenvironment underlies acquired resistance to CSF-1R \\ninhibition in gliomas. Science 352, aad3018 (2016).\\n44. Sugiyama, S. et\\xa0al. Macrophage myeloperoxidase regulation by granulocyte macrophage \\ncolony-stimulating factor in human atherosclerosis and implications in acute coronary \\nsyndromes. Am. J. Pathol. 158, 879–891 (2001).\\n45. Amorim, A. et\\xa0al. IFNγ and GM-CSF control complementary differentiation programs in \\nthe monocyte-to-phagocyte transition during neuroinflammation. Nat. Immunol. 23, \\n217–228 (2022).\\n46. Reynolds, W. F. et\\xa0al. Myeloperoxidase polymorphism is associated with gender specific \\nrisk for Alzheimer’s disease. Exp. Neurol. 155, 31–41 (1999).\\n47. Zilionis, R. et\\xa0al. Single-cell transcriptomics of human and mouse lung cancers reveals \\nconserved myeloid populations across individuals and species. Immunity 50, 1317–1334.\\ne1310 (2019).\\n48. Yanez, A. et\\xa0al. Granulocyte–monocyte progenitors and monocyte–dendritic cell \\nprogenitors independently produce functionally distinct monocytes. Immunity 47, \\n890–902.e894 (2017).\\n49. Lagasse, E. & Weissman, I. L. Mouse MRP8 and MRP14, two intracellular calcium-binding \\nproteins associated with the development of the myeloid lineage. Blood 79, 1907–1915 \\n(1992).\\n50. Butowski, N. et\\xa0al. Orally administered colony stimulating factor 1 receptor inhibitor PLX3397 \\nin recurrent glioblastoma: an Ivy Foundation Early Phase Clinical Trials Consortium phase II \\nstudy. Neuro Oncol. 18, 557–564 (2016).\\n51. Lee, A. H. et\\xa0al. Neoadjuvant PD-1 blockade induces T\\xa0cell and cDC1 activation but fails  \\nto overcome the immunosuppressive tumor associated macrophages in recurrent \\nglioblastoma. Nat. Commun. 12, 6938 (2021).\\n52. Ravi, V. M. et\\xa0al. T-cell dysfunction in the glioblastoma microenvironment is mediated by \\nmyeloid cells releasing interleukin-10. Nat. Commun. 13, 925 (2022).\\n53. Xie, Y. et\\xa0al. Key molecular alterations in endothelial cells in human glioblastoma \\nuncovered through single-cell RNA sequencing. JCI Insight 6, e150861 (2021).\\nPublisher’s note Springer Nature remains neutral with regard to jurisdictional claims in \\npublished maps and institutional affiliations.\\nOpen Access This article is licensed under a Creative Commons Attribution \\n4.0 International License, which permits use, sharing, adaptation, distribution \\nand reproduction in any medium or format, as long as you give appropriate \\ncredit to the original author(s) and the source, provide a link to the Creative Commons licence, \\nand indicate if changes were made. The images or other third party material in this article are \\nincluded in the article’s Creative Commons licence, unless indicated otherwise in a credit line \\nto the material. If material is not included in the article’s Creative Commons licence and your \\nintended use is not permitted by statutory regulation or exceeds the permitted use, you will \\nneed to obtain permission directly from the copyright holder. To view a copy of this licence, \\nvisit http://creativecommons.org/licenses/by/4.0/.\\n© The Author(s) 2023', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='da09766f-747a-409c-87b7-fdb01cf9a9b0', embedding=None, metadata={'page_label': '564', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nMethods\\nClinical samples for IMC\\nA cohort of 185 patients underwent surgical resection for primary brain \\ntumours or BrM between 2006–2019. A breakdown of the patients \\nwhose tumour samples were used in this study can be found in Extended \\nData Fig.\\xa01a. The clinical data for all patients was obtained from sur-\\ngical and pathological reports. All tumour samples obtained from \\npatients with glioblastoma were previously untreated and classified \\nby a certified neuropathologist (M.C.G.) following primary surgical \\ndebulking. A subset of tumours was removed during a second follow-up \\nsurgery (residual tumour) or following tumour recurrence (recurrent \\ntumour). In accordance with World Health Organization 2021\\xa0guide-\\nlines23, tumours formerly classified as IDH mutant glioblastoma are \\nnow considered grade IV IDH mutant astrocytoma; therefore, in this \\nstudy, only grade IV IDH wild-type tumours were designated as glio-\\nblastoma. We further distinguished glioblastomas that resulted from \\nprogression from grade II/III tumours (Extended Data Fig.\\xa01a). L TSs were \\ndefined as patients with an overall survival greater than three years \\n(much longer than expected survival time), and STSs were defined \\nas patients with an overall survival less than one year (shorter than \\nexpected survival time).\\xa0Brain metastases samples54 were surgically \\nremoved from patients bearing lung (n\\u2009=\\u200951 images), breast (n\\u2009=\\u200929 \\nimages), and melanoma (n\\u2009=\\u200919 images) primary tumours as well as \\na small number of bladder, colorectal, gastric, gastrointestinal and \\novarian tumours, collectively called ‘other’ (n \\u2009=\\u200920 images) in this \\nstudy. Pre-operative MRI images were used to determine the extent \\nof peritumoural oedema (scored 0–3) by a neuroradiologist (S.L.). \\nLeptomeningeal disease 55 was determined by contrast-enhancing \\nlesions in the subarachnoid or ventricles as determined on MRI by \\na neuroradiologist (S.L.). All patients underwent standard of care \\n(SOC) following surgery, unless otherwise specified. Cores (1–1.5\\u2009mm \\nin diameter) were removed from formalin-fixed, paraffin embedded \\n(FFPE) tissue blocks and assembled into tissue microarrays (TMAs). \\nWithin the glioma cohort, we included tumour-adjacent ‘normal’ tissues \\nas well as primary brain tumour samples extracted from the tumour \\nbulk, which were confirmed using corresponding haematoxylin and \\neosin (H&E) staining by a neuropathologist (M.C.G.). BrM samples \\nwere extracted from the tumour bulk and/or tumour–brain interface \\n(termed BrM-cores and BrM- margins, respectively). A total of 242 tissue \\nregions were sampled across the 185 patients including 139 high-grade \\nglioma, 18 glioma-adjacent normal, 41 BrM-cores and 44 BrM-margins. \\nOf these 242 regions, 142 were sampled in duplicate and 2 were sampled \\nin triplicate for a total of 389 cores. Additionally, 39 patients with BrM \\nhad matched core-margin pairs. All surgical specimens and clinical \\ninformation were obtained following written informed patient con-\\nsent. Clinical information was de-identified and used in accordance \\nwith the institutional review boards of McGill University and Montreal \\nNeurological Institute-Hospital (REB: NEU-10-066, 2018-4150).\\nAntibody optimization\\nAntibodies were optimized on control tissues including spleen, tonsil, \\nlymph node, liver, kidney, normal lung, normal brain, lung cancer, \\nglioblastoma and/or BrM. In Extended Data Fig.\\xa03a and Supplementary \\nFigs.\\xa01 and 2, we show representative optimization images of both immu-\\nnohistofluorescence\\xa0(IHF) and IMC staining for all markers in our panel, \\nwith some exceptions: IHF was not performed for antibodies that were \\ncommercially available with conjugated metal isotopes (except CD20 \\nand CD45; unconjugated forms were used for IHF), and IHF was not per-\\nformed for Ki67 as the B56 clone is routinely used. A list of all antibodies \\ncan be found in Supplementary Table\\xa01. For IHF staining, FFPE sections \\nunderwent deparaffinization and heat-mediated antigen retrieval using \\nthe Ventana Discovery Ultra auto-stainer platform (Roche Diagnostics) \\naccording to manufacturer instructions. FFPE slides were incubated at \\n70\\u2009°C in pre-formulated EZ Prep solution (Roche Diagnostics), followed \\nby incubation at 95\\u2009°C in pre-formulated Cell Conditioning 1 solution \\n(Roche Diagnostics) for a total run time of ~2.5\\u2009h. Slides were rinsed in \\n1× PBS and incubated for 1\\u2009h at room temperature in Dako Serum-free \\nProtein Block solution (Agilent). An antibody cocktail was prepared in \\nDako Antibody Diluent and slides were incubated with primary antibod-\\nies overnight at 4\\u2009°C. Slides were rinsed with 1× PBS and incubated with \\nsecondary antibody cocktail prepared in Dako Antibody Diluent for 1\\u2009h \\nat room temperature. Slides were counterstained with DAPI for 5\\u2009min \\nat room temperature and mounted using Dako Mounting Medium. An \\nAxioScan Z1 scanner was used to capture tissue images.\\nImmunostaining and IMC\\nFFPE TMA slides underwent deparaffinization and heat-mediated anti-\\ngen retrieval using the Ventana Discovery Ultra auto-stainer platform \\n(Roche Diagnostics) according to the manufacturer’s instructions. \\nFFPE slides were incubated at 70\\u2009°C in pre-formulated EZ Prep solution \\n(Roche Diagnostics), followed by incubation at 95\\u2009°C in pre-formulated \\nCell Conditioning 1 solution (Roche Diagnostics) for a total run time \\nof ~2.5\\u2009h. Slides were rinsed in 1× PBS and incubated for 45\\u2009min at room \\ntemperature in Dako Serum-free Protein Block solution (Agilent). An \\nantibody cocktail containing metal-conjugated antibodies was pre-\\npared in Dako Antibody Diluent at optimized dilutions. Slides were \\nstained with primary antibodies at 4\\u2009°C overnight and subsequently \\nwashed with 0.2% Triton X-100 and 1× PBS. A secondary antibody cock-\\ntail containing metal-conjugated anti-biotin was prepared in Dako \\nAntibody Diluent at the optimized dilution. Slides were incubated with \\nanti-biotin for 1\\u2009h at room temperature and subsequently washed with \\n0.2% Triton X-100 and 1× PBS. Slides were counterstained with Cell-ID \\nIntercalator-Ir (Fluidigm) diluted at 1:400 in 1× PBS for 30\\u2009min at room \\ntemperature, rinsed for 5\\u2009min with distilled water, and air-dried prior \\nto IMC acquisition. IMC acquisition was performed using the Hyperion \\nImaging System (Fluidigm).\\nData transformation and normalization\\nAll IMC data presented were not transformed and analyses were based \\non raw measurements. Single-cell marker expressions are summarized \\nby mean pixel values for each channel. For heat map visualization, \\nexpression data were normalized to the 95th percentile and z-scored \\ncluster means were plotted.\\nCell segmentation and lineage assignment\\nAll lineage and functional markers underwent a staining quality check \\nprior to cell segmentation. A subset of functional markers passed ini-\\ntial quality control, but did not stain consistently with IMC, and were \\nsubsequently removed from analysis (GM-CSF-R, M-CSF-R, PD-1, PD-L1 \\nand CTLA-4; see Supplementary Fig.\\xa01). Cell segmentation was done \\nusing a combination of classical and modern machine learning-based \\ncomputer vision algorithms. This pipeline enables high-throughput \\nsegmentation and accurately resolves individual cells across diverse \\ntissues and structures. Importantly, this algorithm fully automates the \\ndetection of cells, thus eliminating subjective bias. The DNA channel is \\npre-processed for nuclei segmentation to obtain foreground regions of \\ninterest using mixtures of generalized Gaussian distributions (MoGG). \\nThe channels are also tiled for segmentation so we can pass them as \\ninputs for inference to the MaskRCNN model. A detailed description of \\nour segmentation and image analysis pipeline is available56. T o assign \\ncell phenotypes, we established a supervised approach based on canon-\\nical lineage markers, expected population abundance, staining qual-\\nity, and maturity of cell lineage (Extended Data Fig.\\xa01b). We first used \\nk-means clustering57 and a mixture of generalized Gaussian models58 \\nto create multi-level image stacks based on the staining intensity of \\neach marker. Masks were curated for each lineage marker in the panel \\nbased on consideration of 6 levels using the following procedure. (1) \\nGreyscale image channel is convolved with a median filter with a 3\\u2009×\\u20093 \\nwindow size. (2) Each pixel is clustered into 6 groups of intensity levels ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1c40f6d7-5093-43a6-ba77-60a3d54a2c22', embedding=None, metadata={'page_label': '565', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='using the k-means algorithm. (3) For each channel we select all groups \\nup to a particular level as foreground (1) and the rest are designated as \\nbackground (0). (4) We apply morphological blob removal to obtain \\nsmoother binary masks, where binary blobs of a particular area are \\nremoved from masks to avoid noisy regions. (5) T o further refine the \\naccuracy of select markers, additional channel-specific morphologi-\\ncal operations are applied by computing an additional binary mask \\nobtained using the adaptive binarization method with a sensitivity \\nof 0.4. This mask is then amalgamated with the mask obtained in step \\n4. (6) T o enhance the image intensity of select channels, we apply a \\nsimple contrast enhancement filter by saturating the bottom and top \\nintensity levels of pixels in particular channels. This process enables \\nus to capture more accurate masks from channels when phenotyping \\ncells within our cores.\\nThe method of lineage assignment is represented in the following \\nformula: for each cell ci we consider the curated mask for each lineage \\nmarker Mk, where k\\u2009=\\u20091,…,n and n is the number of lineage markers. Let \\nus assume pc\\nj\\ni\\n be the jth pixel that lies in the surrounding of ci and each \\npixel has the following presence vector based on the lineage markers:\\n{}()Ep pp p=, ,… ,c\\nj\\nM\\nj\\nM\\nj\\nM\\nj\\nin 12\\nwhere p ={ 0o r1 }Mi\\n, which determines whether the pixel pc\\nj\\ni\\n is positive \\nfor a particular marker. Next, to determine whether each pixel within \\na cell is positive or negative for a given marker, we determine the major-\\nity vector by summing over the presence of all vectors as:\\n\\uf8f4\\uf8f4\\n\\uf8f4\\uf8f4\\uf8f1\\n\\uf8f2\\n\\uf8f3\\n\\uf8fc\\n\\uf8fd\\n\\uf8fe\\n∑∑ ∑Mp pp=, ,… ,c\\nj\\nN\\nM\\nj\\nj\\nN\\nM\\nj\\nj\\nN\\nM\\nj\\n=1 =1 =1\\ni\\nci ci ci\\nn12\\nwhere Nci is the number of pixels in the cell ci. The maximum value in \\nvector Mci  determines the cell type assignment. Cell lineages are \\nassigned in rank priority order (Extended Data Fig.\\xa01b).\\nAll code used to perform these analyses is available at https://github.\\ncom/walsh-quail-labs/IMC-Brain.\\nCell–cell pairwise interaction analysis\\nT o identify significant pairwise interaction and avoidance behaviours \\nbetween cell types, we performed permutation tests of single-cell \\ninteractions as previously described11,59. Cells within a 6 pixel radius \\n(6\\u2009μm) were considered interacting. Significant interaction or avoid-\\nance behaviours were defined as having a P-value of less than 0.01.\\nCellular neighbourhood identification\\nT o identify spatial cellular neighbourhoods, we first computed neigh-\\nbour windows, which we defined as being the number (N) of nearest \\ncells to each cell (as indicated), as previously described60. Each win-\\ndow is a frequency vector consisting of the types of N closest cells to a \\ngiven cell. Neighbour windows were clustered. Cellular neighbourhood \\ndiscovery on glioblastoma and BrM-cores combined (performed in \\n2021) was performed using Scikit-learn, a software machine learning \\nlibrary for Python. Clustering was performed using MiniBatchKMeans \\nclustering algorithm version 0.24.2 with default batch size\\u2009=\\u2009100 and \\nrandom_state\\u2009=\\u20090. BrM-margins were excluded from cellular neighbour-\\nhood discovery due to their variable mix of tumour versus stromal \\ncontent. Cellular neighbourhood analysis on glioblastoma cores alone \\n(performed in 2022) used MiniBatchKMeans clustering algorithm ver-\\nsion 1.1.2 with default batch size\\u2009=\\u20091,024 and random_state\\u2009=\\u20090. Every \\ncell was then assigned to a particular cellular neighbourhood based on \\ntheir neighbour window. Cellular neighbourhood prevalence in each \\ncore was normalized so the sum of cellular neighbourhood prevalence \\nfor that core was 100%. Values were then z-scored and cores with z-score \\nabove or equal to 0 and below 0 were compared for survival outcome.\\nIMC survival analysis\\nGlioblastoma survival analysis was conducted using a clinically con-\\ntrolled cohort of patients that received gross total resection of the \\ntumour prior to treatment, as confirmed by post-surgical MRI, and \\nwere treated with SOC (Extended Data Fig.\\xa09f and Supplementary \\nTable\\xa02). Overall survival was calculated from the date of surgery to \\ndate of death. For patients with glioblastoma whose date of death was \\nnot specified, overall survival was estimated using the date of their \\nlast known follow-up. For BrM survival analyses, local recurrence-free \\nsurvival was assessed in previously untreated lesions with complete \\nmacroscopic gross total resections as confirmed by post-surgical MRI. \\nFor all Kaplan–Meier analyses, images were averaged when multiple \\ncores were collected from the same patient’s tumour (that is, each \\npatient had only one survival value represented in the analysis).\\nt-SNE\\nUsing default parameters, t-SNE dimensionality reduction plots were \\ngenerated in MATLAB (version 2019b). Clustering was performed using \\na customized high-dimensional spectral-based clustering algorithm, \\ndue to the curse dimensionality of our cells (order of million number of \\ncells). In our customized algorithm, we first use the DBSCAN to isolate \\nclusters that have a similar density (with fixed parameters of a maximum \\ndistance of 3 pixels minimum number of 30 points per cluster). This \\napproach produces some small and some big clusters with densities \\nthat are similar to each other. The big cluster group is then re-clustered \\nusing a spectral clustering algorithm. T o be able to achieve a spectral \\nclustering result on our massive dimensional data, we do a subsampling \\nof the data (with a subsampling rate of 10), which gives us the overall \\nshape of the data. Next, we assign each cluster with its cluster labels \\nobtained from spectral clustering. Finally, we fit a k-nearest neighbour \\nclassifier (with k\\u2009=\\u20095) to our labelled subsampled data, to identify the \\ncluster labels of all samples. Markers used for t-SNE analysis include \\nCD14, CD16, CD68, CD163, P2Y12, CC3, Ki67, CD40, CD206, HIF1α, \\nMMP9, MPO, OX40L and pSTAT3. For visualization, expression data \\nwere normalized to the 95th percentile.\\nImmunohistofluorescence co-staining\\nFFPE tissue sections were deparaffinized and underwent heat-mediated \\nantigen retrieval in citrate buffer pH 6.0 or EDTA buffer pH 9.0. Slides \\nwere blocked with Power Block for 5\\u2009min at room temperature and \\nincubated with the primary antibody for 30\\u2009min at room tempera -\\nture. Slides were rinsed with TBS-T and subsequently incubated with \\nsecondary antibody–horseradish peroxidase (HRP) for 30\\u2009min at \\nroom temperature. Slides were rinsed with TBS-T and stained with \\nOpal fluorophore working solution for 10\\u2009min (AKOYA Biosciences; \\nOpal 520: FP1487001KT, lot 202212718; Opal 570: FP1488001KT, lot \\n20212821). This was followed by heat-mediated antibody stripping to \\nremove primary and secondary antibodies. These steps were repeated \\nfor each primary antibody for a total of two rounds of labelling: MPO, \\nAbcam, EPR20257, ab208670, lot GR3390666-13, 1:500; IBA1, Fujifilm \\nWako Pure Chemicals, polyclonal, 019-19741, lot 41375175, 1:400; and \\nHorse Anti-Rabbit IgG HRP Polymer Kit, Vector Laboratories, MP-7801, \\nlot ZH0611, 1:1.\\nAntibody specificities and dilutions were optimized individually \\nbefore multiplexing was performed. Tissue images were captured \\nusing the AxioScan Z1 scanner and processed using HALO software \\n(version 3.5).\\nClinical samples for IHF\\nTissue microarrays containing glioblastoma primary tumour sam -\\nples from n\\u2009=\\u2009135 patients were consolidated from McGill University \\n(Quebec, Canada), University of Calgary (Alberta, Canada) and the \\nNetherlands Cancer Institute (NKI, Amsterdam, The Netherlands). \\nAll patients were previously untreated and classified as IDH wild-type ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='23a1badd-b8b5-4b46-af55-dabd51c1faca', embedding=None, metadata={'page_label': '566', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nglioblastoma by a certified neuropathologist following primary  \\nsurgical debulking, and later treated with SOC. All patient information \\nand tissues were obtained after written informed consent and used in \\naccordance with the following ethics oversight.\\nMcGill University cohort. n\\u2009=\\u200970 patients underwent surgical resection \\nbetween 2006–2019; McGill University Health Centre and the Mon-\\ntreal Neurological Institute and Hospital institutional review boards  \\n(NEU-10-066, 2018-4150); a neuropathologist reviewed all cases and \\nprovided the TMA (M.C.G.). These samples represent a subset of our \\noriginal IMC cohort based on tissue availability\\xa0on\\xa0the TMA (an\\xa0inde-\\npendent section was used). Survival <1 year, n\\u2009=\\u200925 patients; survival\\xa0 \\n1–3 years, n\\u2009=\\u200918 patients; survival\\xa0>3 years, n\\u2009=\\u200927 patients.\\nUniversity of Calgary cohort. n\\u2009=\\u200958 patients underwent surgical resec-\\ntion between 2002 and 2020; Health Research Ethics Board of Alberta, \\nCancer Committee (HREBA.CC-16-0762), Clark Smith Tumour Biobank; \\na neuropathologist reviewed all cases and provided the TMA ( J.A.C.). \\nSurvival\\xa0<1 year,\\u2009n\\xa0=\\xa031 patients; survival\\xa01–3 years, n\\u2009=\\u200923 patients;  \\nsurvival\\xa0>3 years, n\\u2009=\\u20094 patients.\\nNKI cohort. n\\u2009=\\u20097 patients; Institutional Review Board of the NKI-AvL and \\nNKI-biobank (CFMPB541); TMA provided by D.B. Survival\\xa0<1 year,\\u2009n\\xa0=\\xa05 \\npatients; survival\\xa01–3 years, n\\u2009=\\u20091 patient; >3 years, n\\u2009=\\u20091 patient.\\nPublicly available RNA-sequencing data\\nSingle-cell RNA-sequencing datasets were downloaded from\\xa0the  \\nfollowing.\\n(1) GSE154795 (ref. 51) (GSE154795_GBM.AllCell.Integrated.Scaled.\\nClusterRes.0.1.rds.gz). Using the Seurat object file GSE154795_GBM.\\nAllCell.Integrated.Scaled.ClusterRes.0.1.rds, a new Seurat object \\nwas created (Seurat 4.1.1), with the RNA assay counts from the subset  \\nof the 14 new patients with glioblastoma and was normalized with \\nthe default parameters of the Seurat function NormalizeData.\\n(2) GSE162631 (ref. 53) (GSE162631_raw_counts_matrix.zip.gz). A Seurat \\nobject was created using Seurat 4.1.1 from the expression matrix \\ncount files with the parameters min.cells = 0 and min.features = 200.  \\nThe counts were normalized with the default parameters of the \\nSeurat function NormalizeData.\\n(3) https://doi.org/10.17605/OSF .IO/4Q32E52. Using the Seurat object \\nfile seurat.obj_MNN_ref.RDS, a new Seurat object was created using \\nSeurat 4.1.1 with the RNA assay counts of the source Seurat object \\nand were normalized with the default parameters of the Seurat \\nfunction NormalizeData.\\nMDMs from each dataset were characterized as CD68high (normal-\\nized expression\\u2009>0) and P2RY12low (normalized expression\\u2009<0.1) and \\nwere isolated for further downstream analysis. MDMs were subdivided \\nby\\xa0MPOhigh (normalized expression\\u2009>0.05) or MPOlow (normalized \\nexpression\\u2009<0.05). For each individual patient, an average expres-\\nsion matrix was generated from the MPOhigh and MPOlow MDMs. The \\nFindMarkers function in Seurat was used to generate a list of differen-\\ntially expressed genes between the MPOhigh and MPOlow MDMs. Pathway \\nenrichment was assessed using Ingenuity Pathway Analysis software \\nv.01–13 (Qiagen). Differentially expressed genes (adjusted P\\u2009<\\u20090.05) \\nwere selected for each dataset and ‘Core Analysis’ was used with all \\ndefault parameters.\\nThe following data were also used. Transcriptomic data from human \\nimmune cells in blood61, accessed via Human Protein Atlas (https://\\nwww.proteinatlas.org/ENSG00000005381-MPO/immune+cell); glio-\\nblastoma RNA-sequencing data from The Cancer Genome Atlas (TCGA \\nFirehose Legacy), accessed via the cBioPortal for cancer genomics \\n(https://www.cbioportal.org); and bulk RNA-sequencing data from \\nsorted immune cells from brain tumours16, accessed via https://\\njoycelab.shinyapps.io/braintime/.\\nStatistics and reproducibility\\nAll image analysis steps were performed in MATLAB (version 2019b) \\nand Python (version 3.7.12). Statistical analyses were performed using \\nGraphPad Prism 9 statistical software. P-values of <0.05 were con-\\nsidered significant and data were expressed as mean\\u2009±\\u2009s.e.m. unless \\nindicated otherwise in the figure legends. Normal distribution was \\nexamined via the Shapiro–Wilk test. Parametric data were analysed \\nby Student’s t-test, one-way or two-way ANOVA. Non-parametric data \\nwere analysed by Mann–Whitney test; for large sample size com-\\nparisons, Student’s t-test was used62. Survival data were analysed by \\nlog-rank (Mantel–Cox) tests, as indicated in the relevant figure legends.  \\nContingency tables were analysed by Fisher’s exact test. Tukey’s test \\nwas used for multiple comparisons. For all analyses related to survival, \\nincluding Kaplan–Meier analysis and the L TS and STS cohort, images \\nwere averaged when multiple cores were collected from the same \\npatient’s tumour to prevent biasing results toward individuals with \\nmore images. All other analyses unrelated to survival (for example, \\npopulation dynamics) were performed using individual images to \\nappropriately capture heterogeneity within the TME. Area analysis \\nof IMC images was performed using ImageJ (version 1.53k). All anti-\\nbody optimization was repeated at least two times by IHF and an addi-\\ntional two times by IMC, using a broad variety of tissues as shown in \\nExtended Data Fig.\\xa03 and Supplementary Fig.\\xa01. Additional representa-\\ntive IMC images (including Fig.\\xa01b, Extended Data Figs.\\xa06a and 10b and  \\nSupplementary Fig.\\xa03) were selected from 389 total images and depict \\nthe statistical changes and/or staining quality as described; similar \\nresults in staining quality were obtained for all samples included in \\nanalysis. All other representative immunostaining (Fig.\\xa04b, Extended \\nData Fig.\\xa010g and Supplementary Fig.\\xa02) was performed on at least five \\nfull tissue samples with similar results.\\nReporting summary\\nFurther information on research design is available in the\\xa0Nature  \\nPortfolio Reporting Summary linked to this article.\\nData availability\\nThe source data supporting findings in this study, including \\nhigh-dimensional TIFF images and clinical information corresponding \\nto IMC, have been deposited at https://doi.org/10.5281/zenodo.7383719. \\nRaw primary imaging data can be obtained from the authors directly \\nupon reasonable request.\\nCode availability\\nAll original code has been deposited at GitHub and can be accessed at: \\nhttps://github.com/walsh-quail-labs/IMC-Brain.\\n \\n54. Dankner, M. et\\xa0al. Invasive growth associated with cold-inducible RNA-binding protein \\nexpression drives recurrence of surgically resected brain metastases. Neuro Oncol. 23, \\n1470–1480 (2021).\\n55. Dankner, M. et\\xa0al. The underlying biology and therapeutic vulnerabilities of leptomeningeal \\nmetastases in adult solid cancers. Cancers 13, 732 (2021).\\n56. Karimi, E. et\\xa0al. Machine learning meets classical computer vision for accurate  \\ncell identification. Preprint at BioRxiv, https://doi.org/10.1101/2022.02.27.482183  \\n(2022).\\n57. Arthur, D. & Vassilvitskii, S. k-means++: The advantages of careful seeding. In Proc. 18th \\nannual ACM-SIAM Symposium on Discrete Algorithms 1027–1035 (Society for Industrial \\nand Applied Mathematics, 2006).\\n58. Peel, D. M. G. Finite Mixture Models (John & Sons, 2000).\\n59. Schapiro, D. et\\xa0al. histoCAT: analysis of cell phenotypes and interactions in multiplex \\nimage cytometry data. Nat. Methods 14, 873–876 (2017).\\n60. Schurch, C. M. et\\xa0al. Coordinated cellular neighborhoods orchestrate antitumoral \\nimmunity at the colorectal cancer invasive front. Cell 182, 1341–1359.e1319 (2020).\\n61. Uhlen, M. et\\xa0al. A genome-wide transcriptomic analysis of protein-coding genes in human \\nblood cells. Science 366, eaax9198 (2019).\\n62. Fagerland, M. W. t-tests, non-parametric tests, and large studies-a paradox of statistical \\npractice? BMC Med. Res. Methodol. 12, 78 (2012).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c67e0610-fd5d-4478-8d63-67d90cebe50a', embedding=None, metadata={'page_label': '567', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Acknowledgements The authors are grateful for technical support from core facilities from \\nthe Rosalind and Morris Goodman Cancer Institute (GCI) and Life Sciences Complex at McGill \\nUniversity, including the Single Cell and Imaging Mass Cytometry Platform (SCIMAP) and \\nHistology core facility (N. Robinson and P. Cruz). The authors acknowledge financial support \\nfrom the McGill Interdisciplinary Initiative in Infection and Immunity, the Quebec Cancer \\nConsortium and the Ministère de l’Économie et de l’Innovation du Québec through the Fonds \\nd’accélération des collaborations en santé. L.A.W., D.F.Q., L.A., K.P. and M.C.G. acknowledge \\nfunding from the Brain Tumour Funders’ Collaborative. L.A.W acknowledges funding from  \\nthe Canadian Institutes of Health Research (CIHR; PJT-162137) and the Canadian Foundation  \\nfor Innovation, and holds a Rosalind and Morris Goodman Chair in Lung Cancer Research. \\nD.F.Q. acknowledges funding from the CIHR (PJT-159742, PJT-178306), Canadian Foundation  \\nfor Innovation, and the Canada Research Chairs Program. P.M.S. acknowledges funding from \\nthe CIHR (PJT-175066) and is a McGill University William Dawson Scholar. I.R.W. acknowledges \\nfunding from the Terry Fox Research Institute (1084), CIHR (PJT-152975) and Canada Research \\nChairs Program. M.P. acknowledges funding from the Réseau de Recherche en Cancer of  \\nthe Fonds de Recherche du Québec-Santé and the Québec Breast Cancer Foundation, and \\nholds the Diane and Sal Guererra Chair in Cancer Genetics. Fellowship support was provided \\nby the McGill University MD/PhD Program (S.M.M.), the GCI\\xa0Charlotte and Leo Karassik \\nFoundation\\xa0Oncology PhD\\xa0fellowship (M.W.Y. and L.J.M.P.), Fonds de Recherche du Québec \\n(M.W.Y. and L.J.M.P.) and the GCI Canderel Graduate Studentship Program.\\nAuthor contributions P.M.S., D.F.Q. and L.A.W. designed the study. D.F.Q., L.A.W., P.M.S., M.W.Y. \\nand S.M.M. wrote the manuscript and coordinated experiments. E.K., M.R., M.W.Y., S.M.M., \\nP.M.S., D.F.Q. and L.A.W. performed all IMC analysis and interpreted data. E.K., M.R. and B.F. \\nwrote all original code used in the study. L.J.M.P., Y.W., I.R.W. and D.F.Q. designed, optimized \\nand/or performed IMC. M.S., S.D. and B.F. performed analysis of publicly available datasets. \\nL.R., I.R.W., D.J.K. and L.A. provided analysis critical to data interpretation. M.P. and D.Z. \\nprovided materials and optimized and performed IHF for validation of IMC. K.P. and M.C.-G. \\nprovided human tissue samples used for IMC. M.D., P.F., S.L., R.A. and M.C.G. assisted with \\ncollection and/or curation of clinical data for the IMC TMAs. For glioblastoma IHF, the \\nfollowing authors provided TMAs, pathology expertise and/or clinical data for patient samples \\nfrom McGill University (K.P., M.C.G. and R.A.), the University of Calgary (J.A.C., D.L.S. and G.R.U.) \\nand the Netherlands Cancer Institute (D.B. and L.A.). All authors reviewed and approved the \\nmanuscript.\\nCompeting interests The authors declare no competing interests.\\nAdditional information\\nSupplementary information The online version contains supplementary material available at \\nhttps://doi.org/10.1038/s41586-022-05680-3.\\nCorrespondence and requests for materials should be addressed to Peter M. Siegel,  \\nDaniela F. Quail or Logan A. Walsh.\\nPeer review information Nature thanks Robert Prins and the other, anonymous, reviewer(s) for \\ntheir contribution to the peer review of this work.\\nReprints and permissions information is available at http://www.nature.com/reprints.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='0263f902-b8e3-4ad7-98d2-75efb4b91ca7', embedding=None, metadata={'page_label': '568', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 1 | Study cohort and cell lineage assignment. a, Summary \\nof patient samples. Patients underwent surgical resection and tumours were \\nclassified by a neuropathologist (M.C.G.). Gliomas were classified as: primary \\n(first surgical resection), residual (second surgery to remove any remaining \\ntumour), recurrent (surgical resection after tumour recurrence), or progression \\nfrom grade II/III (resection of grade IV tumour that progressed from low grade \\nglioma). Glioma samples were extracted from the tumour bulk or the tumour- \\nadjacent normal. Brain metastasis samples were extracted from the tumour \\nbulk (core) or the tumour/brain interface (margin). All clinical information was \\nobtained from surgical and pathological reports. Number of cells acquired \\ncorresponds to the total number of cells segmented across all images.  \\nb, Schematic for cell lineage assignment strategy. Created with BioRender.com.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='400d299a-98ee-4d5f-bf11-93d6471840e6', embedding=None, metadata={'page_label': '569', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 2 | Cellular dynamics in primary & metastatic brain \\ntumours.  a, Cell frequencies as a percentage of all TME cells in glioblastoma \\n(n\\u2009=\\u2009192 images), BrM-cores (n\\u2009=\\u200972 images) and BrM-margins ( n\\u2009=\\u200947 images). \\nData are\\xa0mean ± s.e.m.; two-way ANOVA; significant adjusted P -values are \\nhighlighted. b, Left: number of cells/image for each cell type in glioblastoma \\n(n\\u2009=\\u2009192 images), BrM-cores (n\\u2009=\\u200972 images) and BrM-margins ( n\\u2009=\\u200947 images); \\ndata are\\xa0mean ± s.e.m. Right: absolute number of cells (total across all images) \\nin glioblastoma (n\\u2009=\\u2009192 images), BrM-core (n\\u2009=\\u200972 images) and BrM-margin \\n(n\\u2009=\\u200947 images).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='95c507e9-a557-4294-b55d-3b368a398a5f', embedding=None, metadata={'page_label': '570', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nHLA-DR\\nCD11c\\nTonsil 1\\nMerge100 µm\\nSpleen 1\\nCD4\\nCD68\\n Merge100 µm\\nCo-expression/exclusion Inset\\nTonsil 1\\nCD3\\nCD20\\n100 µmMerge\\nTonsil 1\\nCD20\\nHLA-DR\\n Merge100 µm\\nCo-expression/exclusion Inset Co-expression/exclusion Inset\\nCD68\\nP2Y12\\n Merge100 µm\\nSpleen 1\\nMPO\\nCD68\\nMerge100 µm\\nSpleen 1\\nCD68\\nHLA-DR\\n Merge100 µm\\nTonsil 1\\nCD68\\nCD206\\n Merge100 µm\\nTonsil 2\\nCD68\\nCD163\\nMerge100 µm\\nTonsil 4\\nCD45\\nCD3\\nTonsil 2\\nMerge100 µm\\nCD3\\nCD4 CD8\\n Merge100 µm\\nTonsil 1\\nCD4\\nCD8\\n Merge100 µm\\nTonsil 1\\nFOXP3\\nCD4\\nMerge100 µm\\nTonsil 3\\nCD3\\nPD1\\nMerge100 µm\\nTonsil 4\\nCD45\\nGFAP\\n Merge100 µm\\nBrMM argin A\\nHLA-DR\\nGFAP\\n Merge100 µm\\nBrMM argin A\\nCD20\\nGFAP\\n Merge100 µm\\nBrMM argin A\\nCD117\\nGFAP\\n Merge100 µm\\nBrMM argin A\\nCD16\\nCD14\\n Merge100 µm\\nBrMM argin B\\na\\nNeutrophils\\nB cells\\nNK cells\\nDendritic cells\\nEndothelial cells\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells SOX2+ or SOX9+\\nOther T cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nNon-Cl Mo\\nInt Mo\\nUndefined\\nCell Segmentation\\nCD3 CD20 CD68 MPO CD11c CD4 CD16 CD163 CD31 HLA-DR\\nSpleenTonsil\\n100 µm\\n100 µm\\nb\\nExtended Data Fig. 3 | Representative staining of control tissues and brain \\ntumours.  a, Representative IMC images of expected co-staining patterns in \\ntonsil, spleen or tumour margins. Blue, DNA (191Ir and 193Ir). b , Representative \\nIMC images of major lineage markers (left, centre) and corresponding cell \\nsegmentation and lineage assignment (right). The colour code for the cell \\nsegmentation image is provided (right).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f363a997-c871-4901-aeda-a07269aeae56', embedding=None, metadata={'page_label': '571', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='a\\nEndothelial cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells\\nOther T cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nNon-Cl Mo\\nInt Mo\\nUndefined\\nP 0.001\\nP =0 .01\\nP = 0.05\\nTotal T cells\\nTotal MDM\\nTotal MG\\nTotal macs\\nTotal Mo\\nGliomaBrM-coresBrM-margins\\nIDH wt\\nIDH wt\\nIDH wt\\nIDH wt\\nIDH wt\\nAdj norm\\nIDH mut\\nRecur\\nMGMT unmethMGMT meth\\nSurvival <1 yr Survival >3 yr\\nMale Female\\n<Mean age> Mean age\\nBrM-cores\\nBrM-margins\\nBrM-cores BrM-margins\\nBreast\\nBreast\\nBrM-cores BrM-margins\\nLung\\nMelanoma\\nLungM elanoma\\nNo recur\\nEarly recur\\nNo recurE arly recur\\nLate recur\\nLate recur\\nWithout LMDW ith LMD\\n<Mean age\\nBreast\\nMale Female\\n>Mean age\\nLung\\nBreast Melanoma\\nNo recur\\nNo recur\\nLungM elanoma\\nEarly recur\\nLate recur\\nEarly recurL ate recur\\nMale\\n<Mean age\\nWithout LMDW ith LMD\\nFemale\\n>Mean age\\nA term vs       B term\\nUnpairedPaired\\nb\\nHigh T Cells (>5% of TME)Low T Cells (<5% of TME)\\n0\\n1\\n2\\n3\\n4\\nCD8+ T cells\\nglioblastoma\\nFrequency (% of total cells)\\nP<0.0001\\n0.0\\n0.5\\n1.0\\n1.5\\n2.0\\n5\\n10\\nCD4+ T cells\\nglioblastoma\\nP<0.0001\\n0\\n1\\n2\\n3\\n4\\nCD8-CD4- T cells\\nglioblastoma\\nP<0.0001\\n0.000\\n0.005\\n0.010\\n0.015\\n0.1\\n0.2\\n0.3\\nTreg\\nglioblastoma\\nns    \\n0.00\\n0.05\\n0.10\\n0.2\\n0.4\\n0.6\\nMast cells\\nglioblastoma\\nP=0.0126\\n0\\n1000\\n2000\\n3000\\n4000\\nglioblastoma\\nOverall survival (days)\\nc\\nP=0.0281\\n0.0\\n0.2\\n0.4\\n0.6\\n0.8\\n2\\n4\\nDCs\\nglioblastoma\\nP=0.0042\\nTreg\\nExtended Data Fig. 4 | Single-cell frequencies correlate with clinical \\nsubgroups in patients with brain tumours.  a, Cell density comparisons \\nbetween clinical subgroups of patients, corresponding to Supplementary \\nFig.\\xa08. Within each row, bubble colour indicates the clinical subgroup (A term \\nversus B\\xa0term, right) with higher cell type representation; bubble size indicates \\nP-value. Two-sided student’s t test, unpaired unless indicated otherwise; paired \\nanalyses are from patient-matched samples. b , Cell frequencies as a percentage \\nof total cells per image in glioblastoma tumours with low T\\xa0cells (<5% of TME, \\nn\\u2009=\\u2009144 images) versus high T\\xa0cells (>5% of TME, n\\u2009=\\u200913 images). Data are\\xa0mean ± \\ns.e.m.; two-sided student’s t test. c , Mean overall survival time of glioblastoma \\npatients (for whom this information was available) with low T\\xa0cells (<5% of TME, \\nn\\u2009=\\u2009142 images) versus high T\\xa0cells (>5% of TME, n\\u2009=\\u200913 images). Data are mean ± \\ns.e.m.; two-sided student’s t test.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='96dd0159-85cd-472d-9eb4-21d89b16e05b', embedding=None, metadata={'page_label': '572', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 5 | Cellular interactions in primary versus metastatic \\nbrain tumours.  a, Avoidance scores corresponding to Fig.\\xa0 2c for cancer cell \\ninteractions with non-cancer lineages in glioblastoma ( n\\u2009=\\u2009192 images) and \\nBrM-cores (n\\u2009=\\u200959 images). Data are mean ± s.e.m.; two-sided student’s t test.  \\nb, Statistical significance of cellular interaction/avoidance scores in \\nglioblastoma ( n\\u2009=\\u2009192 images) and BrM-cores (n\\u2009=\\u200959 images) corresponding to \\nFig.\\xa02c. P-values calculated by two-sided student’s t test. Colours indicate \\nsignificantly greater interaction (red) or avoidance (blue) in BrM versus \\nglioblastoma (upper table) or glioblastoma versus BrM (lower table).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='27dbc7a7-6134-4896-a733-0be81b59f536', embedding=None, metadata={'page_label': '573', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='a\\nGlioblastoma\\nDNACD68CD31 Ki67\\nGlioblastoma\\nDNACancerCD31 Ki67\\n100 µm\\n100µm\\nBrM-core\\nDNACancerCD31 Ki67\\nDNACancerCD31 Ki67\\nBrM-margin\\n100 µm\\n100 µm\\n100 µm\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells\\nEndothelial cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nOther T cells\\nNon-Cl Mo\\nInt Mo\\nUndefined\\nb\\nGlioblastoma\\nBrM-margins\\nGrid layout\\nBrM-cores\\nAvoidInteract\\ncd ef hg\\nCancer cells\\nB cells\\nNeutrophils\\nNK cells\\nDendritic cells\\nEndothelial\\nMast cells\\nAstrocytes\\nTc cells\\nTH cells\\nTreg cells\\nOther T cells\\nM1-like MDM\\nM2-like MDM\\nM1-like MG\\nM2-like MG\\nCl Mo\\nNon-Cl Mo\\nInt Mo\\nUndefined\\n+1-1 0 \\nExtended Data Fig. 6 | Single-cell spatial interactions in patients with brain \\ntumours.  a, IMC images of interacting cells in the perivascular niche of \\nglioblastoma and BrM samples, representative of analysis in Fig.\\xa0 2d–g. First \\nrow, perivascular cancer cell proliferation in glioblastoma; second row, \\nperivascular macrophage proliferation in glioblastoma; third row, peritumoural \\nendothelial proliferation in BrM-cores; fourth row, peritumoural endothelial \\nproliferation in BrM-margins. High-magnification insets are provided to the \\nright of each image. b, Heatmap of Spearman’s correlation between cell types \\nin glioblastoma (top rows, n\\u2009=\\u2009192 images), BrM-cores (middle rows, n\\u2009=\\u200959 \\nimages) and BrM-margins (bottom rows, n\\u2009=\\u200940 images). Red, positive \\ncorrelation; blue, negative correlation. c , Frequency of CD40+ MDMs and \\nCD40- MDMs interacting with Th. Data are mean ± s.e.m.; n\\u2009=\\u20091,048 cells across \\n74 images; two-sided student’s t test. d , CD40 in M1-like MDMs interacting with \\n(red) or avoiding (blue) endothelial cells in glioblastoma. Data are mean ± s.e.m.; \\nn\\u2009=\\u2009415 cells across 188 images/group; two-sided student’s t test. e , Ox40L in \\nM2-like MDMs interacting with (red) or avoiding (blue) endothelial cells in \\nglioblastoma. Data are mean ± s.e.m.; n\\u2009=\\u20094,248 cells across 183 images/group; \\nTwo-sided student’s t test. f, Ki67 in endothelial cells interacting with (red) or \\navoiding (blue) Tc cells in BrM-margins. Data are mean ± s.e.m.; n\\u2009=\\u200998 cells \\nacross 28 images; two-sided student’s t test. g, Claudin-5 in endothelial cells \\ninteracting with (red) or avoiding (blue) cancer cells in BrM-cores and -margins. \\nData are mean ± s.e.m.; BrM-cores, n\\u2009=\\u20091,503 cells across 61 images; BrM-margins, \\nn\\u2009=\\u20091507 cells across 45 images; two-sided student’s t test. h , Claudin-5 \\nexpression in endothelial cells interacting with (red) or avoiding (blue) cancer \\ncells in BrM-cores subdivided by extent of peritumoural edema. Data are mean \\n± s.e.m.; none/low edema, n\\u2009=\\u2009369 cells across 13 images; moderate/high \\nedema, n\\u2009=\\u20091,046 cells across 42 images; two-sided student’s t test. All images \\nlacking pairwise interactions of interest were excluded from analysis.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='dc5d6c5c-7fa0-49e3-a9aa-f25a747b1b00', embedding=None, metadata={'page_label': '574', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 7 | Survival associations of spatial cellular \\nneighbourhoods in glioblastoma.  Heatmaps depicting the cellular \\ncomposition of glioblastoma cellular neighbourhoods (CN), with N\\u2009=\\u20093, 5, 10, 20 \\nor 30 nearest neighbours and CN = 9 neighbourhoods. Tables show P-values of \\nsurvival analyses (Log-rank (Mantel-Cox) test) for samples based on median CN \\nfrequency. CNs enriched in M1-like MDM that are significantly associated with \\nsurvival are highlighted in grey. CN frequencies were averaged when multiple \\nsamples corresponded to the same patient.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5a4648b0-7e59-425f-b662-2bfb7594eba9', embedding=None, metadata={'page_label': '575', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 8 | M1-like MDM enriched cellular neighbourhoods are \\nassociated with survival in glioblastoma.  a, Heatmap depicting the cellular \\ncomposition of glioblastoma cellular neighbourhoods (CN), with N\\u2009=\\u200910 nearest \\nneighbours and CN = 30 neighbourhoods (n\\u2009=\\u2009192 images). 6 CNs are enriched \\nwith >3000 M1-like MDM (grey highlight; number of M1-like MDM in each CN is \\nindicated). Of these, 2 (red text) are associated with prolonged survival. Table \\nshows P-values of survival analyses (Log-rank (Mantel-Cox) test) for samples \\nbased on median CN frequency. CN frequencies were averaged when multiple \\nsamples corresponded to the same patient. b, Cell frequency correlation \\ngraphs of neutrophils, M1-like MDM and M1-like MG. Linear regression with 95% \\nconfidence interval is shown.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f05e78ed-997e-4e58-b13d-fbe61b639769', embedding=None, metadata={'page_label': '576', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\na\\ncd eUndefinedC ells\\n3.61% CD45+\\n96.39% CD45-\\nMelanoma\\nBrM-margins\\nLung\\nBrM-margins\\nBreast\\nBrM-margins\\nMelanoma\\nBrM-cores\\nLung\\nBrM-cores\\nBreast\\nBrM-cores\\nCN1\\nCN2\\nCN3\\nCN4\\nCN5\\nCN6\\nCN7\\nCN8\\nCN9\\nCN1\\nCN2\\nCN3\\nCN4\\nCN5\\nCN6\\nCN7\\nCN8\\nCN9\\nExample1g lioblastoma\\n Example2glioblastoma\\n Example3 BrM-core\\n Example4 BrM-coreb\\nsurvival(% )\\ntime(years) CN_high (n=16) CN_low( n=16)\\n0123\\n0\\n25\\n50\\n75\\n100\\nCN1\\nP=0.5599\\n0123\\n0\\n25\\n50\\n75\\n100\\nCN3\\nP=0.4662\\n0123\\n0\\n25\\n50\\n75\\n100\\nCN4\\nP=0.0635\\n0123\\n0\\n25\\n50\\n75\\n100\\nCN5\\nP=0.3600\\n0123\\n0\\n25\\n50\\n75\\n100\\nCN6\\nP=0.1161\\n0123\\n0\\n25\\n50\\n75\\n100\\nCN8\\nP=0.5202\\nf\\ng\\n0 2468\\n0\\n25\\n50\\n75\\n100\\nP = 0.0722\\nCN_high(n=13)\\nCN_low (n=14)\\nlocalrecurrence( %)\\ntime(years)\\nCN5\\nExtended Data Fig. 9 | The spatial cellular neighbourhoods of glioblastoma \\nand brain metastases.  a, Representative Voronoi diagrams of CNs mapped \\nonto glioblastoma IMC images. b , Representative Voronoi diagrams of CNs \\nmapped onto BrM-core IMC images. c , Kaplan-Meier analysis of BrM patients \\n(lung, melanoma and breast primaries) based on median CN5 frequency in \\nBrM-cores. CN frequencies were averaged when multiple samples corresponded \\nto the same patient. Log-rank (Mantel-Cox) test, n\\u2009=\\u200913 patients/group. d, CD45 \\nexpression in undefined cells across glioblastoma and BrM samples ( n\\u2009=\\u2009389 \\nimages). e, Distribution of CNs across BrM samples. BrM-margins: melanoma \\nn\\u2009=\\u20096 images, lung n\\u2009=\\u200922 images, breast n\\u2009=\\u200912 images; BrM-cores: melanoma \\nn\\u2009=\\u200913 images, lung n\\u2009=\\u200929 images, breast n\\u2009=\\u200917 images. For each image, the \\npercentage of cells from each CN was determined, then averaged for each \\ngroup. f, Schematic of the glioblastoma LTS/STS cohort and inclusion/exclusion \\ncriteria, pertaining to Supplementary Table\\xa02. Created with BioRender.com.  \\ng, Kaplan-Meier analysis of glioblastoma LTS/STS cohort based on median CN \\nfrequency. CN frequencies were averaged when multiple samples corresponded \\nto the same patient. Log-rank (Mantel-Cox) test, n\\u2009=\\u200916 patients/group.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='85d2b327-41f0-4788-a8b4-0ca5acd02408', embedding=None, metadata={'page_label': '577', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 10 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d3a5c54f-14a0-495c-bc13-5890919e773c', embedding=None, metadata={'page_label': '578', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 10 | MPO+ macrophages are associated with enhanced \\ncytotoxic functions.  a, Pie chart depicting the distribution of macrophage \\nsubsets within the total pool of MPO+ CD68+ macrophages in glioblastoma \\n(n\\u2009=\\u2009192 images). Percentages reflect the proportion of total MPO+ CD68+ cells. \\nb, Representative IMC images of MPO+ MDMs in two glioblastoma samples.  \\nc, Normalized expression of MPO  in glioblastoma and BrM immune cell subsets \\nvia RNA-seq (data from ref. 16). Data are mean ± s.e.m.; n-values depicted \\nindicate number of patients. d , Normalized expression of MPO  in human blood \\nimmune cell subsets via RNA-seq (data from ref. 61). e, Heatmap displaying the \\ntop 50 differentially expressed genes in MPO+ versus MPO- macrophages that \\nwere common to three independent publicly available glioblastoma scRNA-seq \\ndatasets (data from refs. 51–53). f, Pairwise interaction/avoidance scores using \\ntwo-sided permutation tests on individual images (1,000 permutations each) \\nbetween endothelial cells and MPO+ versus MPO- M1-like MDMs (n\\u2009=\\u2009192 images). \\nColour indicates interaction (red) or avoidance (blue) and circle size reflects the \\nmagnitude of the interaction score. g, Representative immunohistofluorescence \\nimages of MPO (green) and IBA1 (red; macrophage marker) co-localization  \\nin glioblastoma tumours. Examples of MPO+ IBA1+ macrophages are shown  \\nin insets, along with MPO+ IBA1- neutrophils or MPO- IBA1+ macrophages  \\n(n\\u2009=\\u20095 images). h, Pairwise interaction/avoidance scores using two-sided \\npermutation tests on individual images (1,000 permutations each) for MPO+ \\nM1-like MDM with other cell types (n\\u2009=\\u2009192 images); interaction (red), avoidance \\n(blue). i, Kaplan-Meier analysis of glioblastoma LTS/STS cohort based on \\nmedian MPO+CD163-P2Y12-CD68+ macrophage enrichment. Cell frequencies \\nwere averaged when multiple samples corresponded to the same patient.  \\nLog-rank (Mantel-Cox) test, n\\u2009=\\u200916 patients/group. j, Frequency of MPO+ M1-like \\nMDM as a percentage of total cells in the glioblastoma LTS/STS cohort ( n\\u2009=\\u200916 \\npatients/group) and recurrent glioblastoma (n\\u2009=\\u200910 patients). Frequencies were \\naveraged when multiple samples corresponded to the same patient. Data are \\nmedian ± interquartile range; two-sided Mann-Whitney test.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1b7ddeba-365b-4eae-a929-2f62ebe77a0d', embedding=None, metadata={'page_label': '579', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\"1 nature portfolio  |  reporting summaryMarch 2021\\nCorresponding author(s):\\nPeter M. Siegel, Logan A. Walsh, Daniela F. \\nQuail\\nLast updated by author(s): Dec 2, 2022\\nReporting Summary\\nNature Portfolio wishes to improve the reproducibility of the work that we publish. This form provides structure for consistency and transparency \\nin reporting. For further information on Nature Portfolio policies, see our Editorial Policies and the Editorial Policy Checklist.\\nStatistics\\nFor all statistical analyses, confirm that the following items are present in the figure legend, table legend, main text, or Methods section.\\nn/a Confirmed\\nThe exact sample size (n) for each experimental group/condition, given as a discrete number and unit of measurement\\nA statement on whether measurements were taken from distinct samples or whether the same sample was measured repeatedly\\nThe statistical test(s) used AND whether they are one- or two-sided \\nOnly common tests should be described solely by name; describe more complex techniques in the Methods section.\\nA description of all covariates tested\\nA description of any assumptions or corrections, such as tests of normality and adjustment for multiple comparisons\\nA full description of the statistical parameters including central tendency (e.g. means) or other basic estimates (e.g. regression coefficient) \\nAND variation (e.g. standard deviation) or associated estimates of uncertainty (e.g. confidence intervals)\\nFor null hypothesis testing, the test statistic (e.g. F, t, r) with confidence intervals, effect sizes, degrees of freedom and P value noted \\nGive P values as exact values whenever suitable.\\nFor Bayesian analysis, information on the choice of priors and Markov chain Monte Carlo settings\\nFor hierarchical and complex designs, identification of the appropriate level for tests and full reporting of outcomes\\nEstimates of effect sizes (e.g. Cohen's d, Pearson's r), indicating how they were calculated\\nOur web collection on statistics for biologists contains articles on many of the points above.\\nSoftware and code\\nPolicy information about availability of computer code\\nData collection  Data collection for imaging mass cytometry was performed using the Hyperion Imaging System CyTOF Software version 6.7.1014. \\nData analysis Code for IMC cell segmentation was written in Matlab (version 2019b) and Python (version 3.7.12). All code, libraries and algorithms (with \\nversions) are deposited on GitHub with detailed descriptions and can be accessed at:  https://github.com/walsh-quail-labs/IMC-Brain \\nPrism 9.1.0 (GraphPad) and RStudio (version 4.2.0) were used for statistical analyses. Immunohistofluorescent image analysis was performed \\nin HALO (version 3.5). Pathway enrichment analyses were performed with IPA Software (version 01-13). Area analysis of IMC images was \\nperformed using ImageJ (1.53k). scRNA-seq data was analyzed using Seurat (version 4.1.1).\\nFor manuscripts utilizing custom algorithms or software that are central to the research but not yet described in published literature, software must be made available to editors and \\nreviewers. We strongly encourage code deposition in a community repository (e.g. GitHub). See the Nature Portfolio guidelines for submitting code & software for further information.\\nData\\nPolicy information about availability of data\\nAll manuscripts must include a data availability statement. This statement should provide the following information, where applicable: \\n- Accession codes, unique identifiers, or web links for publicly available datasets \\n- A description of any restrictions on data availability \\n- For clinical datasets or third party data, please ensure that the statement adheres to our policy \\n \\nThe source data supporting the findings of this study, including high-dimensional TIFF images, have been deposited at: https://doi.org/10.5281/zenodo.7383719 \\nPublicly available resources used in this study were accessed via: GEO (GSE154795, GSE162631), OSF (doi.org/10.17605/OSF.IO/4Q32E), Human Protein Atlas \", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a7bfff68-e5d8-4f07-bcee-61d1a3ca6784', embedding=None, metadata={'page_label': '580', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='2 nature portfolio  |  reporting summaryMarch 2021\\n(https://www.proteinatlas.org/ENSG00000005381-MPO/immune+cell), The Cancer Genome Atlas (glioblastoma, Firehose Legacy) accessed via cBioPortal (https://\\nwww.cbioportal.org), and https://joycelab.shinyapps.io/braintime/ (data from Klemm et al 2020) \\nField-specific reporting\\nPlease select the one below that is the best fit for your research. If you are not sure, read the appropriate sections before making your selection.\\nLife sciences Behavioural & social sciences  Ecological, evolutionary & environmental sciences\\nFor a reference copy of the document with all sections, see nature.com/documents/nr-reporting-summary-flat.pdf\\nLife sciences study design\\nAll studies must disclose on these points even when the disclosure is negative.\\nSample size 270 samples from 139 primary brain tumor patients and 119 brain metastasis samples from 46 patients were included in our IMC analysis. A \\nsubset of the glioblastoma patients used for IMC (n=70; based on tissue availability) plus an additional 65 new glioblastoma patients were \\nused for IHF validation (n=135 in total for IHF). We included all samples we had access to for analyses.\\nData exclusions Exclusion criteria were pre-established for the balanced STS and LTS  cohort (depicted in Extended Data Fig. 9f): We excluded patients with \\nunknown IDH status; we excluded tumors representing progression from grade II/III, recurrent or residual disease; we excluded patients that \\ndid not receive standard of care, we excluded patients that received a partial (or unknown) resection; and we excluded patients surviving \\nbetween 1-3 years. Patients with brain metastases from primary sources other than melanoma, breast or lung were excluded from cellular \\ninteraction and neighbourhood analyses. \\nReplication All antibody optimization was repeated at least 2 times by IHF and 2 times by IMC. All representative immunostaining was performed on ≥5 \\nfull tissue sections. All cell types and cellular interactions were identified in multiple patient samples, including replicate samples from the \\nsame patient. \\nRandomization The glioma patient cohort selected for this study was enriched in long term survivors (>3 years); the brain metastasis patients were not pre-\\nselected for inclusion in this study based on any clinical or histopathological features. Clinical covariates were controlled for the balanced \\nglioblastoma STS and LTS cohort (Extended Data Fig. 9f and Supplementary Table 2). \\nBlinding All samples that underwent IMC were stained simultaneously; the order of image acquisition was blinded to clinical data.  \\nFor all image analyses, quantitative methods were used to eliminate subjective interpretation of data.\\nReporting for specific materials, systems and methods\\nWe require information from authors about some types of materials, experimental systems and methods used in many studies. Here, indicate whether each material, \\nsystem or method listed is relevant to your study. If you are not sure if a list item applies to your research, read the appropriate section before selecting a response. \\nMaterials & experimental systems\\nn/a Involved in the study\\nAntibodies\\nEukaryotic cell lines\\nPalaeontology and archaeology\\nAnimals and other organisms\\nHuman research participants\\nClinical data\\nDual use research of concern\\nMethods\\nn/a Involved in the study\\nChIP-seq\\nFlow cytometry\\nMRI-based neuroimaging\\nAntibodies\\nAntibodies used Information for all antibodies including clones can be found in Supplementary Table 1. Information for antibodies used for \\nimmunofluorescent staining can be found in the Methods. A complete list is below:  \\n \\nAntibody / Clone /  Dilution / Catalogue Number / Supplier \\n \\nIMC Antibodies:  \\nCD3 / Polyclonal / 1:50 / 3170019D / Fluidigm \\nCD8a  / C8-144 / 1:50 / 3162034D / Fluidigm \\nCD4 / EPR6855 / 1:100 / ab133616 / Abcam \\nFoxP3 / 236A-E7 / 1:50 / ab20034 / Abcam \\nCD20 / H1 / 1:100 / 3161029D / Fluidigm \\nCD94 / EPR21003 / 1:100 / ab235441 / Abcam ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='7743046a-75fc-45e1-baeb-2fdca4b51dd1', embedding=None, metadata={'page_label': '581', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='3 nature portfolio  |  reporting summaryMarch 2021\\nCD68 / KP1 / 1:50 / 3159035D / Fluidigm \\nCD163 / EDHu-1 / 1:200 / 3147021D / Fluidigm \\nP2Y12/ Polyclonal / 1:100 / AS-55043A / Labscoop \\nCD11c / EP1347Y / 1:100 / ab52632 / Abcam \\nHLA-DR / EPR3692 / 1:100 / ab92511 / Abcam \\nCD14 / SP192 / 1:100 / ab183322 / Abcam \\nCD16 / SP175/ 1:100 / ab183354 / Abcam \\nCD117 / YR145 / 1:100 / ab32363 / Abcam \\nMPO / EPR20257 / 1:100 / ab208670 / Abcam \\nPanCK / AE1+AE3 / 1:100 / ab80826 / Abcam \\nPMEL / HMB-45 / 1:100 / NBP2-34638 / Novus Biologicals \\nMelanA / A103 / 1:100 / sc-20032 / Santa Cruz \\nSox2 / EPR3131 / 1:200 / ab215970 / Abcam \\nSox9 / EPR14335-78 / 1:100 / ab185966 / Abcam \\nOlig2 / EPR2673 / 1:200 / ab220796 / Abcam \\nCD31 / JC-70A / 1:100 / ab9498 / Abcam \\nGFAP / EP672Y / 1:400 / ab33922 / Abcam \\nCD45 / D9M8I / 1:100 / 3152018D / Fluidigm \\nKi67 / B56 / 1:100 / ab279657 / Abcam \\nCC3 / 5A1E / 1:100 / 3172027D / Fluidigm \\nClaudin-5 / EPR7583 / 1:100 / ab131259 / Abcam \\nOx40L / EP1168Y / 1:100 / ab76130 / Abcam \\nMMP9 / EP1255Y / 1:100 / ab137867 / Abcam \\nM-CSF-R / SP211 / 1:100 / ab183316 / Abcam \\nGM-CSF-R / 4H1 / 1:50 / 305902 / Biolegend \\nCTLA-4 / SP355 / 1:100 / ab227709 / Abcam \\nHIF1ɑ / EP1215Y / 1:100 / ab51608 / Abcam \\nCD39 / EPR20627 / 1:100 / ab223842 / Abcam \\nCD40 / EPR20540 / 1:50 / ab213205 / Abcam \\npERK / D13.14.4E / 1:100 / 3171010A / Fluidigm \\npSTAT3 / EP2147Y / 1:100 / ab76315 / Abcam \\nCIRBP / EPR18783 / 1:100 / ab238946 / Abcam \\nCD206 / EPR22489-7 / 1:100 / ab254471 / Abcam \\nPD-L1/ E1L3N / 1:50 / 13684T / Cell Signaling Technologies \\nAnti-Biotin / 1D4-C5 / 1:50/ 3150008B / Fluidigm \\nPD1 / D4W2J / 1:50 / 86163 / Cell Signaling Technologies \\n \\nImmunohistofluorescence Antibodies: \\nMPO / EPR20257 / 1:500 / ab208670 / Abcam \\nIBA1 / Polyclonal / 1:400 / 019-19741 / Fujifilm Wako Pure Chemical Corporation \\nHorse Anti-Rabbit IgG HRP Polymer Kit / 1:1 / MP-7801 / Vector Laboratories \\n \\nValidation All antibodies used in this study were individually titrated for each lot, to determine optimal concentrations.  \\n \\nIMC Antibodies: All in-house conjugated antibodies were validated first by immunohistofluorescence (IHF) imaging prior to \\nconjugating to metal isotopes. IHF staining was validated with secondary-only staining and co-staining for expected overlapping \\nmarkers. All conjugated antibodies (both in-house and commercially available) were validated by imaging mass cytometry (IMC) \\nbased on expected expression pattern in several positive/negative control tissues and co-localization/exclusion with other markers. In \\nall cases, staining specificity was assessed in lymphoid control tissue (tonsil), brain tumor (glioblastoma and/or brain metastasis) and \\na positive or negative control tissue (such as spleen, liver, kidney, lymph node, normal brain, normal lung, lung cancer). HIF1ɑ was \\nadditionally validated using samples from patients with von Hippel Lindau syndrome (positive control) and appendicitis (contained \\npositive and negative cells).  \\n \\nMultiplex IHF Antibodies: Antibody specificity was evaluated via expected cell morphology and co-staining with overlapping markers \\n(ex. IBA1 and CD68). Staining was validated with secondary-only staining.  \\nHuman research participants\\nPolicy information about studies involving human research participants\\nPopulation characteristics We collected biobanked tumor samples from patients aged 25 to 96 for IMC. All other information on population \\ncharacteristics and covariates can be found at https://doi.org/10.5281/zenodo.7383719\\nRecruitment We obtained biobanked tumor samples for IMC from primary (glioma) and brain metastasis patients that underwent surgical \\nresection between 2003-2019, and were contacted to give consent for use of their samples in this study. The primary brain \\ntumor cohort was enriched in samples obtained from long-term survivors (> 3 years), therefore, our findings may be biased \\ntowards this group.\\nEthics oversight The patient material used in this study was obtained and used in accordance with the following institutional review boards: \\n1. McGill University cohort: McGill University Health Centre and the Montreal Neurological Institute and Hospital institutional \\nreview boards (NEU-10-066, 2018-4150) \\n2. University of Calgary cohort: Health Research Ethics Board of Alberta, Cancer Committee (HREBA.CC-16-0762) ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='99e2c6f4-a79f-41f8-bd87-42aefb142c7f', embedding=None, metadata={'page_label': '582', 'file_name': 'karimi_brain.pdf', 'file_path': '/content/data/karimi_brain.pdf', 'file_type': 'application/pdf', 'file_size': 28228397, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='4 nature portfolio  |  reporting summaryMarch 2021\\n3. Netherlands Cancer Institute (NKI) cohort: Institutional Review Board of the NKI-AVL and NKI-biobank (CFMPB541)  \\nNote that full information on the approval of the study protocol must also be provided in the manuscript.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='352d1c66-44a5-48c7-a5a3-801777f9eb21', embedding=None, metadata={'page_label': '585', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 585\\nArticle\\nAn atlas of healthy and injured cell states and \\nniches in the human kidney\\nBlue B. Lake1,40,41, Rajasree Menon2,41, Seth Winfree3,41, Qiwen Hu4,41, Ricardo Melo Ferreira5,41, \\nKian Kalhor1,41, Daria Barwinska5, Edgar A. Otto6, Michael Ferkowicz5, Dinh Diep1,40, \\nNongluk Plongthongkum1, Amanda Knoten7, Sarah Urata1, Laura H. Mariani6, Abhijit S. Naik6, \\nSean Eddy6, Bo Zhang7, Yan Wu1,40, Diane Salamon7, James C. Williams5, Xin Wang4, \\nKarol S. Balderrama8, Paul J. Hoover8, Evan Murray8, Jamie L. Marshall8, Teia Noel8, \\nAnitha Vijayan7, Austin Hartman9, Fei Chen8, Sushrut S. Waikar10, Sylvia E. Rosas11,12, \\nFrancis P . Wilson13, Paul M. Palevsky14, Krzysztof Kiryluk15, John R. Sedor16, Robert D. Toto17, \\nChirag R. Parikh18, Eric H. Kim19, Rahul Satija9, Anna Greka8, Evan Z. Macosko8, \\nPeter V . Kharchenko4,40, Joseph P . Gaut20, Jeffrey B. Hodgin21, KPMP Consortium*, \\nMichael T . Eadon5\\u2009✉, Pierre C. Dagher5\\u2009✉, Tarek M. El-Achkar5\\u2009✉, Kun Zhang1,40\\u2009✉, \\nMatthias Kretzler6\\u2009✉ & Sanjay Jain7,20\\u2009✉\\nUnderstanding kidney disease relies on defining the complexity of cell types  \\nand states, their associated molecular profiles and interactions within tissue \\nneighbourhoods1. Here we applied multiple single-cell and single-nucleus assays \\n(>400,000 nuclei\\xa0or\\xa0cells) and spatial imaging technologies to a broad spectrum of \\nhealthy reference kidneys (45 donors) and diseased kidneys (48 patients). This has \\nprovided a high-resolution cellular atlas of 51 main cell types, which include rare  \\nand previously undescribed cell populations. The multi-omic approach provides \\ndetailed transcriptomic profiles, regulatory factors and spatial localizations spanning \\nthe entire kidney. We also define 28 cellular states across nephron segments and \\ninterstitium that were altered in kidney injury, encompassing cycling, adaptive \\n(successful\\xa0or maladaptive repair), transitioning and degenerative states. Molecular \\nsignatures permitted the localization of these states within injury neighbourhoods \\nusing spatial transcriptomics, while large-scale 3D imaging analysis (around 1.2\\u2009million \\nneighbourhoods) provided corresponding linkages to active immune responses. \\nThese analyses defined biological pathways that are relevant to injury time-course \\nand niches, including signatures underlying epithelial repair that predicted \\nmaladaptive states associated with a decline in kidney function. This integrated \\nmultimodal spatial cell atlas of healthy and diseased human kidneys represents a \\ncomprehensive benchmark of cellular states, neighbourhoods, outcome-associated \\nsignatures and publicly available interactive visualizations.\\nThe human kidneys have vital systemic roles in the preservation of \\nbody fluid homeostasis, metabolic waste product removal and blood \\npressure maintenance. After injury, dynamic acute and chronic changes \\noccur in the renal tubules and surrounding interstitial niche. The bal-\\nance between successful or maladaptive repair processes may ulti-\\nmately contribute to the progressive decline in kidney function 2–5. \\nDefining the underlying molecular diversity at a single-cell level is key \\nto understanding progression of acute kidney injury (AKI) to chronic \\nkidney disease (CKD), kidney failure, heart disease or death—issues \\nthat remain a global concern6,7.\\nWe report a multimodal single-cell and spatial atlas with integrated \\ntranscriptomic, epigenomic and imaging data over three major con-\\nsortia: the Human Biomolecular Atlas Program (HuBMAP)8, the Kidney \\nPrecision Medicine Project (KPMP)9 and the Human Cell Atlas (HCA)10. \\nT o ensure robust cell state profiles, healthy reference tissues were \\nobtained from multiple sources, and biopsies were collected from \\npatients with AKI and CKD under rigorous quality assurance and con-\\ntrol procedures8,9,11. We define niches for healthy and altered states \\nacross different regions of the human kidney spanning the cortex to \\nthe papillary tip, and identify gene expression and regulatory mod-\\nules in altered states associated with worsening kidney function. The \\nresultant atlas greatly expands on existing efforts12–15 and will serve as \\nan important resource for investigators and clinicians working towards \\na better understanding of kidney pathophysiology.\\nConstructing a kidney cellular atlas\\nT o fully examine the molecular profile of kidney cell types, we \\nused droplet-based transcriptomic assays (Chromium v3) for sin-\\ngle nuclei (snCv3) and single cells (scCv3) and the multiomic assay \\nhttps://doi.org/10.1038/s41586-023-05769-3\\nReceived: 31 July 2021\\nAccepted: 30 January 2023\\nPublished online: 19 July 2023\\nOpen access\\n Check for updates\\nA list of affiliations appears at the end of the paper.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a0b05499-ee7b-4d20-9e23-6e853c93c561', embedding=None, metadata={'page_label': '586', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='586 | Nature | Vol 619 | 20 July 2023\\nArticle\\nfor single-nucleus chromatin accessibility and mRNA expression \\nsequencing (SNARE-seq2, or SNARE2)16–18 (Supplementary Tables\\xa01–3).  \\nIntegrative transcriptome analyses were performed on more than \\n400,000 high-quality nuclei/cells (Methods) from 58 reference tissues  \\n(35 donors) and 52 diseased tissues (36 patients) that covered the spec-\\ntrum of conditions from healthy to AKI and CKD (Fig.\\xa01, Extended Data \\nFigs.\\xa01–3 and Supplementary Fig.\\xa01). Unsupervised clustering was first \\nperformed on snCv3 data, permitting the discovery of 100 distinct \\ncell populations, which were annotated to 77 subclasses of epithelial, \\nendothelial, stromal, immune and neural cell types (Fig.\\xa02, Methods, \\nExtended Data Figs.\\xa01 and 2 and Supplementary Tables\\xa04 and 5). T o \\nfurther extend cell type annotations across omic platforms, snCv3 data \\nwere used to anchor scCv3 and SNARE2 datasets to the same embedding \\nspace, and cell type labels were assigned through integrative clustering \\n(Methods, Extended Data Fig.\\xa03 and Supplementary Tables\\xa06 and 7). \\nFor spatial localization of these cell types or states in\\xa0situ, we applied \\n3D label-free imaging, multiplex fluorescence imaging (15 individuals)  \\nand spatial transcriptomic Slide-seq2 19,20 (6 individuals, 67 pucks) \\nand Visium assays (22 individuals, 23 samples) (Fig.\\xa01 , Methods and  \\nSupplementary Table\\xa02). T o ensure consistency and agreement of find-\\nings across technologies and minimize procurement- and assay-related \\nbiases, multiple samples were processed with more than one assay \\n(Supplementary Table\\xa03 and Extended Data Fig.\\xa01a). Our approach per-\\nmitted deep and cross-validated molecular profiles for aligned kidney \\ncell types, leveraging the distinct advantages of each technology; for \\nexample, the addition of cytosolic transcripts from scCv3, regulatory \\nelements from SNARE2 accessible chromatin, and in\\xa0situ cell type/state \\nlocalization and interactions from spatial technologies.\\nReference and altered states\\nWe provide a very high level of complexity for all cell types along the \\ndepth of the kidney from the cortex to the papillary tip, in each nephron \\nsegment and the interstitium (Fig.\\xa02a), identifying 51 canonical human \\nkidney cell types with associated biomarkers (Methods and Supplemen-\\ntary Tables\\xa05−8). This includes cell type epigenetic maps, comprising \\nopen chromatin regions and cis-regulatory elements with enriched \\ntranscription-factor-binding motifs (Supplementary Fig.\\xa01 and Sup -\\nplementary Table\\xa09). T o spatially localize cell types within the tissue, \\nsnCv3 subclasses were used to predict identities in Slide-seq and Visium \\ntranscriptomic data at different resolutions (10\\u2009µm and 55\\u2009µm beads, \\nrespectively) (Fig.\\xa02c–g, Methods and Extended Data Fig.\\xa04–5). This \\nenabled us to recapitulate renal corpuscle, tubular, vascular and inter-\\nstitial cell types with proportions, marker profiles and spatial organiza-\\ntions consistent with expected or observed (Visium) histopathology \\n(Extended Data Fig.\\xa05). Proximity enrichment analysis based on the cell \\ntype composition of adjacent Slide-seq beads across 32 cortical and \\n35 medullary tissue pucks (6 participants) delineated region-specific \\ncellular neighbourhoods (Extended Data Fig.\\xa04d,e), including the renal \\ncorpuscle composition of podocytes (PODs), glomerular capillaries \\n(EC-GC), mesangial cells and parietal epithelial cells. These renal cor-\\npuscle neighbourhoods localized adjacent to the juxtaglomerular \\napparatus cells—renin-producing granular (REN) cells and macula \\ndensa cells—and endothelial cells of the afferent/efferent arterioles \\n(EC-AEA) leading into and out of the renal corpuscle (Fig.\\xa02e–f). This \\nneighbourhood analysis further confirmed a distinct vascular smooth \\nmuscle cell (VSMC) population flanking the afferent/efferent arterioles \\n(Extended Data Fig.\\xa04f). Consistent with these annotations, we validated \\nthe appropriate localization of associated cell type markers across \\nplatforms (Fig.\\xa02f and Extended Data Fig.\\xa05d–j). In addition to the renal \\ncorpuscle, we spatially anchored cell type subpopulations to the cortex \\nor medulla (Fig.\\xa02c and Extended Data Fig.\\xa05a). The transition of the \\nascending thin limbs (ATL) of the inner medulla to the medullary thick \\nascending limb (M-TAL) of the outer medullary stripe was observed in \\nSlide-seq (Fig.\\xa02c), along with the transition from descending thin limb \\n(DTL2) and M-TAL in the medulla to the cortical thick ascending limb \\n(C-TAL) in the cortex in Visium (Fig.\\xa02g and Extended Data Fig.\\xa05d). Thus, \\nthe unique strengths of each spatial technology enabled the validation \\nof our omic-defined cell types.\\nA critical and new element of this reference atlas is the characterization \\nof cellular states associated with pathophysiological stress or injury. We \\ncarefully defined these altered states on the basis of previous studies and \\nsnCv3 SNARE2 scCv3\\n200,338 nuclei 112,008 nuclei 109,741 cells\\nImmune\\ncells\\nEndothelial\\ncells\\nStromal\\ncells\\nNeural\\ncellsFIB\\nEpithelial cells\\nPT\\nDCT\\nCNT\\nPC\\nPOD\\nIC\\nPapE\\nDTL ATL\\nTAL\\nVSM/P\\nNuclei\\nSections\\nCells\\nBiopsy\\n(or Nx/DD)\\nHuman kidney\\n(AKI, CKD, reference)\\nTissue\\nprocessing\\nSpatial technologies\\n3D imaging, cytometry and neighbourhood analysis\\nSpatial transcriptomics\\nLabel free\\nCortex Medulla\\nMulti-/f_luoresence\\nIntegrate\\nCv3\\nSNARE2\\nRNA and\\nATAC\\nDAPI F-actin THP AQP1 MPO CD68 CD3\\nSHG AF\\nRNA\\nAssay\\n10x snRNA-seq\\n10x scRNA-seq\\nSNARE2\\n3D imaging\\nSlide-seq2\\nVisium\\nSex Condition\\n19\\n21\\n3\\n6\\n5\\n12\\n13\\n15\\n–\\n9\\n–\\n10\\n13\\n18\\n7\\n–\\n6\\n6\\n10\\n12\\n–\\n6\\n–\\n6\\nCKD   AKI     Ref\\n17\\n24\\n4\\n9\\n1\\n10\\nPool/split\\nPool/split\\nPool\\nATAC + BC1 BC2 BC3\\na\\nb\\nc\\nFig. 1 | Overview of the technologies used to generate a human kidney cell \\natlas. a, Human kidney samplesconsisted of healthy reference, AKI or CKD \\nnephrectomies (Nx), deceased donors (DD) or biopsies. Tissues were processed \\nfor one or more assays, including snCv3, scCv3, SNARE2, 3D imaging or spatial \\ntranscriptomics (Slide-seq2, Visium). Scale bars, 1\\u2009mm (top) and 300\\u2009µm \\n(bottom). b, Summary of the samples. Ref, reference. c, Omic RNA data were \\nintegrated, as shown by joint UMAP embedding, for alignment of cell type \\nannotations across the three different data modalities. IC, intercalated cells; \\nPC, principal cells;\\xa0VSM/P , vascular smooth muscle cell or pericyte.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='46d33607-93e7-4de4-9c40-83cc9d55a3c2', embedding=None, metadata={'page_label': '587', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 587\\nknown features of injury (Methods and Supplementary Table\\xa010). We \\nestablished multiple putative states—namely cycling, transitioning, adap-\\ntive (successful or maladaptive\\xa0repair) and degenerative (damaged or \\nstressed). These altered states were identified for epithelial cells along the \\nnephron, as well as within the stroma and vasculature (Fig.\\xa02a,d). Altered \\nstates, from reference and disease tissues in different proportions, were \\nfound to exist across technologies (Extended Data Figs.\\xa01 and 3) and \\nshowed distinct expression signatures (Supplementary Tables\\xa011–15).\\nWe used several methods to confirm these altered states. Mapping \\nour annotations onto an existing mouse AKI model4 provided insights \\ninto their timecourse after an acute injury event (Extended Data Fig.\\xa06). \\nDegenerative states, coinciding with elevated expression of the known \\ninjury markers SPP1, CST3, CLU and IGFBP721 in humans (Supplementary \\nFig.\\xa02), arose early in mice after injury (Extended Data Fig.\\xa06c–e). These \\nstates showed a common expression and regulatory signature across \\ncell types associated with FOS/JUN signalling (Supplementary Fig.\\xa02) \\nand were largely depleted in recovered mouse kidneys, consistent with \\npossible cell death or a progression into repair states. Putative adap-\\ntive (successful\\xa0or maladaptive tubular\\xa0repair) states were primarily \\nfound within the proximal tubule (PT) and TAL subclasses in mouse \\nand human kidneys. Both adaptive epithelial (aEpi) cell types showed \\nexpression profiles associated with epithelial differentiation, mor-\\nphogenesis, mesenchymal differentiation and EMT, while also exhibit-\\ning a marked downregulation of transporters critical to their normal \\nPEC\\nPOD\\nMC\\nEC-GC\\nREN\\nEC-AEA\\nVSMC\\nPT-S1\\nPT-S2\\nPT-S3\\naPT\\nEC-PTC\\nEC-LYM\\nFIB\\nMyoF\\naFIB\\nC-TAL\\naTAL1\\nMD\\nDCT\\nCNT\\nC-PC\\nC-IC-A\\nIC-B\\nDTL1\\nDTL2\\nEC-DVR\\nVSMC/P\\nM-TAL\\nM-PC\\nM-IC-A\\nEC-AVR\\nDTL3\\nATL\\nIMCD\\nM-FIB\\nOM\\nIM\\nSubclass\\nSlide-seq2\\nATL\\n40\\n80\\n40\\n80\\n40\\n80\\n0\\n25\\n0\\n4\\n0\\n3\\nEC-DVR M-TALSH3GL3 SLC14A1 SLC12A1\\nSLC12A1\\nSLC14A1\\nSH3GL3\\nMean cell type frequency\\nRNA\\nCx\\nOM\\nO/IM\\nIM01\\nM-TAL\\nATL\\nEC-DVR\\nPOD\\n0.75\\n0 0\\n0.5\\nDTL2M-TAL\\n0\\n0.8\\n0\\n0.8\\nC-TAL\\n1\\n4\\nSLC12A1\\nC\\nOM\\nHistology\\nVisium\\ndPTdPT\\nSC/NEUSC/NEU\\nMASTMAST\\ndPODdPODPODPOD\\nPLPL\\nNN\\nNKTNKT\\nTTBBpDCpDC\\ncDCcDC\\ncycMNPcycMNP\\nMDCMDC\\nncMONncMON\\nMAC-M2MAC-M2\\nEC-LYMEC-LYMEC-GCEC-GC\\nEC-AEAEC-AEA\\nEC-DVREC-DVR\\nEC-PTCEC-PTC\\ndEC-PTCdEC-PTC\\ncycECcycEC\\ndECdEC aFIBaFIB\\nVSMCVSMC\\nRENREN\\ndVSMCdVSMC\\nVSMC/PVSMC/P\\nMCMC\\ncycMyoFcycMyoFFIBFIB\\nMyoFMyoF\\ndFIBdFIB\\ndM-FIBdM-FIBM-FIBM-FIB\\nEC-AVREC-AVR\\nCCD-IC-ACCD-IC-A\\nOMCD-IC-AOMCD-IC-A\\nCNT-IC-ACNT-IC-A\\ndC-IC-AdC-IC-A tPC-ICtPC-IC\\nIC-BIC-B\\nIMCDIMCD OMCD-PCOMCD-PC CCD-PCCCD-PC\\ndOMCD-PCdOMCD-PC\\ndIMCDdIMCD cycCNTcycCNT\\nCNT-PCCNT-PC\\nCNTCNTdCNTdCNT\\nPapEPapE\\ncycDCTcycDCT\\ndDCTdDCT DCT1DCT1\\nDCT2DCT2\\ndM-TALdM-TAL\\ndC-TALdC-TALdPT/DTLdPT/DTL\\ncycPTcycPT\\nPECPEC MDMD\\nC-TALC-TAL\\naTAL2aTAL2\\naTAL1aTAL1\\nATLATL\\ndATLdATL\\ndDTL3dDTL3\\nDTL3DTL3\\nDTL1DTL1\\nDTL2DTL2\\naPTaPT\\nPT-S3PT-S3\\nPT-S2PT-S2\\nPT-S1PT-S1\\nM-TALM-TAL\\nImmune cells\\nEndothelial\\ncells\\nStromal\\ncells\\nEpithelial cells\\nCortex\\nOuter \\nmedulla\\nInner\\nmedulla\\nPT-S1\\naPT\\ncycPT\\ndPT\\ndPT/DTL\\nPT-S2\\nPT-S3\\nPT-S3\\nDTL1\\nDTL2\\nDTL3\\nATL\\ndATL\\nM-TAL\\ndM-TAL\\nC-TAL\\nOuter stripe\\nInner stripe\\ndC-TAL\\naTAL\\nMD\\nDCT1\\nDCT2\\ndDCT\\ncycDCT\\nCNT\\ndCNT\\nCNT-IC-A\\ndC-IC-A\\ncycCNTCNT-PC\\nNephron\\nIC-B\\nCCD-PC\\nOMCD-PC\\nOMCD-IC-A\\ndOMCD-PC\\ntPC-IC\\nCCD-IC-A\\nIMCD\\ndIMCD\\nPapE\\nIC-B\\ndDTL3\\nEC-AA\\nEC-EA\\nMD\\nREN\\nVSMC\\ndVSMC\\nPEC\\nEC-GC\\nMC\\nPOD\\ndPOD\\nPT-S1\\nExp. (%)\\n21025 100\\nAve. Exp. Exp. (%) Ave. Exp. Exp. (%) Ave. Exp.\\n2102 1020 60 25 75\\nsnCv3/scCv3 Slide-seq2 Visium\\nNPHS2\\nEMCN\\nPOSTN\\nVCAM1\\nSLC5A12\\nPALMD\\nMYH11\\nREN\\nNOS1\\nPOD\\nEC-GC\\nMC\\nPEC\\nPT-S1/2\\nEC-AEA\\nVSMC\\nREN\\nMD\\nPOD\\nEC-GC\\nMC\\nPEC\\nPT-S1/2\\nEC-AEA\\nVSMC\\nREN\\nMD\\nPOD\\nEC-GC\\nMC\\nPEC\\nPT-S1/2\\nEC-AEA\\nVSMC\\nMD\\nSlide-seq2\\nNOS1EMCN REN\\nMD\\nMD\\nREN\\nPEC\\nPOD\\nEC-GC\\nEC-GC\\nMD\\nREN\\nRC\\n0 601 00 6\\nRC\\nab Altered states\\nRegions\\naEpi\\nRef\\nCortex\\nCortex/Med.\\nMedulla\\nPapilla\\nDegen\\nTrans\\nCyc\\naStr\\nc\\nf\\ne\\nd\\ng\\nFig. 2 | Spatially resolved atlas of molecular cell types. a, Schematic of the \\nhuman nephron showing cell types and states. b , UMAP embedding showing \\ncell types (subclass level 3) for snCv3. Insets: overlays for both regional origin \\nand altered-state status. Cyc, cycling; degen, degenerative; trans, transitioning. \\nSee Supplementary Table 4 for cell type definitions.\\xa0 c, Heat map of Slide-seq \\ncell type frequencies along the corticomedullary axis (three individuals) (left). \\nMiddle, representative tissue puck region showing the transition of ATL to \\nM-TAL segments. Right, corresponding expression of marker genes (scaled). \\nScale bar, 300\\u2009µm. d, Schematic of the renal corpuscle showing resolved cell \\ntypes. e, The Slide-seq puck area indicated in Extended Data Fig.\\xa04c and \\npredicted cell types for renal corpuscles (top). Bottom, mapped expression \\nvalues for corresponding marker genes (scaled). Scale bar, 100\\u2009µm. f, The \\naverage expression values for renal corpuscle cell types for markers shown in e \\nand Extended Data Fig.\\xa04f for all datasets. Ave., average; Exp., expression.  \\ng, Visium data on a healthy reference kidney (cortex, top; medulla, bottom). \\nLeft, haematoxylin and eosin (H&E)-stained tissue. Right, the per-bead \\npredicted transfer scores for cell types or transcript expression values. Scale \\nbar, 300\\u2009µm. Cx, cortex; OM, outer medulla; IM, inner medulla. The black lines \\noutline histologically confirmed medullary rays leading into medulla.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='989dc72b-5f4e-4347-ab81-41ddf986b6a7', embedding=None, metadata={'page_label': '588', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='588 | Nature | Vol 619 | 20 July 2023\\nArticle\\nfunction (Extended Data Fig.\\xa07a–c). The adaptive PT (aPT) population \\nboth mapped to and correlated with failed repair in rodents (Extended \\nData Fig.\\xa02g), with characteristic expressions of VCAM1, DCDC2 and \\nHAVCR14,22 (Extended Data Fig.\\xa07c). Notably, we now identify a similar \\nstate within the TAL (aTAL), marked in humans by PROM1 (encoding \\nCD133) and DCDC2 (Supplementary Table\\xa013). These are consistent with \\nCD133+PAX2+ lineage-restricted progenitors that are known to exist in \\nthe proximal and distal tubules of the adult kidney23,24. Analysis of the \\nmouse AKI data revealed that these originated predominantly from \\nC-TAL, and followed a similar time course as aPT, persisting 6\\u2009weeks \\nafter AKI, consistent with a potential failed-repair population4. This \\nsuggests a common aEpi state, sharing molecular signatures associ-\\nated with injury and repair, that occurs in higher abundance within \\nthe PT and cortical TAL.\\nDistinct altered states were identified within the stroma (aStr) that \\nwere consistent with cell types involved in wound healing and fibrosis \\nafter tissue injury25 (Extended Data Fig.\\xa02i). These cell populations \\nencompass myofibroblasts (MyoF), cycling MyoF (cycMyoF) and a \\ngroup of adaptive fibroblasts (aFIB) representing potential MyoF \\nprogenitors25. Their expression signatures included genes encoding \\nperiostin (POSTN), fibroblast activation protein alpha (FAP), smooth \\nmuscle actin (ACTA2) and collagens (Extended Data Fig.\\xa07d). aStr cells \\nwere enriched after mouse AKI, and they persisted at later timepoints \\n(Extended Data Fig.\\xa06d,e). Furthermore, they exhibited high matri-\\nsome expression25, consistent with their predicted role in extracellular \\nmatrix deposition and fibrosis (Extended Data Fig.\\xa07e). Thus, careful \\nannotation of altered states across kidney cell types has provided a \\nmeans for labelling injury populations. This is important not only for \\ndiseased tissues, but also in reference tissues in which they might arise \\nfrom ischaemic stress during sample acquisition or normal ageing. Key \\noutcomes are the ability to annotate healthy reference cell clusters \\n(Supplementary Fig.\\xa03) as well as providing insights into the pathoge-\\nnetic mechanisms of disease.\\nSpatially mapped injury neighbourhoods\\nFor spatial localization of injury, altered states were mapped to Visium \\ndata generated on a range of healthy reference, AKI and CKD tissues \\n(Supplementary Tables\\xa02 and 3). As expected, altered cell state sig-\\nnatures were enriched in AKI and CKD samples compared with in ref-\\nerence tissues (Fig.\\xa03a,b). On the basis of cell type colocalization in \\nthe relatively larger area of Visium spots, immune and stromal cells \\ncolocalized more frequently with altered epithelial cells (Fig.\\xa03c), con-\\nsistent with increased fibrosis and inflammation around damaged \\ntubules. Furthermore, cell-type-specific altered states in Visium data \\nthat showed expression profiles consistent with snCv3/scCv3 (Fig.\\xa03d) \\nwere directly mapped to histological areas of injury. For example, stro-\\nmal (fibroblast (FIB)), aStr (aFIB) and immune cells (monocyte-derived \\ncells (MDCs)) localized to a region of fibrosis within the cortex of a CKD \\nbiopsy (Fig.\\xa03e,f). This region abutted dilated and atrophic tubules \\nthat showed an aPT signature marked by CDH622 (Extended Data Fig.\\xa07f \\nand Supplementary Table\\xa011). We also found evidence for injury of the \\nmedullary tubules (Extended Data Fig.\\xa07g–i), with an area showing \\nintraluminal cellular cast formation, cell sloughing and loss of nuclei \\nthat were associated with degenerative CD cells, including degenera-\\ntive medullary principal cells (dM-PCs) and transitioning principal and \\nintercalated cells. This region increased expression of the degenerative \\nmarker DEFB1, which was previously shown to contribute to fibrosis \\nthrough immune cell recruitment26. These results support co-mapping \\nof snCv3/scCv3 reference and altered cell types to histological areas \\nof injury.\\nT o further uncover in\\xa0situ cellular niches and injured microenviron-\\nments across kidney disease, we performed 3D multiplexed immuno-\\nfluorescence imaging and label-free cytometry (3DTC) with second \\nharmonic generation for collagen content27 on KPMP AKI and CKD \\nkidney biopsy samples (Extended Data Fig.\\xa08a and Supplementary \\nTables\\xa02 and 3). 3DTC defined cellular niches for 1,540,563 cells by \\nneighbourhood analysis of 14 classes of cells covering renal cortical and \\nmedullary structures (Fig.\\xa04a, Methods and Extended Data Fig.\\xa08b–i). \\nWe identified 14 cellular niches through community detection that \\nincluded expected niches of cortical or medullary epithelium (N7 \\nand N8 versus N14, N9 and N1, respectively; Fig.\\xa04b,c). The TAL and PT \\nneighbourhoods (N7 and N8) were enriched in areas of injury (Fig.\\xa04c \\nand Extended Data Fig.\\xa08i). Furthermore, areas of injury were associ-\\nated with infiltrating leukocytes, including CD68+ (myeloid), MPO+ \\n(N) and CD3+ (lymphoid or T) cells (N6, N11 and N13, respectively). \\nUniquely, CD3+ cells were almost exclusively detected in a subset of \\nneighbourhoods with areas of tissue damage including presumptive \\nepithelial degeneration (loss of markers and simplification) and fibrosis \\n(N13; Fig.\\xa04a (iii) and 4c and Extended Data Fig.\\xa08h), consistent with \\ndegenerative epithelial enrichment found using Visium (Fig.\\xa03c). By \\ncontrast, myeloid cells were found in cellular diverse niches with corti-\\ncal or medullary epithelium (N6 and N11; Fig.\\xa04c). This is consistent with \\nthe association of M2 macrophages (MAC-M2) with adaptive rather \\nthan degenerative epithelia in Visium data (Fig.\\xa03c) and their sustained \\npresence in mouse ischaemia–reperfusion injury (IRI) (Extended Data \\nFig.\\xa06d). The leukocyte diversity was specific in 3D neighbourhoods, \\nas MPO+ and CD3+ cells were overlapping, whereas CD3+ cells were \\nconspicuously low in neighbourhoods with CD68+ cells (N11 versus N6; \\nFig.\\xa04c and Extended Data Fig.\\xa08g). As neutrophils colocalized with puta-\\ntive adaptive and degenerative states (Fig.\\xa03c) and transiently infiltrate \\nearly in mouse IRI (Extended Data Fig.\\xa06d), neutrophils may infiltrate \\nalong with T\\xa0cells predominantly in areas of acute injury marked by \\nmixed degenerative and adaptive states. Alternatively, myeloid cells \\n(such as MAC-M2) may occur more predominantly within relatively \\nhealthy areas showing active repair (adaptive or maladaptive). Overall, \\nthe results from spatial transcriptomics, histological correlation and \\n3DTC demonstrate that altered states were enriched in PT and TAL \\nneighbourhoods, with distinct immune-active cellular niches associ-\\nated with healthy and injured tubules.\\nStages and niches of epithelial repair\\nT o obtain a deeper understanding of the genetic networks underlying \\nthe progression and potential pathology of altered tubular epithe -\\nlium, we performed trajectory inference on the snCv3/SNARE2 and \\nscCv3 subpopulations (Fig.\\xa05a,b, Methods and Extended Data Fig.\\xa09). \\nAlthough most degenerative states appeared too disconnected, aEpi \\ntrajectories showed dynamic gene expression and regulatory transi-\\ntions from dedifferentiated to mature functional states (Supplemen-\\ntary Tables\\xa016–21). We further identified transitory states or modules \\nthat may be associated with either successful or maladaptive repair. \\nEarly repair cells showed expression signatures associated with pro-\\ngenitor states (PROM1), microtubule reorganization (DCDC1) and AKI \\n(HAVCR1, SPP1) (Fig.\\xa05b and Extended Data Fig.\\xa09c,f). The directionality \\nof these repair trajectories was confirmed from RNA velocities esti -\\nmated from dynamical modelling of transcript splicing kinetics, and \\nthe alignment with mouse AKI subpopulations (Fig.\\xa05a and Extended \\nData Fig.\\xa09b,g). These analyses enabled the identification of TAL \\nrepair signatures that were either conserved across species or human  \\nspecific (Fig.\\xa05b).\\nEpithelial repair signalling was enriched for several growth factors \\nand pathways with known roles in promoting normal tubulogenesis, \\nas well as maladaptive repair, fibrosis and inflammation. These include \\nWnt, Notch, TGF-β, EGF , MAPK (FOS/JUN), JAK/STAT and Rho/Rac signal-\\nling28–36 (Fig.\\xa05c, Extended Data Fig.\\xa09d and Supplementary Tables\\xa019–21),  \\nwith dynamic transcription of several pathway regulators mapped to \\nthe TAL repair modules (Extended Data Fig.\\xa09h, i). In support of MAPK \\nsignalling, PT cells that showed expression of PROM1 were subjacent \\nto phosphorylated JUN (p-JUN) (Extended Data Fig.\\xa09e). Progressively ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f7e4bf51-8d52-4da4-9e15-13ea26cbc20c', embedding=None, metadata={'page_label': '589', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 589\\nactive REL/NF-κB signalling along the aTAL and aPT trajectories further \\nexpands on previous roles for this pathway in injured PTs15 (Fig.\\xa05c and \\nSupplementary Table\\xa019). We also found increased cAMP signalling \\n(CREB transcription factors in aPT) capable of promoting dedifferen-\\ntiation37 and increased ELF3 activities that are potentially required for \\nmesenchymal–epithelial transition38, both indicating that adaptive \\nstates may be poised for re-epithelialization.\\nThrough integration of SNARE2 epigenomic profiles with snCv3 tran-\\nscriptomes, detailed gene regulatory networks (GRNs) were inferred \\nfor TAL trajectory modules. Transcription factors with high network \\nimportance were identified in each repair state, confirming key roles \\nfor several major signalling pathways, including their downstream \\ntarget genes and processes (Extended Data Fig.\\xa09j and Supplementary \\nTables\\xa022–24). This highlighted a critical role for TRAP2B (AP-2β), which \\nwas previously found to be required for terminal differentiation of \\ndistal tubule cells through activated expression of KCTD139. Both fac-\\ntors were active or expressed within mid-repair states (Fig.\\xa05c) and \\nsimulated perturbation of TRAP2B disrupted the repair trajectory tran-\\nsition (Extended Data Fig.\\xa09l,m). We therefore find adaptive epithelial \\ntrajectories sharing common molecular profiles that progressively \\nupregulate cytokine signalling involved in tubule regeneration, while \\nalso providing molecular links to pathways associated with fibrosis, \\ninflammation and end-stage kidney disease.\\nSlide-seq, Visium, immunofluorescence staining and RNA in\\xa0situ \\nhybridization (ISH) experiments confirmed spatial localization of adap-\\ntive states into injury niches (Fig.\\xa05d,e and Extended Data Fig.\\xa010). aTAL \\nFibrosis\\nAltered PT\\nT\\nHistology Predicted cell typee\\nf aPT\\n0 0.25\\naFIB T MDC N\\n4\\n2\\n1\\n0\\n3\\nc\\nFIBaFI\\nB\\nMyoF\\ncycMy\\noF\\ndF\\nIB\\nM-FIBdM-FIB\\nPL B T\\nNK\\nT\\nMAST\\nMA\\nC-M2MD\\nC\\nncMON\\ncDCpDC\\nN\\nOdds ratio\\n***\\n******\\n***\\n********\\n***\\n****\\n* **** ***\\n***\\n*\\n** ******\\n******\\n******\\n*****\\nRef\\nDegen\\nAd/Mal\\nState\\nd\\nPODXL\\nLRP2\\nPDZK1\\nCUBN\\nGATM\\nVCAM1\\nPROM1\\nITGB8\\nUMOD\\nSLC12A1\\nDEFB1\\nS100A6\\nTPT1\\nSPP1\\nCD24\\nPROM1\\nITGB6\\nAQP2\\nFXYD4\\nAQP3\\nCALB1\\nATP6V0D2\\nSLC26A7\\nS100A11\\nFTH1\\nFTL\\nAPOE\\nFBLN5\\nPDGFRA\\nCDH11\\nTPT1\\nVIM\\nACTA2\\nCOL1A1\\nCOL1A2\\nABI3BP\\nPRRX1\\nFLRT2\\nIGF1\\nKRT19\\nS100A6\\nCLU\\nMMP7\\nB2M\\nSLC4A1\\nSLC4A9\\nS100A6\\nGATA3\\nCST3\\nAPOE\\nALDOB\\nSPP1\\nS100A6\\nTPM1\\nCLU\\nNPHS2\\nCLIC5\\nSPOCK2\\nB2M\\nS100A6\\nPOD PT\\naPT\\nPOD PT TALC DF IB\\nPC\\nIC\\ndIC\\ndPC\\nFIB\\naSTR\\ndFIB\\nExp. (%)\\ndPOD dPT\\nTAL\\naTAL\\ndTAL\\nIGFBP7\\n0\\n1000\\n1\\nAve. Exp.\\nPOD\\ndATL\\ndM-PC\\ndEC\\nB\\ndPOD\\naTAL1\\nIMCD\\ncycEC\\nPL\\nPEC\\naTAL2\\ndIMCD\\nEC-LYM\\nT\\nPT-S1\\nM-TAL\\nC-IC-A\\nMC\\nNKT\\nPT-S2\\ndM-TAL\\ndC-IC-A\\nREN\\nMAST\\nPT-S3\\nC-TAL\\nM-IC-A\\nVSMC\\nMAC-M2\\naPT\\ndC-TAL\\ntPC-IC\\nVSMC/P\\nMDC\\ncycPT\\nMD\\nIC-B\\ndVSMC\\ncDC\\ndPT\\nDCT\\nPapE\\nMyoF\\nncMON\\ndPT/DTL\\ndDCT\\nEC-GC\\ncycMyoF\\nN\\nDTL2\\ncycDCT\\nEC-AEA\\nFIB\\nSC/NEU\\nDTL1\\nCNT\\nEC-DVR\\nM-FIB\\nDTL3\\ndCNT\\nEC-PTC\\ndM-FIB\\ndDTL3\\nC-PC\\ndEC-PTC\\naFIB\\nATL\\nM-PC\\nEC-AVR\\ndFIB\\nRef\\n0.6\\n*** ***\\n* ***\\n*\\n****\\n***\\n0\\na b\\nHealthy reference CKD\\n0 0.8\\nCKD\\nAKI\\ne\\nCondition\\nCell state\\nDegen Cyc Trans aEpi\\nREF\\naStr\\nMean proportion\\n(all spots)\\nPrediction weight\\nFig. 3 | Transcriptomically defined injury neighbourhoods.  a, The mean \\nproportion of altered-state expression signatures (see Methods, 10x Visium \\nspatial transcriptomics) for all Visium spots (146,460 total spots over 22 \\nindividuals). P values were calculated using Fisher’s exact tests over the spot \\nproportions. b, Feature plots of the aEpi cell state. Scale bar, 300\\u2009µm. The top \\nbounded region is shown in Extended Data Fig.\\xa07h. c, Colocalization of immune \\nand stromal cells with epithelial cell injury states. The y axis shows the odds \\nratio of colocalization (40,326 total spots over 22 individuals). P values were \\ncalculated using Fisher’s exact tests over the colocalization events. Ad/Mal, \\nadaptive/maladaptive\\xa0representing successful or maladaptive tubular repair. \\nd, The average expression values for healthy reference and altered-state \\nmarkers across cell types identified using Visium. e , Histology and predicted \\ncell types in a cortical region (CKD) of interstitial fibrosis and neighbouring PT \\natrophy (altered PT). The pie charts show the proportions of predicted transfer \\nscores for cell type annotations from snCv3 (Fig.\\xa0 2b). The area corresponds to \\nthe bottom bounded region in b . Scale bar, 100\\u2009µm. f, The per-bead predicted \\ntransfer scores for cell types for area shown in e . Scale bar, 100\\u2009µm. *P\\u2009<\\u20090.01, \\n**P\\u2009<\\u20091\\u2009×\\u200910−5, ***P\\u2009<\\u20091\\u2009×\\u200910−10. Exact P values are provided with the Source Data.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b47d8d5f-fd00-40c7-88e4-b0b509697359', embedding=None, metadata={'page_label': '590', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='590 | Nature | Vol 619 | 20 July 2023\\nArticle\\npopulations in Slide-seq-processed tissues (3 niches, 2 individuals; \\nFig.\\xa05d and Extended Data Fig.\\xa011a) were marked by an upregulation \\nof the aTAL marker ITGB6 and downregulated EGF expression, which \\nis known to occur after TAL injury40. These were identified adjacent to \\nareas of aStr enrichment, evidenced by elevated COL1A1 expression. \\nThese potentially fibrotic regions also showed diverse inflammation \\nfor both lymphoid (T\\xa0cell) and myeloid (MAC-M2/MDC) cell types that \\nco-localized around vessels (Fig.\\xa05d). Analogously, aTAL injury niches \\nwere identified in Visium data as spots (55\\u2009µm) colocalizing with stro-\\nmal, lymphoid and myeloid cells (Fig.\\xa05e, Methods and Extended Data \\nFig.\\xa011b–e). Localization of aTAL states to injured tubules was further \\nconfirmed by ISH, in which PROM1-expressing cells showed clear his-\\ntological evidence of injury, including epithelial simplification (thin-\\nning), loss of nuclei and loss of brush border in PTs (Extended Data \\nFig.\\xa010e). Overall, aTAL, aStr and immune expression profiles from \\nspatial transcriptomics were consistent with those identified from \\nsnCV3 and scCv3, providing both validation and spatial co-localization \\nof these cell types and states into niches of ongoing injury and repair.\\nGiven the upregulation of fibrotic cytokine signalling in epithelial \\nrepair, these regenerating cells may represent maladaptive states if \\nthey accumulate or fail to complete tubulogenesis. We therefore inves-\\ntigated the contribution of these states to cell–cell secreted ligand–\\nreceptor interactions within a fibrotic niche (Supplementary Table\\xa025). \\nFrom spatial assays, this niche may comprise aEpi cells adjacent to \\nnormal and altered arteriole cells and fibroblasts, and immune cells that \\ninclude lymphoid and myeloid cells (Figs.\\xa03–5). Using snCv3 and scCv3 \\ndatasets associated with trajectory modules, we identified aTAL repair \\nstates as having a higher number of interactions first with immune \\ncells (early repair), then with the stroma (mid-repair; Fig.\\xa06a,b). This \\nwas associated with secreted growth factors of the FGF , BMP, WNT, \\nEGF , IGF and TGF-β families and the gain of interactions with MAC-M2 \\nand T\\xa0cells (Extended Data Fig.\\xa011f). This indicates that adaptive tubule \\nstates may recruit activated fibroblasts and MyoF both primarily and \\nsecondarily through their recruitment of immune cells.\\nWe also found additional evidence for the activation of EGF pathway \\nsignalling within the adaptive epithelial trajectories, which in itself \\nmay lead to activation of TGF-β signalling and create a niche capable \\nof promoting fibrosis36. Consistently, EGF ligands NRG1 and NRG3 both \\nbecome expressed in aEpi states for a possible role in stromal cells (STR) \\nand MAC-M2 recruitment (Figs.\\xa05d,e and\\xa06c,d). Early and mid-repair \\nTAL states may also recruit or stimulate T\\xa0cells through expression \\nof the CD226-interacting protein NECTIN2 (Fig.\\xa06c,d). Alternatively, \\nBMP6 signalling from mid-repair states may have a role in preventing \\nfibrosis41 through possible SMAD1 activation of fibroblast differentia-\\ntion within aFIB populations (Fig.\\xa06c,d, Extended Data Fig.\\xa011g,h and \\nSupplementary Tables\\xa026–28). BMP6 expression was also detected \\nin repair states of the mouse AKI model at late timepoints when aFIB \\ncells already showed reduced IGF1 expression (Extended Data Fig.\\xa011g). \\nIGF1 secreted from aFIB cells may signal to both stimulate MYOF dif-\\nferentiation42 and promote regeneration of the repairing epithelial \\ncells through IGF1R43 (Fig.\\xa06c,d). Given the timing of BMP6 and IGF1 \\nexpression after acute injury, BMP6-induced differentiation pathways \\nwithin the aFIB cells may represent a late aTAL signal to dampen the \\nfibroblast response. We therefore identify state- and niche-dependent \\nsignalling for reparative states in proximal and distal tubules that may \\nultimately influence the extent of fibrosis and inflammation.\\nAdaptive states can be maladaptive\\nAlthough recruitment of stromal and immune cells is necessary for \\nnormal wound healing, persistent recruitment by aEpi cells may \\nimpair epithelial function or lead to continued release of cytokines \\npromoting disease progression. Consistent with this, we found that \\naEpi gene signatures that were conserved across snCv3 and scCv3 \\n(Supplementary Table\\xa029) were associated with poor renal function \\nin CKD cases (Extended Data Fig.\\xa012a). Thus, successful\\xa0or maladap -\\ntive repair within the TAL may have a role in the transition to chronic \\ndisease. Notably, aTAL signatures underlying early repair states were \\nsignificantly associated with disease progression using unadjusted and \\nsequentially adjusted survival models within the Nephrotic Syndrome \\nStudy Network (NEPTUNE) cohort of 193 patients44 (Fig.\\xa06e, Methods, \\nExtended Data Fig.\\xa012b and Supplementary Table\\xa030). Furthermore, \\ne\\nPearson’s  coef /f_icient  (P < 0.05)\\nMPO\\nCD68\\nC-T AL\\n1.0\\n20\\n0\\n–20\\n–200 20\\nX1\\nX2\\n–40\\n0.8\\n0.6\\n0.4\\n0.2\\n0\\n–0.2\\n–0.4\\n–0.6\\n–0.8\\n–1.0\\nPT\\nM-TA L\\nGl o m\\nCD3\\nAltere d\\nd\\nAlter ed\\nMPO CD68\\nC-T AL\\nPT\\nC-DN M-TA L\\nDTL\\nCD\\nGlom\\nVe ssel\\nVB\\nCD3\\nPairwise connections in neighbourhoods\\n(1,240,569 neighbourhoods)\\nc\\nMPO\\nCD68\\nCD3\\nGlom\\nV essel\\nVB\\nC-TA L\\nM-TA L\\nPT\\nDTL\\nC-DN\\nCD\\nAlter ed\\nFraction of cell classi/f_ications 14131211109876521 43\\nNeighbourhood clusters\\n0\\n0.50\\n1.00\\n0.75\\n0.25\\nb\\nN13\\nN1 N2\\nN3\\nN4\\nN5\\nN6\\nN7N14\\nN12\\nN11\\nN9\\nN8\\nN10\\nNeighbourhood clusters\\n(sampling of 1,240,569 neighbourhoods)\\nCell classi/f_ication\\nGlom\\nCortex Medulla\\n*\\n#\\n#*\\nPT\\nC-T AL\\nDCT/CD\\nDCT/CD\\nDTL CD\\nVB\\nM-TA L\\nV\\nCD\\nT cells\\nAltere d\\nDAPI F-actin THP AQP1 MPO CD68 CD3\\nV\\nV\\nPT\\nGlom\\n(iii)\\n(iii)\\n(ii)\\n(i)\\n(i) (ii)\\nCortex\\na\\nFig. 4 | Defining cellular niches in renal disease from 3D fluorescence \\nimaging.  a, Maximum-intensity projections of representative biopsies (cortex \\nor medulla) showing classification label examples (insets i–iii). Altered, altered \\nmorphology or injury; C-DN, cortical distal nephron; Glom, glomeruli; V, vessels; \\nVB, vascular bundle. Examples of MPO+ and CD68+ are indicated (i). The symbols \\n* and # indicate CD68+ and MPO+ cells, respectively, in (i) and insets. \\xa0Arrowhead \\nindicates T\\xa0cell in (iii) and inset.\\xa0Scale bars, 1\\u2009mm (biopsy images), 100\\u2009µ m  \\n(i and ii) and 5\\u2009µm (insets). b, Community-based clustering on cell composition \\nfor around 20,000 randomly chosen neighbourhoods (15 individuals). The  \\nred outline indicates neighbourhoods including the medulla. c, The cellular \\ncomposition of the neighbourhoods identified in b . d, Pairwise analysis of cells \\nwithin 1.2\\u2009million neighbourhoods (15 individuals); colours are as indicated in c . \\ne, Pearson’s coefficients for select interactions, the colour indicates both the \\nvalue and direction of the correlation. P values were generated using two-sided \\nt-tests.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1f539324-1e7c-44dc-9a4f-bf9c2d241fec', embedding=None, metadata={'page_label': '591', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 591\\nin an independent cohort of 131 patients with kidney disease in the \\nEuropean Renal cDNA Bank (ERCB) cohort, aEpi scores varied by kid-\\nney disease diagnosis relative to living donors45. Specifically, patients \\nwith diabetes, hypertension and focal segmental glomerular sclerosis \\nhad higher aPT and common aPT–aTAL signatures compared with \\nthat of living donors after adjusting for age and sex. In the diabetes \\ngroup, the aPT and common aPT–aTAL signatures remained higher \\nthan that of living donors even after adjusting for age, sex and esti -\\nmated glomerular filtration rate (eGFR; Methods and Supplementary \\nTable\\xa030). Nevertheless, it is important to note that the clinical cor -\\nrelations are based on a small sample size and should therefore be \\ninterpreted with care.\\nThese findings indicate that altered TAL functionality, including \\nits GFR-regulatory role through tubuloglomerular feedback, may \\nrepresent a major contributing factor to progressive kidney failure. \\nFurthermore, causal variants for eGFR and chronic kidney failure \\nwere enriched within TAL regulatory regions that were also enriched \\nfor oestrogen-related receptor (ESSR) transcription-factor motifs \\n(Extended Data Fig.\\xa012c and Supplementary Table\\xa031). ESRR tran -\\nscription factors (especially ESRRB), which are key players in TAL ion \\ntransporter expression46, are central regulators of the TAL expression \\nnetwork (Extended Data Fig.\\xa012d), become inactivated in adaptive \\nstates (Fig.\\xa05c) and, in experimental models, could exacerbate AKI and \\nfibrosis47. Expression quantitative trait loci (eQTL) associated with \\nsnCv3\\nMouse AKI model\\n(ref. 4)\\nscCv3\\naTAL1\\naTAL2\\nM-TAL\\nC-TALRepair\\n(Ad/Mal)\\nM-TAL\\nC-TAL\\naTAL1\\naTAL2\\nM-TAL\\nC-TALaTAL1\\naTAL2\\nTrajectory mapping\\nRNA velocity\\n01\\nLatent time\\nCell type\\nC-TAL\\nRC\\nC-TAL\\nAEA\\naTAL1\\nEGF\\nSLC12A1\\nITGB6\\n0\\n06\\n0 2\\n5\\nITGB6\\nEGF\\nSLC12A1\\nMyoF\\naTAL\\nT\\nMyoF\\naTAL1\\nT\\n20 60\\n20 50\\n25 75\\n0 9\\n0 15\\n09\\nCD14\\nCOL1A1\\nMYH11\\nCD14\\nCOL1A1\\nMYH11\\nSlide-seq\\nCounts\\nPL\\nT\\nNKT\\nMAST\\nMAC-M2\\nMDC\\nncMON\\ncDC\\npDC\\n03 5\\n–0.5 1.0\\n20 60\\n–1\\n10 30\\n1\\nPOD\\nEC-GC\\nVSMC\\nFIB\\naFIB\\nMyoF\\ncycMyoF\\nEC-AEA\\nC-TAL\\naTAL1\\nPL\\nMAC-M2\\nMDC\\nncMON\\ncycMNP\\nNKT\\nT\\ncDC\\nPROM1\\nMACC1\\nARHGAP29\\nMET\\nERBB2\\nTPM1\\nNRP1\\nTSPAN1\\nCST3\\nBICC1\\nITGB6\\nTFAP2B\\nKCTD1\\nTMEM52B\\nEGF\\nESRRG\\nCOL1A1\\nSORBS1\\nCOL1A2\\nACTA2\\nCOL14A1\\nCOL16A1\\nFGF14\\nLIFR\\nFLRT2\\nLAMA2\\nPDGFRA\\nDCN\\nFBLN5\\nNEGR1\\nSLC1A3\\nCD14\\nHLA-DRA\\nCD163\\nMRC1\\nF13A1\\nSTAB1\\nCD96\\nCD247\\nIL7R\\nMZB1\\nIGKC\\nC-TAL\\naTAL1\\nFIB\\naFIB\\nMyoF\\nPL\\nT\\nMAC-M2\\nMDC\\nAve. exp.\\nExp. (%)\\n–0.4 0.4\\n10 30\\nESRRG\\nESRRB\\nESRRA\\nFOS–JUND\\nFOSL1–JUN\\nFOSB–JUNB\\nFOSL2\\nATF2\\nSTAT3\\nSTAT5A–STAT5B\\nSTAT1\\nSMAD2–SMAD3\\nREL\\nRELB\\nNFKB1\\nTCF7L2\\nNR2F2\\nNR3C1\\nTFAP2B\\nETV6\\nZEB1\\nEHF\\nELF3\\nETS1\\nRFX2\\nCEBPD\\nESRRG\\nPPARGC1A\\nESRRB\\nESRRA\\nJUND\\nFOS\\nJUN\\nJUNB\\nFOSL2\\nATF2\\nARHGAP29\\nROCK2\\nRAC1\\nCDC42\\nRHOA\\nRHOB\\nRHOC\\nSTAT3\\nJAK1\\nSTAT5B\\nSTAT1\\nSMAD3\\nSMAD2\\nTGFBR2\\nTGFBR1\\nITGB6\\nITGB8\\nITGAV\\nSKIL\\nNFKBIZ\\nNFKB1\\nRELB\\nREL\\nBICC1\\nDCDC2\\nTCF7L2\\nTNIK\\nPRICKLE1\\nNR2F2\\nNR3C1\\nTFAP2B\\nKCTD1\\nETV6\\nZEB1\\nEHF\\nELF3\\nETS1\\nRFX2\\nCEBPD\\nFGF13\\nFGFR1\\nFGF2\\nEGF\\nERBB4\\nEGFR\\nNRG1\\nERBB2\\nPLSCR1\\nNRG3\\nMET\\nMACC1\\nJAK/STATFGFEGFMETW NT TGF/uni03B2 Rho/Rac JNK ERRMisc. Rel/NF/uni03BAB\\nAve. access.\\nAccess. (%)\\n01 .6\\n8040\\nAve. exp.\\nExp. (%)\\n–1 1\\n7525\\nTF activities\\n(SNARE2)\\nExpression\\n(snCv3/scCv3)\\n–4 4\\nSAMD4A\\nPROM1\\nGGT2\\nDCDC2\\nAC012593.1\\nMACC1\\nITGA3\\nARHGAP26\\nABCC4\\nTMPRSS4\\nSVIL\\nMYO1D\\nMIR4435-2HG\\nFGF13\\nBICD1\\nRBFOX1\\nOGDHL\\nFXYD6-FXYD2\\nCACNB2\\nALDH1L1\\nAC092078.2\\nVAT1L\\nAC078980.1\\nSPP1\\nPLSCR1\\nNRP1\\nITGB6\\nHAVCR1\\nADAMTS1\\nABCC3\\nTPM1\\nSDK1\\nMET\\nATP11A\\nPRKCA\\nITGB8\\nDGKH\\nTMEM52B\\nRALYL\\nGPC5\\nCLCNKA\\nRepair (Ad/Mal)Ref state\\nEarlyMidLateCM\\nRepair (Ad/Mal)Ref state\\nEarlyMidLateCM\\nsnCv3 (TAL)\\nControl\\n4–12 h\\n2 days\\n14 days\\n16 weeks\\nMapped modules\\nIRI groups\\nFraction of cells\\n0\\n1\\naTAL (3)\\nC-TAL (1)\\nVisium\\nCell type\\nRef Altered\\n0\\n–0.4 0.4\\n2\\nAve. exp. Exp. (%)\\n25\\n20 80\\n100\\nNiche compositions Niche expression\\n0 1Proportion\\nMyeloid (11)\\nLymphoid (7)\\nSTR (5)\\naTAL (3)\\nMD (8)\\nC-TAL (1)\\nM-TAL (2)\\naTAL1\\naTAL2\\nM-TAL\\ndM-TAL\\nC-TAL\\ndC-TAL\\nMD\\nMyoF\\ncycMyoF\\nFIB\\nM-FIB\\ndM-FIB\\naFIB\\ndFIB\\nLymphoid\\nMyeloid\\nOther\\nUMOD\\nSLC12A1\\nEGF\\nWNK1\\nREN\\nCALCR\\nNOS1\\nROBO2\\nPAPPA2\\nSPP1\\nPROM1\\nITGB6\\nITGAV\\nITGA3\\nCOL14A1\\nCOL1A2\\nC7\\nAEBP1\\nIGKC\\nCCL5\\nCCL21\\nFOS\\nJCHAIN\\nIGLC2\\nIGHG3\\nCD14\\nCD163\\nF13A1\\nHLA−DQA1\\nIRAK3\\nS100A9\\nTMEM52B\\nEGF\\nTFAP2B\\nKCTD1\\nESRRG\\nITGB6\\nTSPAN1\\nTPM1\\nPROM1\\nBICC1\\nNRP1\\nMET\\nARHGAP29\\nMACC1\\nab c\\nd\\ne\\n2\\n2\\n1\\n1\\nC-TAL\\nRC\\nC-TAL\\nAEA\\naTAL1\\nC-TAL\\nRC\\nC-TAL\\nAEA\\naTAL1\\nAEA\\nFig. 5 | Expression and regulatory signatures of adaptive epithelial cells.  \\na, Trajectory of TAL cells for snCv3, scCv3 and mouse AKI 4 data, showing  \\nmouse to human mapping. Top right, latent time heat map from RNA velocity \\nestimates. Bottom right, bar plot of collection groups after IRI across mouse \\ntrajectory modules. b, Heat map of smoothened gene expression (conserved \\nor human specific) along the inferred TAL pseudotime. State modules based on \\nthe gene expression profiles are shown. M, M-TAL;\\xa0C, C-TAL; Ad/Mal, adaptive/\\nmaladaptive, representing successful or maladaptive tubular repair.\\xa0c, SNARE2 \\naverage accessibilities (access.) (chromVAR) and the proportion accessible for \\ntranscription-factor-binding sites (TFBSs) (right), and the averaged gene \\nexpression values (log scale) and the proportion expressed for integrated \\nsnCv3/scCv3 modules (left). TF , transcription factor. d , Slide-seq fibrotic \\nregions. Top and bottom right, bead locations for a representative region, \\ncoloured by predicted subclasses, prediction weights or scaled gene \\nexpression values. Marker genes are ITGB6 (aTA L), EGF and SLC12A1 (TAL), CD14 \\n(MAC-M2/MDC), MYH11 (VSMC/MyoF) and COL1A1  (aStr). The bar plot shows \\nthe immune subclass counts and the dot plots show the average expression of \\nmarker genes generated from three fibrotic regions (two individuals; Extended \\nData Fig.\\xa011a). Scale bar, 50\\u2009µm. e, Visium TAL niches identified from all Visium \\nspots and defined by colocalized cells (Methods and Extended Data Fig.\\xa011b–e), \\nshowing the proportion of component cell type signatures. The dot plots show \\nthe niche marker gene average expression values.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='934c83b3-5756-44b3-824e-171a2875dd4e', embedding=None, metadata={'page_label': '592', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='592 | Nature | Vol 619 | 20 July 2023\\nArticle\\nkidney function that were previously shown to be enriched primarily \\nin PTs also showed enrichment within the TAL, along with signatures \\nassociated with acute injury and fibrosis in a human AKI to CKD pro-\\ngression study (Extended Data Fig.\\xa012e). Thus, we demonstrate both \\na potential maladaptive role for the aEpi states and a potential central \\nrole for the TAL segment in maintaining the health and homeostasis \\nof the human kidney. This is consistent with the finding that the top \\nrenal genes showing decline in a mouse ageing cell atlas were associ-\\nated with the TAL48.\\nOur findings implicate an accumulation of maladaptive epithelia \\nduring disease progression that may also be consistent with chroni-\\ncally senescent cells 5. This is supported by both increased expres -\\nsion of ageing-related genes, stress-response transcription factor \\nactivities and an apparent senescence-associated secretory phenotype \\n(SASP) for these cells (Extended Data Fig.\\xa012f,g). As such, we detected \\nCDKN1A (also known as p21cip1), CDKN1B (also known as p27kip1), CDKN2A  \\n(also known as p16ink4a) and CCL2 expression in late aPT and aTAL states. \\nFurthermore, expression signatures for reparative processes in aEpi \\nstates were downregulated in the CKD (n\\u2009=\\u200928) over AKI (n\\u2009=\\u200922) cases \\nused in this study (snCv3/scCv3; Supplementary Table\\xa032). This is dis-\\ntinct from the immune response signatures that were more enriched \\nin AKI cases more globally across cell types (Extended Data Fig.\\xa012h \\nand Supplementary Table\\xa033). Overall, our findings are consistent with \\npro-inflammatory repair processes that may persist after injury22, or \\nmay subsequently transition to maladaptive or senescent pro-fibrotic \\nstates during disease progression.\\na\\nc\\nb\\nd\\ne\\nM-TAL\\nC-TAL\\nLate\\nMid\\nEarly\\nRef\\nRepair\\n(Ad/Mal)\\nM-TAL\\nC-TAL\\nLate\\nMid\\nEarly\\nRef\\nRepair\\n(Ad/Mal)\\n0\\n10\\n0\\n10\\nSignalling\\npathways\\nSignalling\\npathways\\nSignalling\\npathways\\nSignalling\\npathways\\n60\\n60\\n06\\n010\\nCell\\ngroups\\nCell\\ngroups\\nCell\\ngroups\\nCell\\ngroups\\n0\\n0\\n15\\n15\\n0 1\\nRelative\\nsignalling\\nstrength\\nTAL incoming\\nTAL outgoing\\nTAL incoming\\nTAL outgoing\\nTAL IMM signalling (snCv3/scCv3) TAL STR signalling (snCv3/scCv3)\\nP = 0.021\\n0.25\\n0.50\\n0.75\\n1.00\\nProbability\\n63 44 30 17 00\\n65 43 32 20 10\\n65 53 37 27 10\\nNumber at risk\\n0 500 1,000 1,500 2,000 2,500\\nP = 0.003\\n0\\n0.25\\n0.50\\n0.75\\n1.00\\n0\\n0.25\\n0.50\\n0.75\\n1.00\\n0\\n62 43 26 18 00\\n66 49 39 17 00\\n65 48 34 29 20\\n0 500 1,000 1,500 2,000 2,500\\nFollow-up times (days)\\nRepair\\nOdds ratio\\nCM\\nGene enrich.\\n0\\n5\\nP = 0.55\\n65 51 34 23 0 0\\n64 44 32 21 10\\n64 45 33 20 1 0\\n0 500 1,000 1,500 2,000 2,500\\nScore tertile\\nLow\\nMedium\\nHigh\\naPT state genes aTAL state genes Degen state genes\\n15\\n0\\nGene enrich.\\nOdds ratio\\nRepair\\nS1–3\\nsnCv3/scSCv3\\nFIB PL\\nT\\nNKC/T\\nMAC-M2\\nMDC\\nncMON\\naFIB\\nMyoF\\nEarlyLate Mid\\nCD226\\nERBB4\\nNRP1\\n–1 01\\n–1 01\\n–1 01\\nSTR\\nTAL\\nIMM\\n20 60\\n25 75\\n10 40\\nStroma\\nTAL\\nIMM\\nC-TAL Repair (Ad/Mal)\\nAve. Exp. Exp. (%)\\nAve. Exp. Exp. (%)\\nAve. Exp. Exp. (%)\\nAve. Exp.\\nExp. (%)\\nAve. Exp.\\nExp. (%)\\nVisium\\nTAL\\nFBLN5\\nFLRT2\\nFGF14\\nIGF1\\nIGFBP6\\nIGFBP5\\nERBB4\\nBMPR2\\nBMPR1A\\nACVR1\\nBMPR1B\\nNRP1\\nSEMA3C\\nSLC12A1\\nEGF\\nIGF1R\\nNECTIN2\\nNRG1\\nNRG3\\nBMP6\\nNRP1\\nSEMA3C\\nSLC12A1\\nEGF\\nIGF1R\\nNECTIN2\\nNRG1\\nBMP6\\nNRP1\\nSEMA3C\\nFBLN5\\nFLRT2\\nIGF1\\nIGFBP6\\nIGFBP5\\nERBB4\\nBMPR2\\nBMPR1A\\nACVR1\\nBMPR1B\\nNRP1\\nSEMA3C\\naTAL(3)\\nFIB\\naStr\\nC-TAL(1)\\n–0.4 0.4\\n25 75\\nSTR\\n–1 1\\n25 75\\nCollagen\\nLaminin\\nSPP1\\nNRG\\nPTPRM\\nAPP\\nCADM\\nEGF\\nEPHA\\nBMP\\nVisfatin\\nSEMA4\\nOCLN\\nCDH1\\nNEGR\\nFGF\\nSEMA3\\nTHBS\\nAGRN\\nGAS\\nHSPG\\nNRXN\\nNectin\\nALCAM\\nCNTN\\nMPZ\\nKIT\\nGRN\\nCXCL\\nCollagen\\nLaminin\\nSPP1\\nPTPRM\\nNRG\\nFN1\\nEGF\\nCADM\\nBMP\\nEPHA\\nNRXN\\nSEMA3\\nOCLN\\nNotch\\nSEMA4\\nVisfatin\\nCD46\\nFGF\\nNPNT\\nCDH1\\nAPP\\nCNTN\\nPDGF\\nHSPG\\nCSPG4\\nMPZ\\nCDH\\nAGRN\\nJAM\\nCD39\\nCollagen\\nLaminin\\nSPP1\\nPTPRM\\nNRG\\nFN1\\nEGF\\nTenascin\\nCADM\\nBMP\\nEPHA\\nTHBS\\nNRXN\\nSEMA3\\nOCLN\\nNotch\\nSEMA4\\nVisfatin\\nCD46\\nFGF\\nNPNT\\nIGF\\nCDH1\\nCNTN\\nPDGF\\nHSPG\\nCSPG4\\nNCAM\\nMPZ\\nCDH\\nTGFb\\nPTN\\nAGRN\\nActivin\\nGRN\\nPeriostin\\nHGF\\nNectin\\nncWNT\\nJAM\\nGAS\\nComplement\\nCD39\\nNT\\nCollagen\\nLaminin\\nSPP1\\nNRG\\nPTPRM\\nAPP\\nCADM\\nEGF\\nEPHA\\nBMP\\nVisfatin\\nSEMA4\\nOCLN\\nCDH1\\nTGFb\\nNEGR\\nFGF\\nSEMA3\\nTHBS\\nAGRN\\nGAS\\nHSPG\\nNRXN\\nNectin\\nCD6\\nCD226\\nCNTN\\nMPZ\\nncWNT\\nGRN\\nFig. 6 | Maladaptive repair signatures. a,b, The ligand–receptor signalling \\nstrength between TAL states and IMM subclasses ( a) or STR subclasses ( b).  \\nThe coloured bars indicate the total signalling strength of the cell group by \\nsummarizing signalling pathways. The grey bars indicate the total signalling \\nstrength of a signalling pathway by summarizing cell groups. Members of key \\nsignaling pathways described in the main text are in bold.\\xa0 c, The average gene \\nexpression values for select ligand–receptor combinations using snCv3/scCv3 \\nintegrated data. d , Dot plots validating select markers shown in c in the Visium \\ndata. e, Unadjusted Kaplan–Meier curves by cell state scores for composite of \\nend-stage renal disease (ESRD) or for 40% drop in eGFR from time of biopsy in \\nthe NEPTUNE adult patient cohort (199 patients; Supplementary Table\\xa030). \\nPatients who reached the end point between screening and biopsy were \\nexcluded. Enrich., enrichment. P values calculated using log-rank tests for \\ntrend are shown (P\\u2009=\\u20090.021 (aPT), P\\u2009=\\u20090.003 (aTAL), P\\u2009=\\u20090.55 (degenerative)).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='bf2e5588-d616-4154-9585-a3d4add12014', embedding=None, metadata={'page_label': '593', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature | Vol 619 | 20 July 2023 | 593\\nDiscussion\\nIn contrast to recent work to broadly integrate major healthy kidney  \\ncell types across disparate data modalities49, here we present a compre-\\nhensive spatially resolved healthy and injured single-cell atlas across \\nthe corticomedullary axis of the kidney. Signals between tubuli, stroma \\nand immune cells that underlie normal and pathological cell neigh-\\nbourhoods were identified, including putative adaptive or maladap-\\ntive repair signatures within the epithelial segments that may reflect a \\nfailure to complete differentiation and tubulogenesis. Spatial analyses \\nidentified that these epithelial repair states have elevated cytokine \\nproduction, increased interactions with the distinct fibrotic and \\ninflammatory cell types, and expression signatures linked to senes-\\ncence and progression to end-stage kidney disease. Failure of these \\ncells to complete tubulogenesis, which might arise from an incompat-\\nible cytokine milieu within the fibrotic niche, in itself might ultimately \\ncontribute to a progressive decline in kidney function. In turn, the \\nhigh-cytokine-producing nature of these cells may further contribute to \\nkidney disease through promotion of fibrosis. We portray a clear role for \\nthe relatively understudied TAL segment of the nephron, a region that \\nis critical for maintaining osmotic gradient and blood pressure through \\ntubuloglomerular feedback. The insights, discoveries and interactive \\ndata visualization tools provided here will serve as key resources for \\nstudies into normal physiology and sex differences, pathways associ-\\nated with transitions from healthy and injury states, clinical outcomes, \\ndisease pathogenesis and targeted interventions.\\nOnline content\\nAny methods, additional references, Nature Portfolio reporting summa-\\nries, source data, extended data, supplementary information, acknowl-\\nedgements, peer review information; details of author contributions \\nand competing interests; and statements of data and code availability \\nare available at https://doi.org/10.1038/s41586-023-05769-3.\\n1. Schreibing, F. & Kramann, R. Mapping the human kidney using single-cell genomics. Nat. \\nRev. Nephrol. 18, 347–360 (2022).\\n2. Park, J. et\\xa0al. Single-cell transcriptomics of the mouse kidney reveals potential cellular \\ntargets of kidney disease. Science 360, 758–763 (2018).\\n3. Sheng, L. & Zhuang, S. New insights into the role and mechanism of partial \\nepithelial-mesenchymal transition in kidney fibrosis. Front. Physiol. 11, 569322 (2020).\\n4. Kirita, Y., Wu, H., Uchimura, K., Wilson, P. C. & Humphreys, B. D. Cell profiling of mouse \\nacute kidney injury reveals conserved cellular responses to injury. Proc. Natl Acad. Sci. USA \\n117, 15874–15883 (2020).\\n5. Docherty, M.-H., O’Sullivan, E. D., Bonventre, J. V. & Ferenbach, D. A. Cellular senescence \\nin the kidney. J. Am. Soc. Nephrol. 30, 726–736 (2019).\\n6. Lameire, N. H. et\\xa0al. Acute kidney injury: an increasing global concern. Lancet 382, 170–179 \\n(2013).\\n7. Zuk, A. & Bonventre, J. V. Acute kidney injury. Annu. Rev. Med. 67, 293–307 (2016).\\n8. HuBMAP Consortium. The human body at cellular resolution: the NIH Human \\nBiomolecular Atlas Program. Nature 574, 187–192 (2019).\\n9. de Boer, I. H. et\\xa0al. Rationale and design of the Kidney Precision Medicine Project. Kidney \\nInt. 99, 498–510 (2021).\\n10. Regev, A. et\\xa0al. The Human Cell Atlas. eLife 6, e27041 (2017).\\n11. El-Achkar, T. M. et\\xa0al. A multimodal and integrated approach to interrogate human kidney \\nbiopsies with rigor and reproducibility: guidelines from the Kidney Precision Medicine \\nProject. Physiol. Genom. 53, 1–11 (2021).\\n12. Lake, B. B. et\\xa0al. A single-nucleus RNA-sequencing pipeline to decipher the molecular \\nanatomy and pathophysiology of human kidneys. Nat. Commun. 10, 2832 (2019).\\n13. Menon, R. et\\xa0al. Single cell transcriptomics identifies focal segmental glomerulosclerosis \\nremission endothelial biomarker. JCI Insight 5, e133267 (2020).\\n14. Stewart, B. J. et\\xa0al. Spatiotemporal immune zonation of the human kidney. Science 365, \\n1461–1466 (2019).\\n15. Muto, Y. et\\xa0al. Single cell transcriptional and chromatin accessibility profiling redefine \\ncellular heterogeneity in the adult human kidney. Nat. Commun. 12, 2190 (2021).\\n16. Chen, S., Lake, B. B. & Zhang, K. High-throughput sequencing of the transcriptome and \\nchromatin accessibility in the same cell. Nat. Biotechnol. 37, 1452–1457 (2019).\\n17. Bakken, T. E. et\\xa0al. Comparative cellular analysis of motor cortex in human, marmoset and \\nmouse. Nature 598, 111–119 (2021).\\n18. Plongthongkum, N., Diep, D., Chen, S., Lake, B. B. & Zhang, K. Scalable dual-omics profiling \\nwith single-nucleus chromatin accessibility and mRNA expression sequencing 2 (SNARE- \\nseq2). Nat. Protoc. 16, 4992–5029 (2021).\\n19. Rodriques, S. G. et\\xa0al. Slide-seq: a scalable technology for measuring genome-wide \\nexpression at high spatial resolution. Science 363, 1463–1467 (2019).\\n20. Stickels, R. R. et\\xa0al. Highly sensitive spatial transcriptomics at near-cellular resolution with \\nSlide-seqV2. Nat. Biotechnol. 39, 313–319 (2021).\\n21. Murray, P. T. et\\xa0al. Potential use of biomarkers in acute kidney injury: report and summary \\nof recommendations from the 10th Acute Dialysis Quality Initiative consensus conference. \\nKidney Int. 85, 513–521 (2014).\\n22. Gerhardt, L. M. S., Liu, J., Koppitch, K., Cippà, P. E. & McMahon, A. P. Single-nuclear \\ntranscriptomics reveals diversity of proximal tubule cell states in a dynamic response to \\nacute kidney injury. Proc. Natl Acad. Sci. USA 118, e2026684118 (2021).\\n23. Bussolati, B. et\\xa0al. Isolation of renal progenitor cells from adult human kidney. Am. J. Pathol. \\n166, 545–555 (2005).\\n24. Cohen-Zontag, O. et\\xa0al. Human kidney clonal proliferation disclose lineage-restricted \\nprecursor characteristics. Sci Rep. 10, 22097 (2020).\\n25. Kuppe, C. et\\xa0al. Decoding myofibroblast origins in human kidney fibrosis. Nature 589, \\n281–286 (2021).\\n26. Andresen, E., Günther, G., Bullwinkel, J., Lange, C. & Heine, H. Increased expression of \\nbeta-defensin 1 (DEFB1) in chronic obstructive pulmonary disease. PLoS ONE 6, e21898 \\n(2011).\\n27. Ferkowicz, M. J. et\\xa0al. Large-scale, three-dimensional tissue cytometry of the human \\nkidney: a complete and accessible pipeline. Lab. Invest. 101, 661–676 (2021).\\n28. Schueler, M. et\\xa0al. DCDC2 mutations cause a renal-hepatic ciliopathy by disrupting Wnt \\nsignaling. Am. J. Hum. Genet. 96, 81–92 (2015).\\n29. Yu, J. et\\xa0al. A Wnt7b-dependent pathway regulates the orientation of epithelial cell division \\nand establishes the cortico-medullary axis of the mammalian kidney. Development 136, \\n161–171 (2009).\\n30. Park, J.-S., Valerius, M. T. & McMahon, A. P. Wnt/beta-catenin signaling regulates nephron \\ninduction during mouse kidney development. Development 134, 2533–2539 (2007).\\n31. Miller, R. K. & McCrea, P. D. Wnt to build a tube: contributions of Wnt signaling to epithelial \\ntubulogenesis. Dev. Dyn. 239, 77–93 (2010).\\n32. Patel, S. et\\xa0al. Rac-GTPase promotes fibrotic TGF-β1 signaling and chronic kidney disease \\nvia EGFR, p53, and Hippo/YAP/TAZ pathways. FASEB J. 33, 9797–9810 (2019).\\n33. Edeling, M., Ragi, G., Huang, S., Pavenstädt, H. & Susztak, K. Developmental signalling \\npathways in renal fibrosis: the roles of Notch, Wnt and Hedgehog. Nat. Rev. Nephrol. 12, \\n426–439 (2016).\\n34. Meecham, A. & Marshall, J. F. The ITGB6 gene: its role in experimental and clinical biology. \\nGene X 5, 100023 (2020).\\n35. Nanjundan, M. et\\xa0al. Plasma membrane phospholipid scramblase 1 promotes EGF- \\ndependent activation of c-Src through the epidermal growth factor receptor. J. Biol. Chem. \\n278, 37413–37418 (2003).\\n36. Harskamp, L. R., Gansevoort, R. T., van Goor, H. & Meijer, E. The epidermal growth factor \\nreceptor pathway in chronic kidney diseases. Nat. Rev. Nephrol. 12, 496–506 (2016).\\n37. Puri, P. et\\xa0al. Ectopic phosphorylated Creb marks dedifferentiated proximal tubules in \\ncystic kidney disease. Am. J. Pathol. 188, 84–94 (2018).\\n38. Sengez, B. et\\xa0al. The transcription factor Elf3 is essential for a successful mesenchymal to \\nepithelial transition. Cells 8, 858 (2019).\\n39. Marneros, A. G. AP-2β/KCTD1 control distal nephron differentiation and protect against \\nrenal fibrosis. Dev. Cell 54, 348–366 (2020).\\n40. Ju, W. et\\xa0al. Tissue transcriptome-driven identification of epidermal growth factor as a \\nchronic kidney disease biomarker. Sci. Transl. Med. 7, 316ra193 (2015).\\n41. Dendooven, A. et\\xa0al. Loss of endogenous bone morphogenetic protein-6 aggravates \\nrenal fibrosis. Am. J. Pathol. 178, 1069–1079 (2011).\\n42. Chetty, A., Cao, G.-J. & Nielsen, H. C. Insulin-like growth factor-I signaling mechanisms, \\ntype I collagen and alpha smooth muscle actin in human fetal lung fibroblasts. Pediatr. \\nRes. 60, 389–394 (2006).\\n43. Wu, Z., Yu, Y., Niu, L., Fei, A. & Pan, S. IGF-1 protects tubular epithelial cells during injury \\nvia activation of ERK/MAPK signaling pathway. Sci. Rep. 6, 28066 (2016).\\n44. Gadegbeku, C. A. et\\xa0al. Design of the Nephrotic Syndrome Study Network (NEPTUNE) to \\nevaluate primary glomerular nephropathy by a multidisciplinary approach. Kidney Int. 83, \\n749–756 (2013).\\n45. Yasuda, Y., Cohen, C. D., Henger, A., Kretzler, M. & European Renal cDNA Bank (ERCB) \\nConsortium. Gene expression profiling analysis in nephrology: towards molecular \\ndefinition of renal disease. Clin. Exp. Nephrol. 10, 91–98 (2006).\\n46. Krid, H., Dorison, A., Salhi, A., Cheval, L. & Crambert, G. Expression profile of nuclear \\nreceptors along male mouse nephron segments reveals a link between ERRβ and thick \\nascending limb function. PLoS ONE 7, e34223 (2012).\\n47. Wang, X. X. et\\xa0al. Estrogen-related receptor agonism reverses mitochondrial dysfunction \\nand inflammation in the aging kidney. Preprint at bioRxiv https://doi.org/10.1101/755801 \\n(2020).\\n48. Tabula Muris Consortium. A single-cell transcriptomic atlas characterizes ageing tissues \\nin the mouse. Nature 583, 590–595 (2020).\\n49. Hansen, J. et\\xa0al. A reference tissue atlas for the human kidney. Sci. Adv. 8, eabn4965 \\n(2022).\\nPublisher’s note Springer Nature remains neutral with regard to jurisdictional claims in \\npublished maps and institutional affiliations.\\nOpen Access This article is licensed under a Creative Commons Attribution \\n4.0 International License, which permits use, sharing, adaptation, distribution \\nand reproduction in any medium or format, as long as you give appropriate \\ncredit to the original author(s) and the source, provide a link to the Creative Commons licence, \\nand indicate if changes were made. The images or other third party material in this article are \\nincluded in the article’s Creative Commons licence, unless indicated otherwise in a credit line \\nto the material. If material is not included in the article’s Creative Commons licence and your \\nintended use is not permitted by statutory regulation or exceeds the permitted use, you will \\nneed to obtain permission directly from the copyright holder. To view a copy of this licence, \\nvisit http://creativecommons.org/licenses/by/4.0/.\\n© The Author(s) 2023', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='96676586-0252-4a62-9969-67db02497b06', embedding=None, metadata={'page_label': '594', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='594 | Nature | Vol 619 | 20 July 2023\\nArticle\\nKPMP Consortium\\nRichard Knight22, Stewart H. Lecker23, Isaac Stillman23, Afolarin A. Amodu10, Titlayo Ilori10, \\nShana Maikhor10, Insa Schmidt10, Gearoid M. McMahon24, Astrid Weins24, Nir Hacohen8, \\nLakeshia Bush16, Agustin Gonzalez-Vicente16, Jonathan Taliercio16, John O’toole16, \\nEmilio Poggio16, Leslie Cooperman16, Stacey Jolly16, Leal Herlitz16, Jane Nguyen16, \\nEllen Palmer16, Dianna Sendrey16, Kassandra Spates-Harden16, Paul Appelbaum15, \\nJonathan M. Barasch15, Andrew S. Bomback15, Vivette D. D’ Agati15, Karla Mehl15, \\nPietro A. Canetta15, Ning Shang15, Olivia Balderes15, Satoru Kudose15, Laura Barisoni25, \\nTheodore Alexandrov26, Yinghua Cheng5, Kenneth W. Dunn5, Katherine J. Kelly5, \\nTimothy A. Sutton5, Yumeng Wen18, Celia P . Corona-Villalobos18, Steven Menez18, \\nAvi Rosenberg27, Mohammed Atta18, Camille Johansen11, Jennifer Sun11, Neil Roy11, \\nMark Williams11, Evren U. Azeloglu28, Cijang He28, Ravi Iyengar28, Jens Hansen28, \\nYuguang Xiong28, Brad Rovin29, Samir Parikh29, Sethu M. Madhavan29, \\nChristopher R. Anderton30, Ljiljana Pasa-Tolic30, Dusan Velickovic30, Olga Troyanskaya31, \\nRachel Sealfon31, Katherine R. Tuttle32, Zoltan G. Laszik33, Garry Nolan34, Minnie Sarwal33, \\nKavya Anjani33, Tara Sigdel33, Heather Ascani35, Ulysses G. J. Balis35, Chrysta Lienczewski35, \\nBecky Steck35, Yougqun He35, Jennifer Schaub35, Victoria M. Blanc35, Raghavan Murugan36, \\nParmjeet Randhawa36, Matthew Rosengart36, Mitchell Tublin36, Tina Vita36, John A. Kellum36, \\nDaniel E. Hall36, Michele M. Elder36, James Winters36, Matthew Gilliam36, Charles E. Alpers37, \\nKristina N. Blank37, Jonas Carson37, Ian H. De Boer37, Ashveena L. Dighe37, \\nJonathan Himmelfarb37, Sean D. Mooney37, Stuart Shankland37, Kayleen Williams37, \\nChristopher Park37, Frederick Dowd37, Robyn L. McClelland37, Stephen Daniel37, \\nAndrew N. Hoofnagle37, Adam Wilcox37, Shweta Bansal38, Kumar Sharma38, \\nManjeri Venkatachalam38, Guanshi Zhang38, Annapurna Pamreddy38, Vijaykumar R. Kakade39, \\nDennis Moledina39, Melissa M. Shaw39, Ugochukwu Ugwuowo39, Tanima Arora39, \\nJoseph Ardayfio39, Jack Bebiak42, Keith Brown42, Catherine E. Campbell42, John Saul42, \\nAnna Shpigel42, Christy Stutzke42, Robert Koewler42, Taneisha Campbell42, Lynda Hayashi42, \\nNichole Jefferson42, Roy Pinkeney42 & Glenda V . Roberts42\\n22American Association of Kidney Patients, Tampa, FL, USA. 23Beth Israel Deaconess Medical \\nCenter, Boston, MA, USA. 24Brigham and Women’s Hospital, Boston, MA, USA. 25Duke \\nUniversity, Durham, NC, USA. 26Structural and Computational Biology Unit, European \\nMolecular Biology Laboratory, Heidelberg, Germany. 27Department of Pathology, Johns \\nHopkins School of Medicine, Baltimore, MD, USA. 28Icahn School of Medicine at Mount Sinai, \\nNew York, NY, USA. 29Ohio State University, Columbus, OH, USA. 30Pacific Northwest National \\nLaboratories, Richland, WA, USA. 31Princeton University, Princeton, NJ, USA. 32Providence \\nHealth, Spokane, WA, USA. 33University of California, San Francisco, CA, USA. 34Stanford \\nUniversity, Stanford, CA, USA. 35University of Michigan, Ann Arbor, MI, USA. 36University of \\nPittsburgh, Pittsburgh, PA, USA. 37University of Washington, Seattle, WA, USA. 38UT Health San \\nAntonio, San Antonio, TX, USA. 39Yale University, New Haven, CT, USA. 42Unaffiliated. \\n1Department of Bioengineering, University of California, San Diego, La Jolla, CA, USA. \\n2Department of Computational Medicine and Bioinformatics, University of Michigan, Ann \\nArbor, MI, USA. 3Department of Pathology and Microbiology, University of Nebraska Medical \\nCenter, Omaha, NE, USA. 4Department of Biomedical Informatics, Harvard Medical School, \\nBoston, MA, USA. 5Department of Medicine, Indiana University School of Medicine, \\nIndianapolis, IN, USA. 6Department of Internal Medicine, Division of Nephrology, University  \\nof Michigan, Ann Arbor, MI, USA. 7Department of Medicine, Washington University School of \\nMedicine, St Louis, MO, USA. 8Broad Institute of Harvard and MIT, Cambridge, MA, USA. 9New \\nYork Genome Center, New York, NY, USA. 10Section of Nephrology, Boston University School \\nof Medicine and Boston Medical Center, Boston, MA, USA. 11Kidney and Hypertension Unit, \\nJoslin Diabetes Center, Boston, MA, USA. 12Harvard Medical School, Boston, MA, USA. \\n13Department of Medicine, Yale University School of Medicine, New Haven, CT, USA. \\n14Department of Medicine, University of Pittsburgh School of Medicine, Pittsburgh, PA, USA. \\n15Department of Medicine, Columbia University, New York, NY, USA. 16Lerner Research and \\nGlickman Urology and Kidney Institutes, Cleveland Clinic, Cleveland, OH, USA. 17Department \\nof Internal Medicine, UT Southwestern Medical Center, Dallas, TX, USA. 18Division of \\nNephrology, Johns Hopkins School of Medicine, Baltimore, MD, USA. 19Department of Surgery, \\nWashington University School of Medicine, St Louis, MO, USA. 20Department of Pathology and \\nImmunology, Washington University School of Medicine, St Louis, MO, USA. 21Department of \\nPathology, University of Michigan, Ann Arbor, MI, USA. 40Present address: San Diego Institute \\nof Science, Altos Labs, San Diego, CA, USA. 41These authors contributed equally: Blue B. Lake, \\nRajasree Menon, Seth Winfree, Qiwen Hu, Ricardo Melo Ferreira, Kian Kalhor. *A list of authors \\nand their affiliations appears at the end of the paper; A full list of members other than authors \\nand their affiliations appears in the Supplementary Information. ✉e-mail: meadon@iupui.edu; \\npdaghe2@iu.edu; telachka@iu.edu; kzhang@bioeng.ucsd.edu; kretzler@med.umich.edu; \\nsanjayjain@wustl.edu', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f8aebeaa-0550-46ce-b48f-1daced837fd1', embedding=None, metadata={'page_label': '595', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Methods\\nStatistics and reproducibility\\nFor 3D imaging and immunofluorescence staining experiments, each \\nstaining was repeated on at least two separate individuals or separate \\nregions. For immunofluorescence validation studies, commercially \\navailable antibodies were used; 13 out of the 15 tissue samples were also \\nanalysed using snCv3 or scCv3. For ISH, 6 tissue samples (4 biopsies \\nand 2 nephrectomies) were analysed. For Slide-seq, 67 tissue pucks \\n(6 individuals) were analysed, with 2 individuals also analysed using \\nsnCv3 or Visium. For Visium, 23 kidney tissue sections (22 individu-\\nals) were imaged, including 6 that were also analysed using snCv3 or \\nscCv3 and one examined using Slide-seq. Orthogonal validation of \\nspatial transcriptomic annotations revealed similar marker gene \\nexpression in snCv3/scCv3 and these technologies, as well as spatial \\nlocalization\\xa0that\\xa0corresponded with histologically validated Visium \\nspot mapping. Although multiomic data from the same samples would \\nbe the most informative, this remains technically challenging. However, \\nwherever possible, several technologies were performed on a subset \\nof samples from the same patient and, in some cases, the same tissue \\nblock was used to generate multimodal data (Extended Data Fig.\\xa01a \\nand Supplementary Table\\xa03). This heterogeneous sampling approach \\nensured cell type discovery while minimizing assay-dependent biases \\nor artifacts encountered when using different sources of kidney tissue. \\nWe recognize that the heterogeneity of sample sources for several \\ntechnologies is a potential limitation due logistics and limited patient \\nbiopsy material.\\nEthical compliance\\nWe have complied with all ethical regulations related to this study. All \\nexperiments on human samples followed all relevant guidelines and \\nregulations. Human samples (Supplementary Table\\xa01) collected as \\npart of the KPMP consortium (https://KPMP.org) were obtained with \\ninformed consent and approved under a protocol by the KPMP single \\nIRB of the University of Washington Institutional Review Board (IRB \\n20190213). Samples as part of the HuBMAP consortium were collected \\nby the Kidney Translational Research Center (KTRC) under a protocol \\napproved by the Washington University Institutional Review Board (IRB \\n201102312). Informed consent was obtained for the use of data and \\nsamples for all participants at Washington University, including living \\npatients undergoing partial or total nephrectomy or rejected kidneys \\nfrom deceased donors. Cortical and papillary biopsy samples from \\npatients with stone disease were obtained with informed consent from \\nIndiana University and approved by the Indiana University Institutional \\nReview Board (IRB 1010002261). For Visium spatial gene expression, \\nreference nephrectomies and kidney biopsy samples were obtained \\nfrom the KPMP under informed consent or the Biopsy Biobank Cohort \\nof Indiana (BBCI)50 under waived consent as approved by the Indiana \\nUniversity Institutional Review Board (IRB 1906572234). Living donor \\nbiopsies as part of the HCA were obtained with informed consent under \\nthe Human Kidney Transplant Transcriptomic Atlas (HKTTA) under the \\nUniversity of Michigan IRB HUM00150968. Deidentified leftover frozen \\nCOVID-19 AKI kidney biopsies were obtained from the Johns Hopkins \\nUniversity pathology archive under waived consent approved by the \\nJohns Hopkins Institutional Review Board (IRB 00090103).\\nSingle-cell and single-nucleus human tissue samples\\nFor single-nucleus omic assays, tissues were processed according \\nto a protocol available online (https://doi.org/10.17504/protocols.\\nio.568g9hw). For nucleus preparation, around 7 sections of 40\\u2009µm thick-\\nness were collected and stored in RNAlater solution (RNA assays) or kept \\non dry ice (accessible chromatin assays) until processing or used fresh. \\nT o confirm tissue composition, 5\\u2009µm sections flanking these thick sec-\\ntions were obtained for histology and the relative amount of cortex or \\nmedulla composition including glomeruli was determined. For single- \\ncell omic assays, tissues used (15 CKD,12 AKI and 18 living donor biopsy \\ncores) were preserved using CryoStor (StemCell T echnologies).\\nSingle-cell, single-nucleus and SNARE2 RNA-seq, quality control \\nand clustering\\nIsolation of single nuclei. Nuclei were isolated from cryosec -\\ntioned tissues according to a protocol available online (https://\\ndoi.org/10.17504/protocols.io.ufketkw ) with the exception that \\n4′,6-diamidino-2-phenylindole (DAPI) was excluded from the nuclear \\nextraction buffer and used only to stain a subset of nuclei used for \\ncounting. Nuclei were used directly for omic assays.\\nIsolation of single cells. Single cells were isolated from frozen tissues \\naccording to a protocol available online (https://doi.org/10.17504/\\nprotocols.io.7dthi6n). The single-cell suspension was immediately \\ntransferred to the University of Michigan Advanced Genomics Core \\nfacility for further processing.\\n10x Chromium v3 RNA-seq analysis. 10x single-nucleus RNA-seq \\nand 10x single-cell RNA-seq were performed according to protocols \\navailable online (https://doi.org/10.17504/protocols.io.86khzcw and \\nhttps://doi.org/10.17504/protocols.io.7dthi6n, respectively), both \\nusing the 10x Chromium Single-Cell 3′ Reagent Kit v3. Sample demulti-\\nplexing, barcode processing and gene expression quantifications were \\nperformed using the 10x Cell Ranger (v.3) pipeline using the GRCh38 \\n(hg38) reference genome with the exception of a subset of scCv3  \\nexperiments that used hg19 (indicated in Supplementary Table\\xa01). For \\nsingle-nucleus data, introns were included in the expression estimates.\\nSNARE2 dual RNA and ATAC-seq analysis. SNARE-seq217, as outlined \\npreviously18, was performed according to a protocol available online \\n(https://doi.org/10.17504/protocols.io.be5gjg3w). Accessible chroma-\\ntin and RNA libraries were sequenced separately on the NovaSeq 6000 \\n(Illumina) system (NovaSeq Control Software v.1.6.0 and v.1.7.0) using \\nthe 300 cycle and 200 cycle reagent kits, respectively.\\nSNARE2 data processing. Detailed step-by-step processing for \\nSNARE2 data has been outlined previously18. This has now been devel-\\noped as an automated data processing pipeline that is available at \\nGitHub (https://github.com/huqiwen0313/snarePip). snarePip (v.1.0.1) \\nwas used to process all the SNARE2 datasets. The pipeline provides \\nan automated framework for complex single-cell analysis, including \\nquality assessment, doublet removal, cell clustering and identification, \\nrobust peak generation and differential accessible region identification, \\nwith flexible analysis modules and generation of summary reports for \\nboth quality assessment and downstream analysis. The directed acyclic \\ngraph was used to incorporate the entirety of the data-processing steps \\nfor better error control and reproducibility. For RNA processing, this \\ninvolved removal of accessible chromatin contaminating reads using \\ncutadapt (v.3.1)51, dropEst (v.0.8.6)52 to extract cell barcodes and STAR \\n(version 2.5.2b)53 to align tagged reads to the genome (GRCh38). For  \\naccessible chromatin data, this involved snaptools (v.1.2.3)54 and mini-\\nmap (v.2-2.20)55 for alignment to the genome (GRCh38).\\nQuality control of sequencing data. 10x snRNA-seq (snCv3). Cell \\nbarcodes passing 10x Cell Ranger filters were used for downstream \\nanalyses. Mitochondrial transcripts (MT-*) were removed, doublets \\nwere identified using the DoubletDetection software (v.2.4.0)56 and \\nremoved. All of the samples were combined across experiments and \\ncell barcodes with greater than 400 and less than 7,500 genes detected \\nwere retained for downstream analyses. T o further remove low-quality \\ndatasets, a gene UMI ratio filter (gene.vs.molecule.cell.filter) was ap-\\nplied using Pagoda2 (https://github.com/hms-dbmi/pagoda2).\\n10x scRNA-seq (scCv3). As a quality-control step, a cut-off of <50% mito-\\nchondrial reads per cell was applied. The ambient mRNA contamination ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='05556620-31fe-48bd-97cb-9468bf99bcb4', embedding=None, metadata={'page_label': '596', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nwas corrected using SoupX (v.1.5.0)57. The mRNA content and the num-\\nber of genes for doublets are comparatively higher than for single cells. \\nT o reduce doublets or multiplets from the analysis, we used a cut-off \\nof >500 and <5,000 genes per cell.\\nSNARE2 RNA. Cell barcodes for each sample were retained with the fol-\\nlowing criteria: having an DropEst cell score of greater than 0.9; having \\ngreater than 200\\u2009UMI detected; having greater than 200 and less than \\n7,500 genes detected. Doublets identified by both DoubletDetection \\n(v.3.0) and Scrublet (https://github.com/swolock/scrublet; v.0.2.2) \\nwere removed. T o further remove low-quality datasets, a gene UMI \\nratio filter (gene.vs.molecule.cell.filter) was applied using Pagoda2.\\nSNARE2 A T AC. Cell barcodes for each sample that had already passed \\nquality filtering from RNA data were further retained with the follow-\\ning criteria: having transcriptional start site (TSS) enrichment greater \\nthan 0.15; having at least 1,000 read fragments and at least 500\\u2009UMI; \\nhaving fragments overlapping the promoter region ratio of greater \\nthan 0.15. Samples were retained only if they exhibited greater than \\n500 dual omic cells after quality filtering.\\nClustering snCv3. Clustering analysis was performed using Pagoda2, \\nwhereby counts were normalized to the total number per nucleus, \\nbatch variations were corrected by scaling expression of each gene \\nto the dataset-wide average. After variance normalization, all 5,526 \\nsignificantly variant genes were used for principal component analysis \\n(PCA). Clustering was performed at different k values (50, 100, 200, \\n500) on the basis of the top 50 principal components, with cluster  \\nidentities determined using the infomap community detection  \\nalgorithm. The primary cluster resolution (k\\u2009=\\u2009100) was chosen on the \\nbasis of the extent of clustering observed. Principal components and \\ncluster annotations were then imported into Seurat (v.4.0.0) and uni-\\nform manifold approximation and projection (UMAP) dimensional-\\nity reduction was performed using the top 50 principal components \\nidentified using Pagoda2. Subsequent analyses were then performed  \\nin Seurat. A cluster decision tree was implemented to determine  \\nwhether a cluster should be merged, split further or labelled as an alt-\\nered state. For this, differentially expressed genes between clusters \\nwere identified for each resolution using the FindAllMarkers function \\nin Seurat (only.pos\\u2009=\\u2009TRUE, max.cells.per.ident = 1000, logfc.thresh-\\nold\\u2009=\\u20090.25, min.pct\\u2009=\\u20090.25). Possible altered states were initially defined  \\nfor clusters with one or more of the following features: low genes  \\ndetected, a high number of mitochondrial transcripts, a high number of  \\nendoplasmic-reticulum-associated transcripts, upregulation of injury \\nmarkers (CST3, IGFBP7, CLU, FABP1, HAVCR1, TIMP2, LCN2) or enrich-\\nment in AKI or CKD samples. Clusters (k\\u2009=\\u2009100) that showed no distinct \\nmarkers were assessed for altered-state features; if present, then these \\nclusters were tagged as possible altered states, if absent then clusters \\nwere merged on the basis of their cluster resolution at k\\u2009=\\u2009200 or 500. \\nIf this merging occurred across major classes (epithelial, endothelial, \\nimmune, stromal) at higher k values, then these clusters were instead \\nlabelled as ambiguous or low quality (including possible multiplets). \\nFor k\\u2009=\\u2009100 clusters (non-epithelial only) that did show distinct mark-\\ners, their k\\u2009=\\u200950 subclusters were assessed for distinct marker genes; if \\npresent, then these clusters were split further. The remaining split and \\nunsplit clusters were then assessed for altered-state features. If present, \\nthey were tagged as possible altered states, if absent they were assessed \\nas the final cluster. Annotations of clusters were based on known posi-\\ntive and negative cell type markers11,12,58–60 (Supplementary Table\\xa05), \\nthe regional distribution of the clusters across the corticomedullary \\naxis and altered state (including cell cycle) features. For separation of \\nEC-DVR from EC-AEA, the combined population was independently clus-\\ntered using Pagoda2 and clusters associated with medullary sampling \\nwere annotated as EC-DVR. For separation of the REN cluster, stromal \\ncells expressing REN were selected on the basis of normalized expression \\nvalues of greater than 3. Final overall assessment of clustering accuracy \\nwas performed using the Single Cell Clustering Assessment Framework \\n(SCCAF v.0.0.10) using the default settings, and compared against that \\nassociated with broad cell type classifications (subclass level 1).\\nAnnotating snCv3 clusters. T o overcome the challenge of dis -\\nparate nomenclature for kidney cell annotations, we leveraged a \\ncross-consortium effort to use the extensive knowledge base from \\nhuman and rodent single-cell gene expression datasets, as well as the \\ndomain expertise from pathologists, biologists, nephrologists and \\nontologists11,12,22,58–61 (see also Supplementary Tables\\xa04 and 5 and the \\nHuBMAP ASCT+B Reporter at GitHub (https://hubmapconsortium.\\ngithub.io/ccf-asct-reporter)). This enabled the adoption of a standard-\\nized anatomical and cell type nomenclature for major and minor cell \\ntypes and their subclasses (Supplementary Table\\xa04), showing distinct \\nand consistent expression profiles of known markers and absence of \\nspecific segment markers for some of the cell types (Extended Data \\nFig.\\xa02a and Supplementary Table\\xa05). The knowledge of the regions dis-\\nsected and histological composition of snCv3 data further enabled \\nstratification of distinct cortical and outer and inner medullary cell \\npopulations (Fig.\\xa02b and Extended Data Fig.\\xa01). The cell type identities \\nand regional locations were confirmed through orthogonal valida-\\ntion using spatial technologies presented here and correlations with \\nexisting human or rodent stromal, immune, endothelial and epithelial \\ndatasets4,25,58,59,61,62 (Extended Data Fig.\\xa02b–l).\\nAtlas cell type resolution\\nOur atlas now includes a higher granularity for the loop of Henle, distal \\nconvoluted tubule and collecting duct segments, now resolving three \\ndescending thin limb cell types (DTL1, 2, 3); different subpopulations \\nof medullary or cortical thick ascending limb cells (M-TAL/C-TAL); two \\ntypes of distal convoluted tubule cells (DCT1, 2); intercalated and prin-\\ncipal cells of the connecting tubules (CNT-IC and CNT-PC); cortical, \\nouter medullary and inner medullary collecting duct subpopulations \\n(CCD, OMCD, IMCD); and papillary tip epithelial cells abutting the calyx \\n(PapE). Molecular profiles for rare cell types important in homeostasis \\nwere annotated, including juxtaglomerular renin-producing granular \\ncells (REN); macula densa cells (MD); and a cell population with enriched \\nSchwann/neuronal (SCI/NEU) genes NRXN1, PLP1 and S100B. Major \\nendothelial cell types were stratified, including endothelial cells of \\nthe lymphatics (EC-LYM) and vasa recta (EC-AVR, EC-DVR). Specific \\nstromal and immune cell types were distinguished, including distinct \\nfibroblast populations across the cortico-medullary axis and 12 immune \\ncell types from lymphoid and myeloid lineages.\\nIntegrating snCv3 and SNARE2 datasets\\nIntegration of snCv3 and SNARE RNA data was performed using Seurat \\n(v.4.0.0) using snCv3 as reference. All counts were normalized using \\nsctransform, anchors were identified between datasets based on the \\nsnCv3 Pagoda2 principal components. SNARE2 data were then pro-\\njected onto the snCv3 UMAP structure and snCv3 cell type labels were \\ntransferred to SNARE2 using the MapQuery function. Both datasets \\nwere then merged and UMAP embeddings were recomputed using the \\nsnCv3 projected principal components. Integrated clusters were identi-\\nfied using Pagoda2, with the k-nearest neighbour graph (k\\u2009=\\u2009100) based \\non the integrated principal components and using the infomap com-\\nmunity detection algorithm. The SNARE2 component of the integrated \\nclusters was then annotated to the most overlapping, correlated and/\\nor predicted snCv3 cluster label, with manual inspection of cell type \\nmarkers used to confirm identities. Integrated clusters that overlapped \\ndifferent classes of cell types were labelled as ambiguous or low-quality \\nclusters. Segregation of EC-AEA, EC-DVR and REN subpopulations was \\nperformed as described for snCv3 above.\\nIntegrating snCv3 and scCv3 datasets\\nIntegration of snCv3 and scCv3 data was performed using Seurat \\nv.4.0.0 with snCv3 as a reference. All counts were normalized using ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='dc5fcbae-316d-4ce4-9b05-4a5f1ec9a510', embedding=None, metadata={'page_label': '597', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='sctransform, anchors were identified between datasets based on the \\nsnCv3 Pagoda2 principal components. scCv3 data were then projected \\nonto the snCv3 UMAP structure and snCv3 cell type labels were trans-\\nferred to scCv3 using the MapQuery function. Both datasets were \\nthen merged and UMAP embeddings recomputed using the snCv3 \\nprojected principal components. Integrated clusters were identified \\nusing Pagoda2, with the k-nearest neighbour graph (k\\u2009=\\u2009100) based \\non the integrated principal components and using the infomap com-\\nmunity detection algorithm. The scCv3 component of the integrated \\nclusters was then annotated to the most overlapping or correlated \\nsnCv3 subclass, with manual inspection of cell type markers used to \\nconfirm identities. Cell types that could not be accurately resolved \\n(PT-S1/PT-S2) were kept merged. Integrated clusters that overlapped \\ndifferent classes of cell types or that were too ambiguous to anno -\\ntate were considered to be low quality and were removed from the \\nanalysis. Segregation of EC-AEA, EC-DVR and REN subpopulations was \\nperformed as described above.\\nAssessment of snCv3, scCv3 and SNARE2 data integration\\nAs described above, we used the demonstrated Seurat v.4.0.0 integra-\\ntion strategy63 to project query datasets (scCv3, SNARE2 RNA) into the \\nsame PCA space as our snCv3 reference. These imputed principal com-\\nponents were used to generate an integrated embedding and integrated \\nclustering through Pagoda2. Query datasets within these integrated \\nclusters were manually annotated on the basis of co-clustering with the \\nreference data, predicted subclass levels and the manual inspection of \\nmarker genes. This process was necessary to account for misalignments \\nthat occurred for altered states showing more ambiguous marker gene \\nexpression profiles, especially for mapping between single-nucleus and \\nsingle-cell technologies. T o assess the accuracy in our alignments, we \\nperformed correlation of average expression signatures between the \\nassigned query cell populations and the original reference cell popula-\\ntions (Extended Data Fig.\\xa03e). Although several samples were examined \\nusing more than one platform (Supplementary Table\\xa03 and Extended \\nData Fig.\\xa01a), not all conditions could be covered by all technologies, \\nwith AKI/CKD biopsies too limited in size to process with SNARE2 and \\ndeeper medullary region capture being less likely for needle biopsies. \\nDespite the differences in patient conditions and regions sampled, \\nwe were able to confirm cross-platform sampling with minimal batch \\ncontributions for a majority of our subclass (level 3) assignments (77 \\ntotal). This was demonstrated through integrated bar plots for assay, \\npatient, sex and condition contributions (Extended Data Fig.\\xa03e). The \\ndegree to which cells/nuclei between assays were mixed within these \\nsubclasses was confirmed using normalized relative entropy weighted \\nby subclass size64, with an average assay entropy across subclasses \\n(covered by more than one technology) of 0.71 and an average patient \\nentropy of 0.71 (out of 1). Mixing within the subclasses was also assessed \\non the cell embeddings (principal components) using the average \\nsilhouette width or ASW (scib.metrics.silhouette_batch function of \\nthe scIB package v.1.0.365), with an average score of 0.86 for assays and \\n0.82 for patients (out of 1). Finally, the average of k-nearest neighbour \\nbatch effect test (kBET) score per subclass, computed for all patients \\nusing the scib.metrics.kBET function of the scIB package, was 0.49 (out \\nof 1), which is consistent with other integration efforts65.\\nIntegrating snCv3 with published datasets\\nIntegration with published data was performed using Seurat v.4.0.0 \\nwith snCv3 as a reference. All counts were normalized using sctrans-\\nform, anchors were identified between datasets on the basis of the \\nsnCv3 Pagoda2 principal components. Published data were then pro-\\njected onto the snCv3 UMAP structure and snCv3 cell type labels were \\ntransferred to the published dataset using the MapQuery function. \\nRef. 12 snDrop-seq data are available at the Gene Expression Omnibus \\n(GEO: GSE121862). Ref. 15 single-nucleus RNA-seq and ref. 14 single-cell \\nRNA-seq count matrices and metadata tables were downloaded from \\nthe UCSC Cell Browser (Cell Browser dataset IDs human-kidney-atac \\nand kidney-atlas, respectively).\\nNSForest marker genes\\nT o identify a minimal set of markers that can identify snCv3 clusters \\nand subclasses (subclass.l3), or scCv3 integrated subclasses (subclass.\\nl3), we used the Necessary and Sufficient Forest66 (NSForest v.2; https://\\ngithub.com/JCVenterInstitute/NSForest/releases/tag/v2.0) software \\nusing the default settings.\\nCorrelation analyses\\nFor correlation of RNA expression values between snCv3 and scCv3, \\nor SNARE2, average scaled expression values were generated, and \\npairwise correlations were performed using variable genes identi-\\nfied from Pagoda2 analysis of snCv3 (top 5,526 genes). For compari-\\nson with mouse single-cell RNA-seq data of healthy reference tissue59, \\nraw counts were downloaded from the GEO (GSE129798). For com-\\nparison with mouse single-cell RNA-seq from IRI tissue4, raw counts \\nwere downloaded from the GEO (GSE139107). For human fibroblast \\nand myofibroblast data25, raw counts were downloaded from Zenodo \\n(https://doi.org/10.5281/zenodo.4059315 ). For each dataset, raw \\ncounts were processed using Seurat: counts for all cell barcodes were \\nscaled by total UMI counts, multiplied by 10,000 and transformed to \\nlog space. For comparison with mouse single-cell types of the distal \\nnephron61, the precomputed Seurat object was downloaded from the \\nGEO (GSE150338). For mouse bulk distal segment data61, normalized \\ncounts were downloaded from the GEO (GSE150338) and added to the \\n‘data’ slot in a Seurat object. Bulk-sorted immune cell reference data \\nwere obtained using the celldex package67 using the MonacoImmune-\\nData()62 and ImmGenData()67,68 functions and log counts imported \\ninto the ‘data’ slot of Seurat. For correlation against these reference \\ndatasets, averaged scaled gene expression values for each cluster were \\ncalculated (Seurat) using an intersected set of variable genes identi-\\nfied for each dataset (identified using Padoda2 for snCv3 and Seurat \\nfor reference datasets). For immune reference correlations, a list of \\nimmune-related genes downloaded from ImmPort (https://immport.\\norg) was used instead of the variable genes. Correlations were plot-\\nted using the corrplot package (https://github.com/taiyun/corrplot). \\nImmune annotations within our atlas were further confirmed by manual \\ncomparison with recently reported data14.\\nCross-species alignment of cell types/states\\nFor mouse single-nucleus RNA-seq data from IRI tissue4, raw counts \\nwere downloaded from the GEO (GSE139107). Integration was per-\\nformed using Seurat v.4.0.0 with snCv3 as a reference. All counts were \\nnormalized using sctransform, anchors were identified between data-\\nsets on the basis of the snCv3 Pagoda2 principal components. Mouse \\ndata were then projected onto the snCv3 UMAP structure and snCv3 \\ncell type labels were transferred using the MapQuery function.\\nComputing single-nucleus/cell-level expression scores\\nT o identify markers associated with altered states (degenerative;  \\nadaptive—epithelial or aEpi; adaptive—stromal or aStr; cycling), snCv3 \\nand scCv3 data were independently used to identify differentially \\nexpressed genes between reference and corresponding altered states \\nfor each subclass level 1 (subclass.l1). T o ensure general state-level \\nmarkers, differentially expressed genes were identified using the \\nFindConservedMarkers function (grouping.var\\u2009=\\u2009“condition.l1” ,  \\nmin.pct\\u2009=\\u20090.25, max.cells.per.ident\\u2009=\\u2009300) in Seurat. A minimal set \\nof general degenerative conserved genes was identified as enriched \\n(P\\u2009<\\u20090.05) in the degenerative state of each condition.l1 (reference, AKI \\nand CKD) and in at least 4 out of the 11 (snCv3) or 9 (scCv3) subclass.l1 \\ncell groupings. A minimal set of conserved aEpi genes was identified as \\nenriched (P\\u2009<\\u20090.05) in the adaptive state of each condition.l1 (reference, \\nAKI and CKD) in both the aPT and aTAL cell populations. This aEpi gene ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='dfa437c2-ffe9-4c03-bb65-142b5f51197f', embedding=None, metadata={'page_label': '598', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nset was then further trimmed to include only those genes that were \\nenriched within the adaptive epithelial population (aPT/aTAL) versus all \\nothers using the FindMarkers function and with a minimum P value of \\n0.05 and average log2-transformed fold change of >0.6. A minimal set of \\nconserved aStr genes was identified as enriched (P\\u2009<\\u20090.05) in the adap-\\ntive state of each condition.l1 (reference, AKI and CKD for snCv2; refer-\\nence and AKI for scCv3) for stromal cells. T o increase representation \\nfrom MyoF in scCv3 showing a small number of these cells, MyoF-alone \\nenriched genes (average log2-transformed fold change\\u2009≥\\u20090.6; adjusted \\nP\\u2009<\\u20090.05) were included for the scCv3 gene set. The aStr gene sets were \\nthen further trimmed to include only those genes that were enriched \\nwithin the adaptive stromal population (aFIB and MYOF) compared with \\nall others using the FindMarkers function and with a minimum P value \\nof 0.05 and average log2-transformed fold change of >0.6. A minimal set \\nof cycling-associated genes was identified as those enriched (adjusted \\nP\\u2009<\\u20090.05 and average log2-transformed fold change\\u2009>\\u20090.6) in the cycling \\nstate across all associated subclass.l1 cell groupings.\\nScores for altered state, extracellular matrix and for gene sets associ-\\nated with ageing or SASP were computed for each cell from averaged \\nnormalized counts using only the genes showing a minimum correla-\\ntion to the averaged whole gene set of 0.1 (ref. 25) (https://github.com/\\nmahmoudibrahim/KidneyMap). Ageing and SASP genes were obtained \\nfrom ref. 48 (top 20 genes upregulated in ageing kidney)48, ref. 69 (genes \\nfrom table\\xa0S3, group.age A), ref. 70 (SASP genes from figure\\xa02c) or ref. 71  \\n(from table\\xa0S1 (sheet IR Epithelial SASP), having a positive AVE log2 \\nratio)71.\\nGene set enrichment analyses (GSEA)\\nT o compute gene set enrichments for aPT and aTAL, conserved genes \\ndifferentially expressed in the adaptive over reference states were iden-\\ntified as indicated above. Gene set ontologies from the Molecular Signa-\\ntures Database (MSigDB) were downloaded from https://gsea-msigdb.\\norg and pathway enrichments were computed using fgsea72 and gage73, \\nretaining only Gene Ontology terms that were significant (P\\u2009<\\u20090.05) for \\nboth. Redundant pathways were collapsed using the fgsea function \\ncollapsePathways and visualized using ggplot.\\nSNARE2 accessible chromatin analyses\\nSNARE2 chromatin data were analysed using Signac74 (v.1.1.1). Peak call-\\ning was performed using the CallPeaks function and MACS (v.3.0.0a6; \\nhttps://github.com/macs3-project/MACS) separately for clusters, sub-\\nclass.l1 and subclass.l3 annotations. Peak regions were then combined \\nand used to generate a peak count matrix using the FeatureMatrix func-\\ntion, then used to create a new assay within the SNARE2 Seurat object \\nusing the CreateChromatinAssay function. Gene annotation of the \\npeaks was performed using GetGRangesFromEnsDb(ensdb\\u2009=\\u2009EnsDb. \\nHsapiens.v86). TSS enrichment, nucleosome signal and blacklist frac-\\ntions were all computed using Signac. Jaspar motifs ( JASPAR2020, all \\nvertebrate) were used to generate a motif matrix and motif object that \\nwas added to the Seurat object using the AddMotifs function. For motif \\nactivity scores, chromVAR75 (v.1.12.0; https://greenleaflab.github.io/\\nchromVAR) was performed using the RunChromVAR function. The \\nchromVAR deviation score matrix was then added to a separate assay \\nslot of the Seurat object. T o assess the chromatin data, UMAP embed-\\ndings were computed from cis-regulatory topics that were identified \\nthrough latent Dirichlet allocation using CisT opic76 (v.0.3.0; https://\\ngithub.com/aertslab/cisT opic) and the runCGSModels function. Only \\nregions accessible in 50 nuclei and nuclei with 200 of these accessible \\nregions were used for cisT opic and downstream analyses. The UMAP \\ncoordinates for the remaining nuclei were added to the Seurat object. \\nT o ensure high-quality accessible chromatin profiles, only clusters \\nwith more than 50 nuclei were retained for downstream analyses \\n(Supplementary Table\\xa07). For joint embedding of SNARE2 accessible \\nchromatin and gene expression, a weighted nearest-neighbour graph \\nwas computed using the FindMultiModalNeighbors function (Seurat) \\nbased on PCA (RNA) and latent semantic indexing or LSI (accessible \\nchromatin) dimensionality reductions. UMAP dimensionality reduc-\\ntion was performed to visualize the joint embedding.\\nDAR analyses\\nSites that were differentially accessible for a given cell grouping (sub-\\nclass) were identified against a selection of background cells with the \\nbest matched total peak counts, to best account for technical differ-\\nences in the total accessibility for each cell. For this, the total peaks in \\neach cell were used for estimation of the distribution of total peaks \\n(depth distribution) for the cells belonging to the test cluster, and \\n10,000 background cells with a similar depth distribution to the test \\ncluster were randomly selected. Differentially accessible sites (DARs) \\nwere then identified as significantly enriched in the positive cells over \\nselected background cells using the CalcDiffAccess function (https://\\ngithub.com/yanwu2014/chromfunks), where P values were calculated \\nusing Fisher’s exact tests on a hypergeometric distribution and adjusted \\nP values (or q values) were calculated using the Benjamini–Hochberg \\nmethod. For subclass level 2 DARs, VSM/P clusters were merged and the \\nMD was combined with C-TAL before to DAR calling. Subclasses with \\n>100 DARs with q\\u2009<\\u20090.01 were used for further analysis. Co-accessibility \\nbetween all peak regions was computed using Cicero77 (v.1.8.1). Sites \\nwere then linked to genes on the basis of co-accessibility with promoter \\nregions, occurring within 3,000\\u2009bp of a gene’s TSS, using the Region-\\nGeneLinks function (https://github.com/yanwu2014/chromfunks) \\nand the ChIPSeeker package78. DARs associated with markers for each \\nsubclass (identified at the subclass.l2 level using snCv3, P\\u2009<\\u20090.05) and \\nshowing q\\u2009<\\u20090.01 and a log-transformed fold change of >1 were selected \\nfor visualization. For this, DAR accessibility (peak counts) was averaged, \\nscaled (trimming values to a minimum of 0 and a maximum of 5) and \\nvisualized using the ggHeat plotting function of the SWNE package79. \\nMotif enrichment within cell type DARs was computed using the hyper-\\ngeometric test (FindMotifs function) in Signac.\\nTranscription factor analyses\\nT o identify active transcription factors from SNARE2 accessible chro-\\nmatin data, differential activities (or deviation scores) of TFBSs between \\ndifferent populations were assessed using the Find[All]Markers func-\\ntion through logistic regression and using the number of peak counts \\nas a latent variable. Only transcription factors with expression detected \\nwithin the corresponding cluster, subclass or state grouping were \\nincluded. For PT and TAL clusters, TFBSs that were differentially active \\n(P\\u2009<\\u20090.05, average log2-transformed fold change of >0.35) and associ-\\nated with transcription factors with expression detected in at least \\n2.5% of nuclei (SNARE2) were identified between clusters. Common \\naPT/aTAL TFBS activities were identified from an intersection of those \\ndifferentially active and expressed within adaptive PT and TAL clusters. \\nFor aPT and aTAL trajectory modules, TFBSs showing differential activ-\\nity between modules (adjusted P\\u2009<\\u20090.05, average log2-transformed fold \\nchange of >0.35) and expression detected within at least 2.5% of nuclei/\\ncells (snCv3/scCv3) were identified. For common degenerative state \\nTFBS activities, differentially active TFBSs were identified between \\nreference and degenerative states for each level 1 subclass. Significant \\ndegenerative state TFBS activities (P\\u2009<\\u20090.05, average log2-transformed \\nfold change of >0.35) in three or more subclass.l1 were trimmed to those \\nshowing expression detected in more than 20% of the degenerative \\nstate nuclei/cells for snCv3/scCv3.\\nLigand–receptor interaction analyses\\nLigand–receptor analyses were performed on the basis of the Cell-\\nChat package (v.1.0.0; https://github.com/sqjin/CellChat). Only cells \\nin TAL, immune and stroma of subclass level 2 (immune: cDC, cycMNP, \\nMAC-M2, MAST, MDC, N, ncMON, NKT, pDC, PL, T and B; stroma: MyoF , \\nFIB, dFIB, cycMyoF and aFIB) and interactions for secreted ligands \\nwere used to infer the cell–cell communication. For cells in the TAL ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='411f7143-56e3-4278-ad4a-a5ae8e1782cf', embedding=None, metadata={'page_label': '599', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='trajectory, we computed the intercellular cell communication probabil-\\nity between each module and other cell populations using the CellChat \\nfunction computeCommunProb (see ref. 80 for a detailed description of \\nthe method). The overall scaled communication probability was then \\nvisualized based on a circle plot using a customized plot_communica-\\ntion function (Code availability). T o further understand which signals \\ncontribute most to the ligand–receptor (LR) interaction pathways, \\nwe generated the pathway enrichment heat map of each interaction \\nfor incoming, outgoing and overall signals using the plotSigHeatmap \\nfunction (Code availability). The contribution of significant LR pairs \\nof each interaction was also identified using netAnalysis_contribution \\nin the CellChat package.\\nGWAS analyses\\nT o link SNARE2 cell types to kidney genome-wide association study \\n(GWAS) traits and diseases, we first summed the binary peak acces-\\nsibility profiles for all cells belonging to the same cell type to create a \\npseudobulk peak-by-subclass accessibility matrix. Pseudobulk analyses \\ngive more stable results, especially as SNARE2 accessibility data can be \\nsparse. T o ensure sufficient coverage, we used subclass level 2 group-\\nings with the following modifications: VSM/P clusters were merged; MD \\nwas combined with C-TAL; subclasses with <100 DARs with q\\u2009<\\u20090.01 were \\nexcluded. We used g-chromVAR81 (v.0.3.2), an extension of chromVAR \\nfor GWAS data, to identify cell types with higher than expected acces-\\nsibility of genomic regions overlapping GWAS-linked single-nucleotide \\npolymorphisms (SNPs). Running g-chromVAR requires first identi-\\nfying GWAS-linked SNPs that are more likely to be causal, a process \\nknown as fine-mapping. For the chronic kidney failure GWAS traits, we \\nused existing fine-mapped SNPs from the CausalDB database, using \\nthe posterior probabilities generated by CAVIARBF82,83. The original \\nGWAS summary statistics files were obtained from an atlas of genetic \\nassociations from the UK BioBank84. We manually fine-mapped the \\nCKD, eGFR, blood urea nitrogen and gout traits using the same code \\nthat was used to generate the CausalDB database (https://github.com/\\nmulinlab/CAUSALdb-finemapping-pip). The summary statistics for all \\nof these traits are available at the CKDGen Consortium site (https://\\nckdgen.imbi.uni-freiburg.de/)85,86. We also manually fine-mapped the \\nhypertension trait and the original summary statistics can be found \\non the EBI GWAS Catalog87. We looked only at causal SNPs with a pos-\\nterior causal probability of at least 0.05 to ensure that SNPs with low \\ncausal probabilities did not cause false-positive signals. Moreover, as \\ng-chromVAR selects a semi-random set of peaks with similar average \\naccessibility and GC content as background peaks, the method has an \\nelement of randomness. T o ensure stable results, we ran g-chromVAR \\n20 times and averaged the results. Cluster/trait z-scores were plotted \\nusing ggheat (https://github.com/yanwu2014/swne).\\nT o link causal SNPs to genes, we used functions outlined in the \\nchromfunks repository (https://github.com/yanwu2014/chromfunks;  \\n/R/link_genes.R). This involved the identification of causal peaks for \\neach cell type and trait (minimum peak Z score of 1, minimum peak \\nposterior probability score of 0.025). Sites were then linked to genes on \\nthe basis of co-accessibility (Cicero) with promoter regions, occurring  \\nwithin 3,000\\u2009bp of a gene’s TSS. Only sites associated with genes \\ndetected as expressed in 10% of TAL nuclei/cells (snCv3/scCv3) were \\nincluded. Motif enrichment within the causal SNP and TAL-associated \\npeaks was performed using the FindMotifs function in Seurat and only \\nmotifs for transcription factors expressed in 10% of TAL nuclei/cells  \\n(snCv3/scCv3) were included (Supplementary Table\\xa031). For a TAL- \\nassociated ESRRB transcription factor subnetwork, peaks were linked \\nto genes using Cicero, then subset to those associated with TAL (C-TAL, \\nM-TAL) marker genes that were identified using the Find[All]Markers \\nfunction in Seurat for subclass.l3 (P\\u2009<\\u20090.05). Transcription factors were \\nthen linked to gene-associated peaks on the basis of the presence of \\nthe motif and correlation of peak and TFBS co-accessibility (chrom-\\nVAR), using a correlation cut-off of 0.3. Only transcription factors with \\nexpression detected within 20% of TAL cells or nuclei (snCv3/scCv3) \\nwere included. Eigenvector centralities were then computed using \\nigraph and the transcription-factor-to-gene network was visualized \\nusing PlotNetwork in chromfunks.\\nDisease-associated gene set enrichment analyses\\nGenes linked with CKDGen consortium GWAS loci for the kidney func-\\ntional traits eGFR and urinary albumin-creatinine ratio (UACR) were \\nobtained from table S14 of ref. 88. These included the top 500 genes \\nper trait or only those genes also implicated in monogenic glomerular \\ndiseases. eQTLs associated with eGFR, systolic blood pressure and \\ngeneral kidney function were obtained from tables S20, S21 and S22 of \\nref. 89, respectively. Genes associated with the transition from acute to \\nchronic organ injury after ischaemia–reperfusion were obtained from \\nref. 90 from the following supplementary tables: Acute_Human_Specific \\n(table\\xa0S3, Human specific column); Acute_Mouse_Overlap (table\\xa0S3, \\nShared column); Mid_Acute (table\\xa0S8, cluster 2 genes); Late_Human_\\nSpecific (table\\xa0S9, Human specific column); Late_Mouse_Overlap \\n(table\\xa0S9, Shared column); Late_Fibrosis (table\\xa0S6, positive logFC); \\nLate_Recovery (table\\xa0S6, negative logFC). Each gene set was assessed \\nfor its enrichment within combined snCv3 and scCv3 subclass (level 3)  \\ndifferentially expressed genes (adjusted P\\u2009<\\u20090.05, log-transformed \\nfold change of >0.25). Enrichments were performed using Fisher’s \\nexact tests and the resultant −log10[P] values were scaled and visual-\\nized using ggplot2.\\nPatient cohorts used for clinical association analyses\\nNEPTUNE91 (193 adult patients) and ERCB45 (131 patients) expression \\ndata were used as validation cohorts to determine the significance \\nbetween patients with different levels of cell state gene expression. \\nNEPTUNE (NCT01209000 ) is a multicentre (21 sites) prospective \\nstudy of children and adults with proteinuria recruited at the time \\nof first clinically indicated kidney biopsy (Supplementary Table\\xa034). \\nThe study participants were followed prospectively, every 4\\u2009months \\nfor the first year, and then biannually thereafter for up to 5\\u2009years. At \\neach study visit, medical history, medication use and standard local \\nlaboratory test results were recorded, while blood and urine samples \\nwere collected for central measurement of serum creatinine and urine \\nprotein/creatinine ratio (UPCR) and eGFR (ml per min per 1.73\\u2009m2). \\nEnd-stage kidney disease (ESKD) was defined as initiation of dialysis, \\nreceipt of kidney transplant or eGFR <15\\u2009ml per min per 1.73\\u2009m2 meas-\\nured at two sequential clinical visits; and the composite end point of \\nkidney functional loss by a combination of ESKD or 40% reduction in \\neGFR92. Genome-wide transcriptome analysis was performed on the \\nresearch core obtained at the time of a clinically indicated biopsy using \\nRNA-seq by the University of Michigan Advanced Genomics Core using \\nthe Illumina HiSeq2000 system. Read counts were extracted from the \\nfastq files using HTSeq (v.0.11). NEPTUNE mRNA-seq data and clinical \\ndata are controlled access data and will be available to researchers on \\nrequest to NEPTUNE-STUDY@umich.edu.\\nERCB is the European multicentre study that collects biopsy tissue \\nfor gene expression profiling across 28 sites. Transcriptional profiles \\nof biopsies from patients in the ERCB were obtained from the GEO \\n(GSE104954).\\nClinical association of cell state scores\\nThe gene expression data from the tubulointerstitial compartment of \\nthe kidney biopsies from NEPTUNE patients was used to calculate the \\ncomposite scores for the genes enriched in degenerative, aPT, aTAL and \\naStr states. The expression of the genes that were uniquely enriched \\nin the cell state (described above) and that were found in both snCv3 \\nand scCv3 were used to calculate the composite cell state score (Sup-\\nplementary Table\\xa029). As scCv3 did not efficiently identify all stromal \\ncell types, the union of the enriched genes from scCv3 and snCv3 data \\nwere used to calculate the aStr cell state score. We also generated a ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='418fb7f6-af2b-448e-86db-0422e29bf31c', embedding=None, metadata={'page_label': '600', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\ncell state score for the genes that were commonly enriched in aPT and \\naTAL cells (common).\\nFor outcome analyses (40% loss of eGFR or ESKD) in the NEPTUNE \\ncohort, patient profiles were binned according to the degree of cell state \\nscore by tertile. Clinical outcomes were available on 193 participants \\nwith a total of 30 events. Kaplan–Meier analyses were performed using \\nlog-rank tests to determine significance between patients in different \\ntertiles of cell state gene expression. Moreover, for the different cell \\nstate scores, multivariable adjusted Cox proportional hazard analyses \\nwere performed using five statistical models adjusting for different sets \\nof potential confounding effects given the overall few number of events: \\n(1) age, sex and race; (2) baseline eGFR and UPCR; (3) immunosuppressive \\ntreatment and FSGS status; (4) eGFR, UPCR and race self-reported as Black \\n(factors that were associated with outcome in this dataset); and (5) immu-\\nnosuppressive treatment, eGFR and UPCR (Supplementary Table\\xa030). \\nNote that the adjusted models simply assess whether the association with \\noutcome persists after adjusting for common clinical features (that is, \\nconfounding effects), but do not assess for prediction accuracy.\\nIn the ERCB cohort, differential expression analyses using multi-\\nvariable regression modelling were performed between the cell state \\nscores in the disease groups and living donors. Age and sex were used \\nas covariates. The cell state scores for both NEPTUNE and ERCB bulk \\nmRNA transcriptomics data were generated93. In brief, the cell state \\nscores were generated by creating Z scores for each of the cell state gene \\nsets and then using the average Z score as the cell state composite score. \\nThese analyses found scores for all adaptive epithelial, but not degen-\\nerative, states were significantly higher in the patients with diabetic \\nnephropathy patients compared to that of living donors (Supplemen-\\ntary Table\\xa030). After adjusting for sex and age, both aPT and aTAL were \\nsignificant when scores from patients with diabetic nephropathy were \\ncompared with those of living donors and aPT scores were significant \\neven after correcting for the different disease groups.\\nSample-level analysis and clustering on clinical association  \\ngene sets\\nT o find association of patients based on altered-state gene signatures that \\nwere used in clinical association analyses (Supplementary Table\\xa030), we \\nperformed sample-level clustering. All of the cells from the same patient \\nin snCv3 and scCv3 were aggregated to get pseudo-bulk count matrices \\non the basis of the associated clinical gene set. The matrices were further \\nnormalized by RPKM followed by t-distributed stochastic neighbour \\nembedding (t-SNE) dimensionality reduction. Groups of patients were \\nthen identified based on k-means clustering and density-based spatial \\nclustering (DBSCAN) methods in the reduced space. T o associate the \\npatient clusters with clinical features, we calculated the distribution \\nof eGFR in each identified group (Code availability).\\nT o identify gene sets that best differentiate between AKI and CKD \\npatients in the PT and TAL cell populations, we trained a gene-specific \\nlogistic regression model based on the sample-level gene expression. \\nThe model was used to assess the predictive power that differentiate \\npatients with AKI and CKD in both snCv3 and scCv3 measured by area \\nunder the curve (AUC). The genes with AUC\\u2009>\\u20090.65 on both snCV3 and \\nscCv3 were selected for downstream analysis (Supplementary Table\\xa032).\\nT o identify genes that were differentially expressed between AKI and \\nCKD across all cell types, we aggregated the cells associated with each \\nsubclass (level 1) to generate cell-type-specific pseudocounts for each \\nsample and performed differential gene expression analysis based on \\nthe DEseq2 method using the estimatePerCellTypeDE function in the \\nCacoa package (v.0.2.0; https://github.com/kharchenkolab/cacoa).\\nPseudotime analysis of PT and T AL cells\\nT o find cells associated with disease progression, we performed tra-\\njectory analysis for PT and TAL cells. T o get accurate pseudotime and \\ntrajectory estimation, we removed degenerative cell populations in \\nboth PT and TAL and inferred the trajectory for single nuclei and single \\ncells separately using the Slingshot package94 (v.2.0.0). We specified \\nnormal cell populations as the end points for trajectory inference  \\n(S1–S3 in PT and M-TAL in TAL) using the Slingshot parameter end.\\nclus. The correspondent trajectory embedding was visualized using \\nthe plotEmbedding function in the Pagoda2 package.\\nT o identify whether the gene expression was statistically significantly \\nassociated with the inferred trajectory, we modelled the expression \\nof a gene as a function of the estimated pseudotime by fitting a gam \\nmodel with cubic spline regression using formula expi\\u2009= f(t) + ϵ, where \\nt is the pseudotime and f is the function of cubic spline. The model is \\nthen compared to a reduced model expi = f(1) + ϵ to get P-value estimates \\nusing the F-test. The Benjamin–Hochberg method was used to calculate \\nthe adjusted P values. T o further identify candidate genes showing \\npotential differences between patients with AKI and CKD, we extended \\nthe base gam model by fitting a conditional-smooth interaction using \\nCKD as a reference.\\nGene module detection and cell assignment\\nT o identify expression modules for significant gene sets along the \\nestimated trajectories, we applied the module detection algorithm \\nimplemented in the WGCNA package95 (v.1.70-3) based on the smoothed \\ngene expression matrix with parameters softPower\\u2009=\\u200910 and minMod-\\nuleSize\\u2009=\\u200920. The similar modules detected by the original parameters \\nwere further merged. In total, we identified five different modules in \\nPT and six modules in TAL cells. For the gene sets in each module, we \\nfurther performed pathway analysis using the Reactome online tool96 \\n(https://reactome.org/PathwayBrowser/). The enrichment of clinical \\nassociated gene sets for each module (Fig.\\xa06e) was assessed by per-\\nforming log ratio enrichment tests. T o predict the transcription factor \\nactivities of PT and TAL subclass genes, we used the DoRothEA package \\n(v.1.7.2) as targets. DoRothEA transcription factors and transcriptional \\ntargets were curated from both human and mouse evidence. The tran-\\nscription factor activity scores for each cell type were calculated based \\non the run_viper function of the viper package (v.3.15; https://biocon-\\nductor.org/packages/release/bioc/html/viper.html).\\nT o identify cells that are associated with each module, we developed \\na systematic approach. In brief, for the cells in the smoothed expres-\\nsion matrix, we performed dimension reduction using PCA followed \\nby Louvain clustering. This enabled the identification of cell clusters \\nalong the trajectory. For the identified cell clusters, we then performed \\nhierarchical clustering to calculate the correlation of each module on \\nthe basis of mean gene expression values and further linked the clus-\\nters with associated modules by cutting the hierarchical tree. Finally, \\nmodule labels for each cell were assigned on the basis of its associated \\nclusters. T o link single-cell datasets with single-nucleus modules, we \\nperformed k-means clustering based on the joint embedding of PT or \\nTAL cells and assigned the cells in scCv3 to modules on the basis of the \\nmajority voting from its k’s nearest neighbours (Code availability).\\nT o further investigate cluster-free compositional change between \\ndisease conditions, we also performed cell density analysis, in which \\nwe compared the normalized cell density between AKI and CKD condi-\\ntions through 2D kernel estimates using Cacoa Package. Z scores were \\ncalculated to identify the regions that showed significant differences \\nin cell density.\\nT o validate the direction of modules inferred from human data, \\nwe performed joint alignment of the human and mouse trajectories. \\nThe individual trajectories inferred separately from these two species \\n(Slingshot, described above) were aligned to generate a joint trajectory \\nusing CellAlign (https://github.com/shenorrLab/cellAlign) with param-\\neters winSz\\u2009=\\u20090.1 and NumPts\\u2009=\\u20091000. The collection groups (timepoints \\nfrom injury) derived from mouse data were then projected to human \\ncells based on the joint trajectory. The genes that were conserved or \\ndivergent between the two species were specified as overlapping/ \\ndistinct gene sets that were tested for significance based on a gam \\nmodel inferred from the trajectory (see above).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c7fbf6df-ede2-48ef-b3c4-4e0bcffd1669', embedding=None, metadata={'page_label': '601', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='RNA velocity analyses\\nSpliced and unspliced reads were counted from Cell Ranger BAM files \\nfor each snCv3 run using velocyto97 (v.0.17.17) and using the GRCh38 \\ngene annotations prepackaged with the Cell Ranger pipeline. Repeti-\\ntive elements were downloaded from the UCSC genome browser and \\nmasked from these counts. Corresponding loom files were loaded into \\nR using the SeuratWrappers function ReadVelocity and converted to \\nSeurat objects using the as.Seurat function. aPT or aTAL trajectory \\npopulations were then subset and RNA velocity estimates were calcu-\\nlated using scVelo98 (v.0.2.4) through a likelihood-based dynamical \\nmodel. Velocity embeddings on the trajectory UMAPs were visual -\\nized using the pl.velocity_embedding_stream function. Latent times \\nbased on transcriptional dynamics predicted from splicing kinetics \\nwere computed and the top 300 dynamical genes were plotted using \\nthe pl.heatmap function. T op likelihood genes were computed for \\neach TAL module to identify potential drivers for these states. Spliced \\nversus unspliced or latent time scatter plots were generated using the \\npl.scatter function.\\nGRN analyses\\nGRNs associated with TAL trajectory modules were constructed using \\nCelloracle (v.0.9.1) with the default parameters outlined in the pro -\\nvided tutorials (https://morris-lab.github.io/CellOracle.documenta-\\ntion). The base GRN was first constructed from SNARE2 accessible \\nchromatin data. Co-accessible peaks across cell types identified using \\nCicero (v.1.8.1) were linked to genes through their TSS peaks to iden-\\ntify accessible promoter/enhancer DNA regions. Peaks were then \\nscanned for transcription-factor-binding motifs (gimme.vertebrate.\\nv5.0) to generate a base GRN. snCv3 data were then used to identify \\nTAL state-specific GRNs. T o ensure that relevant genes were used, we \\nincluded genes that varied across the aTAL trajectory (Supplemen -\\ntary Table\\xa017), showed dynamic module-specific transcription from \\nscVelo analyses (Supplementary Table\\xa021), were variably expressed \\nacross TAL cells (Pagoda2) or that were associated with differential \\ntranscription factor activities (Supplementary Table\\xa020). GRN infer-\\nence through regularized machine learning regression models was \\nperformed to prune inactive (insignificant or weak) connections \\nand to select active edges associated with regulatory connections \\nwithin each module or state, retaining the top 2,000 edges ranged by \\nedge strength. Network scores for different centrality metrics were \\nthen calculated and visualized using Celloracle plotting functions. \\nFor in silico transcription factor perturbation analyses, target gene \\nexpression was set to 0 and resultant gene expression values were \\nextrapolated or interpolated using the default parameters of Cellora-\\ncle and according to the provided tutorial. Stromal GRN construction \\nwas performed as indicated above, except using a gene subset that \\nincluded variable STR genes identified using Pagoda2; subclass level \\n3 markers for FIB, aFIB, MyoF (adjusted P \\u2009<\\u20090.05); or transcription \\nfactors with expression detected in at least 2.5% of nuclei (SNARE2) \\nand having binding sites that were differentially active between STR \\nsubclasses (P\\u2009<\\u20090.05). T o ensure BMP target SMADs were represented, \\nSMAD1/5/8 were also included.\\nSLIDE-seq2\\nPuck preparation and sequencing. Tissue pucks were prepared  \\nfrom fresh frozen kidney tissue either embedded in Optimal Cut -\\nting T emperature (OCT) compound or frozen in liquid nitrogen and  \\nsequenced20,99 according to a step-by-step protocol (https://doi.org/ \\n10.17504/protocols.io.bvv6n69e). Libraries were sequenced on the \\nNovaSeq S2 flowcell (NovaSeq 6000) with a standard loading con-\\ncentration of 2\\u2009nM (read structure: read 1, 42\\u2009bp; index 1, 8\\u2009bp; read 2, \\n60\\u2009bp; index 2, 0\\u2009bp). Demultiplexing, genome alignment and spatial \\nmatching was performed using Slide-seq tools (https://github.com/\\nMacoskoLab/slideseq-tools/releases/tag/0.1).\\nDeconvolution. We used Giotto100 (v.1.0.3) for handling the slide-seq \\ndata and RCTD101 (v.1.2.0) for the cell type deconvolution. As only ref-\\nerence tissue was used for slide-seq, all degenerative states as well as \\nPapE, NEU, B and N were removed from the snCv3 Seurat object prior \\nto deconvolution. The Seurat object was randomly subsampled to \\nhave at most 3,000 cells from each level 2 (l2) subtype and the level 1  \\n(l1) subclasses of ATL and DTL were merged. For each data source, that \\nis, HuBMAP or KPMP (Supplementary Table\\xa02), the counts from all \\nbeads across all pucks were pooled and deconvolved hierarchically: \\nfirst, beads with less than 100\\u2009UMIs and genes detected in less than 150 \\nbeads were removed. Then, the broad l1 subclass annotations in the \\nSeurat object were used to deconvolve all beads (gene_cutoff\\u2009=\\u20090.0003, \\ngene_cutoff_reg\\u2009=\\u20090.00035, fc_cutoff\\u2009=\\u20090.45, fc_cutoff_reg\\u2009=\\u20090.6, manu-\\nally adding REN in the RCTD gene list and merging ATL and DTL subtypes \\nas TL). The prediction weights were normalized to sum to 100 per bead. \\nBeads for which one cell type had a relative weight of 40% or higher were \\nclassified as that l1 subclass. Then, for each l1 subclass, all classified \\nbeads were further deconvolved using the l2 annotation of that sub-\\nclass, as well as the remaining subclass l1 annotations (same parameters \\nas l1). Note that, for each l2 deconvolution, the bulk parameters in RCTD \\nwere fitted using all beads and then the RCTD object was subsetted to \\nonly contain the selected beads for the l2 deconvolution. Classifica-\\ntion at subclass l2 was done similar to l1 with the maximum relative \\nweight cut-off of 30% or 50% depending on the stringency needed for \\nan analysis (50% for Figs.\\xa02c,f and Extended Data Fig.\\xa04b and 30% in other \\nanalyses). For plotting gene counts, the scaling was performed with the \\ncommand normalizeGiotto(gObj, scalefactor\\u2009=\\u200910000, log_norm\\u2009=\\u2009T, \\nscale_genes\\u2009=\\u2009T, scale_cells\\u2009=\\u2009F). The marker gene dot plots were plotted \\nusing the DotPlot function in Seurat (v.4.0.0).\\nCell type interaction. Coarse cell–cell interactions can be revealed by \\nlooking for cell types that tend to be in close proximity. For each puck, \\nwe created a neighbourhood graph based on Delaunay triangulation \\nin which each bead is connected by an edge to at least one other neigh-\\nbouring bead, provided that their distance is smaller than 50\\u2009µm. For \\neach pair of cell types, we count the number of times they are connected \\nby edges. Then, the cell type labels are randomly permuted 2,500 times \\nto form the null distribution of the number of connections. The expec-\\nted number of connections between pairs of cell types is calculated \\nfrom this simulation and the proximity enrichment is defined as the \\nratio of the observed over the expected frequency of connections. \\nThe network construction and enrichment analysis were performed \\nusing Giotto’s createSpatialNetwork and cellProximityEnrichment \\ncommands, respectively. Those beads with maximum level 2 weight \\nless than 30% were removed. We further excluded spurious beads that \\nwere outside of the visual boundary of the tissue (only for the pucks \\nof which the names start with ‘Puck_210113’) by manually specifying \\nstraight lines that follow the tissue boundary. For cortical pucks (Sup-\\nplementary Table\\xa02), M-PC, C-PC and IMCD labels were relabelled as PC; \\nM-TAL and C-TAL as TAL; and EC-DVR was merged into EC-AEA. Other \\nmedullary and cycling subtypes were removed. For medullary pucks, \\nM-PC and C-PC were relabelled as PC; M-TAL and C-TAL as TAL; all DTL \\nsubtypes as DTL; and EC-AEA was merged into EC-DVR. Other cortical \\nand cycling subtypes were removed.\\nT o generate the proximity plots in Extended Data Fig.\\xa04, the enrich-\\nment values for each cell type pair were averaged across all pucks from \\nthe same region and plots were generated using the R package ggGally \\n(v.2.1.2). For the cortex and medulla, respectively, only the connections \\nwith mean enrichment values higher than 0.7 and 0.8 were plotted.\\n10x Visium spatial transcriptomics\\nPreparation, imaging and sequencing. Human kidney tissue was \\nprepared and imaged according to the Visium Spatial Gene Expression \\n(10x Genomics) manufacturer’s protocol (CG000240, Visium Tissue ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='36f8577a-2b42-43b9-9b7e-e5b70c055865', embedding=None, metadata={'page_label': '602', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nPreparation Guide) and as previously described102. Nephrectomy (n\\u2009=\\u20096), \\nAKI (n\\u2009=\\u20096) and CKD (n\\u2009=\\u200911) samples were sectioned at 10\\u2009µm thickness \\nfrom OCT-compound-embedded blocks. These 23 samples represent 22 \\nparticipants because 2 samples (1 cortex and 1 medulla) were obtained \\nfrom the same participant with CKD. A Keyence BZ-X810 microscope \\nequipped with a Nikon ×10 CFI Plan Fluor objective was used to acquire \\nH&E-stained bright-field mosaics, which were subsequently stitched. \\nmRNA was isolated from stained tissue sections after permeabiliza-\\ntion for 12\\u2009min. Released mRNA was bound to oligonucleotides in the \\nfiducial capture areas. mRNA was then reverse-transcribed and under-\\nwent second strand synthesis, denaturation, cDNA amplification and \\nSPRIselect cDNA cleanup (Visium CG000239 protocol) as part of library \\npreparation. Sequencing was performed on the Illumina NovaSeq 6000 \\nsystem103.\\nGene expression analysis. Space Ranger (v.1.0 or higher) with the  \\nreference genome GRCh38 was used to perform expression analysis, \\nmapping, counting and clustering. Summary statistics and quality- \\ncontrol metrics are included in Extended Data Fig.\\xa05 and Supplementary \\nTable\\xa02. Normalization was performed using SCTransform104. Final data \\nprocessing was performed in Seurat (v.3.2.3). Expression feature plots \\ndepict the intensity of transcript expression in each spot. In each Visium \\nsample, the outermost layer of spots was eliminated from comparative \\nanalyses if the edge was manually cut by a razor.\\nDeconvolution. Using Seurat (v.3.2.0), a transfer score system was \\nused to assess and map the proportion of signatures arising from each \\n55\\u2009µm spot. The transfer score reflects a probability between each spot’s \\nsignature and its association with a given snCv3 subclass (level 2). The \\nhighest probability transfer scores have the highest proportion mapped \\nwithin each spatial transcriptomics spot pie graph. For cell type feature \\nplots (Figs.\\xa02g and 3f and Extended Data Fig.\\xa07i), subclass level 2 cell type \\ntransfer scores were mapped to convey the proportion of signature \\nunderlying each spot. For cell state feature plots (Fig.\\xa03b), instead of \\nmapping subclass level 2 cell types, the aEpi cell state annotated in \\nsnCv3 was mapped across all spots in the samples. We summed the \\nproportion of signatures arising from all cell types corresponding to \\neach of the 6 cell states in all spots of all samples (Fig.\\xa03a). The propor-\\ntions of cell state were compared across nephrectomy, AKI and CKD \\nsamples using Fisher’s exact tests.\\nColocalization of epithelial, immune and stromal cells. In all spots \\nacross all samples, we categorized spots into healthy, adaptive or \\ndegenerative epithelial cell states on the basis of the highest propor-\\ntion of epithelial cell state signature as calculated in Fig.\\xa03a. For stro-\\nmal or immune cell type colocalization, we first selected spots with \\nnon-zero transfer scores of each cell type in all 23 samples. The presence \\nof stromal or immune cell signature was considered colocalized with an \\nepithelial cell if its stromal or immune transfer score exceeded its mean \\ntransfer score across all selected spots. An odds ratio was calculated \\nfor colocalization between the healthy, adaptive and degenerative \\nepithelial cell state with stromal or immune cell signature.\\nCell state marker expression. T o compare marker gene expression \\nassociated with the healthy, adaptive and degenerative cell states \\n(Fig.\\xa03d), we first categorized a subset of spots from AKI and CKD sam-\\nples into 1 of 5 predominant cell types: POD, PT, TAL, CD or FIB. For \\nthe PT, TAL and fibroblasts, a spot was selected if the highest propor-\\ntion of its signature (level 1 mapping) corresponded to one of these \\ncell types. For the CD subset, a spot was selected if the sum of level 1 \\nmapping proportions for the PC and IC contributed most to its signa-\\nture. POD spots were defined by the presence of a minimum of 20% \\nsignature arising from the level 1 POD label. Once the subsets of PT, \\nTAL, fibroblast, CD and POD spots were selected, each spot was fur-\\nther divided into healthy, adaptive or degenerative cell state groups \\nbased on the highest proportion of cell state signature as calculated \\nin Fig.\\xa03a. For PODs, the presence of EC-GC signature was considered \\nto be a degenerative equivalent given that a loss of POD markers was \\nassociated with an observed gain in EC-GC signatures within DKD  \\nsamples.\\nNiche analysis. T o examine the diversity of cell types colocalizing with \\nTAL epithelial cells, we selected spots with more than 20% TAL signa-\\nture and in which the highest proportion of signature arose from level \\n1 TAL mapping. Using Seurat clustering methodology, selected spots \\nwere reclustered after Seurat label transfer scores were substituted in \\nlieu of gene expression. Spots with similar proportions of signature \\narising from TAL cell types and states, stromal cells and immune cells \\nwere clustered into 13 niches. Niches were mapped over the 23 kidney \\nsamples and the marker gene expression in each niche was determined. \\nT o depict the relative proportion of each cell type, the transfer score \\naverage was first computed in each niche. Next, a z score for each cell \\ntype was calculated across the niches.\\nHistological validation. T o determine whether the 74 snCv3 subclasses \\n(level 2) were appropriately mapped to histological structures, the \\nproportion of signature in each spot was compared to a histologically \\nvalidated set of six unsupervised clusters defined by Space Ranger102 \\n(Extended Data Fig.\\xa05a). These six unsupervised clusters (glomerulus, \\nPT, loop of Henle, distal convoluted tubule, connecting tubule and col-\\nlecting duct, and the interstitium) had an overall alignment of 97.6% \\nwith the underlying histopathologic structures in the H&E image. In \\neach sample, regions of cortex and medulla were defined by histological \\nevaluation, including the presence of glomeruli. Of the 23 samples, 18 \\nsamples were composed of only cortex, 4 samples were a combination \\nof cortex and medulla and 1 sample was completely medulla.\\nLabel-free and multifluorescence large-scale 3D imaging\\nKidney biopsy cores frozen in OCT from patients with AKI or CKD \\nenrolled in KPMP were used for label-free imaging followed by \\nmultiplexed-fluorescence large-scale 3D imaging as outlined in the pro-\\ntocol (https://doi.org/10.17504/protocols.io.9avh2e6) and described in \\na recent publication27. Frozen biopsies were sectioned to a thickness of \\n50\\u2009µm using a cryostat and then immediately fixed in 4% fresh paraform-\\naldehyde (PFA) for 24\\u2009h and subsequently stored at 4\\u2009°C in 0.25% PFA.\\nThe first step in imaging consists of label-free imaging with mul-\\ntiphoton microscopy to collect autofluorescence and second harmonic \\nimages of the unlabelled tissue mounted in non-hardening mounting \\nmedium. Imaging was conducted using a Leica SP8 confocal scan-head \\nmounted to an upright DM6000 microscope. For large-scale imaging \\nof tissues at the sub-micrometer resolution, the Leica Tile Scan func-\\ntion was used to collect a mosaic of smaller image volumes using a \\nhigh-power, high-numerical aperture objective. Leica LASX software \\n(v.3.5) was then used to stitch these component volumes into a single \\nimage volume of the entire sample. The scanner zoom and focus motor \\ncontrol were set to provide voxel dimensions of 0.5\\u2009×\\u20090.5\\u2009µm laterally \\nand 1\\u2009µm axially.\\nLabelling of tissue for fluorescence microscopy was preceded by \\nwashing in phosphate-buffered saline (PBS) and blocking with PBS \\nwith 0.1% Triton X-100 (MP Biomedical) and 10% normal donkey serum \\n( Jackson Immuno Research). Antibodies for indirect immunofluores-\\ncence were applied first for 8–16\\u2009h at room temperature, followed by \\nwashing cycles of PBS with 0.1% Triton X-100. An incubation cycle with \\nsecondary antibodies occurred next, followed by washing and finally \\napplication of directly labelled antibodies. Antibodies targeting mark-\\ners for tubular cells and structures (aquaporin-1, uromodulin, F-actin) \\nand immune cells (myeloperoxidase, CD68, CD3, siglec 8) were used, \\nin addition to nuclei labelling using DAPI (Supplementary Table\\xa035). \\nAfter the final washing cycles, the tissue was mounted in Prolong Glass \\n(Thermo Fisher Scientific).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c83d1a3a-0723-4dbe-9f93-eba5b08c39b0', embedding=None, metadata={'page_label': '603', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Confocal microscopy was conducted using a Leica ×20/0.75\\u2009NA \\nmulti-immersion objective (adjusted for oil immersion), with excita-\\ntion sequentially provided by a solid-state laser launch with laser lines \\nat 405\\u2009nm, 488\\u2009nm, 552\\u2009nm and 635\\u2009nm. Images in 16 channels (emission \\nspectra collected by PMT detectors adjusted for the following ranges: \\n410–430\\u2009nm, 430–450\\u2009nm, 450–470\\u2009nm, 470–490\\u2009nm, 500–509\\u2009nm, \\n510–519\\u2009nm, 520–530\\u2009nm, 530–540\\u2009nm, 570–590\\u2009nm, 590–610\\u2009nm, \\n610–630\\u2009nm, 631–651\\u2009nm, 643–664\\u2009nm, 664–685\\u2009nm, 685–706\\u2009nm \\nand 706–726\\u2009nm) were collected for each focal plane of each panel \\nof the 3D mosaic. The resulting 16-channel image was then spectrally \\ndeconvolved (by linear unmixing using the Leica LASX linear unmixing \\nsoftware) to discriminate the eight fluorescent probes in the sample. \\nValidation of the linear unmixing was described previously27.\\nConfocal immunofluorescence microscopy\\nHuman kidney tissue samples from the cortex or medulla were fixed in \\n4% PFA, cryopreserved in 30% sucrose and frozen in OCT cryomolds, \\nand were cut into 5\\u2009µm sections. The sections were post-fixed with 4% \\nPFA for 15\\u2009min at room temperature, blocked in blocking buffer (1% BSA, \\n0.2% skimmed milk, 0.3% Triton X-100 in 1× PBS) for 30\\u2009min at room tem-\\nperature and then immunofluorescence microscopy was performed, \\nfirst by overnight incubation at 4\\u2009°C with primary antibodies, fol -\\nlowed by labelling with secondary antibodies. The primary antibodies \\nincluded NRXN-1β, TUJ1, collagen I and III, synapsin-1, NPSH-1, SLC14A2, \\nUMOD, CD31, CD34, CD11b, PROM1, KIM1, VCAM1, AQP1, AQP2, CD45 \\nand S100 (Supplementary Table\\xa036). After washing, labelling with the \\nsecondary antibodies was performed using Alexa-488-conjugated goat \\nanti-mouse IgG, Cy3-conjugated goat anti-rabbit IgG or Cy5-conjugated \\ndonkey anti-goat IgG at room temperature for 1\\u2009h. After washing, the \\nsections were counterstained with DAPI for nuclear staining. Images \\nwere acquired with a Nikon 80i C1 confocal microscope.\\nIn\\xa0situ hybridization\\nHuman kidney tissues were sectioned with 3\\u2009µm from formalin-fixed, \\nparaffin-embedded (FFPE) blocks. In\\xa0situ detections of PROM1, CST3 \\nand EGF mRNA transcripts were performed with the use of RNAscope \\nProbes Hs-PROM1 (311261, Advanced Cell Diagnostics), Hs-CST3 \\n(528181, Advanced Cell Diagnostics), and Hs-EGF (605771, Advanced \\nCell Diagnostics) and RNAscope kit (322330, Advanced Cell Diagnos-\\ntics) according to the manufacturer’s protocol. RNAscope Positive \\nControl Probe Hs-UBC (310041, Advanced Cell Diagnostics) was used \\nas a positive control. A horseradish-peroxidase-based signal amplifica-\\ntion system (322310, RNAscope 2.0 HD Detection Kit-Brown, Advanced \\nCell Diagnostics) was used to hybridize with target probes followed by \\nDAB staining. The sections were then counterstained with haematoxy-\\nlin (3535-16, RICCA Chemical Company). Positive staining was deter-\\nmined by brown dots. After rehydrating, the sections were immersed \\nin periodic acid solution (0.5%, P7875, Sigma-Aldrich) for 5\\u2009min, rinsed \\nin three changes of distilled water, incubated with Schiff’s reagent \\n(3952016, Sigma-Aldrich) for 15\\u2009min and then rinsed in running tap water \\nfor 5\\u2009min. Nuclei were counterstained with haematoxylin 2 (220-102, \\nThermo Fisher Scientific) for 2\\u2009min. The sections were then rinsed in \\nwater, dehydrated in alcohol, cleared in xylene and finally mounted \\nwith Cytoseal XYL (8312-4, Thermo Fisher Scientific).\\nTissue cytometry and in\\xa0situ cell classification\\nTissue cytometry and analysis were conducted using the Volumetric \\nTissue Exploration and Analysis (VTEA) software (v.1.0a-r9). VTEA is \\na 3D image processing workspace that was developed as a plug-in for \\nImageJ105. The version of VTEA, which includes the supervised and \\nunsupervised labelling of cells and combining spatial and features \\nbased gating strategies, used here is available at GitHub (https://github.\\ncom/icbm-iupui/volumetric-tissue-exploration-analysis) and through \\nthe FIJI updater. In this analytical pipeline, each individual nucleus was \\nsegmented using intensity thresholding and connected component \\nsegmentation built into VTEA and ImageJ. Each surveyed nucleus \\nbecame a surrogate for a cell, to which the location and marker staining \\naround or within the nucleus could be registered. This captured infor-\\nmation could be used to classify cells on the basis of marker intensity \\nor spatial features using scatterplot displays that enable various gating \\nstrategies and statistical analysis, including export as .csv files of all \\nsegmented cells and the associated features106. Cells classified on the \\nbasis of marker intensity are summarized in Supplementary Table\\xa037. \\nGated cells were mapped back directly into the image volumes, which \\nenabled immediate validation of the gates. Moreover, direct gating \\non the image could be performed, which could trace all of the cells \\nwithin the chosen region-of-interest back to the data display on the \\nscatter plot. Thus, cell classification could also be performed based \\non direct annotation of regions-of-interest (ROIs) within the image \\nvolumes. Annotated ROIs were determined by the pixel-wise agree-\\nment between 3 of 4 experts who performed annotation on each biopsy \\nspecimen separately.\\nUsing tissue cytometry, 14 cell classes were defined based on the \\nfollowing features: (1) PT cells: AQP1+ cells in cortex ± brush border \\nstaining. (2) C-TAL cells: UMOD+ cells in cortex. (3) Glomerular cells \\n(which encompass PODs, glomerular endothelium and mesangial cells) \\nannotated ROIs based on morphology and F-actin staining. (4) Cortical \\nlarge and medium vessel cells: annotated ROIs based on morphology \\nand F-actin staining. (5) Cortical distal nephron cells (distal tubules \\n(CD), connecting tubules (CNT) and collecting ducts (C-CD) or cortical \\ndistal nephrons): AQP1−UMOD− and annotated ROIs based on unique \\nmorphology in cortex. (6) M-TAL cells: UMOD+ cells in the medulla. \\n(7) DTL: AQP1+ cells in the medulla. (8) Medullary collecting ducts: \\nAQP1−UMOD− and annotated ROIs based on unique morphology in the  \\nmedulla. (9) Vascular bundles in the medulla: annotated ROIs based \\nonunique morphology in the medulla and F-actin staining. (10) Neu-\\ntrophils: MPO+ cells. (11) Activated macrophages: MPO−CD68+ cells.  \\n(12) T\\xa0cells: CD3+ cells. (13) Cells in altered regions: annotated ROIs \\nbased on loss of (unrecognizable) tubular morphology, expanded inter-\\nstitium, increased fibrosis (by second harmonic generation imaging)  \\nand cell infiltrates. (14) Not determined: unable to be classified on the \\nbasis of the above criteria.\\nUsing such an approach,1,540,563 cells were labelled from all the \\nbiopsies used in this analysis.\\n3D neighbourhood building and representation\\n3D neighbourhoods were calculated for every cell in each biopsy \\nusing VTEA and a radius of 25\\u2009µm (50 voxels in x and y and 25 voxels \\nin z). We reasoned the largest measurable neighbourhood/niche in \\nour 3D approach is limited by the 50\\u2009µm thickness of the sections \\nimaged (z dimension). Thus, per Nyquist sampling, the radius used \\nwas about 25\\u2009µm, which is consistent with previous approaches107–109. \\nFor each 3D neighbourhood, VTEA was used to calculate the features: \\nfraction-of-total and sum of each labelled cell per neighbourhood. A \\nlist of neighbourhoods, positions in 3D and their features was exported \\nby biopsy sample as .csv files.\\nNeighbourhood visualization and statistical analysis\\nCSV files of neighbourhoods by biopsy sample were generated in VTEA \\nand imported into R (v.4.0.4), parsed for the sum of each labelled cell \\nand monotypic neighbourhoods removed. These features were scaled \\nby Z-standardization and used for Louvain community detection  \\n(R packages FNN (v.1.1.3) and igraph (v.1.2.6)) and t -SNE manifold \\nprojection (R package Rtsne (v.0.15)). T o understand the interactions  \\nwithin neighbourhoods, pairwise interactions by neighbourhood  \\nwere tallied and plotted on a chord plot (R package: circlize (v.0.4.12)) \\nand Pearson’s correlation coefficients were calculated and plotted  \\n(R packages Hmisc (v.4.5.0) and corrplot (v.0.84)). Subclasses of neigh-\\nbourhoods, those with at least one cell with a specific label were selected  \\nand plotted as network plots (R package igraph (v.1.2.6)) with edges in ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='89edab8f-0a5f-4ba9-9ca6-00bea6564a21', embedding=None, metadata={'page_label': '604', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nCD3 and Altered neighbourhoods scaled at 40% of all other subclasses \\nto facilitate visualization. All scripts are provided as an annotated  \\nRStudio notebook file (.rmd).\\nPlots and figures\\nUMAP, feature, dot and violin plots for snCv3, scCv3, SNARE2 and \\nVisium data were generated using Seurat. Correlation plots were \\ngenerated using the corrplot package. Genome coverage plots were \\nperformed using Signac. Plots for 3D cytometry and neighbourhood \\nanalysis were generated in R with circlize, ggplot2 and igraph.\\nReporting summary\\nFurther information on research design is available in the\\xa0Nature  \\nPortfolio Reporting Summary linked to this article.\\nData availability\\nProcessed data, interactive and visualization tools: the snCv3, scCv3, \\nSNARE2, Slide-seq and Visium processed data files are all available \\nfor download at the GEO (Superseries GSE183279). snCv3 healthy \\nreference data are available for reference-based single-cell mapping \\nusing the Azimuth tool (https://azimuth.hubmapconsortium.org/). \\nAll snCv3 and scCv3 processed data can be accessed and viewed at \\ncellxgene (https://cellxgene.cziscience.com/collections/bcb61471-\\n2a44-4d00-a0af-ff085512674c). snCv3 (excluding COVID-AKI and CKD \\nnephrectomy samples), scCv3, Visium (KPMP biopsies) and 3D imaging \\ncan all be visualized and examined using the KPMP Data Atlas Explorer \\n(https://atlas.kpmp.org/explorer/). For 3D imaging, the cytometry data, \\ncell classifications, gates and neighbourhood analysis data are available \\nat Zenodo (https://doi.org/10.5281/zenodo.7120941). Raw sequencing \\nand imaging data: raw sequencing data are under controlled access \\n(human data) as they are potentially identifiable and can be accessed \\nfrom the respective sources indicated below (summarized in Supple-\\nmentary Tables\\xa01 and 2). Raw and processed sequencing and imaging \\ndata (snCv3, scCv3, 3D imaging and Visium) generated as part of the \\nKPMP have been deposited (https://atlas.kpmp.org/repository/) and \\ncompiled (https://doi.org/10.48698/3z31-8924) online. 3D imaging \\nraw data are freely available to download; however, KPMP raw sequenc-\\ning data (snCv3, scCv3, Visium) have restricted access. These can be \\nrequested from KPMP by contacting A.L.D. (info@kpmp.org) and are \\navailable by signing a data use agreement (DUA) promising to abide \\nby KPMP security standards and to not re-identify participants, share \\ndata outside those named on the DUA Exhibit A or sell the data. Data \\naccess is granted to anyone signing the KPMP DUA as is. KPMP will \\nrespond to initial data requests within 12–36\\u2009h and provide data up to \\none month after the DUA has been signed. Manuscripts resulting from \\nKPMP controlled access data are requested to go through the KPMP \\npublications and presentations (P&P) committee to ensure that KPMP is \\nacknowledged appropriately and authorship follows ICJME standards. \\nThe KPMP P&P committee reviews and approves manuscripts every \\n2\\u2009weeks and, to date, no manuscript has been rejected. Any analysis \\nresulting from KPMP data may be published or shared provided that \\nit does not re-identify KPMP participants. Slide-seq raw sequencing \\ndata generated as part of KPMP pilot nephrectomy tissue are available \\nfor download from the GEO (Superseries GSE183279). Raw sequencing \\ndata (snCv3, SNARE2, Slide-seq) generated as part of the Human Bio-\\nmolecular Atlas Project (HuBMAP) have been deposited (https://portal.\\nhubmapconsortium.org/) and compiled (https://doi.org/10.35079/\\nhbm776.rgsw.867) online. The HuBMAP raw sequencing data have \\nrestricted access and are available for download from the database \\nof Genotypes and Phenotypes (dbGaP: phs002249) by requesting for \\nauthorized access following instructions on the dbGaP website. The \\nprocess to request access to this dbGaP study is available online (https://\\ndbgap.ncbi.nlm.nih.gov/aa/wga.cgi?adddataset=phs002249&page=\\nlogin). In brief, to download the human sequencing data for this study \\nafter obtaining authorization from the NIH DAC, one would go through \\nthe SRA (https://www.ncbi.nlm.nih.gov/bioproject/PRJNA671343). \\nsnCv3 data not deposited to KPMP or HuBMAP are available from the \\nGEO (GSE183279) or, for Covid AKI raw sequencing files, on request \\nfrom Washington University Kidney Translational Research Center \\n(KTRC) to S.J. (sanjayjain@wustl.edu) due to patient confidentiality. \\nResponse to requests or questions will be addressed within a week. \\nCode access and data use agreement forms can be accessed online \\n(https://research.wustl.edu/core-facilities/ktrc/). Once the executed \\nform is received and approved, data will be distributed within a month. \\nThere is no authorship restriction on the use of COVID data. Additional \\npublished/public datasets: the following publicly available RNA-seq \\ndatasets were used in this study: mouse kidney single cell (GEO: \\nGSE129798); mouse kidney injury single nucleus (GEO: GSE139107); \\nhuman fibroblast and myofibroblast single cell (Zenodo: https://doi.\\norg/10.5281/zenodo.4059315); mouse distal nephron single cell and \\nbulk distal segment (GEO: GSE150338); human kidney mature immune \\nsingle cell (https://kidney-atlas.cells.ucsc.edu); and human kidney \\nsingle nucleus (GEO: GSE151302; https://human-kidney-atac.cells.ucsc.\\nedu). GWAS summary statistics were from the CKDGen Consortium \\n(all eGFR; https://ckdgen.imbi.uni-freiburg.de/files/Wuttke2019), EBI \\nGWAS Catalog (hypertension; https://www.ebi.ac.uk/gwas/efotraits/\\nEFO_0000537) and the CausalDB database (release 1.1 2019-09-29; \\nhttp://www.mulinlab.org/causaldb). NEPTUNE sequencing and clinical \\ndata were obtained from NEPTUNE. Owing to patient confidentiality, \\nthese data have restricted access and are available only on request to \\nNEPTUNE-STUDY@umich.edu. ERCB data were obtained from the GEO \\n(GSE104954). Raw sequencing data (scCv3) on living donor biopsies as \\npart of the Chan Zuckerberg Initiative (CZI) and HCA are available from \\nthe GEO (GSE169285). Additional Visium spatial transcriptomic data \\nnot in the KPMP repository are available from the GEO (GSE171406). \\nFigures: schemata of the human nephron and renal corpuscle were \\ndeveloped by the KPMP and HuBMAP (https://doi.org/10.48698/DEM4-\\n0Q93).\\xa0Source data are provided with this paper.\\nCode availability\\nCode to reproduce figures are available to download from GitHub \\n(https://github.com/KPMP/Cell-State-Atlas-2022). No additional  \\ncustom computational code was generated in this study.\\n \\n50. Eadon, M. T. et\\xa0al. Kidney histopathology and prediction of kidney failure: a retrospective \\ncohort study. Am. J. Kidney Dis. 76, 350–360 (2020).\\n51. Martin, M. Cutadapt removes adapter sequences from high-throughput sequencing \\nreads. EMBnet J. 17, 10–12 (2011).\\n52. Petukhov, V. et\\xa0al. dropEst: pipeline for accurate estimation of molecular counts in droplet- \\nbased single-cell RNA-seq experiments. Genome Biol. 19, 78 (2018).\\n53. Dobin, A. et\\xa0al. STAR: ultrafast universal RNA-seq aligner. Bioinformatics 29, 15–21 (2013).\\n54. Fang, R. et\\xa0al. Comprehensive analysis of single cell ATAC-seq data with SnapATAC. Nat. \\nCommun. 12, 1337 (2021).\\n55. Li, H. Minimap2: pairwise alignment for nucleotide sequences. Bioinformatics 34,  \\n3094–3100 (2018).\\n56. Gayoso, A., Shor, J., Carr, A. J., Sharma, R. & Pe’er, D. GitHub: DoubletDetection (2019); \\nhttps://doi.org/10.5281/zenodo.2678042.\\n57. Young, M. D. & Behjati, S. SoupX removes ambient RNA contamination from droplet- \\nbased single-cell RNA sequencing data. Gigascience 9, giaa151 (2020).\\n58. Chen, L., Chou, C.-L. & Knepper, M. A. A comprehensive map of mRNAs and their isoforms \\nacross all 14 renal tubule segments of mouse. J. Am. Soc. Nephrol. 32, 897–912 (2021).\\n59. Ransick, A. et\\xa0al. Single-cell profiling reveals sex, lineage, and regional diversity in the \\nmouse kidney. Dev. Cell 51, 399–413 (2019).\\n60. Börner, K. et\\xa0al. Anatomical structures, cell types and biomarkers of the Human Reference \\nAtlas. Nat. Cell Biol. 23, 1117–1128 (2021).\\n61. Chen, L., Chou, C.-L. & Knepper, M. A. Targeted single-cell RNA-seq identifies minority \\ncell types of kidney distal nephron. J. Am. Soc. Nephrol. 32, 886–896 (2021).\\n62. Monaco, G. et\\xa0al. RNA-seq signatures normalized by mRNA abundance allow absolute \\ndeconvolution of human immune cell types. Cell Rep. 26, 1627–1640 (2019).\\n63. Hao, Y. et\\xa0al. Integrated analysis of multimodal single-cell data. Cell 184, 3573–3587 (2021).\\n64. Barkas, N. et\\xa0al. Joint analysis of heterogeneous single-cell RNA-seq dataset collections. \\nNat. Methods 16, 695–698 (2019).\\n65. Luecken, M. D. et\\xa0al. Benchmarking atlas-level data integration in single-cell genomics. \\nNat. Methods 19, 41–50 (2022).\\n66. Aevermann, B. D. et\\xa0al. Cell type discovery using single-cell transcriptomics: implications \\nfor ontological representation. Hum. Mol. Genet. 27, R40–R47 (2018).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='00781740-19f5-4886-b633-b63b25edcb8f', embedding=None, metadata={'page_label': '605', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='67. Aran, D. et\\xa0al. Reference-based analysis of lung single-cell sequencing reveals a transitional \\nprofibrotic macrophage. Nat. Immunol. 20, 163–172 (2019).\\n68. Heng, T. S. P., Painter, M. W. & Immunological Genome Project Consortium. The \\nImmunological Genome Project: networks of gene expression in immune cells. Nat. \\nImmunol. 9, 1091–1094 (2008).\\n69. Takemon, Y. et\\xa0al. Proteomic and transcriptomic profiling reveal different aspects of aging \\nin the kidney. eLife 10, e62585 (2021).\\n70. Ruscetti, M. et\\xa0al. NK cell-mediated cytotoxicity contributes to tumor control by a cytostatic \\ndrug combination. Science 362, 1416–1422 (2018).\\n71. Basisty, N. et\\xa0al. A proteomic atlas of senescence-associated secretomes for aging \\nbiomarker development. PLoS Biol. 18, e3000599 (2020).\\n72. Korotkevich, G. et\\xa0al. Fast gene set enrichment analysis. Preprint at bioRxiv https://doi.\\norg/10.1101/060012 (2016).\\n73. Luo, W., Friedman, M. S., Shedden, K., Hankenson, K. D. & Woolf, P. J. GAGE: generally \\napplicable gene set enrichment for pathway analysis. BMC Bioinform. 10, 161 (2009).\\n74. Stuart, T. et\\xa0al. Single-cell chromatin state analysis with Signac. Nat. Methods 18, 1333–1341 \\n(2021).\\n75. Schep, A. N., Wu, B., Buenrostro, J. D. & Greenleaf, W. J. chromVAR: inferring transcription- \\nfactor-associated accessibility from single-cell epigenomic data. Nat. Methods 14, 975–978 \\n(2017).\\n76. Bravo González-Blas, C. et\\xa0al. cisTopic: cis-regulatory topic modeling on single-cell  \\nATAC-seq data. Nat. Methods 16, 397–400 (2019).\\n77. Pliner, H. A. et\\xa0al. Cicero predicts cis-regulatory DNA interactions from single-cell chromatin \\naccessibility data. Mol. Cell 71, 858–871 (2018).\\n78. Yu, G., Wang, L.-G. & He, Q.-Y. ChIPseeker: an R/Bioconductor package for ChIP peak \\nannotation, comparison and visualization. Bioinformatics 31, 2382–2383 (2015).\\n79. Wu, Y., Tamayo, P. & Zhang, K. Visualizing and interpreting single-cell gene expression \\ndatasets with similarity weighted nonnegative embedding. Cell Syst. 7, 656–666  \\n(2018).\\n80. Jin, S. et\\xa0al. Inference and analysis of cell-cell communication using CellChat. Nat. \\nCommun. 12, 1088 (2021).\\n81. Ulirsch, J. C. et\\xa0al. Interrogation of human hematopoiesis at single-cell and single-variant \\nresolution. Nat. Genet. 51, 683–693 (2019).\\n82. Wang, J. et\\xa0al. CAUSALdb: a database for disease/trait causal variants identified using \\nsummary statistics of genome-wide association studies. Nucleic Acids Res. 48,  \\nD807–D816 (2020).\\n83. Chen, W. et\\xa0al. Fine mapping causal variants with an approximate bayesian method using \\nmarginal test statistics. Genetics 200, 719–736 (2015).\\n84. Canela-Xandri, O., Rawlik, K. & Tenesa, A. An atlas of genetic associations in UK Biobank. \\nNat. Genet. 50, 1593–1599 (2018).\\n85. Tin, A. et\\xa0al. Target genes, variants, tissues and transcriptional pathways influencing \\nhuman serum urate levels. Nat. Genet. 51, 1459–1474 (2019).\\n86. Wuttke, M. et\\xa0al. A catalog of genetic loci associated with kidney function from analyses \\nof a million individuals. Nat. Genet. 51, 957–972 (2019).\\n87. Zhu, Z. et\\xa0al. Genetic overlap of chronic obstructive pulmonary disease and cardiovascular \\ndisease-related traits: a large-scale genome-wide cross-trait analysis. Respir. Res. 20, 64 \\n(2019).\\n88. Li, Y. et\\xa0al. Integration of GWAS summary statistics and gene expression reveals  \\ntarget cell types underlying kidney function traits. J. Am. Soc. Nephrol. 31, 2326–2340 \\n(2020).\\n89. Sheng, X. et\\xa0al. Mapping the genetic architecture of human traits to cell types in the \\nkidney identifies mechanisms of disease and potential treatments. Nat. Genet. 53,  \\n1322–1333 (2021).\\n90. Cippà, P. E. et\\xa0al. Transcriptional trajectories of human kidney injury progression. JCI \\nInsight 3, e123151 (2018).\\n91. Barisoni, L. et\\xa0al. Digital pathology evaluation in the multicenter Nephrotic Syndrome \\nStudy Network (NEPTUNE). Clin. J. Am. Soc. Nephrol. 8, 1449–1459 (2013).\\n92. Levey, A. S. et\\xa0al. GFR decline as an end point for clinical trials in CKD: a scientific workshop \\nsponsored by the National Kidney Foundation and the US Food and Drug Administration. \\nAm. J. Kidney Dis. 64, 821–835 (2014).\\n93. Tao, J. et\\xa0al. JAK-STAT signaling is activated in the kidney and peripheral blood cells of \\npatients with focal segmental glomerulosclerosis. Kidney Int. 94, 795–808 (2018).\\n94. Street, K. et\\xa0al. Slingshot: cell lineage and pseudotime inference for single-cell \\ntranscriptomics. BMC Genom. 19, 477 (2018).\\n95. Zhang, B. & Horvath, S. A general framework for weighted gene co-expression network \\nanalysis. Stat. Appl. Genet. Mol. Biol. 4, Article17 (2005).\\n96. Jassal, B. et\\xa0al. The reactome pathway knowledgebase. Nucleic Acids Res. 48, D498–D503 \\n(2020).\\n97. La Manno, G. et\\xa0al. RNA velocity of single cells. Nature 560, 494–498 (2018).\\n98. Bergen, V., Lange, M., Peidli, S., Wolf, F. A. & Theis, F. J. Generalizing RNA velocity to transient \\ncell states through dynamical modeling. Nat. Biotechnol. 38, 1408–1414 (2020).\\n99. Stickels, R. et\\xa0al. Library generation using slide-seqV2 v2. Protocols.io https://doi.org/ \\n10.17504/protocols.io.bvv6n69e (2021).\\n100. Dries, R. et\\xa0al. Giotto: a toolbox for integrative analysis and visualization of spatial expression \\ndata. Genome Biol. 22, 78 (2021).\\n101. Cable, D. M. et\\xa0al. Robust decomposition of cell type mixtures in spatial transcriptomics. \\nNat. Biotechnol. 40, 517–526 (2021).\\n102. Melo Ferreira, R. et\\xa0al. Integration of spatial and single cell transcriptomics localizes \\nepithelial-immune cross-talk in kidney injury. JCI Insight 6, e147703 (2021).\\n103. Janosevic, D. et\\xa0al. The orchestrated cellular and molecular responses of the kidney to \\nendotoxin define a precise sepsis timeline. eLife 10, e62270 (2021).\\n104. Melo Ferreira, R., Freije, B. J. & Eadon, M. T. Deconvolution tactics and normalization in \\nrenal spatial transcriptomics. Front. Physiol. 12, 812947 (2021).\\n105. Winfree, S. et\\xa0al. Quantitative three-dimensional tissue cytometry to study kidney tissue \\nand resident immune cells. J. Am. Soc. Nephrol. 28, 2108–2118 (2017).\\n106. Winfree, S. et\\xa0al. Large-scale 3-dimensional quantitative imaging of tissues: \\nstate-of-the-art and translational implications. Transl. Res. 189, 1–12 (2017).\\n107. Stoltzfus, C. R. et\\xa0al. CytoMAP: a spatial analysis toolbox reveals features of myeloid cell \\norganization in lymphoid tissues. Cell Rep. 31, 107523 (2020).\\n108. Stoltzfus, C. R. et\\xa0al. Multi-parameter quantitative imaging of tumor microenvironments \\nreveals perivascular immune niches associated with anti-tumor immunity. Front. Immunol. \\n12, 726492 (2021).\\n109. Leal, J. M. et\\xa0al. Innate cell microenvironments in lymph nodes shape the generation of \\nT\\xa0cell responses during type I inflammation. Sci. Immunol. 6, eabb9435 (2021).\\nAcknowledgements We thank the patients who volunteered to donate tissue primarily for \\nresearch purposes despite no direct immediate benefit to their clinical care; collaborators  \\nL. J. Arend, S. M. Bagnasco, C. J. Sperrati and M. Atta for providing samples from patients with \\nCOVID-AKI; Y. Yang for his expertise in performing the in\\xa0situ hybridization experiments; the \\nmembers of the KPMP nomenclature working group for establishing the nephron schema and \\nstandardized nomenclature used, specifically, T. Valerius, W. Kriz, B. Kaissling and M. Rose; the \\nmembers of the HuBMAP consortium including the HuBMAP HIVE for building the infrastructure \\nfor data storage and public access to reference samples, in particular, J. Silverstein, P. Kant,  \\nK. Borner and N. Gehlenborg; J. Hilton and J. Yu-Sheng Chien for building the CellxGene instance \\nof the integrated datasets as part of the Chan Zuckerburg Initiative; the staff at the Indiana \\nCenter for Biological Microscopy for imaging assistance; the staff at the Indiana Center for \\nMedical Genomics for sequencing of the 10x Visium samples; the staff at the Indiana Biobank \\nfor hosting the BBCI; the staff at the Kidney Translational Research Center (KTRC) at the \\nWashington University (Division of Nephrology) and Mid America Transplant in St Louis for \\ninfrastructural support for HuBMAP samples; and the S. Steffan family for support of new \\nomics technologies in kidney research (B1401-40 to S.J.). This publication is part of the HCA \\n(https://humancellatlas.org/publications). The snCv3 and SNARE2 sequencing data were \\ngenerated at the UC San Diego IGM Genomics Center supported by the National Institutes of \\nHealth (SIG grant S10 OD026929) and Washington University Genome Technology Access \\nCenter at the McDonnell Genome Institute partially supported by NCI Cancer Center Support \\n(P30 CA91842) to the Siteman Cancer Center and by ICTS/CTSA (UL1TR002345) from the \\nNational Center for Research Resources (NCRR). The KPMP data presented here were funded \\nby the following grants from the NIDDK: U2C DK114886, UH3DK114861, UH3DK114866, \\nUH3DK114870, UH3DK114908, UH3DK114915, UH3DK114926, UH3DK114907, UH3DK114923  \\nand UH3DK114933. The HuBMAP data presented here were supported by OT2D033760  \\n(to R.S.), U54HL145608 (to S.J. and K.Z.) and U54DK134301 (to S.J.). Additional NIH support \\nwas provided by NIH/NIDDK K08DK107864 (to M.T.E.); Indiana Grand Challenge Precision \\nHealth fund (to R.M.F.); R01DK111651 (to T.M.E.-A.); P30 DK079312 (to T.M.E.-A. and P.C.D.);  \\nP01 DK056788 (to J.C.W.); U2CDK114886, U54DK083912, P30 DK081943/HCA: Kidney  \\nSeed Network (to M.K.); K23DK125529 (to A.S.N.); U54DK083912 (to S.E.); U01MH114828  \\n(to K. Kalhor); UH3CA246632 (to E.Z.M.). The content is solely the responsibility of the authors \\nand does not necessarily represent the official views of the National Institutes of Health.\\nAuthor contributions Coordination of manuscript writing and project: B.B.L. and S.J. \\nContribution to patient recruitment and tissue collection: A.K., A.S.N., C.R.P., D.S., E.H.K.,  \\nF.P.W., J.C.W., J.R.S., K. Kiryluk, M.K., P.M.P., R.D.T., S.J., S.R. and S.S.W. Contribution to tissue \\nprocessing: A.K., A.S.N., D.B., D.S., E.A.O., J.R.S., M.F., M.K., M.T.E., P.C.D., S.J., S.R., S.W. and \\nT.M.E.-A. Contribution to RNA data generation: A.G., A.S.N., B.B.L., D.D., E.A.O., E.M., E.Z.M., \\nF.C., J.B.H., J.L.M., K.S.B., K.Z., M.K., M.T.E., N.P., P.C.D., R.M., S.J., S.U. and T.M.E.-A. Contribution \\nto imaging data generation: B.Z., D.B., J.B.H., M.F., M.T.E., P.C.D., S.J., S.W. and T.M.E.-A. \\nContribution to ATAC data generation: B.B.L., D.D., K.Z., N.P. and S.J. Contribution to data \\narchive/infrastructure: B.B.L., D.D., K.S.B., M.F., M.K., M.T.E., P.C.D., Q.H., R.M., R.M.F., S.W., \\nT.M.E.-A. and X.W. Contribution to data analysis: B.B.L., D.B., E.A.O., J.B.H., J.P.G., K. Kalhor, K.Z., \\nL.H.M., M.F., M.K., M.T.E., P.C.D., P.V.K., Q.H., R.M., R.M.F., S.E., S.J., S.W., T.M.E.-A., T.N., X.W. and \\nY.W. Contribution to the Azimuth tool: A.H. and R.S. Contribution to data interpretation: A.S.N., \\nA.V., B.B.L., E.A.O., J.B.H., J.P.G., K. Kalhor, K.Z., L.H.M., M.F., M.K., M.T.E., P.C.D., P.J.H., P.V.K., \\nQ.H., R.M., R.M.F., S.E., S.J., S.W. and T.M.E.-A. Contribution to writing the manuscript: B.B.L.,  \\nK. Kalhor, K.Z., M.K., M.T.E., P.C.D., P.V.K., Q.H., R.M., R.M.F., S.J., S.W. and T.M.E.-A.\\nCompeting interests P.V.K. serves on the scientific advisory board to Celsius Therapeutics  \\nand Biomage. A.V. is a consultant for Astute and NxStage. C.R.P. is a member of the advisory \\nboard of and owns equity in RenalytixAI, and serves as a consultant for Genfit and Novartis. \\nM.K. has grants from JDRF, Astra-Zeneca, NovoNordisc, Eli Lilly, Gilead, Goldfinch Bio, Janssen, \\nBoehringer-Ingelheim, Moderna, European Union Innovative Medicine Initiative, Chan \\nZuckerberg Initiative, Certa, Chinook, amfAR, Angion Pharmaceuticals, RenalytixAI, Travere \\nTherapeutics, Regeneron, IONIS Pharmaceuticals, Astellas, Poxel and a patent (PCT/\\nEP2014/073413; ‘Biomarkers and methods for progression prediction for chronic kidney \\ndisease’) licensed. F.C. and E.Z.M. are paid consultants for Atlas Bio. F.P.W. receives research \\nsupport from Astrazeneca, Boeringher-Ingelheim, Vifor Pharma and Whoop. P.M.P. is a \\nconsultant for Janssen. S.R. has research funding from AstraZeneca and Bayer Healthcare. \\nS.S.W. is a consultant for GSK, GEHC, JNJ, Strataca, Roth Capital Partners, Venbio, and an \\nexpert witness on litigation for Davita and Pfizer. J.R.S. consults for Maze and Goldfinch and \\nreceives royalties from Sanfi Genzyme. K.Z. is a co-founder, equity holder and serves on the \\nscientific advisory board of Singlera Genomics. A.S.N. is on the external advisory board for \\nCareDX. L.H.M. is a consultant for Reata Pharmaceuticals, Travere Therapeutics and Calliditas. \\nS.J. is a paid Blue SKy mentor for Meharry Medical College, Nashville and receives royalties \\nfrom Elsevier. J.L.M. is an employee and shareholder of Solid Biosciences. The other authors \\ndeclare no competing interests.\\nAdditional information\\nSupplementary information The online version contains supplementary material available at \\nhttps://doi.org/10.1038/s41586-023-05769-3.\\nCorrespondence and requests for materials should be addressed to Michael T. Eadon, Pierre \\nC. Dagher, Tarek M. El-Achkar, Kun Zhang, Matthias Kretzler or Sanjay Jain.\\nPeer review information Nature thanks Carlos Talavera-Lopez and the other, anonymous, \\nreviewer(s) for their contribution to the peer review of this work.\\nReprints and permissions information is available at http://www.nature.com/reprints.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f03012c2-7fc4-460e-b713-13ffa4ef13cf', embedding=None, metadata={'page_label': '606', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 1 | snCv3 cell types and quality metrics. a. Number of \\nsamples processed across technologies assessed both individually and in \\ncombination. b. UMAP plots for snCv3 clusters. c . UMAP plots as in (b) showing \\nthe corresponding tissue regions, sex, patient identities and conditions. d . Bar \\nand violin plots for snCv3 patients shown in ( c). Barplots showing the total \\nnumber of post-QC nuclei used in the snCv3 clustering analysis, and the \\nproportions that were associated with level 1 subclasses, regions sampled or \\nthe health or disease conditions. Violin plots show the percentage of transcripts \\nassociated with the mitochondria (Mt) or endoplasmic reticulum (ER), as well \\nas mean genes and mean transcripts detected per patient sample. e. Receiver \\noperating characteristic (ROC) curve showing snCv3 clustering quality as \\nassessed by the descrimination between subclasses (level 1) or clusters (b) \\nusing the Single Cell Clustering Assessment Framework (SCCAF). f. Bar and \\nviolin plots as in (d) for snCv3 clusters shown in (b), including proportion of \\nnuclei contributed by each patient.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ef602ef5-bb97-4417-85d0-49ced66b3db6', embedding=None, metadata={'page_label': '607', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 2 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e9cb19cd-86b9-47dd-87c5-9ed3e582c9f4', embedding=None, metadata={'page_label': '608', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 2 | snCv3 marker genes and comparison with reference \\ndata. a. Dot plot showing averaged marker gene expression values (log scale) \\nand proportion expressed for snCv3 clusters. b . Cell type labels predicted from \\nLake et. al. 201912 mapped on the snCv3 UMAP embedding. Inset shows the \\ncorresponding prediction score values. c . UMAP of Lake et. al. 201912 data \\nmapped to snCv3 embeddings showing subclass level 3 predicted labels. Inset \\nshows the corresponding prediction score values. d. UMAP of Muto et\\xa0al. 2021 15 \\ndata mapped to snCv3 embeddings showing subclass level 3 predicted labels. \\nInset shows the corresponding prediction score values. e. Heatmap showing \\ncorrelation of averaged scaled gene expression values for snCv3 epithelial \\n(reference state) clusters and mouse bulk segmental RNA-seq data from Chen \\net\\xa0al., 202161. f. Heatmap showing correlation of averaged scaled gene expression \\nvalues for snCv3 distal tubule clusters (reference states) and mouse scRNA-seq \\ndata from Chen et\\xa0al., 202161. g. Heatmap showing correlation of averaged \\nscaled gene expression values for snCv3 clusters (reference and altered/\\nadaptive states) and mouse snRNA-seq clusters from Kirita et\\xa0al., 2020 4.  \\nh. Heatmap showing correlation of averaged scaled gene expression values \\n(reference states) for snCv3 clusters and mouse scRNA-seq clusters from \\nRansick et\\xa0al., 201959. i. Heatmap showing correlation of averaged scaled gene \\nexpression values for snCv3 stromal clusters (reference and altered/adaptive \\nstates) against human scRNA-seq clusters from Kuppe et\\xa0al., 202025. j. Heatmap \\nshowing correlation of averaged scaled gene expression values for snCv3 \\nimmune cell clusters and mouse immune cell types from Immgen.org.  \\nk. Heatmap showing correlation of averaged scaled gene expression values for \\nsnCv3 immune cell clusters and human immune cell types from Monaco et\\xa0al. \\n201962. l. UMAP of Stewart et\\xa0al., 201914 immune single-cell RNA-seq data \\nmapped to snCv3 embeddings showing subclass level 3 predicted labels (top) \\nand the prior published cell type annotations (bottom). Inset shows the \\ncorresponding prediction score values.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='6ca04fa7-2a00-41c5-b09f-da376401a2fc', embedding=None, metadata={'page_label': '609', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 3 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5cc1ab9b-f6f9-41ec-8f10-920560976984', embedding=None, metadata={'page_label': '610', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 3 | scCv3 integration and quality metrics. a. UMAP  \\nplot showing integrated snCv3, scCv3 and SNARE2 (RNA) subclass level 3 \\nannotations. scCv3 and SNARE2 (RNA) datasets were projected onto the snCv3 \\nembeddings. b. UMAP plots as in (a) show mapping of the corresponding sex, \\npatient identities and conditions for scCv3 and SNARE2 datasets. c . Joint \\nembedding of SNARE2 RNA and AC modalities. d . Barplots showing the total \\nnumber of post-QC nuclei and subclass level 1 cell types detected per scCv3 or \\nSNARE2 patient. Violin plots show the percentage of transcripts associated \\nwith the mitochondria (Mt) or endoplasmic reticulum (ER), as well as mean \\ngenes, mean transcripts, mean accessible peaks or mean TSS enrichment \\nscores detected per patient. e. Barplots showing the total number of post-QC \\nnuclei/cells per subclass (level 3) combined across platforms (snCv3, scCv3, \\nSNARE2). Patient entropy as well as tissue type, region, condition, sex and \\nassay proportions are shown. Heatmap of correlation values for each scCv3  \\nand SNARE2 subclass against the corresponding snCv3 subclass is shown (top \\npanel). Grey values indicate absence of a comparison where subclasses were \\nnot covered by one or more of the technologies.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='99706102-1d2a-4667-b862-36be3c25a2c3', embedding=None, metadata={'page_label': '611', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 4 | Slide-seq predicted cell types. a. UMI counts per  \\nbead for classified beads. Normalized RCTD weights for the beads classified at \\nsubclass level 2 (Methods). Region of the tissue associated with beads for each \\nsubclass. Frequency of cell types predicted across pucks. b. Dot plot showing \\nexpression of cell type markers identified by snCv3 in the classified Slide-seq \\nbeads. c. Representative pucks showing subclass level 2 classifications. Cell \\ntypes are grouped into 3 categories and plotted separately for clarity. Scale bar \\nis 300\\u2009µm. d-e . Cell proximity networks for Slide-seq cell types associated with \\ncortical or medullary regions. For panels a , b, d and e all pucks (6 individuals) \\nwere combined. f. Left panel: Slide-seq puck area indicated in ( c) and predicted \\ncell types for the AEAs and surrounding cell types. Right panel: mapped \\nexpression values for corresponding marker genes (scaled). AEA mapping over \\nVisium histology is depicted in Extended Data Fig.\\xa05j, colocalized with REN  \\nexpression. Scale bar is 100\\u2009µm.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='851948d6-0743-4a31-a00c-f4f6ddda787b', embedding=None, metadata={'page_label': '612', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 5 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='83bd9c16-809c-4955-9127-0aede1f3cee4', embedding=None, metadata={'page_label': '613', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 5 | 10X Visium predicted cell types. a. Analysis of subclass \\n(level 2) predictions on 10x Visium spots (23 samples, 22 individuals). The top \\npanel presents the distribution of transfer scores for the subclass (level 2) with \\nthe highest score in each spot. The UMI count panel presents the UMI counts \\nassociated with these spots. The cell type proportion panel depicts the \\nproportion of transcriptomic signatures for each subclass, corresponding to \\nits transfer score relative to all non-zero transfer scores in that spot. The \\nrelative proportion of cell type subclass signatures arising from the cortex or \\nmedulla in the 23 samples is shown. The bottom panel reveals the alignment \\nbetween the predicted cell type subclass and unsupervised clusters that were \\nhistologically validated (Methods). b. Dot plot showing gene expression of \\nselect cell markers by predicted subclass (level 2) for all 23 Visium samples.  \\nc. The proportion of transcriptomic signatures in the 23 samples revealed a \\nsimilar distribution of cell types across healthy reference nephrectomies, \\nchronic kidney disease (CKD), and acute kidney injury (AKI) samples. d. Cortical \\n(left, I) and medullary (right, U) portions of specimen 21-0063 reveal POD \\nsignatures confined to the cortex, while M-TAL signatures were found in the \\nmedulla. White arrows denote the connection point between the cortex and \\nmedulla portions of the sample. e . A histologic image of the cortex (bounded  \\nin d) reveals level 1 cell type mapping of POD, EC-GC, and VSM/P cells to a \\nglomerulus. PT and TAL signatures were seen mapped over distinct regions of \\ntubules. f. Expression of NPHS2  (for glomeruli), ALDOB (for PT), and SLC12A1 \\n(for TAL) in the cortex. g. A histologic image of the medulla (bounded in d ) \\nreveals level 1 cell type mapping of a high proportion of TAL cells within the \\nmedulla. h. Feature plots showing SLC12A1 but not NPHS2 or ALDOB expression \\nin the medulla. i . Proportion of cortex and medulla cell types for sample  \\n21-0063 (9555 total spots over two sections of the same individual). j . A cortical \\nimage in a healthy reference sample (19-M61) showing EC-AEA entering the \\nglomerular corpuscle near the MD. Two glomeruli contain signatures arising \\npredominantly from POD and EC-GC. Two TAL niches are outlined. TAL niche 1 \\nis enriched in healthy cortical TAL signature and TAL niche 8 near the afferent \\narteriole is enriched for Macula Densa (MD) signature. NPHS2 expression is \\nfound within the glomeruli and renin ( REN) expression is highest in the EC-AEA. \\nA full level 2 cell type deconvolution is provided in the final panel (right). Scale \\nbars are 300\\u2009µm in length.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='7b3bd1d4-c98a-4c45-8883-1f21d9b06919', embedding=None, metadata={'page_label': '614', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 6 | Altered states in a mouse model of AKI. a. UMAP \\nshowing mouse AKI (IRI) data 4 with cell types predicted from snCv3. Mouse \\ndatasets were projected onto the snCv3 UMAP embeddings (Fig.\\xa0 2b). \\nHistograms of prediction scores for subclasses (level 1 and 3) are shown.  \\nb. UMAP plots as in (a) showing the original cell type annotations 4 and injury \\ngroups (time points following IRI) for mouse data. c . Barplot showing the \\nproportion of altered states for each mouse injury group. d . Barplot showing \\nproportion of each injury group for a subset of predicted subclasses. Arrows \\nindicate altered states or immune cells (MAC-M2) that persisted at 6 weeks \\nfollowing injury. e. UMAP as in (a) showing the distribution of reference and \\naltered states over the different injury groups.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5b24011a-8dfd-436a-8fd4-0e89cd28a349', embedding=None, metadata={'page_label': '615', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 7 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f2db0f59-a722-44c5-8a01-57c4bb6f9a2e', embedding=None, metadata={'page_label': '616', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 7 | Altered state expression signatures. a-b. Gene Set \\nEnrichment Analyses (GSEA) for genes upregulated or downregulated in \\nadaptive PT (a) and TAL (b) states compared to reference states. c . Dot plot \\nshowing averaged marker gene expression values (log scale) and proportion \\nexpressed for snCv3 clusters. d . Dot plot showing averaged marker gene \\nexpression values (log scale) and proportion expressed for integrated snCv3/\\nscCv3 reference, degenerative and adaptive stromal clusters. e . Violin plots \\nshowing aSTR and ECM (matrisome) scores for snCv3 clusters. f. Visium feature \\nplots of normalized counts for select markers mapped to regions shown in \\nFig.\\xa03e. Scale bar is 100\\u2009µm. g . Visium feature plot of normalized counts for a \\nselect marker mapped to region shown in ( h). Scale bar is 100\\u2009µm. h. Histology \\nand predicted cell types for a medullary region of acute tubular necrosis (cellular \\ncast formation within tubular lumens, loss of brush border, loss of nuclei, and \\nepithelial simplification). Pie charts are proportions of predicted transfer \\nscores. Area corresponds to the upper bounded region in Fig.\\xa03b. Scale bar is \\n100\\u2009µm. i. Predicted transfer scores for area shown in ( h). Scale bar is 100\\u2009µm.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1eb6bea7-29af-4a82-80b3-607a1ca34fd3', embedding=None, metadata={'page_label': '617', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 8 | 3D imaging identifies injury neighbourhoods.  \\na. Maximum intensity projections of immunofluorescence and second \\nharmonic images for 13 example biopsies, scale bars 500\\u2009µm. b. Overview of \\nneighbourhood classes as given in Fig.\\xa0 4b for reference. c. Distribution of \\nneighbourhoods by specimen in neighbourhood clusters plotted in tSNE space \\nfrom Fig.\\xa04. d. Feature plots of the number of cells per neighbourhood for \\ncortical TAL (C-TAL), altered morphology and proximal tubule (PT). C-TALs and \\nPTs are found in neighbourhoods with altered morphology, cyan and orange vs. \\nred and magenta arrowheads. e-h. Neighbourhoods with at least one cell for \\nthe labels indicated were subsetted and neighbourhood graphs generated to \\nindicate the pairwise interaction between cell labels. At right: maximum \\nZ-projections of 3D confocal fluorescence images with white arrow indicating \\nMPO+ cells (e and f) or CD68+ cells (g), orange arrows indicating CD3+ cells and \\nasterisks highlighting fibrosis (white) or areas of altered morphology/injury \\n(yellow). Scale bar\\u2009=\\u2009100\\u2009µm. h and i, pairwise subset analysis of CD3+, PT and \\nTAL (orange, magenta and cyan arrows respectively). CD3+ cells cluster in \\nregions of fibrosis (orange arrowhead and white asterisks). UMOD positive \\ncasts associate with regions of injury and CD3+ cells (orange asterisk), the \\ntubular epithelium is intact with brush borders (white #), has evidence of \\nepithelial simplification (orange #) or less AQP1 marker and epithelial \\nsimplification (red #). Scale bar\\u2009=\\u2009100\\u2009µm.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='76ad27f2-e49c-4395-bc3c-10d6c08e5686', embedding=None, metadata={'page_label': '618', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 9 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='88d1e1c7-97a5-45a8-92b1-6cda137371b2', embedding=None, metadata={'page_label': '619', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 9 | PT and TAL repair trajectories. a. Trajectory of PT  \\ncells for snCv3 and scCv3 datasets. Bottom UMAPs are coloured by cell density \\nfor each condition (AKI/CKD), including the cell density difference between \\nAKI and CKD. b. UMAP of PT subclasses (PT-S1-S3, aPT) with projected RNA \\nvelocities, derived from a dynamical model of PT repair modules, visualized as \\nstreamlines (Methods). c. Heatmap of smoothed gene expression profiles \\nalong the inferred pseudo-time for PT cells. Colour blocks on the left show \\ndifferent repair states or modules identified based on the gene expression \\nprofiles. d. Right panel: dot plot of SNARE2 average accessibilities (chromVAR) \\nand proportion accessible for TFBSs showing differential activity in aPT \\nmodules. Left panel: dot plot of averaged gene expression values (log scale) \\nand proportion expressed for integrated snCv3/scCv3 modules. e . 3D confocal \\nimaging of a reference kidney tissue section stained for PROM-1 (red), Phopho-\\nc-Jun (p-c-JUN, yellow), F-actin (with FITC phalloidin, green) and DNA with DAPI \\n(cyan) (scale bar 100\\u2009µm). Regions of PROM-1 within a glomerulus (G) and a \\nproximal tubule (PT) are indicated and enlarged in the right panels (rendered \\n3D volumes, scale bar 10\\u2009µm). This area shows the association of PROM-1 \\nexpression with p-c-Jun+ cells in the tubules. 3D rendering was performed \\nusing the Voxx software from the Indiana Center for Biological Microscopy \\n(voxx.sitehost.iu.edu/). f. Top panels: TAL UMAPs as in Fig.\\xa05a (snCv3) showing \\ncondition densities as in ( a). Bottom panels: changes of smoothed gene \\nexpression (snCv3) for representative genes as a function of inferred pseudotime \\ncoloured by disease conditions. g. TAL UMAP as in Fig.\\xa05a (snCv3) with projected \\nRNA velocities, derived from a dynamical model for TAL repair modules, \\nvisualized as streamlines (Methods). h. Heatmap showing expression value \\ndynamics (snCv3) along latent time inferred from RNA velocities for the top \\n300 likelihood-ranked genes. Top colour bar indicates aTAL repair modules.  \\ni. Scatter plots (u, unspliced; s, spliced; t, latent time) for putative driver genes \\n(snCv3) identified by high likelihoods in the dynamical model. j. Gene regulatory \\nnetworks associated with TAL repair modules (Methods, see Supplementary \\nTable\\xa023). Eigenvector centrality scores were plotted for select factors with \\nhigh influence on different states. k. UMAP embedding (snCv3) showing \\npseudotime gradient and the derived vector field associated with TAL repair.  \\nl-m. UMAP embedding showing simulated vector fields following TFAP2B  (l) or \\nNR3C1 (m) perturbation. Barplots show inner product calculations (perturbation  \\nscores) comparing directionality and size of TAL repair flow vectors and the \\nsimulated perturbation vectors. Negative perturbation scores indicate a block \\nin differentiation.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='55b882a0-93b9-4090-acfd-583729b50521', embedding=None, metadata={'page_label': '620', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 10 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='80e5e416-9e51-4263-b6eb-692989e8a8c2', embedding=None, metadata={'page_label': '621', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 10 | Adaptive epithelia localized to areas of injury.  \\na. Immunofluorescent (IF) staining of VCAM1, AQP1, KIM1 (HAVCR1) in the  \\naPT (performed on replicate sections from 3 individuals). Scale bars represent \\n20\\u2009µm. b. IF staining of UMOD, PROM1 and KIM1 in the TAL (performed on \\nreplicate sections from 3 individuals). Scale bars represent 20\\u2009µm. c-e . RNA \\nin\\xa0situ hybridization (ISH) for PROM1 , CST3 or EGF (performed on adjacent \\nsections from 6 individuals). c . ISH for PROM1 and CST3 in adjacent sections. \\nPROM1 is localized to an area showing interstitial fibrosis and tubular atrophy. \\nScale bar is 100\\u2009µm. d. RNA ISH for PROM1 (left panel) and EGF (right panel) in \\nadjacent corticomedullary sections. PROM1  positive epithelial cells seen in \\ninjured tubules (epithelial simplification, loss of nuclei) that are EGF  negative \\n(blue asterisks, upper inset image) and EGF  positive healthy TAL (red asterisks, \\nlower inset image). Scale bar is 100\\u2009µ m. e. ISH for PROM1 and EGF (healthy TAL) \\nshowing PROM1 localization to PT (blue asterisks, left inset) and TAL (red \\nasterisks, right inset) showing histological evidence of injury (epithelial thinning, \\nnuclei loss, brush border loss in PT). Adjacent section (lower panel) shows EGF \\npositivity in healthy TAL cells. Scale bar is 50\\u2009µ m.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c821ed0c-dae0-429d-9de5-db9bd882debf', embedding=None, metadata={'page_label': '622', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 11 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ff5212ff-a28d-486d-b658-97034ab5f7ff', embedding=None, metadata={'page_label': '623', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 11 | TAL adaptive or maladaptive repair niches.  \\na. Slide-seq fibrotic/inflammatory niches from Fig.\\xa0 5d showing full predicted \\nsubclass level 3 cell type distributions. Scale bar is 100\\u2009µ m. b. Visium TAL niches \\nwere identified by clustering TAL dominant spots according to Seurat label \\ntransfer scores. The UMAP denotes 13 TAL niches which were distributed across \\nthe 23 samples (patient inset) and across disease state conditions (condition \\ninset). c. Visium niche cluster compositions. Signature proportions of TAL cell \\ntypes, injury cell states, stromal cells, and immune cells. Niche 5 contained \\nsignificant stromal, niche 7 contained lymphoid, and niche 11 contained myeloid \\ncell signatures. Some niches (e.g. 9) had significant contributions from \\nneighbouring non-TAL epithelial cells (“Proportion Other” bar plot). The \\ncolocalization score (Methods) for cell types within each niche is based on \\nSeurat label transfer scores and provided as a dot plot. d . A subset of TAL niches \\n(1, 3, 5, 7) were overlaid upon a histologic image of the cortex in sample M19-F52_3, \\nwith each niche often represented by multiple contiguous spots. Scale bar is \\n300\\u2009µm in length. e. Representative region (patient 28-12265) showing niche 5 \\n(STR) localized in proximity to interstitial fibrosis, and niche 3 (aTAL) localized \\nadjacent to myeloid cell infiltration. Scale bar is 300\\u2009µm. f. Circle plot of ligand-\\nreceptor cell cell communications between TAL repair modules or states and \\nimmune cell subclasses. Dot size indicates relative proportion of the subclasses \\nand TAL module, edge width represents strength of the communication.  \\ng. Dotplots showing expression level and percent expressed for select ligands \\nor receptors within the mouse AKI data. Data were grouped into injury groups \\nless than or equal to 2 days (including control cells) and groups greater than  \\n2 days post-injury. The asterix highlights an IGF1 expression difference found \\nbetween early and late injury groups of the aFIB population. h . Gene regulatory \\nnetworks associated with STR cell types (see Supplementary Table\\xa027). \\nEigenvector centrality scores were plotted for select factors with high influence \\non different subclasses. Ontologies for target genes downstream of select \\ntranscription factors are shown.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='30891c5b-eea7-4da8-83db-6daea2426fb4', embedding=None, metadata={'page_label': '624', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Article\\nExtended Data Fig. 12 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='91352a80-f21d-4553-b2a4-4f70987fc05b', embedding=None, metadata={'page_label': '625', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Extended Data Fig. 12 | Association of cell state scores with clinical \\nphenotypes.  a. Embedding plots: grouping of patient-level expression profiles \\nfor the aTAL, aStr, Degen, and aPT genesets used for clinical outcome association \\n(Supplementary Table\\xa027) for snCv3 (Top) and scCv3 (Bottom). Barplots: the \\ndistribution of eGFR among the identified groups. b . Unadjusted Kaplan Meier \\ncurves by aStr (P\\u2009=\\u20090.001) and common aPT and aTAL (P\\u2009=\\u20090.03) state scores for \\ncomposite of ESRD or 40% drop in eGFR from time of biopsy in Neptune adult \\npatient cohort (see Supplementary Table\\xa030). A score generated using 100 \\nrandomly selected genes failed to show any correlation (P\\u2009=\\u20090.52) with disease \\nsurvival. c. Heatmap of causal variants (z-scores) that were enriched in SNARE2 \\ncell-type specific accessible chromatin. Dots represent Z-scores > 2 (or P value \\n< 0.05). Dotplots show averaged ESRRB binding site accessibility or gene \\nexpression (log values) and percent accessible or expressed. d. ESRRB \\nsubnetwork of TF connections to target genes generated using SNARE2 RNA \\nand AC data, demonstrating a central role for ESRRB in regulating TAL marker \\ngenes. Inset shows the ESRRB motif. Boxes represent ESRRB target genes \\nshowing causal variant enrichment (c) within linked regulatory regions (AC \\npeaks). e. Heatmap showing enrichment scores (scaled -log10(p values)) for  \\nthe RNA expression (snCv3/scCv3) of gene sets associated with eQTL linked to \\nkidney function or disease 88,89 or associated with progression of acute to \\nchronic injury 90. f. Dot plots of averaged gene expression values (snCv3/scCv3) \\nor TF binding site accessibilities (SNARE) and proportion expressed/accessible. \\nViolin plots show gene expression scores for gene sets associated with aging \\n(Tabula Muris Consortium48 and Takemon et\\xa0al.69) or SASP (Ruscetti et\\xa0al. 70 or \\nBasisty et\\xa0al. 71). g. Violin plots showing expression scores for gene sets shown in \\n(f) for all non-immune subclasses. h. Bottom: Number of differentially expressed \\ngenes between AKI and CKD cases for each major cell type in snCv3 and scCv3 \\ndatasets. Top: enrichment of functional gene ontology terms for each major \\ncell type. Colour indicates -log adjusted p-value (derived from GSEA and \\ncalculated based on permutation).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='89fb8505-4821-4481-a0b8-266077098e65', embedding=None, metadata={'page_label': '626', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\"@ \\nMichael T. Eadon, Pierre C. Dagher, TarekM. \\nu O | el-Achkar, Kun Zhang, Matthias Kretzler, \\nCorresponding author(s): Sanjay Jain \\n  \\nLast updated by author(s): Oct 20, 2022 \\n  \\nReporting Summary \\nNature Portfolio wishes to improve the reproducibility of the work that we publish. This form provides structure for consistency and transparency \\nin reporting. For further information on Nature Portfolio policies, see our Editorial Policies and the Editorial Policy Checklist. \\n  \\n= \\njad) gy = \\noO) \\nTe) fe) \\n=e \\nor 5 \\na) Oo fe) \\n= 5 i) \\n12a) \\nS \\n3 \\n3 \\n= < \\nStatistics \\nFor all statistical analyses, confirm that the following items are present in the figure legend, table legend, main text, or Methods section. \\nConfirmed \\na ~ \\now \\nThe exact sample size (n) for each experimental group/condition, given as a discrete number and unit of measurement \\nA statement on whether measurements were taken from distinct samples or whether the same sample was measured repeatedly xX! Xl \\nThe statistical test(s) used AND whether they are one- or two-sided \\nOnly common tests should be described solely by name; describe more complex techniques in the Methods section. \\nx! \\nA description of all covariates tested \\nA description of any assumptions or corrections, such as tests of normality and adjustment for multiple comparisons \\nA full description of the statistical parameters including central tendency (e.g. means) or other basic estimates (e.g. regression coefficient) \\nAND variation (e.g. standard deviation) or associated estimates of uncertainty (e.g. confidence intervals) x\\n xX x \\nFor null hypothesis testing, the test statistic (e.g. F, t, r) with confidence intervals, effect sizes, degrees of freedom and P value noted \\nGive P values as exact values whenever suitable. \\nx \\nFor Bayesian analysis, information on the choice of priors and Markov chain Monte Carlo settings \\nFor hierarchical and complex designs, identification of the appropriate level for tests and full reporting of outcomes \\n  O\\nX\\nm\\nM\\n O \\nOOOO\\n \\nOO\\n \\nxX OO\\n \\nEstimates of effect sizes (e.g. Cohen's d, Pearson's r), indicating how they were calculated \\nOur web collection on statistics for biologists contains articles on many of the points above. \\nSoftware and code \\nPolicy information about availability of computer code \\nData collection 10x Chromium v3 and Illumina Novaseq 6000 instrument control software (v1.6.0 and 1.7.0); Leica LASX software (v. 3.5); 3D label free \\nautofluorescence and fluorescence imaging data were captured using a Leica SP8 confocal scan-head mounted to an upright DM6000 \\nmicroscope. For large-scale imaging of tissues at submicron resolution, the Leica Tile Scan function was used to collect a mosaic of smaller \\nimage volumes using a high-power, high-numerical aperture objective. Leica LASX software (v. 3.5) was then used to stitch these component \\nvolumes into a single image volume of the entire sample. The scanner zoom and focus motor control were set to provide voxel dimensions of \\n0.5 x 0.5 um laterally and 1 um axially. 2D Immunofluorescence images and data were captured using Nikon EZ-C1 (3.91) confocal system and \\nimages produced using NIS-elements software (BR3.2 64 bit). \\nData analysis Code to reproduce figures are available to download from github.com/KPMP/Cell-State-Atlas-2022. \\nsnCv3 and scCv3 sample demultiplexing, barcode processing, and gene expression quantifications were performed with the 10X Cell Ranger \\nv3 pipeline using the GRCh38 (hg38) or GRCh37 (hg19, indicated in Comments column of Supplementary Table 1)) reference genome. For \\nsingle nucleus data, introns were also included in the expression estimates. SNARE2 data processing pipeline (snarePIP v1.0.1) is available at \\ngithub.com/huqiwen0313/snarePip. For SNARE2 RNA processing, this involved removal of AC contaminating reads using cutadapt (version \\n3.1), dropEst (version 0.8.6) to extract cell barcodes and STAR (version 2.5.2b) to align tagged reads to the genome (GRCh38). For SNARE2 AC \\ndata, this involved snaptools (version v1.2.3) and minimap (version 2-2.20) for alignment to the genome (GRCh38). snCv3 doublets were \\nidentified using DoubletDetection software (version 2.4.0). SNARE2 doublets were identified by both DoubletDetection (version 3.0) and \\nScrublet (github.com/swolock/scrublet, version 0.2.2). Ambient RNA in scCv3 was corrected using SoupX (version 1.5.0). snCv3/scCv3/SNARE \\nanalyses involved the following R packages: Seurat (version 4.0.0), Pagoda2 (version 1.0.2), corrplot (version 0.84), Signac (version 1.1.1), \\nMACS (version 3.0.0a6), chromVAR (version 1.12.0), CisTopic (version 0.3.0), Cicero (version 1.8.1), swne (version 0.6.20), ggdendro (version \\n0.1.20), circlize (version 0.4.12), g-chromVAR (version 0.3.2), Slingshot (version 2.0.0), WGCNA package (version 1.70-3), Cacao (version 0.2.0), \\nSCCAF (version 0.0.10), sclB (version 1.0.3), DoRothEA (version 1.7.2), viper (version 3.15), CellAlign (https://github.com/shenorrLab/cellAlign), \\nvelocyto (version 0.6), CellChat (version 1.0.0). The following python packages were also used: NSForest (version 2.0), velocyto (version \\n0.17.17), scVelo (version 0.2.4), Celloracle (version 0.9.1), CausalDB database (github.com/mulinlab/CAUSALdb-finemapping-pip). Additional \\nYD \\nKaira \\n \\n\", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='639a7650-b020-4919-96cf-056846f04d01', embedding=None, metadata={'page_label': '627', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='code for analysis of chromatin data is provided at github.com/yanwu2014/chromfunks. \\nSlide-seq2 demultiplexing, genome alignment and spatial matching was performed using Slide-seq tools github.com/MacoskoLab/slideseq- \\ntools/releases/tag/0.1. Slide-seq analysis was performed using: Giotto (version 1.0.3), RCTD (version 1.2.0), ggGally (version 2.1.2) and Seurat \\n(version 4.0.0). 10X visium expression analysis, mapping, counting, and clustering was performed using Space Ranger (version 1.0.0) and final \\ndata processing was done in Seurat (version 3.2.0 and 3.2.3). Tissue cytometry and analysis were conducted using the Volumetric Tissue \\nExploration and Analysis (VTEA) software (version 1.0a-r9, www.github.com/icbm-iupui/volumetric-tissue-exploration-analysis) and RStudio \\n(version 1.4) with R (version 4.0.2), corrplot (version 0.84), igraph (version 1.2.6), FNN (version 1.1.3), circlize (version 0.4.12), Hmisc \\n(version 4.5.0), corrplot (version 0.84) and Rtsne (version 0.15). \\nFor manuscripts utilizing custom algorithms or software that are central to the research but not yet described in published literature, software must be made available to editors and \\nreviewers. We strongly encourage code deposition in a community repository (e.g. GitHub). See the Nature Portfolio guidelines for submitting code & software for further information. \\nData \\n  \\nPolicy information about availability of data \\nAll manuscripts must include a data availability statement. This statement should provide the following information, where applicable: \\n- Accession codes, unique identifiers, or web links for publicly available datasets \\n- A description of any restrictions on data availability \\n- For clinical datasets or third party data, please ensure that the statement adheres to our policy \\nProcessed data, interactive and visualization tools: The snCv3, scCv3, SNARE2, Slide-seq and Visium processed data files are all available for download from GEO \\n(Superseries GSE183279). snCv3 healthy reference data is available for reference-based single cell mapping by the Azimuth tool: https:// \\nazimuth.hubmapconsortium.org/. All snCv3 and scCv3 processed data can be accessed and viewed at cellxgene (https://cellxgene.cziscience.com/collections/ \\nbcb61471-2a44-4d00-a0af-ff085512674c). snCv3 (excluding COVID-AKI and CKD nephrectomy samples), scCv3, Visium (KPMP biopsies) and 3D imaging can all be \\nvisualized and interrogated using the KPMP Data Atlas Explorer: https://atlas.kpmp.org/explorer/. For 3D imaging, the cytometry, cell classifications, gates and \\nneighborhood analysis data are located at: https://doi.org/10.5281/zenodo.7120941. \\nRaw sequencing and imaging data: Raw sequencing data are under controlled access (human data) as they are potentially identifiable and can be accessed from the \\nrespective sources indicated below (summarized in Supplementary Table 1). Raw and processed sequencing and imaging data (snCv3, scCv3, 3D imaging, Slide-seq, \\nVisium) generated as part of the Kidney Precision Medicine Project (KPMP) has been deposited at https://atlas.kpmp.org/repository/ and compiled at https:// \\ndoi.org/10.48698/3z31-8924. Raw sequencing data can be requested and are available by signing a data use agreement with KPMP. Raw sequencing data (snCv3, \\nSNARE2, Slide-seq) generated as part of the Human Biomolecular Atlas Project (HuBMAP) has been deposited at https://portal.hubmapconsortium.org/ and \\ncompiled at https://doi.org/10.35079/hbm776.rgsw.867. The HUBMAP raw data are available for download from the database of Genotypes and Phenotypes \\n(dbGaP, phs002249). snCv3 data not deposited to KPMP or HUBMAP are available from GEO (GSE183279) or, for Covid AKI raw sequencing files, upon request from \\nWU KTRC (sanjayjain@wustl.edu) due to patient confidentiality. \\nAdditional published/public data sets: The following publicly available RNA-seq data sets were used in this study: mouse kidney single-cell (GEO, GSE129798); mouse \\nkidney injury single-nucleus (GEO, GSE139107); human fibroblast and myofibroblast single-cell (Zenodo, 10.5281/zenodo.4059315); mouse distal nephron single- \\ncell and bulk distal segment (GEO, GSE150338); human kidney mature immune single-cell (https://kidney-atlas.cells.ucsc.edu); and human kidney single-nucleus \\n(GEO, GSE151302; https://human-kidney-atac.cells.ucsc.edu). GWAS summary statistics were from the CKDGen Consortium (all eGFR, https://ckdgen.imbi.uni- \\nfreiburg.de/files/Wuttke2019), EB! GWAS Catalog (hypertension, https://www.ebi.ac.uk/gwas/efotraits/EFO_0000537), and the CausalDB database (Release 1.1 \\n2019-09-29, http://www.mulinlab.org/causaldb). NEPTUNE sequencing and clinical data were obtained from the Nephrotic Syndrome Study Network and are \\navailable upon request to NEPTUNE-STUDY @umich.edu due to patient confidentiality. ERCB data was obtained from GEO (GSE104954). Raw sequencing data \\n(scCv3) on living donor biopsies as part of the Chan Zuckerberg Initiative (CZI) and Human Cell Atlas (HCA) are available from GEO (GSE169285). Additional visium \\nspatial transcriptomic data not in the KPMP repository are available from GEO (GSE171406). \\nFigures: Source data are provided with this paper. Additional figures can be accessed at Zenodo https://doi.org/10.5281/zenodo.6987337. Schemata of the human \\nnephron and renal corpuscle were developed by the Kidney Precision Medicine Project and HuBMAP (https://doi.org/10.48698/DEM4-0Q93). \\nField-specific reporting \\n  \\nPlease select the one below that is the best fit for your research. If you are not sure, read the appropriate sections before making your selection. \\nLife sciences [| Behavioural & social sciences [| Ecological, evolutionary & environmental sciences \\nFor a reference copy of the document with all sections, see nature.com/documents/nr-reporting-summary-flat.pdf \\nLife sciences study design \\n  \\nAll studies must disclose on these points even when the disclosure is negative. \\nSample size Sample sizes were not predetermined by statistical methods due to nature of this study. The strength lies in the number of individuals \\nanalyzed, technologies represented for orthogonal validation and cells analyzed (more than any existing study for the kidney). For snCv3 (n = \\n36), scCv3 (n = 45), SNARE2 (n = 7), 3D imaging (n = 15), 10X Visium (n = 22) and Slide-seq (n = 6) single nuclei, single cells or tissue sections \\nwere obtained from living or deceased donor tissues (\"\\'n\" here refers to individuals, the number of independent samples is explained in detail \\nin the \"Replication\" section below). These were obtained from healthy reference, AKI or CKD individuals. To ensure robust cell state profiles, \\nreference tissues were obtained from multiple sources, and biopsies were collected from AKI and CKD patients under rigorous quality \\nassurance and control procedures. This ensured that cell type clusters were not driven by technical artifacts and that our analyses showed \\nrigor and reproducibility. \\n2 \\nsad) \\noy \\nS o \\nre) e) \\n=a \\no e \\na ze) (e) \\n= 2 a \\nZa) \\n= \\n3 \\n= \\n= = \\nYD (é COC \\n1K \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b75357b1-fd8c-4c75-b382-610f7d3936fd', embedding=None, metadata={'page_label': '628', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Data exclusions \\nReplication \\nRandomization \\nBlinding \\nBehaviou \\nLow quality cells or nuclei were excluded from analyses based on established quality filtering metrics: \\nsnCv3: \\nCellRanger Empty barcode filter \\nDoublets identified using DoubletDetection software \\n< 400 or > 7500 genes detected per cell \\nGene/UMI ratio filter (Pagoda2) \\nscCv3: \\n> 50% mitochondrial reads \\n< 500 or > 5000 genes detected per cell \\nSNARE2 - RNA: \\nDropEst cell score < 0.9 \\nDoublets identified using DoubletDetection and Scrublet software \\n< 200 or > 7500 genes detected per cell \\nGene/UMI ratio filter (Pagoda2) \\nSNARE2 - AC: \\nCell barcodes not passing RNA QC filters \\n< 0.15 tss enrichment \\n< 1000 read fragments or 500 UMI per cell \\n< 0.15 of read fragments overlapping promoter regions \\nsamples showing < 500 dual omic cells after quality filtering \\nGene/UMI ratio filter (Pagoda2) \\nVisium 10x: \\nIn each Visium sample, spots were eliminated if they did not overly tissue. In addition, the outermost layer of spots was eliminated from \\ncomparative analyses if the edge was manually cut by a razor. \\nRNA-Seq: snCv3 data was generated from 44 independent samples or experiments to cover 36 individuals, scCv3 was generated from 49 \\nsamples covering 45 individuals, and SNARE2 was generated from 17 samples covering 7 individuals. snCv3 clustering analysis was performed \\nat multiple k values and cluster assignments were performed using a defined process (see Methods). Reproducibility of assigned cell type \\nannotations was evident from consistent aligned populations found across technologies (scCv3, SNARE, Slide-seq, Visium) and high correlation \\nvalues with reference (published) data sets. \\nImaging: For 3D imaging and immunofluorescence staining experiments, each staining was repeated on at least 2 separate individuals or \\nseparate regions. For ISH, each stain was performed on 6 separate individuals. For Visium spatial transcriptomics, 23 samples from 22 \\nindividuals were included in the analysis. These included at least 6 samples from each of the reference, CKD, and AKI categories. For Slide-seq \\nwe generated 31 cortical and 36 medullary pucks from 6 individuals. For immunofluorescence validation studies, commercially available \\nantibodies were used; the immunostaining included tissue from patients not contributing to omics data. Similarly, orthogonal validation of \\nomics annotations and spatial localization in Visium studies also included more than four samples each from reference and disease biopsies \\nthat were not used to generate single cell gene expression data. This heterogeneity in sampling demonstrated the reproducibility and rigor of \\nthe atlas. All attempts at replication were successful for these imaging experiments. \\nFurther, several technologies were performed on samples from the same individual and in some cases the same tissue block was used to \\ngenerate multimodal data. \\nRandomization was not used as it was not relevant for this study design as healthy and disease samples were obtained as available. \\nGeneration of data and processed files were agnostic to the disease conditions. Batch effects were corrected by scaling expression of each \\ngene to the dataset-wide average and shown to have minimal effect from cell type or cluster contribution plots. \\nAll human specimens used in this study were de-identified, however select attributes (condition, age, sex) were available to all investigators. A \\nmajority of the analyses were not performed blind as these sample attributes were needed for accurate annotation of cell types or states and \\nfor the design of downstream analyses to create maps. \\nral & social sciences study design \\nAll studies must disclose on these points even when the disclosure is negative. \\nStudy description \\nResearch sample \\nSampling strategy \\nData collection \\ncribe the stuay typ nciuding whether data a quanutaui \\n  \\n    \\n    \\n2 \\nm \\n(om \\n‘S \\no \\nro) \\nfe) \\nBe o F \\na \\n12) \\n(e} \\n& 5 \\n© \\nZa) \\n= \\n= \\n=| \\n= = \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='8427c2d4-864f-440b-b3b2-79be0fc7cb10', embedding=None, metadata={'page_label': '629', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='or audio equipment) whether anyone was present ides the participant(s) and the researcher, and     \\n  \\n    Data collection   1s blind to experimental condition and/or the study hypothesis during data collection. | y ny] ( \\n    \\n     \\n     \\n= \\n(oY) \\n; ¢ ‘ 3 j ; f (ear \\nTiming 2 > start and stop dates of data collection. If there is a gap between collection periods, state the dates for each sample (S \\ncohort. (@) \\n12) \\n. “A ; . oy ie) Data exclusions if no data were excluded from the analyses, state so OR if data were excluded, provide the exact nurnber of exclusions and the = \\nck 7 . . » - =) \\nrationale behind them, indicating whether exclusion criteria were pre-established. Q \\ne) \\nNon-participation State how many participants dropped out/declined participation and the reason(s) given OR provide response rate OR state that no arr \\nparticipants dropped out/declined participation. ay \\nxe) \\n. \" £ 7 1 . ni I je) Randomization if participants were not allocated into experimental groups, state so OR describe how participants were allocated to groups, and if = \\nallocation was not random, describe how covariates were controlled. ie \\nWw \\n= \\n3) \\nlogical luti ironm | sci dy desi 3 i) Ecological, evolutionary & environmental sciences study design E \\n  \\nAll studies must disclose on these points even when the disclosure is negative. \\nStudy description Briefly describe the study. For quantitative data include treatment factors and interactions, design struclure (e.g. factorial, nested, \\nhierarchicol), nature and number of experimental units and replicates. \\nResearch sample Describe the research sample (e.g. a group of tagged Passer domesticus, all Stenocereus thurberi within Organ Pipe Cactus National \\nMonument), and provide a rationale for the sample choice. When relevant, describe the organism taxa, source, sex, age range and \\nimple is meant to represent when applicable. For studies involving existing da \\n  \\n    \\n    \\nany manipulations. State what population the els, \\ndescribe the data and its source \\nSampling strategy Note the sampling procedure. Describe the statistical methods that were used to predetermine sample size OR if no sample-size \\ncalculation was performed, describe how sample sizes were chosen and provide a rationale for why these sample sizes are sufficient. \\nData collection Describe the dato collection procedure, including who recorded the dato and how \\nTiming and spatial scale = /ndicate the start and stop dotes of data collection, neting the frequency and periodicity of sampling and providing a rationale for g p g the | y E y pling ( | \\nthese choices. If there is a gap between collection periods, state the dates for each sample cohort. Specify the spatial scale from which   \\nthe data are taken \\n    \\n  \\n5, state so OR if data were ided, describe the exclusions and the rationale behind them, \\nablished. \\nData exclusions If no data were excluded from the analys \\nindicating whether exclusion criteria were pre \\ne \\n  \\nReproducibility Describe the measures taken to verify the reproducibility of experimental findings. For each experiment, note whether any attempts to \\nepeat the experiment failed OR state that all attempts to repeat the experiment were successful. t | f I f } \\nRandomization Describe how samples/organisms/participants were allocated into groups. If allocation was not random, describe how covariates were \\ncontrolled. tf this is not relevant to your study, explain why. \\n    ‘plain why \\n  \\nscribe the extent of blinding used during data acquisition and analysis. If blinding was not possible, describe why OR « Blinding De \\nblinding was not relevant tu your study. \\nDid the study involve field work? [_| Yes No \\nField work, collection and transport \\n  \\nemperature, rainfall).     i ndition Describe the study conditions for field work, providing relevant purameters (e. S J g k \\nle the location of the sampling or experiment, providing relevant parameters (e.g. latitude and longitude, elevation, water depth). J ung | ! ¢ ! u g | \\n  \\nLocation   \\n   ponsible manner and in \\n  \\nport your samples ina re \\n  \\nss habitats and to collect and impo   Access & import/export Describe the efforts you have made to ac \\ncompliance with local, national and international laws, noting any permits that were         obtained (give the name of the issuing authority,    \\nthe date of issue, and any identifying information). \\ncaused by the study and how it was minimized.     any disturban Disturbance Descr \\nSy a \\n; \\nReporting for specific materials, systems and methods \\n  \\nWe require information from authors about some types of materials, experimental systems and methods used in many studies. Here, indicate whether each material, \\nsystem or method listed is relevant to your study. If you are not sure if a list item applies to your research, read the appropriate section before selecting a response. \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='88d17286-b5aa-4fcb-9bd2-c7468431a2d0', embedding=None, metadata={'page_label': '630', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Materials & experimental systems Methods \\n  \\nn/a Involved in the study \\nLIX Antibodies \\nEukaryotic cell lines \\nPalaeontology and archaeology \\nn/a | Involved in the study \\nXL] chip-seq \\nXx EC] Flow cytometry \\nXi LE] MRI-based neuroimaging \\nAnimals and other organisms \\nHuman research participants \\nClinical data UU\\nk\\nX\\n \\nX K\\nOO\\n \\n[DX]}[_] Dual use research of concern \\n  \\n) \\njad) \\not \\n(= 5 \\n(@>) \\nxe) \\n(e} \\n2 =e \\nio} 5 \\n1 \\n@) \\n12) \\nfo) \\na \\n=) \\n© \\nn \\n(= \\n3 \\n3} \\nje) \\nSs \\nAntibodies \\nAntibodies used The antibodies used and associated details are tabulated in Supplemental tables 35 and 36. \\nFor 3D cytometry studies \\nPrimary antibody or Fluorescent probe, Target, Vendor, Dilution, Secondary antibody, Vendor, Dilution \\nGoat anti-aquaporin1(AQP1), Proximal tubules, Santa Cruz (sc-9878), 1:50, (please note, very minimal content about this \\ndiscontinued antibody) \\nAlexa568 donkey anti-goat, ThermoFisher(A-11057), 1:200 \\nRabbit anti-myeloperoxidase(MPO), Neutrophils, Abcam (ab9535), 1:50 \\nDyLight 594 donkey anti-rabbit, ThermoFisher(SA5-10040), 1:200 \\nMouse anti-CD68, Macrophages, Dako (M0876), 1:50, Alexa 633donkey anti-mouse, ThermoFisher(A16019)(Conjugated in-house \\nwith ThermoFisher (A20170), 1:200 \\nMouse Alexa660 anti-SIGLEC8, Eosinophils, Biolegend (347102)(Conjugated in-house with ThermoFisher (A20171)), 1:50 \\nSheep Alexa546 anti-Uromodulin(UMOD), Thick ascending limb, R&D Systems (AF5144), 1:200, conjugated in house with kit (https:// \\nwww.thermofisher.com/order/catalog/product/A20183) \\nMouse Alexa647 anti-CD3, T-cells, BD Pharmingen (557706), 1:50 \\nDAPI, Nuclei, ThermoFisher (D1306), 1:100 \\nOregon Green Phalloidin, Filamentous actin (vasculature, brush border), ThermoFisher 07466, 1:200 \\nFor 2D confocal immunofluorescence microscopy \\nPrimary Antibody, Against Raised in, Company Cat #, Primary antibody dilution, secondary Antibody, secondary Antibody dilution, \\nUromodulin human mouse Ray biotech 119-13298 1:100 Goat anti-mouse alexa-488 1:400 \\nCD133 (PROM1) human, mouse, rat Rabbit ThermoFisher PAS-38014 1:50 goat anti-ratbbit-cy3 1:400 \\nKIM1 human Rabbit ThermoFisher PAS-79345 1:250 goat anti-rabbit -cy3 1:400 \\nVCAM1 human, rat mouse ThermoFisher MA5-11447 1:50 Goat anti-mouse alexa-488 1:400 \\nAQP1 human Rabbit santa cruz sc-20810 1:100 goat anti-rabbit -cy3 1:400 \\nValidation Validation of antibodies and confidence in their staining is derived from several sets of data including vendors specifications, omitting \\nprimary antibody, well-established expected cell-type staining pattern for the indicated antibodies in the literature, referring to \\nhuman protein atlas data where available and orthogonal validations in the multiomics data presented. \\n3D IF antibodies: \\nAQP1 - https://www.scbt.com/p/aqp1-antibody-I-19 \\nAF-568 - https://www.thermofisher.com/antibody/product/Donkey-anti-Goat-lgG-H-L-Cross-Adsorbed-Secondary-Antibody- \\nPolyclonal/A-11057 \\nMPO - https://www.abcam.com/myeloperoxidase-antibody-ab9535.html \\nDyLight-594 - https://www.thermofisher.com/antibody/product/Donkey-anti-Rabbit-IgG-H-L-Cross-Adsorbed-Secondary-Antibody- \\nPolyclonal/SA5-10040 \\nCD68 - https://www.agilent.com/en/product/immunohistochemistry/antibodies-controls/primary-antibodies/cd68-%28concentrate \\n%29-76550 \\nConjugated in-house to AF-633 — kit # https://www.thermofisher.com/order/catalog/product/A20170 \\nSIGLEC8 (AF660) - https://www.biolegend.com/de-at/products/purified-anti-human-siglec-8-antibody-6383 \\nConjugated in-house to AF-660 — kit # https://www.thermofisher.com/order/catalog/product/A20171 \\nUMOD - https://www.rndsystems.com/products/human-uromodulin-antibody_af5144 \\nConjugated in-house to AF-546 — kit # https://www.thermofisher.com/order/catalog/product/A20183 (https:// \\nwww.ncbi.nlm.nih.gov/pmc/articles/PMC8363780/) \\nCD3 - https://www.bdbiosciences.com/en-us/products/reagents/flow-cytometry-reagents/research-reagents/single-color- \\nantibodies-ruo/alexa-fluor-647-mouse-anti-human-cd3.557706 \\nDAPI - https://www.thermofisher.com/order/catalog/product/D21490 \\nPhalloidin - https://www.thermofisher.com/order/catalog/product/O7466 \\n2D Antibodies: \\nUromodulin - https://www.raybiotech.com/mouse-anti-human-uromodulin/ \\nCD133 (PROM1) - https://www.thermofisher.com/antibody/product/CD133-Antibody-Polyclonal/PA5-38014 \\nKIM1 - https://www.thermofisher.com/antibody/product/KIM-1-Antibody-Polyclonal/PAS-79345 \\nVCAM1 - https://www.thermofisher.com/antibody/product/VCAM-1-Antibody-clone-1-4C3-Monoclonal/MA5-11447 \\nAQP1 - https://www.citeab.com/antibodies/789773-sc-20810-aqp1-antibody-h-55 \\na Ova \\nBH eNh] \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d5d12ec0-4031-4caa-b1df-23ce918c95da', embedding=None, metadata={'page_label': '631', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Eukaryotic cell lines \\n  \\nPolicy information about cell lines \\n    \\n    \\n  \\nCell line source(s) State the source of each cell line used. \\nAuthentication Describe the authenticat for each cell line used OR declare that none of the cell lines used were authenticated \\nMycoplasma contamination Confirm that ¢ e the results o, > tesling for \\nmy sma contamination. \\nCommonly misidentified lines Name any commonly misidentified cell lines used in the study and provide a rationale for their use. \\n(See ICLAC register) \\nPalaeontology and Archaeology \\n  \\n          \\nvenance information for specimens and describe permit Specimen provenance Provide p \\nissuing authority, the date of issue, and any identifying information). Permits \\n  \\nexport, \\n    Specimen deposition Indicate where \\n»y were obtained (e.g. collectio je, sample pretreatment and measurement), where \\ne ORs \\nDating methods           oration program and t e that no new dates are \\n  \\n[| Tick this box to confirm that the raw and calibrated dates are available in the paper or in Supplementary Information. \\nganization(s) that approved or provided guidance on the study protocol, OR state that na ethical approval or guidance Ethics oversight Identify th \\nwas required and explain why not 4 F \\n  \\nNote that full information on the approval of the study protocol must also be provided in the manuscript. \\nAnimals and other organisms \\n  \\nPolicy information about studies involving animals; ARRIVE guidelines recommended for reporting animal research \\n     \\n      \\n   \\n   \\n  \\n   \\n  \\nLaboratory animals For laboratory animals, report species, strain, sex and age OR state that the study did not involve laboratory \\nWild animals ide detail e where possible. Describe how animals were \\ncaught and trans «plain why and describe method; if released \\n  \\nay where and \\nField-collected samples = (or aboratory u \\nphotoperioc \\n  \\nEthics oversight Identify the organi \\nvas required and \\n  \\nNote that full information on the approval of the study protocol must also be provided in the manuscript. \\nHuman research participants \\n  \\nPolicy information about studies involving human research participants \\nPopulation characteristics The population used here were adults in the age interval 20-80 and included both sexes and participants of different races. \\nThe associated clinical metadata includes age, sex, race, comorbidities, eGFR, certain medications and is detailed in \\nsupplemental table 3. The clinical conditions include AKI and CKD. \\nRecruitment Participants were recruited from different sites and IRB approval was obtained for use of tissue and data for research ina \\ndeidentifiable manner. To obtain consent, the coordinators would approach the participant after consultations with the \\nclinical team, go over the study with them, address any questions and concerns. Once consent was obtained, samples are \\nprocured and preserved in a timely manner using standardized protocols that have been published and available on \\nKPMP.org. Recruitment of AKI and CKD patients were per established clinical criteria (https://www.kpmp.org/for-clinicians). \\nThe reference tissue samples were selected from patients with normal kidney function and/or age appropriate \\nhistopathology as they became available. Samples under waived consent are described in the ethics statement. The \\nassociated clinical and pathological data is provided in Supplemental Table3 for readers to interpret the study results. \\nEthics oversight We have complied with all ethical regulations related to this study. Human samples (Supplementary Table 1) collected as part \\nof the Kidney Precision Medicine Project (KPMP) consortium (KPMP.org) were obtained with informed consent and approved \\nunder a protocol by the KPMP single IRB of the University of Washington Institutional Review Board (IRB#20190213). Samples \\nas part of the Human Biomolecular Atlas Program (HUBMAP) consortium were collected by the Kidney Translational Research \\nCenter (KTRC) under a protocol approved by the Washington University Institutional Review Board (IRB #201102312). \\n3 \\njad) \\nss \\nSs o) \\n18) \\n(e) 2 \\no a \\na \\nxe) \\nOo \\n2 =] \\n© \\nwn \\n= 5 \\n3} \\nad) = = \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='99d160ae-cf5b-40ff-b38d-0745616606f1', embedding=None, metadata={'page_label': '632', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Informed consent was obtained for the use of data and samples for all participants at Washington University, including living \\npatients undergoing partial or total nephrectomy or from discarded deceased kidney donors. Cortical and papillary biopsy \\nsamples from patients with stone disease were obtained with informed consent from Indiana University and approved by the \\nIndiana University Institutional Review Board (IRB #1010002261). For Visium Spatial Gene Expression, reference \\nnephrectomies and kidney biopsy specimens were obtained from the KPMP under informed consent or the Biopsy Biobank \\nCohort of Indiana (BBCI)49 under waived consent as approved by the Indiana University Institutional Review Board (IRB # \\n1906572234). Living donor biopsies as part of the Human Cell Atlas (HCA) were obtained with informed consent under the \\nHuman Kidney Transplant Transcriptomic Atlas (HKTTA) under IRB HUM00150968. Deidentified leftover frozen COVID-19 AKI \\nkidney biopsies were obtained from the Johns Hopkins University pathology archive under waived consent approved by the \\nJohns Hopkins Institutional Review Board (IRB 00090103). \\nNote that full information on the approval of the study protocol must also be provided in the manuscript. \\nClinical data \\n  \\nPolicy information about clinical studies \\nAll manuscripts should comply with the ICMJE guidelines for publication of clinical research and a completed CONSORT checklist must be included with all submissions. \\nClinical trial registration NOT APPLICABLE \\nR if not available, \\n  \\nStudy protocol Note where the full trial protocol can be accesse explain WAY. \\n  \\nf data collection, noting the time periods of recruitment and dota collection. Data collection Describe the settings ¢ \\nOutcomes Describe how you pre-defined primary and secondary outcome measures and how you assessed these measure \\nDual use research of concern \\n  \\nPolicy information about dual use research of concern \\nHazards \\nCould the accidental, deliberate or reckless misuse of agents or technologies generated in the work, or the application of information presented \\nin the manuscript, pose a threat to: \\nNo | Yes \\nXI}L_] Public health \\n[_] National security \\n[| Crops and/or livestock \\n| Ecosystems \\nx [| Any other significant area \\nExperiments of concern \\nDoes the work involve any of these experiments of concern: \\nDemonstrate how to render a vaccine ineffective \\nConfer resistance to therapeutically useful antibiotics or antiviral agents \\nEnhance the virulence of a pathogen or render a nonpathogen virulent \\nIncrease transmissibility of a pathogen \\nAlter the host range of a pathogen \\nEnable evasion of diagnostic/detection modalities \\nEnable the weaponization of a biological agent or toxin \\nRR\\nKRRR\\nMM\\nS \\nOOOO\\noCoOoo\\ns \\nAny other potentially harmful combination of experiments and agents   \\nChIP-seq \\n  \\nData deposition \\n[| Confirm that both raw and final processed data have been deposited in a public database such as GEO. \\n[| Confirm that you have deposited or provided access to graph files (e.g. BED files) for the called peaks. \\nData access links For “initial submissio r \"Re ? version” documents, provide revie \\nMay remain private before publication provide Glink to th id \\n5) \\nsad) 2 \\nSs \\n© \\nre} e) g a aa \\neh 5 \\n- \\nie) 2 \\nct. \\n= i) \\nWa) \\nme 5 \\n=) \\n3 \\n2 S \\npy (é a oye \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='bd4590c0-e07e-41e2-9a8a-b5217dff3972', embedding=None, metadata={'page_label': '633', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='submission     f all files available in the d¢ Files in database submission Provide al \\n  \\n‘ond \"Revised version\" documents only, to \\n  \\n“Initial su Genome browser session Provide a link to an anonymized genor \\n(e.g. UCSC) enable peer review. Write \"no longe \\n          \\nlal submission\" documents \\nMethodology \\nReplicates Describe the experimental replicates, specifying number, type and replicate agreement \\n    mapped reads, length of reads and \\n  \\ne total number of reads, unique \\n  \\nSequencing depth \\n  \\nprovide supplier name, catalog number, clone name, and lot Antibodies \\n  \\nnumber, \\n{for read mapping and peak calling, including the ChIP, control and index files \\n  \\nPeak calling parameters Specify the command line prograrn and parameters   \\n  \\ne the methods used to ensure data quality in full detail, including how many peaks are at FOR 5% and above 5-fold enrichment. \\n  \\nData quality \\nhe software used to collect and analyze the ChiP-seq data. For custom code that has been deposited into a community Software \\nry, provide accession details \\n  \\nFlow Cytometry \\nPlots \\nConfirm that: \\n| The axis labels state the marker and fluorochrome used (e.g. CD4-FITC). \\n[| The axis scales are clearly visible. Include numbers along axes only for bottom left plot of group (a ‘group’ is an analysis of identical markers). \\n| All plots are contour plots with outliers or pseudocolor plots. \\n[| A numerical value for number of cells or percentage (with statistics) is provided. \\nMethodology \\nsource of the cells and any tissue processing steps used, \\n  \\nSample preparation Describe the sample preparation, detailing the bialoc \\n  \\n, specifying make and model number Instrument Identify the instrument used for data colle \\ncode that has been deposited into a vare used to collect and anal!     Software Describe the sof low cytometry data. For ¢ \\ncommunity repository, provide acce       \\nyst-sorl fractions, providing \\n  \\nCell population abundance Desc \\n  \\n2cifying the preliminal     Gating strategy Describe the gating strateay used for all re \\npopulation, indicating where boundaries be \\n  \\n\"negative\" staining cell populations are defined \\n  \\n[| Tick this box to confirm that a figure exemplifying the gating strategy is provided in the Supplementary Information. \\nMagnetic resonance imaging \\nExperimental design \\n  \\nor DIOCK design. Design type indicate task or resting state; event-relate \\n  \\nDesign specifications \\n   \\n      Behavioral performance measures \\n     expected (e.g. mean, \\n  \\n>) \\njad} \\not \\n‘s \\n(q>) \\nxe) \\nfe a o = \\noO \\nae) \\nfe) \\net. \\n>) \\nCo} \\nWN \\nSs \\n=: \\n= \\nst \\n< \\n \\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='66755a15-7cb0-4149-888d-133dcbc2a56f', embedding=None, metadata={'page_label': '634', 'file_name': 'lake_kidney.pdf', 'file_path': '/content/data/lake_kidney.pdf', 'file_type': 'application/pdf', 'file_size': 29830863, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text=\"Acquisition \\n  \\nImaging type(s) Specify: functional, structural, diffusion \\nField strength specify in Tesla \\n  \\nSequence & imaging parameters Specify the pulse sequence type (gradient echo, spin echo, etc.), imaging type (EPI, spiral, etc.), field of view, matrix sf: \\nslice thickness, orientation and TE/TR/flip angle \\n  \\n   \\nArea of acquisition State whether a whole brain sce \\n  \\nwas used OR define the area of acquisition, describing how the region was determined. \\nDiffusion MRI [| Used [_] Not used \\nPreprocessing \\n  \\nre version and revision number and on specific parameters (model/functions, brain extraction, Preprocessing software Provide detail on soft \\nsegmentation, smoothing kernel size, ete.). \\n= \\na8) = = \\noO \\n8} \\n2) ca \\njee \\n(fe) \\no O \\nie ae 3) i) \\nVa} \\n= \\n= o \\n3) \\na9) \\nSZ \\n   \\n    \\nNormalization ' if data \\ntransformation OR indicate that data were not normatized \\n  \\nre normalized/standardized, describe the approach(es): specify linear or non-linear and define image types used for \\n    and explain rationale for lack of normatizatior \\n  \\nNormalization template Describe the template used for normalization/transformation, specifying subject space or group standardized s \\noriginal Talairach, MNI305, ICBM152) OR indicate that the data we     ere Not normalized. \\nNoise and artifact removal Describe your procedure(s) for artifact and structured noise removal, specifying motion parameters, tissue signals and \\nIs (heart re     physiological signe e, respiration) \\nVolume censoring Define your software and/or method and criteria for volu censoring, and state the extent of such censoring. \\n  \\nStatistical modeling & inference \\n    (mass univariate, multivariate, RSA, predictive, etc.) and describe essential details of the model at the first and Model type and settings \\n   second levels (e.g. fixed, random or mixed effects; drift or auto-correlation). \\nEffect(s) tested tin terms of the task or stimulus conditions instead of psychological concepts and indicate whether \\n    torial designs were used   \\nSpecify type of analysis: [ ]Whole brain [ | ROl-based [| Both \\nStatistic type for inference Specify voxel-wise or cluster-wise and report all relevant parameters for cluster-w \\n(See Eklund et al. 2016) \\n> methods \\n  \\nCorrection Desc > of correction and how it is obtained for multiple comparisons (e.g. FWE, FDR, permutation ar Monte Carlo) \\n  \\nModels & analysis \\nn/a | Involved in the study \\n[| [| Functional and/or effective connectivity \\n[| [| Graph analysis \\n[ ] [| Multivariate modeling or predictive analysis \\n      rson correla Functional and/or effective connectivity \\n  \\nor dinar oh    \\n  \\nGraph analysis Re \\n  \\n    \\nclion, model, training and evaluation \\n  \\nMultivariate modeling and predictive analysis \\n  \\n \\n\", path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='764bc332-71b2-4183-b086-3d3e3b3e633c', embedding=None, metadata={'page_label': '1534', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1534\\nnature cancer\\nArticle\\nhttps://doi.org/10.1038/s43018-024-00828-8\\nSpatial analysis reveals targetable \\nmacrophage-mediated mechanisms of \\nimmune evasion in hepatocellular carcinoma \\nminimal residual disease\\nLea Lemaitre\\u2009  \\u20091,9, Nia Adeniji1,9, Akanksha Suresh1,9, Reshma Reguram\\u2009  \\u20091, \\nJosephine Zhang1, Jangho Park1, Amit Reddy\\u2009  \\u20091, Alexandro E. Trevino\\u2009  \\u20092, \\nAaron T . Mayer\\u2009  \\u20092, Anja Deutzmann\\u2009  \\u20093, Aida S. Hansen4, Ling Tong3, \\nVinodhini Arjunan1, Neeraja Kambham5, Brendan C. Visser6, Monica M. Dua6, \\nC. Andrew Bonham6, Nishita Kothary7, H. Blaize D’ Angio2, Ryan Preska2, \\nYanay Rosen8, James Zou\\u2009  \\u20098, Vivek Charu5, Dean W. Felsher\\u2009  \\u20093  & \\nRenumathy Dhanasekaran\\u2009  \\u20091 \\nHepatocellular carcinoma (HCC) frequently recurs from minimal residual \\ndisease (MRD), which persists after therapy. Here, we identified mechanisms \\nof persistence of residual tumor cells using post-chemoembolization human \\nHCC (n\\u2009=\\u2009108 patients, 1.07 million cells) and a transgenic mouse model of \\nMRD. Through single-cell high-plex cytometric imaging, we identified a \\nspatial neighborhood within which PD-L1\\u2009+\\u2009M2-like macrophages interact \\nwith stem-like tumor cells, correlating with CD8+ T cell exhaustion and poor \\nsurvival. Further, through spatial transcriptomics of residual HCC, we showed \\nthat macrophage-derived TGFβ1 mediates the persistence of stem-like tumor \\ncells. Last, we demonstrate that combined blockade of Pdl1 and Tgfβ e xc lu-\\nded i mm un os up pr essive macrophages, recruited activated CD8+ T cells and \\neliminated residual stem-like tumor cells in two mouse models: a transgenic \\nmodel of MRD and a syngeneic orthotopic model of doxorubicin-resistant \\nHCC. Thus, our spatial analyses reveal that PD-L1+ macrophages sustain MRD \\nby activating the TGFβ pathway in stem-like cancer cells and targeting this \\ninteraction may prevent HCC recurrence from MRD.\\nHCC is associated with high rates of recurrence, which contribute \\nto a grim 5-year survival rate below 20% (ref. 1). In patients with  \\nnonmetastatic HCC who respond well to transarterial chemoembo -\\nlization (TACE), most but not all tumor cells are eliminated, leaving  \\nbehind a small fraction of residual tumor cells. Even after prolonged \\nlatency, these residual tumor cells can lead to recurrence, which is \\ngenerally unresponsive to therapy and contributes to poor survival2–5. \\nThus, deciphering how residual tumor cells evade post-therapy \\nimmune surveillance is crucial for preventing recurrence and  \\nenhancing outcomes.\\nResidual tumor cells may employ two major mechanisms to persist \\nafter therapy and instigate recurrence: cell-intrinsic or immune-based \\nmechanisms. First, recurrence can arise from a subset of residual cells \\nthat could function as stem-like tumor cells, characterized by unique \\nattributes such as chemoresistance and dormancy6–8. Previously, we \\nhave reported a transgenic mouse model of HCC dormancy with such \\nReceived: 15 September 2023\\nAccepted: 14 August 2024\\nPublished online: 20 September 2024\\n Check for updates\\nA full list of affiliations appears at the end of the paper. \\u2009e-mail: dfelsher@stanford.edu; dhanaser@stanford.edu', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='75c79d19-8a13-496b-b1a6-9236c6bea6b4', embedding=None, metadata={'page_label': '1535', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1535\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nCD206−/HLA-DR−) and HLA-DR+ M1-like (HLA-DR+/S100A4+/CD206−/\\nPD-L1−) (Fig. 1c,d). The expression of other macrophage markers \\naligned with this classification (Extended Data Fig. 1c). Third, we \\ndefined three subsets among the tumor-infiltrating CD8+ T cells (17,647 \\ncells, 0.01%): exhausted (PD-1 +/TIM3+/CD44−), effector (CD45RO +/\\nCD44+/PD-1−/TIM3−) or memory (CD45RO+/CD44−/PD-1−) CD8+ T cells \\n(Fig. 1c,d). Thus, through CODEX analysis we classified discrete subsets \\nof tumor cells, macrophages and CD8+ T cells (Fig. 1d and Supplemen-\\ntary Table 2).\\nT o validate our tumor and immune cell classification we used \\nthree approaches. First, we overlaid our unsupervised cell-calling on \\nimmunostained images and confirmed the accuracy of classification \\nbased on marker expression (Fig. 1e). Second, we demonstrated a \\nconsistent proportion of immune and stromal cell types across various \\nHCC clinical subgroups, indicating the reliability of the classification, \\nwith NASH HCC notably displaying increased CD8+ T cell exhaustion as \\npreviously reported19 (Extended Data Fig. 1d–f). Third, we noted that \\nthe observed subset identification and proportion of macrophage \\nand T cell subsets are similar to previous reports 20–22; however, we \\ndid note a higher proportion of neutrophils, possibly attributable to \\ntheir underrepresentation in transcriptome-based deconvolution \\nanalyses23,24 (Extended Data Fig. 2a). Thus, our CODEX analysis provides \\nan accurate and quantitative single-cell spatial profile of 20 major cell \\ntypes in human HCC.\\nResidual human HCC has enrichment of immunosuppressive \\ncells\\nNext, we used the above single-cell phenotyping to identify dif -\\nferences in tumor and immune cell subsets that would distinguish \\nresidual and primary HCC. The above 108 samples were composed \\nof two cohorts- residual HCC (n\\u2009=\\u200955; 427,126 cells) and primary HCC \\n(n\\u2009=\\u200953; 638,158 cells). Residual HCC was derived from patients with \\nHCC who had undergone transarterial chemoembolization (mean of \\n2.42 cycles; s.e.m.\\u2009=\\u20091.32) and retained post-treatment viable HCC in \\nthe liver explant. The control group was derived from patients with \\ntreatment-naive primary HCC. The two groups were matched for key \\nvariables, including age, sex, race, ethnicity, comorbidities, grade, \\nmicrovascular invasion, alphafetoprotein (AFP), clinical tumor stage \\nand pathological tumor stage, but not matched for etiology and cir-\\nrhosis (Supplementary Table 3). We could measure all 20 cell types in \\nboth residual and primary HCC. Thus, we could distinguish residual \\nand primary HCC through their distinct distribution of cellular sub-\\ntypes (Fig. 2a,b).\\nWe found higher infiltration of immunosuppressive tumor \\nand immune cell subsets in residual HCC than in primary HCC. \\nSpecifically, PD-L1 + macrophages (P adj\\u2009=\\u20092.2\\u2009×\\u200910 −6,), PD-L1 + tumor \\ncells (P adj\\u2009=\\u20093.4\\u2009×\\u200910 −5), mast cells (P adj\\u2009=\\u20093.1\\u2009×\\u200910 −4), exhausted CD8 \\nT cells (Padj\\u2009=\\u20096.6\\u2009×\\u200910−4) and natural killer (NK) cells (Padj\\u2009=\\u20091.8\\u2009×\\u200910−3) \\nwere more abundant in residual HCC (Fig. 2c–e and Extended Data \\nFig. 2b). Conversely, neutrophils (P adj\\u2009=\\u20091.4\\u2009×\\u200910 −4) and fibroblasts \\n(Padj\\u2009=\\u20091.5\\u2009×\\u200910−2) were relatively depleted in residual HCC compared to \\nprimary HCC (Fig. 2c and Extended Data Fig. 2c). These observations \\npersistence of stem cell-like residual tumor cells9. Second, dormancy \\nand recurrence can be promoted by protumor macrophages10–12. Our \\nprevious findings support this, as we observed an M2-like macrophage \\npopulation in mouse models of HCC that facilitates immune cell eva-\\nsion13 and metastasis 14. While tumor and immune factors may both \\ndrive HCC recurrence, their interplay remains unclear.\\nWe aimed to evaluate the interplay between cell-intrinsic and \\nimmune mechanisms contributing to the persistence of residual \\ntumor cells. T o do this, we analyzed both human samples of post-TACE \\nresidual human HCC and mouse models of MRD. First, we generated \\na spatial map of residual human HCC, which showed that interactions \\nbetween stem-like tumor cells and protumor macrophages, medi -\\nated by the TGFβ pathway, were linked to CD8+ T cell exhaustion. Sec-\\nond, we employed three distinct yet complementary mouse models: \\ntwo transgenic models mimicking MRD (TRE-MYC-Cebpb-tTA 9 and \\nTRE-MYC/Twist1/Luc-Cebpb-tTA 14), where oncogene inactivation \\ncaused tumor regression but allowed undetectable MRD to persist \\nand lead to recurrence, and a syngeneic orthotopic allograft model \\nmimicking chemoresistance. In these mouse models, we showed that \\ntargeting the TGFβ and PD-L1 pathways eliminated residual tumor cells \\nand prevented recurrence.\\nResults\\nSingle-cell spatial atlas reveals heterogeneity in human HCC\\nT o elucidate the spatial organization of tumor and immune cells in \\npost-chemoembolization residual HCC, we constructed a comprehen-\\nsive single-cell spatial map of human HCC. We microdissected areas \\nof viable HCC, constructed formalin-fixed paraffin-embedded (FFPE) \\ntissue microarrays (TMA) and used co-detection by indexing (CODEX)15 \\nto analyze the spatial architecture. We imaged a total of 1.07 million \\ncells from 108 HCC samples with multiplex staining using 41 antibod-\\nies targeting tumor, immune, stromal and functional markers (Fig. 1a, \\nSupplementary Fig. 1 and Supplementary Table 1). Through cellular \\nsegmentation, marker quantification, normalization and unsupervised \\nclustering, we identified 11 major cell types within HCC (Fig. 1b  and \\nExtended Data Fig. 1a). Thus, by utilizing FFPE-optimized CODEX, we \\nachieved highly multiplexed single-cell marker visualization and cel-\\nlular phenotyping in human HCC.\\nWe used canonical marker expression in CODEX to further subset \\nthree major cell types: tumor cells, macrophages and T cells (Fig. 1c,d). \\nFirst, among the tumor cells we identified three discrete subsets. Two \\nwere stem cell-like, cytokeratin 19 (CK19) + (6.7%, 34,436 cells) and \\nepithelial cellular adhesion molecule (EpCAM) + (5.2%, 26,622 cells) \\ntumor cells (Fig. 1c,d). They expressed stemness, prosurvival and mes-\\nenchymal markers (Extended Data Fig. 1b). A third subset of tumor \\ncells expressed the immune checkpoint PD-L1 (3.8%, 19,737 cells). The \\nheterogeneity observed among these tumor cell subsets aligned with \\nprevious reports from single-marker studies16–18. Second, among the \\nhost immune cells, we identified the CD68 + tumor-associated mac -\\nrophages (TAMs) as the most common immune cell subset (14.5%, \\n154,899 cells). Three subsets of TAMs were identified: CD206+ M2-like \\n(CD206+/CD163+/HLA-DR−/PD-L1−), PD-L1+ M2-like (PD-L1 +/CD11b+/\\nFig. 1 | Single-cell spatial atlas of the tumor immune microenvironment in \\nHCC. a, Single-cell spatial analysis workflow of human HCC tumor cores using \\nCODEX to determine cell frequency, cell–cell interactions and neighborhoods. \\nb, Heatmap dendrogram of unsupervised clustering using canonical marker \\nexpression to define major tumor and immune cell subsets. Heatmap scaled \\nby column. c, Subclassification of tumor cells, macrophages and T cells \\nusing cell-specific canonical markers. Heatmap is scaled by row. d, Graphical \\nrepresentation of the 20 cell types and subtypes with their absolute cell counts \\nand proportions in the CODEX dataset: tumor cells (tumor cell CK+ n\\u2009=\\u2009433,918 \\ncells (40.73%), tumor cell CK19+ n\\u2009=\\u200934,436 (3.23%), tumor cell EpCAM+ n\\u2009=\\u200926,622 \\n(2.50%), tumor cell PD-L1+ n\\u2009=\\u200919,737 (1.85%)); macrophages (macro CD206+ \\nn\\u2009=\\u200945,628 (4.28%), macro HLA-DR+ n\\u2009=\\u200955,558 (5.22%), macro PD-L1+ n\\u2009=\\u200953,713 \\n(5.04%)); T cells (CD4+ T cell n\\u2009=\\u200944,396 (4.17%), CD8+ T cell Eff n\\u2009=\\u20094,366 \\n(0.41%), CD8+ T cell Exh n\\u2009=\\u20099,539 (0.90%), CD8+ T cell mem n\\u2009=\\u20093,742 (0.35%), \\nTreg n\\u2009=\\u200923,280 (2.19%)); B cell n\\u2009=\\u200929,155 (2.74%), DC n\\u2009=\\u20095,769 (0.54%); mast \\ncell n\\u2009=\\u20097,389 (0.69%); monocyte n\\u2009=\\u20091,490 (0.14%); NK cell n\\u2009=\\u200913,812 (1.30%); \\nneutrophil n\\u2009=\\u200955,927 (5.25%); endothelial cell n\\u2009=\\u200945,618 (4.28%); fibroblast \\nn\\u2009=\\u2009151,189 (14.19%). e, CODEX imaging representations of the canonical markers \\nused to verify the validity of the unsupervised clustering for each cell type. The \\ncell type (white filling) is overlaid on the immunostained images. abs, antibodies; \\nPanCK, pan cytokeratin; CK19, cytokeratin 19; CD, cluster of differentiation; \\nHLA-DR, human leukocyte antigen DR isotype; TIM3, T cell immunoglobulin and \\nmucin domain-containing protein 3; DC, dendritic cell; Treg, regulatory T cell; Exh, \\nexhausted; Eff, effector.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e1d25d3f-9adf-4730-8d9c-ac1f243d475b', embedding=None, metadata={'page_label': '1536', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1536\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nremained consistent within the subgroup of HCCs arising in a cir -\\nrhotic background (Extended Data Fig. 2d). Moreover, we show that \\nM2-like macrophage enrichment was specific to the presence of \\ntherapy-resistant residual tumors post-TACE, rather than treatment \\neffects, as evidenced by higher macrophage levels in TACE-refractory \\nHCC25 compared to TACE-exposed peritumoral cirrhotic liver tissue \\nt-SNE2\\nt-SNE1\\n0\\n0\\n–20\\n–20\\n20\\n20\\nSpatial immune\\npro/f.shortiling\\nCODEX\\n1. Collect HCC tissue \\nand create TMA\\n2. Microdissect \\nareas of interest\\n3. Stain with oligo- \\ntagged antibodies\\n4. Repeat cycles \\nwith 41-plex panel\\n5. Segmentation, \\nclustering\\n6. Spatial analysis\\nCycle1\\nCycle2\\nCycle3\\nCycle4\\nCycle5\\nCycle6\\nCycle7\\nCycle8\\nDye1 Dye2 Dye3\\na. Cell frequencies\\nb. Cell interactions c. Neighborhoods\\nTumor cell \\nCK19 +\\nTumor cell \\nEpCAM +\\nTumor cell \\nPD-L1\\nTumor cell\\nPan CK Pan CK Pan CK Pan CK\\nCK19 EpCAM PD-L1 E cadherin\\nCell calling Cell calling Cell calling Cell calling\\nCD8 +  T \\nexhausted\\nCD8 +  T \\nmemory\\nCD8 + T \\neﬀector\\nCD4 +  T cell\\nCell calling Cell calling Cell calling\\nCell calling\\nCD8 CD8 CD8\\nPD-1 CD45RO CD44\\nFibroblast Endothelial cell\\nαSMA\\nVimentin\\nCD31\\nVimentin\\nCell calling Cell calling\\nHLA-DR\\nM1-like\\nHLADR +\\nM2-like\\nCD206 +\\nM2-like\\nPD-L1 +\\nTumor cells Macrophages CD8 +  T cells\\nStromal cells Lymphoid cells\\nCell calling\\nCD68 CD68\\nCell calling Cell calling\\nCD68\\nCD206 PD-L1\\nTumor cells Macrophages\\nT cells CD8T cells\\nB cell\\nCD19\\nCD20\\nCell callingCell calling\\nCD4\\nCD4CD3\\nFoxP3\\nNK cell\\nCell calling\\nCD11b\\nCD56\\nT reg\\nPanCK\\nCD68\\nαSMA\\nCD31\\nCD20\\nCD45\\nCD3\\nCD4\\nCD8\\nCD15\\nCD56\\nCD117\\nCD11c\\nCD14\\nTumor cell\\nTumor cell CK\\n+\\nTumor cell CK19\\n+\\nTumor cell EpCAM\\n+\\nTumor cell PDL1\\n+\\nMacrophage Fibroblast Endothelial cell B cell T cell Neutrophil NK cell Mast cell Dendritic cell Monocyte\\nCK19\\nEpCAM\\nPDL1\\nCD8\\nCD4\\nFoxP3\\nPD-L1\\nHLA-DR\\nCD206\\nCD45RO\\nPD1\\nTIM3\\nCD44\\nM1-like macro M2-like macro M2-like macro \\nCD4\\n+  T cell\\nT reg\\n cells\\nCD8\\n+  T cell\\nCD8\\n+  T eﬀector\\nCD8\\n+  T exhausted\\nCD8\\n+  T memory\\nT cells\\nCD8 +  T\\nmem \\nCD8 +  T\\nexh\\n(62,043, 5.8%)\\nCD8 +  T\\neﬀ\\nCK19 +\\n(34,436, 6.7%)\\n  CK +\\n(433,918, 84.3%)\\nEpCAM +\\nPD-L1 +\\n(19,737, \\n3.8%)\\n(26622, \\n5.2%)\\nTumor cells\\n(514,713 cells, 48.3%)\\nCD206 +\\nHLA-DR +\\n(55,558, \\n35.9%)\\n(45,628, \\n29.5%)\\n(53,713, \\n34.7%)\\nPD-L1 +\\nFibroblasts\\nMacrophages\\nCD4 +  T\\n(151,189, 14.2%)\\nB cells\\nNeutro\\nphils\\n(55,927,\\n5.3%)\\nT reg\\nNK cell\\nEndothelial cell\\n(45,618, 4.3%) (29,155, \\n2.7%)\\n(154,899, 14.5%)\\n(23,280,\\n2.2%)\\nMast\\nDC\\nMono\\nMarker expression\\nCluster\\n10 µm 10 µm 10 µm\\nMyeloid cells\\nMast cell\\nCell calling\\nCD45\\nCD117\\nDendritic cell Neutrophil\\nCD45\\nCD15\\nCD45\\nCD11c\\nCell calling Cell calling\\nMonocyte\\nCD45\\nCD14\\nCell calling\\n10 µm10 µm10 µm\\na\\nb c\\ne\\nd', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='581e76c7-a442-4db8-a357-f3ed77c39c49', embedding=None, metadata={'page_label': '1537', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1537\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\n(Extended Data Fig. 2e–g) or sites of post-TACE complete response \\n(Extended Data Fig. 2h), with results being consistent across dif -\\nferent TACE methods (Extended Data Fig. 2i). Hence, residual HCC \\nshowed a more protumor redistribution of specific immune cells \\nthan primary HCC.\\nIn addition to changes in cellular distribution, we observed four dif-\\nferences in cellular phenotype between residual and primary HCC. First, \\ntumor cells in residual HCC showed a more stem-like (CD44high, CK19high) \\nphenotype (Fig. 2f). Second, the abundant PD-L1+ macrophages within \\nthe residual HCC were more likely to be monocyte-derived (CD11bhigh) \\nand migratory (podoplaninhigh) (Fig. 2g). Third, the CD8+ T and NK cells \\ninfiltrating residual HCCs were more exhausted, with higher PD-1 and \\nTIM3 expression (Fig. 2h). Thus, residual HCC seems to be character-\\nized by cancer stemness, enrichment of protumor macrophages and \\ndecreased immune surveillance.\\nSpatial interactions in residual HCC drive immune evasion\\nWe hypothesized that the differences observed between residual \\nand primary HCC in both the tumor and immune cellular subtypes \\nmay be related to cellular interactions. T o test this, we integrated \\nsingle-cell phenotype data with spatial coordinates to analyze interac-\\ntions between nonhomotypic cells in direct contact with each other, \\nwhich we defined as within a 25-μm radius. Based on the above results, \\nwe focused on interactions among tumor cells, macrophages and \\nCD8+ T cells.\\nWe first compared the interactions of tumor cells in residual ver-\\nsus primary HCC (Fig. 3a, Extended Data Fig. 3a). The most significant \\nchange was seen in tumor cell interactions with PD-L1+ macrophages \\n(Fig. 3b). In residual HCC, all four types of tumor cells we identified \\nhad more frequent direct interactions with PD-L1+ macrophages com-\\npared to primary HCC (Fig. 3b ). This was true, whether tumors had \\nhigh or low levels of PD-L1+ macrophages (Extended Data Fig. 3b). In \\ncontrast, tumor cells did not differentially interact with exhausted \\nCD8+ T cells, despite the increased presence of the latter in residual \\nHCC (Fig. 3c). Among the interactions of tumor cells, those between \\nEpCAM+ tumor cells and either M2-like PD-L1+ or CD206+ macrophages \\nwere associated with poor recurrence-free survival in residual HCC \\n(Fig. 3d). We validated this finding in The Cancer Genome Atlas (TCGA) \\ncohort by showing that overexpression of genes representing the \\ninteraction between stem-like cancer cells (EPCAM, KRT19 and CD44) \\nand M2-like macrophages (CD68, CD274, MRC1 and CD163) was indeed \\nassociated with a significantly poorer recurrence-free survival in \\nHCC (P\\u2009=\\u20095.2\\u2009×\\u200910 −5) (Extended Data Fig. 3c). Further, by employing \\nthree-dimensional (3D) co-culture tumoroid in vitro experiments, \\nwe showed that co-culture of HCC cancer cells with M2-like mac -\\nrophages induced more cancer stemness than co-culture with M1-like \\nmacrophages (Extended Data Fig. 3d). Thus, in residual HCC, stem-like \\ntumor cells spatially interact with M2-like PD-L1 + macrophages, cor-\\nrelating with worse clinical outcomes.\\nNext, we focused on the interactions of macrophages (Extended \\nData Fig. 3e). M2-like PD-L1 + macrophages were found to more fre -\\nquently directly interact with exhausted CD8 + T cells (P \\u2009=\\u20091.11×10−3) \\n(Fig. 3e) but not effector (P\\u2009=\\u20090.17) or memory CD8+ T cells (P\\u2009=\\u20090.38) \\nin residual HCC. These interactions of PD-L1+ macrophages in residual \\nHCC seemed to occur predominantly within fibrovascular bundles. \\nPD-L1+ macrophages interacted more frequently with both fibroblasts \\n(Padj\\u2009=\\u20092.99\\u2009×\\u200910−3) and endothelial cells (P\\u2009=\\u20097.7\\u2009×\\u200910−3) in residual than \\nprimary HCC (Fig. 3f ). Similarly, we found that there was closer spa-\\ntial proximity between PD-L1 + macrophages and fibroblasts (44.99 \\nversus 179.75\\u2009μm; P\\u2009=\\u20090.027) or endothelial cells (36.2 versus 68.5\\u2009μm; \\nP\\u2009=\\u20090.004) in residual than primary HCC. This was not observed for the \\nother two subsets of HLA-DR+ or CD206+ macrophages and fibroblasts \\n(42.2 versus 38.6\\u2009μm P\\u2009=\\u20090.329; 30.3 versus 31.9, P\\u2009=\\u20090.197, respectively). \\nThus, in residual HCC, PD-L1+ macrophages seem to directly interact \\nwith, and may result in the exhaustion of CD8+ T cells within fibrovas-\\ncular bundles.\\nWe next examined interactions that did not occur by direct contact \\nbut could still be mediated indirectly by intermediary cells. T o first \\nestablish the range of indirect influence of a central cell, we developed \\na regression model (Supplementary Fig. 2). We found that the variance \\nin a given central cell’s expression of Ki67 and BCL2 could be explained \\nby evaluating an interaction radius ranging from 25–100\\u2009μm (Fig. 3g \\nand Extended Data Fig. 3f). Based on this, we evaluated indirect interac-\\ntions between nonhomotypic cells lying between 25\\u2009μm and 100\\u2009μm \\nof a given cell. We found that in residual versus primary HCC, all four \\ntumor cell subsets more frequently indirectly interacted with PD-L1+ \\nmacrophages (Fig. 3h and Extended Data Fig. 3g,h); however, differ-\\nent from what we observed for direct interactions, we found that in \\nresidual HCC, the tumor cells did indirectly interact more frequently \\nwith exhausted CD8 + T cells (Fig. 3i). This raises the possibility that \\nPD-L1+ macrophages could serve as an intermediary facilitating indirect \\ninteractions between tumor cells and exhausted CD8 + T cells. Thus, \\nresidual HCC exhibits remodeling in cellular interactions between \\nstem-like tumor cells, PD-L1+ macrophages, and exhausted CD8+ T cells \\nwhich could be responsible for changes in immune surveillance, as we \\nexplain below.\\nSpatial neighborhoods reprogram immune cells in residual \\nHCC\\nWe wondered whether the remodeling of cell–cell interactions in \\nresidual HCC had a higher-order spatial organization. T o test this, we \\ndefined ‘cellular neighborhoods (CNs)’ aiming to capture the intricate \\nFig. 2 | Immunosuppressive cell enrichment and tissue remodeling in \\nresidual HCC. a, Immune and tumor cell proportions in residual HCCs from \\nchemoembolization-treated liver explants (residual HCC, n\\u2009=\\u200955 patients) \\ncompared to primary resected primary HCC (n\\u2009=\\u200953 patients). b, H&E and CODEX \\nimaging representations of a residual and primary HCC tumor core. In each \\nCODEX image, eight canonical markers (DAPI, CD31, CD4, CD15, CD68, PanCK, \\nCD8 and αSMA) are overlaid onto the image in different colors. c, Volcano plot \\nshowing the differential enrichment of tumor and immune cell populations \\n(with statistical significance) in residual (n\\u2009=\\u200955 patients) versus primary HCCs \\n(n\\u2009=\\u200953 patients). Two-tailed unpaired t-test shown with Bonferroni correction \\nfor multiple comparisons was used. d, Proportion of PD-L1+ macrophages, PD-L1+ \\ntumor cells, mast cells and exhausted T cells in residual (n\\u2009=\\u200955 patients) versus \\nprimary HCCs (n\\u2009=\\u200953 patients). Box indicates 25th–75th percentile, whiskers \\nshow 5th–95th percentile and the line shows the median e, Representative \\nVoronoi plot of cores showing PD-L1+ macrophage, PD-L1+ tumor cells, mast cells \\nand exhausted T cells density in residual (n\\u2009=\\u200955 patients) versus primary HCCs \\n(n\\u2009=\\u200953 patients). f, Comparison of normalized expression of stemness marker \\nCD44 and CK19 marker in tumor cells from residual (n\\u2009=\\u2009217,105 cells) and primary \\nHCC (n\\u2009=\\u2009297,608 cells) (both Padj value\\u2009<\\u20092.2\\u2009×\\u200910−308). Two-tailed unpaired \\nt-test shown with Bonferroni correction for multiple comparisons. The curve \\nrepresents the probability density function of the normalized expression levels, \\nwith the peak indicating the mode. g, Comparison of normalized expression of \\nCD11b and podoplanin in PD-L1+ macrophages from residual (n\\u2009=\\u200947,605 cells) and \\nprimary HCC (n\\u2009=\\u20096,108 cells) (both Padj value\\u2009<\\u20092.2\\u2009×\\u200910−308). Two-tailed unpaired \\nt-test shown with Bonferroni correction for multiple comparisons. The curve \\nrepresents the probability density function of the normalized expression levels, \\nwith the peak indicating the mode. h, Comparison of normalized expression \\nof PD-1 and TIM3 in T cells and NK cells from residual (T cells, n\\u2009=\\u200916,549 cells; \\nNK cells n\\u2009=\\u200911,701 cells) and primary HCC (T cells, n\\u2009=\\u200945,494 cells; NK cells \\nn\\u2009=\\u20092,111 cells) (both Padj value\\u2009<\\u20092.2\\u2009×\\u200910−308). Two-tailed unpaired t-test shown \\nwith Bonferroni correction for multiple comparisons. The curve represents \\nthe probability density function of the normalized expression levels, with the \\npeak indicating the mode. Statistical significance was assessed by an unpaired, \\ntwo-tailed t-test, Bonferroni adjustment was used for P values. CD, cluster of \\ndifferentiation; αSMA, smooth muscle alpha actin; BCL2, B cell lymphoma 2;  \\nSig, significant.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='7aa1162e-c34c-4341-ad52-e1cbce5c8a75', embedding=None, metadata={'page_label': '1538', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1538\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nNK cell Monocyte B cell T reg CD8 +  T cell exh CD4 +  T cell Macro HLADR + Endothelial cell Tumor cell PD-L1 + Tumor cell CK19 +\\nNeutrophil Mast cell Denditic cell CD8 +  T cell mem CD8 +  T cell eﬀ Macro PD-L1 + Macro CD206 + Fibroblast Tumor cell EpCAM + Tumor cell CK +\\nProportion\\n1.00\\n0.75\\n0.50\\n0.25\\n0\\n–2.5 0 2.5 5.0\\nSig\\nNot sig\\nPD-L1 +  macrophage\\n–log P value\\nT-statistic (standard mean difference)\\nPD-L1 +  tumor cell\\nMast cell\\nCD8 +  T cell exh\\nNK cell\\nNeutrophil\\nFibroblast\\n4\\n3\\n2\\n1\\n0\\nResidual HCC versus primary HCC\\n20 µm \\n100 µm \\nResidual HCC (H&E) Residual HCC (CODEX) Primary HCC (H&E) Primary HCC (CODEX)\\nExpression on tumor cells Expression on T cells Expression on NK cellsExpression on PD-L1 +  macrophages\\nPrimary HCC Residual HCC\\nResidual HCC\\nPrimary HCC\\nCD44 CK19 CD11b Podoplanin PD-1 TIM3 PD-1 TIM3\\n–2 0 2 4 –2 0 2 4 –2 0 2 4 –2 0 2 4 –2 0 2 4 –2 0 2 4 –2 0 2 4–0.5 0 0.5 1.0\\nP < 2 × 10\\n–308\\nP < 2 × 10\\n–308\\nP < 2 × 10\\n–308\\nNormalized\\nexpression\\nNormalized\\nexpression\\nNormalized\\nexpression\\nNormalized\\nexpression\\nNormalized\\nexpression\\nNormalized\\nexpression\\nNormalized\\nexpression\\nNormalized\\n`expression\\nP < 2 × 10\\n–308\\nP < 2 × 10\\n–308\\nP < 2 × 10\\n–308\\nP < 2 × 10\\n–308 P < 2 × 10\\n–308\\na\\nb c\\nd\\nf g h\\nDAPI\\nCD31\\nCD4\\nCD15\\nCD68\\nPanCK\\nCD8\\nαSMA\\nDAPI\\nCD31\\nCD4\\nCD15\\nCD68\\nPanCK\\nCD8\\nαSMA\\n51\\n85\\n62\\n130\\n127\\n115\\n78\\n141\\n129\\n87\\n117\\n112\\n76\\n99\\n114\\n119\\n113\\n122\\n159\\n139\\n123\\n110\\n126\\n75\\n135\\n109\\n125\\n147\\n86\\n155\\n176\\n102\\n163\\n152\\n111\\n98\\n118\\n101\\n124\\n140\\n90\\n138\\n89\\n100\\n133\\n74\\n168\\n116\\n131\\n10\\n88\\n142\\n120\\n45\\n33\\n48\\n44\\n29\\n69\\n70\\n64\\n52\\n57\\n49\\n143\\n38\\n53\\n63\\n60\\n13\\n46\\n50\\n58\\n11\\n24\\n23\\n37\\n2\\n71\\n12\\n22\\n16\\n27\\n47\\n4\\n61\\n36\\n39\\n56\\n25\\n40\\n21\\n34\\n3\\n41\\n72\\n178\\n26\\n9\\n59\\n35\\n15\\n10\\n14\\n32\\n1\\n28\\n17\\nProportion\\nPrimary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC\\n0.5\\n0.4\\n0.3\\n0.2\\n0.1\\n0\\n0.3\\n0.2\\n0.1\\n0\\n0.20\\n0.15\\n0.10\\n0.05\\n0\\n0.100\\n0.075\\n0.050\\n0.025\\n0\\n0 1 2 3 4\\nx (1,000 px)\\ny (1,000 px)\\n4\\n3\\n2\\n1\\n0\\n0 1 2 3 4\\nx (1,000 px)\\n0 1 2 3 4\\nx (1,000 px)\\n0 1 2 3 4\\nx (1,000 px)\\ny (1,000 px)\\n4\\n3\\n2\\n1\\n0\\n0 1 2 3 4\\nx (1,000 px)\\ny (1,000 px)\\n4\\n3\\n2\\n1\\n0\\n0 1 2 3 4\\nx (1,000 px)\\ny (1,000 px)\\n4\\n3\\n2\\n1\\n0\\n0 1 2 3 4\\nx (1,000 px)\\n0 1 2 3 4\\nx (1,000 px)\\nTumor cell PD-L1 +\\nResidual-HCC\\nResidual-HCC\\nPrimary-HCC Residual-HCCPrimary-HCC\\nResidual-HCCPrimary-HCCPrimary-HCC\\nMacrophage PD-L1 + CD8 +  T cell exhausted\\nMast cell\\nMacrophage \\nPD-L1 +\\nTumor cell\\nPD-L1 + Mast cell\\nCD8 +  T cell\\nexhausted\\nP = 4.5 × 10 –5 P = 3.4 × 10 –4 P = 1.5 × 10 –3 P = 2.7 × 10 –3\\ne', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d13629a7-9de3-43fd-a812-0e92738c2429', embedding=None, metadata={'page_label': '1539', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1539\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\n0.2\\n0.4\\n0.6\\n0 0\\n0\\n0.5\\n1.0\\n0\\n0.5\\n1.0\\n0.1\\n0.2\\n0.3\\nTumor cell\\nCK +\\nTumor cell\\nCK19 +\\nTumor cell\\nEpCAM +\\nTumor cell\\nPD-L1 +\\nTumor cell CK +\\nTumor cell\\nCK19 +\\nTumor cell\\nEpCAM +\\nTumor cell PD-L1 +\\nCD4 +  T cell\\nEndothelial cell\\nMacro CD206 +\\nMacro PD-L1 +\\nMast cell\\nNeutrophil\\nNK cell\\nMacro HLADR +\\nPrimary HCC Residual HCC\\nTumor cell CK +\\nTumor cell CK19 +\\nTumor cell\\nEpCAM +\\nTumor cell PD-L1 +\\nCD4 +  T cell\\nEndothelial cell\\nMacro CD206 +\\nMacro PD-L1 +\\nMast cell\\nNeutrophil\\nNK cell\\nMacro HLADR +\\nSigni/f.shorticant differential interactions of tumor cells Direct interactions of tumor cells: PD-L1 +  macrophages\\nCD68/CD206 DAPI/CD68/CD206\\nPanCK/EpCAMDAPI PanCK/EpCAM\\n5 µm\\nEpCAM+ tum cell: CD206+ macrophage interactions\\nEpCAM +  tum cell: PDL1 +  macrophage interactions\\nPD-L1 +  macrophage: CD8 +  T exh interactions\\nIndirect interactions\\n(25–100 µm)\\nDirect interactions\\n(<25 µm)\\nCentral cell\\nDe/f.shortining indirect\\ninteractions\\nVariance in marker expression\\nPrim HCC Res HCC Prim HCC Res HCC\\nPD-L1 +  macrophage: /f.shortibroblast interactions\\nTumor cell CK +\\nTumor cell CK19 +\\nTumor cell PD-L1 +\\nTumor cell EpCAM +\\nTumor cell CK +\\nTumor cell CK19 +\\nTumor cell PD-L1 +\\nTumor cell EpCAM +\\n0 0.80.4\\n0 0.80.4\\n0 0.80.4 0 0.80.4\\nPrim HCC Res HCC\\nP = 2.58 × 10 –10\\nP = 3.09 × 10 –8\\nP = 1.27 × 10 –5\\nP = 2.42 × 10 –2\\nDirect interactions of tumor cells: CD8 +  T exh cells\\nP = 0.201\\nP = 0.836\\nP = 0.778\\nP = 0.443\\nCD68/PDL1 DAPI/CD68/PDL1\\nPanCK/aSMA/CD31DAPI/PanCK aSMA/CD31\\n5 µm\\nCD68/CD8 DAPI/CD68/PDL1\\nPanCK/CD8/PD1DAPI/PanCK PDL1/PD1\\n5 µm\\nCD68/PDL1 DAPI/CD68/PDL1\\nPanCK/EpCAMDAPI/PanCK PanCK/EpCAM\\n5 µm\\nNormalized freq\\nof interaction\\nNormalized freq\\nof interaction\\nPrim\\nHCC\\nRes \\nHCC\\nPrim\\nHCC\\nRes \\nHCC\\nPD-L1\\n+\\n mac:\\nFibroblast\\nPD-L1\\n+\\n mac:\\nEndothelial cell\\nPD-L1\\n+\\n mac:\\nCD8\\n+\\n T exh\\nPD-L1\\n+\\n mac:\\nCD8\\n+\\n T eﬀ\\n1.1 × 10\\n–3\\n0.172\\n2.9 × 10\\n–3 7.7 × 10\\n–3\\nPrim HCC Res HCC\\nP = 3.48 × 10 –16\\nP = 3.36 × 10 –11\\nP = 1.60 × 10 –2\\nP = 7.55 × 10 –7\\nP = 1.26 × 10 –3\\nP = 3.51 × 10 –4\\nP = 0.334\\nP = 4.47 × 10 –2\\nIndirect interactions \\ntumor cells: PD-L1 +  mac\\nIndirect interactions \\ntumor cells: CD8 +  T exh\\n25–40 µm ring\\n40–100 µm ring\\n100 µm\\n40 µm\\n25\\nRadius (µm)\\nR2\\nKi67 BCL2\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\n0 5 10\\n0\\n50\\n100\\nRecurrence-free survival\\nEpCAM +  tum: CD206 +  Mac\\nHigh\\nEpCAM +  tum: CD206 +  Mac\\nLow\\nP = 0.01\\nHR 5.3 (1.3–16.0)\\nSurvival (%)\\nn = 31\\nn = 24\\n0 5 10\\n0\\n50\\n100\\nRecurrence-free survival\\nEpCAM +  tum: PD-L1 +  Mac\\nHigh\\nEpCAM +  tum: PD-L1 +  Mac\\nLow\\nP = 0.04\\nHR 3.9 (1.1–14.6)\\nSurvival (%)\\nn = 26\\nn = 29\\na b\\nc\\nd e\\nf\\ng h i\\nFig. 3 | Remodeling of spatial cellular interactions in residual HCC.  \\na, Alluvial plot of tumor cell subtype and immune cell interactions in residual \\n(n\\u2009=\\u200955 patients) and primary HCCs (n\\u2009=\\u200953 patients). The height of each unit \\nis proportional to the frequency of interaction between two cell types. Only \\nsignificant interactions with Padj\\u2009<\\u20090.05 are depicted. b, Size-modulated circular \\nheatmap showing mean frequency and adjusted P values of direct interactions \\nbetween tumor cell subtypes and PD-L1+ macrophages in residual (n\\u2009=\\u200955 \\npatients) and primary HCCs (n\\u2009=\\u200953 patients). c, Size-modulated circular heatmap \\nshowing mean frequency and adjusted P values of direct interactions between \\ntumor cell subtypes and exhausted CD8 T cells in residual (n\\u2009=\\u200955 patients) and \\nprimary HCCs (n\\u2009=\\u200953 patients). d, Kaplan–Meier plots signifying the recurrence-\\nfree survival of tumors stratified by median frequency of interaction between \\nEpCAM+ tumor cells and either CD206+ or PD-L1+ macrophages in residual HCC \\n(n\\u2009=\\u200955 patients). CODEX representative IF images demonstrate the interaction \\nbetween representative cells. log-rank test used to statistically compare the \\ngroups. e, CODEX imaging representations of PD-L1+ macrophage interactions \\nwith exhausted and effector CD8+ T cells, shown with box plots quantifying the \\nproportion of interactions in primary (n\\u2009=\\u200953 patients) compared to residual \\nHCC (n\\u2009=\\u200955 patients). Box shows 25th–75th percentiles, whiskers show 5th–95th \\npercentiles and the line indicates the median. f, CODEX imaging representations \\nof PD-L1+ macrophage interactions with fibroblasts and endothelial cells, shown \\nwith bar plots quantifying the proportion of interactions in primary (n\\u2009=\\u200953 \\npatients) compared to residual HCC (n\\u2009=\\u200955 patients). g, Model of direct and \\nindirect interactions with a single central cell using the scalar variable ‘cell–cell \\ndistance’ . R2 values for predicting Ki67 and BCL2 marker expression in radius \\nof increasing sizes. Volcano plots show that increasing neighborhood sizes \\nexplain more variance in expression of a central cell’s Ki67 and BCL2 (n\\u2009=\\u20091.07 \\nmillion cells each). R2 values for Ki67 are higher on average for a neighborhood \\nradius of 100\\u2009μm versus a neighborhood radius of 25\\u2009μm (two-sided Wilcoxon \\nrank-sum Padj value 8.48\\u2009×\\u200910−6) and for BCL2 (two-sided Wilcoxon rank-sum \\nPadj value 1.76\\u2009×\\u200910−5). Violin/boxplot shows 25th–75th percentiles, whiskers \\nshow 5th–95th percentiles and the line indicates the median. h, Size-modulated \\ncircular heatmap showing mean frequency and adjusted P values of indirect \\ninteractions between tumor cell subtypes and PD-L1+ macrophages in residual \\n(n\\u2009=\\u200955 patients) and primary HCCs (n\\u2009=\\u200953 patients). i, Size-modulated circular \\nheatmap showing mean frequency and adjusted P values of frequency of indirect \\ninteractions between tumor cell subtypes and exhausted CD8+ T cells in residual \\n(n\\u2009=\\u200955 patients) and primary HCCs (n\\u2009=\\u200953 patients). Statistical significance was \\nassessed by an unpaired, two-tailed t-test and BH adjustment was used for  \\nP values.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='94aae96d-6e77-4a2e-8e64-59543bdae999', embedding=None, metadata={'page_label': '1540', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1540\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\n–0.75\\n–0.50\\n–0.25\\n0\\n0.25\\nMean\\nf\\ng\\nh\\ni j\\na b\\nc\\nd e\\nC3 C3 C3 C3C4 C4 C4 C4\\nExpression level\\nExpression level\\n1\\n2\\n3\\n0\\n4\\n2\\n4\\n0\\n6\\n0\\n2\\n4\\n6\\n0\\n1\\n2\\n3\\n4\\nNS NS\\n50\\n25\\n0\\n–25\\n50\\n25\\n0\\n–25\\n–40 –20 0 20 40 –40 –20 0 20 40\\nDoxRPatient-derived\\n3D tumoroids\\nPatient-derived\\nmonocytes\\nDifferentate to\\nmacrophages\\nCheck PD-L1\\non macrophages\\n1 2 3\\nTreat MoM with CM of \\npatient-derived tumoroids\\nControl\\n0\\n0.25\\n0.50\\n0.75\\n1.00\\nPrimary HCC Residual HCC\\nProportion\\nSpatial structures 10-cell clusters\\nClusters\\nNeighborhood 1\\nNeighborhood 2\\nNeighborhood 3\\nNeighborhood characterization\\n#1\\n#2\\n#3\\nIterative classi/f.shortication\\nVascular in/f.shortlamm\\ntumor CN\\nInnate immune\\nCN\\nT cell \\nimmune CN\\nPauci immune\\ntumor CN\\nFibro in/f.shortlamm\\nimmune CN\\nCK19\\n+\\n tumor\\nCN\\nEpCAM\\n+\\n tumor\\nCN\\nExpression on T cellsExpression on tumor cells\\nM2-macrophage CN in residual HCC Stem-like tumor cell CN in residual HCC\\nNeighborhood distribution\\nPauci immune CN Pauci immune CN Pauci immune CN\\nM2 macro CN\\nT cell immune CN\\nM2 macro CN\\n4 × 10\\n–5\\n0.023 8.5 × 10\\n–7\\n0.012 0.7 0.32 0.18 0.17\\n0\\n0.3\\n0.6\\n0.9\\nProportion\\nPrimary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC Primary HCC Residual HCC\\nT reg\\nTumor cell EpCAM +\\nTumor cell CK19 +\\nFibroblast\\nCD8 +  T cell exh\\nCD8 +  T cell mem\\nCD4 +  T cell\\nCD8 +  T cell eﬀ\\nMacro HLADR +\\nNeutrophil\\nDendritic cell\\nMonocyte\\nTumor cell CK +\\nNK cell\\nMast cell\\nTumor cell PD-L1 +\\nB cell\\nEndothelial cell\\nMacro CD206 +\\nMacro PD-L1 +\\nM2-macrophage immune CN\\n4\\n2\\n0\\n–2\\n–4\\nT cell immune CN\\nCK19 +  tumor CN\\nEpCAM +  tumor CN\\nVascular in/f.shortlammatory tumor CN\\nPauci immune tumor CN\\nFibro in/f.shortlammatory immune CN\\nInnate immune CN\\nOther\\nM2-macrophage immune CN\\nT cell immune CN\\nCK19\\n+\\n tumor CN\\nEpCAM\\n+\\n tumor CN\\nVascular in/f.shortlammatory tumor CN\\nOther\\nPauci immune tumor CN\\nFibro in/f.shortlammatory immune CN\\nInnate immune CN\\n0 5 10 15\\n0\\n50\\n100\\nEpCAM CN–\\nEpCAM CN+\\nP = 0.02\\nHR 5.0 (1.2–21.3)\\nYears\\nSurvival (%)\\nEpCAM +  tumor CN \\nin residual HCC\\nMean\\nPD-1\\nTIM3\\nFoxP3\\nCD44\\nCD45RO\\nCD45RA\\nEpCAM\\nKeratin 19\\nCD44\\nVimentin\\nBCL2\\nKi67\\n0\\n0.5\\n1.0\\nP < 1.0 × 10\\n–305\\nP = 1.3 × 10\\n–204\\nP < 1.0 × 10\\n–305\\nP < 1.0 × 10\\n–305\\nP = 3.3 × 10\\n–17\\nP = 1.9 × 10\\n–131\\nP = 8.5 × 10\\n–32\\nP < 1.0 × 10\\n–305\\nP < 1.0 × 10\\n–305\\nP = 1.1 × 10\\n–196\\nP = 1.2 × 10\\n–315\\nP = 1.6 × 10\\n–199\\nEpCAM\\n+  tumor CN\\nCK19\\n+  tumor CN\\nCD163\\nPD-1\\nTIM3\\nCD206\\nCD163\\nPD-1\\nCD206\\nTIM3\\n00 0.50.5 1.0\\n1.0\\nExpression on macrophages\\nExpression on T cells\\nP = 1.2 × 10\\n–38\\nP = 5.6 × 10\\n–33\\nP = 7.9 × 10\\n–02\\nP = 3.6 × 10\\n–108\\nP = 1.7 × 10\\n–30\\nP = 1.2 × 10\\n–7\\nP = 1.1 × 10\\n–2\\nP = 9 × 10\\n–2\\nMean\\nCancer cellMacrophagePD-L1Merge\\nControl HCC DoxR HCC\\n0.8\\n0.6\\n0.4\\n0.2\\n0\\nP = 1.6 × 10\\n–5\\nPD-L1 +  macrophages\\nCtrl- HCC\\ncancer cell\\nMacrophages\\nDoxR \\nHCC cancer cell\\nPool cell, single cell\\ncapture, scRNA-seq\\nAnalyze data and identify\\n cell clusters\\n0\\n3\\n4\\n5\\nEnrichment\\nscore\\nNES = 2.6\\nPadj = 0.001\\nC3: M2 macrophage signature (GSE 32164)\\nP = 3.3 × 10\\n–13\\nP = 1.1 × 10\\n–61\\nP = 3.7 × 10\\n–4\\nP = 3.0 × 10\\n–83\\nP = 3.6 × 10\\n–114\\nP = 1.3 × 10\\n–67\\nC0: HCC 1\\nC1: Macs indet\\nControl HCC:\\nmacrophage tumoroid\\nDoxR HCC:\\nmacrophage tumoroid\\nC2: Macs indet\\nC3: Macs M2-like\\nC4: Macs M1-like\\nC5: HCC2\\nCtrl DoxR\\nCtrl DoxR\\n0\\n100\\n50\\nDistribution of clusters\\nDoxRCtrl\\nProp of PD-L1 +\\nmacs\\nCXCR4 LPL STAT1 ISG15\\nHCC cell/\\nmacrophage\\nControl DoxR\\nPercentage\\nCtrl PDO CMDoxR PDO CM\\nDAPI PDL1 DAPI/PDL1\\nCtrl PDO CM DoxR PDO CM\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\nPD-L1 +  MoM\\nProportion of PD-L1 +  Macs\\nP = 1.3 × 10\\n–4Patient-derived \\ntumoroid\\n100 µm\\nCalcein\\n0.1\\n0.2\\n0.3\\n0\\n0.4\\n1\\n2\\n0\\n3\\n4\\n5\\n1\\n2\\nM2-macrophage\\nimmune CN\\n0\\n0\\n10\\n10\\n–10\\n–10\\n–20\\n–20\\n–30\\n–30\\n20\\n20\\n30\\n30\\n100 µm 10 µm\\n100 µm\\nCell\\ntypes\\nCell\\nfunction', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c53aa867-a25d-4fb2-8379-7795e171afbc', embedding=None, metadata={'page_label': '1541', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1541\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nspatial arrangements within HCC as opposed to viewing them just as \\nuniform sheets of cells. T o determine these spatial CNs, we employed a \\npreviously reported approach that defined neighborhoods by cluster-\\ning individual cells and their neighbors to identify broad patterns of \\nspatial organization26 (Fig. 4a). We identified nine distinct neighbor-\\nhoods across all the HCC tissues (Fig. 4b ). Among these, four were \\ntumor-cell-dominant, four were immune-cell-dominant and one was of \\nmixed cell population (termed ‘other’ , of unclear significance) (Fig. 4b). \\nThus, through CN analysis, we delineated higher-order spatial organi-\\nzation in HCC.\\nT o confirm that we were identifying real spatial structures, we \\nused three approaches. First, we overlaid the neighborhoods with \\nhematoxylin and eosin (H&E)-stained sections and fluorescent images, \\nconfirming the accurate recapitulation of known spatial structures \\nsuch as fibrovascular bundles and lymphoid infiltrates (Extended Data \\nFig. 4a). Second, we observed the presence of these CNs in tumors \\nacross all stages, grades and etiologies, suggesting that their pres -\\nence was a shared feature in the microenvironment (Extended Data \\nFig. 4b). Third, we confirmed that the canonical marker of the dominant \\ncell within each CN was indeed overexpressed in its respective CN \\n(Extended Data Fig. 4c). Thus, the microenvironment of HCC seems \\nto be organized into spatial CNs.\\nWe examined whether the distribution of the identified spatial CNs \\nwas different between residual and primary HCCs (Fig. 4c ). Residual \\nHCCs exhibited a higher prevalence of two neighborhoods, the M2-like \\nmacrophage immune CN (P\\u2009=\\u20094\\u2009×\\u200910−5) and the vascular inflammatory \\ntumor CN ( P\\u2009=\\u20090.02). In contrast, the CNs with abundant antitumor \\nimmune cells, the innate immune CN (P\\u2009=\\u20098.5\\u2009×\\u200910−7) and T cell immune \\nCN (P\\u2009=\\u20090.012), were less frequent in residual HCC (Fig. 4c ). These \\nfindings show that in residual versus primary HCC there is a protumor \\nrestructuring of the microenvironment.\\nWe further examined the M2-like macrophage immune CN \\nenriched in residual HCC. We hypothesized this CN may influence the \\ncellular phenotype of tumor cells and T cells residing within it. The \\ntumor cells residing in the M2-macrophage CN (17,046 cells, 7.4%) \\nexhibited a more cancer stem-like (CK19 high/EpCAMhigh/CD44high), \\nmesenchymal (vimentin high) and prosurvival (BCL2 high and Ki67high) \\nphenotype than the tumor cells within another CN abundant in tumor \\ncells but devoid of M2-like macrophages, the pauci-immune tumor \\nCN (109,391 cells, 47.2%) (Fig. 4d). Additionally, T cells within the \\nM2-macrophage CN (4,299 cells, 25.9%) exhibited a more exhausted \\nphenotype (PD-1high, TIM3high, CD44low and CD45ROlow) than the T cells \\nresiding within another CN enriched in T cells (8,345 cells, 50.3%) \\n(Fig. 4d). The observations at the cellular level were also true at the \\ntissue level, where residual HCC with higher M2-like macrophage CN \\nhad greater infiltration of EpCAM + tumor cells and exhausted CD8 + \\nT cells (Extended Data Fig. 4d). Thus, M2-like macrophages seem to \\npromote stemness and CD8+ T cell exhaustion within specific spatial \\nCNs rather than across the entire tumor.\\nNext, we investigated the phenotype of macrophages and T cells \\nresiding within the two stem-like tumor cell CNs in residual HCC. Mac-\\nrophages residing within the two stem-like tumor CNs were more \\nFig. 5 | TGFβ pathway activation promotes persistence of residual tumor \\ncells in HCC. a, Workflow of NanoString spatial transcriptomics analysis. \\nThe expression of targeted transcriptomes of cancer and immune-related \\ngenes were quantified in tumor cell and macrophage AOIs. b, Representative \\nimmunofluorescent ROIs showing tumor cell and macrophage-enriched AOI \\nbased on the expression of panCK, CD45, CD68 and DAPI. c, PCA and volcano \\nplot showing differential expression of genes in the tumor cell (n\\u2009=\\u200952 AOIs) and \\nmacrophage (n\\u2009=\\u200953 AOI). Welch’s two-tailed unpaired t-test was performed \\non log2-transformed normalized count data and the Benjamini–Yekutieli false \\ndiscovery rate correction was applied. d, Molecular pathways activated in tumor \\ncells (n\\u2009=\\u200919 AOI versus n\\u2009=\\u200911 AOI) and macrophage AOIs (n\\u2009=\\u200918 AOI versus n\\u2009=\\u20099 \\nAOI) of residual HCC compared to nontumorous liver samples. e, Schematic \\nshowing derivation of tumor cell and macrophage signatures from the spatial \\ntranscriptome data of primary and residual HCC. f, Kaplan–Meier curve showing \\nthe prognostic significance of tumor cell and macrophage signatures applied \\nto a validation cohort of human HCC (n\\u2009=\\u2009340 patients). A log-rank test was used \\nto compare the groups. g, Upstream regulators of transcriptional regulators \\nof gene expression changes in the tumor cell and macrophage AOIs of residual \\nHCC compared to nontumorous liver samples. h, Plot comparing TGFB1 gene \\nexpression levels in tumor cells (n\\u2009=\\u200919 AOIs) and macrophage AOIs (n\\u2009=\\u200918 AOI) \\nof residual HCC. Data are presented as mean\\u2009±\\u2009s.e.m. A two-tailed unpaired t-test \\nwas used. i, Correlation plot showing coexpression of SMAD2 and CD274 (PD-L1) \\nin residual HCC (n\\u2009=\\u200938 AOI) but not primary HCC (n\\u2009=\\u200948 AOI). A two-sided \\nSpearman test was used to correlate expressions. j, Representative fluorescent \\nmRNA FISH images along with quantification of TGFB1 mRNA expression in \\nprimary (n\\u2009=\\u200951 patients) and residual HCC (n\\u2009=\\u200946 patients). Representative \\nfluorescent mRNA FISH images along with quantification of TGFB1 and CD68 \\nmRNA expression in primary (n\\u2009=\\u200931 patients) and residual HCC (n\\u2009=\\u200930 patients). \\nData are presented as mean\\u2009±\\u2009s.e.m. A two-tailed unpaired t-test was used. rHCC, \\nresidual hepatocellular carcinoma; TGFB1, transcription growth factor β1.\\nFig. 4 | Synchronized spatial remodeling of cellular neighborhoods in \\nresidual HCC. a, Schematic showing CN identification based on an iterative \\nten-cell clustering algorithm. Color codes show hypothetical spatial structures \\nwithin the tumor microenvironment. b, Heatmap demonstrating the cellular \\ncompositions of the nine CNs defined in this study. c, Comparison of tumor  \\nand immune cell neighborhood distributions in primary (n\\u2009=\\u200953 patients) and \\nresidual HCCs (n\\u2009=\\u200955 patients). Box plots comparing the proportion of eight  \\nCNs in primary (n\\u2009=\\u200953 patients) and residual HCCs (n\\u2009=\\u200955 patients). Box shows \\n25th–75th percentiles, whiskers show 5th–95th percentiles and the line indicates \\nthe median. Statistical significance was assessed by unpaired, two-tailed t-test. \\nBH adjustment was used for P values. d, Size-modulated circular heatmap \\nshowing mean expression of markers on y axis in tumor cells and T cells in \\nresidual HCC within the M2-macrophage CN (tumor cells, n\\u2009=\\u200917,046 cells;  \\nT cells, n\\u2009=\\u20094,299 cells) compared to the pauci-immune tumor CN (tumor cells, \\nn\\u2009=\\u2009109,391 cells) or T cell immune CN (T cells, n\\u2009=\\u20098,345 cells) respectively, Padj \\nvalues shown next to the plot. Size-modulated circular heatmap showing mean \\nexpression of markers on y axis on macrophages and T cells within the EpCAM+ \\ntumor cell CN (macrophages, n\\u2009=\\u2009272 cells; T cells, n\\u2009=\\u200913 cells) and CK19+ tumor \\ncell CNs (macrophages, n\\u2009=\\u20091,201 cells; T cells, n\\u2009=\\u2009354 cells) compared to the \\npauci-immune tumor CN (macrophages, n\\u2009=\\u20094,401 cells; T cells, n\\u2009=\\u2009239 cells), \\nrespectively, Padj values are shown next to the plot. Statistical significance was \\nassessed by unpaired, two-tailed t-test. BH adjustment was used for P values. \\ne, Kaplan–Meier plot showing recurrence-free survival in patients with tumors \\nstratified based on the presence (n\\u2009=\\u200912 patients) or absence (n\\u2009=\\u200943 patients) \\nof EpCAM+ tumor cell CN in residual HCC (n\\u2009=\\u200955 patients). A log-rank test was \\nused to statistically compare the groups. f, Schematic of in vitro 3D tumoroid \\nco-culture of control and doxorubicin-resistant (DoxR) Huh7 HCC cell lines \\nwith THP1 macrophages followed by scRNA-seq followed by cell clustering \\nand quantification. g, Characterizing the macrophage C3 (5,176 cells) and C4 \\n(3,372 cells) between DoxR and control samples. Gene set expression analysis \\nshows enrichment of the M2-like macrophage signature in the C3 cluster \\nof macrophages enriched in the DoxR samples. h, Violin plot shows mean \\nexpression of key differentially expressed genes in C3 (5,176 cells) and C4 (3,372 \\ncells) clusters. i, Quantification of PD-L1+ macrophages by IF in 3D heterotypic \\ntumoroids of control (n\\u2009=\\u20094 tumoroids) or DoxR (n\\u2009=\\u20093 tumoroids) Huh7 cancer \\ncells with THP1 macrophages. Data are presented as mean\\u2009±\\u2009s.e.m. Statistical \\nsignificance was assessed by unpaired, two-tailed t-test. j, Schematic of in vitro \\nassay using primary HCC patient-derived 3D tumoroids and monocyte-derived \\nmacrophages from patients with HCC. Calcein staining demonstrates viability \\nof 3D patient-derived tumoroids. PD-L1 expression in monocyte-derived \\nmacrophages treated with conditioned medium from DoxR patient-derived \\ntumoroid (n\\u2009=\\u20094 patients) versus control (n\\u2009=\\u20094 patients). Unpaired t-tests were \\nused to compare the proportion of PD-L1+ macrophages between the two groups. \\nStatistical significance was assessed by unpaired, two-tailed t-test.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c72e7e0d-3ef7-4dfe-9f89-daf4cd567fab', embedding=None, metadata={'page_label': '1542', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1542\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nlikely to be M2-like (CD206 high and CD163high), and CD8+ T cells were \\nmore likely to be exhausted (PD-1 high and TIM3high), than in the other \\ntwo nonstem-like tumor CNs (Fig. 4d  and Extended Data Fig. 4e). \\nAdditionally, the abundance of EpCAM + tumor CNs was associated \\nwith poor recurrence-free survival in residual HCC (P \\u2009=\\u20090.02, hazard \\nratio (HR) 5.0) (Fig. 4e). Taken together, these data indicate that the \\nspatial organization into CNs is distinct in residual HCC than pri -\\nmary HCC. Specifically, the M2-macrophage CN seems to promote \\nspatially constrained stemness and CD8+ T cell exhaustion. This sug-\\ngests that such spatial organization into M2-macrophage CN may \\nserve as a mechanism by which residual tumor cells evade CD8 +  \\nT cell surveillance.\\n0 1 2 3–1–2\\nKRT18\\nSERPINA1\\nTTR\\nHLA-DRB\\nCD74\\n–log 10  P value\\n1 (37%)\\n2 (17%)\\n8 (6%)\\n1. TMA-residual HCC, \\nprim HCC and \\nnontumorous liver\\n2. Select ROIs \\nwith pathologist 3. Attach photocleavable \\noligonucleotide \\nprobes to the TMA\\n4. Stain with PanCK, CD45, \\nCD68, DAPI. Select \\nAOIs\\na. Diﬀerential genes\\nb. Altered pathways\\nc. Molecular functions\\n5. Retreive probes \\nfrom AOI and\\n sequence mRNA\\nHCC tissue\\nMac AOI (CD68+)Tumor AOI (CK + ve)ROI\\nPathways enriched in residual HCC\\nTumor cell AOI\\n–4 –2 0 2–6 4\\nMacrophage activation\\nT H 1 pathway\\nIl-12 signaling in macrophages\\nPTEN signaling\\nPD-L1 pathway\\nIL-10 pathway\\nNormalized\\nenrichment score\\nNormalized\\nenrichment score\\nResidual\\nversus\\nprimary HCC\\nCompare the two \\nspatial compartments\\nDerive 2 classi/f.shortiers Stratify independent\\ncohort of human HCC\\nDetemine\\nprognostic\\nsigni/f.shorticance\\nRes-HCC tum sig\\nRes-HCC mac sig\\nMacrophage AOI\\nTumor cell AOI\\nResidual \\nHCC\\nResidual \\nHCC\\nPrimary\\nHCC\\nPrimary\\nHCC\\n0 2 4 6 8\\n0\\n0.5\\n1.0\\nYears\\nProbability of survival\\nRes HCC mac sig High\\nRes HCC mac sig Low\\nP = 8.2 × 10 –4\\nHR 1.9 (1.3–3.1)\\nn = 257\\nn = 83\\n0 2 4 6 8 10\\n0\\n0.5\\n1.0\\nYears\\nProbability of survival\\nRes HCC tum sig High\\nRes HCC tum sig Low\\nP = 0.628\\nHR 1.1 (0.8–1.6)\\nn = 212\\nn = 128\\nRes-HCC macrophage signature Res-HCC tumor cell signature\\nTGFB1 mRNA /f.shortish TGFB1/CD68 \\nmRNA /f.shortish\\n20 30 40 50 0 2 4 6 8 100 10\\nTGFBR1/2\\nCDKN1A\\nPIK3CA\\nIL1RN\\nDKK1\\nNTN1\\nCXCL17\\nIL37\\nCAB39L\\nTGFB1\\nEDN1\\nCCN5\\nAGT\\nMTOR\\nMAP2K1\\nFGF2\\nIL6\\nPRL\\nIL17A\\n–log P value of overlap –log P value of overlap\\nTumor cell AOI Macrophage AOI\\nUpstream regulators of transcriptional changes in residual HCC TGFB1\\nResidual HCC Primary HCC\\nTGFB1 mRNA /f.shortish\\nTumor cell AOI\\nMacrophage AOI\\nlog 2  fold change\\nTumor cell AOI versus macrophage AOI\\n(n = 334)\\nCCL2\\nDAPI/CD68/TGFB1 mRNA /f.shortish\\nPrimary HCCResidaul HCC 10 µm50 µm\\n200 µm\\nPrimary HCC Residual HCC\\n0\\n0.5\\n1.0\\n1.5\\nProportion of CD68 + TGFB1 +\\ncells/TGFB1 +  cells \\nP = 1.1 × 10\\n–3\\nPrimary HCC Residual HCC\\n0\\n0.2\\n0.4\\n0.6\\n0.8\\n1.0\\nProportion of TGFB1\\npos cells\\nTGFB1 +  cells\\nP = 2.1 × 10\\n–8\\nPrimary HCC Residual HCC\\n0\\n1\\n2\\n3\\n4\\n5\\nTGFB1 +  copies\\nMean number of\\ncopies per cell\\nP = 7.9 × 10\\n–4\\nTum cell AOI\\nMac AOI\\n0\\n50\\n100\\n150\\n200\\nNormalized mRNA expression\\n1.9 × 10\\n–7\\n0 20 40 60\\n30\\n40\\n50\\n60\\n70\\n80\\nCD274 (PDL1)\\nSMAD2\\nResidual HCC\\nPDL1:SMAD2\\nr = 0.46\\nP = 2.8 × 10\\n–3\\n0 20 40 60\\n0\\n50\\n100\\n150\\nCD274 (PDL1)\\nSMAD2\\nPrimary HCC\\nPDL1:SMAD2\\nr = 0.14\\nP = 0.32\\nba\\nc\\nj\\nd\\ne f\\nig h\\nROI\\n–4 –2 0 2 4\\nC el l Ne cr os is\\nI nf la mm at io n of  o rg an\\nA ct iv at io n of  l ym ph oc yt es\\nA ng io ge ne si s\\nB in di ng  o f my el oi d ce ll s\\nI nv as io n of  c el ls\\nMacrophage AOI', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b66b3e2e-2978-4679-82b6-4edb6add7fdf', embedding=None, metadata={'page_label': '1543', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1543\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nT o experimentally test our observations from post-TACE residual \\nhuman HCC, we conducted in vitro experiments to investigate how \\nresistance to doxorubicin, the most commonly used chemotherapy \\nagent in TACE, reprograms cancer cells and macrophages. Employing \\nsingle-cell RNA sequencing (scRNA-seq), we analyzed 3D heterotypic \\ntumoroids (n\\u2009=\\u200931,058 cells) composed of macrophages co-cultured \\nwith either doxorubicin-resistant or control HCC cells (Fig. 4f ). In \\nthe doxorubicin-resistant tumoroids, cancer cells demonstrated a \\nstem-cell-like phenotype with higher expression of stemness, chem-\\noresistance and cytokines, which can drive M2-like macrophage \\npolarization (Extended Data Fig. 5a,b). Among the macrophages, a \\nspecific macrophage cluster (C3), enriched in doxorubicin-resistant \\ntumoroids, showed a distinct M2-like phenotype, in contrast to the \\nless-abundant C4 cluster (CXCR4/LPLPhigh; STAT1/ISG15low) (Fig. 4g,h). \\nAdditionally, immunofluorescence (IF) analysis confirmed that mac-\\nrophages co-cultured with doxorubicin-resistant cancer cells exhibited \\nPD-L1 overexpression compared to those co-cultured with control cells \\n(P\\u2009=\\u20091.6\\u2009×\\u200910−6) (Fig. 4i). In concordance, IF and flow cytometry analysis \\nconfirmed that monocyte-derived macrophages from patients with \\nHCC treated with conditioned medium from doxorubicin-resistant \\npatient-derived tumoroids exhibited significantly higher PD-L1 expres-\\nsion compared to those treated with conditioned medium from con-\\ntrol tumoroids (P \\u2009=\\u20091.6\\u2009×\\u200910−6) (Fig. 4j and Extended Data Fig. 5c) or \\nwith doxorubicin alone (Extended Data Fig. 5d). Overall, our findings \\ndemonstrate that doxorubicin-resistant cancer cells not only adopt \\na more stem-cell-like phenotype but also significantly influence mac-\\nrophage polarization toward an M2-like phenotype with elevated \\nPD-L1 expression.\\nTGFβ pathway activation drives persistence of residual tumor\\nT o investigate the transcriptional changes within the tumor cell and \\nmacrophage neighborhoods identified by CODEX analysis, we used \\nspatially resolved transcriptomics. We separately quantified the \\nexpression of 1,812 genes within tumor cell (pan cytokeratin (panCK)+, \\nCD45−/CD68−) and macrophage (panCK−, CD45+/CD68+) areas of inter-\\nest (AOIs) using NanoString GeoMx DSP (Fig. 5a,b). A subset of the \\nsamples which were analyzed by CODEX and exhibited clearly defined \\ntumor and macrophage areas were included in spatial transcriptom-\\nics; total of 105 AOIs (tumor cell (n \\u2009=\\u200952) and macrophage (n \\u2009=\\u200953); \\npatient details in Supplementary Table 4). The robustness of AOI clas-\\nsification was confirmed by canonical gene expression of each AOI \\n(Fig. 5c). We examined pathways activated within the AOIs of residual \\nHCC compared to nontumorous liver AOIs. Residual HCC tumor cell \\nAOIs displayed upregulated immunosuppressive interleukin (IL)-10 \\npathway (P\\u2009=\\u20094.7\\u2009×\\u200910−47) and PD-L1 pathway (P\\u2009=\\u20092.79\\u2009×\\u200910−37) (Fig. 5d \\nand Supplementary Table 5). On the other hand, macrophage AOIs \\nshowed upregulated angiogenesis (P\\u2009=\\u20093.75\\u2009×\\u200910−10) and invasiveness \\n(P\\u2009=\\u20098.33\\u2009×\\u200910−9) pathways (Supplementary Table 6) (Fig. 5d). Thus, using \\nspatial transcriptomics, we revealed distinct yet complementary pro-\\ntumor pathway activation within the tumor cell and macrophage AOIs  \\nof residual HCC.\\nT o examine whether transcriptional changes within tumor cell \\nor macrophage AOIs were prognostic, we established two gene sig -\\nnatures enriched in the respective AOIs of residual, but not primary \\nHCC (Fig. 5e). We applied the residual HCC tumor cell and mac -\\nrophage signatures to stratify the independent cohort of human \\nTCGA HCC (n\\u2009=\\u2009340). Enrichment of the macrophage signature was \\nassociated with poor overall- and recurrence-free survival on multi-\\nvariable analysis of TCGA HCC, adjusting for age, sex, and tumor stage \\n(P\\u2009=\\u20098.2\\u2009×\\u200910−4, HR 1.9 (1.3–3.1)) (Fig. 5f and Extended Data Fig. 5e). In \\ncontrast, the tumor cell signature did not show significant associations \\nwith overall or recurrence-free survival ( P\\u2009=\\u20090.628, HR 1.1 (0.8–1.6)) \\n(Fig. 5f and Extended Data Fig. 5e). This suggests that macrophages \\ninfluence the trajectory of residual HCC toward recurrence and  \\npoor prognosis.\\nNext, we assessed upstream regulators of the molecular path -\\nways activated within tumor cell and macrophage AOIs. The recep -\\ntor TGFBR1/2 kinase pathway was the top upstream regulator of the \\ntranscriptional changes in the tumor cell AOIs (P\\u2009=\\u20093.1\\u2009×\\u200910−44, z-score \\n2.5), whereas its corresponding ligand, TGFB1 was the top upstream \\nregulator of transcriptional changes in the macrophage AOIs of resid-\\nual HCC (P \\u2009=\\u20092.7\\u2009×\\u200919 −9, z-score 1.2) (Fig. 5g, Extended Data Fig. 5f). \\nMoreover, TGFB1 was expressed at a higher level in the macrophage \\nAOIs than tumor AOIs of residual HCC (Fig. 5h). The TGFB1 pathway \\neffector gene SMAD2 positively correlated with CD274 (PD-L1) within \\nresidual, but not primary HCC (Fig. 5i). Further, we used messenger \\nRNA fluorescence in situ hybridization (FISH) to show that TGFB1  \\nmRNA expression and the proportion of TGFB1-expressing cells that \\nwere CD68+ macrophages were higher in residual than primary HCC \\n(Fig. 5j). Collectively, our CODEX and spatial transcriptomics analy -\\nses highlight the central role of macrophage-mediated immunosup-\\npressive pathways in residual tumor cell persistence, suggesting that \\nPD-L1+ macrophages, may activate the TGFβ1 pathway in residual  \\ntumor cells.\\nFig. 6 | Recurrence of HCC arises from MRD in a transgenic mouse model \\nof HCC. a, Schematic illustrates the process of oncogene activation triggering \\ntumor progression, subsequent inactivation leading to tumor regression and \\nthen reactivation promoting recurrence. Representative H&E and IHC for MYC \\nshow the temporal evolution of tumor progression and tumor regression in \\nMYC/Twist1 HCC in the liver and lungs. b, Representative bioluminescence \\nimaging, MRI liver and CT lungs show regression of tumors upon oncogene \\ninactivation and recurrence upon oncogene reactivation. MRD-bearing mice \\ndo not show any radiological evidence of tumor. c, Overall survival of MYC-HCC \\n(n\\u2009=\\u200917 mice) and MT-HCC (n\\u2009=\\u200921 mice) mice with primary tumor from oncogene \\nactivation (MYC-HCC n\\u2009=\\u200912, MT-HCC n\\u2009=\\u200916 mice) or with recurrent tumor from \\noncogene reactivation in MRD-bearing mice (MYC-HCC n\\u2009=\\u20095, MT-HCC n\\u2009=\\u20095 \\nmice). A log-rank test was used to compare the survival in the Kaplan–Meier \\nanalysis. d, Intravital microscopy (MYC-HCC n\\u2009=\\u20093, MT-HCC n\\u2009=\\u20093 mice) and \\nsubcutaneous transplant allograft model (MYC-HCC n\\u2009=\\u200910, MT-HCC n\\u2009=\\u200910 \\nmice) confirms persistence of a small proportion of viable residual tumor cells \\nin perivascular niches, which serve as a source of recurrence upon oncogene \\nreactivation. e, Cross-species transcriptome analysis of a 45-gene signature \\nenriched in residual murine HCC (n\\u2009=\\u20096 mice) and not primary (n\\u2009=\\u200910 mice) \\nor recurrent HCC (n\\u2009=\\u20093 mice) was able to predict survival in an independent \\nhuman HCC cohort (n\\u2009=\\u2009372 patients). A log-rank test was used to compare \\nsurvival between Kaplan–Meier curves. f, Experimental scheme for scRNA-seq \\nof microdissected areas containing in vivo MRD. Graph-based clustering of \\nscRNA-seq represented as a Uniform Manifold Approximation and Projection \\n(UMAP), with each color representing the correspondingly named cell type \\n(n\\u2009=\\u200915,300 cells). The residual tumor cells (n\\u2009=\\u2009355 cells) are highlighted in a red \\nbox. g, Overlap with cell types and enriched pathways in residual tumor cells \\n(n\\u2009=\\u2009355 cells) compared to hepatocyte clusters 1 (3,607 cells) and 2 (2,144 cells) in \\nscRNA-seq of MRD. Bar graphs indicate −log(P value) from hypergeometric tests \\nto assess significance of pathway enrichment. h, Expression of Tg fb1 and Tgfbr1 \\nin primary (MYC-HCC n\\u2009=\\u20095, MT-HCC n\\u2009=\\u20095 mice), residual HCC (MYC-HCC n\\u2009=\\u20093, \\nMT-HCC n\\u2009=\\u20093 mice) and recurrent HCC (MYC-HCC n\\u2009=\\u20093 mice). Data are presented \\nas mean\\u2009±\\u2009s.e.m. A two-tailed unpaired t-test was used to compare the mean \\nbetween the groups. Horizontal bar plot showing activation of Tgfb1 pathway as \\nupstream regulator of transcriptional changes in MYC-HCC and MT-HCC MRD \\nand MYC pathway inactivation in MRD. i, Expression of Tgfbr1 by IF staining of \\nprimary cancer cell lines derived from MYC-HCC upon MYC inactivation and \\nreactivation. Representative IF images and quantification of phosphorylation \\nof smad2/3 primary cancer cell lines derived from MYC-HCC upon MYC \\ninactivation. Mean and s.e.m. are shown (n\\u2009=\\u200912 biological replicates each). A \\ntwo-tailed unpaired t-test was used to compare the mean between the groups. \\nIHC, immunohistochemistry; BLI, bioluminescence imaging; IVM, intravital \\nmicroscopy; Prim, primary; Res, residual; Rec, recurrent.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='cca26707-b622-402c-bb94-15c5b6df0ec8', embedding=None, metadata={'page_label': '1544', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1544\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nTransgenic mouse model shows recurrence can arise from \\nMRD\\nBased on our analysis of residual human HCC, we hypothesized \\nthat PD-L1+ macrophages promote persistence of residual disease \\nby concurrently activating the TGFβ1 pathway in residual tumor \\ncells, and also driving CD8+ T cell exhaustion. We next sought to \\ndirectly test this hypothesis using transgenic mouse models of  \\nMRD in HCC.\\nT o do this, we employed hepatocyte-specific transgenic mouse \\nmodels of MYC-driven HCC, which express key HCC-specific genes \\nand proteins, and show concordance with gene expression patterns \\nin multiple human HCC datasets (Extended Data Fig. 6a–c). The non-\\nmetastatic MYC-HCC model has been previously reported to retain \\nintrahepatic dormant residual tumor cells upon oncogene inactiva -\\ntion and tumor regression9. T o more closely mimic human HCC and \\nsimulate disseminated residual tumors, we additionally employed \\nSurvival (%)\\nSurvival (%)\\n0\\n0\\n20\\n40\\n40\\n–20\\n–40\\n–40\\n0 week 2 week 4 week 12 week\\nLiver\\nLung\\nTumor progressionTumor regressionTumor recurrence\\nH&EH&EH&E MYCMYCMYC\\nTumor initiation\\n(oncogenes on)\\nTumor progresion\\n(oncogenes on)\\nTumor regression\\n(oncogenes oﬀ)\\n2–3 months 0.5–1 months\\nTumor recurrence\\n(oncogenes re-on)\\nBLI MRI liver CT lung\\n40 80 120\\nPercent survival0\\n40\\n60\\n100\\n20\\n80\\n0\\nMonths\\nTCGA HCC: Overall survival\\n200 µm 10 µm\\n(×40)(×10)\\nMT-HCC cell/dextranIVM\\nPrimary HCCMRDRecurrent HCC\\nPrimary HCCMRDRecurrent HCC\\nBLI Tumor\\n0\\n2,000\\n4,000\\n6,000\\nSubcutaneous transplantation\\nHigh 45-gene sig\\nLow 45-gene sig\\nP = 5.4 × 10\\n–4\\nHR = 1.8\\nPrimary Recurrent\\nResidual\\n(45 genes)\\nPrimary ResidualPrimary Residual Recurrent\\nPhosphto Smad 2/3\\nExpression/HPF10 µm\\n3 µm\\nMYC/DAPI Tg/f.shortbr1/DAPI\\npSmad 2/3/DAPI\\nSingle cell dissociation\\nSequencing and clustering\\n–30\\n–20\\n–10\\n0\\n10\\n20\\n30\\nMTOn\\nMTOﬀ\\nHepatocyte 1\\nFibroblasts\\nHepatocyte 2\\nCholangiocyte\\nC5: Macrophages\\nC9: Macrophages\\nEndothelial cells\\nDendritic cells\\nC6: Macrophages\\nPrimary HCC MRD\\nSelect MRD\\n–30–20–10 0 102030\\nPathways\\nCell type\\nPrim HCC\\nCENPK\\nCD163\\nRASA3\\nBPGM\\nLPAR2\\nTOMM5\\nRPL28\\nCTDSPL2\\nKRT18\\nAPOH\\nMAP4K4\\nNDUFB10\\nSSBP3\\nRFS9\\nGLS\\nSERPING1\\nNCOA3\\nFAM207A\\nMCOLN2\\nZRANB1\\nE2F6\\nUSP35\\nIFT20\\nPOLR2J\\nTOMM6\\nEPHA8\\nFUNDC2\\nABHD14B\\nAIMP1\\nNRG4\\nRHO4\\nPHB2\\nEEF1B2\\nPRR7\\nTFG\\nGATM\\nDUSP6\\nNARS2\\nKLHL38\\nSLC25A16\\nUTY\\nSECISBP2L\\nRPL27A\\nITFG2\\nRNF207\\nLPCAT3\\nRec HCC\\nRes HCC\\n0 2 4 6 8 10\\nEmbryonic stem cell line V262\\nEmbryonic stem cell line Bruce4\\nNIH 3T3 mouse embyonic fibroblast\\nHematopoetic stem cells\\nC2C12 progenitor cells\\n0 5 10\\nVillanueva liver cancer KRT19\\nMueller plurinet\\nWong embryonic stem cell core\\nEppert progenitor\\nZhang cancer progrenitor\\n200 µm\\n200 µm\\n200 µm\\nMT-HCC survival\\nTime (weeks) Time (weeks)\\nMYC-HCC survival\\nP = 4.3 × 10\\n–7\\nHR 7.8\\nPrimary\\nRecurrent\\nPrimary\\nRecurrent\\nP = 5.4 × 10\\n–5\\nHR 6.4\\nlog adjusted P value\\nlog adjusted P value\\nPrimary HCCMRDRecurrent HCC\\nIntensity of\\nnuclear exp\\nMRD\\n300\\n0\\n100\\n200\\nmRNA expression\\n8.9 × 10\\n–4\\n3 × 10\\n–2\\nNS\\nNS\\nTg/f.shortb1Tg/f.shortbr1\\nUpstream regulators MYC-HCC: \\nMRD versus primary\\nUpstream regulators MT-HCC: \\nMRD versus primary\\nMYC\\n–5 0 5 –5 0 5\\nActivation z score Activation z score\\nTg/f.shortbr1\\nPadj = 4.9\\n × 10\\n–22\\nTg/f.shortb1\\nMYC\\nTg/f.shortb1Padj = 7.8\\n × 10\\n–49\\nPadj = 1.8\\n × 10\\n–50\\nPadj = 4.6\\n × 10\\n–28\\na b c\\nd\\ne f\\ng\\nh\\ni\\nPrim tum cellsRes tum cellsRec tum cells\\n0\\n4\\n8\\n3 × 10\\n–8\\n2 × 10\\n–8\\nPrim tum cellsRes tum cells\\n0\\n4\\n8 P = 2.3 × 10\\n–5\\nPrim HCC\\nMRD\\nRes HCCPrim HCC\\nMRD\\nRes HCC\\ntSNE1\\ntSNE1\\n0\\n0\\n50\\n100\\n0\\n50\\n100\\n10 20 30 40 50 0 10 20 30 40\\nResidual tumor cells\\n', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2c8e6e7b-ab98-4d4b-a5c3-8ca51c534276', embedding=None, metadata={'page_label': '1545', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1545\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nlog 2  fold change\\n–2\\n0\\n2\\n4\\n6\\n8\\n10\\n0 2 4\\nIl10RA\\nIdo2\\nCxcl1\\nIl1r1\\nCd274\\nNlrp6\\nCsf1r\\nVcam1\\n–log (Padj)\\nC5: Macs\\nC9: Macs\\nC6: Macs\\nC5: Macs\\nC9: Macs\\nC6: Macs\\nC6 M2-like macrophages versus\\nC5, C9 macrophages\\nM8524 IL-4/IL-13 pathway\\nC6 M2-like macrophages\\nversus  C5,C9 macrophages\\nGO_BP monocyte derived mac\\nC6 M2-like macrophages versus \\nC5,C9 macrophages\\nTg/f.shortb1\\n0\\n2\\n4\\n6\\n8\\n10\\nNumber of cells/HPF\\nPdl1 Hi /Tgfβ1 Hi\\nmacrophages\\nPdl1 Hi /F4/80 +\\nmacrophages\\nP = 3.3 × 10\\n–3\\nResidual tumor\\ncells\\nCYBA\\nWIPF1\\nRGS2\\nBCL2A1B\\nH2-AA\\nIL10RA\\nPRKCD\\nFAM105A\\nHBB-BS\\nMPEG1\\nIL7R\\nETS1\\nFGL2\\nDOK2\\nCCR2\\nPARP8\\nKCTD12\\nCCR5\\nTSPAN32\\nLMO2\\nSLC25A36\\nVCAM1\\nENTPD1\\nIL10RB\\nITGB2\\nAHNAK\\nIL18RAP\\nIL18R1\\nCD22\\nRASSF2\\nSOAT2\\nULK1\\nAPP\\nGATA2\\nPF4\\nLIF\\nSLPI\\nTLR2\\nGAB3\\nMMP9\\nIL15\\nGATA3\\nVPS54\\nTGFB1\\nUCP2\\nRB1\\nCSF2\\nIFNG\\nZBTB46\\nCDC42\\nC1QC\\nHCLS1\\nPARP1\\nIL1RL1\\nCSF1R\\nINHA\\nCSF1\\nCD4\\nTSPAN2\\nNRROS\\nIL34\\nCASP8\\nFADD\\nIL33\\nSOCS1\\nPRKCA\\nBMP4\\nHepatocyte 1\\nFibroblasts\\nHepatocyte 2\\nCholangiocyte\\nC5: Macrophage\\nC9: Macrophages\\nEndothelial cells\\nDendritic cells\\nC6: Macrophages\\nC6: M2-like macrophages\\nH&E DAPI CD133 F4/80 Pdl1 DAPI/CD133/F4/80/Pdl1\\nPD1-BV785\\nTIM3-PE-Cy7\\nControl MRD\\nControl\\nMRD\\n0\\n10\\n20\\n30\\nCD4 +  T cells\\nFrequency\\n(% all/single cells/viab/\\nCD45 + /TCRb\\nNS\\nControl\\nMRD\\n0\\n10\\n20\\n30\\nCD8 +  T cells\\nFrequency\\n(% all/single cells/viab/\\nCD45 + /TCRb\\nNS\\nControl\\nMRD\\n0\\n20\\n40\\n60\\n80\\nFrequency\\n(% all/single cells/viab/\\nCD45 + /TCRb/CD8 + )\\nExhausted CD8 +  T cells\\nP = 0.002\\nExhausted CD8 +  T cells\\n20 µm 20 µm\\nDAPIH&E F4/80 Tg/f.shortb1Pdl1 DAPI/F4/80/Pdl1/Tg/f.shortb1\\n20 µm20 µm\\n5.77 0.00 48.4 61.1\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0 10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0 10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0 10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nPhagocytosis assay co-culture \\nFar MRD Near MRD\\nFar MRD Near MRD\\nFar MRDNear MRDFar MRDNear MRD\\nZymogen\\nNegative control:\\n Dox alone\\nPrimary tumor cells:\\nConditioned medium\\nResidual tumor cells:\\nConditioned medium\\n0\\n200\\n400\\n600\\n800 2.7 × 10\\n–37.3 × 10\\n–7\\n0\\n20\\n40\\n60\\n80 2.7 × 10\\n–4\\n0\\n50\\n100\\n150\\n200\\n0\\n2,000\\n4,000\\n6,000\\n8,000 4.0 × 10\\n–4\\nPrim tumor cells Res tumor cells Prim tumor cells Res tumor cells Prim tumor cells Res tumor cells Prim tumor cells Res tumor cells\\nCXCL5 IL-6 IL-22 IL-18\\nConcentration (pg ml –1 )\\nConditioned medium cytokine ELISA\\nNeg ctrl\\nPrim tum cells Res tum cells\\n0\\n20\\n40\\n60\\n80\\n100\\nPhagocytosis\\nPercentage fluorescent cells\\nP = 0.036\\nCo-culture assay\\nPrimary \\ncells\\nResidual\\ncells\\nConditioned \\nmedium\\nMacrophages\\nMYC On MYC Oﬀ\\n0\\n2\\n4\\n6\\n8\\n10\\nNumber of cells/HPF\\nP = 8.4 × 10\\n–3\\ne\\na b c\\nd\\ng\\nh\\nf\\n400 µm\\n–40\\n–40\\n–30\\n–20\\n–10\\n0\\n10\\n20\\n30\\n40\\n–20 0\\ntSNE1\\ntSNE2\\n20 40\\n–40 –20 0\\ntSNE1\\n20 40\\n–40\\n–30\\n–20\\n–10\\n0\\n10\\n20\\n30\\n40\\ntSNE2\\nConditioned\\nmedium', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='ecdb2e87-c353-44bb-9217-306d95218517', embedding=None, metadata={'page_label': '1546', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1546\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nthe MYC/Twist1 model of metastatic HCC9,14 (MT-HCC). We now show \\nthat MT-HCC retains small clusters of hyperchromatic, nonprolif -\\nerative (phospho histone 3 −), nonapoptotic (cleaved casp3 −) viable \\nresidual tumor cells, both within the liver and at extrahepatic sites, \\nupon oncogene inactivation (Fig. 6a,b and Extended Data Fig. 6d). \\nNotably, these areas or residual HCC are radiologically undetectable \\n(Fig. 6b), thus fulfilling criteria to be referred to as minimal residual \\ndisease (MRD)27. Thus, our models effectively mimic the persistence of \\nresidual tumor cells post-tumor regression, as observed in post-TACE \\nhuman residual HCC.\\nWe evaluated whether the MYC-HCC and MT-HCC model are com-\\nparable to post-chemoembolization residual human HCC in four ways. \\nFirst, we used time-series analysis to show that oncogene reactiva -\\ntion led to the recurrence of tumors from residual tumor cells within \\nthe regressed tumor bed, recapitulating the clinical phenomenon of \\nrecurrence (Fig. 6a,b). These recurrent tumors had shorter latency \\nand poorer survival rates, pointing to their origin from MRD (Fig. 6c). \\nSecond, we unambiguously traced the origin of recurrent tumors to \\nMRD using a combination of subcutaneous transplantation studies and \\nintravital microscopy to track live, residual RFP-labeled MT-HCC cells \\n(Fig. 6d). Third, we found that the overexpression of a gene signature \\nfrom mouse MRD was associated with poor overall (P\\u2009=\\u20095.4\\u2009×\\u200910−4, HR \\n1.8) and recurrence-free survival (P\\u2009=\\u20090.03, HR 1.4) in the TCGA cohort of \\nhuman HCC (Fig. 6e). Fourth, we confirmed the translational relevance \\nof our model to the context of chronic inflammation by demonstrat-\\ning that tumor regression and MRD-driven recurrence also occur in \\nthe background of NASH induced by a high-fat diet. (Extended Data \\nFig. 6e–g). These results validate using MYC-HCC and MT-HCC mouse \\nmodels to study MRD-driven recurrence in human HCC and to test \\nadjuvant therapies.\\nResidual tumor cells in mouse MRD demonstrate cancer \\nstemness\\nIn our analysis of human HCC, we found that residual tumor cells \\nexhibit cancer stem cell-like features. We characterized the in vivo \\nresidual tumor cells in our mouse models by performing scRNA-seq \\nof a microdissected tumor bed containing MRD (n \\u2009=\\u200915,300 cells) \\n(Fig. 6f). A subset of cells (n\\u2009=\\u2009355, 0.02%) clustered with hepatocytes \\nbut differentially expressed several genes compared to the two main \\nhepatocyte clusters (586 differentially expressed genes; 460 up and \\n126 down) (Fig. 6f and Supplementary Table 7). Gene set enrichment \\nanalysis revealed that these cells were transcriptionally analogous to \\nembryonic stem cell lines (Fig. 6g). Additionally, gene signatures associ-\\nated with progenitor or cancer stem cells were enriched in the residual \\ntumor cells (Fig. 6g). Particularly, a gene signature representing liver \\ncancer stem cell CK19+ tumors associated with human HCC recurrence28 \\nwas strongly upregulated in this cluster (Padj\\u2009=\\u20091.1\\u2009×\\u200910−11) (Fig. 6g). \\nFurther, the residual tumor cells in the liver and metastatic sites \\nexpressed liver cancer stem cell markers CD133 and CK19 (Extended \\nData Fig. 6h,i). Thus, as seen in human post-chemoembolization resid-\\nual tumor cells, the tumor cells in MRD in our mouse models exhibit  \\ncancer stemness.\\nFig. 7 | Mechanisms of immune evasion by cancer stem cells in the mouse MRD \\nniche. a, Graph-based clustering of scRNA-seq of MRD represented as UMAP \\nwith each color representing the correspondingly named cell type (n\\u2009=\\u200915,300 \\ncells). The three subsets of macrophages are highlighted in a red box. The M2-like \\nmacrophage cluster (C6) (1,061 cells) is shown separately in the UMAP below. \\nb, Volcano plot showing the differentially expressed genes between the M2-like \\nC6 cluster (1,061 cells) of macrophages and the C5 (1,097 cells) and C9 cluster \\n(681 cells) of M1-like macrophages (Padj\\u2009<\\u20090.05) are shown in green if under-\\nexpressed or red if overexpressed in the M2-like macrophage cluster. DESeq2 was \\nused in differential expression analysis to calculate fold changes and P values. \\nc, Heatmap showing the enrichment of gene signature of M2-like macrophage \\npolarization and monocyte-derived macrophage differentiation in the M2-like C6 \\ncluster of macrophages (1,061 cells) than the C5 (1,097 cells) and C9 cluster (681 \\ncells) of M1-like macrophages. d, Representative H&E and IF images of CD133+ \\nstem-like cancer cells and Pdl1+ macrophages near MRD (<200\\u2009μm) compared \\nto liver distant from MRD (n\\u2009=\\u20098 areas, each group). Data are presented as \\nmean\\u2009±\\u2009s.e.m. Bar plot shows comparison for mean PD-L1+ macrophages between \\ngroups using two-sided unpaired t-tests. e, Representative H&E and IF images \\nof Tgfβ1 expression on Pdl1+ macrophages near MRD (<200\\u2009μm) compared \\nto liver distant from MRD (n\\u2009=\\u20099 areas, each group). Data are presented as \\nmean\\u2009±\\u2009s.e.m. Bar plot shows comparison for mean Tgfβ1+/PD-L1+ macrophages \\nbetween groups using two-sided unpaired t-tests. f, Schematic showing \\ncollection of conditioned medium from primary or residual HCC cells from \\nMYC-HCC to treat macrophages in vitro. Bar plot shows comparison for mean \\nconcentration of cytokines secreted by residual tumor cells (n\\u2009=\\u20096 biologically \\nindependent samples) and primary tumor cells (n\\u2009=\\u20096 biologically independent \\nsamples) using two-sided unpaired t-tests. Data are presented as mean\\u2009±\\u2009s.e.m. \\ng, Macrophages are treated with conditioned medium (CM) from negative \\ncontrol (n\\u2009=\\u20092 biologically independent samples), primary (n\\u2009=\\u20093 biologically \\nindependent samples) or residual tumor cells (n\\u2009=\\u20093 biologically independent \\nsamples) in vitro. Bar plot shows comparison for phagocytosis between CM from \\nprimary or residual tumor cell groups using two-sided unpaired t-tests. Data are \\npresented as mean\\u2009±\\u2009s.e.m. h, Bar plots showing comparison of mean CD4+ T, \\nCD8+ T and exhausted CD8+ T cells in livers with MRD (n\\u2009=\\u20094 mice) versus control \\nnon-MRD livers (n\\u2009=\\u20094 mice) using two-sided unpaired t-tests. Representative flow \\ncytometry plots of exhausted CD8+ T cells based on TIM3 and PD-1 expression. \\nData are presented as mean\\u2009±\\u2009s.e.m. Ctrl, control.\\nFig. 8 | Blockade of Tgfβr1 and Pdl1 prevents recurrence from MRD in mouse \\nHCC. a, Experimental scheme for treatment of oncogene-deprived MRD-bearing \\nmice with control antibody or monotherapy with Tgfbr inhibitor (TgfbrI) or \\nanti-Pdl1 or their combination. Treatment is followed by oncogene reactivation \\nto induce tumor recurrence. b, Representative gross images of recurrent tumor \\nburden upon oncogene reactivation in the liver of mice treated with control \\nantibody or monotherapy with TgfbrI or anti-Pdl1 inhibitor or their combination. \\nc, Quantification of recurrent tumor burden upon oncogene activation in the \\nliver of MYC-HCC and MT-HCC mice treated with control antibody (n\\u2009=\\u200911 mice)  \\nor monotherapy with Tgfbr inhibitor (n\\u2009=\\u20099 mice) or anti-Pdl1 inhibitor (n\\u2009=\\u20099 \\nmice) or their combination (n\\u2009=\\u200916 mice). Plots compare the mean between \\nthe groups with two-sided unpaired t-tests. Box shows 25th–75th percentiles, \\nwhiskers show 5th–95th percentiles and the line represents the median.  \\nd, Representative images and quantification from H&E staining and IHC staining \\nfor CD4 and CD8 in recurrent tumors in mice treated with control antibody (n\\u2009=\\u20095 \\nmice) or monotherapy with TgfbrI (n\\u2009=\\u20095 mice) or anti-Pdl1 inhibitor (n\\u2009=\\u20095 mice) \\nor their combination (n\\u2009=\\u20095 mice). Plots compare the mean between the groups \\nwith two-sided unpaired t-tests. Box shows 25th–75th percentiles, whiskers show \\n5th–95th percentiles and the line represents the median. e, Representative flow \\ncytometry images and quantification of activated CD4+ and CD8+ T cells, which \\nare CD69+/CD44high in the recurrent tumor in the liver and lungs of MYC-HCC and \\nMYC/Twist1 HCC mice treated with control antibody (n\\u2009=\\u20093 mice) or combination \\ntherapy with TgfbrI and anti-Pdl1 inhibitor (n\\u2009=\\u20094 mice). Bar plots compare the \\nmean between the groups with two-sided unpaired t-tests. Data are presented \\nas mean\\u2009±\\u2009s.e.m. f, Experimental scheme for treatment of oncogene-deprived \\nresidual HCC-bearing mice with control antibody or Tgfbr1 or Pdl1 inhibitors or \\ncombination therapy with TgfbrI and anti-Pdl1 inhibitor. Residual tumor niches \\nare then evaluated for immune response. g, Quantification of IHC staining for \\nCD8 and PD-L1 in residual HCC niche in mice treated with control antibody (n\\u2009=\\u20095 \\nmice) or Tgfbr1 (n\\u2009=\\u20095 mice) or Pdl1 inhibitors (n\\u2009=\\u20095 mice) or combination therapy \\nwith Tgfbr1 and Pdl1 inhibitors (n\\u2009=\\u20095 mice). Bar plots compare the mean between \\nthe groups with two-sided unpaired t-tests. Data are presented as mean\\u2009±\\u2009s.e.m. \\nh, Representative H&E staining and IHC staining for CD8 and PD-L1 in MRD \\n(red arrows in gross image and black boxes in H&E) in mice treated with control \\nantibody (n\\u2009=\\u20095 mice) or Tgfbr1 (n\\u2009=\\u20095 mice) or Pdl1 inhibitors (n\\u2009=\\u20095 mice) or \\ncombination therapy with Tgfbr1 and Pdl1 inhibitors (n\\u2009=\\u20095 mice).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a2009977-1d54-464b-bda1-707d04bd9760', embedding=None, metadata={'page_label': '1547', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1547\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nControlTgfbrIAnti-Pdl1\\nAb\\nTgfbrI+  \\nanti-Pdl1 Ab\\nControlTgfbrI\\nAnti-Pdl1\\nAb\\nTgfbrI+\\nanti-Pdl1 Ab\\nControl\\nTgfbrI\\nAnti-Pdl1 Ab\\nTgfbrI+\\nAnti-Pdl1 Ab\\nControl TgfbrI\\nAnti-Pdl1 Ab\\nTgfbrI+\\nanti-Pdl1 Ab\\n0\\n0.1\\n0.1\\n0.2\\n0.2\\n0.3\\nLiver/body weight ratio\\nP = 2.3 × 10\\n–3\\nP = 3.0 × 10\\n–4\\nP = 1.25 × 10\\n–8\\n0\\n5\\n10\\n15\\nNumber of surface\\nnodules\\nP = 3.0 × 10\\n–4\\nP = 3.0 × 10\\n–4\\nP = 1.1 × 10\\n–8\\n0\\n10\\n20\\n30\\nTumor volume (mm 3 )\\nP = 2.7 × 10\\n–6\\nP = 8.6 × 10\\n–5\\nP = 2.8 × 10\\n–10\\nH&E CD8 CD8 CD4 CD4\\n5.0\\n10.0\\n0\\n15.0\\nP = 0.02\\nP = 0.009\\nP = 0.0004\\n5\\n10\\n15\\n0\\n20\\nP = 0.02\\nP = 0.005\\nP = 0.004\\nCD8 +  T cells\\nCD4 +  T cells\\nNo. of +cells/HPFNo. of +cells/HPF\\n50 µm 10 µm 10 µm\\nRecurrent tumor burden\\nRecurrent HCC\\nOncogene on Oncogene off\\nRandomize mice \\nwith MRI-con/f.shortirmed \\nprimary HCC\\nRandomize mice \\nafter MRI-con/f.shortirmed \\ntumor regression\\nKill and assess\\ntumor burden and \\nimmune response\\nOncogene\\nre-on\\nCtrl IgG\\nAnti-Pdl1 Ab\\nTgfβrI+\\nanti-Pdl1 Ab\\nTgfβrI\\nPrimary HCC MRD\\n16.1\\n56.0\\n16.3 8.35\\n1.51\\n72.7 23.1\\n27.2\\n47.2\\n7.72 13.2\\n18.1\\n42.7 37.1\\n67.8\\n15.7\\n3.31 9.09\\n59.6\\n12.6 25.9\\n15.277.1\\n4.9837.1\\n36.7\\n14.8 7.12\\n10\\n5\\n10\\n5\\n10\\n4\\n10\\n4\\n10\\n3\\n10\\n3\\n–10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n0\\nCD4 +  T cells CD8 +  T cells CD4 +  T cells CD8 +  T cells\\nRecurrent HCC liver Recurrent HCC lung\\n20\\n40\\n60\\n0\\n80\\nActivated CD4 +  T cells\\nPercentage of CD4 +\\nT cells\\nP = 8.1 × 10\\n–3\\nControl Ab\\nTgfbrI\\n+  anti-Pdl1 Ab\\n20\\n40\\n60\\n0\\n80\\nActivated CD8 +  T cells\\nPercentage of CD8 +\\nT cells\\nP = 0.035\\n20\\n40\\n0\\n60\\nActivated CD4 +  T cells\\nPercentage of CD4 +\\nT cells\\nP = 0.23\\nControl Ab\\nTgfbrI\\n+  anti-Pdl1 Ab\\n5\\n10\\n15\\n0\\n20\\nActivated CD8 +  T cells\\nPercentage of CD8 +\\nT cells\\nP = 0.013\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD44 (PerCP-Cy5.5)\\nCD69 (PE-Cy7)\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD69 (PE-Cy7)\\nControl Ab\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD69 (PE-Cy7)\\nControl Ab\\nTgbrI+ anti-Pdl1 Ab\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\n10\\n5\\n10\\n4\\n10\\n3\\n–10\\n3\\n0\\nCD69 (PE-Cy7)\\nTgbrI+ anti-Pdl1 Ab\\nOncogene on Oncogene off\\nRandomize mice \\nwith MRI-con/f.shortirmed \\nprimary HCC\\nRandomize mice \\nafter MRI-con/f.shortirmed \\ntumor regression\\nEuthanize and assess\\nimmune response in\\nMRD niche\\nCtrl IgG\\nAnti Pdl1 Ab\\nTgfβrI+\\nanti Pdl1 Ab\\nTgfβrIPrimary HCC MRD MRD\\n200 µm 20 µm\\nControl\\nH&E H&E H&E CD8 PDL1H&E CD8 PDL1\\nTg/f.shortbr1I+\\nanti-Pdl1 Ab\\nAnti-Pdl1\\nAb\\nTg/f.shortbr1I\\n200 µm 20 µm\\n10\\n20\\n0\\n30\\nNumber of cells/HPF\\nNumber of cells/HPF\\nP = 2.3 × 10\\n–6\\nCD8T cells\\nP = 7.1 × 10\\n–6\\nP = 6.2 × 10\\n–4\\n0\\n5\\n10\\nP = 9.6 × 10\\n–3\\nPDL1+ cells\\nControl Ab\\nTgfbrI + anti Pdl1 Ab\\nTgfbrI\\nAnti Pdl1 Ab\\nP = 7.0 × 10\\n–3\\nP = 9.2 × 10\\n–2\\nf g\\nh\\na\\nb\\nc\\nd\\ne', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='be55dda3-f6ca-421f-86e5-fa4388558901', embedding=None, metadata={'page_label': '1548', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1548\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nTgfβ pathway is activated in residual tumor cells in mouse \\nmodels of MRD\\nNext, we found that the TGFβ pathway, identified as a key regulator \\nof residual human HCC, was also activated in mouse MRD using three \\napproaches. First, we show that the TGFβ pathway was the top activated \\nupstream regulator of the transcriptional changes in MRD of both \\nMYC-HCC (Padj\\u2009=\\u20091.8\\u2009×\\u200910−50; enrichment score (ES) 6.6) and MT-HCC \\n(Padj\\u2009=\\u20097.9\\u2009×\\u200910−49; ES 6.2) upon oncogene inactivation (Fig. 6h). Second, \\nwe corroborated that the TGFβ pathway activation was specifically \\nconfined to a subset of stem-like tumor cells in vitro using scRNA-seq \\n(Extended Data Fig. 7a–e). Third, we found that residual tumor cells \\nin vitro have increased phosphorylation of smad2/3 and reversible over-\\nexpression of the receptor Tgfbr1 but did not expressTg fb1 (Fig. 6h,i). \\nThus, in our mouse model of MRD, stem-like residual tumor cells show \\nTGFβ pathway activation.\\nMechanisms of immune evasion by residual tumor cells\\nWe investigated if in our mouse models, as suggested by human residual \\nHCC, macrophages within the MRD niche contribute to TGFβ activation \\nin residual tumor cells (Fig. 5g). Single-cell analysis of the in vivo mouse \\nMRD demonstrated three distinct macrophage subsets (Fig. 7a). Two \\nmacrophage subsets had a proinflammatory M1-like phenotype with \\nhigher expression of genes such as Il1r1, Cxcl1, Ido2 and Nlrp6 (Fig. 7b). \\nOne of the macrophage clusters demonstrated an immunosuppressive \\nphenotype (n\\u2009=\\u20091,061 cells) with overexpression Cd274 (Pdl1) and enrich-\\nment of M2-like macrophage29 and monocyte-derived macrophage \\ngene signatures (979 differentially expressed genes; 669 up and 310 \\ndown) (Fig. 7b,c and Supplementary Table 8). Moreover, this M2-like \\nmacrophage cluster expressed higher levels of Tg fb1 (Padj\\u2009=\\u20093.95\\u2009×\\u200910−7). \\nThus, Pdl1high macrophages seem to be the source for the Tgfβ pathway \\nactivation in residual tumor cells in our mouse MRD.\\nWe found that Pdl1high macrophages were enriched in MRD and \\nnot in non-MRD-bearing areas of the liver and also found that they \\nclosely interacted with stem-like residual tumor cells, both in the liver \\n(Fig. 7d) and the lungs (Extended Data Fig. 8a,b). Notably, these Pdl1high \\nmacrophages were confirmed to coexpress Tgfβ1 (Fig. 7e). Further \\nusing a multiplex ELISA assay, we show that residual tumor cells, and \\nnot primary tumor cells, secrete four cytokines, Cxcl5, Il6, Il22 and Il18, \\nwhich are known to promote M2-like macrophage polarization and \\nincrease PD-L1 expression30–33 (Fig. 7f). Also, macrophages exposed to \\nthe medium from residual tumor cells showed more impaired phago-\\ncytosis than those exposed to the medium from primary tumor cells \\n(Fig. 7g). Thus, residual tumor cells in our mouse models seem to use a \\ncytokine-mediated mechanism to recruit Tgfβ1-secreting nonphago-\\ncytic Pdl1+ macrophages to MRD.\\nWe examined immune cell functionality within MRD using flow \\ncytometry. While infiltration of other innate immune cells (Extended \\nData Fig. 8c) CD4+ T and CD8+ T cells (Fig. 7h), including naive, central \\nmemory or effector phenotypes (Extended Data Fig. 8d), were similar \\nbetween MRD-bearing and control livers (Supplementary Fig. 3), we \\nobserved a significant enrichment of exhausted (Pd1+/Tim3+) CD8+ T cells \\nin MRD-bearing liver (Fig. 7h). Thus, as we see in residual human HCC, \\nin our mouse model of MRD we observed exhaustion of CD8+ T cells, \\nsuggesting that this is a mechanism of impaired immune surveillance.\\nInhibition of Tgfβr1 and Pdl1 blocks HCC recurrence from MRD\\nOur analysis of human residual HCC and mouse MRD suggests that \\ntargeting the Tgfβ1 pathway and Pdl1high macrophages could eliminate \\nresidual tumor cells and prevent tumor recurrence. T o experimentally \\naddress this, we examined in our residual MYC-HCC or MT-HCC-bearing \\ntransgenic mice, whether a Tgfbr inhibitor (TgfbrI) or anti-Pdl1 anti-\\nbody, either alone or in combination, prevented recurrence (Fig. 8a). We \\nconfirmed the absence of macroscopic tumors by magnetic resonance \\nimaging (MRI) imaging before randomizing transgenic mice with MRD \\nto the four treatment groups (Fig. 8a). Upon oncogene ractivation, \\nmice receiving combined anti TgfbrI and Pdl1 therapy showed an eight-\\nfold reduction in recurrent tumor burden and a fivefold reduction in \\nrecurrence rate, compared to controls (both P\\u2009<\\u20090.05; Fig. 8b,c). These \\ndifferences were also significant when compared to monotherapy \\nwith TgfbrI or anti-Pdl1 (all P\\u2009<\\u20090.05; Fig. 8b,c). Combination anti-TgfbrI \\nand Pdl1 therapy-treated livers demonstrated areas of higher CD4+ \\nand CD8+ T cell recruitment than either control or monotherapies (all \\nP\\u2009<\\u20090.05; Fig. 8d), Moreover, CD4+ T and CD8+ T cells of mice treated \\nwith combination therapy were more activated (CD69+/CD44+) than \\nthe control group (Fig. 8e). The combined inhibition of Tgfbr1 and \\nPdl1 in a subcutaneous allograft model improved recurrence-free \\nsurvival, confirming its effectiveness in preventing recurrence from \\nMRD (Extended Data Fig. 9a–d). Additionally, we found that combined \\ninhibition of Tgfbr1 and Pdl1 lacked efficacy in preventing the progres-\\nsion of primary MT-HCC, suggesting mechanistic specificity to MRD \\n(Extended Data Fig. 9e,f). Thus, combined blockade of Tgfβ1 pathway \\nand Pdl1 promoted recruitment and activation of T cell response and \\nprevented tumor recurrence from MRD.\\nT o determine whether blocking Tgfβ1 and Pdl1 eliminated MRD, \\nwe administered either control or TgfbrI or anti-Pdl1 antibodies or \\ntheir combination to mice bearing residual HCC and euthanized mice \\nimmediately after the completion of treatment, without reactivating \\nthe oncogenes (Fig. 8f). Transgenic mice treated with the combination \\ntherapy exhibited increased recruitment of CD8+ T cells and decreased \\nPdl1+ macrophages within MRD than treatment with control or the \\nrespective monotherapies (Fig. 8g,h). We also saw an increased number \\nof apoptotic cells in MRD with combination therapy than control or \\nmonotherapies (Extended Data Fig. 9g). Flow cytometry analysis con-\\nfirmed that CD8+ T cells in mice treated with combination anti TgfbrI/\\nPdl1 therapy were more activated (CD69 +/CD44+) than with control \\n(Extended Data Fig. 9h and Supplementary Fig. 4). Hence, combined \\nblockade of Tgfβ and Pdl1 decreased immunosuppressive macrophage \\ninfiltration, elicited a robust CD8+ T cell response, eliminated MRD and \\nprevented HCC recurrence.\\nWe note that our transgenic mouse models of MRD recapitulate \\nthe natural history of residual tumor cell persistence and HCC recur-\\nrence, but they do not involve persistence of residual tumors after \\nexposure to therapeutic agents such as doxorubicin, as is used in TACE \\nfor human HCC34. Hence, we complemented our MRD mouse model by \\ndeveloping a immunocompetent mouse model of syngeneic ortho -\\ntopic allografts in wild-type C57/BL6 mice with Hep 53.4 HCC cells \\nthat underwent in vitro selection to be either doxorubicin-resistant \\nor control (Extended Data Fig. 10a). Treatment with either control or \\ncombined TgfbrI and anti-Pdl1 antibodies in the doxorubicin-resistant \\northotopic allografts resulted in a significant reduction in liver tumor \\nburden (P\\u2009=\\u20090.01) (Extended Data Fig. 10b,c). Immune analysis revealed \\nincreased CD8 + T cell infiltration, elimination of PD-L1 +/CD206 + \\nM2-like macrophages and increased infiltration of CD86 +/MHCII+ \\nM1-like macrophages (Extended Data Fig. 10d,e). In contrast, control \\nallografts (Extended Data Fig. 10f) showed no significant response \\nto this treatment (Extended Data Fig. 10b,c,g and Supplementary \\nFig. 5). These observations confirm the findings from our transgenic \\nMRD model, highlighting the complementary nature of both models \\nand underscoring the therapeutic potential of dual Tgfβ1 and Pdl1 \\nblockade in eliminating both oncogene-deprived stem-like MRD and \\ndoxorubicin-resistant murine HCC.\\nDiscussion\\nWe combined spatial analysis of both human clinical samples and trans-\\ngenic mouse models of HCC to determine the mechanism of persistence \\nof residual disease in HCC and suggest a possible therapeutic approach \\nto improve clinical outcome (Extended Data Fig. 10h). We identified \\nthat interactions between stem-like tumor cells and immunosuppres-\\nsive PD-L1+ macrophages within spatially constrained neighborhoods, \\nin both human and mouse residual HCC, were linked to CD8+ T cell ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='03ad263c-067c-4271-9108-a14bfb75bd23', embedding=None, metadata={'page_label': '1549', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1549\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nexhaustion. Further, macrophage-mediated TGFβ pathway activation \\nwithin residual tumor cells was found to enable persistence of both \\nhuman and mouse MRD. Finally, we provide preclinical evidence that \\ncombined inhibition of PD-L1+ macrophage-mediated TGFβ activation \\neliminates MRD in mouse HCC and prevents recurrence. Our results \\nprovide mechanistic insight that suggests a therapeutic strategy for \\neliminating residual disease to prevent cancer recurrence.\\nWe harness the power of multiple complementary spatial biology \\ntechnologies to draw mechanistic insights into residual HCC. Nota -\\nbly, residual disease that persists after therapy is a poorly character-\\nized entity, especially in human solid tumors. Previous studies have \\nemployed scRNA-seq of primary HCC35–37 or CODEX of treatment-naive \\nHCC38,39; however, the spatial organization of post-TACE residual HCC \\nis not known. Our spatial mapping provides two key insights. First, \\nwe show that spatial interactions of cancer stem cells and protumor \\nmacrophages, not just their abundance as previously suggested40–42, \\nare critical drivers of recurrence from residual HCC. Second, we reveal \\na spatial restructuring of residual HCC into neighborhoods enriched \\nwith M2-like macrophages. Within these spatially constrained areas, \\nM2-like macrophages promote CD8+ T cell exhaustion, thus facilitat-\\ning immune evasion by stem-like tumor cells. This interplay adds a \\nspatial dimension to the known role in PD-L1+ macrophages in cancer \\nprogression43–45. Thus, our spatial analysis identified tumor cell and \\nhost immune cell interactions that could enable evasion of immune \\nsurveillance and eventual tumor recurrence of HCC.\\nWe studied MRD mechanisms in HCC using transgenic mouse \\nmodels9,14, which, in contrast to previous studies using xenograft mod-\\nels in immunocompromised hosts46–48, allow for in situ investigation \\nof MRD with an intact immune system. A signature derived from these \\ntransgenic MRD models was indicative of shorter recurrence-free \\nsurvival in human HCC, demonstrating translational relevance. Using \\nthese models, we identify that stem-like tumor cells reversibly over -\\nexpress Tgfbr1 and spatially interact with the Tgfβ1-secreting Pdl1 + \\nmacrophages abundant in MRD. Further, blocking the Tgfβ pathway \\nalone was not effective in preventing HCC recurrence. Rather, the con-\\ncordant elimination of Pdl1+ macrophages, which are a source of Tgfb1, \\nwas required to induce a robust T cell response and eliminate residual \\ntumor cells. Additionally, we validated the efficacy of this combination \\nin a mouse model of doxorubicin-resistant HCC. Combined blockade \\nof TGFβ1 and PD-L1 has been trialed in primary cancers 49–51, yielding \\ninconsistent results52 and exhibiting toxicity with extended use. Our \\nstudy suggests that targeting MRD through brief adjuvant therapy can \\npotentially mitigate these limitations and prevent HCC recurrence.\\nOur study presents a comprehensive single-cell spatial map of \\npost-chemoembolization residual human HCC, highlighting critical \\ninteractions between stem-like tumor cells, M2-like macrophages and \\nexhausted CD8+ T cells, insights possible only through the preservation \\nof spatial context. Moreover, using a transgenic mouse model for dis-\\nseminated MRD we demonstrate that the insights gained from spatial \\nanalysis can indeed be actionable, thus guiding us to target Tgfβ and \\nPdl1 to eliminate MRD in HCC. Thus, our results suggest a new adjuvant \\ntherapeutic strategy for reducing recurrence in HCC and improving \\npatient outcomes.\\nMethods\\nEthical approval\\nAll procedures and methods were conducted in compliance with fed-\\neral, state and Stanford University guidelines. Tissue collection from \\npatients with HCC was approved by the Stanford Institutional Review \\nBoard (no. 28374). Written patient consent was obtained as per Insti-\\ntutional Review Board approval for the applicable components of \\nthe study. Animal experiments were approved by Stanford’s Animal \\nProtocol and Laboratory Animal Care (APLAC) and adhere to the USDA \\nAnimal Welfare Act and PHS Policy on Humane Care and Use of Labora-\\ntory Animals.\\nPatient cohort selection\\nIn this study, we identified two cohorts of patients diagnosed with \\nHCC who met the following inclusion criteria. For cohort 1 (residual \\nHCC), patients with a confirmed diagnosis of HCC who received \\nbridging therapy with TACE, subsequently underwent liver transplan-\\ntation and had viable residual HCC in the explanted liver tissue, were \\nselected. Additional criteria included availability of sufficient tissue \\nin FFPE blocks. We carefully selected those nodules which had been \\ntargeted for TACE and ensured adequate samples for our downstream \\nanalysis. Exclusion criteria included a history of other malignancies, \\nreceipt of resection before liver transplantation, receipt of radia -\\ntion therapy before transplant or evidence of metastatic cancer. All \\npatients in this cohort (n = 55) had received doxorubicin-based TACE \\n(total procedures n  = 116), a majority of which was DEB-TACE with \\ndoxorubicin-eluting beads (n \\u2009=\\u2009102, 89%) and a smaller proportion \\nreceived conventional TACE ( n\\u2009=\\u200914, 11%), none had received bland \\nTACE. A consistent team of experienced interventional radiologists at \\na single institution performed all TACE procedures. Patients undergo-\\ning TACE underwent follow-up computed tomography (CT) or MRI \\nscans 8–12 weeks post-procedure to assess therapeutic response, \\nwhich was evaluated using the modified Response Evaluation Criteria \\nin Solid Tumors (mRECIST) system53. Decisions regarding additional \\nTACE treatments were made during multidisciplinary tumor board \\nmeetings. For the control group (primary HCC), patients who under-\\nwent surgical resection for HCC and had not received any locore -\\ngional therapy or systemic therapy before resection were selected. \\nAdditionally, these patients did not have any other form of cancer \\nor metastatic tumors. In both groups, we collected comprehensive \\nclinical and pathological data, including patient demographics, \\ntumor characteristics, details of the treatments received and out -\\ncomes following the treatment. Patients were followed until death or  \\nuntil August 2023.\\nConstruction of tissue microarrays\\nAll tumor tissues were processed uniformly in the Stanford clinical \\npathology laboratory. H&E-stained sections from each FFPE block \\nwere carefully reviewed by the pathologist and areas of viable tumor \\nor nontumorous liver selected. We generated a total of five TMAs with \\n1.5-mm diameter cores which were assembled using a TMA Grand \\nMaster automated tissue microarrayer.\\nCODEX panel development and staining\\nCODEX multiplex staining and analysis . A 41-plex custom CODEX \\nantibody panel was developed and validated (Enable Medicine) for \\nultra-high-plex imaging utilizing purified, carrier-free antibodies con-\\njugated to unique DNA oligonucleotide barcodes (Akoya Biosciences) \\n(Supplementary Table 1). Image processing and analysis were per -\\nformed as described previously26.\\nCODEX antibodies were validated on FFPE tonsil sections, and \\nstaining patterns were confirmed via comparison with online databases \\n(The Human Protein Atlas, www.proteinatlas.org; Pathology Outlines, \\nwww.pathologyoutlines.com) and the published literature. Tissue \\nmicroarrays containing FFPE biopsies from the cohort described in this \\nstudy were sectioned at 5\\u2009μm and placed on 15\\u2009×\\u200915-mm glass coverslips \\n(Electron Microscopy Sciences, 72204-01) precoated with poly-l-lysine \\n(Sigma, P8920). Coverslip staining was performed by Enable Medicine.\\nBriefly, FFPE tissue sections on coverslips were pretreated by heat-\\ning on a slide warmer for 25\\u2009min at 55\\u2009°C. Tissue deparaffinization and \\nhydration were next performed by incubating the FFPE tissue sections \\non coverslips for 5\\u2009min each following a solvent series (Histochoice \\nClearing Agent, Histochoice Clearing Agent, 100% ethanol, 100% etha-\\nnol, 90% ethanol, 70% ethanol, 50% ethanol, 30% ethanol, ddH20 then \\nddH20). Antigen retrieval was performed in 0.01\\u2009M citrate buffer at high \\npressure. The tissue was washed and equilibrated before staining for \\n3\\u2009h at room temperature with the 41-plex CODEX antibody cocktail in ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='667c20c3-e94b-4edb-9776-a479ed487548', embedding=None, metadata={'page_label': '1550', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1550\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\na staining buffer containing blocking solution (Akoya Biosciences). \\nPost-staining, the tissues were washed and fixed in 1.6% PFA, followed by \\nan ice-cold methanol incubation. After washing, the final tissue fix was \\nperformed using Fixative reagent (Akoya Biosciences). FFPE tissues on \\ncoverslips were stored in a six-well plate containing the storage buffer \\nat 4\\u2009°C until CODEX acquisition.\\nCODEX multiplexed imaging and processing.  Stained coverslips \\nwere mounted onto the CODEX stage plate v.2 (Akoya) and secured \\nonto the stage of a BZ-X810 inverted fluorescence microscope \\n(Keyence). Reporter plates were prepared by adding fluorescently \\nlabeled oligonucleotides (Atto550, Cy5, AF750) made up in a reporter \\nstock solution of nuclease free water, 10× CODEX buffer, assay rea -\\ngent and nuclear stain to a black Corning 96-well plate. Automated \\nimage acquisition of tissue regions was performed at Enable Medi -\\ncine using a CFI Plan Apo λ ×20/0.75 objective (Nikon) and fluidics \\nexchange managed via the CODEX instrument and CODEX Instru -\\nment Manager software (v.1.29.3.6, Akoya Biosciences), according \\nto the manufacturer’s instructions, with slight modifications. Raw \\nfluorescent TIFF image files were processed, deconvolved and back-\\nground subtracted utilizing the Enable Processor Pipeline (Enable \\nMedicine) and antibody staining was visually assessed for each bio -\\nmarker and tissue region using the Enable Visualizer (Enable BIOS, \\nEnable Medicine). OME-TIFF hyper stacks were segmented based on \\n4,6-diamidino-2-phenylindole (DAPI) stain, pixel intensities were quan-\\ntified, and spatial fluorescence compensation was performed, which \\ngenerated comma-separated value and flow cytometry standard files for  \\ndownstream analysis.\\nCODEX data analysis. All analyses were run in R v.4.0.5 unless other-\\nwise indicated. R functions are specified using the following notation: \\n‘<package_name>::<function_name>’ .\\nCell clustering. For cell clustering and cell neighborhood analysis, data \\nfrom 108 cores were analyzed. Possible batch effects were addressed \\nby performing an inverse hyperbolic sine transform (‘base::asinh’) \\non cell expression values for every marker, in every region of interest \\n(ROI). Next, normalized values were z-scaled across both cells and mark-\\ners. T o cluster cells, dimensionality reduction was first performed on \\nscaled expression values using principal-component analysisA (PCA) \\nwith 20 components (‘stats::prcomp’). Next, a k -nearest-neighbor \\ngraph was constructed to build a similarity network between cells in \\nprincipal-component space (‘dbscan::kNN’ , k\\u2009=\\u200930). Finally, cells were \\nclustered using the Leiden graph clustering algorithm (‘igraph::cluster_\\nleiden’ , cluster_resolution\\u2009=\\u20091.0).\\nCell populations were defined using iterative unsupervised clus-\\ntering using subsets of the full markers-by-cells expression matrix. \\nAn initial set of coarse cell clusters was first defined by unsupervised \\nclustering on all cells that passed quality control (QC) and the major cell \\nlineage markers in the panel: CD20 (B cell), CD15 (neutrophil), CD68 \\n(macrophage), FoxP3 (Treg), CD31 (endothelial cell), CD56 (NK cell), \\nCD8/CD4/CD45/CD3e (T cell), CD117 (mast cell), CD11c (dendritic cell), \\nPanCK (tumor) and αSMA (fibroblast). Clustering parameters, includ-\\ning granularity, nearest-neighbor number and marker subsets, were \\noptimized by assessing clustering results visually, overlaid on images \\n(Enable Medicine Visualizer); by manually examining the distribution \\nof expression values in each cluster; and by quantifying cluster purity \\nusing the silhouette score. Next, cell subtypes were defined by sub -\\nclustering of major cell categories. The full marker-by-cell expression \\nmatrix was subsetted with the following criteria: tumor cells: PanCK, \\nEpCAM, CK19 and PD-L1; macrophages: HLA-DR, PD-L1 and CD206; and \\nT cells: CD4, PD-1, TIM3, CD44 and CD45RO. Other canonical markers of \\nthese cell types were evaluated to further confirm the cell definitions. \\nIn each case, subclustered cells were examined visually for proper \\nexpression of lineage and subtype markers.\\nConstruction of a spatial cellular interaction graph. T o perform spatial \\nanalyses on the data, we first constructed a spatial nearest-neighbor \\ngraph. Cell coordinates were derived by taking the centroid of each \\nsegmented cell nucleus relative to the corner of the ROI. A fixed \\nradius neighbor algorithm was next used on these coordinates \\n(‘dbscan::fixedrad’ , r\\u2009=\\u200925 or r\\u2009=\\u2009100). This graph thus represents, for \\neach cell, its closest neighbors in two-dimensional space within the \\nspecified fixed radius. Each cell was assessed for its interaction with \\nneighboring cells based on their proximity. Direct interactions were \\ndefined as those occurring between cells within a 25-μm radius. Indirect \\ninteractions were considered for cells located between 25 and 100\\u2009μm \\nfrom each other. Homotypic interactions, where a cell interacted with \\nanother of the same type, were excluded from the analysis. For each \\nsample, we further calculated the proportion of each cell type engaged \\nin interactions with other distinct cell types. This was determined for \\neach central cell by dividing the number of interacting cells of another \\nspecific type by the total number of that central cell type present in the \\nsample. This normalized the interaction data for varying cell densities \\nacross regions.\\nCell neighborhood analysis. T o define CNs, the number of neighbors \\nof each cell type was counted, resulting in a matrix of cells by cell clus-\\nters, with each row representing a cell, each column representing a \\ncell annotation (cell type) from the clustering above and each value \\nrepresenting the count of neighbors of the given annotation. The \\nneighbor cell proportion was computed for each row. The resulting \\nmatrix was clustered using k-means clustering (‘stats::kmeans’), where \\nthe optimal k was determined empirically by maximizing the silhou -\\nette score metric (‘cluster::silhouette’). Each cluster was defined as \\na CN. Thus, each cell was given both a cell type annotation, which \\ndepends only on the cell’s own marker expression and a cell neighbor-\\nhood annotation, which depends on the cell type and the identities of \\nits nearest neighbors. T o compare CNs between patient cohorts, we \\ndetermined the proportion of cells in each ROI belonging to each CN. \\nProportions were transformed using the inverse hyperbolic tangent \\n(‘base::asinh’) and split by cohort. We then performed pairwise t-tests \\n(‘stat::t.test’) on the transformed proportions, comparing each CN \\nbetween primary and residual HCC cohorts. The resulting P  values \\nwere corrected for multiple testing by the Bonferroni method (‘stat::p.\\nadjust’ , method\\u2009=\\u2009‘Bonferroni’).\\nNanoString digital spatial profiling\\nWe used the published experimental methods for the NanoString \\nGeoMx analysis54. The GeoMx Digital Spatial Profiling instrument \\nfrom NanoString T echnologies was used for IF imaging. We used 5-μm \\nsections of the FFPE human HCC TMAs for H&E and Cancer Transcrip-\\ntome Atlas (CTA) processing. CTA processing involved baking slides \\nat 60\\u2009°C, deparaffinization, antigen retrieval, proteinase K digestion, \\nhybridization to RNA probes, washing to remove off-target probes and \\ncounterstaining with morphology markers. The morphology mark -\\ners used were anti-panCK-Alexa Fluor 532, anti-CD45-Alexa Fluor 594 \\nand anti-CD68-Alexa Fluor 647. The GeoMx Digital Spatial Profiling \\ninstrument from NanoString T echnologies was used for IF imaging, \\nROI selection, AOI segmentation and spatially indexed barcode cleav-\\nage and collection. The tissue microarrays underwent staining with \\nfour markers, including panCK, CD45, CD68 and DAPI. These slides \\nwere imaged on the GeoMX platform, which functions in part as a \\nfluorescent slide scanner. ROIs; (n\\u2009=\\u200912 per slide) were selected based \\non the visualization markers, using a custom-designed web-based con-\\ntrol program (NanoString). Both H&E slides and multiplex IF-stained \\nslides were visually examined in collaboration with a pathologist. ROIs \\nwere selected based on criteria such as sufficient cellularity and the \\nabsence of artifacts. Within each ROI, two specific AOIs were identi -\\nfied: one enriched in epithelial cells (PanCK +/CD45−/CD68−) and the \\nother in macrophages (PanCK−/CD45/CD68−). After AOIs were chosen, ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='1bc5c7ce-4d05-4b23-baa4-5df746593c25', embedding=None, metadata={'page_label': '1551', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1551\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nthe GeoMX platform utilized an automatically controlled UV laser to \\nilluminate each AOI in turn, specifically cleaving oligonucleotide tags \\nwithin the AOI but not in the surrounding tissue. A microcapillary col-\\nlection system then collected the liberated oligonucleotides from each \\nregion and plated them into an individual well on a microtiter plate. \\nThis process was repeated in turn for each AOI. After AOI collection \\nwas complete, oligonucleotides were hybridized to complementary \\nNanoString counting beads and counted using an nCounter analysis \\nplatform (NanoString).\\nLibrary preparation involved PCR amplification and sequencing \\non a NovaSeq S2 to achieve a minimum sequencing depth of 150–200 \\nreads per μm2 of illumination area. Digital counts from barcodes cor-\\nresponding to gene probes were normalized using internal spike-in \\ncontrols to account for system variation. Subsequently, these counts \\nwere further normalized to the area of their respective compartments. \\nDifferential gene expression and gene set enrichment analysis were \\nperformed in the NanoString GeoMx web portal. A Benjamini–Hoch-\\nberg correction was used to decrease the false discovery rate. PCA \\nwas conducted in the Omics Explorer software from QluCore (Lund, \\nSweden, v.3.7)). Upstream regulators of transcription were discovered \\nusing the Ingenuity Pathway Analysis software from QIAGEN Digital \\nInsights (v.94302991).\\nMultiplex fluorescent in situ hybridization\\nThe RNAscope (Advanced Cell Diagnostics) fluorescent in situ hybridi-\\nzation technology designed primarily for use on FFPE sections was used \\nto detect TGFB1 and CD68 mRNA in human HCC tissue microarrays. The \\nprocedure was performed as described previously55. Post-fixed TMA \\nsections were subjected to staining using the RNAscope assay protocol, \\nemploying the TGFB1 and CD68 primary target probe. Simultaneously, \\npositive and negative in-house control probes were incorporated \\nwithin each run. The stained sections were scanned on a fluorescence \\nmicroscope (BZ-X800) from Keyence Corporation of America and \\nquantified using QuPath analysis software56.\\nIn vitro 3D co-culture and scRNA-seq\\nMacrophage polarization for the in vitro co-culture experiment was \\nperformed as previously described57. In brief, the human THP1 mono-\\ncytes were cultured in RPMI medium with 10% heat-inactivated fetal \\nbovine serum. THP1 monocytes were differentiated into macrophages \\nby 24\\u2009h incubation with 150\\u2009nM phorbol 12-myristate 13-acetate (PMA, \\nSigma, P8139) followed by 24\\u2009h incubation in RPMI medium. Mac -\\nrophages were polarized in M1 macrophages by incubation with \\n20\\u2009ng\\u2009ml−1 IFNγ (R&D system, 285-IF) and 10\\u2009pg\\u2009ml−1 LPS (Sigma, 8630) \\nfor 48\\u2009h. Macrophage M2 polarization was obtained by a 48-h incuba-\\ntion with 20\\u2009ng\\u2009ml−1 IL-4 (R&D Systems, 204-IL) and 20\\u2009ng\\u2009ml−1 IL-13 (R&D \\nSystems, 213-ILB). After staining the macrophages with Cell Trace Violet \\n(C34557), 3,000 cells of M1 and M2 type cells were added separately to \\nan ultra-low attachment 96-well plate containing 3,000 Huh7 cells in \\neach well. The cells were centrifuged at 1,400\\u2009rpm for 3\\u2009min following \\nwhich they were co-cultured at 37\\u2009°C for 72\\u2009h. IF was performed on the \\nresultant spheroids. Tubulin antibody (CST, 5335S) was used as a com-\\nmon marker for all the cells and preconjugated AF488 CK19 (ab87014) \\nantibody was used to identify the stem cells in the spheroids.\\nIn this study, we evaluated how doxorubicin-resistant cancer cells \\nreprogrammed macrophages. The human HCC cell line Huh7 (Gift \\nfrom the T orok laboratory) and SNU449 (CRL-2234, ATCC) were cul-\\ntured in DMEM, 10% v/v serum, 2\\u2009mM l-glutamine and antibiotics in a \\nhumidified atmosphere of 5% CO2 at 37\\u2009°C. Cells were tested negative \\nfor Mycoplasma contamination. Cells were cultured in the presence \\nof increasing concentrations of doxorubicin for 96\\u2009h to determine \\na half-maximum inhibitory concentration (IC 50) of 1\\u2009μM. Our deter -\\nmined IC50 for doxorubicin in HCC cell lines were consistent with previ-\\nous in vitro studies58–60 and mirrored the concentrations observed in \\nhuman HCC explants following TACE61. Clones of cells that remained \\nviable after doxorubicin exposure or cells exposed to control treat -\\nment for 96\\u2009h were counted (2,500 cells per well) and co-cultured with \\nmacrophages (THP1, TIB-202 and ATCC) (2,500 cells) in 3D spheroids \\ncreated in 96-well low attachment plates. After 96\\u2009h of co-culture, the \\nheterotypic 3D tumoroids from the two experimental conditions were \\npooled separately as single-cell suspensions followed by scRNA-seq \\nusing a BD Rhapsody Single-Cell Analysis System62 to investigate molec-\\nular changes in both the cancer cells and macrophages. A total of 34,474 \\npassed initial QC and were sequenced. Analysis of scRNA-seq data was \\nperformed using the Seurat platform (v.5 of our Seurat toolkit)63.\\nHuman monocyte-derived macrophages and patient-derived \\n3D tumoroids\\nPeripheral blood samples were collected from patients with HCC and \\nused to isolate peripheral blood mononuclear cells, which were then \\nsorted into monocytes using CD14 + beads (130-097-052, Miltenyi). \\nThese monocytes were cultured in six-well plates with 10\\u2009ng\\u2009ml −1 of \\nM-CSF (300-25, Peprotech) and GM-CSF (G5035 Sigma) for 5 days \\nto generate monocyte-derived macrophages (MoMs), as described \\nbefore 64. MoMs were subsequently treated with doxorubicin \\n(130\\u2009ng\\u2009ml−1, AAJ64000MF , Fisher Scientific) or control for 48\\u2009h and \\nthen the MoMs were trypsinized, stained and analyzed by flow cytom-\\netry. In parallel, these MoMs were cultured in a eight-well chamber slide \\n(177402 Lab-T ek) and IF was performed for PD-L1 (ab205921 Abcam).\\nIn parallel, tumor samples from patients with HCC who underwent \\nresection were dissociated into single cells and cultured in tumoroid \\nmedium until forming spheres in a 96-well plate, as described pre -\\nviously65. Upon sphere formation, patient-derived organoids were \\ntreated with doxorubicin (130\\u2009ng\\u2009ml−1) or control. After 72\\u2009h, the con-\\nditioned medium from the patient-derived organoid cultures, treated \\nwith control or doxorubicin, was collected and added to the MoMs. \\nFollowing an additional 48\\u2009h of culture with conditioned medium, \\nthe MoMs were trypsinized, stained and analyzed by flow cytometry.\\nTransgenic mice and in vivo treatment\\nAnimals were housed in a pathogen-free environment at Stanford \\nUniversity and all procedures were performed in accordance with \\nStanford’s APLAC protocols. Mice were generally maintained on a 12-h \\ndark–light cycle at 70\\u2009°F (±2\\u2009°F) and 50% (±20%) humidity. LAP-tTA/\\ntet-O-MYC and LAP-tTA/TRE-Twist1/Luc transgenic lines were used, \\nas previously described 9,14. Only male mice of this strain carry the \\ntet-O-MYC transgene and develop liver cancer upon doxycycline \\nwithdrawal. Littermate controls were used for all experiments. Mice \\nwere administered weekly doses of 0.1\\u2009mg\\u2009ml −1 doxycycline (Sigma) \\nin drinking water during mating and until 4 weeks of age. At 4 weeks, \\nmice were taken off doxycycline. Mice were screened for tumors via \\nMRI at approximately 2–3 months of age, at which time they developed \\ntumors between 50–150\\u2009mm 3. Stanford University’s APLAC allows a \\nmaximal tumor diameter of 1.70\\u2009cm, and this was not exceeded. Once \\nliver tumors and/or lung metastases were confirmed, mice were placed \\non doxycycline to induce tumor regression. MRI was used to confirm \\ncomplete tumor regression. T o induce tumor recurrence, mice were \\ntaken off doxycycline and monitored closely. T o evaluate the effective-\\nness of therapies in preventing tumor recurrence, two weeks following \\ncomplete radiologic tumor resolution, mice were enrolled into one of \\nfour treatment groups and treated for 2 weeks (Fig. 8a). Control rat IgG \\n(BioXCell) and anti-PD-L1 (clone 10F .9G2, BioXCell) antibodies were \\ngiven intraperitoneally (i.p.) (100\\u2009μg per mouse) every other day. The \\nTgfbr1 inhibitor SB431542 was dosed 10\\u2009mg\\u2009kg−1 (dissolved in dimethyl-\\nsulfoxide) daily i.p. as previously described66. At the end of treatment, \\nmice were taken off doxycycline and monitored for tumor recurrence \\nwith MRI. Humane end points for early euthanasia included signs of \\npain, labored breathing or the inability to move and stand normally. \\nOther early euthanasia criteria were a total body weight loss greater \\nthan 20% or more than 10% body weight loss in a single week. Mice ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='cac6f5f2-8cd1-469e-a54f-6a6655e4a70d', embedding=None, metadata={'page_label': '1552', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1552\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nwere killed by gradual carbon dioxide displacement and subsequent \\nconfirmatory cervical dislocation.\\nEstablishment and treatment of immunocompetent \\nsyngeneic allograft models of HCC\\nThe murine HCC cell line Hep 53.4 (ref. 67 ) (gift the Mark Yarchoan \\nLaboratory) derived from HCC arising in the C57BL/6 background \\nwas used to establish orthotopic syngeneic allografts. The cells were \\ncultured in DMEM, 10% v/v serum, 2\\u2009mM l-glutamine and antibiotics in a \\nhumidified atmosphere of 5% CO2 at 37\\u2009°C. Cells were tested as negative \\nfor Mycoplasma contamination. Cells were cultured in the presence of \\nincreasing concentrations of doxorubicin for 96\\u2009h to determine an IC50 \\nof 1\\u2009μM. Clones of cells that remained viable after doxorubicin treat -\\nment or cells exposed to control treatment for 96\\u2009h were used to create \\northotopic allografts. Female C57BL/6J mice at 6–8 weeks old were \\npurchased from The Jackson Laboratories and maintained in accord-\\nance with the APLAC protocol. A total of 3\\u2009×\\u2009106 cells in 40\\u2009μl Matrigel \\n(Corning) and a phosphate-buffered saline (PBS) mixture per mouse \\nwere injected orthotopically into the left lobe of the livers of mice as \\npreviously described 68,69. Two weeks after engraftment, mice were \\nenrolled into two treatment groups and treated for 2 weeks (Extended \\nData Fig. 10a). Control rat IgG (BioXCell) and anti-PD-L1 (clone 10F .9G2, \\nBioXCell) antibodies were given i.p. (100\\u2009μg per mouse) every other \\nday. The Tgfbr1 inhibitor SB431542 was dosed 10\\u2009mg\\u2009kg −1 (dissolved \\nin dimethylsulfoxide) daily i.p. as previously described 66. At the end \\nof treatment, mice were killed and evaluated for treatment response.\\nAnimal magnetic resonance imaging\\nMRI was performed using a 7T small animal MRI (Agilent conversion) \\nwith a 40-mm Varian Millipede RF coil (ExtendMR) at the Stanford Small \\nAnimal Imaging Facility as previously described70,71. In brief, animals \\nwere anesthetized with 1–3% isoflurane and placed into the MRI scanner \\ncontaining a 40-mm Varian Millipede RF coil (ExtendMR). ParaVision \\n(PV6.01) was used to acquire the DICOM images and tumor volumes \\nwere quantified from images using Osirix image processing software \\n(Osirix, UCLA).\\nIntravital microscopy\\nThe window chambers were prepared following previously described \\nmethods72. Under sterile conditions and with appropriate anesthesia, \\nthe titanium window frames were implanted onto the back of NSG mice, \\ncreating a 1-cm diameter hole in the dorsal skin-fold flap while preserv-\\ning the integrity of the opposing dermis, fascial plane and vasculature. \\nRFP-labeled primary HCC cells derived from MYC-HCC and MYC/Twist1 \\nHCC tumors were injected near a major vessel between the fascia and \\ndermis. The window-chamber mice were anesthetized using 1.0–2.0% \\nisoflurane in oxygen flow. Intravital images were obtained using an A1 \\nMP+ Multiphoton microscope (Nikon) with a ×10 water immersion \\nobjective. Throughout the imaging process, the mouse’s tempera -\\nture was maintained at 37\\u2009°C with a warm plate. Large-field imaging \\ncombined with blood vessel imaging (Dextran labeling) was used as a \\n‘position mark’ to focus on the same imaging region and obtain tumor \\nimages on different days.\\nWhole-transcriptome sequencing of mouse tumors\\nRNA sequencing of MYC-HCC was performed at the Beijing Genom-\\nics Institute using their BGIseq 500 platform single-end 150-bp,  \\n20 million reads per sample. MYC-activated tumors (MYC-On, MT-On, \\nn\\u2009=\\u200910), tumors 7 days after MYC inactivation (MYC-Off, MT-Off, n\\u2009=\\u20096), \\nand tumors that recurred upon MYC reactivation (n \\u2009=\\u20093) under -\\nwent whole-transcriptome sequencing. The gene expression level  \\nwas quantified by a software package called RSEM. DESeq software \\nwas used to perform differential expression analysis. Ingenuity Path-\\nway Analysis (QIAGEN) was used to perform functional pathway and \\nsimilarity analysis.\\nscRNA-seq of mouse liver\\nMRD in MYC/Twist-HCC liver. We identified an area of the liver that \\ncontained MRD based on the presence of scar of tumor regression on \\nthe liver surface. We microdissected this area of the liver and isolated \\nnuclei from mouse liver tissue and constructed 3′ single-cell gene \\nexpression libraries (Next GEM v.3.1) using the 10x Genomics Chro -\\nmium system. The library was sequenced on Illumina NovaSeq (PE150). \\nAfter sequencing, clean reads were analyzed with mouse reference \\ngenome mm10-2020-A using Cell Ranger v.5.0. A total of 15,300 cells \\npassed QC metrics, had an average of 24,020 reads per cell and median \\nof 1,646 genes per cell.\\nResidual tumor cells from in vitro studies. MYC was inactivated in a \\nclonal population of primary cells derived from MYC-HCC for 1 week. \\nMYC inactivation was confirmed by immunoblotting and IF . Around \\n90% of the cells died over the period of 1 week; the residual viable cells \\nwere submitted for single-cell sequencing. The 10x barcoding and \\ncomplementary DNA synthesis were performed using 10x chromium \\n3′ scRNA-seq V2 chemistry according to the manufacturer’s instruc -\\ntions. The final libraries were sequenced with the Illumina Hiseq2500 \\naccording to recommended specifications. A total of 4,367 cells passed \\nour QC metrics, had an average of 93,743 reads per cell and a median \\nof 1,347 genes per cell.\\nData were analyzed by ROSALIND (https://rosalind.onramp.bio/), \\nwith a HyperScale architecture developed by ROSALIND. Quality scores \\nwere assessed using FastQC. Cell Ranger (v.5.0) was used to align reads \\nto the Mus musculus genome-build GRCm38, count UMIs, call cell \\nbarcodes and perform clustering. Individual sample reads were nor-\\nmalized via relative log expression using the DESeq2 R library. Read \\ndistribution percentages, violin plots, identity heatmaps and sample \\nMDS plots were generated as part of the QC step using RSeQC. DESeq2 \\nwas also used to calculate fold changes and P  values and perform \\noptional covariate correction. Clustering of genes for the final heatmap \\nof differentially expressed genes (fold change ≥2, P adj value\\u2009<\\u20090.05) \\nwas carried out using the Partitioning Around Medoids method using \\nthe fpc R library. Hypergeometric distribution was used to analyze \\nthe enrichment of pathways, Gene Ontology, domain structure and \\nother ontologies.\\nImmunohistochemistry\\nTissues were fixed in 10% paraformaldehyde and embedded in par -\\naffin for sectioning. Sections were deparaffinized by incubation in \\nxylene and rehydrated by sequential incubation in 100%, 95%, 80%, \\n60% ethanol and deionized water. Antigen retrieval was performed in \\na prewarmed container using the Dako pH 6.1 Target Retrieval Solution \\nfrom Agilent T echnologies. The sections were treated with 3% hydrogen \\nperoxide to quench endogenous peroxidase activity. The sections were \\ncovered in the Dako Serum-Free Protein Block from Agilent T echnolo-\\ngies. The sections were incubated with primary antibody overnight at \\n4\\u2009°C in a humid chamber. Details of the antibodies used and dilutions \\nare provided in Supplementary Table 9. Subsequently, the sections \\nwere incubated with biotinylated anti-mouse (1:300 dilution, Vec -\\ntor Laboratories) or biotinylated anti-rabbit (1:300 dilution, Vector \\nLaboratories) for 30\\u2009min at room temperature. Sections were placed \\nfor 30\\u2009min at room temperature in an ABC reagent (1:300 dilution, \\nVectastain ABC kit, Vector Laboratories). Sections were developed \\nusing 3,3′-diaminobenzidine, counterstained with hematoxylin and \\nmounted with Permount.\\nImmunofluorescence\\nFor paraffin sections, tissues were fixed in 10% paraformaldehyde and \\nembedded in paraffin for sectioning. The sections were deparaffinized \\nby incubation in xylene and rehydrated by sequential incubation in \\n100%, 95%, 80% and 60% ethanol and deionized water. For frozen sec-\\ntions, tissues were frozen in Tissue-T ek O.C.T. Compound (Sakura ', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b8a72af0-386e-4b5b-a4bb-b25d5dcf514f', embedding=None, metadata={'page_label': '1553', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1553\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nFinetek). The sections were equilibrated to room temperature in 1× \\nPBS and then fixed with 4% methanol-free formaldehyde. The sections \\nwere covered in 100% methanol at −20\\u2009°C for 10\\u2009min. The sections were \\ntreated with 3% hydrogen peroxide. The slides were incubated in the \\nImmunofluorescence Blocking Buffer from Cell Signaling T echnolo-\\ngies. The sections were incubated with primary antibody overnight \\nat 4\\u2009°C in a humid chamber. Subsequently, the sections were incu -\\nbated with fluorescent-labeled secondary anti-mouse or anti-rabbit \\nantibodies (1:300 dilution, Abcam) for 30\\u2009min at room temperature. \\nThe sections were mounted with DAPI. The sections were scanned on \\nthe All-in-One Fluorescence Microscope (BZ-X800) from Keyence \\nCorporation of America and quantified on ImageJ software (National \\nInstitutes of Health; NIH).\\nFlow cytometry\\nLiver tissues were collected from treated mice and processed into \\nsingle-cell suspensions through mechanical disaggregation and \\nenzymatic digestion with a collagenase solution. The obtained cells \\nwere then stained with fluorochrome-conjugated antibodies specific \\nto murine immune cell markers. Following staining, cells were ana -\\nlyzed using flow cytometry to identify and quantify distinct immune \\ncell populations based on their fluorescent characteristics. LIVE/\\nDEAD Fixable near-IR (Thermo Fisher Scientific, L34975) was used for \\nlive/dead discrimination. Data were analyzed using FlowJo (v.10.10),  \\nwith cell populations identified through gating strategies that dis -\\ncriminated based on size (forward scatter) and granularity (side \\nscatter). Details of the antibodies used and dilutions are provided in \\nSupplementary Table 9.\\nMacrophage phagocytosis assay\\nConditional HCC cell lines were derived from LAP-tTA and T et-O-MYC \\nor MYC/Twist1 mice. Cells were grown in DMEM (Invitrogen), supple-\\nmented with 10% FBS (Invitrogen) and cultured at 37\\u2009°C in a humidi -\\nfied incubator with 5% CO 2. Cell lines were confirmed to be negative \\nfor Mycoplasma contamination. Raw 264.7 (macrophage) cell sus -\\npensions from ATCC (TIB-71) were prepared with a concentration of \\nabout 3 million cells per ml. Cell suspension was pipetted onto a plate \\nin 100-μl aliquots. Conditioned medium from MYC-HCC cells and \\nconditioned medium from residual MYC-HCC cells (created through \\nchronic treatment with 0.01% doxycycline) were collected. The nega-\\ntive control for the conditioned medium was DMEM (Invitrogen), \\nsupplemented with 10% FBS (Invitrogen) and 0.01% doxycycline solu-\\ntion. The cell suspensions were treated with the conditioned medium \\nfor 2\\u2009h. Then, the phagocytosis assay was performed (Phagocyto -\\nsis Assay kit, ab234054) from Abcam). The reagents were prepared \\naccording to the package instructions. Zymosan slurry was added \\nto the plated cells and the mixture was incubated at a controlled \\ntemperature of 37\\u2009°C with 5% CO2 for 2\\u2009h. The cells were washed in \\nan ice-cold phagocytosis assay buffer. Scanning was completed on \\nthe All-in-One Fluorescence Microscope (BZ-X800) from Keyence \\nCorporation of America and quantification was performed on ImageJ  \\nsoftware (NIH).\\nLuminex assay\\nConditioned medium from MYC-HCC cells (derived from a conditional \\nMYC transgenic mouse model of liver cancer) and conditioned medium \\nfrom residual MYC-HCC cells (created through chronic treatment with \\n0.01% doxycycline) were collected and frozen at −80\\u2009°C. A Luminex \\nassay using a 48-plex mouse cytokine array was performed by the \\nHuman Immune Monitoring Center, Stanford University. Concentra-\\ntion data was analyzed and visualized in GraphPad Prism (v.9.5.1).\\nStatistical analysis and reproducibility . All statistical tests were \\nperformed in R (v.4.05 and 4.3.1). All statistical tests were two-sided \\nunless otherwise stated. Differences between groups were analyzed \\nusing a Student’s t -test or one-way analysis of variance. The Benja -\\nmini–Hochberg method was used for adjusting P values. A chi-squared \\ntest was used to compare categorical variables. Kaplan–Meier analysis \\nwith a log-rank test was performed for survival analysis. All graphs \\nare presented as the mean\\u2009±\\u2009s.e.m. An adjusted P value <0.05 was con-\\nsidered to be significant. No statistical method was used to prede -\\ntermine sample size and no data were excluded. Data normalization \\nwas performed before analysis but this was not formally tested, we \\nhave shown data distribution in the plots as individual data points. If \\nnot stated otherwise, the experiments were not randomized and the \\ninvestigators were not blinded to allocation during experiments and \\noutcome assessment. T o ensure the reproducibility of our results, all \\nexperiments were conducted with adequate replicates. Specific details \\non the number of replicates for each experiment are provided in the \\nrespective methods sections.\\nReporting summary\\nFurther information on research design is available in the Nature \\nPortfolio Reporting Summary linked to this article.\\nData availability\\nSource data for all figures have been provided as Source Data files. \\nAll other data supporting the findings of this study are available from \\nthe corresponding author on reasonable request. CODEX data (the \\nraw data matrix), which includes cell-level information on normalized \\nexpression for all markers, centroid coordinates and the assigned \\ncell type and cohort for each cell, have been deposited in Figshare  \\n(ref. 73). In addition, if raw images are sought, we will make them \\navailable at https://app.enablemedicine.com/portal/visualizer upon \\nrequest to the corresponding author (dhanaser@stanford.edu) within \\n10 days of receiving the request. Nanostring data: the gene expression \\ndata matrix, which includes sample-level information on normalized \\nexpression for all genes, and the assigned AOI type and cohort for each \\nsample, has been deposited in Figshare (ref. 74). The mouse single-cell \\nand whole-transcriptome sequencing data generated in this study are \\npublicly available in the National Center for Biotechnology Information \\nGene Expression Omnibus database under accession codes GSE243176, \\nGSE242746, GSE242745 and GSE242743. Publicly available TCGA liver \\ncancer data were used for analysis from https://www.cbioportal.org/\\nstudy/summary?id=lihc_tcga_pan_can_atlas_2018 . Source data are \\nprovided with this paper.\\nCode availability\\nNo custom code or algorithms were generated in this manuscript.\\nReferences\\n1. Lee, Y.-T. et al. The mortality and overall survival trends of primary \\nliver cancer in the United States. J. Natl. Cancer Inst. 113,  \\n1531–1541 (2021).\\n2. Lei, J. et al. Response to transarterial chemoembolization may \\nserve as selection criteria for hepatocellular carcinoma liver \\ntransplantation. Oncotarget 8, 91328–91342 (2017).\\n3. Kim, D. J. et al. Recurrence of hepatocellular carcinoma: \\nimportance of mRECIST response to chemoembolization and \\ntumor size. Am. J. Transplant 14, 1383–1390 (2014).\\n4. Agopian, V. G. et al. Impact of pretransplant bridging locoregional \\ntherapy for patients with hepatocellular carcinoma within Milan \\ncriteria undergoing liver transplantation: analysis of 3601 patients \\nfrom the US multicenter HCC transplant consortium. Ann. Surg. \\n266, 525–535 (2017).\\n5. Adeniji, N. et al. Impact of bridging locoregional therapies for \\nhepatocellular carcinoma on post-transplant clinical outcome. \\nClin. Transplant. 34, e14128 (2020).\\n6. Ghajar, C. M. Metastasis prevention by targeting the dormant \\nniche. Nat. Rev. Cancer 15, 238–247 (2015).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='3474d96a-6371-432d-bea5-fcea34b12f10', embedding=None, metadata={'page_label': '1554', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1554\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\n7. Lee, T. K.-W., Guan, X.-Y. & Ma, S. Cancer stem cells in \\nhepatocellular carcinoma - from origin to clinical implications. \\nNat. Rev. Gastroenterol. Hepatol. 19, 26–44 (2022).\\n8. Price, T. T. et al. Dormant breast cancer micrometastases reside \\nin specific bone marrow niches that regulate their transit to and \\nfrom bone. Sci. Transl. Med. 8, 340ra73 (2016).\\n9. Shachaf, C. M. et al. MYC inactivation uncovers pluripotent \\ndifferentiation and tumour dormancy in hepatocellular cancer. \\nNature 431, 1112–1117 (2004).\\n10. Ding, W. et al. Clinicopathologic and prognostic significance of \\ntumor-associated macrophages in patients with hepatocellular \\ncarcinoma: a meta-analysis. PLoS ONE 14, e0223971 (2019).\\n11. Zhu, X.-D. et al. High expression of macrophage \\ncolony-stimulating factor in peritumoral liver tissue is associated \\nwith poor survival after curative resection of hepatocellular \\ncarcinoma. J. Clin. Oncol. 26, 2707–2716 (2008).\\n12. Borriello, L. et al. Primary tumor associated macrophages activate \\nprograms of invasion and dormancy in disseminating tumor cells. \\nNat. Commun. 13, 626 (2022).\\n13. Dhanasekaran, R. et al. MYC overexpression drives immune \\nevasion in hepatocellular carcinoma that is reversible through \\nrestoration of proinflammatory macrophages. Cancer Res. 83, \\n626–640 (2023).\\n14. Dhanasekaran, R. et al. MYC and Twist1 cooperate to drive \\nmetastasis by eliciting crosstalk between cancer and innate \\nimmunity. eLife https://doi.org/10.7554/elife.50731 (2020).\\n15. Black, S. et al. CODEX multiplexed tissue imaging with \\nDNA-conjugated antibodies. Nat. Protoc. 16, 3802–3835 (2021).\\n16. Uenishi, T. et al. Cytokeratin 19 expression in hepatocellular \\ncarcinoma predicts early postoperative recurrence. Cancer Sci. \\n94, 851–857 (2003).\\n17. Khosla, R. et al. EpCAM+ liver cancer stem-like cells exhibiting \\nautocrine wnt signaling potentially originate in cirrhotic patients. \\nStem Cells Transl. Med. 6, 807–818 (2017).\\n18. Liu, C.-Q. et al. Expression patterns of programmed death ligand 1 \\ncorrelate with different microenvironments and patient prognosis \\nin hepatocellular carcinoma. Br. J. Cancer 119, 80–88 (2018).\\n19. Pfister, D. et al. NASH limits anti-tumour surveillance in \\nimmunotherapy-treated HCC. Nature 592, 450–456 (2021).\\n20. Cancer Genome Atlas Research Network. Comprehensive \\nand integrative genomic characterization of hepatocellular \\ncarcinoma. Cell 169, 1327–1341.e23 (2017).\\n21. Guo, Y. et al. The heterogeneity of immune cell infiltration \\nlandscape and its immunotherapeutic implications in \\nhepatocellular carcinoma. Front. Immunol. 13, 861525 (2022).\\n22. Yu, S. et al. Tumor-infiltrating immune cells in hepatocellular \\ncarcinoma: Tregs is correlated with poor overall survival. PLoS \\nONE 15, e0231003 (2020).\\n23. Veglia, F. et al. Analysis of classical neutrophils and \\npolymorphonuclear myeloid-derived suppressor cells in cancer \\npatients and tumor-bearing mice. J. Exp. Med. 218, e20201803 \\n(2021).\\n24. Salcher, S. et al. High-resolution single-cell atlas reveals diversity \\nand plasticity of tissue-resident neutrophils in non-small cell lung \\ncancer. Cancer Cell 40, 1503–1520.e8 (2022).\\n25. Singal, A. G. et al. AASLD practice guidance on prevention, \\ndiagnosis, and treatment of hepatocellular carcinoma. \\nHepatology 78, 1922–1965 (2023).\\n26. Schürch, C. M. et al. Coordinated cellular neighborhoods \\norchestrate antitumoral immunity at the colorectal cancer \\ninvasive front. Cell 183, 838 (2020).\\n27. Dasari, A., Grothey, A. & Kopetz, S. Circulating tumor DNA-defined \\nminimal residual disease in solid tumors: opportunities to \\naccelerate the development of adjuvant therapies. J. Clin. Oncol. \\n36, JCO2018789032 (2018).\\n28. Villanueva, A. et al. Combining clinical, pathology, and gene \\nexpression data to predict recurrence of hepatocellular \\ncarcinoma. Gastroenterology 140, 1501–12.e2 (2011).\\n29. Pello, O. M. et al. Role of c-MYC in alternative activation of human \\nmacrophages and tumor-associated macrophage biology. Blood \\n119, 411–421 (2012).\\n30. Li, Z. et al. Cancer-associated fibroblasts promote PD-L1 \\nexpression in mice cancer cells via secreting CXCL5. Int. J. Cancer \\n145, 1946–1957 (2019).\\n31. Zhang, W. et al. IL-6 promotes PD-L1 expression in monocytes \\nand macrophages by decreasing protein tyrosine phosphatase \\nreceptor type O expression in human hepatocellular carcinoma. \\nJ. Immunother. Cancer 8, e000285 (2020).\\n32. Xi, X. et al. Interleukin-22 promotes PD-L1 expression via STAT3 in \\ncolon cancer cells. Oncol. Lett. 22, 716 (2021).\\n33. Kobori, T. et al. Interleukin-18 amplifies macrophage polarization \\nand morphological alteration, leading to excessive angiogenesis. \\nFront. Immunol. 9, 334 (2018).\\n34. Patel, M. & Pillai, A. Management of intermediate-stage \\nhepatocellular carcinoma: systemic versus locoregional therapy. \\nSurg. Oncol. Clin. N. Am. 33, 159–172 (2024).\\n35. Lu, Y. et al. A single-cell atlas of the multicellular ecosystem of \\nprimary and metastatic hepatocellular carcinoma. Nat. Commun. \\n13, 4594 (2022).\\n36. Ma, L. et al. Single-cell atlas of tumor cell evolution in response \\nto therapy in hepatocellular carcinoma and intrahepatic \\ncholangiocarcinoma. J. Hepatol. 75, 1397–1408 (2021).\\n37. Ma, L. et al. Tumor cell biodiversity drives microenvironmental \\nreprogramming in liver cancer. Cancer Cell 36, 418–430.e6 \\n(2019).\\n38. Ruf, B. et al. Tumor-associated macrophages trigger MAIT cell \\ndysfunction at the HCC invasive margin. Cell 186, 3686–3705.e32 \\n(2023).\\n39. Maestri, E. et al. Spatial proximity of tumor-immune interactions \\npredicts patient outcome in hepatocellular carcinoma. \\nHepatology https://doi.org/10.1097/hep.0000000000000600 \\n(2023).\\n40. Govaere, O. et al. Keratin 19: a key role player in the invasion of \\nhuman hepatocellular carcinomas. Gut 63, 674–685 (2014).\\n41. Zhang, J. et al. Overexpression of Epcam and CD133 correlates \\nwith poor prognosis in dual-phenotype hepatocellular carcinoma. \\nJ. Cancer 11, 3400–3406 (2020).\\n42. Noh, C.-K. et al. EpCAM as a predictive marker of tumor \\nrecurrence and survival in patients who underwent surgical \\nresection for hepatocellular carcinoma. Anticancer Res. 38, \\n4101–4109 (2018).\\n43. Zhou, C. et al. Disruption of SLFN11 deficiency-induced CCL2 \\nsignaling and macrophage M2 polarization potentiates anti-PD-1 \\ntherapy efficacy in hepatocellular carcinoma. Gastroenterology \\n164, 1261–1278 (2023).\\n44. Murai, H. et al. Multiomics identifies the link between intratumor \\nsteatosis and the exhausted tumor immune microenvironment in \\nhepatocellular carcinoma. Hepatology 77, 77–91 (2023).\\n45. Li, X. et al. Targeting of tumour-infiltrating macrophages \\nvia CCL2/CCR2 signalling as a therapeutic strategy against \\nhepatocellular carcinoma. Gut 66, 157–167 (2017).\\n46. Martin-Padura, I. et al. Residual dormant cancer stem-cell foci are \\nresponsible for tumor relapse after antiangiogenic metronomic \\ntherapy in hepatocellular carcinoma xenografts. Lab. Invest. 92, \\n952–966 (2012).\\n47. Haraguchi, N. et al. CD13 is a therapeutic target in human liver \\ncancer stem cells. J. Clin. Invest. 120, 3326–3339 (2010).\\n48. Schrader, J. et al. Matrix stiffness modulates proliferation, \\nchemotherapeutic response, and dormancy in hepatocellular \\ncarcinoma cells. Hepatology 53, 1192–1205 (2011).', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='c53d2447-c11c-4b79-9ef5-d6ad24a01df3', embedding=None, metadata={'page_label': '1555', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556\\n 1555\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\n49. Ravi, R. et al. Bifunctional immune checkpoint-targeted \\nantibody-ligand traps that simultaneously disable TGFβ enhance \\nthe efficacy of cancer immunotherapy. Nat. Commun. 9, 741 \\n(2018).\\n50. Mariathasan, S. et al. TGFβ attenuates tumour response to PD-L1 \\nblockade by contributing to exclusion of T cells. Nature 554, \\n544–548 (2018).\\n51. Lan, Y. et al. Enhanced preclinical antitumor activity of M7824, a \\nbifunctional fusion protein simultaneously targeting PD-L1 and \\nTGF-β. Sci. Transl. Med. 10, eaan5488 (2018).\\n52. Metropulos, A. E., Munshi, H. G. & Principe, D. R. The difficulty \\nin translating the preclinical success of combined TGFβ and \\nimmune checkpoint inhibition to clinical trial. eBioMedicine 86, \\n104380 (2022).\\n53. Lencioni, R. & Llovet, J. M. Modified RECIST (mRECIST) \\nassessment for hepatocellular carcinoma. Semin. Liver Dis. 30, \\n52–60 (2010).\\n54. Merritt, C. R. et al. Multiplex digital spatial profiling of proteins \\nand RNA in fixed tissue. Nat. Biotechnol. 38, 586–599  \\n(2020).\\n55. Cosi, C., Millar, M., Beltran, M., Sherry, L. & Gatti-McArthur, S. \\nQuantitative analysis of RNAscope staining for expression in \\nmouse brain tissue as a measure of neuronal activation. MethodsX \\n8, 101348 (2021).\\n56. Bankhead, P. et al. QuPath: open source software for digital \\npathology image analysis. Sci. Rep. 7, 16878 (2017).\\n57. Genin, M., Clement, F., Fattaccioli, A., Raes, M. & Michiels, C. \\nM1 and M2 macrophages derived from THP-1 cells differentially \\nmodulate the response of cancer cells to etoposide. BMC Cancer \\n15, 577 (2015).\\n58. Gambari, R., Hau, D. K.-P., Wong, W.-Y. & Chui, C.-H. Sensitization \\nof Hep3B hepatoma cells to cisplatin and doxorubicin by \\ncorilagin. Phytother. Res. 28, 781–783 (2014).\\n59. Hu, Q.-D. et al. NSC 74859 enhances doxorubicin cytotoxicity  \\nvia inhibition of epithelial-mesenchymal transition in \\nhepatocellular carcinoma cells. Cancer Lett. 325, 207–213  \\n(2012).\\n60. Buschauer, S. et al. Hepatocellular carcinoma cells surviving \\ndoxorubicin treatment exhibit increased migratory potential and \\nresistance to doxorubicin re-treatment in vitro. Oncol. Lett. 15 \\nhttps://doi.org/10.3892/ol.2018.7887 (2018).\\n61. Namur, J. et al. Embolization of hepatocellular carcinoma with \\ndrug-eluting beads: doxorubicin tissue concentration and \\ndistribution in patient liver explants. J. Hepatol. 55, 1332–1338 \\n(2011).\\n62. Ulbrich, J., Lopez-Salmeron, V. & Gerrard, I. BD RhapsodyTM \\nsingle-cell analysis system workflow: from sample to multimodal \\nsingle-cell sequencing data. Methods Mol. Biol. 2584, 29–56 \\n(2023).\\n63. Hao, Y. et al. Dictionary learning for integrative, multimodal and \\nscalable single-cell analysis. Nat. Biotechnol. https://doi.org/ \\n10.1038/s41587-023-01767-y (2023).\\n64. Nielsen, M. C., Andersen, M. N. & Møller, H. J. Monocyte isolation \\ntechniques significantly impact the phenotype of both isolated \\nmonocytes and derived macrophages in vitro. Immunology 159, \\n63–74 (2020).\\n65. Wu, P. V. & Nusse, R. 3D culture of primary patient-derived \\nhepatoblastoma tumoroids. Methods Mol. Biol. 2544, 259–267 \\n(2022).\\n66. Wang, S. K., Xue, Y. & Cepko, C. L. Microglia modulation by TGF-β1 \\nprotects cones in mouse models of retinal degeneration. J. Clin. \\nInvest. 130, 4360–4369 (2020).\\n67. Kress, S. et al. p53 mutations are absent from carcinogen-induced \\nmouse liver tumors but occur in cell lines established from these \\ntumors. Mol. Carcinog. 6, 148–158 (1992).\\n68. Brown, Z. J., Heinrich, B. & Greten, T. F. Establishment of \\northotopic liver tumors by surgical intrahepatic tumor injection in \\nmice with underlying non-alcoholic fatty liver disease. Methods \\nProtoc. 1, 21 (2018).\\n69. Zabransky, D. J. et al. Profiling of syngeneic mouse HCC  \\ntumor models as a framework to understand anti-PD-1 sensitive \\ntumor microenvironments. Hepatology 77, 1566–1579  \\n(2023).\\n70. Lai, I. et al. Lipid nanoparticles that deliver IL-12 messenger RNA \\nsuppress tumorigenesis in MYC oncogene-driven hepatocellular \\ncarcinoma. J. Immunother. Cancer 6, 125 (2018).\\n71. Dhanasekaran, R. et al. MYC ASO impedes tumorigenesis and \\nelicits oncogene addiction in autochthonous transgenic mouse \\nmodels of HCC and RCC. Mol. Ther. Nucleic Acids 21, 850–859 \\n(2020).\\n72. Palmer, G. M. et al. In vivo optical molecular imaging and analysis \\nin mice using dorsal window chamber models applied to hypoxia, \\nvasculature and fluorescent reporters. Nat. Protoc. 6, 1355–1366 \\n(2011).\\n73. Dhanasekaran, R. Spatial analysis reveals targetable \\nmacrophage-mediated mechanisms of immune evasion in \\nhepatocellular carcinoma minimal residual disease. Figshare \\nhttps://doi.org/10.6084/m9.figshare.26539345  \\n(2024).\\n74. Dhanasekaran, R. Nanostring_GeoMx_HCC_9_2_24.csv. Figshare \\nhttps://doi.org/10.6084/m9.figshare.26898886.v1 (2024).\\nAcknowledgements\\nR.D. acknowledges NIH grant CA222676 from the National Cancer \\nInstitute, American College of Gastroenterology Junior Faculty Career \\nDevelopment Grant and Cancer League Award. D.F. acknowledges NIH \\ngrant CA208735 and CA253180 from the National Cancer Institute.  \\nP. Chu helped with mouse histology services.\\nAuthor contributions\\nR.D. and D.F. conceptualized the study. L.L., N.A., A.S., R.R., J.Z., J.P., \\nA.R., A.D., A.S.H., L.T. and V.A. carried out experiments under the \\nsupervision of R.D. and D.F. A.T., A.M., H.B.D.’A., R.P., Y.R., J.Z. and V.C. \\ncontributed to method development. L.L., N.A., A.S., A.D., A.S.H. and \\nY.R. analyzed the data. N. Kambham, B.C.V., M.M.D., C.A.B., N. Kothary, \\nV.C. and R.D. who contributed through clinical data retrieval, clinical \\nfollow-up and sample acquisition. The paper was written by R.D. with \\ninput from all authors.\\nCompeting interests\\nA.T.M., A.E.T., R.P. and H.B.D. are employees of Enable Medicine. All \\nother authors declare no competing interests.\\nAdditional information\\nExtended data is available for this paper at  \\nhttps://doi.org/10.1038/s43018-024-00828-8.\\nSupplementary information The online version  \\ncontains supplementary material available at  \\nhttps://doi.org/10.1038/s43018-024-00828-8.\\nCorrespondence and requests for materials should be addressed to \\nDean W. Felsher or Renumathy Dhanasekaran.\\nPeer review information Nature Cancer thanks Bertram Bengsch, Tim \\nGreten and the other, anonymous, reviewer(s) for their contribution to \\nthe peer review of this work.\\nReprints and permissions information is available at  \\nwww.nature.com/reprints.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='311a5a77-8122-4343-9d0b-0d5d1483017d', embedding=None, metadata={'page_label': '1556', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer | Volume 5 | October 2024 | 1534–1556 1556\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nPublisher’s note Springer Nature remains neutral with regard  \\nto jurisdictional claims in published maps and institutional  \\naffiliations.\\nSpringer Nature or its licensor (e.g. a society or other partner) holds \\nexclusive rights to this article under a publishing agreement with \\nthe author(s) or other rightsholder(s); author self-archiving of the \\naccepted manuscript version of this article is solely governed by the \\nterms of such publishing agreement and applicable law.\\n© The Author(s), under exclusive licence to Springer Nature America, \\nInc. 2024\\n1Division of Gastroenterology and Hepatology, Stanford University, Stanford, CA, USA. 2Enable Medicine, Menlo Park, CA, USA. 3Division of Oncology, \\nDepartments of Medicine and Pathology, Stanford University, Stanford, CA, USA. 4Department of Biomedicine, Aarhus University, Aarhus, Denmark. \\n5Department of Pathology, Stanford University, Stanford, CA, USA. 6Department of Surgery, Stanford University, Stanford, CA, USA. 7Department of \\nRadiology, Stanford University, Stanford, CA, USA. 8Department of Biomedical Data Science and Computer Science, Stanford University, Stanford, CA, \\nUSA. 9These authors contributed equally: Lea Lemaitre, Nia Adeniji, Akanksha Suresh. \\u2009e-mail: dfelsher@stanford.edu; dhanaser@stanford.edu', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='512af5fc-7ce6-45ad-af26-08a953523e59', embedding=None, metadata={'page_label': '1557', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 1 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b8d62922-d8f3-4bc0-b3bf-f58e26038481', embedding=None, metadata={'page_label': '1558', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 1 | Identification, validation and relative proportion of \\nmajor cell types using CODEX. a. UMAP representation of 12 major cell subtypes \\nidentified by CODEX analysis of human HCC. b. Comparison of normalized \\nexpression patterns of stemness markers (EPCAM, CK19, CD44), survival markers \\n(Ki67, BCL2), and mesenchymal markers (vimentin, podoplanin) between \\nstem-like and non-stem-like tumor cell types identified by CODEX analysis of \\nhuman HCCs (Tumor cell EpCAM+ n= 26622 cells, Tumor cell CK19+ n= 34436 \\ncells, Tumor cell CK+ n= 433,918 cells). c. Comparison of normalized expression \\npatterns of macrophage markers between three macrophage subsets identified \\nby CODEX analysis of human HCCs (Macro 206+ n=45628 cells; Macro HLA-DR+ \\nn=55558 cells; Macro PD-L1+ n=53713 cells). Box: 25–75 percentile, whiskers: 5–95 \\npercentile, line: median. d. Stacked bar chart comparing relative proportions \\nof all 20 immune and tumor cell subtypes identified by CODEX between self-\\nreported sex (female n=35 patients, male n=73 patients), HCC etiologies (NASH \\nn=28 patients, ALD n=11 patients, HepC n=35 patients, and HepB n=21 patients), \\nAJCC Stage (I n=58 vs II-III n=50 patients), and grade (1 n=40, 2 n=57, and 3 n=9 \\npatients). e. CODEX image representations comparing exhausted CD8+ T cells, \\nendothelial cell, and neutrophil presence in NASH HCC compared to HCV-HCC. f. \\nBoxplot comparisons of the proportions of exhausted CD8+ T cells, endothelial \\ncells, and neutrophils in NASH HCC (n=28) vs non-NASH HCC (n=80). Box: \\n25–75 percentile, whiskers: 5–95 percentile, line: median. Two-tailed unpaired \\nt-test was used to compare the groups. Abbreviations: UMAP- uniform manifold \\napproximation and projection for dimension reduction, CODEX- co-detection \\nby indexing, HCC- hepatocellular carcinoma, CD- cluster of differentiation, CK- \\ncytokeratin, EPCAM- epithelial cell adhesion molecule, BCL2- B cell lymphoma 2, \\nNASH-nonalcoholic steatohepatitis, ALD- alcoholic liver disease, HepC- hepatitis \\nC, HCV- hepatitis C virus, HepB- hepatitis B, HLA-DR - major histocompatibility \\ncomplex II cell surface receptor, PDL1- programmed cell death ligand 1, DAPI- \\n4′,6-diamidino-2-phenylindole, PanCK- pan cytokeratin.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='70ac13f8-2239-4cbe-ad47-188e55c5de6d', embedding=None, metadata={'page_label': '1559', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 2 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='b22e7bd9-08be-4d65-8956-ab6d5d345af2', embedding=None, metadata={'page_label': '1560', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 2 | Comparison of immune cell distribution in residual \\nHCC. a. Stacked Bar graph containing relatively similar proportions of immune \\ncell populations in HCC as reported by CODEX analysis in this study, the Cancer \\nGenome Atlas Project, the ICGC project and a single cell RNA sequencing \\nresearch study PMID 35355983. b. Boxplot comparisons and Voronoi plot \\nof representative cores of NK cell proportions in primary (n=53 patients) vs \\nresidual HCCs (n=55 patients). Box: 25–75 percentile, whiskers: 5–95 percentile, \\nline: median. Two-tailed unpaired t-test was used to compare the groups. c. \\nBoxplot comparisons and Voronoi plot of representative cores of neutrophil \\nproportions in primary (n=53 patients) vs residual HCCs (n=55 patients). Box: \\n25–75 percentile, whiskers: 5–95 percentile, line: median. Two-tailed unpaired \\nt-test was used to compare the groups. d. Volcano plot showing key tumor and \\nimmune cell populations (with statistical significance) in primary vs residual \\nHCCs in the subgroup of HCCs arising in the cirrhotic liver (n=96 patients). Two-\\ntailed unpaired t-test was used to compare the groups. e. Boxplot comparisons \\nof PDL1+ macrophages and exhausted CD8 T cells between peritumoral TACE-\\nexposed cirrhotic livers (n=3 patients), primary (n=53 patients) vs residual \\nHCCs (n=55 patients). Box: 25–75 percentile, whiskers: 5–95 percentile, line: \\nmedian. Two-tailed unpaired t-test was used to compare the groups. f. Boxplot \\ncomparisons of PDL1+ macrophages and exhausted CD8 T cells between \\nprimary (n=53 patients), residual HCC- not refractory to TACE (n=21 patients) \\nvs residual HCC refractory to TACE (n=24 patients). Box: 25–75 percentile, \\nwhiskers: 5–95 percentile, line: median. Two-tailed unpaired t-test was used to \\ncompare the groups. g. Stacked bar chart comparing relative proportions of all \\n20 immune and tumor cell subtypes identified by CODEX between primary (n=53 \\npatients), residual HCC- not refractory to TACE (n=21 patients) vs residual HCC \\nrefractory to TACE (n=24 patients). h. H&E and Immunofluorescence for PDL1+ \\nmacrophages in hepatic tissue with complete response to TACE (n=5 patients) \\ncompared to residual HCC (n=5 patients). Bar plots compare the proportion \\nof PDL1+ cells between the two groups. Data are presented as mean values +/− \\nSEM. Two-tailed unpaired t-test was used to compare the groups. i. Stacked bar \\nchart comparing relative proportions of all 20 immune and tumor cell subtypes \\nidentified by CODEX between conventional TACE (cTACE, n=6 patients) and \\ndoxorubicin-eluting beads TACE (DEB-TACE, n=49 patients). Abbreviations: HCC- \\nhepatocellular carcinoma, CODEX- co-detection by indexing, RNA-Ribonucleic \\nacid, NK- natural killer, CD- cluster of differentiation, PDL1- programmed cell death \\nligand 1, TACE-transarterial chemoembolization, H&E-hematoxylin and eosin, \\ncTACE- conventional transarterial chemoembolization, DEB-TACE- doxorubicin-\\neluting beads transarterial chemoembolization.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='de24b217-35f7-483e-a283-dacd962ac4d7', embedding=None, metadata={'page_label': '1561', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 3 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='037fe8fd-4a6d-4e75-9a56-b41931251e3c', embedding=None, metadata={'page_label': '1562', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 3 | Interactions of tumor and immune cells in residual \\nHCC. a. Heatmap showing different patterns of direct interactions between \\nthe four tumor cell subtypes in primary (n=53 patients) vs residual HCC (n=55 \\npatients). Two-tailed unpaired t-test was used to compare the groups. * Indicates \\npAdj value<0.05. b. Dot plot showing mean frequency of interaction between \\ntumor cell subtypes and PDL1+ macrophages in primary and residual HCC \\nstratified by median frequency of PDL1+ macrophages in tumor. Two-tailed \\nunpaired t-test was used to compare the groups. * Indicates pAdj value<0.05. c. \\nKaplan Meir curve showing survival analysis of human HCC TCGA cohort (n=372 \\npatients) classified based on expression of genes related to M2-like macrophages \\nand cancer stem cells. Log rank test used. d. IF analysis of human HCC cell lines \\nHuh7 co-cultured in 3D tumoroids with THP1 macrophages polarized to M1-like \\n(n=6 tumoroids) or M2-like (n=8 tumoroids) macrophages. Two-tailed unpaired \\nt-test was used to compare the groups. Bar plot shows quantification of CK19 \\nexpression on the 3D tumoroids. Data are presented as mean values +/− SEM. e. \\nHeatmap of direct interactions between immune cells identified by CODEX in \\nprimary (n=53 patients) and residual HCC (n=55 patients). f. Large neighborhood \\nsizes demonstrate reduced, but still significant spatial autocorrelation for Ki67 \\nand BCL2 markers. Spatial autocorrelation is calculated using Geary’s C statistic. \\nThe spatial autocorrelation for each marker is calculated independently for all \\ntumor regions (n=1.07 million cells) and reported as C′ = 1 - C. The maximal spatial \\nautocorrelation possible is 1. Box: 25–75 percentile, whiskers: 5–95 percentile, \\nline: median. g. Alluvial plot of statistically significant indirect tumor cell \\nsubtype-immune cell interactions in primary (n=53 patients) vs residual HCCs \\n(n=55 patients). h. Heatmap showing different patterns of indirect (25-100um) \\ninteractions of tumor cell subtypes with immune cells in primary (n=53 patients) \\nvs residual HCC (n=55 patients). Two-tailed unpaired t-test was used to compare \\nthe groups. * Indicates pAdj value<0.05. Abbreviations: HCC- hepatocellular \\ncarcinoma, PDL1- programmed cell death ligand 1, TCGA- the cancer genome atlas \\nproject, M2- Type 2 Macrophage, IF- Immunofluorescence, 3D- 3 Dimensional, THP1-\\n, M1- Type 1 Macrophage, CK- cytokeratin, CODEX- co-detection by indexing, CD- \\ncluster of differentiation, EPCAM- epithelial cell adhesion molecule, HLA-DR - major \\nhistocompatibility complex II cell surface receptor, NK- natural killer, Eff- Effector, \\nExh- Exhausted, Mem- Memory, DC- Dendritic cell, EC-, Macro- Macrophage, CSC- \\nCancer Stem cell, Mac- Macrophage, BCL2- B cell lymphoma 2.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='eac27eb3-dda9-436a-9969-6233c81d4906', embedding=None, metadata={'page_label': '1563', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 4 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='a1733ddc-2143-4319-8e3f-96817f70ebd0', embedding=None, metadata={'page_label': '1564', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 4 | Cellular neighborhood analysis of residual HCC. \\na. Cross-matching of Voronoi plots showing spatial neighborhoods and \\nthe associated H&E and CODEX immunofluorescence staining images \\nrepresentations of the same tumor core. The canonical markers of key cells \\nrepresented in each neighborhood are shown. b. Stacked bar chart comparing \\nrelative proportions of all 9 cellular neighborhoods identified by CODEX by \\nHCC AJCC Stage (I n=58 vs II-III n=50 patients), grade (1 n=40, 2 n=57, and 3 n=9 \\npatients), and etiologies (NASH n=28 patients, ALD n=11 patients, HepC n=35 \\npatients, and HepB n=21 patients). c. Boxplot comparisons of CK19, EpCAM, \\nCD3, aSMA, CD15, and CD68 expression amongst the identified cellular \\nneighborhoods (CK19+Tumor CN=33708 cells, EpCAM+ Tumor CN=28190, \\nFibroinflammatory immune CN= 140,273, Innate immune CN= 53,912, M2-\\nmacrophage Immune CN= 188,407, Other= 6,130, T cell Immune CN=51,987, \\nPauci-immune tumor CN=310,977, Vascular inflammatory Tumor CN=251,581). \\nBox: 25–75 percentile, whiskers: 5–95 percentile, line: median. d. Comparison of \\nproportion of EpCAM+ tumor cells and exhausted CD8 T cells in residual HCC \\nstratified by median frequency of M2-macrophage CN (Low n=17; High n=38 \\npatients). e. Comparison of specific canonical marker expression on macrophage \\nand T cells in the EpCAM+ tumor CN (macrophages n=272 cells, T cells n=13 \\ncells), CK19+ tumor CN (macrophages n=1201 cells, T cells n=354 cells) compared \\nto vascular inflammatory tumor CN (macrophages n=16143 cells, T cells \\nn=1275 cells) in residual HCC. Abbreviations: H&E-hematoxylin and eosin, HCC- \\nhepatocellular carcinoma, CODEX- co-detection by indexing, NASH-nonalcoholic \\nsteatohepatitis, ALD- alcoholic liver disease, HepC- hepatitis C, HepB- hepatitis \\nB, CK- cytokeratin, CD- cluster of differentiation, EPCAM- epithelial cell adhesion \\nmolecule, aSMA- alpha smooth muscle actin, CN- Cellular neighborhood, PD1- \\nprogrammed cell death protein 1, TIM3- T cell immunoglobulin and mucin domain-\\ncontaining protein 3, Macro- macrophages, DAPI-4′,6-diamidino-2-phenylindole, \\nPanCK- Pan-cytokeratin, ECad- E cadherin.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='10364b1a-e9a3-412a-b71c-3975269ce2d7', embedding=None, metadata={'page_label': '1565', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 5 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='9704a1ad-4320-425b-98e7-88a205de3940', embedding=None, metadata={'page_label': '1566', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 5 | Pathway analysis of Residual tumor cells and \\nMacrophages. a. Characterizing the HCC cancer cell clusters C0 cluster between \\ndoxorubicin-resistant (DoxR) (n=4320 cells) and control samples (n=3627 cells). \\nGene set expression analysis shows enrichment of HCC progenitor signature \\nand doxorubicin resistance signature in the cancer cells of the DoxR samples. \\nb. Dot plot shows mean expression of key differentially expressed genes in C0 \\nand C5 clusters between control (C0 n=3627 cells; C5 n=161 cells) and DoxR (C0 \\nn=4320 cells; C5 n=57 cells) samples. c. Flow cytometry of monocyte-derived \\nmacrophages (MoM) from HCC peripheral blood mononuclear cells (PBMCs) \\nwere cultured for 48 hours with conditioned media from patient-derived \\norganoid (PDO) which were either treated with doxorubicin (MoM+ DoxoR \\nPDO CM; n=11,753 cells) or control (MoM+ ctrl PDO CM; n=2089 cells), d. Flow \\ncytometry of monocyte-derived macrophages (MoM) from HCC peripheral \\nblood mononuclear cells (PBMCs) were treated with doxorubicin-containing \\nmedia (MoM+Doxorubicin, n=7559 cells) or doxorubicin-free control media \\n(MoM+ctrl media, n=15,546 cells). This experiment was repeated using MoM \\nfrom patients with HCC which were then treated with doxorubicin (n=3 \\npatients) or control (n=3 patients). Two-tailed unpaired t-test used to compare \\nthe proportion of PDL1+ MoM on flow cytometry. e. Recurrence-free survival \\npredicted by the macrophage and tumor cell signatures derived from the spatial \\ntranscriptomic analysis. Kaplan Meir analysis with log rank test was performed \\n(n=372 patients, unique biological samples). f. TGFBR1/2 pathway displayed \\nas an upstream regulator of the network of transcriptional changes in the \\ntumor cell AOI and the corresponding ligand TGFB1 in the macrophage AOI of \\nresidual HCC. Upstream analysis in Ingenuity Pathway Analysis was performed \\nusing a two-tailed Fisher’s Exact T est to identify likely upstream regulators \\nbased on differential gene expression data. Abbreviations: HCC- hepatocellular \\ncarcinoma; DoxR- Doxorubicin-Resistant; PDO- Patient-derived organoid; CM- \\nConditioned media; Ctrl- Control; MoM- Monocyte-derived macrophage; PBMCs- \\nPeripheral blood mononuclear cells, PDL1- programmed cell death ligand 1, TGFB1 \\ntransforming growth factor beta 1; Res- Residual; Tum- Tumor; HR- hazard ratio. \\nAOI- Area of interest.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='349e2355-d91e-42a8-b3ed-a229a3af7c99', embedding=None, metadata={'page_label': '1567', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 6 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='e4b40bc7-f315-4b51-8861-831364cf08b7', embedding=None, metadata={'page_label': '1568', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 6 | Translational relevance of MRD model of MYC-HCC \\nand MT -HCC. a. Representative H&E images and IHC for HCC-specific proteins \\nglutamine synthetase, HNF4A and stem cell marker CK19. This was performed \\non multiple tumor samples (MYC-HCC=5, MT-HCC=5 mice). b. Expression of \\nHCC-specific genes in primary (MYC-HCC=5, MT-HCC=5 mice) compared to \\nWT FVB mouse liver (n=2 mice). Data are presented as mean values +/− SEM. c. \\nHeatmap shows similarity between MYC-HCC=5, MT-HCC=5 and five human HCC \\ntranscriptome datasets including the TCGA cohort; the top 5 upregulated and \\ndownregulated genes across the different datasets are shown d. Representative \\nH&E and IHC images showing phospho histone 3 (pH3) and cleaved caspase 3 \\n(cC3) staining in primary HCC, minimal residual disease (MRD), and recurrent \\nHCC in MYC-HCC and MYC-Twist1 HCC livers. Bar plots comparing quantification \\nof phospho histone 3, and cleaved caspase 3, in primary tumors, MRD, and \\nrecurrent tumors in the liver of MYC-HCC. This was repeated n=5 mice in each \\ngroup. Data are presented as mean values +/− SEM. e. Experimental scheme \\nto induce NASH in MT-HCC. Gross liver images, H&E, and trichrome staining \\ndemonstrating induction of NASH, fibrosis, and HCC in this model. Experiments \\nwere performed on 3–5 animals in each group. f. Confirming induction of NASH \\nby demonstrating hepatic inflammation, obesity, hyperlipidemia with high-fat \\ndiet (HFD) (c1 chow=5, HFD=3; c2-5 chow=4; HFD=4 mice). Data are presented \\nas mean values +/− SEM. Two-sided unpaired t-tests were used for comparison. \\ng. Confirming reversible residual HCC upon oncogene inactivation in MT-HCC \\nwith diet-induced NASH. Experiments were performed on 5 mice on HFD diet. \\nh. Representative H&E and IHC images showing persistence of CD133+ stem-like \\ncancer cells in perivascular niches of oncogene inactivated MRD in the liver, \\nlungs and subcutaneous spaces. Experiments were performed on 5 mice in each \\ngroup. i. Representative H&E and IHC images showing persistence of CK19+ \\nstem-like cancer cells in perivascular niches of MRD in the subcutaneous spaces. \\nBar plots comparing quantification of CK19 and CD133 in primary tumors, MRD, \\nand recurrent tumors in the liver of MYC-HCC. Experiments were performed on \\n5 mice in each group. Data are presented as mean values +/− SEM Abbreviations: \\nH&E-hematoxylin and eosin, IHC- Immunohistochemistry, HCC- hepatocellular \\ncarcinoma, HNF4A- Hepatocyte nuclear factor-4 alpha, CK- cytokeratin, MYC-\\nHCC- MYC-driven hepatocellular carcinoma, MT-HCC- MYC/Twist hepatocellular \\ncarcinoma, WT FVB- wild-type FVB, TCGA- the cancer genome atlas project, pH3- \\nphospho histone 3, cC3- cleaved caspase 3, MRD- Minimal residual disease, NASH \\nNonalcoholic steatohepatitis, HFD- High-fat diet, AL T-Alanine aminotransferase, \\nLDL - Low density lipoprotein, CD- cluster of differentiation.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='29c4589c-3a1e-45c9-90c3-a552be4c2884', embedding=None, metadata={'page_label': '1569', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 7 | Residual tumor cells demonstrate stemness and \\nactivation of Tgfβ pathway. a. Experimental scheme for single cell sequencing \\non MYC inactivation residual tumor cells (n=4367 cells). b. Two major clusters \\nupon MYC inactivation, stem cell-like cluster (n=1184 cells) and non-stem-\\nlike cluster (n=3183 cells). c. Heatmap shows differentially expressed genes \\nbetween the stem cell-like cluster and non-stem-like clusters. d. Activation \\nof Tgfβ1 pathway in the stem cell-like cluster of residual tumor cells. e. T op \\nmolecular pathways activated in the stem cell-like cluster. Abbreviations: TGFB1 \\ntransforming growth factor beta 1, HCC- hepatocellular carcinoma, scRNA-seq- \\nsingle cell RNA sequencing.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d5d6091e-0f0f-4052-9fcf-1e58f5931392', embedding=None, metadata={'page_label': '1570', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 8 | Interaction of stem-like cancer cells and macrophages \\nin MRD. a. Comparison of macrophages in the spatial proximity of MRD (n=6 \\nmice) versus areas without MRD (n=6 mice) in the lungs of MYC/Twist1 mice. Data \\nare presented as mean values +/− SEM. b. Representative images demonstrating \\nspatial interactions of stem-like cancer cells and macrophages in the spatial \\nproximity of MRD in the lungs and liver of MYC/Twist1 mice. c. Comparison of \\nimmune cell subsets between livers without MRD ctrl Liver (n=4 mice) and livers \\nwith MRD (n=4 mice). Bar plots compared using two-tailed unpaired t-tests. \\nData are presented as mean values +/− SEM. d. Comparison of CD8 T cell subsets \\nbetween livers without MRD ctrl Liver (n=4 mice) and livers with MRD (n=4 mice). \\nBar plots compared using two-tailed unpaired t-tests. Data are presented as \\nmean values +/− SEM. Abbreviations: MRD- Minimal residual disease, ctrl- control, \\nCD- cluster of differentiation, H&E-hematoxylin and eosin, DAPI-4′,6-diamidino-2-\\nphenylindole, HCC- hepatocellular carcinoma.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='5d63e5ec-ce37-4869-a384-b4af42a138da', embedding=None, metadata={'page_label': '1571', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 9 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d9291cbd-0124-400f-a0c5-366345877738', embedding=None, metadata={'page_label': '1572', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 9 | Combined blockade of Tgfβr1 and Pdl1 in mouse \\nMRD HCC. a. Experimental scheme for treatment of oncogene-deprived \\nsubcutaneous MRD-bearing mice with control antibody or combination therapy \\nwith Tgfbr1 and Pdl1 inhibitors. Treatment is followed by oncogene reactivation \\nto induce tumor recurrence. Kaplan–Meier analysis with log rank test was \\nperformed. b. Kaplan–Meier curves show time to recurrence in oncogene-\\ndeprived subcutaneous MRD-bearing mice with control antibody (n=4 mice) or \\ncombination therapy with Tgfbr1 and Pdl1 inhibitors (n=5 mice). c. Experimental \\nscheme for treatment of oncogene-deprived subcutaneous MRD-bearing mice \\nwith control antibody or combination therapy with TgfbrI and anti-Pdl1 inhibitor. \\nResidual tumor niches are then evaluated at the end of treatment. d. Macroscopic \\nand microscopic evaluation of subcutaneous MRD sites shows elimination of \\nresidual tumor cells in mice treated with combination therapy with TgfbrI and \\nPdl1 inhibitors (n=4 mice) than control (n=4 mice). e. Experimental scheme for \\ntreatment of oncogene-activated primary MT-HCC with control antibody or \\ncombination therapy with TgfbrI and anti-Pdl1 inhibitor. Liver tumor burden is \\nassessed at the end of treatment. f. Quantification of liver tumor burden, gross \\nimages, and H&E images of primary MT-HCC treated with control antibody (n=4 \\nmice) or combination therapy with TgfbrI and anti-Pdl1 inhibitor (n=4 mice). \\nData are presented as mean values +/− SEM. Two-tailed unpaired t-test was \\nused to compare the groups. g. Apoptosis measured by cleaved caspase 3+ cells \\nin liver MRD of MYC-HCC mice treated with control (n=5 mice) versus Tgfbr1 \\n(n=5 mice) or Pdl1 inhibitors (n=5 mice) or combination therapy with Tgfbr1 \\nand Pdl1 inhibitors (n=5 mice). Data are presented as mean values +/− SEM. \\nTwo-tailed unpaired t-test was used to compare the groups. h. Flow cytometry \\nquantification of activated CD4 and CD8 T cells which are CD69+/CD44high in \\nMRD of mice treated with control antibody (n=3 mice) or combination therapy \\nwith Tgfbr1 and Pdl1 inhibitors (n=3 mice). Bar plots compare the mean between \\nthe groups with unpaired t-tests. Data are presented as mean values +/− SEM. \\nTwo-tailed unpaired t-test was used to compare the groups. Abbreviations: \\nHCC- hepatocellular carcinoma, PDL1- programmed cell death ligand 1, TGFBr1 \\ntransforming growth factor receptor beta 1, H&E-hematoxylin and eosin, Ctrl-\\ncontrol, MRD- Minimal residual disease, MT-HCC- MYC/Twist hepatocellular \\ncarcinoma, CD- cluster of differentiation.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='f83dbd58-efe2-4c64-9059-77b0d09c0346', embedding=None, metadata={'page_label': '1573', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 10 | See next page for caption.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='17f695b6-aa6e-4d2c-9415-8151994876ca', embedding=None, metadata={'page_label': '1574', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='Nature Cancer\\nArticle https://doi.org/10.1038/s43018-024-00828-8\\nExtended Data Fig. 10 | Combined blockade of Tgfβr1 and Pdl1 eliminates \\ndoxorubicin-resistant mouse HCC. a. Schematic showing the establishment \\nof mouse model of control (n=5 mice) or doxorubicin-resistant syngeneic \\northotopic allografts (n=11 mice) which were then treated with control antibody \\n(n=6 mice) or combination therapy with TgfbrI and anti-Pdl1 inhibitor (n=10 \\nmice). b.Representative gross images, H&E and IHC images of control or \\ndoxorubicin-resistant orthotopic HCC-bearing mice treated with control \\nantibody (n=6 mice) or combination therapy with TgfbrI and anti-Pdl1 inhibitor \\n(n=10 mice). IHC for CD8 T cells shows in the bottom panel. c. Quantification \\nof tumor burden and CD8T cells in the liver of control or doxorubicin-resistant \\northotopic HCC-bearing mice treated with control antibody (n=6 mice) or \\ncombination therapy with TgfbrI and anti-Pdl1 inhibitor (n=10 mice). Bar plots \\ncompare the mean between the groups with two-sided unpaired t-tests. Data \\nare presented as mean values +/− SEM. d. Flow cytometry-based quantification \\nof tumor-infiltrating leukocytes, T cells, M2-like (PDL1+ or CD206+) and \\nM1-like macrophage (CD86+ or MHCII+) subsets in doxorubicin-resistant \\northotopic HCC-bearing mice treated with control antibody (n=4 mice) or \\ncombination therapy with TgfbrI and anti-Pdl1 inhibitor (n=4 mice). Bar plots \\ncompare the mean between the groups two-sided with unpaired t-tests. Data \\nare presented as mean values +/− SEM. e. Representative flow cytometry images \\nand quantification of M1-like and M2-like macrophages within the tumor in \\nthe doxorubicin-resistant orthotopic HCC-bearing mice treated with control \\nantibody or combination therapy with TgfbrI and anti-Pdl1 inhibitor. f. Hep 53.4 \\nHCC cell lines are treated with control or doxorubicin for 96 hours and viable \\ncells are selected for orthotopic implantation into mouse liver, confirmed by \\nMRI. g. Flow cytometry analysis shows no difference in leukocyte or T cell or \\nmacrophage infiltration in WT mice bearing control HCC treated with either \\ncontrol antibody (n=2 mice) of combined inhibition of Tgfbr1 and anti-Pdl1 \\nantibody (n=3 mice). Data are presented as mean values +/− SEM. h. Schematic \\nsummarizing the main findings- i. The spatial organization of post-TACE residual \\nHCC is unveiled through integrated analysis of single-cell spatial profiling \\nemploying CODEX and GeoMx spatial transcriptomics of human HCC. ii. Mouse \\nmodel of minimal residual disease (MRD) reveals that TGFβ1 derived from \\nPDL1+ macrophages enable the persistence of residual stem-like tumor cells and \\ninduces exhaustion in CD8T cells. iii. In two mouse models of MRD, we target the \\nTGFβ pathway and PDL1, eliminating residual tumor cells and preventing HCC \\nrecurrence. Abbreviations: HCC- hepatocellular carcinoma, Tgfb -transforming \\ngrowth factor beta, H&E-hematoxylin and eosin, Ctrl-control, Res-residual, MRD- \\nMinimal residual disease, DoxR- doxorubicin-resistant.', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='22d0c92d-1b35-480c-9ec8-9b3233bf21fd', embedding=None, metadata={'page_label': '1575', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='d8f7e4ec-1ab7-488f-94f9-fbd467fbc093', embedding=None, metadata={'page_label': '1576', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='2d107014-fa45-4501-9d7b-ba6d7e82ad7a', embedding=None, metadata={'page_label': '1577', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='897aa83a-abbb-49eb-9c5f-b1998195456e', embedding=None, metadata={'page_label': '1578', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='4879f54c-1f18-4feb-ab38-4b9240b2bc58', embedding=None, metadata={'page_label': '1579', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=None, image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}'),\n",
              " Document(id_='43043bda-1d17-44a6-93e5-3f504b7bfbd0', embedding=None, metadata={'page_label': '1580', 'file_name': 'lemaitre_hepatocellular.pdf', 'file_path': '/content/data/lemaitre_hepatocellular.pdf', 'file_type': 'application/pdf', 'file_size': 36730865, 'creation_date': '2025-04-28', 'last_modified_date': '2025-04-28'}, excluded_embed_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], excluded_llm_metadata_keys=['file_name', 'file_type', 'file_size', 'creation_date', 'last_modified_date', 'last_accessed_date'], relationships={}, metadata_template='{key}: {value}', metadata_separator='\\n', text_resource=MediaResource(embeddings=None, data=None, text='-', path=None, url=None, mimetype=None), image_resource=None, audio_resource=None, video_resource=None, text_template='{metadata_str}\\n\\n{content}')]"
            ]
          },
          "execution_count": 27,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "documents"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mFEgvBZ1NFY9",
        "outputId": "56b1ec84-c7c6-4b30-90e8-213586f07235"
      },
      "outputs": [
        {
          "data": {
            "text/plain": [
              "<llama_index.core.indices.vector_store.base.VectorStoreIndex at 0x7e6890c4e850>"
            ]
          },
          "execution_count": 28,
          "metadata": {},
          "output_type": "execute_result"
        }
      ],
      "source": [
        "index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Uor2s7O8Q4GP",
        "outputId": "50a7780f-3e80-41b5-88c5-cc254ae16813"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Cellular neighborhoods enriched in M1-like MDM showed positive clinical variable association and had no negative association.\n"
          ]
        }
      ],
      "source": [
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"which cellular neighborhoods were positive clinical variable association and had no negative association?\")\n",
        "print(response)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h5WX-9rkNlNV",
        "outputId": "eb7b987b-b6e7-4314-d8cd-cd74bdd6eb68"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Cellular neighborhoods enriched in M1-like MDM showed positive clinical variable association and had no negative association.\n"
          ]
        }
      ],
      "source": [
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"which cellular neighborhoods were positive clinical variable association and had no negative association?\")\n",
        "print(response)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hGm9eAympR2s",
        "outputId": "aa6e5131-092d-4c7c-a745-4a289f6dcec2"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "The unique cellular neighborhoods identified include:\n",
            "1. Neutrophil-Enriched neighborhood: characterized by a high density of neutrophils associated with vasculature and innate immune cells.\n",
            "2. Neuroendocrine-Enriched neighborhood: a mixture of epithelial and immune cell types enriched, with denser neuroendocrine cells compared to other epithelial neighborhoods.\n",
            "3. Paneth-Cell-Enriched neighborhood: observed only within the small intestine, enriched with Paneth cells known to be restricted to the small intestine and enriched within the intestinal crypt.\n"
          ]
        }
      ],
      "source": [
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"list the unique cellular neighborhoods identified with details of the type of cells\")\n",
        "print(response)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Yja2FYZUpAxg",
        "outputId": "94fc70a6-4c36-42c3-8db0-247fab4d7c9c"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Vimentin+ and fibronectin+ fibroblast abundances were not correlated, but their respective neighborhoods were inversely correlated, suggesting a negative clinical variable association.\n"
          ]
        }
      ],
      "source": [
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"which cellular neighborhood was striclty negative clinical variable association?\")\n",
        "print(response)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ruiaUxAbpDPz",
        "outputId": "9a43f772-0388-4c67-b131-6c8d276a128a"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Precision Oncology\n"
          ]
        }
      ],
      "source": [
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"Disease full name\")\n",
        "print(response)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bEx7eAZepKtL",
        "outputId": "eb58080e-32b1-4db0-cc80-87f969b58dd8"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "The paper focuses on survival, response to immunotherapy, and histology type.\n"
          ]
        }
      ],
      "source": [
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"what factors does the paper focus on? : survival,response to immunotherapym,histology type,None\")\n",
        "print(response)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "GBcl9hEnPVTV"
      },
      "outputs": [],
      "source": [
        "from llama_index.core.retrievers import VectorIndexRetriever\n",
        "from llama_index.core.query_engine import RetrieverQueryEngine\n",
        "from llama_index.core.indices.postprocessor import SimilarityPostprocessor\n",
        "retriever = VectorIndexRetriever(\n",
        "    index=index,\n",
        "    similarity_top_k=4,\n",
        ")\n",
        "postprocessor = SimilarityPostprocessor(similarity_cutoff=0.8)\n",
        "query_engine = RetrieverQueryEngine(\n",
        "    retriever=retriever,node_postprocessors=[postprocessor])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JQqMLFkcSGmv",
        "outputId": "f3dc749a-2099-436b-8257-f8294f6fbd2e"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Final Response: The paper focuses on survival, response to\n",
            "immunotherapy, and histology type.\n",
            "______________________________________________________________________\n",
            "Source Node 1/2\n",
            "Node ID: 5e09b69e-7c50-448f-838e-4ffdb9b0ec6f\n",
            "Similarity: 0.8244290984390621\n",
            "Text: comparisons. If the ANOVA result was signi ﬁcant, a Tukey\n",
            "honestly signiﬁcant difference post-hoc test was conducted to\n",
            "determine which groups were signi ﬁcantly different from one another.\n",
            "A Benjamini–Hochberg correction was used to account for multiple\n",
            "hypothesis testing in analyses that involved systematically testing\n",
            "multiple variables. p-va...\n",
            "______________________________________________________________________\n",
            "Source Node 2/2\n",
            "Node ID: 219e162f-df82-4f55-b5d1-9d9c0c5074c9\n",
            "Similarity: 0.824426433156154\n",
            "Text: As opposed to Vodnala et al., we did not detect a complete\n",
            "reduction for the 2HC-treated T cell condition in the B16F10 model,\n",
            "which could be attributed to the aggressive nature of the B16F10\n",
            "model.11 Three days post-treatment, CODEX imaging of the tumors\n",
            "indicated immune inﬁltration (CD45), tumor inﬂammation (H2Kb, PDL1),\n",
            "and downregulation of ...\n"
          ]
        }
      ],
      "source": [
        "from llama_index.core.response.pprint_utils import pprint_response\n",
        "pprint_response(response, show_source=True)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "pq8hpPbvectL",
        "outputId": "a972eabf-e4dd-4fc7-c722-137b6da50cb7"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Cellular neighbourhoods play a crucial role in understanding the spatial organization and interactions of different cell types within tissues. By analyzing cellular neighbourhoods, researchers can identify patterns of cell distribution, cell-cell interactions, and how these neighborhoods may impact various biological processes such as survival, immune responses, and disease progression.\n"
          ]
        }
      ],
      "source": [
        "'''Storing indexed data in a disk to avoid relying on in-memory storage and retrieving data from disk'''\n",
        "import os.path\n",
        "from llama_index.core import (\n",
        "    VectorStoreIndex,\n",
        "    SimpleDirectoryReader,\n",
        "    StorageContext,\n",
        "    load_index_from_storage,\n",
        ")\n",
        "\n",
        "PERSIST_DIR = \"./storage\"\n",
        "if not os.path.exists(PERSIST_DIR):\n",
        "    documents = SimpleDirectoryReader(\"data\").load_data()\n",
        "    index = VectorStoreIndex.from_documents(documents)\n",
        "    index.storage_context.persist(persist_dir=PERSIST_DIR)\n",
        "else:\n",
        "    storage_context = StorageContext.from_defaults(persist_dir=PERSIST_DIR)\n",
        "    index = load_index_from_storage(storage_context)\n",
        "\n",
        "query_engine = index.as_query_engine()\n",
        "response = query_engine.query(\"Cellular neighbourhoods\")\n",
        "print(response)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 40,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UU1rix5rEymQ",
        "outputId": "eab4c7e8-7b1a-45ba-9822-ab3cccd60e8d"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Processing PDF: /content/data/ak_prostate.pdf\n",
            "Extracted Title: Multiplex imaging of localized prostate tumors reveals altered spatial organization of AR-positivecells in the microenvironment\n",
            "Extracted for disease_name: The disease focus of the study is prostate cancer.\n",
            "Extracted for clinical_variable: The clinical variables analyzed in the study include AR + rCN frequency between TAN and tumors (G3 and G4), as well as the mean AR + rCN frequency differences between TAN and tumors (G3 and G4) in subsampling analysis. Additionally, the study also looked at the AR + rCN (CN1) frequency differences between TAN and tumors (G3 and G4) in spatial significance tests 1 and 2.\n",
            "Extracted for unique_neighborhoods: Four neighborhoods were identified:\n",
            "1. AR+ non-immune stromal cells interacting with neuroendocrine cells.\n",
            "2. Mesenchymal/endothelial cells interacting with immune cells.\n",
            "3. Smooth muscle and M2 macrophages.\n",
            "4. Lack of AR+ stroma interacting with various cell types such as luminal epithelial cells, EMT/infiltrating VIM+ cells, AMACR-high luminal cells with APC immune cells, basal cells, neuroendocrine cells, and luminal cells.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods identified were rCN0, rCN1, rCN2, rCN3, rCN4, rCN5, rCN6, rCN7, rCN8, and rCN9. The clinical variable associations observed were as follows:\n",
            "- rCN1 and rCN2 showed a statistically significant positive association with TAN samples.\n",
            "- rCN5 had a notable negative association with B cells.\n",
            "- rCN6 had a positive association with AMACR-high luminal cells in G3 and G4 tumors.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/blise_head_and_neck.pdf\n",
            "Extracted Title: Single-cell spatial architectures associated with clinical outcome in head and neck squamous cell carcinoma\n",
            "Extracted for disease_name: The disease focus of the study is head and neck squamous cell carcinoma (HNSCC).\n",
            "Extracted for clinical_variable: HPV status, tobacco and alcohol use, treatment regimens, and survival outcomes were the clinical variables analyzed.\n",
            "Extracted for unique_neighborhoods: Clusters 1 and 2 primarily consisted of αSMA+cells, while Cluster 3 had the highest proportion of neoplastic tumor cell neighbors. Clusters 4, 5, 6, and 7 were predominantly composed of immune cells as neighbors, with varying proportions of different immune cell types.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods identified were clusters 1, 2, 3, 4, 5, 6, and 7. Cluster 1 was predominantly composed of αSMA+cells, cluster 3 had the highest proportion of neoplastic tumor cell neighbors, and clusters 4, 5, 6, and 7 consisted mostly of immune cells as neighbors. The association between these cellular neighborhoods and the clinical variable was as follows: cluster 1 had a positive association with αSMA+cells, cluster 3 had a positive association with neoplastic tumor cells, and clusters 4, 5, 6, and 7 had a positive association with immune cells.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/blise_pancreatic.pdf\n",
            "Extracted Title: 1 Machine learning links T cell function and spatial localization to neoadjuvant immunotherapy and clinical outcome in pancreatic cancer\n",
            "Extracted for disease_name: The disease focus of the study is pancreatic cancer.\n",
            "Extracted for clinical_variable: The clinical variables analyzed in the study include treatment cohorts, disease-free survival (DFS) groups, and tissue regions (T, IA, TAS, NAP sites).\n",
            "Extracted for unique_neighborhoods: Seven unique cellular neighborhoods were identified in the study. Each neighborhood had distinct compositions of cell types. One neighborhood, RCN1, consisted mainly of B cells. Another neighborhood, RCN5, contained the highest proportion of CD44+ T cells, including both CD8+ T cells and CD4+ T helper cells.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods identified were RCN1, RCN2, RCN3, RCN4, RCN5, RCN6, and RCN7. These neighborhoods showed different associations with clinical variables. For example, RCN5 had a positive association with CD44+ T cells, including both CD8+ T cells and CD4+ T helper lineages. The specific associations of the other cellular neighborhoods with clinical variables were not explicitly mentioned in the provided context.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/eng_breast.pdf\n",
            "Extracted Title: TITLE: Robust biomarker discovery through  multiplatform multiplex image analysis of Jennifer Eng1,2, Elmar Bucher2, Zhi Hu2, Melinda Sanders3,4, Bapsi Chakravarthy3, Paula\n",
            "Extracted for disease_name: The disease focus of the study is breast cancer, specifically exploring the breast tumor microenvironment and identifying biomarkers related to patient outcomes in breast cancer.\n",
            "Extracted for clinical_variable: Patient age, tumor size, and stage were the clinical variables analyzed in the study.\n",
            "Extracted for unique_neighborhoods: The unique cellular neighborhoods identified include tumor cell neighborhoods with T cells and macrophages, mixed neighborhoods containing vimentin+ fibroblasts, quiescent stroma tumor cell neighborhoods, vimentin+ fibroblast neighborhoods, and mixed fibroblast neighborhoods with vimentin-positive and negative fibroblasts. These neighborhoods consist of various cell types such as T cells, macrophages, vimentin+ fibroblasts, and tumor cells.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods identified were tumor cell neighborhoods with T cells and adjacent macrophage-rich tumor neighborhoods, transitioning into mixed neighborhoods and vimentin+ fibroblast neighborhoods. The clinical variable associations observed were as follows:\n",
            "1. Increased vimentin+ fibroblast neighborhoods around tumor cells were associated with shorter overall survival (OS) in TNBC cohorts.\n",
            "2. Increased vimentin+ fibroblast neighborhoods were associated with shorter OS and recurrence-free survival (RFS) in the multivariate model.\n",
            "3. Increased mixed fibroblast neighborhoods containing vimentin-positive and negative fibroblasts around tumor cells were associated with shorter OS in ER+ tumors.\n",
            "4. Directly clustering the neighborhood counts using Kmeans did not result in robust prediction of prognosis.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/hickey_intestine.pdf\n",
            "Extracted Title: 572 | Nature | Vol 619 | 20 July 2023 ArticleOrganization of the human intestine at\n",
            "Extracted for disease_name: The disease focus of the study is on intestinal diseases, specifically mentioning inflammatory bowel disease (IBD).\n",
            "Extracted for clinical_variable: The clinical variables analyzed include the expression of B3GAT1 and four CEACAM transcripts in different epithelial cell types from various regions of the intestine, cell-cell co-localization patterns in the large colon and small bowel, SEMA4D ligand expression in plasma cells, MET receptor gene expression in TA2 cells, and differences in pairwise cell-type colocalization patterns across tissue locations.\n",
            "Extracted for unique_neighborhoods: Twenty unique cellular neighborhoods were identified, each enriched with specific cell types. Some of the unique cellular neighborhoods include Neuroendocrine enriched, Paneth enriched, CD8+ T enriched IEL, Plasma cell enriched, and Adaptive immune enriched. Each neighborhood consists of a distinct combination of cell types that are enriched compared to the overall percentage of cell types in the samples.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods mentioned in the context are Microvasculature and CD8+ T cell Enriched IEL neighborhoods. The association with a clinical variable, in this case, a history of hypertension, was positive for Microvasculature neighborhoods (p value = 0.0065) and CD8+ T cell Enriched IEL neighborhoods (p value = 0.0017).\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/hickey_melanoma.pdf\n",
            "Extracted Title: T cell-mediated curation and restructuring of tumor tissue coordinates an effective immune response\n",
            "Extracted for disease_name: The disease focus of the study is melanoma.\n",
            "Extracted for clinical_variable: The clinical variables analyzed include gene expression measurements, survival data, age, and disease stage.\n",
            "Extracted for unique_neighborhoods: Seven major T cell-specific neighborhoods (TCNs) were identified in the study. These neighborhoods included:\n",
            "1. TCN A: Associated with tumor cells and vasculature.\n",
            "2. TCN B: Enriched with vasculature.\n",
            "3. TCN C: Enriched with dendritic cells (DCs) and CD4+ T cells.\n",
            "4. TCN D: Characterized by an inflamed phenotype of tumor cells.\n",
            "5. TCN E: Associated with other immune cells.\n",
            "6. TCN F: Indicative of active tumor killing.\n",
            "7. TCN G: Associated with other immune cells.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods mentioned in the context are Immune Infiltrate, Productive T cell & Tumor, Unproductive T cell & Tumor, CD8+T cell, CD8+PD1+T cell, and Tumor PDL1+MHCI+ combination. The kind of clinical variable association they had varied based on the analysis conducted. For example, the analysis included spatial context maps, instance segmentation, cell interaction analysis, random forest analysis, likelihood ratio test, and canonical correlation analysis to study different aspects of cellular neighborhoods and their relationships. The associations could be positive, negative, or neutral depending on the specific analysis and variables being studied.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/jin_lymphoma.pdf\n",
            "Extracted Title: reveals tumor immune contexture associatedwith clinical outcomes after CAR T cell therapy Jin Jin,1,2,3,10Li Lin,1,2,10Jiao Meng,1,4,10Lijun Jiang,1,2,10Man Zhang,5Yuekun Fang,1,2Wanying Liu,1,2\n",
            "Extracted for disease_name: The disease focus of the study is Diffuse Large B-Cell Lymphoma (DLBCL).\n",
            "Extracted for clinical_variable: Age, sex, tumor burden, tumor stage, \"double\" and \"triple\" hit status, extranodal invasion sites, prior hematopoietic stem cell transplant (HSCT), prior treatment lines, refractory or relapsed status, CAR T cell dose, cytokine release syndrome (CRS) grade, outcomes, and follow-up time were the clinical variables analyzed.\n",
            "Extracted for unique_neighborhoods: The unique cellular neighborhoods identified include the proximity between CD4+ T cells and macrophages, as well as myeloid-derived suppressor cells (MDSCs) in patients with progressive disease (PD) compared to those with complete remission (CR). Additionally, fibroblasts and vascular endothelial cells were found to be enriched near tumor cells in patients with poor outcomes after CAR T therapy.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods identified in the study were CN1 and CN2. CN1, characterized by an abundance of immune cells with CD8+ T cells as the majority, showed a positive association with longer remission duration after CAR T therapy. On the other hand, CN2, where tumor cells were predominant, exhibited a negative association with fast recurrence (within 3 months) post-CAR T therapy.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/karimi_brain.pdf\n",
            "Extracted Title: Nature | Vol 614 | 16 February 2023 | 555 ArticleSingle-cell spatial immune landscapes of\n",
            "Extracted for disease_name: The disease focus of the study is on primary brain tumors, specifically glioblastoma, and brain metastases.\n",
            "Extracted for clinical_variable: The clinical variables analyzed include IDH status, tumor grade, disease progression, treatment received, extent of resection, survival duration, primary cancer source, and clinical covariates for the balanced glioblastoma patient cohort.\n",
            "Extracted for unique_neighborhoods: CN1: M1-like MDM, neutrophils, M1-like MG  \n",
            "CN2: CD45+, CD45-  \n",
            "CN3: Undefined cells  \n",
            "CN4: Melanoma cells  \n",
            "CN5: Lung cells  \n",
            "CN6: Breast cells  \n",
            "CN7: M1-like MDM  \n",
            "CN8: M1-like MDM, M1-like MG  \n",
            "CN9: M1-like MDM, M1-like MG, CD45+\n",
            "Extracted for clinical_variable_association: CN1: Tumour boundary - No association  \n",
            "CN2: Pan-immune hotspot - No association  \n",
            "CN3: Astrocyte-enriched - No association  \n",
            "CN4: Astrocyte-enriched - No association  \n",
            "CN5: Undefined cells - Negative association with reduced time to local recurrence  \n",
            "CN6: Vascular niche - No association  \n",
            "CN7: Macrophage-enriched - No association  \n",
            "CN8: Tumour compartment - No association  \n",
            "CN9: Immune subset deficiencies - No association\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/lake_kidney.pdf\n",
            "Extracted Title: Nature | Vol 619 | 20 July 2023 | 585 ArticleAn atlas of healthy and injured cell states and\n",
            "Extracted for disease_name: The disease focus of the study is Chronic Kidney Disease (CKD) and Acute Kidney Injury (AKI).\n",
            "Extracted for clinical_variable: Age and sex were the clinical variables analyzed in the study.\n",
            "Extracted for unique_neighborhoods: The unique cellular neighborhoods identified include renal corpuscle neighborhoods consisting of podocytes (PODs), glomerular capillaries (EC-GC), mesangial cells, and parietal epithelial cells. Additionally, the juxtaglomerular apparatus cells such as renin-producing granular (REN) cells and macula densa cells, as well as endothelial cells of the afferent/efferent arterioles (EC-AEA), were also part of these neighborhoods. Vascular smooth muscle cell (VSMC) populations flanking the afferent/efferent arterioles were identified as well.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods mentioned in the context are adaptive epithelial (aPT) and adaptive thick ascending limb (aTAL). The association between these cellular neighborhoods and patients with diabetic nephropathy compared to living donors was positive.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "Processing PDF: /content/data/lemaitre_hepatocellular.pdf\n",
            "Extracted Title: Nature Cancer | Volume 5 | October 2024 | 1534–1556 1534 immune evasion in hepatocellular carcinoma\n",
            "Extracted for disease_name: The disease focus of the study is hepatocellular carcinoma (HCC).\n",
            "Extracted for clinical_variable: The clinical variables analyzed include the liver/body weight ratio, number of surface nodules, tumor volume, and the number of CD4+ and CD8+ T cells.\n",
            "Extracted for unique_neighborhoods: Nine distinct neighborhoods were identified across all the HCC tissues. Among these, four were tumor-cell-dominant, four were immune-cell-dominant, and one was of mixed cell population. The tumor-cell-dominant neighborhoods included stem-like tumor cells, mesenchymal tumor cells, and tumor cells with prosurvival phenotype. The immune-cell-dominant neighborhoods included M2-like macrophages, antitumor immune cells, innate immune cells, and T cell immune cells.\n",
            "Extracted for clinical_variable_association: The unique cellular neighborhoods identified were the M2-like macrophage immune CN, vascular inflammatory tumor CN, innate immune CN, and T cell immune CN. The association between these neighborhoods and the clinical variable of residual versus primary HCC was as follows: Residual HCC showed a higher prevalence of the M2-like macrophage immune CN and the vascular inflammatory tumor CN, while the innate immune CN and T cell immune CN were less frequent in residual HCC. This indicates a positive association between the M2-like macrophage immune CN and vascular inflammatory tumor CN with residual HCC, and a negative association with the innate immune CN and T cell immune CN.\n",
            "Row added to CSV: ./result_data2.csv\n",
            "CSV file updated with data from 10 PDFs!\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "import pandas as pd\n",
        "from PyPDF2 import PdfReader\n",
        "from llama_index.core import (\n",
        "    VectorStoreIndex,\n",
        "    Document,\n",
        ")\n",
        "\n",
        "def update_csv_from_extracted_data(csv_path, extracted_data):\n",
        "    \"\"\"\n",
        "    Updates a CSV file with extracted data from the query responses.\n",
        "    \"\"\"\n",
        "    columns_order = [\"paper_name\", \"disease_name\", \"clinical_variable\", \"unique_neighborhoods\", \"clinical_variable_association\"]\n",
        "    extracted_data = {col: extracted_data.get(col, \"\") for col in columns_order}\n",
        "\n",
        "    if os.path.exists(csv_path):\n",
        "        df = pd.read_csv(csv_path)\n",
        "    else:\n",
        "        df = pd.DataFrame(columns=extracted_data.keys())\n",
        "\n",
        "\n",
        "    new_row_df = pd.DataFrame([extracted_data])\n",
        "\n",
        "\n",
        "    df = pd.concat([df, new_row_df], ignore_index=True)\n",
        "\n",
        "    df.to_csv(csv_path, index=False)\n",
        "    print(f\"Row added to CSV: {csv_path}\")\n",
        "\n",
        "def extract_title_from_pdf(pdf_path):\n",
        "    \"\"\"\n",
        "    Attempts to extract the title from the first page of the PDF, accounting for multi-line titles.\n",
        "    \"\"\"\n",
        "    reader = PdfReader(pdf_path)\n",
        "    first_page = reader.pages[0].extract_text()\n",
        "\n",
        "\n",
        "    lines = first_page.splitlines()\n",
        "\n",
        "\n",
        "    filtered_lines = [line.strip() for line in lines if len(line.split()) > 4]\n",
        "\n",
        "\n",
        "    if len(filtered_lines) >= 2:\n",
        "        title = f\"{filtered_lines[0]} {filtered_lines[1]}\".strip()\n",
        "    elif filtered_lines:\n",
        "        title = filtered_lines[0].strip()\n",
        "    else:\n",
        "        title = \"Unknown Title\"  # Default if no title is found\n",
        "\n",
        "\n",
        "    non_title_keywords = [\"ARTICLE\", \"OPEN\", \"CHECK FOR UPDATES\"]\n",
        "    for keyword in non_title_keywords:\n",
        "        title = title.replace(keyword, \"\").strip()\n",
        "\n",
        "    return title\n",
        "\n",
        "queries = {\n",
        "    \"disease_name\": \"What is the disease focus of the study?\",\n",
        "    \"clinical_variable\": \"What are the clinical variables analyzed?\",\n",
        "    \"unique_neighborhoods\": \"list the unique cellular neighborhoods identified with details of the type of cells\",\n",
        "    \"clinical_variable_association\": \"list the unique cellular neighborhoods and the kind of clinical variable association they had. whethe the association was positive negative or no association at all\",\n",
        "}\n",
        "\n",
        "def query_fields(queries, query_engine):\n",
        "    \"\"\"\n",
        "    Queries the engine for each field and collects the responses.\n",
        "    \"\"\"\n",
        "    extracted_data = {}\n",
        "    for field, query in queries.items():\n",
        "        response = query_engine.query(query)\n",
        "        extracted_data[field] = str(response).strip()  # Clean the response text\n",
        "        print(f\"Extracted for {field}: {response}\")\n",
        "    return extracted_data\n",
        "\n",
        "\n",
        "csv_path = \"./result_data2.csv\"\n",
        "\n",
        "\n",
        "pdf_paths = [\n",
        "  '/content/data/ak_prostate.pdf',\n",
        "  '/content/data/blise_head_and_neck.pdf',\n",
        "  '/content/data/blise_pancreatic.pdf',\n",
        "  '/content/data/eng_breast.pdf',\n",
        "  '/content/data/hickey_intestine.pdf',\n",
        "  '/content/data/hickey_melanoma.pdf',\n",
        "  '/content/data/jin_lymphoma.pdf',\n",
        "  '/content/data/karimi_brain.pdf',\n",
        "  '/content/data/lake_kidney.pdf',\n",
        "  '/content/data/lemaitre_hepatocellular.pdf'\n",
        "]\n",
        "\n",
        "\n",
        "\n",
        "for pdf_path in pdf_paths:\n",
        "    print(f\"Processing PDF: {pdf_path}\")\n",
        "\n",
        "\n",
        "    title = extract_title_from_pdf(pdf_path)\n",
        "    print(f\"Extracted Title: {title}\")\n",
        "\n",
        "\n",
        "    reader = PdfReader(pdf_path)\n",
        "    pdf_content = \"\"\n",
        "    for page in reader.pages:\n",
        "        pdf_content += page.extract_text()\n",
        "\n",
        "\n",
        "    document = Document(text=pdf_content, metadata={\"file_name\": os.path.basename(pdf_path)})\n",
        "\n",
        "\n",
        "    index = VectorStoreIndex.from_documents([document])\n",
        "\n",
        "\n",
        "    query_engine = index.as_query_engine()\n",
        "\n",
        "\n",
        "    extracted_data = query_fields(queries, query_engine)\n",
        "\n",
        "\n",
        "    extracted_data[\"paper_name\"] = title\n",
        "\n",
        "\n",
        "    update_csv_from_extracted_data(csv_path, extracted_data)\n",
        "\n",
        "print(f\"CSV file updated with data from {len(pdf_paths)} PDFs!\")\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 39,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xfRAh12dlyuD",
        "outputId": "f5b029cd-cb75-49cf-e497-cf9059705e87"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Processing PDF: /content/data/blise_pancreatic.pdf\n",
            "Extracted Title: Machine learning links T cell function and spatial localization to neoadjuvant immunotherapy and clinical outcome in pancreatic cancer  Katie E. Blise , Shamilene Sivagnanam\n",
            "Extracted for disease_name: Pancreatic Cancer\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: neoplastic epithelial cell, B cell\n",
            "neoplastic epithelial cell, T cell\n",
            "neoplastic epithelial cell, myeloid cell\n",
            "neoplastic epithelial cell, mesenchymal fibroblast-like cell\n",
            "Extracted for clinical_variable_association: positive,positive,none,negative,negative,negative,negative\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/hickey_melanoma.pdf\n",
            "Extracted Title: T cell-mediated curation and restructuring of tumor tissue coordinates an effective immune response\n",
            "Extracted for disease_name: Melanoma\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: Tumor cell, vasculature, DCs, CD4+ T cells, endothelial cells, macrophages, PDL1+ MHCI+ tumor cells, lymphatic cells, Ki67+ tumor cells, marker negative cells.\n",
            "Extracted for clinical_variable_association: positive, negative, none\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/blise_head_and_neck.pdf\n",
            "Extracted Title: Single-cell spatial architectures associated with clinical outcome in head and neck squamous cell carcinoma\n",
            "Extracted for disease_name: Head and neck squamous cell carcinoma\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: neoplastic tumor cell, stromal cell (mesenchymal), B cell, CD45+ immune cell, CD3+ immune cell, CD8+ immune cell, CD20+ immune cell, CD66B+ immune cell, Ki-67+ cell\n",
            "Extracted for clinical_variable_association: positive,positive,positive,negative,negative,negative,negative\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/lemaitre_hepatocellular.pdf\n",
            "Extracted Title: Spatial analysis reveals targetable macrophage-mediated mechanisms of immune evasion in hepatocellular carcinoma minimal residual disease\n",
            "Extracted for disease_name: Hepatocellular carcinoma\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: Tumor cell, Fibroinflammatory immune, Innate immune, M2-macrophage Immune, Other, T cell Immune, Pauci-immune tumor, Vascular inflammatory Tumor.\n",
            "Extracted for clinical_variable_association: positive,positive,positive,positive,positive,positive,positive,negative,negative\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/eng_breast.pdf\n",
            "Extracted Title: T :\n",
            "Extracted for disease_name: Breast cancer\n",
            "Extracted for clinical_variable: Survival\n",
            "Extracted for unique_neighborhoods: Tumor cell, T cell, macrophage, vimentin+ fibroblast, CD4 T cell, quiescent stroma, luminal ER+ tumor, ER+ HER2+ tumor, fibronectin+ fibroblast.\n",
            "Extracted for clinical_variable_association: positive, negative, none\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/jin_lymphoma.pdf\n",
            "Extracted Title: High-multiplex single-cell imaging analysis reveals tumor immune contexture associated with clinical outcomes after CAR T cell therapy\n",
            "Extracted for disease_name: Lymphoma\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: B cell, cytotoxic CD8+ T cell, CD4+ T cell, NK cell, macrophage, granulocyte, myeloid-derived suppressor cell, fibroblast, vascular endothelium.\n",
            "Extracted for clinical_variable_association: positive,none\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/karimi_brain.pdf\n",
            "Extracted Title: Single-cell spatial immune landscapes of primary and metastatic brain tumours\n",
            "Extracted for disease_name: Glioblastoma\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: CN1: Astrocytes, T cells, M1-like MDM  \n",
            "CN2: Neutrophils, NK cells, Mast cells  \n",
            "CN3: Dendritic cells, Tc cells, TH cells  \n",
            "CN4: Treg cells, Other T cells, M1-like MG  \n",
            "CN5: M2-like MDM, M2-like MG, Cl Mo  \n",
            "CN6: Non-Cl Mo, Int Mo, Undefined  \n",
            "CN7: Total T cells, Total MDM, Total MG  \n",
            "CN8: Total macs, Total Mo, Glioma  \n",
            "CN9: BrM-cores, BrM-margins, IDH wt\n",
            "Extracted for clinical_variable_association: positive,none,none,none,none,none,none,none,positive\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/lake_kidney.pdf\n",
            "Extracted Title: An atlas of healthy and injured cell states and niches in the human kidney\n",
            "Extracted for disease_name: Kidney Disease\n",
            "Extracted for clinical_variable: histology type\n",
            "Extracted for unique_neighborhoods: Podocytes, glomerular capillaries, mesangial cells, parietal epithelial cells, renin-producing granular cells, macula densa cells, endothelial cells of the afferent/efferent arterioles, vascular smooth muscle cells.\n",
            "Extracted for clinical_variable_association: positive, negative, none\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/hickey_intestine.pdf\n",
            "Extracted Title: Organization of the human intestine at single-cell resolution\n",
            "Extracted for disease_name: Intestinal diseases\n",
            "Extracted for clinical_variable: histology type\n",
            "Extracted for unique_neighborhoods: Paneth cell, CD4/CD8 ICCs, Immune, Stromal, Epithelial, Neuroendocrine, MUC1+ enterocyte, CD57+ enterocyte, CD8+ T, CD7+ immune, CD66+ enterocyte, Cycling TA, TA, Enterocyte, Goblet, Plasma, Smooth muscle, Nerve, NK, DC, Stroma, CD4+ T cell, M2 macrophage, Lymphatic, ICC, Endothelial, M1 macrophage, Neutrophil, Glandular epithelial, CD66+ mature epithelial, Secretory epithelial, Transit-amplifying zone, CD8+ T enriched IEL, Mature epithelial, Inner follicle, Innate immune enriched, Plasma cell enriched, Adaptive immune enriched, Outer follicle, Stroma and innate immune, Microvasculature, Macrovasculature, Innervated smooth muscle, Innervated stroma, Smooth muscle and innate immune.\n",
            "Extracted for clinical_variable_association: positive,none,negative\n",
            "CSV updated: ./result_data1.csv\n",
            "Processing PDF: /content/data/ak_prostate.pdf\n",
            "Extracted Title: Multiplex imaging of localized prostate tumors reveals altered spatial organization of AR-positive cells in the microenvironment\n",
            "Extracted for disease_name: Prostate cancer\n",
            "Extracted for clinical_variable: response to immunotherapy\n",
            "Extracted for unique_neighborhoods: AR+ non-immune stromal cells interacting with neuroendocrine, mesenchymal/endothelial, immune, and smooth muscle and M2 macrophages; luminal epithelial, EMT/infiltrating VIM+, AMACR-high luminal cells interacting with APC immune cells, basal and neuroendocrine, and basal, neuroendocrine, and luminal cells.\n",
            "Extracted for clinical_variable_association: positive,positive,none,none,none,none,none,positive,negative,negative\n",
            "CSV updated: ./result_data1.csv\n",
            "All documents processed and results saved to CSV.\n"
          ]
        }
      ],
      "source": [
        "import os\n",
        "import pandas as pd\n",
        "import fitz\n",
        "from llama_index.core import VectorStoreIndex, SimpleDirectoryReader, Document\n",
        "\n",
        "\n",
        "def update_csv(csv_path, data_row):\n",
        "    \"\"\"\n",
        "    Updates the CSV file with extracted data row by row.\n",
        "    Ensures correct formatting and missing field handling.\n",
        "    \"\"\"\n",
        "    columns = [\"paper_name\", \"disease_name\", \"clinical_variable\", \"unique_neighborhoods\",\n",
        "               \"clinical_variable_association\"]\n",
        "\n",
        "\n",
        "    if os.path.exists(csv_path):\n",
        "        df = pd.read_csv(csv_path)\n",
        "    else:\n",
        "        df = pd.DataFrame(columns=columns)\n",
        "\n",
        "\n",
        "    data_row = {col: data_row.get(col, \"Not Available\") for col in columns}\n",
        "\n",
        "\n",
        "    df = pd.concat([df, pd.DataFrame([data_row])], ignore_index=True)\n",
        "    df.to_csv(csv_path, index=False)\n",
        "    print(f\"CSV updated: {csv_path}\")\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "def extract_title_with_font_size(pdf_path):\n",
        "\n",
        "    doc = fitz.open(pdf_path)\n",
        "    first_page = doc[0]\n",
        "\n",
        "    text_instances = []\n",
        "    blocks = first_page.get_text(\"dict\")[\"blocks\"]\n",
        "    for block in blocks:\n",
        "        if \"lines\" in block:  # Ensure \"lines\" exists\n",
        "            for line in block[\"lines\"]:\n",
        "                for span in line[\"spans\"]:\n",
        "                    text_instances.append({\n",
        "                        \"text\": span[\"text\"].strip(),\n",
        "                        \"font_size\": span[\"size\"],\n",
        "                        \"y_position\": span[\"bbox\"][1]\n",
        "                    })\n",
        "\n",
        "    if text_instances:\n",
        "        text_instances.sort(key=lambda x: (-x[\"font_size\"], x[\"y_position\"]))\n",
        "\n",
        "        largest_font_size = text_instances[0][\"font_size\"]\n",
        "\n",
        "\n",
        "        largest_font_spans = [t for t in text_instances if t[\"font_size\"] == largest_font_size]\n",
        "\n",
        "        largest_font_spans.sort(key=lambda x: x[\"y_position\"])\n",
        "        top_three_spans = largest_font_spans[:5]\n",
        "\n",
        "        title = \" \".join([t[\"text\"] for t in top_three_spans])\n",
        "        return title.strip()\n",
        "\n",
        "    return \"Unknown Title\"\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "def query_fields(queries, query_engine):\n",
        "    \"\"\"\n",
        "    Executes the queries using the query engine and extracts the data.\n",
        "    \"\"\"\n",
        "    extracted_data = {}\n",
        "    for field, query in queries.items():\n",
        "        try:\n",
        "            response = query_engine.query(query)\n",
        "            extracted_data[field] = str(response).strip()\n",
        "            print(f\"Extracted for {field}: {response}\")\n",
        "        except Exception as e:\n",
        "            print(f\"Error extracting {field}: {e}\")\n",
        "            extracted_data[field] = \"Not Available\"\n",
        "    return extracted_data\n",
        "\n",
        "\n",
        "def process_pdf(pdf_path, csv_path, queries):\n",
        "    \"\"\"\n",
        "    Processes a single PDF file to extract relevant data and save it to the CSV.\n",
        "    \"\"\"\n",
        "    print(f\"Processing PDF: {pdf_path}\")\n",
        "\n",
        "\n",
        "    title = extract_title_with_font_size(pdf_path)\n",
        "    print(f\"Extracted Title: {title}\")\n",
        "\n",
        "\n",
        "    doc = fitz.open(pdf_path)\n",
        "    pdf_content = \"\"\n",
        "    for page in doc:\n",
        "        pdf_content += page.get_text()\n",
        "\n",
        "\n",
        "    document = Document(text=pdf_content, metadata={\"file_name\": os.path.basename(pdf_path)})\n",
        "    index = VectorStoreIndex.from_documents([document])\n",
        "    query_engine = index.as_query_engine()\n",
        "\n",
        "\n",
        "    extracted_data = query_fields(queries, query_engine)\n",
        "\n",
        "\n",
        "    extracted_data[\"paper_name\"] = title\n",
        "\n",
        "\n",
        "    update_csv(csv_path, extracted_data)\n",
        "\n",
        "\n",
        "queries = {\n",
        "    \"disease_name\": \"Disease Name\",\n",
        "    \"clinical_variable\": \"what factors does the paper focus on? Choose one of these option: survival,response to immunotherapym,histology type,None\",\n",
        "    \"unique_neighborhoods\": \"Just List the unique cellular neighborhoods identified with details of the type of cells.format example tumor cell,bcell\",\n",
        "    \"clinical_variable_association\" :\"what associations did the cellular neighborhoods have to clinical variable. format example positive,negative,none maintain the order with respect to the unique neighborhoods\",\n",
        "}\n",
        "\n",
        "csv_path = \"./result_data1.csv\"\n",
        "\n",
        "data_folder = \"/content/data\"\n",
        "pdf_files = [os.path.join(data_folder, f) for f in os.listdir(data_folder) if f.endswith(\".pdf\")]\n",
        "\n",
        "for pdf_path in pdf_files:\n",
        "    process_pdf(pdf_path, csv_path, queries)\n",
        "\n",
        "print(\"All documents processed and results saved to CSV.\")\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ZfCmbmzMlHtu",
        "outputId": "3df90555-da0b-4abb-863e-62f5d0840f9f"
      },
      "outputs": [
        {
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "Fuzzy Matching Accuracy: 33.33%\n",
            "Total Matches: 5\n",
            "Total Cells: 15\n",
            "Mismatch details saved to 'mismatch_details_fuzzy.csv'\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "from fuzzywuzzy import fuzz\n",
        "\n",
        "expected_results_path = \"/content/labelpapers.csv\"\n",
        "generated_results_path = \"/content/result_data.csv\"\n",
        "\n",
        "expected_results = pd.read_csv(expected_results_path)\n",
        "generated_results = pd.read_csv(generated_results_path)\n",
        "\n",
        "\n",
        "if expected_results.shape != generated_results.shape:\n",
        "    print(\"Mismatch in shape between the two files.\")\n",
        "    print(f\"Expected shape: {expected_results.shape}\")\n",
        "    print(f\"Generated shape: {generated_results.shape}\")\n",
        "else:\n",
        "\n",
        "    total_cells = expected_results.size\n",
        "    matches = 0\n",
        "    threshold = 80\n",
        "\n",
        "\n",
        "    for col in expected_results.columns:\n",
        "        for i in range(len(expected_results)):\n",
        "            expected = str(expected_results.at[i, col])\n",
        "            generated = str(generated_results.at[i, col])\n",
        "\n",
        "\n",
        "            similarity = fuzz.ratio(expected, generated)\n",
        "\n",
        "            if similarity >= threshold:\n",
        "                matches += 1\n",
        "\n",
        "\n",
        "    accuracy = (matches / total_cells) * 100\n",
        "\n",
        "\n",
        "    print(f\"Fuzzy Matching Accuracy: {accuracy:.2f}%\")\n",
        "    print(f\"Total Matches: {matches}\")\n",
        "    print(f\"Total Cells: {total_cells}\")\n",
        "\n",
        "\n",
        "    mismatches = []\n",
        "    for col in expected_results.columns:\n",
        "        for i in range(len(expected_results)):\n",
        "            expected = str(expected_results.at[i, col])\n",
        "            generated = str(generated_results.at[i, col])\n",
        "            similarity = fuzz.ratio(expected, generated)\n",
        "            if similarity < threshold:\n",
        "                mismatches.append({\n",
        "                    \"Row\": i + 1,\n",
        "                    \"Column\": col,\n",
        "                    \"Expected\": expected,\n",
        "                    \"Generated\": generated,\n",
        "                    \"Similarity\": similarity\n",
        "                })\n",
        "\n",
        "\n",
        "    if mismatches:\n",
        "        mismatches_df = pd.DataFrame(mismatches)\n",
        "        mismatches_df.to_csv(\"mismatch_details_fuzzy.csv\", index=False)\n",
        "        print(\"Mismatch details saved to 'mismatch_details_fuzzy.csv'\")\n",
        "    else:\n",
        "        print(\"All entries are sufficiently similar!\")\n"
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}